--- lib/ext2fs/llseek.c
+++ lib/ext2fs/llseek.c
@@ -59,7 +59,7 @@
 #define __NR__llseek            140
 #endif
 
-#ifndef __i386__
+#if !defined __i386__ && !defined __UCLIBC__
 static int _llseek (unsigned int, unsigned long,
 		   unsigned long, ext2_loff_t *, unsigned int);
 
@@ -73,7 +73,7 @@
 	ext2_loff_t result;
 	int retval;
 
-#ifndef __i386__
+#if !defined __i386__ && !defined __UCLIBC__
 	retval = _llseek(fd, ((unsigned long long) offset) >> 32,
 #else
 	retval = syscall(__NR__llseek, fd, (unsigned long long) (offset >> 32),
--- lib/blkid/llseek.c
+++ lib/blkid/llseek.c
@@ -58,7 +58,7 @@
 #define __NR__llseek            140
 #endif
 
-#ifndef __i386__
+#if !defined __i386__ && !defined __UCLIBC__
 static int _llseek(unsigned int, unsigned long, unsigned long,
 		   blkid_loff_t *, unsigned int);
 
@@ -72,7 +72,7 @@
 	blkid_loff_t result;
 	int retval;
 
-#ifndef __i386__
+#if !defined __i386__ && !defined __UCLIBC__
 	retval = _llseek(fd, ((unsigned long long) offset) >> 32,
 			 ((unsigned long long)offset) & 0xffffffff,
 			 &result, origin);
