--- signal.c
+++ signal.c
@@ -626,7 +626,7 @@
 #endif
 	tprints(", ");
 	printflags(sigact_flags, sa.sa_flags, "SA_???");
-#ifdef SA_RESTORER
+#if defined(SA_RESTORER) && !defined(MIPS)
 	if (sa.sa_flags & SA_RESTORER)
 		tprintf(", %p", sa.sa_restorer);
 #endif
@@ -1219,7 +1219,7 @@
 	tprints(", ");
 
 	printflags(sigact_flags, sa.sa_flags, "SA_???");
-#ifdef SA_RESTORER
+#if defined(SA_RESTORER) && !defined(MIPS)
 	if (sa.sa_flags & SA_RESTORER)
 		tprintf(", %p", sa.sa_restorer);
 #endif
