--- etc/hotplug/storage.orig	2008-02-17 14:30:19.000000000 +0100
+++ etc/hotplug/storage	2008-02-17 14:32:09.000000000 +0100
@@ -15,6 +15,9 @@
 MSTART=/sbin/start_mediasrv
 MSTOP=/sbin/stop_mediasrv
 
+# Do we have a Ftp server (ftpd) or was ist stripped from the firmware?
+_ftpd=$(basename $(which ftpd) 2> /dev/null)
+
 # Ask the system's configuration for the status of the readonly flag:
 # (Translates flag's yes/no to ro/rw)
 #
@@ -220,13 +223,15 @@
 	fi
 	# ftp server control
 	ENABLED=`echo usbhost.ftp_server_enabled|ar7cfgctl -s`
-	if [ "$ENABLED" != "no" ] ; then
-		if [ "`pidof ftpd`" = "" ] ; then
-			USERS_ENABLED=`echo usbhost.users_enabled|ar7cfgctl -s`
-			if [ "$USERS_ENABLED" = "yes" ] ; then
-				ftpd -D -q -t 120 -m 15 -U -h "$CONFIG_PRODUKT_NAME" &
-			else
-				ftpd -D -q -t 120 -m 15 -h "$CONFIG_PRODUKT_NAME" &
+	if [ "$_ftpd" ] ; then
+		if [ "$ENABLED" != "no" ] ; then
+			if [ "`pidof ftpd`" = "" ] ; then
+				USERS_ENABLED=`echo usbhost.users_enabled|ar7cfgctl -s`
+				if [ "$USERS_ENABLED" = "yes" ] ; then
+					ftpd -D -q -t 120 -m 15 -U -h "$CONFIG_PRODUKT_NAME" &
+				else
+					ftpd -D -q -t 120 -m 15 -h "$CONFIG_PRODUKT_NAME" &
+				fi
 			fi
 		fi
 	else
