--- module/setup.py
+++ module/setup.py
@@ -109,7 +109,7 @@
         print ""
 
         if len(avail) < 5:
-            print _("Featues missing: ")
+            print _("Features missing: ")
             print ""
 
             if not self.check_module("Crypto"):
--- pyLoadCore.py
+++ pyLoadCore.py
@@ -172,7 +172,7 @@
         print "  --configdir=<dir>", " " * 6, "Run with <dir> as config directory"
         print "  -p, --pidfile=<file>", " " * 3, "Set pidfile to <file>"
         print "  --changedir", " " * 12, "Change config dir permanently"
-        print "  --daemon", " " * 15, "Daemonmize after start"
+        print "  --daemon", " " * 15, "Daemonize after start"
         print "  --no-remote", " " * 12, "Disable remote access (saves RAM)"
         print "  --status", " " * 15, "Display pid if running or False"
         print "  --clean", " " * 16, "Remove .pyc/.pyo files"
