--- configure
+++ configure
@@ -13837,7 +13837,7 @@
 #else
 typedef locale_t xsltLocale;
 #endif
-#if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ <= 2
+#if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ <= 2 && !defined(__UCLIBC__)
 #define newlocale __newlocale
 #define freelocale __freelocale
 #define strxfrm_l __strxfrm_l
--- libxslt/xsltlocale.c
+++ libxslt/xsltlocale.c
@@ -19,7 +19,7 @@
 #include "xsltlocale.h"
 #include "xsltutils.h"
 
-#if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ <= 2
+#if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ <= 2 && !defined(__UCLIBC__)
 #define newlocale __newlocale
 #define freelocale __freelocale
 #define strxfrm_l __strxfrm_l
