--- include/bitset
+++ include/bitset
@@ -210,7 +210,7 @@
 			return *this;
 		}
 		bitset<N>& reset(){
-			for(size_t i = 0; i <= num_bytes; ++i){
+			for(size_t i = 0; i < num_bytes; ++i){
 				data[i] = 0;
 			}
 			return *this;
@@ -226,7 +226,7 @@
 		}
 
 		bitset<N>& flip(){
-			for(size_t i = 0; i <= num_bytes; ++i){
+			for(size_t i = 0; i < num_bytes; ++i){
 				data[i] =  ~data[i];
 			}
 			return *this;
