--- usr/www/all/html/de/home/home.js.orig	2009-03-27 15:20:16.000000000 +0100
+++ usr/www/all/html/de/home/home.js	2009-04-01 01:18:45.000000000 +0200
@@ -754,15 +754,13 @@
 var nPartTotal = Number("<? query ctlusb:settings/storage-part-total ?>");
 if (isNaN(nPartTotal)) nPartTotal = 0;
 var unplugOk = parseInt("<? query ctlusb:settings/unplug-ok ?>", 10) == 1;
-if (nPartTotal > nStor) {
+if (nPartTotal > 0) {
 var strStor = "";
 if (unplugOk) {
 strStor = "{?de.home.home.js:912?}";
 }
 else {
-strStor = "{?de.home.home.js:5308?} ";
-strStor += "<a href=\"javascript:jslGoTo('home', 'syslog');\">{?de.home.home.js:3744?}</a>";
-strStor = strStor.fontcolor("#cc0000");
+strStor = "Es wurden " + nStor + " von " + nPartTotal + " Partitionen eingebunden.";
 }
 strArr.push(strStor);
 }
