--- Makefile.mipsel.orig	2007-04-17 15:41:20.000000000 +0200
+++ Makefile.mipsel	2007-07-01 03:12:46.000000000 +0200
@@ -1,19 +1,15 @@
-CROSS=mipsel-linux-
-PJPATH=./pjproject
+CROSS:=mipsel-linux-
 
-CC=$(CROSS)gcc
-STRIP=$(CROSS)strip
+CC:=$(CROSS)gcc
+STRIP:=$(CROSS)strip
 
-include $(PJPATH)/build.mak
+CFLAGS:=$(CFLAGS)
+CPPFLAGS:=${CFLAGS} -w 
+LDFLAGS:=$(LDFLAGS)
+LDLIBS:=-Wl,-Bstatic $(LDLIBS) \
+	    -Wl,-Bdynamic -lpthread -lcapi20
 
-CC = $(APP_CC)
-LDFLAGS = $(APP_LDFLAGS)
-LDLIBS = -Wl,-Bstatic $(APP_LDLIBS) \
-	-static-libgcc -Wl,-Bdynamic -lpthread -lcapi20
-
-CFLAGS = $(APP_CFLAGS)
-CPPFLAGS = ${CFLAGS} -w 
-DESTDIR=
+DESTDIR:=
 
 HEADERS=dtmfbox.h \
 	config.h \
