--- linux-2.6.28/drivers/net/avm_cpmac/Makefile.26
+++ linux-2.6.28/drivers/net/avm_cpmac/Makefile.26
@@ -99,7 +99,7 @@
 # Create source files on demand
 EXTRACT_SOURCE: $(SOURCE_FILES:%=$(src)/%)
 
-extra-y	+= cpmacconfig
+#extra-y	+= cpmacconfig
 
 cpmacconfig-objs	:= cpmacconfig.o
 
--- linux-2.6.28/drivers/net/ifxmips_switch_api/Makefile
+++ linux-2.6.28/drivers/net/ifxmips_switch_api/Makefile
@@ -9,7 +9,7 @@
 drv_switch_api-objs	:= ifx_ethsw_linux.o
 drv_switch_api-objs	+= ifx_ethsw_init.o
 
-extra-y			+= switch_utility
+#extra-y			+= switch_utility
 ##################### AR9 ######################
 ifeq ($(CONFIG_AR9),y)
   EXTRA_CFLAGS += -DAR9 
