--- Makefile.am
+++ Makefile.am
@@ -45,7 +45,7 @@
 usrlib_LTLIBRARIES = libpopt.la
 
 libpopt_la_SOURCES = popt.c poptparse.c poptconfig.c popthelp.c poptint.c
-libpopt_la_LDFLAGS = -no-undefined @LTLIBINTL@ @LTLIBICONV@
+libpopt_la_LDFLAGS = -no-undefined
 
 pkgconfigdir = $(prefix)/lib/pkgconfig
 pkgconfig_DATA = popt.pc
--- Makefile.in
+++ Makefile.in
@@ -230,15 +230,15 @@
 INTL_MACOSX_LIBS = @INTL_MACOSX_LIBS@
 LD = @LD@
 LDFLAGS = @LDFLAGS@
-LIBICONV = @LIBICONV@
-LIBINTL = @LIBINTL@
+LIBICONV =
+LIBINTL =
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
 LIBTOOL = @LIBTOOL@
 LIPO = @LIPO@
 LN_S = @LN_S@
-LTLIBICONV = @LTLIBICONV@
-LTLIBINTL = @LTLIBINTL@
+LTLIBICONV =
+LTLIBINTL =
 LTLIBOBJS = @LTLIBOBJS@
 LT_AGE = @LT_AGE@
 LT_CURRENT = @LT_CURRENT@
@@ -368,7 +368,7 @@
 usrlibdir = $(libdir)
 usrlib_LTLIBRARIES = libpopt.la
 libpopt_la_SOURCES = popt.c poptparse.c poptconfig.c popthelp.c poptint.c
-libpopt_la_LDFLAGS = -no-undefined @LTLIBINTL@ @LTLIBICONV@ \
+libpopt_la_LDFLAGS = -no-undefined \
 	$(am__append_1)
 pkgconfigdir = $(prefix)/lib/pkgconfig
 pkgconfig_DATA = popt.pc
