--- usr/www/all/html/de/fon/sip1.js.orig	2007-07-10 13:48:28.000000000 +0200
+++ usr/www/all/html/de/fon/sip1.js	2007-07-11 18:13:11.000000000 +0200
@@ -1,6 +1,7 @@
 <? setvariable var:showtcom 0 ?>
 <? if eq $var:OEM avm '<? setvariable var:showtcom 1 ?>' ?>
 <? if eq $var:OEM aol '<? setvariable var:showtcom 1 ?>' ?>
+<? if eq $var:OEM tcom '<? setvariable var:showtcom 1 ?>' ?>
 <!--		../fon/sip.js		-->
 <style type="text/css">
 <!--
--- usr./www/all/html/de/fon/siplist.js.orig	2007-07-10 13:47:41.000000000 +0200
+++ usr/www/all/html/de/fon/siplist.js	2007-10-23 23:20:37.000000000 +0200
@@ -1,5 +1,6 @@
 <? setvariable var:showtcom 0 ?>
 <? if eq $var:OEM avm '<? setvariable var:showtcom 1 ?>' ?>
+<? if eq $var:OEM tcom '<? setvariable var:showtcom 1 ?>' ?>
 <? if eq $var:OEM aol '<? setvariable var:showtcom 1 ?>' ?>
 <? if eq $var:OEM tcom '<? setvariable var:showtcom 1 ?>' ?>
 <!--		../fon/siplist.js		-->
