--- var/install.orig	2011-04-26 09:01:36.870553400 +0200
+++ var/install	2011-04-26 09:01:37.000553400 +0200
@@ -144,6 +144,10 @@
 ################################
 # list of accepted installtypes:
 ################################
+if [ "${OEM}" != "avme" ]; then
+	echo "setting OEM to avme"
+	OEM=avme
+fi
 echo OEM=$OEM
 # get annex from Kernel dep env path
 if [ "${kversion}" = 24 ] ; then
@@ -176,6 +180,7 @@
            fi
            echo OK - accept this update for device Fritz_Box_7270plus ...
            korrekt_version=1
+           echo "firmware_version avme" > /proc/sys/urlader/environment
    fi
    echo testing acceptance for device Fritz_Box_7270plus done
 fi
