--- var/install
+++ var/install
@@ -79,6 +79,10 @@
 ################################
 # list of accepted installtypes:
 ################################
+if [ "${OEM}" != "avme" ]; then
+	echo "setting OEM to avme"
+	OEM=avme
+fi
 echo OEM=$OEM
 # get annex from Kernel dep env path
 if [ "${kversion}" = 24 ] ; then
@@ -111,6 +115,7 @@
            fi
            echo OK - accept this update for device Fritz_Box_7270plus ...
            korrekt_version=1
+           echo "firmware_version avme" > /proc/sys/urlader/environment
    fi
    echo testing acceptance for device Fritz_Box_7270plus done
 fi
