diff -ua linux-2.6.32.orig/drivers/net/avm_cpmac/cpphy_entry.c linux-2.6.32/drivers/net/avm_cpmac/cpphy_entry.c
--- linux-2.6.32/drivers/net/avm_cpmac/cpphy_entry.c	2012-05-10 10:29:01+02:00
+++ linux-2.6.32/drivers/net/avm_cpmac/cpphy_entry.c	2012-05-10 10:29:01+02:00
@@ -605,6 +605,8 @@
     /* Print driver version */
     DEB_ERR("Version: %s\n", AVM_CPMAC_VERSION);
 
+    printk(KERN_INFO "cpmac: compiled with sizeof(struct sk_buff) = %d\n", sizeof(struct sk_buff));
+
     /* Initialize administrative array */
     memset(&cpmac_global, 0, sizeof(cpmac_global_t));
     cpmac_global.products = &cpmac_products;
