--- base/etc/default.callmonitor/callmonitor.cfg
+++ base/etc/default.callmonitor/callmonitor.cfg
@@ -5,7 +5,7 @@
 export CALLMONITOR_REVERSE='yes'
 export CALLMONITOR_REVERSE_CACHE='transient'
 export CALLMONITOR_REVERSE_PROVIDER=''
-export CALLMONITOR_AREA_PROVIDER='billiger'
+export CALLMONITOR_AREA_PROVIDER=''
 export CALLMONITOR_READ_FONBUCH='no'
 export CALLMONITOR_PHONEBOOKS='callers cache avm'
 export CALLMONITOR_MON_HOST='127.0.0.1'
--- base/usr/lib/callmonitor/reverse/provider.cfg
+++ base/usr/lib/callmonitor/reverse/provider.cfg
@@ -10,4 +10,3 @@
 R	pronto_it	39!	pronto.it	pronto.it
 R	wittegids_be	32!	www.wittegids.be	wittegids.be
 R	userdef	*	.	$(lang de:"Benutzerdefiniert" en:"User-defined")
-A	billiger	*	www.billiger-telefonieren.de	billiger-telefonieren.de
