--- etc/init.d/rc.net.orig	2011-01-13 15:34:48.800905570 +0100
+++ etc/init.d/rc.net	2011-01-13 15:34:49.010905570 +0100
@@ -73,6 +73,8 @@
 #   a) it has the parameter at all AND
 #   b) upnpd binary does *not* exist
 [ "$_dsld_upnp" ] && [ ! "$_upnpd" ] && DSLDDPARAM="-g"
+# Do we have a Samba server (smbd) or was ist stripped from the firmware?
+_smbd=$(basename $(which smbd) 2> /dev/null)
 
 
 upnpdstart()
@@ -330,6 +332,7 @@
 #we have the samba cfg in /var, so crate them there
 sambastart()
 {
+   [ "$_smbd" ] || return;
    cd /
    tar xf /var.tar
    /bin/smbd
