--- usr/www/all/html/de/home/home.js.orig	2009-07-09 09:59:07.000000000 +0000
+++ usr/www/all/html/de/home/home.js	2009-08-07 08:30:47.000000000 +0000
@@ -585,14 +585,13 @@
 var nPartTotal = Number("<? query ctlusb:settings/storage-part-total ?>");
 if (isNaN(nPartTotal)) nPartTotal = 0;
 var unplugOk = parseInt("<? query ctlusb:settings/unplug-ok ?>", 10) == 1;
-if (nPartTotal > nStor) {
+if (nPartTotal > 0) {
 var strStor = "";
 if (unplugOk) {
 strStor = "Alle USB-Speicher wurden \"sicher entfernt\" und können jetzt abgezogen werden.";
 }
 else {
-strStor = "Mindestens ein USB-Speicher oder Teile davon sind nicht eingebunden. ";
-strStor += "<a href=\"javascript:jslGoTo('home', 'syslog');\">Weitere Informationen.</a>";
+strStor = "Es wurden " + nStor + " von " + nPartTotal + " Partitionen eingebunden.";
 }
 strArr.push(strStor);
 }
