--- src/mount_davfs.c
+++ src/mount_davfs.c
@@ -672,7 +672,7 @@
     struct mntent *ft = getmntent(fstab);
     while (ft) {
         if (ft->mnt_dir) {
-            char *mp = canonicalize_file_name(ft->mnt_dir);
+            char *mp = realpath(ft->mnt_dir, NULL);
             if (mp) {
                 if (strcmp(mp, mpoint) == 0) {
                     free(mp);
@@ -911,7 +911,7 @@
             url = ne_strdup(argv[i]);
         }
         i++;
-        mpoint = canonicalize_file_name(argv[i]);
+        mpoint = realpath(argv[i], NULL);
         if (!mpoint)
             error(EXIT_FAILURE, 0,
                   _("can't evaluate path of mount point %s"), mpoint);
@@ -2109,7 +2109,7 @@
                 error_at_line(EXIT_FAILURE, 0, filename, lineno,
                               _("malformed line"));
             *(parmv[0] + strlen(parmv[0]) - 1) = '\0';
-            char *mp = canonicalize_file_name(parmv[0] + 1);
+            char *mp = realpath(parmv[0] + 1, NULL);
             if (mp) {
                 applies = (strcmp(mp, mpoint) == 0);
                 free(mp);
@@ -2364,7 +2364,7 @@
             if (scheme && !port)
                 port = ne_uri_defaultport(scheme);
 
-            char *mp = canonicalize_file_name(parmv[0]);
+            char *mp = realpath(parmv[0], NULL);
 
             char *ccert = NULL;
             if (args->clicert) {
--- src/umount_davfs.c
+++ src/umount_davfs.c
@@ -26,6 +26,9 @@
 #ifdef HAVE_LIBINTL_H
 #include <libintl.h>
 #endif
+#ifdef HAVE_LIMITS_H
+#include <limits.h>
+#endif
 #ifdef HAVE_LOCALE_H
 #include <locale.h>
 #endif
@@ -118,7 +121,7 @@
     if (optind < (argc - 1))
         error(EXIT_FAILURE, 0, _("too many arguments"));
 
-    char *mpoint = canonicalize_file_name(argv[optind]);
+    char *mpoint = realpath(argv[optind], NULL);
 
     char *umount_command = NULL;
     if (mpoint) {
