--- configure
+++ configure
@@ -5987,6 +5987,7 @@
 ALIGNOF_UINT64_T="$ac_cv_alignof_uint64_t"
 
 
+if test "$enable_openssl" = yes; then
 for ac_header in openssl/blowfish.h openssl/des.h openssl/cast.h openssl/aes.h
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
@@ -6002,6 +6003,7 @@
 fi
 
 done
+fi
 
 
 # The Ultrix 4.2 mips builtin alloca declared by alloca.h only works
