--- gdb/Makefile.in
+++ gdb/Makefile.in
@@ -669,9 +669,9 @@
 demangle_h =    $(INCLUDE_DIR)/demangle.h
 obstack_h =     $(INCLUDE_DIR)/obstack.h
 opcode_m68hc11_h = $(INCLUDE_DIR)/opcode/m68hc11.h
-readline_h = 	$(READLINE_SRC)/readline.h
-readline_tilde_h =	$(READLINE_SRC)/tilde.h
-readline_history_h =	$(READLINE_SRC)/history.h
+#readline_h = 	$(READLINE_SRC)/readline.h
+#readline_tilde_h =	$(READLINE_SRC)/tilde.h
+#readline_history_h =	$(READLINE_SRC)/history.h
 frv_desc_h =	$(OPCODES_SRC)/frv-desc.h
 mep_desc_h =	$(OPCODES_SRC)/mep-desc.h
 mep_opc_h =	$(OPCODES_SRC)/mep-opc.h
