--- configure
+++ configure
@@ -7020,6 +7020,9 @@
 else
    MISC_TERMINFO_DIRS=""
 fi
+if test "$cross_compiling" = yes; then
+      { $as_echo "$as_me:${as_lineno-$LINENO}: result: assuming yes" >&5; $as_echo " assuming yes" >&6; }
+else
 JD_Terminfo_Dirs="$MISC_TERMINFO_DIRS \
                   /usr/lib/terminfo \
                   /usr/share/terminfo \
@@ -7037,6 +7040,7 @@
       break
    fi
 done
+fi
 if test "$TERMCAP"; then
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
