--- configure.orig	2008-11-24 22:05:27.000000000 +0100
+++ configure	2008-11-24 22:08:14.000000000 +0100
@@ -45691,8 +45691,7 @@
 
   if test x$GLIB_GENMARSHAL = xno; then
     { { $as_echo "$as_me:$LINENO: error: Could not find a glib-genmarshal in your PATH" >&5
-$as_echo "$as_me: error: Could not find a glib-genmarshal in your PATH" >&2;}
-   { (exit 1); exit 1; }; }
+$as_echo "$as_me: error: Could not find a glib-genmarshal in your PATH" >&2;} }
   fi
 fi
 
