--- glib-2.0.pc.in.orig	2008-10-03 22:32:17.000000000 +0200
+++ glib-2.0.pc.in	2008-10-03 22:32:42.000000000 +0200
@@ -10,6 +10,5 @@
 Name: GLib
 Description: C Utility Library
 Version: @VERSION@
-Libs: -L${libdir} -lglib-2.0 @INTLLIBS@
-Libs.private: @ICONV_LIBS@
+Libs: -L${libdir} -lglib-2.0 @INTLLIBS@ @ICONV_LIBS@
 Cflags: -I${includedir}/glib-2.0 -I${libdir}/glib-2.0/include @GLIB_EXTRA_CFLAGS@
