--- tests/Makefile.am	2006-02-25 13:44:06.720915467 +0100
+++ tests/Makefile.am	2006-02-25 13:42:24.136569782 +0100
@@ -21,7 +21,7 @@
 TESTS = prime register ac basic tsexp keygen pubkey benchmark random
 
 INCLUDES = -I$(top_srcdir)/src
-LDADD = ../src/libgcrypt.la
+LDADD = ../src/libgcrypt.la @LTLIBOBJS@ @GPG_ERROR_LIBS@
 
 EXTRA_PROGRAMS = testapi
 noinst_PROGRAMS = $(TESTS)
--- tests/Makefile.in	2006-02-25 13:44:06.721915315 +0100
+++ tests/Makefile.in	2006-02-25 13:42:24.136569782 +0100
@@ -263,7 +263,7 @@
 target_vendor = @target_vendor@
 TESTS = prime register ac basic tsexp keygen pubkey benchmark random
 INCLUDES = -I$(top_srcdir)/src
-LDADD = ../src/libgcrypt.la
+LDADD = ../src/libgcrypt.la @LTLIBOBJS@ @GPG_ERROR_LIBS@
 AM_CFLAGS = @GPG_ERROR_CFLAGS@
 all: all-am
 
