--- configure
+++ configure
@@ -44430,14 +44430,15 @@
 # }}}
 
 # pkg-config --exists 'libxml-2.0'; pkg-config --exists libvirt {{{
-with_libxml2="no (pkg-config isn't available)"
+#with_libxml2="no (pkg-config isn't available)"
 with_libxml2_cflags=""
 with_libxml2_ldflags=""
-with_libvirt="no (pkg-config isn't available)"
+#with_libvirt="no (pkg-config isn't available)"
 with_libvirt_cflags=""
 with_libvirt_ldflags=""
 if test "x$PKG_CONFIG" != "x"
 then
+	if test "${with_libxml2+set}" != set; then
 	pkg-config --exists 'libxml-2.0' 2>/dev/null
 	if test "$?" = "0"
 	then
@@ -44445,7 +44446,9 @@
 	else
 		with_libxml2="no (pkg-config doesn't know library)"
 	fi
+	fi
 
+	if test "${with_libvirt+set}" != set; then
 	pkg-config --exists libvirt 2>/dev/null
 	if test "$?" = "0"
 	then
@@ -44453,6 +44456,7 @@
 	else
 		with_libvirt="no (pkg-config doesn't know library)"
 	fi
+	fi
 fi
 if test "x$with_libxml2" = "xyes"
 then
@@ -44953,7 +44957,7 @@
 # }}}
 
 # $PKG_CONFIG --exists OpenIPMIpthread {{{
-with_libopenipmipthread="yes"
+#with_libopenipmipthread="yes"
 with_libopenipmipthread_cflags=""
 with_libopenipmipthread_libs=""
 
