--- Makefile.in
+++ Makefile.in
@@ -12455,7 +12455,7 @@
 	$(INSTALL_DIR) $(DESTDIR)$(bindir)
 	$(INSTALL_DIR) $(DESTDIR)$(libdir)/wx/config
 	$(INSTALL_PROGRAM) lib/wx/config/$(TOOLCHAIN_FULLNAME) $(DESTDIR)$(libdir)/wx/config
-	(cd $(DESTDIR)$(bindir) && rm -f wx-config && $(LN_S) $(libdir)/wx/config/$(TOOLCHAIN_FULLNAME) wx-config)
+	(cd $(DESTDIR)$(bindir) && rm -f wx-config && $(LN_S) $(DESTDIR)$(libdir)/wx/config/$(TOOLCHAIN_FULLNAME) wx-config)
 
 locale_install: 
 	$(INSTALL_DIR) $(DESTDIR)$(datadir)/locale
--- wx-config.in
+++ wx-config.in
@@ -365,9 +365,9 @@
 
 
 # Determine the base directories we require.
-prefix=${input_option_prefix-${this_prefix:-@prefix@}}
-exec_prefix=${input_option_exec_prefix-${input_option_prefix-${this_exec_prefix:-@exec_prefix@}}}
-wxconfdir="@libdir@/wx/config"
+prefix=${input_option_prefix-${this_prefix:-TARGET_TOOLCHAIN_STAGING_DIR_PLACEHOLDER@prefix@}}
+exec_prefix=${input_option_exec_prefix-${input_option_prefix-${this_exec_prefix:-TARGET_TOOLCHAIN_STAGING_DIR_PLACEHOLDER@exec_prefix@}}}
+wxconfdir="TARGET_TOOLCHAIN_STAGING_DIR_PLACEHOLDER@libdir@/wx/config"
 
 installed_configs=`cd "$wxconfdir" 2> /dev/null && ls | grep -v "^inplace-"`
 
@@ -903,9 +903,9 @@
 prefix=${this_prefix-$prefix}
 exec_prefix=${this_exec_prefix-$exec_prefix}
 
-includedir="@includedir@"
-libdir="@libdir@"
-bindir="@bindir@"
+includedir="TARGET_TOOLCHAIN_STAGING_DIR_PLACEHOLDER@includedir@"
+libdir="TARGET_TOOLCHAIN_STAGING_DIR_PLACEHOLDER@libdir@"
+bindir="TARGET_TOOLCHAIN_STAGING_DIR_PLACEHOLDER@bindir@"
 
 # Trivial queries we can answer now.
 [ -z "$output_option_prefix"        ] || echo $prefix
