--- var/install
+++ var/install
@@ -217,11 +217,11 @@
 export ANNEX=`cat ${CONFIG_ENVIRONMENT_PATH}/annex`
 if [ -z "${ANNEX}" ] ; then echo ANNEX=${ANNEX} not supported ; exit $INSTALL_WRONG_HARDWARE ; fi
 echo ANNEX=$ANNEX
-# Fritz_Box_7170_AnnexA
-if [ -z "$DISABLE_Fritz_Box_7170_AnnexA" ] ; then
-   echo testing acceptance for device Fritz_Box_7170_AnnexA ...
-   if /etc/version --install=ar7_8MB_xilinx_4eth_3ab_isdn_nt_te_pots_wlan_usb_host_25762 ; then
-       echo device has installtype ar7_8MB_xilinx_4eth_3ab_isdn_nt_te_pots_wlan_usb_host_25762
+# Fritz_Box_7140_AnnexA
+if [ -z "$DISABLE_Fritz_Box_7140_AnnexA" ] ; then
+   echo testing acceptance for device Fritz_Box_7140_AnnexA ...
+   if /etc/version --install=ar7_8MB_xilinx_4eth_2ab_isdn_te_pots_wlan_usb_host_62068 ; then
+       echo device has installtype ar7_8MB_xilinx_4eth_2ab_isdn_te_pots_wlan_usb_host_62068
        # passt der ANNEX ?
        accept_annex=A
        echo assumed ANNEX ${accept_annex} -- found ANNEX "${ANNEX}"
@@ -242,11 +242,11 @@
                    exit $INSTALL_WRONG_HARDWARE
                fi
            fi
-           echo OK - accept this update for device Fritz_Box_7170_AnnexA ...
+           echo OK - accept this update for device Fritz_Box_7140_AnnexA ...
            korrekt_version=1
        fi
    fi
-   echo testing acceptance for device Fritz_Box_7170_AnnexA done
+   echo testing acceptance for device Fritz_Box_7140_AnnexA done
 fi
 
 kernel_start=0x90010000
