--- dist/configure
+++ dist/configure
@@ -17654,7 +17654,7 @@
 
 fi
 
-for ac_header in execinfo.h sys/select.h sys/socket.h sys/time.h
+for ac_header in sys/select.h sys/socket.h sys/time.h
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
 ac_fn_c_check_header_mongrel "$LINENO" "$ac_header" "$as_ac_Header" "$ac_includes_default"
@@ -20652,7 +20652,7 @@
 
 # Check for system functions we optionally use.
 for ac_func in \
-	_fstati64 backtrace backtrace_symbols directio fchmod fclose\
+	_fstati64 directio fchmod fclose\
 	fcntl fdatasync fgetc fgets fopen fwrite getgid\
 	getrusage getuid hstrerror mprotect pstat_getdynamic\
 	pthread_self pthread_yield random sched_yield select setgid setuid\
