--- configure.ac
+++ configure.ac
@@ -747,8 +747,6 @@
 	RtlSecureZeroMemory \
 	SecureZeroMemory \
 	accept4 \
-	backtrace \
-	backtrace_symbols_fd \
 	eventfd \
 	explicit_bzero \
 	timingsafe_memcmp \
@@ -1685,7 +1683,6 @@
 		  unistd.h \
 		  arpa/inet.h \
 		  crt_externs.h \
-		  execinfo.h \
 		  gnu/libc-version.h \
 		  grp.h \
 		  ifaddrs.h \
--- configure
+++ configure
@@ -8884,8 +8884,6 @@
 	RtlSecureZeroMemory \
 	SecureZeroMemory \
 	accept4 \
-	backtrace \
-	backtrace_symbols_fd \
 	eventfd \
 	explicit_bzero \
 	timingsafe_memcmp \
@@ -13053,7 +13051,6 @@
 		  unistd.h \
 		  arpa/inet.h \
 		  crt_externs.h \
-		  execinfo.h \
 		  gnu/libc-version.h \
 		  grp.h \
 		  ifaddrs.h \
