--- configure.ac
+++ configure.ac
@@ -377,8 +377,6 @@
 	RtlSecureZeroMemory \
 	SecureZeroMemory \
         accept4 \
-        backtrace \
-        backtrace_symbols_fd \
 	eventfd \
 	explicit_bzero \
 	timingsafe_memcmp \
@@ -1005,7 +1003,6 @@
                   unistd.h \
                   arpa/inet.h \
                   crt_externs.h \
-                  execinfo.h \
                   grp.h \
                   ifaddrs.h \
                   inttypes.h \
--- configure
+++ configure
@@ -7943,8 +7943,6 @@
 	RtlSecureZeroMemory \
 	SecureZeroMemory \
         accept4 \
-        backtrace \
-        backtrace_symbols_fd \
 	eventfd \
 	explicit_bzero \
 	timingsafe_memcmp \
@@ -11246,7 +11244,6 @@
                   unistd.h \
                   arpa/inet.h \
                   crt_externs.h \
-                  execinfo.h \
                   grp.h \
                   ifaddrs.h \
                   inttypes.h \
