--- configure.ac
+++ configure.ac
@@ -392,8 +392,6 @@
 	RtlSecureZeroMemory \
 	SecureZeroMemory \
         accept4 \
-        backtrace \
-        backtrace_symbols_fd \
 	eventfd \
 	explicit_bzero \
 	timingsafe_memcmp \
@@ -989,7 +987,6 @@
 AC_CHECK_HEADERS(
         arpa/inet.h \
         crt_externs.h \
-        execinfo.h \
         grp.h \
         ifaddrs.h \
         inttypes.h \
--- configure
+++ configure
@@ -7048,8 +7048,6 @@
 	RtlSecureZeroMemory \
 	SecureZeroMemory \
         accept4 \
-        backtrace \
-        backtrace_symbols_fd \
 	eventfd \
 	explicit_bzero \
 	timingsafe_memcmp \
@@ -10245,7 +10243,6 @@
 
 for ac_header in arpa/inet.h \
         crt_externs.h \
-        execinfo.h \
         grp.h \
         ifaddrs.h \
         inttypes.h \
