--- configure.ac
+++ configure.ac
@@ -581,8 +581,6 @@
 	RtlSecureZeroMemory \
 	SecureZeroMemory \
 	accept4 \
-	backtrace \
-	backtrace_symbols_fd \
 	eventfd \
 	explicit_bzero \
 	timingsafe_memcmp \
--- configure
+++ configure
@@ -8607,8 +8607,6 @@
 	RtlSecureZeroMemory \
 	SecureZeroMemory \
 	accept4 \
-	backtrace \
-	backtrace_symbols_fd \
 	eventfd \
 	explicit_bzero \
 	timingsafe_memcmp \
@@ -12646,7 +12644,6 @@
 		  unistd.h \
 		  arpa/inet.h \
 		  crt_externs.h \
-		  execinfo.h \
 		  gnu/libc-version.h \
 		  grp.h \
 		  ifaddrs.h \
