--- configure.ac
+++ configure.ac
@@ -588,8 +588,6 @@
 	RtlSecureZeroMemory \
 	SecureZeroMemory \
         accept4 \
-        backtrace \
-        backtrace_symbols_fd \
 	eventfd \
 	explicit_bzero \
 	timingsafe_memcmp \
@@ -1416,7 +1414,6 @@
                   unistd.h \
                   arpa/inet.h \
                   crt_externs.h \
-                  execinfo.h \
                   gnu/libc-version.h \
                   grp.h \
                   ifaddrs.h \
--- configure
+++ configure
@@ -8554,8 +8554,6 @@
 	RtlSecureZeroMemory \
 	SecureZeroMemory \
         accept4 \
-        backtrace \
-        backtrace_symbols_fd \
 	eventfd \
 	explicit_bzero \
 	timingsafe_memcmp \
@@ -12367,7 +12365,6 @@
                   unistd.h \
                   arpa/inet.h \
                   crt_externs.h \
-                  execinfo.h \
                   gnu/libc-version.h \
                   grp.h \
                   ifaddrs.h \
