--- configure.ac
+++ configure.ac
@@ -392,8 +392,6 @@
 	RtlSecureZeroMemory \
 	SecureZeroMemory \
         accept4 \
-        backtrace \
-        backtrace_symbols_fd \
         clock_gettime \
 	eventfd \
 	explicit_bzero \
@@ -963,7 +961,6 @@
 AC_CHECK_HEADERS(
         arpa/inet.h \
         crt_externs.h \
-        execinfo.h \
         grp.h \
         ifaddrs.h \
         inttypes.h \
--- configure
+++ configure
@@ -7048,8 +7048,6 @@
 	RtlSecureZeroMemory \
 	SecureZeroMemory \
         accept4 \
-        backtrace \
-        backtrace_symbols_fd \
         clock_gettime \
 	eventfd \
 	explicit_bzero \
@@ -10192,7 +10190,6 @@
 
 for ac_header in arpa/inet.h \
         crt_externs.h \
-        execinfo.h \
         grp.h \
         ifaddrs.h \
         inttypes.h \
