--- configure.ac
+++ configure.ac
@@ -743,8 +743,6 @@
 	RtlSecureZeroMemory \
 	SecureZeroMemory \
 	accept4 \
-	backtrace \
-	backtrace_symbols_fd \
 	eventfd \
 	explicit_bzero \
 	timingsafe_memcmp \
@@ -1681,7 +1679,6 @@
 		  unistd.h \
 		  arpa/inet.h \
 		  crt_externs.h \
-		  execinfo.h \
 		  gnu/libc-version.h \
 		  grp.h \
 		  ifaddrs.h \
--- configure
+++ configure
@@ -8871,8 +8871,6 @@
 	RtlSecureZeroMemory \
 	SecureZeroMemory \
 	accept4 \
-	backtrace \
-	backtrace_symbols_fd \
 	eventfd \
 	explicit_bzero \
 	timingsafe_memcmp \
@@ -13040,7 +13038,6 @@
 		  unistd.h \
 		  arpa/inet.h \
 		  crt_externs.h \
-		  execinfo.h \
 		  gnu/libc-version.h \
 		  grp.h \
 		  ifaddrs.h \
