--- configure
+++ configure
@@ -8635,6 +8635,7 @@
 
 
 
+if false; then
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler accepts -fasynchronous-unwind-tables" >&5
 $as_echo_n "checking whether the compiler accepts -fasynchronous-unwind-tables... " >&6; }
 if ${tor_cv_cflags__fasynchronous_unwind_tables+:} false; then :
@@ -8690,6 +8691,7 @@
   if test x$tor_cv_cflags__fasynchronous_unwind_tables = xyes; then
     CFLAGS="$CFLAGS -fasynchronous-unwind-tables"
   fi
+fi
 
 
 
