--- etc/hotplug/run_mount.orig	2010-02-26 20:32:06.886023369 +0100
+++ etc/hotplug/run_mount	2010-02-26 20:32:19.026033808 +0100
@@ -133,6 +133,10 @@
 /etc/ftpd_control reconfig
 fi
 test -x $TR069START && $TR069START $MNTNAME
+AUTORUN=$MNTPATH/autorun.sh
+[ -x $AUTORUN ] && $AUTORUN &
+
+[ -r /etc/external.pkg ] && /etc/init.d/rc.external start $MNTPATH &
 if test -x /etc/samba_control ; then
 /etc/samba_control reconfig
 fi
