--- etc/hotplug/run_mount.orig	2010-02-23 19:56:57.000000000 +0100
+++ etc/hotplug/run_mount	2010-02-23 19:57:31.000000000 +0100
@@ -125,6 +125,10 @@
 /etc/ftpd_control reconfig
 fi
 test -x $TR069START && $TR069START $MNTNAME
+AUTORUN=$MNTPATH/autorun.sh
+[ -x $AUTORUN ] && $AUTORUN & 
+
+[ -r /etc/external.pkg ] && /etc/init.d/rc.external start $MNTPATH &
 if test -x /etc/samba_control ; then
 /etc/samba_control reconfig
 fi
