--- etc/hotplug/storage.orig	2010-02-26 20:36:38.977274669 +0100
+++ etc/hotplug/storage	2010-02-26 20:36:50.787272517 +0100
@@ -212,6 +212,10 @@
 local MPOINT=$2
 local MPNAME="${MPOINT##*/}"
 /etc/hotplug/morphstick lost_all_partitions
+AUTOEND=$MPOINT/autoend.sh
+[ -x $AUTOEND ] && $AUTOEND
+
+[ -r /etc/external.pkg ] && /etc/init.d/rc.external stop $MPOINT
 test -x $MSTOP && $MSTOP
 for MP in /var/media/ftp/* ; do
 rmdir "$MP" > /dev/null
