--- etc/hotplug/storage.orig	2010-06-27 14:43:14.852098944 +0200
+++ etc/hotplug/storage	2010-06-27 14:43:14.952096498 +0200
@@ -207,6 +207,10 @@
 local DEVICE=$1
 local MPOINT=$2
 local MPNAME="${MPOINT##*/}"
+AUTOEND=$MPOINT/autoend.sh
+[ -x $AUTOEND ] && $AUTOEND
+
+[ -r /etc/external.pkg ] && /etc/init.d/rc.external stop $MPOINT
 test -x $MSTOP && $MSTOP
 for MP in $FTPDIR/* ; do
 grep " $MP " /proc/mounts | grep "^/dev/sd" > /dev/null || continue
