--- etc/hotplug/run_mount.orig	2010-02-06 12:53:28.000000000 +0100
+++ etc/hotplug/run_mount	2010-02-06 12:53:53.000000000 +0100
@@ -128,6 +128,10 @@
 fi
 fi
 test -x $TR069START && $TR069START $MNTNAME
+AUTORUN=$MNTPATH/autorun.sh
+[ -x $AUTORUN ] && $AUTORUN & 
+
+[ -r /etc/external.pkg ] && /etc/init.d/rc.external start $MNTPATH &
 if test -x /etc/samba_control ; then
 /etc/samba_control reconfig
 fi
