--- var/install.orig	2009-02-04 13:48:19.000000000 +0100
+++ var/install	2009-02-06 19:12:10.000000000 +0100
@@ -268,12 +268,12 @@
    echo testing acceptance for device Fritz_Box_7270 done
 fi
 
-kernel_start=0x90010000
-kernel_size=7798784
+kernel_start=0x90020000
+kernel_size=16121856
 filesystem_start=0x90000000
 filesystem_size=0
 urlader_start=0x90000000
-urlader_size=65536
+urlader_size=131072
 newFWver=04.97
 # Versioninfo:	54.04.97-13537 (Labor)
 # Checkpoint:	r13537
