--- var/install.org	2008-11-24 17:26:03.000000000 +0100
+++ var/install	2008-12-11 01:47:20.000000000 +0100
@@ -224,12 +224,12 @@
    echo testing acceptance for device Fritz_Box_7270 done
 fi
 
-kernel_start=0x90010000
-kernel_size=7798784
+kernel_start=0x90020000
+kernel_size=16121856
 filesystem_start=0x90000000
 filesystem_size=0
 urlader_start=0x90000000
-urlader_size=65536
+urlader_size=131072
 newFWver=04.99
 # Versioninfo:	54.04.99-12606 (Labor)
 # Checkpoint:	r12606
