--- var/install.orig	2009-02-25 10:42:39.000000000 +0100
+++ var/install	2009-02-27 09:06:16.000000000 +0100
@@ -268,12 +268,12 @@
    echo testing acceptance for device Fritz_Box_7270 done
 fi
 
-kernel_start=0x90010000
-kernel_size=7798784
+kernel_start=0x90020000
+kernel_size=16121856
 filesystem_start=0x90000000
 filesystem_size=0
 urlader_start=0x90000000
-urlader_size=65536
+urlader_size=131072
 newFWver=04.70
 # Versioninfo:	54.04.70
 # Checkpoint:	r13713
