--- usr/www/all/menus/menu_data.lua
+++ usr/www/all/menus/menu_data.lua
@@ -619,6 +619,12 @@
 ["lua"] = "system/infoled.lua",
 ["help"] = (forLuaOnly and "hilfe_system_infoanzeige") or true
 } or nil,
+-- TODO: untested, check if "led" is the right key and if adding pageData["led"] is enough
+["led"] = {
+["show"] = data.expertMode,
+["lua"] = "system/led_display.lua",
+["help"] = forLuaOnly and "hilfe_system_anzeige"
+} or nil,
 ["keyLo"] = {
 ["show"] = true,
 ["js"] = "system/keylocker.js",
@@ -2186,6 +2192,7 @@
 ["par"] = "mfn",
 ["pos"] = 10
 } or nil,
+-- AVM defines menuData["led"] but never defines pageData["led"], we make this entry work by defining pageData["led"]
 ["led"] = pageData["led"] and {
 ["txt"] = TXT([[{?menuLed?}]]),
 ["par"] = "mfn",
