--- usr/www/all/menus/menu_data.lua
+++ usr/www/all/menus/menu_data.lua
@@ -545,6 +545,12 @@
 ["lua"] = "system/infoled.lua",
 ["help"] = (forLuaOnly and "hilfe_system_infoanzeige") or true
 } or nil,
+-- TODO: untested, check if "led" is the right key and if adding pageData["led"] is enough
+["led"] = {
+["show"] = data.expertMode,
+["lua"] = "system/led_display.lua",
+["help"] = forLuaOnly and "hilfe_system_anzeige"
+} or nil,
 ["keyLo"] = {
 ["show"] = true,
 ["lua"] = "system/keylock.lua",
