--- var/install
+++ var/install
@@ -217,11 +217,11 @@
 export ANNEX=`cat ${CONFIG_ENVIRONMENT_PATH}/annex`
 if [ -z "${ANNEX}" ] ; then echo ANNEX=${ANNEX} not supported ; exit $INSTALL_WRONG_HARDWARE ; fi
 echo ANNEX=$ANNEX
-# Fritz_Box_7170
-if [ -z "$DISABLE_Fritz_Box_7170" ] ; then
-   echo testing acceptance for device Fritz_Box_7170 ...
-   if /etc/version --install=ar7_8MB_xilinx_4eth_3ab_isdn_nt_te_pots_wlan_usb_host_25762 ; then
-       echo device has installtype ar7_8MB_xilinx_4eth_3ab_isdn_nt_te_pots_wlan_usb_host_25762
+# Fritz_Box_DECT_W900V
+if [ -z "$DISABLE_Fritz_Box_DECT_W900V" ] ; then
+   echo testing acceptance for device Fritz_Box_DECT_W900V ...
+      if /etc/version --install=ar7_8MB_xilinx_4eth_2ab_isdn_nt_te_pots_wlan_usb_host_dect_37264 ; then
+       echo device has installtype ar7_8MB_xilinx_4eth_2ab_isdn_nt_te_pots_wlan_usb_host_dect_37264
        # passt der ANNEX ?
        accept_annex=B
        echo assumed ANNEX ${accept_annex} -- found ANNEX "${ANNEX}"
@@ -230,7 +230,7 @@
            # passt der OEM ?
            if [ ! -z "${OEM}" ] ; then
                oem_found=0
-               for i in  avm 1und1 ; do
+               for i in  avm 1und1 tcom ; do
                    if [ "$i" = "${OEM}" ] ; then
                        echo "OK - OEM ${OEM} is supported"
                        oem_found=1
