--- var/install
+++ var/install
@@ -180,6 +180,11 @@
 urlader_size=262144
 newFWver=04.91
 flash_start=268435456
+
+found=$(grep  -e  '^mtd[1\|5]' /proc/sys/urlader/environment | \
+	grep -c -e  '0x90040000,0x907E0000\|0x907E0000,0x90F80000')
+[ 2 -eq $found ] || korrekt_version=0
+
 # Versioninfo:	75.04.91
 # Checkpoint:	r19965
 #! /bin/sh
@@ -470,6 +475,10 @@
     echo install: ${kversion} setting files to install...
     if [ -f /var/tmp/kernel.image ] ; then
         crc_param=1
+        ### 7570_HN: set mtd1 to max size and unset mtd5
+        echo mtd1 0x90040000,0x90F80000 > /proc/sys/urlader/environment
+        echo mtd5 > /proc/sys/urlader/environment
+        ###
         echo "install: /var/tmp/kernel.image to start($kernel_update_start) size($kernel_update_len)"
         echo "update_parameter=flash_update_file0=\"/var/tmp/kernel.image,$kernel_update_start,$kernel_update_len,crc=${crc_param}\"" >>/var/post_install
         need_reboot=$INSTALL_SUCCESS_REBOOT
