--- etc/fstab	2008-07-23 09:03:32.000000000 +0200
+++ etc/fstab.new	2008-07-23 09:03:55.000000000 +0200
@@ -3,3 +3,4 @@
 # <file system> <mount point>   <type>  <options>               <dump>  <pass>
 proc            /proc           proc    nosuid,nodev,noexec     0       0
 tmpfs           /var            tmpfs   defaults                0       0
+UUID=d03cb7bb-d50e-446e-8312-b8e27ed107a7 /media/export ext3 noatime,nodiratime 0 1
