--- etc/fstab	2008-09-11 22:47:36.000000000 +0200
+++ etc/fstab.new	2008-09-11 22:51:43.000000000 +0200
@@ -3,4 +3,3 @@
 # <file system> <mount point>   <type>  <options>               <dump>  <pass>
 proc            /proc           proc    nosuid,nodev,noexec     0       0
 tmpfs           /var            tmpfs   defaults                0       0
-sysfs           /sys            sysfs   nosuid,nodev,noexec     0       0
