--- etc/init.d/rc.S	2007-11-16 05:08:36.000000000 +0100
+++ etc/init.d/rc.S	2007-11-16 05:15:07.000000000 +0100
@@ -570,6 +570,15 @@
 echo "<? setvariable var:ServiceportalUrl '${CONFIG_SERVICEPORTAL_URL}' ?>" >>/var/config.def
 echo "<? setvariable var:AccessoryUrl '${CONFIG_ACCESSORY_URL}' ?>" >>/var/config.def
 echo "<? setvariable var:FirmwareVersion '${CONFIG_VERSION}${CONFIG_SUBVERSION}' ?>" >>/var/config.def
+echo "<? setvariable var:isOrange 1 ?>" >>/var/config.def
+echo "<? setvariable var:isWizz 0 ?>" >>/var/config.def
+echo "<? setvariable var:isDSmod 1 ?>" >>/var/config.def
+echo "<? setvariable var:isTSB 0 ?>" >>/var/config.def
+echo "<? setvariable var:isCallPatch 1 ?>" >>/var/config.def
+echo "<? setvariable var:isWOL 0 ?>" >>/var/config.def
+echo "<? setvariable var:menuFreetz 'ds-mod' ?>" >>/var/config.def
+echo "<? setvariable var:menuTSB 'TSB-LCR' ?>" >>/var/config.def
+echo "<? setvariable var:menuWOL 'WoL' ?>" >>/var/config.def
 fi
 #########################################################################
 ## NODES init
