--- etc/init.d/rc.S	2007-11-16 04:07:32.000000000 +0100
+++ etc/init.d/rc.S	2007-11-16 04:07:32.000000000 +0100
@@ -572,6 +572,18 @@
 echo "<? setvariable var:FirmwareVersion '${CONFIG_VERSION}${CONFIG_SUBVERSION}' ?>" >>/var/config.def
 fi
 #########################################################################
+## Orangebox
+#########################################################################
+echo "<? setvariable var:isOrange 1 ?>" >>/var/config.def
+echo "<? setvariable var:isWizz 0 ?>" >>/var/config.def
+echo "<? setvariable var:isDSmod 1 ?>" >>/var/config.def
+echo "<? setvariable var:isTSB 0 ?>" >>/var/config.def
+echo "<? setvariable var:isCallPatch 1 ?>" >>/var/config.def
+echo "<? setvariable var:isWOL 0 ?>" >>/var/config.def
+echo "<? setvariable var:menuDSMOD 'ds-mod' ?>" >>/var/config.def
+echo "<? setvariable var:menuTSB 'TSB-LCR' ?>" >>/var/config.def
+echo "<? setvariable var:menuWOL 'WoL' ?>" >>/var/config.def
+#########################################################################
 ## NODES init
 #########################################################################
 major=`grep tffs /proc/devices`
