--- etc/init.d/rc.S	2007-11-21 21:20:36.000000000 +0100
+++ etc/init.d/rc.S	2007-11-21 21:20:37.000000000 +0100
@@ -588,6 +588,15 @@
 echo "<? setvariable var:OnlinehelpUrl '${CONFIG_ONLINEHELP_URL}' ?>" >>/var/config.def
 echo "<? setvariable var:FirmwareVersion '${CONFIG_VERSION}${CONFIG_SUBVERSION}' ?>" >>/var/config.def
 fi
+echo "<? setvariable var:isOrange 1 ?>" >>/var/config.def
+echo "<? setvariable var:isWizz 0 ?>" >>/var/config.def
+echo "<? setvariable var:isDSmod 1 ?>" >>/var/config.def
+echo "<? setvariable var:isTSB 0 ?>" >>/var/config.def
+echo "<? setvariable var:isCallPatch 1 ?>" >>/var/config.def
+echo "<? setvariable var:isWOL 0 ?>" >>/var/config.def
+echo "<? setvariable var:menuFreetz 'freetz' ?>" >>/var/config.def
+echo "<? setvariable var:menuTSB 'TSB-LCR' ?>" >>/var/config.def
+echo "<? setvariable var:menuWOL 'WoL' ?>" >>/var/config.def
 #########################################################################
 ## NODES init
 #########################################################################
