--- etc/init.d/rc.S	2008-01-24 18:08:44.000000000 +0100
+++ etc/init.d/rc.S	2008-02-10 15:38:46.000000000 +0100
@@ -590,6 +590,16 @@
 echo "<? setvariable var:AccessoryUrl '${CONFIG_ACCESSORY_URL}' ?>" >>${CONFIG_DEF}
 echo "<? setvariable var:OnlinehelpUrl '${CONFIG_ONLINEHELP_URL}' ?>" >>${CONFIG_DEF}
 echo "<? setvariable var:FirmwareVersion '${CONFIG_VERSION}${CONFIG_SUBVERSION}' ?>" >>${CONFIG_DEF}
+echo "<? setvariable var:isOrange 1 ?>" >>${CONFIG_DEF}
+echo "<? setvariable var:isWizz 0 ?>" >>${CONFIG_DEF}
+echo "<? setvariable var:isDSmod 1 ?>" >>${CONFIG_DEF}
+echo "<? setvariable var:isTSB 0 ?>" >>${CONFIG_DEF}
+echo "<? setvariable var:isCallPatch 1 ?>" >>${CONFIG_DEF}
+echo "<? setvariable var:isWOL 0 ?>" >>${CONFIG_DEF}
+echo "<? setvariable var:menuFreetz 'ds-mod' ?>" >>${CONFIG_DEF}
+echo "<? setvariable var:menuTSB 'TSB-LCR' ?>" >>${CONFIG_DEF}
+echo "<? setvariable var:menuWOL 'WoL' ?>" >>${CONFIG_DEF}
+
 fi
 #########################################################################
 ## NODES init
