--- etc/init.d/rc.S	2007-07-04 15:26:02.000000000 +0200
+++ etc/init.d/rc.S	2007-07-04 15:26:06.000000000 +0200
@@ -254,6 +254,15 @@
 echo "<? setvariable var:EthCount ${CONFIG_ETH_COUNT} ?>" >>/var/config.def
 echo "<? setvariable var:AbCount ${CONFIG_AB_COUNT} ?>" >>/var/config.def
 echo "<? setvariable var:FirmwareVersion '${CONFIG_VERSION}${CONFIG_SUBVERSION}' ?>" >>/var/config.def
+echo "<? setvariable var:isOrange 1 ?>" >>/var/config.def
+echo "<? setvariable var:isWizz 0 ?>" >>/var/config.def
+echo "<? setvariable var:isDSmod 1 ?>" >>/var/config.def
+echo "<? setvariable var:isTSB 0 ?>" >>/var/config.def
+echo "<? setvariable var:isCallPatch 1 ?>" >>/var/config.def
+echo "<? setvariable var:isWOL 0 ?>" >>/var/config.def
+echo "<? setvariable var:menuDSMOD 'ds-mod' ?>" >>/var/config.def
+echo "<? setvariable var:menuTSB 'TSB-LCR' ?>" >>/var/config.def
+echo "<? setvariable var:menuWOL 'WoL' ?>" >>/var/config.def
 #file "stdin", 8
 #file "./arch/etc/init.d/rc.tffs.sh", 0
 #########################################################################
