--- etc/init.d/rc.S.orig	2008-04-20 01:09:27.000000000 +0200
+++ etc/init.d/rc.S	2008-04-20 01:09:27.000000000 +0200
@@ -856,8 +856,7 @@
 ###########################
 tmp_iso_path=/lib/modules/cdrom.iso
 ## Warte auf Aktivierung CDROM - kein Fallback.
-modprobe avalanche_usb use_cdrom=1 iso_path=${tmp_iso_path} cdrom_fallback=1
-echo "usb client: CDROM Fallback=$CONFIG_CDROM_FALLBACK ISO=${tmp_iso_path}"
+modprobe avalanche_usb use_cdrom=0
 #########################################################################
 ## Set hostname
 #########################################################################
