--- usr/www/all/menus/menu_data.lua	2015-03-26 11:54:28.000000000 +0100
+++ usr/www/all/menus/menu_data.lua	2015-08-02 12:59:44.814095528 +0200
@@ -416,6 +416,12 @@
 page = "/dvb/signal.lua"
 }
 menu.add_item{
+text = [[TV-Dienste]],
+short = [[TV-Dienste]],
+menu = "dvb",
+page = "/dvb/services.lua"
+}
+menu.add_item{
 text = [[{?menuMaintenance?}]],
 short = [[{?menuMaintenanceShort?}]],
 menu = "diagnosis",
