--- etc/hotplug/storage.orig	2007-07-10 13:48:27.000000000 +0200
+++ etc/hotplug/storage	2007-07-11 13:41:08.000000000 +0200
@@ -91,7 +91,7 @@
 	grep -v ":$MPNAME$" $DEVMAP > /var/dev-$$.map
 	mv -f /var/dev-$$.map $DEVMAP
 
-	rm -rf $MPOINT
+	rmdir $MPOINT
 	eventadd 141 $MPNAME
 	sleep 1
 
@@ -270,13 +270,13 @@
 		rmmod usb-storage
 		rmmod scsi_mod
 		
-		rm -rf /var/media /var/REMOVING
+		rm -rf /var/REMOVING
 	fi
 	
 	if ! test -f /var/STARTING; then
 		for MP in /var/media/ftp/*; do
 			test -d $MP || continue
-			grep " $MP " /proc/mounts > /dev/null 2>&1 || rm -r $MP
+			grep " $MP " /proc/mounts > /dev/null 2>&1 || rmdir $MP
 		done
 	fi
 	vrijgeven
@@ -305,11 +305,11 @@
 	if ! test -f /var/STARTING; then
 		for MP in /var/media/ftp/*; do
 			test -d $MP || continue
-			grep " $MP " /proc/mounts > /dev/null 2>&1 || rm -r $MP
+			grep " $MP " /proc/mounts > /dev/null 2>&1 || rmdir $MP
 		done
 	fi
 
-	rm -rf /var/media /var/REMOVING
+	rm -rf /var/REMOVING
 
 	vrijgeven
 	;;
