--- etc/hotplug/run_mount.orig	2009-03-27 12:22:37.000000000 +0100
+++ etc/hotplug/run_mount	2009-03-27 12:22:38.000000000 +0100
@@ -36,31 +36,8 @@
 # Generate a frontend'able 'nice name' to be used as mount point:
 #
 nicename () {
-	local PROD
-	local SNUM
-	local MANU
-	local NAME="USB"
 	local NICE=""
-
-	# Find device's strings, remove/translate evil chars...
-	for VAR in `/sbin/lsusb -s -h $1|tr -d ' '`; do
-		if echo $VAR | grep "='" > /dev/null 2>&1; then
-			eval "${VAR%%=*}=`echo ${VAR##*=}|tr -c "\n\'a-zA-Z0-9" '-' | tr -d \' `"
-		fi
-	done
-	
-	# Find a string for our new nice name...
-	if ! test -z "$PROD"; then
-		NAME="$PROD"
-	elif ! test -z "$MANU"; then
-		NAME="$MANU"
-	fi
-	if ! test -z "$NAME"; then
-		NAME=`echo $NAME|tr ':' ' '`
-	fi
-	
-	# Build nice name: [Product/Vendor name]-Partition-<x>-<y>
-	NICE="$NAME-Partition-$2-$3"
+	NICE="uStor`echo $1|sed 's/^..//;s/a/0/;s/b/1/;s/c/2/;s/d/3/;s/e/4/;s/f/5/;s/g/6/;s/h/7/;s/i/8/;s/j/9/'`$3"
 	echo $NICE
 }
 
@@ -74,6 +51,36 @@
 USBLOCKFILE=/var/USBLOCK_storage
 test -f /etc/hotplug/rc.usbsema && . /etc/hotplug/rc.usbsema
 
+mount_fs () {
+ # mount according to type of filesystem 
+ # return exit code: true - all went well; other - something went wrong 
+ # gobal variables: 
+ #  $DEVNODE, $MNTPATH, READMODE, FTPUID, FTPGID 
+	FSTYPE="vfat"   # default as long as we dont know better 
+	[[ -x /usr/bin/fstyp ]] && FSTYPE=`/usr/bin/fstyp $DEVNODE` 
+	
+	case $FSTYPE in
+	vfat)
+		mount -t vfat -o $READMODE,uid=$FTPUID,gid=$FTPGID,fmask=0000,dmask=0000 $DEVNODE $MNTPATH
+		;;
+	ext2)
+		mount -t ext2 $DEVNODE $MNTPATH -o noatime,nodiratime,rw,async
+		;;
+	ext3)
+		mount -t ext3 $DEVNODE $MNTPATH -o noatime,nodiratime,rw,async
+		;;
+	ntfs)
+		/usr/bin/ntfs-3g $DEVNODE $MNTPATH -o force
+		;;
+	reiserfs)
+		mount -t reiserfs $DEVNODE $MNTPATH -o noatime,nodiratime,rw,async
+		;;
+	*) # fs type unknown
+		mount $DEVNODE $MNTPATH
+		;;
+	esac
+}
+
 # Actually perform the mount operation...
 # (TODO Checking the partition for known filesystem types)
 #
@@ -83,9 +90,10 @@
 	local DEVNODE=$2
 	local PARTNUM=$3
 	local BLKDEV=${DEVNODE##/dev/}
+	local BLKDEVMAIN=${BLKDEV:0:3}
 	local TEMP=${BLKDEV##sd}
 	local HOSTNUM=`echo ${TEMP%%[0-9]*}|tr 'abcd' '0123'`
-	local MNTNAME=`nicename $1 $HOSTNUM $PARTNUM`
+	local MNTNAME=`nicename $BLKDEVMAIN $HOSTNUM $PARTNUM`
 	local MNTPATH=$FTPDIR/$MNTNAME
 	
 	echo "Mounting $MNTNAME to device $DEVNODE..." > /dev/ttyS0
@@ -100,34 +108,8 @@
         local OLD_UMASK=`umask`
 	umask 0
 	local new_filesystem=false
-	if mount -t vfat -o $READMODE,uid=$FTPUID,gid=$FTPGID $DEVNODE $MNTPATH; then
-		new_filesystem=true
-	else
-		if test -x /bin/ntfs-3g ; then
-			ntfs-3g $DEVNODE $MNTPATH -o $READMODE
-			local NTFS_RET=$?
-			case "$NTFS_RET" in
-			0 )
+	if mount_fs; then
 				new_filesystem=true
-				;;
-
-			12 )	# NTFS_VOLUME_NOT_NTFS
-				# ignore - not formatted with NTFS
-				;;
-			16 )    # NTFS_VOLUME_UNCLEAN_UNMOUNT
-				# need option -o force, but we don't do this, because its not safe!
-				eventadd 144 $MNTNAME
-				# no fail event later on
-				FAIL_EVENT=0
-				;;
-			* )
-				# NTFS mount error
-				eventadd 145 $MNTNAME $NTFS_RET
-				# no fail event later on
-				FAIL_EVENT=0
-				;;
-			esac
-		fi
 	fi
 	if [ $new_filesystem = true ] ; then
 		umask $OLD_UMASK
@@ -161,6 +143,7 @@
 	else
 		umask $OLD_UMASK
 		rm -rf $MNTPATH
+		eventadd 142 $MNTNAME NOTMOUNTABLE
 	fi
 
 	if grep $MNTPATH /proc/mounts > /dev/null; then
