--- Makefile.in
+++ Makefile.in
@@ -463,7 +463,7 @@
 LIBGMPXX_LT_CURRENT = 8
 LIBGMPXX_LT_REVISION = 0
 LIBGMPXX_LT_AGE = 4
-SUBDIRS = tests mpn mpz mpq mpf printf scanf rand cxx demos tune doc
+SUBDIRS =       mpn mpz mpq mpf printf scanf rand cxx       tune
 
 # The "test -f" support for srcdir!=builddir is similar to the automake .c.o
 # etc rules, but with each foo.c explicitly, since $< is not portable
