--- configure.orig	2009-07-15 23:02:50.000000000 +0200
+++ configure	2009-07-15 23:03:01.000000000 +0200
@@ -36085,15 +36085,15 @@
 png_write_image()
 ; return 0; }
 EOF
-if { (eval echo configure:36089: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
+#if { (eval echo configure:36089: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   rm -rf conftest*
   eval "ac_cv_lib_$ac_lib_var=yes"
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -rf conftest*
-  eval "ac_cv_lib_$ac_lib_var=no"
-fi
+#else
+#  echo "configure: failed program was:" >&5
+#  cat conftest.$ac_ext >&5
+#  rm -rf conftest*
+#  eval "ac_cv_lib_$ac_lib_var=no"
+#fi
 rm -f conftest*
 LIBS="$ac_save_LIBS"
 
