--- configure	2007-02-12 02:10:24.000000000 +0100
+++ configure	2007-02-12 02:06:57.000000000 +0100
@@ -80175,7 +80175,11 @@
 echo "configure:80176: checking for working ttyname_r() implementation" >&5
   if test "$cross_compiling" = yes; then
   
-    echo "$ac_t""no, cannot detect working ttyname_r() when cross compiling. posix_ttyname() will be thread-unsafe" 1>&6
+    #evaluated by running test program on target platform                                                                          .
+    echo "$ac_t""yes" 1>&6                                                                                                         .
+    cat >> confdefs.h <<\EOF                                                                                                       .
+#define HAVE_TTYNAME_R 1                                                                                                           .
+EOF
   
 else
   cat > conftest.$ac_ext <<EOF
