--- configure.orig	2009-06-17 14:35:22.000000000 +0200
+++ configure	2009-06-20 11:55:21.000000000 +0200
@@ -82221,7 +82221,11 @@
 echo "configure:82222: checking for working ttyname_r() implementation" >&5
   if test "$cross_compiling" = yes; then
   
-    echo "$ac_t""no, cannot detect working ttyname_r() when cross compiling. posix_ttyname() will be thread-unsafe" 1>&6
+    #evaluated by running test program on target platform
+    echo "$ac_t""yes" 1>&6
+    cat >> confdefs.h <<EOF
+#define HAVE_TTYNAME_R 1
+EOF
   
 else
   cat > conftest.$ac_ext <<EOF
