--- configure.orig	2010-02-20 17:05:18.000000000 +0000
+++ configure	2010-02-20 17:05:19.000000000 +0000
@@ -82221,7 +82221,11 @@
 echo "configure:82222: checking for working ttyname_r() implementation" >&5
   if test "$cross_compiling" = yes; then
   
-    echo "$ac_t""no, cannot detect working ttyname_r() when cross compiling. posix_ttyname() will be thread-unsafe" 1>&6
+    #evaluated by running test program on target platform                                                                          .
+    echo "$ac_t""yes" 1>&6                                                                                                         .
+    cat >> confdefs.h <<\EOF                                                                                                       .
+#define HAVE_TTYNAME_R 1                                                                                                           .
+EOF
   
 else
   cat > conftest.$ac_ext <<EOF
