--- configure.orig	2008-12-05 18:48:06.000000000 +0100
+++ configure	2008-12-05 18:48:06.000000000 +0100
@@ -81951,7 +81951,11 @@
 echo "configure:81952: checking for working ttyname_r() implementation" >&5
   if test "$cross_compiling" = yes; then
   
-    echo "$ac_t""no, cannot detect working ttyname_r() when cross compiling. posix_ttyname() will be thread-unsafe" 1>&6
+    #evaluated by running test program on target platform                                                                          .
+    echo "$ac_t""yes" 1>&6                                                                                                         .
+    cat >> confdefs.h <<\EOF                                                                                                       .
+#define HAVE_TTYNAME_R 1                                                                                                           .
+EOF
   
 else
   cat > conftest.$ac_ext <<EOF
