--- configure
+++ configure
@@ -1362,9 +1362,9 @@
	cd "$srcdir" && test -r "./$ac_unique_file" || as_fn_error $? "$ac_msg"
	pwd)`
 # When building in place, set srcdir=.
-if test "$ac_abs_confdir" = "$ac_pwd"; then
-  srcdir=.
-fi
+#if test "$ac_abs_confdir" = "$ac_pwd"; then
+#  srcdir=.
+#fi
 # Remove unnecessary trailing slashes from srcdir.
 # Double slashes in file names in object file debugging info
 # mess up M-x gdb in Emacs.
