--- extensions/libxt_multiport.c
+++ extensions/libxt_multiport.c
@@ -474,13 +474,13 @@
 		.name          = "multiport",
 		.revision      = 0,
 		.version       = XTABLES_VERSION,
-		.size          = XT_ALIGN(sizeof(struct xt_multiport)),
-		.userspacesize = XT_ALIGN(sizeof(struct xt_multiport)),
-		.help          = multiport_help,
-		.x6_parse      = multiport_parse,
+		.size          = XT_ALIGN(sizeof(struct xt_multiport_v1)),
+		.userspacesize = XT_ALIGN(sizeof(struct xt_multiport_v1)),
+		.help          = multiport_help_v1,
+		.x6_parse      = multiport_parse_v1,
 		.x6_fcheck     = multiport_check,
-		.print         = multiport_print,
-		.save          = multiport_save,
+		.print         = multiport_print_v1,
+		.save          = multiport_save_v1,
 		.x6_options    = multiport_opts,
 	},
 	{
