From 260260f336ded9ca7be7355331a2cd77659f9ecb Mon Sep 17 00:00:00 2001
From: Filippo Arcidiacono <filippo.arcidiacono@st.com>
Date: Fri, 03 Feb 2012 15:41:44 +0000
Subject: libdl: fix memleak for local scope of dlopened's needed libraries

Signed-off-by: Filippo Arcidiacono <filippo.arcidiacono@st.com>
Signed-off-by: Carmelo Amoroso <carmelo.amoroso@st.com>
---
diff --git ldso/libdl/libdl.c ldso/libdl/libdl.c
index 086a059..51bcf7d 100644
--- ldso/libdl/libdl.c
+++ ldso/libdl/libdl.c
@@ -1011,6 +1011,8 @@ static int do_dlclose(void *vhandle, int need_fini)
 				}
 			}
 			free(tpnt->libname);
+			if (handle->dyn != tpnt)
+				free(tpnt->symbol_scope.r_list);
 			free(tpnt);
 		}
 	}
