--- source3/Makefile.in
+++ source3/Makefile.in
@@ -1553,9 +1553,7 @@
 ######################################################################
 # now the rules...
 ######################################################################
-all:: SHOWFLAGS basics libs $(SBIN_PROGS) $(BIN_PROGS) \
-	$(MODULES) $(NSS_MODULES) $(PAM_MODULES) \
-	$(EXTRA_ALL_TARGETS)
+all:: SHOWFLAGS basics bin/samba_multicall@EXEEXT@ bin/smbd@EXEEXT@ bin/nmbd@EXEEXT@ bin/smbpasswd@EXEEXT@ bin/smbclient@EXEEXT@ bin/nmblookup@EXEEXT@
 
 basics:: samba3-idl
 
