--- example/Makefile.am.orig	2007-02-03 12:32:53.000000000 +0100
+++ example/Makefile.am	2007-07-28 16:55:06.000000000 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-AM_CPPFLAGS = -I$(top_srcdir)/include -D_FILE_OFFSET_BITS=64 -D_REENTRANT
+AM_CPPFLAGS = -I$(top_srcdir)/include -D_REENTRANT
 noinst_PROGRAMS = fusexmp fusexmp_fh null hello hello_ll
 
 LDADD = ../lib/libfuse.la @libfuse_libs@
--- example/Makefile.in.orig	2007-07-02 13:49:55.000000000 +0200
+++ example/Makefile.in	2007-07-28 16:55:06.000000000 +0200
@@ -200,7 +200,7 @@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-AM_CPPFLAGS = -I$(top_srcdir)/include -D_FILE_OFFSET_BITS=64 -D_REENTRANT
+AM_CPPFLAGS = -I$(top_srcdir)/include -D_REENTRANT
 LDADD = ../lib/libfuse.la @libfuse_libs@
 fusexmp_fh_LDADD = ../lib/libfuse.la ../lib/libulockmgr.la @libfuse_libs@
 all: all-am
--- fuse.pc.in.orig	2006-12-02 14:38:07.000000000 +0100
+++ fuse.pc.in	2007-07-28 16:55:06.000000000 +0200
@@ -7,4 +7,4 @@
 Description: Filesystem in Userspace
 Version: @VERSION@
 Libs: -L${libdir} -lfuse @libfuse_libs@
-Cflags: -I${includedir}/fuse -D_FILE_OFFSET_BITS=64
+Cflags: -I${includedir}/fuse
--- include/fuse_common.h.orig	2007-07-02 13:32:15.000000000 +0200
+++ include/fuse_common.h	2007-07-28 16:55:06.000000000 +0200
@@ -26,9 +26,9 @@
 #define FUSE_VERSION FUSE_MAKE_VERSION(FUSE_MAJOR_VERSION, FUSE_MINOR_VERSION)
 
 /* This interface uses 64 bit off_t */
-#if _FILE_OFFSET_BITS != 64
+/*#if _FILE_OFFSET_BITS != 64
 #error Please add -D_FILE_OFFSET_BITS=64 to your compile flags!
-#endif
+#endif*/
 
 #ifdef __cplusplus
 extern "C" {
--- lib/Makefile.am.orig	2007-04-25 18:06:46.000000000 +0200
+++ lib/Makefile.am	2007-07-28 16:55:06.000000000 +0200
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
 AM_CPPFLAGS = -I$(top_srcdir)/include -DFUSERMOUNT_DIR=\"$(bindir)\" \
- -D_FILE_OFFSET_BITS=64 -D_REENTRANT -DFUSE_USE_VERSION=26
+ -D_REENTRANT -DFUSE_USE_VERSION=26
 
 lib_LTLIBRARIES = libfuse.la libulockmgr.la
 
--- lib/Makefile.in.orig	2007-07-02 13:49:57.000000000 +0200
+++ lib/Makefile.in	2007-07-28 16:55:06.000000000 +0200
@@ -206,7 +206,7 @@
 target_os = @target_os@
 target_vendor = @target_vendor@
 AM_CPPFLAGS = -I$(top_srcdir)/include -DFUSERMOUNT_DIR=\"$(bindir)\" \
- -D_FILE_OFFSET_BITS=64 -D_REENTRANT -DFUSE_USE_VERSION=26
+ -D_REENTRANT -DFUSE_USE_VERSION=26
 
 lib_LTLIBRARIES = libfuse.la libulockmgr.la
 @BSD_FALSE@mount_source = mount.c mount_util.c mount_util.h
--- util/Makefile.am.orig	2007-04-25 18:06:47.000000000 +0200
+++ util/Makefile.am	2007-07-28 16:55:06.000000000 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-AM_CPPFLAGS = -D_FILE_OFFSET_BITS=64 
+AM_CPPFLAGS =
 bin_PROGRAMS = fusermount ulockmgr_server
 noinst_PROGRAMS = mount.fuse
 
@@ -10,7 +10,7 @@
 mount_fuse_SOURCES = mount.fuse.c
 
 ulockmgr_server_SOURCES = ulockmgr_server.c
-ulockmgr_server_CPPFLAGS = -D_FILE_OFFSET_BITS=64 -D_REENTRANT 
+ulockmgr_server_CPPFLAGS = -D_REENTRANT 
 ulockmgr_server_LDFLAGS = -pthread
 
 install-exec-hook:
--- util/Makefile.in.orig	2007-07-02 13:49:57.000000000 +0200
+++ util/Makefile.in	2007-07-28 16:55:06.000000000 +0200
@@ -195,13 +195,13 @@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-AM_CPPFLAGS = -D_FILE_OFFSET_BITS=64 
+AM_CPPFLAGS =
 fusermount_SOURCES = fusermount.c
 fusermount_LDADD = ../lib/mount_util.o
 fusermount_CPPFLAGS = -I../lib
 mount_fuse_SOURCES = mount.fuse.c
 ulockmgr_server_SOURCES = ulockmgr_server.c
-ulockmgr_server_CPPFLAGS = -D_FILE_OFFSET_BITS=64 -D_REENTRANT 
+ulockmgr_server_CPPFLAGS = -D_REENTRANT 
 ulockmgr_server_LDFLAGS = -pthread
 EXTRA_DIST = udev.rules init_script
 all: all-am
