--- kernel/Makefile.in.orig	2007-02-23 12:17:45.000000000 +0100
+++ kernel/Makefile.in	2007-07-28 16:55:07.000000000 +0200
@@ -25,11 +25,9 @@
 install-y: all
 	$(mkdir_p) $(DESTDIR)$(fusemoduledir)
 	$(INSTALL) -m 644 $(fusemodule) $(DESTDIR)$(fusemoduledir)/$(fusemodule)
-	-/sbin/depmod -a
 
 uninstall-y:
 	rm -f $(DESTDIR)$(fusemoduledir)/$(fusemodule)
-	-/sbin/depmod -a
 
 clean:
 	-rm -f $(fusemodule) *.o .*.cmd *.mod.c *.ko *.s */*.o
