--- lib/fuse.c.orig	2007-07-02 13:33:08.000000000 +0200
+++ lib/fuse.c	2007-07-29 03:49:35.000000000 +0200
@@ -3289,12 +3289,14 @@
                                   sizeof(struct fuse_operations_compat1), 11);
 }
 
+/*
 __asm__(".symver fuse_exited,__fuse_exited@");
 __asm__(".symver fuse_process_cmd,__fuse_process_cmd@");
 __asm__(".symver fuse_read_cmd,__fuse_read_cmd@");
 __asm__(".symver fuse_set_getcontext_func,__fuse_set_getcontext_func@");
 __asm__(".symver fuse_new_compat2,fuse_new@");
 __asm__(".symver fuse_new_compat22,fuse_new@FUSE_2.2");
+*/
 
 #endif /* __FreeBSD__ */
 
@@ -3305,5 +3307,6 @@
     return fuse_new_common_compat25(fd, args, (struct fuse_operations *) op,
                                     op_size, 25);
 }
-
+/*
 __asm__(".symver fuse_new_compat25,fuse_new@FUSE_2.5");
+*/
--- lib/fuse_lowlevel.c.orig	2007-07-02 13:32:17.000000000 +0200
+++ lib/fuse_lowlevel.c	2007-07-29 03:49:35.000000000 +0200
@@ -1406,11 +1406,11 @@
     }
     return 0;
 }
-
+/*
 __asm__(".symver fuse_reply_statfs_compat,fuse_reply_statfs@FUSE_2.4");
 __asm__(".symver fuse_reply_open_compat,fuse_reply_open@FUSE_2.4");
 __asm__(".symver fuse_lowlevel_new_compat,fuse_lowlevel_new@FUSE_2.4");
-
+*/
 #else /* __FreeBSD__ */
 
 int fuse_sync_compat_args(struct fuse_args *args)
@@ -1432,5 +1432,6 @@
                                     (const struct fuse_lowlevel_ops *) op,
                                     op_size, userdata);
 }
-
+/*
 __asm__(".symver fuse_lowlevel_new_compat25,fuse_lowlevel_new@FUSE_2.5");
+*/
--- lib/fuse_mt.c.orig	2007-04-25 18:14:26.000000000 +0200
+++ lib/fuse_mt.c	2007-07-29 03:49:35.000000000 +0200
@@ -111,5 +111,6 @@
 
     return fuse_session_loop_mt(fuse_get_session(f));
 }
-
+/*
 __asm__(".symver fuse_loop_mt_proc,__fuse_loop_mt@");
+*/
--- lib/helper.c.orig	2007-07-02 13:32:17.000000000 +0200
+++ lib/helper.c	2007-07-29 03:49:35.000000000 +0200
@@ -389,13 +389,13 @@
     (void) args;
     return fuse_mount_compat22(mountpoint, NULL);
 }
-
+/*
 __asm__(".symver fuse_setup_compat2,__fuse_setup@");
 __asm__(".symver fuse_setup_compat22,fuse_setup@FUSE_2.2");
 __asm__(".symver fuse_teardown,__fuse_teardown@");
 __asm__(".symver fuse_main_compat2,fuse_main@");
 __asm__(".symver fuse_main_real_compat22,fuse_main_real@FUSE_2.2");
-
+*/
 #endif /* __FreeBSD__ */
 
 
@@ -426,8 +426,9 @@
 {
     return fuse_kern_mount(mountpoint, args);
 }
-
+/*
 __asm__(".symver fuse_setup_compat25,fuse_setup@FUSE_2.5");
 __asm__(".symver fuse_teardown_compat22,fuse_teardown@FUSE_2.2");
 __asm__(".symver fuse_main_real_compat25,fuse_main_real@FUSE_2.5");
 __asm__(".symver fuse_mount_compat25,fuse_mount@FUSE_2.5");
+*/
--- lib/mount_bsd.c.orig	2006-05-03 16:46:04.000000000 +0200
+++ lib/mount_bsd.c	2007-07-29 03:49:35.000000000 +0200
@@ -359,5 +359,6 @@
     free(mo.kernel_opts);
     return res;
 }
-
+/*
 __asm__(".symver fuse_unmount_compat22,fuse_unmount@FUSE_2.2");
+*/
--- lib/mount.c.orig	2007-07-02 13:32:17.000000000 +0200
+++ lib/mount.c	2007-07-29 03:49:35.000000000 +0200
@@ -578,6 +578,7 @@
     free(mo.mtab_opts);
     return res;
 }
-
+/*
 __asm__(".symver fuse_mount_compat22,fuse_mount@FUSE_2.2");
 __asm__(".symver fuse_unmount_compat22,fuse_unmount@FUSE_2.2");
+*/
