diff -burN fuse-2.6.3/lib/fuse.c fuse-2.6.3/lib/fuse.c
--- lib/fuse.c	2007-02-02 12:57:48.000000000 +0100
+++ lib/fuse.c	2007-03-04 22:27:19.000000000 +0100
@@ -3289,12 +3289,14 @@
                                   sizeof(struct fuse_operations_compat1), 11);
 }
 
+/*
 __asm__(".symver fuse_exited,__fuse_exited@");
 __asm__(".symver fuse_process_cmd,__fuse_process_cmd@");
 __asm__(".symver fuse_read_cmd,__fuse_read_cmd@");
 __asm__(".symver fuse_set_getcontext_func,__fuse_set_getcontext_func@");
 __asm__(".symver fuse_new_compat2,fuse_new@");
 __asm__(".symver fuse_new_compat22,fuse_new@FUSE_2.2");
+*/
 
 #endif /* __FreeBSD__ */
 
@@ -3305,5 +3307,6 @@
     return fuse_new_common_compat25(fd, args, (struct fuse_operations *) op,
                                     op_size, 25);
 }
-
+/*
 __asm__(".symver fuse_new_compat25,fuse_new@FUSE_2.5");
+*/
diff -burN fuse-2.6.3/lib/fuse_lowlevel.c fuse-2.6.3/lib/fuse_lowlevel.c
--- lib/fuse_lowlevel.c	2006-12-09 19:52:22.000000000 +0100
+++ lib/fuse_lowlevel.c	2007-03-04 22:26:41.000000000 +0100
@@ -1394,11 +1394,11 @@
     }
     return 0;
 }
-
+/*
 __asm__(".symver fuse_reply_statfs_compat,fuse_reply_statfs@FUSE_2.4");
 __asm__(".symver fuse_reply_open_compat,fuse_reply_open@FUSE_2.4");
 __asm__(".symver fuse_lowlevel_new_compat,fuse_lowlevel_new@FUSE_2.4");
-
+*/
 #else /* __FreeBSD__ */
 
 int fuse_sync_compat_args(struct fuse_args *args)
@@ -1420,5 +1420,6 @@
                                     (const struct fuse_lowlevel_ops *) op,
                                     op_size, userdata);
 }
-
+/*
 __asm__(".symver fuse_lowlevel_new_compat25,fuse_lowlevel_new@FUSE_2.5");
+*/
diff -burN fuse-2.6.3/lib/fuse_mt.c fuse-2.6.3/lib/fuse_mt.c
--- lib/fuse_mt.c	2006-12-09 19:51:14.000000000 +0100
+++ lib/fuse_mt.c	2007-03-04 22:25:20.000000000 +0100
@@ -119,5 +119,6 @@
 
     return fuse_session_loop_mt(fuse_get_session(f));
 }
-
+/*
 __asm__(".symver fuse_loop_mt_proc,__fuse_loop_mt@");
+*/
diff -burN fuse-2.6.3/lib/helper.c fuse-2.6.3/lib/helper.c
--- lib/helper.c	2006-12-09 19:52:22.000000000 +0100
+++ lib/helper.c	2007-03-04 22:27:00.000000000 +0100
@@ -389,13 +389,13 @@
     (void) args;
     return fuse_mount_compat22(mountpoint, NULL);
 }
-
+/*
 __asm__(".symver fuse_setup_compat2,__fuse_setup@");
 __asm__(".symver fuse_setup_compat22,fuse_setup@FUSE_2.2");
 __asm__(".symver fuse_teardown,__fuse_teardown@");
 __asm__(".symver fuse_main_compat2,fuse_main@");
 __asm__(".symver fuse_main_real_compat22,fuse_main_real@FUSE_2.2");
-
+*/
 #endif /* __FreeBSD__ */
 
 
@@ -426,8 +426,9 @@
 {
     return fuse_kern_mount(mountpoint, args);
 }
-
+/*
 __asm__(".symver fuse_setup_compat25,fuse_setup@FUSE_2.5");
 __asm__(".symver fuse_teardown_compat22,fuse_teardown@FUSE_2.2");
 __asm__(".symver fuse_main_real_compat25,fuse_main_real@FUSE_2.5");
 __asm__(".symver fuse_mount_compat25,fuse_mount@FUSE_2.5");
+*/
diff -burN fuse-2.6.3/lib/mount_bsd.c fuse-2.6.3/lib/mount_bsd.c
--- lib/mount_bsd.c	2006-12-09 19:51:14.000000000 +0100
+++ lib/mount_bsd.c	2007-03-04 22:26:51.000000000 +0100
@@ -359,5 +359,6 @@
     free(mo.kernel_opts);
     return res;
 }
-
+/*
 __asm__(".symver fuse_unmount_compat22,fuse_unmount@FUSE_2.2");
+*/
diff -burN fuse-2.6.3/lib/mount.c fuse-2.6.3/lib/mount.c
--- lib/mount.c	2007-01-02 18:10:33.000000000 +0100
+++ lib/mount.c	2007-03-04 22:26:47.000000000 +0100
@@ -303,6 +303,7 @@
     free(mo.kernel_opts);
     return res;
 }
-
+/*
 __asm__(".symver fuse_mount_compat22,fuse_mount@FUSE_2.2");
 __asm__(".symver fuse_unmount_compat22,fuse_unmount@FUSE_2.2");
+*/
