--- linux-2.6.28/arch/mips/kernel/module.c	2013-01-11 00:44:28.192253641 +0100
+++ linux-2.6.28/arch/mips/kernel/module.c	2013-01-11 00:46:42.888633185 +0100
@@ -55,7 +55,7 @@
     unsigned long addr;
     char name[64];
     enum _module_alloc_type_ type;
-} module_alloc_size_list[50];
+} module_alloc_size_list[99];
 
 unsigned long module_alloc_size_list_base;
 unsigned int module_alloc_size_list_size;
