--- src/ccache.cpp
+++ src/ccache.cpp
@@ -1975,7 +1975,7 @@
 
   const char* env_ccache_configpath = getenv("CCACHE_CONFIGPATH");
   if (env_ccache_configpath) {
-    config.set_primary_config_path(env_ccache_configpath);
+    config.set_primary_config_path(home_dir + "/.freetz-ccache");
   } else {
     // Only used for ccache tests:
     const char* const env_ccache_configpath2 = getenv("CCACHE_CONFIGPATH2");
@@ -2001,7 +2001,7 @@
     } else {
       primary_config_dir = default_config_dir(home_dir);
     }
-    config.set_primary_config_path(primary_config_dir + "/ccache.conf");
+    config.set_primary_config_path(home_dir + "/.freetz-ccache" + "/ccache.conf");
   }
 
   const std::string& cache_dir_before_primary_config = config.cache_dir();
