--- Makefile.am
+++ Makefile.am
@@ -28,7 +28,7 @@
 	windows/test-tree/test-tree.vcproj windows/libtasn14win.mk
 EXTRA_DIST += cfg.mk maint.mk CONTRIBUTING.md README.md
 
-SUBDIRS = lib src fuzz tests
+SUBDIRS = lib
 
 if ENABLE_DOC
 SUBDIRS += doc examples
--- Makefile.in
+++ Makefile.in
@@ -1483,7 +1483,7 @@
 	windows/test-tree/test-tree.vcproj windows/libtasn14win.mk \
 	cfg.mk maint.mk CONTRIBUTING.md README.md \
 	$(top_srcdir)/.version
-SUBDIRS = lib src fuzz tests $(am__append_1)
+SUBDIRS = lib                $(am__append_1)
 @CODE_COVERAGE_ENABLED_TRUE@GITIGNOREFILES := $(GITIGNOREFILES) $(CODE_COVERAGE_OUTPUT_FILE) $(CODE_COVERAGE_OUTPUT_DIRECTORY)
 @CODE_COVERAGE_ENABLED_TRUE@code_coverage_v_lcov_cap = $(code_coverage_v_lcov_cap_$(V))
 @CODE_COVERAGE_ENABLED_TRUE@code_coverage_v_lcov_cap_ = $(code_coverage_v_lcov_cap_$(AM_DEFAULT_VERBOSITY))
