--- Makefile.am
+++ Makefile.am
@@ -26,7 +26,7 @@
 	windows/test-parser/test-parser.vcproj				\
 	windows/test-tree/test-tree.vcproj windows/libtasn14win.mk
 
-SUBDIRS = gl lib src examples tests
+SUBDIRS = gl lib
 
 if ENABLE_DOC
 SUBDIRS += doc
--- Makefile.in
+++ Makefile.in
@@ -934,7 +934,7 @@
 	windows/test-parser/test-parser.vcproj \
 	windows/test-tree/test-tree.vcproj windows/libtasn14win.mk \
 	cfg.mk maint.mk .clcopying
-SUBDIRS = gl lib src examples tests $(am__append_1)
+SUBDIRS = gl lib                    $(am__append_1)
 ACLOCAL_AMFLAGS = -I m4 -I gl/m4 -I lib/glm4
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
