--- Makefile.am
+++ Makefile.am
@@ -26,7 +26,7 @@
 	windows/test-parser/test-parser.vcproj				\
 	windows/test-tree/test-tree.vcproj windows/libtasn14win.mk
 
-SUBDIRS = gl lib src examples tests doc
+SUBDIRS = gl lib
 
 ACLOCAL_AMFLAGS = -I m4 -I gl/m4 -I lib/glm4
 
--- Makefile.in
+++ Makefile.in
@@ -916,7 +916,7 @@
 	windows/test-parser/test-parser.vcproj \
 	windows/test-tree/test-tree.vcproj windows/libtasn14win.mk \
 	cfg.mk maint.mk .clcopying
-SUBDIRS = gl lib src examples tests doc
+SUBDIRS = gl lib
 ACLOCAL_AMFLAGS = -I m4 -I gl/m4 -I lib/glm4
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
