--- aconfigure
+++ aconfigure
@@ -5657,7 +5657,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking ioqueue backend" >&5
 $as_echo_n "checking ioqueue backend... " >&6; }
 # Check whether --enable-epoll was given.
-if test "${enable_epoll+set}" = set; then :
+if test "${enable_epoll}" = yes; then :
   enableval=$enable_epoll;
 		ac_os_objs=ioqueue_epoll.o
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: /dev/epoll" >&5
@@ -6291,7 +6291,7 @@
 
 
 # Check whether --enable-libsamplerate was given.
-if test "${enable_libsamplerate+set}" = set; then :
+if test "${enable_libsamplerate}" = yes; then :
   enableval=$enable_libsamplerate;  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for src_new in -lsamplerate" >&5
 $as_echo_n "checking for src_new in -lsamplerate... " >&6; }
 if test "${ac_cv_lib_samplerate_src_new+set}" = set; then :
