--- configure
+++ configure
@@ -6090,6 +6090,8 @@
 
 config_appended_defs="$config_appended_defs${newline}LIBC_CONTAINS_LIBINTL=${LIBC_CONTAINS_LIBINTL}"
 
+NO_GETTEXT=YesPlease
+if false; then
 #
 # Define NO_GETTEXT if you don't want Git output to be translated.
 # A translated Git requires GNU libintl or another gettext implementation
@@ -6100,6 +6102,7 @@
 else $as_nop
   NO_GETTEXT=YesPlease
 fi
+fi
 
 
 config_appended_defs="$config_appended_defs${newline}NO_GETTEXT=${NO_GETTEXT}"
