--- configure
+++ configure
@@ -5978,6 +5978,8 @@
 
 config_appended_defs="$config_appended_defs${newline}LIBC_CONTAINS_LIBINTL=${LIBC_CONTAINS_LIBINTL}"
 
+NO_GETTEXT=YesPlease
+if false; then
 #
 # Define NO_GETTEXT if you don't want Git output to be translated.
 # A translated Git requires GNU libintl or another gettext implementation
@@ -5987,6 +5989,7 @@
 else
   NO_GETTEXT=YesPlease
 fi
+fi
 
 
 
