--- configure
+++ configure
@@ -3398,6 +3398,7 @@
 
 config_appended_defs="$config_appended_defs${newline}INLINE=${INLINE}"
 
+if false; then
 # which switch to pass runtime path to dynamic libraries to the linker
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking if linker supports -R" >&5
 $as_echo_n "checking if linker supports -R... " >&6; }
@@ -3509,6 +3510,7 @@
 fi
 
 config_appended_defs="$config_appended_defs${newline}CC_LD_DYNPATH=${CC_LD_DYNPATH}"
+fi
 #AC_PROG_INSTALL		# needs install-sh or install.sh in sources
 if test -n "$ac_tool_prefix"; then
   for ac_prog in gar ar
--- Makefile
+++ Makefile
@@ -1057,16 +1057,6 @@
 	PTHREAD_LIBS =
 endif
 
-ifndef CC_LD_DYNPATH
-	ifdef NO_R_TO_GCC_LINKER
-		# Some gcc does not accept and pass -R to the linker to specify
-		# the runtime dynamic library path.
-		CC_LD_DYNPATH = -Wl,-rpath,
-	else
-		CC_LD_DYNPATH = -R
-	endif
-endif
-
 ifdef NO_LIBGEN_H
 	COMPAT_CFLAGS += -DNO_LIBGEN_H
 	COMPAT_OBJS += compat/basename.o
@@ -1076,7 +1066,7 @@
 	BASIC_CFLAGS += -DUSE_LIBPCRE
 	ifdef LIBPCREDIR
 		BASIC_CFLAGS += -I$(LIBPCREDIR)/include
-		EXTLIBS += -L$(LIBPCREDIR)/$(lib) $(CC_LD_DYNPATH)$(LIBPCREDIR)/$(lib)
+		EXTLIBS += -L$(LIBPCREDIR)/$(lib)
 	endif
 	EXTLIBS += -lpcre
 endif
@@ -1090,7 +1080,7 @@
 	ifdef CURLDIR
 		# Try "-Wl,-rpath=$(CURLDIR)/$(lib)" in such a case.
 		BASIC_CFLAGS += -I$(CURLDIR)/include
-		CURL_LIBCURL = -L$(CURLDIR)/$(lib) $(CC_LD_DYNPATH)$(CURLDIR)/$(lib) -lcurl
+		CURL_LIBCURL = -L$(CURLDIR)/$(lib) -lcurl
 	else
 		CURL_LIBCURL = -lcurl
 	endif
@@ -1118,7 +1108,7 @@
 	ifndef NO_EXPAT
 		ifdef EXPATDIR
 			BASIC_CFLAGS += -I$(EXPATDIR)/include
-			EXPAT_LIBEXPAT = -L$(EXPATDIR)/$(lib) $(CC_LD_DYNPATH)$(EXPATDIR)/$(lib) -lexpat
+			EXPAT_LIBEXPAT = -L$(EXPATDIR)/$(lib) -lexpat
 		else
 			EXPAT_LIBEXPAT = -lexpat
 		endif
@@ -1130,7 +1120,7 @@
 
 ifdef ZLIB_PATH
 	BASIC_CFLAGS += -I$(ZLIB_PATH)/include
-	EXTLIBS += -L$(ZLIB_PATH)/$(lib) $(CC_LD_DYNPATH)$(ZLIB_PATH)/$(lib)
+	EXTLIBS += -L$(ZLIB_PATH)/$(lib)
 endif
 EXTLIBS += -lz
 
@@ -1138,7 +1128,7 @@
 	OPENSSL_LIBSSL = -lssl
 	ifdef OPENSSLDIR
 		BASIC_CFLAGS += -I$(OPENSSLDIR)/include
-		OPENSSL_LINK = -L$(OPENSSLDIR)/$(lib) $(CC_LD_DYNPATH)$(OPENSSLDIR)/$(lib)
+		OPENSSL_LINK = -L$(OPENSSLDIR)/$(lib)
 	else
 		OPENSSL_LINK =
 	endif
@@ -1162,7 +1152,7 @@
 ifdef NEEDS_LIBICONV
 	ifdef ICONVDIR
 		BASIC_CFLAGS += -I$(ICONVDIR)/include
-		ICONV_LINK = -L$(ICONVDIR)/$(lib) $(CC_LD_DYNPATH)$(ICONVDIR)/$(lib)
+		ICONV_LINK = -L$(ICONVDIR)/$(lib)
 	else
 		ICONV_LINK =
 	endif
