--- configure.ac
+++ configure.ac
@@ -198,6 +198,29 @@
 
 # Testing section
 
+internal_libbind="\${top_srcdir}/bind"
+libbind=
+AC_ARG_WITH(libbind,
+	AS_HELP_STRING([--with-libbind=PATH],[bind includes and libraries are in PATH
+		        (default is ./bind)]),
+	use_libbind="$withval", use_libbind="no")
+case "$use_libbind" in
+yes)
+	libbind="$internal_libbind"
+	;;
+no)
+	libbind="$internal_libbind"
+	;;
+*)
+	libbind="$use_libbind"
+	if test ! -d "bind"; then
+		AC_MSG_WARN(empty bind directory)
+	fi
+	;;
+esac
+AM_CONDITIONAL(USE_INTERNAL_LIBBIND, test "$libbind" = "$internal_libbind")
+AC_SUBST([libbind])
+
 atf_path="no"
 AC_ARG_WITH([atf],
     AS_HELP_STRING([--with-atf=PATH],[specify location where atf was installed (or "bind")]),
@@ -205,7 +228,7 @@
 AM_CONDITIONAL(BIND_ATF, test "$atf_path" = "bind")
 if test "$atf_path" = "bind" ; then
     atf_pcp="bind"
-    atf_path="\${top_srcdir}/bind/atf"
+    atf_path="$internal_libbind/atf"
     ATF_CFLAGS="-I$atf_path/include -DUNIT_TEST"
     ATF_LDFLAGS="-L$atf_path/lib -latf-c"
     ATF_BIN=`pwd`/bind/atf/bin
@@ -604,26 +627,6 @@
     [AC_DEFINE([VLAN_TCI_PRESENT], [1], [tpacket_auxdata.tp_vlan_tci present])]
     ,, [#include <linux/if_packet.h>])
 
-libbind=
-AC_ARG_WITH(libbind,
-	AS_HELP_STRING([--with-libbind=PATH],[bind includes and libraries are in PATH 
-		        (default is ./bind)]),
-	use_libbind="$withval", use_libbind="no")
-case "$use_libbind" in 
-yes)
-	libbind="\${top_srcdir}/bind"
-	;;
-no)
-	libbind="\${top_srcdir}/bind"
-	;;
-*)
-	libbind="$use_libbind"
-	if test ! -d "bind"; then
-		AC_MSG_WARN(empty bind directory)
-	fi	
-	;;
-esac
-
 # OpenLDAP support.
 AC_ARG_WITH(ldap,
     AS_HELP_STRING([--with-ldap],[enable OpenLDAP support in dhcpd (default is no)]),
--- client/Makefile.am
+++ client/Makefile.am
@@ -4,14 +4,18 @@
 # production code. Sadly, we are not there yet.
 SUBDIRS = . tests
 
+libbind = @libbind@
+
+AM_CPPFLAGS = -I$(libbind)/include
+
 dist_sysconf_DATA = dhclient.conf.example
 sbin_PROGRAMS = dhclient
 dhclient_SOURCES = clparse.c dhclient.c dhc6.c \
 		   scripts/bsdos scripts/freebsd scripts/linux scripts/macos \
 		   scripts/netbsd scripts/nextstep scripts/openbsd \
 		   scripts/solaris scripts/openwrt
-dhclient_LDADD = ../common/libdhcp.a ../omapip/libomapi.a ../bind/lib/libirs.a \
-		 ../bind/lib/libdns.a ../bind/lib/libisccfg.a ../bind/lib/libisc.a
+dhclient_LDADD = ../common/libdhcp.a ../omapip/libomapi.a $(libbind)/lib/libirs.a \
+		 $(libbind)/lib/libdns.a $(libbind)/lib/libisccfg.a $(libbind)/lib/libisc.a
 man_MANS = dhclient.8 dhclient-script.8 dhclient.conf.5 dhclient.leases.5
 EXTRA_DIST = $(man_MANS)
 
--- client/tests/Makefile.am
+++ client/tests/Makefile.am
@@ -1,7 +1,9 @@
 SUBDIRS = .
 
+libbind = @libbind@
+
 AM_CPPFLAGS = $(ATF_CFLAGS) -DUNIT_TEST -I$(top_srcdir)/includes
-AM_CPPFLAGS += -I$(top_srcdir)/bind/include -I$(top_srcdir)
+AM_CPPFLAGS += -I$(libbind)/include -I$(top_srcdir)
 AM_CPPFLAGS += -DLOCALSTATEDIR='"."'
 AM_CPPFLAGS += -DCLIENT_PATH='"."'
 
@@ -16,9 +18,9 @@
 DHCPSRC = ../clparse.c ../dhc6.c ../dhclient.c
 
 DHCPLIBS = $(top_builddir)/common/libdhcp.a $(top_builddir)/omapip/libomapi.a    \
-          $(top_builddir)/dhcpctl/libdhcpctl.a $(top_builddir)/bind/lib/libirs.a \
-	  $(top_builddir)/bind/lib/libdns.a $(top_builddir)/bind/lib/libisccfg.a \
-	  $(top_builddir)/bind/lib/libisc.a
+          $(top_builddir)/dhcpctl/libdhcpctl.a $(libbind)/lib/libirs.a \
+	  $(libbind)/lib/libdns.a $(libbind)/lib/libisccfg.a \
+	  $(libbind)/lib/libisc.a
 
 ATF_TESTS =
 if HAVE_ATF
--- common/Makefile.am
+++ common/Makefile.am
@@ -1,4 +1,6 @@
-AM_CPPFLAGS = -I.. -DLOCALSTATEDIR='"@localstatedir@"'
+libbind = @libbind@
+
+AM_CPPFLAGS = -I.. -I$(libbind)/include -DLOCALSTATEDIR='"@localstatedir@"'
 AM_CFLAGS = $(LDAP_CFLAGS)
 
 noinst_LIBRARIES = libdhcp.a
--- common/tests/Makefile.am
+++ common/tests/Makefile.am
@@ -1,5 +1,7 @@
 SUBDIRS = .
 
+libbind = @libbind@
+
 AM_CPPFLAGS = $(ATF_CFLAGS) -I$(top_srcdir)/includes
 
 EXTRA_DIST = Atffile
@@ -13,26 +15,26 @@
 alloc_unittest_SOURCES = test_alloc.c $(top_srcdir)/tests/t_api_dhcp.c
 alloc_unittest_LDADD = $(ATF_LDFLAGS)
 alloc_unittest_LDADD += ../libdhcp.a  \
-	../../omapip/libomapi.a ../../bind/lib/libirs.a \
-	../../bind/lib/libdns.a ../../bind/lib/libisccfg.a  ../../bind/lib/libisc.a
+	../../omapip/libomapi.a $(libbind)/lib/libirs.a \
+	$(libbind)/lib/libdns.a $(libbind)/lib/libisccfg.a  $(libbind)/lib/libisc.a
 
 dns_unittest_SOURCES = dns_unittest.c $(top_srcdir)/tests/t_api_dhcp.c
 dns_unittest_LDADD = $(ATF_LDFLAGS)
 dns_unittest_LDADD += ../libdhcp.a  \
-	../../omapip/libomapi.a ../../bind/lib/libirs.a \
-	../../bind/lib/libdns.a ../../bind/lib/libisccfg.a  ../../bind/lib/libisc.a
+	../../omapip/libomapi.a $(libbind)/lib/libirs.a \
+	$(libbind)/lib/libdns.a $(libbind)/lib/libisccfg.a  $(libbind)/lib/libisc.a
 
 misc_unittest_SOURCES = misc_unittest.c $(top_srcdir)/tests/t_api_dhcp.c
 misc_unittest_LDADD = $(ATF_LDFLAGS)
 misc_unittest_LDADD += ../libdhcp.a  \
-	../../omapip/libomapi.a ../../bind/lib/libirs.a \
-	../../bind/lib/libdns.a ../../bind/lib/libisccfg.a  ../../bind/lib/libisc.a
+	../../omapip/libomapi.a $(libbind)/lib/libirs.a \
+	$(libbind)/lib/libdns.a $(libbind)/lib/libisccfg.a  $(libbind)/lib/libisc.a
 
 ns_name_unittest_SOURCES = ns_name_test.c $(top_srcdir)/tests/t_api_dhcp.c
 ns_name_unittest_LDADD = $(ATF_LDFLAGS)
 ns_name_unittest_LDADD += ../libdhcp.a  \
-	../../omapip/libomapi.a ../../bind/lib/libirs.a \
-	../../bind/lib/libdns.a ../../bind/lib/libisccfg.a  ../../bind/lib/libisc.a
+	../../omapip/libomapi.a $(libbind)/lib/libirs.a \
+	$(libbind)/lib/libdns.a $(libbind)/lib/libisccfg.a  $(libbind)/lib/libisc.a
 
 check: $(ATF_TESTS)
 	sh ${top_srcdir}/tests/unittest.sh
--- dhcpctl/Makefile.am
+++ dhcpctl/Makefile.am
@@ -1,3 +1,7 @@
+libbind = @libbind@
+
+AM_CPPFLAGS = -I$(libbind)/include
+
 bin_PROGRAMS = omshell
 lib_LIBRARIES = libdhcpctl.a
 noinst_PROGRAMS = cltest
@@ -6,12 +10,12 @@
 
 omshell_SOURCES = omshell.c
 omshell_LDADD = libdhcpctl.a ../common/libdhcp.a ../omapip/libomapi.a \
-	        ../bind/lib/libirs.a ../bind/lib/libdns.a \
-	        ../bind/lib/libisccfg.a ../bind/lib/libisc.a
+	        $(libbind)/lib/libirs.a $(libbind)/lib/libdns.a \
+	        $(libbind)/lib/libisccfg.a $(libbind)/lib/libisc.a
 
 libdhcpctl_a_SOURCES = dhcpctl.c callback.c remote.c
 
 cltest_SOURCES = cltest.c
 cltest_LDADD = libdhcpctl.a ../common/libdhcp.a ../omapip/libomapi.a \
-	       ../bind/lib/libirs.a ../bind/lib/libdns.a \
-               ../bind/lib/libisccfg.a ../bind/lib/libisc.a
+	       $(libbind)/lib/libirs.a $(libbind)/lib/libdns.a \
+               $(libbind)/lib/libisccfg.a $(libbind)/lib/libisc.a
--- Makefile.am
+++ Makefile.am
@@ -25,7 +25,11 @@
 	     bind/Makefile.in bind/bind.tar.gz bind/version.tmp \
 	     common/tests/Atffile server/tests/Atffile
 
-SUBDIRS = bind includes tests common omapip client dhcpctl relay server
+SUBDIRS =
+if USE_INTERNAL_LIBBIND
+SUBDIRS += bind
+endif
+SUBDIRS += includes tests common omapip client dhcpctl relay server
 
 nobase_include_HEADERS = dhcpctl/dhcpctl.h
 
--- omapip/Makefile.am
+++ omapip/Makefile.am
@@ -1,3 +1,7 @@
+libbind = @libbind@
+
+AM_CPPFLAGS = -I$(libbind)/include
+
 lib_LIBRARIES = libomapi.a
 noinst_PROGRAMS = svtest
 
@@ -10,6 +14,6 @@
 EXTRA_DIST = $(man_MANS)
 
 svtest_SOURCES = test.c
-svtest_LDADD = libomapi.a ../bind/lib/libirs.a ../bind/lib/libdns.a \
-		../bind/lib/libisccfg.a ../bind/lib/libisc.a
+svtest_LDADD = libomapi.a $(libbind)/lib/libirs.a $(libbind)/lib/libdns.a \
+		$(libbind)/lib/libisccfg.a $(libbind)/lib/libisc.a
 
--- relay/Makefile.am
+++ relay/Makefile.am
@@ -1,10 +1,12 @@
-AM_CPPFLAGS = -DLOCALSTATEDIR='"@localstatedir@"'
+libbind = @libbind@
+
+AM_CPPFLAGS = -DLOCALSTATEDIR='"@localstatedir@"' -I$(libbind)/include
 
 sbin_PROGRAMS = dhcrelay
 dhcrelay_SOURCES = dhcrelay.c
 dhcrelay_LDADD = ../common/libdhcp.a ../omapip/libomapi.a \
-		 ../bind/lib/libirs.a ../bind/lib/libdns.a \
-		 ../bind/lib/libisccfg.a ../bind/lib/libisc.a
+		 $(libbind)/lib/libirs.a $(libbind)/lib/libdns.a \
+		 $(libbind)/lib/libisccfg.a $(libbind)/lib/libisc.a
 man_MANS = dhcrelay.8
 EXTRA_DIST = $(man_MANS)
 
--- server/Makefile.am
+++ server/Makefile.am
@@ -4,7 +4,9 @@
 # production code. Sadly, we are not there yet.
 SUBDIRS = . tests
 
-AM_CPPFLAGS = -I.. -DLOCALSTATEDIR='"@localstatedir@"'
+libbind = @libbind@
+
+AM_CPPFLAGS = -I.. -I$(libbind)/include -DLOCALSTATEDIR='"@localstatedir@"'
 
 dist_sysconf_DATA = dhcpd.conf.example
 sbin_PROGRAMS = dhcpd
@@ -14,8 +16,8 @@
 
 dhcpd_CFLAGS = $(LDAP_CFLAGS)
 dhcpd_LDADD = ../common/libdhcp.a ../omapip/libomapi.a \
-	      ../dhcpctl/libdhcpctl.a ../bind/lib/libirs.a \
-	      ../bind/lib/libdns.a ../bind/lib/libisccfg.a ../bind/lib/libisc.a \
+	      ../dhcpctl/libdhcpctl.a $(libbind)/lib/libirs.a \
+	      $(libbind)/lib/libdns.a $(libbind)/lib/libisccfg.a $(libbind)/lib/libisc.a \
 	      $(LDAP_LIBS)		
 
 man_MANS = dhcpd.8 dhcpd.conf.5 dhcpd.leases.5
--- server/tests/Makefile.am
+++ server/tests/Makefile.am
@@ -1,7 +1,9 @@
 SUBDIRS = .
 
+libbind = @libbind@
+
 AM_CPPFLAGS = $(ATF_CFLAGS) -DUNIT_TEST -I$(top_srcdir)/includes
-AM_CPPFLAGS += -I$(top_srcdir)/bind/include -I$(top_srcdir)
+AM_CPPFLAGS += -I$(libbind)/include -I$(top_srcdir)
 AM_CPPFLAGS += -DLOCALSTATEDIR='"."'
 
 EXTRA_DIST = Atffile
@@ -18,9 +20,9 @@
           ../ldap.c ../ldap_casa.c ../dhcpd.c ../leasechain.c
 
 DHCPLIBS = $(top_builddir)/common/libdhcp.a $(top_builddir)/omapip/libomapi.a    \
-          $(top_builddir)/dhcpctl/libdhcpctl.a $(top_builddir)/bind/lib/libirs.a \
-	  $(top_builddir)/bind/lib/libdns.a $(top_builddir)/bind/lib/libisccfg.a \
-	  $(top_builddir)/bind/lib/libisc.a
+          $(top_builddir)/dhcpctl/libdhcpctl.a $(libbind)/lib/libirs.a \
+	  $(libbind)/lib/libdns.a $(libbind)/lib/libisccfg.a \
+	  $(libbind)/lib/libisc.a
 
 ATF_TESTS =
 if HAVE_ATF
--- tests/Makefile.am
+++ tests/Makefile.am
@@ -1,3 +1,5 @@
+libbind = @libbind@
+
 EXTRA_DIST = failover/dhcp-1.cf failover/dhcp-2.cf failover/new-failover \
 	     DHCPv6/000-badmsgtype.pl \
 	     DHCPv6/010-solicit-noclientid.pl \
@@ -26,7 +28,7 @@
 	     HOWTO-unit-test \
 	     unit_test_sample.c
 
-AM_CPPFLAGS = -I..
+AM_CPPFLAGS = -I.. -I$(libbind)/include
 
 check_LIBRARIES = libt_api.a
 libt_api_a_SOURCES = t_api.c t_api_dhcp.c
