Kconfig entry for CONFIG_FUSIV_USB_OPTIMIZATION is missing

--- linux-2.6.28/fusiv_src/kernel/Kconfig.orig	2008-09-02 12:36:15.000000000 +0200
+++ linux-2.6.28/fusiv_src/kernel/Kconfig	2011-10-02 20:00:13.718988407 +0200
@@ -333,5 +333,12 @@
 
 endchoice
 
+config FUSIV_USB_OPTIMIZATION
+	depends on MIPS_FUSIV
+	bool "FUSIV USB optimization"
+	default y
+	help
+	FUSIV USB optimization
+
 endmenu
 
