--- plugins/Makefile	2006-08-03 04:02:01.000000000 +0200
+++ plugins/Makefile	2009-08-11 10:08:58.000000000 +0200
@@ -1,6 +1,6 @@
 CC	= gcc
 COPTS	= -O2 -g
-CFLAGS	= $(COPTS) -I.. -I../../include -fPIC
+CFLAGS	= $(COPTS) -I.. -I../../include $(PLUGINS_CFLAGS) -fPIC
 LDFLAGS	= -shared
 LDADD   = -lutil
 INSTALL	= install -o root
--- plugins/pptpd-logwtmp.c	2007-04-16 02:21:02.000000000 +0200
+++ plugins/pptpd-logwtmp.c	2009-08-11 10:01:40.000000000 +0200
@@ -12,7 +12,7 @@
 #include <unistd.h>
 #include <utmp.h>
 #include <string.h>
-#include "pppd.h"
+#include <pppd/pppd.h>
 
 char pppd_version[] = VERSION;
 
