--- libc/inet/resolv.c.orig	2007-02-22 00:27:09.000000000 +0100
+++ libc/inet/resolv.c	2007-07-29 05:16:13.000000000 +0200
@@ -1474,7 +1474,7 @@
 	buf+=sizeof(char **)*(ALIAS_DIM);
 	buflen-=sizeof(char **)*(ALIAS_DIM);
 
-	if (action!=GETHOSTENT) {
+	{
 #ifdef __UCLIBC_HAS_IPV6__
 		char *p=buf;
 		size_t len=buflen;
@@ -1514,10 +1514,12 @@
 		if (buflen < 80)
 			return ERANGE;
 
-		__open_etc_hosts(&fp);
-		if (fp == NULL) {
-			result=NULL;
-			return errno;
+		if (action!=GETHOSTENT) {
+			__open_etc_hosts(&fp);
+		    if (fp == NULL) {
+			    result=NULL;
+			    return errno;
+		    }
 		}
 	}
 
@@ -1546,7 +1548,6 @@
 
 		if (action==GETHOSTENT) {
 			/* Return whatever the next entry happens to be. */
-			break;
 		} else if (action==GET_HOSTS_BYADDR) {
 			if (strcmp(name, alias[0]) != 0)
 				continue;
@@ -1661,7 +1662,7 @@
 #endif /* __UCLIBC_HAS_IPV6__ */
 					sizeof(char *)*(ALIAS_DIM) +
 					80/*namebuffer*/ + 2/* margin */];
-    struct hostent *host;
+    struct hostent *host = 0;
 
     __UCLIBC_MUTEX_LOCK(mylock);
     gethostent_r(&h, buf, sizeof(buf), &host, &h_errno);
--- libc/sysdeps/linux/arm/syscall.c.orig	2005-08-14 15:29:05.000000000 +0200
+++ libc/sysdeps/linux/arm/syscall.c	2007-07-29 05:16:13.000000000 +0200
@@ -23,6 +23,18 @@
 #include <sys/types.h>
 #include <sys/syscall.h>
 
+#ifndef __NR_syscall			
+#ifndef __NR_SYSCALL_BASE	
+
+#if defined(__thumb__)
+#define __NR_SYSCALL_BASE	0
+#else
+#define __NR_SYSCALL_BASE	0x900000
+#endif
+#endif
+
+#define __NR_syscall			(__NR_SYSCALL_BASE+113)
+#endif
 
 long syscall(long sysnum, long a, long b, long c, long d, long e, long f)
 {
--- utils/Makefile.orig	2007-01-26 01:22:03.000000000 +0100
+++ utils/Makefile	2007-07-29 05:16:13.000000000 +0200
@@ -51,7 +51,8 @@
 readelf.c ldconfig.c ldd.c:	headers
 
 readelf: readelf.c
-	$(CC) $(CFLAGS) $^ -o $@
+#	$(CC) $(CFLAGS) $^  -L $(INSTALL_DIR)/lib -l dl -o $@
+	$(CC) $(CFLAGS) $^  -o $@
 	$(STRIPTOOL) -s -x -R .note -R .comment $@
 
 ifeq ($(strip $(UCLIBC_STATIC_LDCONFIG)),y)
@@ -67,6 +68,7 @@
 	$(STRIPTOOL) -s -x -R .note -R .comment $@
 
 ldd: ldd.c
+#	$(CC) $(CFLAGS) $(PIEFLAG) $(LDPIEFLAG) -L $(INSTALL_DIR)/lib -l dl
 	$(CC) $(CFLAGS) $(PIEFLAG) $(LDPIEFLAG) \
 		-DUCLIBC_RUNTIME_PREFIX=\"$(RUNTIME_PREFIX)\" \
 		-DUCLIBC_LDSO=$(UCLIBC_LDSO) -I. -I../ldso/include \
@@ -74,7 +76,7 @@
 	$(STRIPTOOL) -s -x -R .note -R .comment $@
 
 iconv: ../libc/misc/wchar/wchar.c
-	$(CC) $(CFLAGS) $(PIEFLAG) $(LDPIEFLAG) \
+	$(CC) $(CFLAGS) $(PIEFLAG) $(LDPIEFLAG) -L $(INSTALL_DIR)/lib -l dl \
 		-DL_iconv_main \
 		$^ -o $@
 	$(STRIPTOOL) -s -x -R .note -R .comment $@
