--- Configure
+++ Configure
@@ -2885,9 +2885,12 @@
 
     # Test for IPv6 support.
 
-    if test -r ${LSOF_INCLUDE}/netinet/ip6.h	# {
+    if test "X$LINUX_HASIPV6" = "XY" -o "X$LINUX_HASIPV6" = "Xy" # {
     then
-      LSOF_CFGF="$LSOF_CFGF -DHASIPv6"
+      if test -r ${LSOF_INCLUDE}/netinet/ip6.h	# {
+      then
+        LSOF_CFGF="$LSOF_CFGF -DHASIPv6"
+      fi	# }
     fi	# }
 
     # Test for <rpc/rpc.h>.
