--- usr/www/all/menus/menu_data.lua
+++ usr/www/all/menus/menu_data.lua
@@ -2174,6 +2174,20 @@
 ["lnk"] = pageData["liveTv"].lnk,
 ["pos"] = -65
 } or nil
+-- TODO: find out how to check if Freetz WOL files are avalaible
+local freetzWolInstalled = false
+if freetzWolInstalled then
+menuData["freetz-wol"] = {
+["txt"] = "Freetz WOL",
+["lnk"] = "/cgi-bin/freetz_wol",
+["pos"] = -55
+}
+end
+menuData["freetz"] = {
+["txt"] = "Freetz",
+["lnk"] = "/cgi-bin/freetz_status",
+["pos"] = -60
+}
 if config.NAS then
 menuData["nas"] = {
 ["txt"] = TXT([[{?757:400?}]]),
