--- etc/init.d/rc.net.orig	2008-08-09 21:12:03.000000000 +0200
+++ etc/init.d/rc.net	2008-08-09 21:12:03.000000000 +0200
@@ -68,6 +68,8 @@
 #   a) it has the parameter at all AND
 #   b) igdd binary does *not* exist
 [ "$_dsld_upnp" ] && [ ! "$_igdd" ] && DSLDDPARAM="-g"
+# Do we have a Samba server (smbd) or was ist stripped from the firmware?
+_smbd=$(basename $(which smbd) 2> /dev/null)
 
 
 igddstart()
@@ -303,6 +305,7 @@
 #we have the samba cfg in /var, so crate them there
 sambastart()
 {
+   [ "$_smbd" ] || return;
    cd /
    tar xf /var.tar
    /bin/smbd
