--- etc/init.d/rc.net.orig	2009-09-16 11:54:11.866914469 +0200
+++ etc/init.d/rc.net	2009-09-16 11:54:11.886921348 +0200
@@ -71,6 +71,8 @@
 #   a) it has the parameter at all AND
 #   b) upnpd binary does *not* exist
 [ "$_dsld_upnp" ] && [ ! "$_upnpd" ] && DSLDDPARAM="-g"
+# Do we have a Samba server (smbd) or was ist stripped from the firmware?
+_smbd=$(basename $(which smbd) 2> /dev/null)
 
 
 upnpdstart()
@@ -323,6 +325,7 @@
 #we have the samba cfg in /var, so crate them there
 sambastart()
 {
+   [ "$_smbd" ] || return;
    cd /
    tar xf /var.tar
    /bin/smbd
