--- src/mklibs
+++ src/mklibs
@@ -450,8 +450,8 @@
                     if debuglevel >= DEBUG_VERBOSE:
                         print("Adding rpath " + rpath_elem + " for " + obj)
                     lib_rpath.append(rpath_elem)
-        else:
-            print("warning: " + obj + " may need rpath, but --root not specified")
+#        else:
+#            print("warning: " + obj + " may need rpath, but --root not specified")
 
 lib_path.extend(lib_rpath)
 
