--- Makefile
+++ Makefile
@@ -287,10 +287,9 @@
   USE_DL          = implicit
 else
 ifeq ($(TARGET),linux26)
-  # This is for standard Linux 2.6 with netfilter and standard epoll()
+  # This is for standard Linux 2.6 with netfilter and without epoll()
   USE_NETFILTER   = implicit
   USE_POLL        = implicit
-  USE_EPOLL       = implicit
   USE_TPROXY      = implicit
   USE_LIBCRYPT    = implicit
   USE_FUTEX       = implicit
