--- Makefile.org
+++ Makefile.org
<<<<<<< HEAD:make/openssl/patches/1.0.2j/140-makefile-dirs.openssl.patch
@@ -136,8 +136,8 @@
=======
@@ -137,7 +137,7 @@
>>>>>>> cae5c75f6c2d5c1c45d1446e5fa414fa8e1caf30:make/openssl/patches/1.0.2j/140-makefile-dirs.openssl.patch
 
 BASEADDR=
 
-DIRS=   crypto ssl engines apps test tools
-ENGDIRS= ccgost
+DIRS=   crypto ssl         apps
+ENGDIRS= 
 SHLIBDIRS= crypto ssl
 
 # dirs in crypto to build
