--- SConstruct
+++ SConstruct
@@ -220,8 +220,8 @@
 else:
   LIBNAMESTATIC = 'serf-${MAJOR}'
 
-env.Append(RPATH=libdir,
-           PDB='${TARGET.filebase}.pdb')
+#env.Append(RPATH=libdir,
+#           PDB='${TARGET.filebase}.pdb')
 
 if sys.platform == 'darwin':
 #  linkflags.append('-Wl,-install_name,@executable_path/%s.dylib' % (LIBNAME,))
@@ -370,8 +370,8 @@
 # On some systems, the -R values that APR describes never make it into actual
 # RPATH flags. We'll manually map all directories in LIBPATH into new
 # flags to set RPATH values.
-for d in env['LIBPATH']:
-  env.Append(RPATH=':'+d)
+#for d in env['LIBPATH']:
+#  env.Append(RPATH=':'+d)
 
 # Set up the construction of serf-*.pc
 pkgconfig = env.Textfile('serf-%d.pc' % (MAJOR,),
