--- linux-2.4.17_mvl21.orig/drivers/atm/Makefile	2006-02-05 18:17:10.906017208 +0100
+++ linux-2.4.17_mvl21/drivers/atm/Makefile	2006-02-05 18:22:32.408141456 +0100
@@ -28,7 +28,7 @@
 ##########################################################################################
 ifeq ($(CONFIG_MIPS_SANGAM),y)
 
-subdir-$(CONFIG_MIPS_SANGAM_ATM) += sangam_atm
+#subdir-$(CONFIG_MIPS_SANGAM_ATM) += sangam_atm
 
 EXTRA_CFLAGS += -DEL -I$(TOPDIR)/drivers/atm/sangam_atm -DPOST_SILICON -DCOMMON_NSP -DNO_ACT -DNO_ADV_STATS  
 #EXTRA_CFLAGS += -DAVM_2MB -DDEBUGATM -DDEBUG_BUILD #-DAVM_CLEAR_EOC -DAVM_USE_BITFIELD -DAVM_DSL_FEATURE_CONTROL
@@ -68,7 +68,7 @@
 tiatm-objs  += dsl_hal_export.o
 endif
 
-obj-$(CONFIG_MIPS_SANGAM_ATM) += tiatm.o
+#obj-$(CONFIG_MIPS_SANGAM_ATM) += tiatm.o
 
 endif
 ##########################################################################################
--- linux-2.4.17_mvl21.orig/drivers/char/Makefile	2006-02-05 18:17:12.946706976 +0100
+++ linux-2.4.17_mvl21/drivers/char/Makefile	2006-02-05 18:27:53.661303552 +0100
@@ -213,8 +213,8 @@
 endif
 
 ifeq ($(CONFIG_UBIK2),m)
-obj-$(CONFIG_UBIK2) += ubik2/ubik2.o
-subdir-$(CONFIG_UBIK2) += ubik2
+#obj-$(CONFIG_UBIK2) += ubik2/ubik2.o
+#subdir-$(CONFIG_UBIK2) += ubik2
 endif
 
 #
@@ -226,8 +226,8 @@
 endif
 
 ifeq ($(CONFIG_PIGLET),m)
-obj-$(CONFIG_PIGLET) += Piglet/Piglet.o
-subdir-$(CONFIG_PIGLET) += Piglet
+#obj-$(CONFIG_PIGLET) += Piglet/Piglet.o
+#subdir-$(CONFIG_PIGLET) += Piglet
 endif
 
 #
@@ -285,8 +285,8 @@
 
 
 ifeq ($(CONFIG_MIPS_AVALANCHE_VLYNQ),m)                                                         
-subdir-$(CONFIG_MIPS_AVALANCHE_VLYNQ) += avalanche_vlynq
-obj-$(CONFIG_MIPS_AVALANCHE_VLYNQ) += avalanche_vlynq/avalanche_vlynq.o                                                        
+#subdir-$(CONFIG_MIPS_AVALANCHE_VLYNQ) += avalanche_vlynq
+#obj-$(CONFIG_MIPS_AVALANCHE_VLYNQ) += avalanche_vlynq/avalanche_vlynq.o                                                        
 endif
 
 
@@ -313,8 +313,8 @@
 endif
 
 ifeq ($(CONFIG_AVM_SAMMEL),m)
-obj-$(CONFIG_AVM_SAMMEL) += avm/avm.o
-subdir-$(CONFIG_AVM_SAMMEL) += avm
+#obj-$(CONFIG_AVM_SAMMEL) += avm/avm.o
+#subdir-$(CONFIG_AVM_SAMMEL) += avm
 endif
 
 #
@@ -326,8 +326,8 @@
 endif
 
 ifeq ($(CONFIG_AVM_I2C),m)
-obj-$(CONFIG_AVM_I2C) += avm_i2c/avm_i2c.o
-subdir-$(CONFIG_AVM_I2C) += avm_i2c
+#obj-$(CONFIG_AVM_I2C) += avm_i2c/avm_i2c.o
+#subdir-$(CONFIG_AVM_I2C) += avm_i2c
 endif
 
 
