--- pppd/plugins/Makefile.linux
+++ pppd/plugins/Makefile.linux
@@ -9,7 +9,9 @@
 MANDIR = $(DESTDIR)/share/man/man8
 LIBDIR = $(DESTDIR)/lib/pppd/$(VERSION)
 
-SUBDIRS := rp-pppoe pppoatm pppol2tp
+SUBDIRS := rp-pppoe
+#SUBDIRS += pppoatm
+#SUBDIRS += pppol2tp # doesn't compile with all kernel versions used by freetz
 # Uncomment the next line to include the radius authentication plugin
 SUBDIRS += radius
 PLUGINS := minconn.so passprompt.so passwordfd.so winbind.so
