--- configure
+++ configure
@@ -7477,7 +7477,7 @@
   # Note, version OpenSSL 0.9.7j requires -lcrypto even on Linux.
   { echo "$as_me:$LINENO: checking for libssl" >&5
 echo $ECHO_N "checking for libssl... $ECHO_C" >&6; }
-  if test "${sc_cv_have_libssl+set}" = set; then
+  if false; then # ignore cached value otherwise SSL libs are not added to LIBS
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
    LIBS0="$LIBS"
