--- configure
+++ configure
@@ -21873,7 +21873,7 @@
   # next, we search for the wrap library (libwrap.*)
   { echo "$as_me:$LINENO: checking for libwrap" >&5
 echo $ECHO_N "checking for libwrap... $ECHO_C" >&6; }
-  if test "${sc_cv_have_libwrap+set}" = set; then
+  if false; then # ignore cached value otherwise libwrap is not added to LIBS
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
    LIBS0="$LIBS"
