--- configure
+++ configure
@@ -64972,6 +64972,7 @@
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: $msg" >&5
 $as_echo "$msg" >&6; }
 
+if false; then
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for known struct flock definition" >&5
 $as_echo_n "checking for known struct flock definition... " >&6; }
     if test "$cross_compiling" = yes; then :
@@ -65021,6 +65022,7 @@
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
 fi
+fi
 
 
 
