--- ptunnel_org.c	2009-06-22 10:41:21.000000000 +0200
+++ ptunnel.c	2011-04-13 17:34:48.000000000 +0200
@@ -980,7 +980,7 @@ void		handle_packet(char *buf, int bytes
 						else
 							init_state	= kProto_data;
 						cur			= create_and_insert_proxy_desc(pt_pkt->id_no, pkt->identifier, 0, addr, pt_pkt->dst_ip, ntohl(pt_pkt->dst_port), init_state, kProxy_flag);
-						if (init_state == kProto_authenticate) {
+						if (cur && init_state == kProto_authenticate) {
 							pt_log(kLog_debug, "Sending authentication challenge..\n");
 							//	Send challenge
 							cur->challenge	= generate_challenge();
