diff --strip-trailing-cr -x CVS -x .svn -urN boehm-gc/Makefile.in boehm-gc/Makefile.in
--- boehm-gc/Makefile.in	2010-05-22 20:02:44.000000000 +0200
+++ boehm-gc/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -679,6 +679,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -720,7 +721,7 @@
 
 install-data-am:
 
-install-exec-am: install-multi
+install-exec-am:
 
 install-info: install-info-recursive
 
diff --strip-trailing-cr -x CVS -x .svn -urN libffi/Makefile.in libffi/Makefile.in
--- libffi/Makefile.in	2010-05-22 20:02:44.000000000 +0200
+++ libffi/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -1093,6 +1093,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -1166,7 +1167,7 @@
 
 install-data-am:
 
-install-exec-am: install-multi install-toolexeclibLTLIBRARIES
+install-exec-am: install-toolexeclibLTLIBRARIES
 
 install-info: install-info-recursive
 
diff --strip-trailing-cr -x CVS -x .svn -urN libgcc/Makefile.in libgcc/Makefile.in
--- libgcc/Makefile.in	2008-06-17 11:32:34.000000000 +0200
+++ libgcc/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -107,7 +107,7 @@
 all-multi:
 	# If this is the top-level multilib, build all the other
 	# multilibs.
-	@: $(MAKE) ; exec $(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=all
+	@: $(MAKE) ; $(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=all
 
 .PHONY: check installcheck
 check:
@@ -900,7 +900,7 @@
 	done
 
 install: install-leaf
-	@$(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=install
+	@: $(MAKE) ; $(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=install
 
 .PHONY: install install-shared install-libunwind
 
diff --strip-trailing-cr -x CVS -x .svn -urN libgfortran/Makefile.in libgfortran/Makefile.in
--- libgfortran/Makefile.in	2010-05-22 20:02:44.000000000 +0200
+++ libgfortran/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -4630,6 +4630,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-am
 install-strip:
@@ -4672,7 +4673,7 @@
 
 install-data-am:
 
-install-exec-am: install-multi install-myexeclibLTLIBRARIES \
+install-exec-am: install-myexeclibLTLIBRARIES \
 	install-toolexeclibLTLIBRARIES
 
 install-info: install-info-am
diff --strip-trailing-cr -x CVS -x .svn -urN libgomp/Makefile.in libgomp/Makefile.in
--- libgomp/Makefile.in	2010-05-06 19:19:46.000000000 +0200
+++ libgomp/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -747,6 +747,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -793,7 +794,7 @@
 install-data-am: install-info-am install-nodist_fincludeHEADERS \
 	install-nodist_libsubincludeHEADERS
 
-install-exec-am: install-multi install-nodist_toolexeclibHEADERS \
+install-exec-am: install-nodist_toolexeclibHEADERS \
 	install-toolexeclibLTLIBRARIES
 
 install-info: install-info-recursive
diff --strip-trailing-cr -x CVS -x .svn -urN libiberty/Makefile.in libiberty/Makefile.in
--- libiberty/Makefile.in	2007-07-25 08:26:45.000000000 +0200
+++ libiberty/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -101,7 +101,7 @@
 
 # FIXME: add @BUILD_INFO@ once we're sure it works for everyone.
 all: stamp-picdir $(TARGETLIB) needed-list required-list all-subdir
-	@: $(MAKE) ; exec $(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=all
+	@: $(MAKE) ; $(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=all
 
 .PHONY: check installcheck
 check: check-subdir
@@ -366,14 +366,14 @@
 	    ${INSTALL_DATA} $$h $(DESTDIR)$${thd}; \
 	  done; \
 	fi
-	@$(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=install
+	@: $(MAKE) ; $(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=install
 
 install_to_tooldir: all
 	${mkinstalldirs} $(DESTDIR)$(tooldir)/lib/$(MULTIOSDIR)
 	$(INSTALL_DATA) $(TARGETLIB) $(DESTDIR)$(tooldir)/lib/$(MULTIOSDIR)/$(TARGETLIB)n
 	( cd $(DESTDIR)$(tooldir)/lib/$(MULTIOSDIR) ; chmod 644 $(TARGETLIB)n; $(RANLIB) $(TARGETLIB)n )
 	mv -f $(DESTDIR)$(tooldir)/lib/$(MULTIOSDIR)/$(TARGETLIB)n $(DESTDIR)$(tooldir)/lib/$(MULTIOSDIR)/$(TARGETLIB)
-	@$(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=install
+	@: $(MAKE) ; $(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=install
 
 # needed-list is used by libstdc++.  NEEDED is the list of functions
 # to include there.  Do not add anything LGPL to this list; libstdc++
diff --strip-trailing-cr -x CVS -x .svn -urN libjava/classpath/Makefile.in libjava/classpath/Makefile.in
--- libjava/classpath/Makefile.in	2010-05-06 19:19:46.000000000 +0200
+++ libjava/classpath/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -570,6 +570,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -609,7 +610,7 @@
 
 install-data-am:
 
-install-exec-am: install-multi
+install-exec-am:
 
 install-info: install-info-recursive
 
diff --strip-trailing-cr -x CVS -x .svn -urN libjava/libltdl/Makefile.in libjava/libltdl/Makefile.in
--- libjava/libltdl/Makefile.in	2010-05-22 20:02:44.000000000 +0200
+++ libjava/libltdl/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -445,6 +445,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-am
 install-strip:
@@ -487,7 +488,7 @@
 
 install-data-am: install-data-local install-includeHEADERS
 
-install-exec-am: install-libLTLIBRARIES install-multi
+install-exec-am: install-libLTLIBRARIES
 
 install-info: install-info-am
 
diff --strip-trailing-cr -x CVS -x .svn -urN libjava/Makefile.in libjava/Makefile.in
--- libjava/Makefile.in	2010-05-22 20:02:44.000000000 +0200
+++ libjava/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -9702,6 +9702,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -9810,7 +9811,7 @@
 
 install-exec-am: install-binPROGRAMS install-binSCRIPTS \
 	install-dbexecDATA install-dbexecLTLIBRARIES \
-	install-libexecsubPROGRAMS install-multi \
+	install-libexecsubPROGRAMS \
 	install-toolexeclibLTLIBRARIES install-toolexecmainlibDATA
 	@$(NORMAL_INSTALL)
 	$(MAKE) $(AM_MAKEFLAGS) install-exec-hook
@@ -11831,9 +11832,9 @@
 maintainer-clean-recursive: maintainer-clean-multi
 
 all-multi:
-	: $(MAKE) ; exec $(MULTIDO) $(AM_MAKEFLAGS) DO=all multi-do
+	: $(MAKE) ; $(MULTIDO) $(AM_MAKEFLAGS) DO=all multi-do
 install-multi:
-	$(MULTIDO) $(AM_MAKEFLAGS) DO=install multi-do
+	: $(MAKE) ; $(MULTIDO) $(AM_MAKEFLAGS) DO=install multi-do
 mostlyclean-multi:
 	$(MULTICLEAN) $(AM_MAKEFLAGS) DO=mostlyclean multi-clean
 clean-multi:
diff --strip-trailing-cr -x CVS -x .svn -urN libmudflap/Makefile.in libmudflap/Makefile.in
--- libmudflap/Makefile.in	2010-05-06 19:19:46.000000000 +0200
+++ libmudflap/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -652,6 +652,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -693,7 +694,7 @@
 
 install-data-am: install-nobase_libsubincludeHEADERS
 
-install-exec-am: install-multi install-toolexeclibLTLIBRARIES
+install-exec-am: install-toolexeclibLTLIBRARIES
 
 install-info: install-info-recursive
 
diff --strip-trailing-cr -x CVS -x .svn -urN libobjc/Makefile.in libobjc/Makefile.in
--- libobjc/Makefile.in	2007-06-02 03:35:59.000000000 +0200
+++ libobjc/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -138,7 +138,7 @@
 	"tooldir=$(tooldir)"
 
 all: libobjc$(libsuffix).la $(OBJC_BOEHM_GC)
-	: $(MAKE) ; exec $(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=all
+	: $(MAKE) ; $(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=all
 
 # User-visible header files.
 
@@ -339,7 +339,7 @@
 	  $(LIBTOOL_INSTALL) $(INSTALL) libobjc_gc$(libsuffix).la \
 				$(DESTDIR)$(toolexeclibdir);\
 	fi
-	$(MULTIDO) $(FLAGS_TO_PASS) multi-do DO="$@"
+	: $(MAKE) ; $(MULTIDO) $(FLAGS_TO_PASS) multi-do DO="$@"
 	@-$(LIBTOOL) --mode=finish $(DESTDIR)$(toolexeclibdir)
 
 # Copy Objective C headers to installation include directory.
diff --strip-trailing-cr -x CVS -x .svn -urN libssp/Makefile.in libssp/Makefile.in
--- libssp/Makefile.in	2010-05-06 19:19:46.000000000 +0200
+++ libssp/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -534,6 +534,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-am
 install-strip:
@@ -575,7 +576,7 @@
 
 install-data-am: install-nobase_libsubincludeHEADERS
 
-install-exec-am: install-multi install-toolexeclibLTLIBRARIES
+install-exec-am: install-toolexeclibLTLIBRARIES
 
 install-info: install-info-am
 
diff --strip-trailing-cr -x CVS -x .svn -urN libstdc++-v3/Makefile.in libstdc++-v3/Makefile.in
--- libstdc++-v3/Makefile.in	2010-05-06 19:19:46.000000000 +0200
+++ libstdc++-v3/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -543,6 +543,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -582,7 +583,7 @@
 
 install-data-am:
 
-install-exec-am: install-multi
+install-exec-am:
 
 install-info: install-info-recursive
 
diff --strip-trailing-cr -x CVS -x .svn -urN Makefile.in Makefile.in
--- Makefile.in	2009-04-25 07:55:24.000000000 +0200
+++ Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -51898,6 +51898,7 @@
 all-target-libobjc: maybe-all-target-boehm-gc
 all-target-libstdc++-v3: maybe-all-target-libiberty
 configure-target-libstdc++-v3: maybe-configure-target-libgomp
+install-target-libiberty: maybe-install-libiberty
 all-target-libgloss: maybe-all-target-newlib
 all-target-winsup: maybe-all-target-libiberty
 all-target-winsup: maybe-all-target-libtermcap
diff --strip-trailing-cr -x CVS -x .svn -urN zlib/Makefile.in zlib/Makefile.in
--- zlib/Makefile.in	2010-05-22 20:02:44.000000000 +0200
+++ zlib/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -529,6 +529,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-am
 install-strip:
@@ -567,7 +568,7 @@
 
 install-data-am:
 
-install-exec-am: install-multi install-toolexeclibLIBRARIES
+install-exec-am: install-toolexeclibLIBRARIES
 
 install-info: install-info-am
 
