--- etc/profile.orig	2007-02-08 15:33:08.000000000 +0100
+++ etc/profile	2007-06-25 13:35:53.000000000 +0200
@@ -1,8 +1,11 @@
 # /etc/profile
 #
-VERBOSE_RC_CONF=y
+VERBOSE_RC_CONF=n
 . /etc/init.d/rc.conf
 
+export PATH=/sbin:/bin:/usr/sbin:/usr/bin:/mod/sbin:/mod/bin:/mod/usr/sbin:/mod/usr/bin
+export LD_LIBRARY_PATH=/mod/lib
+
 ####################################################################
 echo "ermittle die aktuelle TTY"                                                
 tty=`ls -l /proc/$$/fd/0`                                                       
