--- usr/bin/system_status	2005-08-08 14:07:22.962084424 +0200
+++ usr/bin/system_status	2005-08-08 14:09:00.848203472 +0200
@@ -13,24 +13,10 @@
 if [ "${OEM}" = "avme" ] ; then
     echo -e "-${Language}\c"
 fi
-##########################################################################
-# An alle Kollegen, die die Firmware der Fritz Boxen fuer 
-# lokale Entwicklungen modifizieren:
-##########################################################################
-# 
-# Seit so nett und hinterlasst euer Logo, und in Stichworten, welche
-# Modifikationen ihr durchgeführt habt. Ihr erleichtert den Nutzern
-# eurer Version das Leben.
-#
-# Im file /etc/version koennt ihr die Variable FIRMWARE_SUBVERSION modifizieren
-# hier koennt ihr eurer eigenen Subversion einen Namen geben
-#
-# export FIRMWARE_SUBVERSION=-Fritz-Build-77
-#
-###########################################################################
-# echo -e "<br>"
-# echo -e "Fritz der große Modifizierer"
-# echo -e "--- telnet demon aktiviert"
-# echo -e "<br>"
+
+echo -e "<br>"
+echo -e "FRITZ!Box Modifikation <subversion>, <a href=\"http://www.ip-phone-forum.de/showthread.php?t=85371\" target=\"_blank\">http://www.ip-phone-forum.de/showthread.php?t=85371</a>"
+echo -e " --- Optionen: <options>"
+echo -e "<br>"
 
 echo -e "</body></html>\n"
