--- configure
+++ configure
@@ -11075,6 +11075,7 @@
 
 fi
 
+if false; then
 	# -lresolv needs to be at the end of LIBS or DNS lookups break
 	{ echo "$as_me:$LINENO: checking for res_query in -lresolv" >&5
 echo $ECHO_N "checking for res_query in -lresolv... $ECHO_C" >&6; }
@@ -11140,6 +11141,7 @@
 if test $ac_cv_lib_resolv_res_query = yes; then
    LIBS="$LIBS -lresolv"
 fi
+fi
 
 	IPADDR_IN_DISPLAY=yes
 	cat >>confdefs.h <<\_ACEOF
@@ -13462,6 +13464,7 @@
 if test $ac_cv_func_strcasecmp = yes; then
   :
 else
+if false; then
    { echo "$as_me:$LINENO: checking for strcasecmp in -lresolv" >&5
 echo $ECHO_N "checking for strcasecmp in -lresolv... $ECHO_C" >&6; }
 if test "${ac_cv_lib_resolv_strcasecmp+set}" = set; then
@@ -13526,6 +13529,7 @@
 if test $ac_cv_lib_resolv_strcasecmp = yes; then
   LIBS="$LIBS -lresolv"
 fi
+fi
 
 
 fi
@@ -31053,6 +31057,7 @@
 
 	{ echo "$as_me:$LINENO: result: no" >&5
 echo "${ECHO_T}no" >&6; }
+if false; then
 		    saved_LIBS="$LIBS"
 		    LIBS="$LIBS -lresolv"
 		    { echo "$as_me:$LINENO: checking for res_query in -lresolv" >&5
@@ -31113,6 +31118,7 @@
       conftest$ac_exeext conftest.$ac_ext
 
 fi
+fi
 
 rm -f core conftest.err conftest.$ac_objext conftest_ipa8_conftest.oo \
       conftest$ac_exeext conftest.$ac_ext
