--- configure
+++ configure
@@ -7880,6 +7880,7 @@
 
 fi
 
+if false; then
 	# -lresolv needs to be at the end of LIBS or DNS lookups break
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for res_query in -lresolv" >&5
 $as_echo_n "checking for res_query in -lresolv... " >&6; }
@@ -7920,6 +7921,7 @@
 if test "x$ac_cv_lib_resolv_res_query" = xyes; then :
    LIBS="$LIBS -lresolv"
 fi
+fi
 
 	IPADDR_IN_DISPLAY=yes
 	$as_echo "#define USE_PIPES 1" >>confdefs.h
@@ -8819,6 +8821,7 @@
 if test "x$ac_cv_func_strcasecmp" = xyes; then :
 
 else
+if false; then
    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for strcasecmp in -lresolv" >&5
 $as_echo_n "checking for strcasecmp in -lresolv... " >&6; }
 if ${ac_cv_lib_resolv_strcasecmp+:} false; then :
@@ -8858,6 +8861,7 @@
 if test "x$ac_cv_lib_resolv_strcasecmp" = xyes; then :
   LIBS="$LIBS -lresolv"
 fi
+fi
 
 
 fi
@@ -16348,6 +16352,7 @@
 else
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
+if false; then
 		    saved_LIBS="$LIBS"
 		    LIBS="$LIBS -lresolv"
 		    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for res_query in -lresolv" >&5
@@ -16383,6 +16388,7 @@
     conftest$ac_exeext conftest.$ac_ext
 
 fi
+fi
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 		for ac_func in _getshort _getlong
