--- configure
+++ configure
@@ -6435,6 +6435,7 @@
 fi
 rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
 } # clang
+if false; then
 	{
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking if $LD supports link flag -Wl,-z,retpolineplt" >&5
 $as_echo_n "checking if $LD supports link flag -Wl,-z,retpolineplt... " >&6; }
@@ -6481,6 +6482,7 @@
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 }
+fi
 	{
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking if $CC supports compile flag -D_FORTIFY_SOURCE=2" >&5
 $as_echo_n "checking if $CC supports compile flag -D_FORTIFY_SOURCE=2... " >&6; }
