--- cyassl/openssl/dh.h
+++ cyassl/openssl/dh.h
@@ -5,6 +5,9 @@
 #define CYASSL_DH_H_
 
 
+/* supports only a subset of openssl-functions, thus OPENSSL_NO_DH for now */
+#define OPENSSL_NO_DH
+
 #include <cyassl/openssl/ssl.h>
 #include <cyassl/openssl/bn.h>
 
--- cyassl/openssl/ecdsa.h
+++ cyassl/openssl/ecdsa.h
@@ -1,2 +1,8 @@
 /* ecdsa.h for openssl */
 
+#ifndef CYASSL_ECDSA_H
+#define CYASSL_ECDSA_H
+
+#define OPENSSL_NO_ECDSA
+
+#endif
--- cyassl/openssl/ec.h
+++ cyassl/openssl/ec.h
@@ -1,2 +1,8 @@
 /* ec.h for openssl */
 
+#ifndef CYASSL_EC_H
+#define CYASSL_EC_H
+
+#define OPENSSL_NO_EC
+
+#endif
--- cyassl/openssl/lhash.h
+++ cyassl/openssl/lhash.h
@@ -1,2 +1,8 @@
 /* lhash.h for openSSL */
 
+#ifndef CYASSL_LHASH_H
+#define CYASSL_LHASH_H
+
+#define OPENSSL_NO_LHASH
+
+#endif
--- cyassl/openssl/md4.h
+++ cyassl/openssl/md4.h
@@ -1 +1,10 @@
 /* md4.h for libcurl */
+
+#ifndef CYASSL_MD4_H
+#define CYASSL_MD4_H
+
+#ifdef NO_MD4
+#define OPENSSL_NO_MD4
+#endif
+
+#endif
--- cyassl/openssl/ocsp.h
+++ cyassl/openssl/ocsp.h
@@ -1 +1,8 @@
 /* ocsp.h for libcurl */
+
+#ifndef CYASSL_OCSP_H
+#define CYASSL_OCSP_H
+
+#define OPENSSL_NO_OCSP
+
+#endif
