--- Makefile.in
+++ Makefile.in
@@ -167,8 +167,8 @@
 @BUILD_ECC_TRUE@am__append_10 = ctaocrypt/src/ecc.c
 @BUILD_OCSP_TRUE@am__append_11 = src/ocsp.c
 @BUILD_CRL_TRUE@am__append_12 = src/crl.c
-@BUILD_CRL_MONITOR_TRUE@am__append_13 = $(PTHREAD_CFLAGS)
-@BUILD_CRL_MONITOR_TRUE@am__append_14 = $(PTHREAD_LIBS)
+am__append_13 = $(PTHREAD_CFLAGS)
+am__append_14 = $(PTHREAD_LIBS)
 subdir = .
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/m4/ax_append_compile_flags.m4 \
