--- Makefile.in
+++ Makefile.in
@@ -225,8 +225,8 @@
 @BUILD_ECC_TRUE@am__append_13 = ctaocrypt/src/ecc.c
 @BUILD_OCSP_TRUE@am__append_14 = src/ocsp.c
 @BUILD_CRL_TRUE@am__append_15 = src/crl.c
-@BUILD_CRL_MONITOR_TRUE@am__append_16 = $(PTHREAD_CFLAGS)
-@BUILD_CRL_MONITOR_TRUE@am__append_17 = $(PTHREAD_LIBS)
+am__append_16 = $(PTHREAD_CFLAGS)
+am__append_17 = $(PTHREAD_LIBS)
 @BUILD_EXAMPLES_TRUE@am__append_18 = examples/client/client \
 @BUILD_EXAMPLES_TRUE@	examples/server/server \
 @BUILD_EXAMPLES_TRUE@	examples/echoclient/echoclient \
