--- configure.orig	2007-10-02 22:35:11.000000000 +0200
+++ configure	2007-10-02 22:35:28.000000000 +0200
@@ -5116,7 +5116,7 @@
 fi
 
 
-CPPFLAGS="-D__UNIX__"
+CPPFLAGS+=" -D__UNIX__"
 
 
 THREADLIBS=no
@@ -8641,7 +8641,7 @@
 
   { echo "$as_me:$LINENO: checking for Ogg" >&5
 echo $ECHO_N "checking for Ogg... $ECHO_C" >&6; }
-  no_ogg=""
+  no_ogg="yes"
 
 
   if test "x$enable_oggtest" = "xyes" ; then
@@ -8769,7 +8769,7 @@
 
   { echo "$as_me:$LINENO: checking for Vorbis" >&5
 echo $ECHO_N "checking for Vorbis... $ECHO_C" >&6; }
-  no_vorbis=""
+  no_vorbis="yes"
 
 
   if test "x$enable_vorbistest" = "xyes" ; then
--- Makefile.in.orig	2007-09-01 19:19:15.000000000 +0200
+++ Makefile.in	2007-10-02 22:33:24.000000000 +0200
@@ -627,7 +627,7 @@
 	esac;
 
 $(top_builddir)/config.status: $(top_srcdir)/configure $(CONFIG_STATUS_DEPENDENCIES)
-	$(SHELL) ./config.status --recheck
+	#$(SHELL) ./config.status --recheck
 
 $(top_srcdir)/configure:  $(am__configure_deps)
 	cd $(srcdir) && $(AUTOCONF)
