--- configure.orig   2009-03-31 03:51:13.000000000 +0200
+++ configure       2010-02-18 11:03:32.000000000 +0100
@@ -4851,7 +4851,7 @@
 fi
 
 
-CPPFLAGS="-D__UNIX__"
+CPPFLAGS+=" -D__UNIX__"
 
 
 
@@ -8764,7 +8764,7 @@
 
   { $as_echo "$as_me:$LINENO: checking for Ogg" >&5
 $as_echo_n "checking for Ogg... " >&6; }
-  no_ogg=""
+  no_ogg="yes"
 
 
   if test "x$enable_oggtest" = "xyes" ; then
@@ -8894,7 +8894,7 @@
 
   { $as_echo "$as_me:$LINENO: checking for Vorbis" >&5
 $as_echo_n "checking for Vorbis... " >&6; }
-  no_vorbis=""
+  no_vorbis="yes"
 
 
   if test "x$enable_vorbistest" = "xyes" ; then
--- Makefile.in.orig 2009-03-31 03:51:10.000000000 +0200
+++ Makefile.in     2010-02-18 11:03:32.000000000 +0100
@@ -636,7 +636,7 @@
 	esac;
 
 $(top_builddir)/config.status: $(top_srcdir)/configure $(CONFIG_STATUS_DEPENDENCIES)
-	$(SHELL) ./config.status --recheck
+	#$(SHELL) ./config.status --recheck
 
 $(top_srcdir)/configure:  $(am__configure_deps)
 	cd $(srcdir) && $(AUTOCONF)
