diff -burN streamripper-1.61.27/configure streamripper-1.61.27/configure
--- configure	2006-09-09 18:48:21.000000000 +0200
+++ configure	2007-02-18 14:48:01.000000000 +0100
@@ -3059,7 +3059,7 @@
   force_use_included_argv=no
 fi;
 
-CPPFLAGS="-Ilib -D__UNIX__"
+CPPFLAGS="$CPPFLAGS -Ilib -D__UNIX__"
 
 
 THREADLIBS=no
diff -burN streamripper-1.61.27/configure.ac streamripper-1.61.27/configure.ac
--- configure.ac	2006-09-09 18:48:03.000000000 +0200
+++ configure.ac	2007-02-18 14:47:22.000000000 +0100
@@ -39,7 +39,7 @@
   force_use_included_argv=no)
 dnl AC_MSG_RESULT($force_use_included_argv)
 
-CPPFLAGS="-Ilib -D__UNIX__"
+CPPFLAGS="$CPPFLAGS -Ilib -D__UNIX__"
 AC_SUBST(CPPFLAGS)
 
 dnl -------- Tests for threading library ----------
diff -burN streamripper-1.61.27/Makefile.in streamripper-1.61.27/Makefile.in
--- Makefile.in	2006-09-09 18:48:19.000000000 +0200
+++ Makefile.in	2007-02-18 14:50:40.000000000 +0100
@@ -553,7 +553,7 @@
 	esac;
 
 $(top_builddir)/config.status: $(top_srcdir)/configure $(CONFIG_STATUS_DEPENDENCIES)
-	$(SHELL) ./config.status --recheck
+#	$(SHELL) ./config.status --recheck
 
 $(top_srcdir)/configure:  $(am__configure_deps)
 	cd $(srcdir) && $(AUTOCONF)
