--- configure.orig	2008-11-19 15:04:30.000000000 +0100
+++ configure	2008-11-22 16:03:02.000000000 +0100
@@ -4694,7 +4694,7 @@
 fi
 
 
-CPPFLAGS="-D__UNIX__"
+CPPFLAGS+=" -D__UNIX__"
 
 
 
@@ -8122,7 +8122,7 @@
 
   { echo "$as_me:$LINENO: checking for Ogg" >&5
 echo $ECHO_N "checking for Ogg... $ECHO_C" >&6; }
-  no_ogg=""
+  no_ogg="yes"
 
 
   if test "x$enable_oggtest" = "xyes" ; then
@@ -8250,7 +8250,7 @@
 
   { echo "$as_me:$LINENO: checking for Vorbis" >&5
 echo $ECHO_N "checking for Vorbis... $ECHO_C" >&6; }
-  no_vorbis=""
+  no_vorbis="yes"
 
 
   if test "x$enable_vorbistest" = "xyes" ; then
--- Makefile.in.orig	2008-11-19 15:04:29.000000000 +0100
+++ Makefile.in	2008-11-22 16:03:02.000000000 +0100
@@ -627,7 +627,7 @@
 	esac;
 
 $(top_builddir)/config.status: $(top_srcdir)/configure $(CONFIG_STATUS_DEPENDENCIES)
-	$(SHELL) ./config.status --recheck
+	#$(SHELL) ./config.status --recheck
 
 $(top_srcdir)/configure:  $(am__configure_deps)
 	cd $(srcdir) && $(AUTOCONF)
