--- etc/init.d/rc.net	2011-11-28 01:54:01+01:00
+++ etc/init.d/rc.net	2011-11-28 01:54:01+01:00
@@ -68,6 +68,8 @@
 #   a) it has the parameter at all AND
 #   b) upnpd binary does *not* exist
 [ "$_dsld_upnp" ] && [ ! "$_upnpd" ] && DSLDDPARAM="-g"
+# Do we have a Samba server (smbd) or was ist stripped from the firmware?
+_smbd=$(basename $(which smbd) 2> /dev/null)
 
 
 upnpdstart()
@@ -325,6 +327,7 @@
 #we have the samba cfg in /var, so crate them there
 sambastart()
 {
+   [ "$_smbd" ] || return;
    cd /
    tar xf /var.tar
    /bin/smbd
