--- configure
+++ configure
@@ -2610,7 +2610,11 @@
 # Checks for libraries.
 # Checks for libmatrixssl.
 
+if test "$enable_semistatic" = yes; then
 
+LIBS="$MATRIXSSL_DIR/src/libmatrixsslstatic.a -lpthread $LIBS"
+ echo $ECHO_N "STATIC! no -lmatrixssl... $ECHO_C" >&6
+else
 echo "$as_me:$LINENO: checking for matrixSslOpen in -lmatrixssl" >&5
 echo $ECHO_N "checking for matrixSslOpen in -lmatrixssl... $ECHO_C" >&6
 if test "${ac_cv_lib_matrixssl_matrixSslOpen+set}" = set; then
@@ -2687,7 +2691,7 @@
    { (exit 1); exit 1; }; }
 fi
 
-
+fi
 
 # Checks for header files.
 ac_ext=c
