--- gdb/gdbserver/configure.orig	2008-02-01 01:08:25.000000000 +0100
+++ gdb/gdbserver/configure	2009-04-17 02:44:27.000000000 +0200
@@ -1239,7 +1239,7 @@
       ac_cache_corrupted=: ;;
     ,);;
     *)
-      if test "x$ac_old_val" != "x$ac_new_val"; then
+      if test "`echo x$ac_old_val`" != "`echo x$ac_new_val`"; then
 	{ echo "$as_me:$LINENO: error: \`$ac_var' has changed since the previous run:" >&5
 echo "$as_me: error: \`$ac_var' has changed since the previous run:" >&2;}
 	{ echo "$as_me:$LINENO:   former value:  $ac_old_val" >&5
--- gdb/configure.orig	2008-01-13 13:23:04.000000000 +0100
+++ gdb/configure	2009-04-17 02:44:27.000000000 +0200
@@ -272,7 +272,7 @@
 PACKAGE_BUGREPORT=
 
 ac_unique_file="main.c"
-ac_subdirs_all="$ac_subdirs_all doc testsuite"
+ac_subdirs_all="$ac_subdirs_all doc"
 # Factoring default headers for most tests.
 ac_includes_default="\
 #include <stdio.h>
@@ -3079,7 +3079,7 @@
 
 
 
-subdirs="$subdirs doc testsuite"
+subdirs="$subdirs doc"
 
 
 # Check whether to support alternative target configurations
