--- etc/init.d/rc.net	2011-10-23 14:35:53+02:00
+++ etc/init.d/rc.net	2011-10-23 14:35:53+02:00
@@ -60,6 +60,8 @@
 #   a) it has the parameter at all AND
 #   b) upnpd binary does *not* exist
 [ "$_dsld_upnp" ] && [ ! "$_upnpd" ] && DSLDDPARAM="-g"
+# Do we have a Samba server (smbd) or was ist stripped from the firmware?
+_smbd=$(basename $(which smbd) 2> /dev/null)
 
 
 upnpdstart()
@@ -314,6 +316,7 @@
 #we have the samba cfg in /var, so crate them there
 sambastart()
 {
+   [ "$_smbd" ] || return;
    cd /
    tar xf /var.tar
    /bin/smbd
