--- linux-2.6.32/drivers/block/compcache/Makefile
+++ linux-2.6.32/drivers/block/compcache/Makefile
@@ -6,7 +6,7 @@
 ramzswap-objs	:=	ramzswap_drv.o xvmalloc.o
 
 ifneq ($(CONFIG_COMPCACHE),n)
-extra-y			+= rzscontrol ramzswap.sh
+#extra-y			+= rzscontrol ramzswap.sh
 
 rzscontrol-objs	:= rzscontrol.o
 
--- linux-2.6.32/drivers/rtc/Makefile
+++ linux-2.6.32/drivers/rtc/Makefile
@@ -91,7 +91,7 @@
 ifeq ($(CONFIG_RTC_DRV_AVM_REF_CLOCK),m)
 CFLAGS_rtc-avm.o += -DRTC_AVM_DEVICE_COUNT=1 # -DRTC_AVM_DEBUG=1
 
-extra-y			 += S09-rtc
+#extra-y			 += S09-rtc
 
 $(src)/S09-rtc: FORCE
 	rm -rf $(src)/S09-rtc
--- linux-2.6.32/drivers/net/avm_cpmac/Makefile
+++ linux-2.6.32/drivers/net/avm_cpmac/Makefile
@@ -23,13 +23,13 @@
 			   cpphy_entry.o cpphy_ar8216_ar.o cpphy_mgmt.o \
 			   cpmac_fusiv_if.o cpmac_puma_if.o cpphy_switch.o cpmac_product_conf.o
 			
-$(obj)/cpphy_entry.o: $(src)/cpphy_entry.c DO_COPY FORCE
+$(obj)/cpphy_entry.o: $(src)/cpphy_entry.c FORCE
 
 DO_COPY:
 	mkdir -p $(FRITZ_BOX_BUILD_DIR)/$(KERNEL_BUILD)/filesystem/include/linux 
 	cp -f $(src)/linux_avm_cpmac.h $(FRITZ_BOX_BUILD_DIR)/$(KERNEL_BUILD)/filesystem/include/linux/avm_cpmac.h
 
-extra-y	+= cpmacconfig
+#extra-y	+= cpmacconfig
 
 cpmacconfig-objs	:= cpmacconfig.o
 
