--- squashfs-tools/error.h
+++ squashfs-tools/error.h
@@ -81,6 +81,15 @@
 
 #define EXIT_UNSQUASH(s, args...) BAD_ERROR(s, ##args)
 
+#define ERROR_EXIT_UNSQUASHFS(s, args...)\
+	do {\
+		if (exit_on_error) {\
+			EXIT_UNSQUASH(s, ##args);\
+		} else {\
+			ERROR(s, ##args);\
+		}\
+	} while(0)
+
 #define MEM_ERROR() \
 	do {\
 		progressbar_error("FATAL ERROR: Out of memory (%s)\n", \
--- squashfs-tools/unsquashfs.c
+++ squashfs-tools/unsquashfs.c
@@ -78,6 +78,8 @@
 int no_xattrs = XATTR_DEF;
 int user_xattrs = FALSE;
 
+int exit_on_error = FALSE;
+
 int lookup_type[] = {
 	0,
 	S_IFDIR,
@@ -700,7 +702,7 @@
 			outlen, &error);
 
 		if(res == -1) {
-			ERROR("%s uncompress failed with error code %d\n",
+			ERROR_EXIT_UNSQUASHFS("%s uncompress failed with error code %d\n",
 				comp->name, error);
 			goto failed;
 		}
@@ -746,7 +748,7 @@
 			block_size, &error);
 
 		if(res == -1) {
-			ERROR("%s uncompress failed with error code %d\n",
+			ERROR_EXIT_UNSQUASHFS("%s uncompress failed with error code %d\n",
 				comp->name, error);
 			goto failed;
 		}
@@ -2123,7 +2125,7 @@
 			&error);
 
 		if(res == -1)
-			ERROR("%s uncompress failed with error code %d\n",
+			ERROR_EXIT_UNSQUASHFS("%s uncompress failed with error code %d\n",
 				comp->name, error);
 		else
 			memcpy(entry->data, tmp, res);
@@ -2627,6 +2629,8 @@
 		} else if(strcmp(argv[i], "-regex") == 0 ||
 				strcmp(argv[i], "-r") == 0)
 			use_regex = TRUE;
+		else if(strcmp(argv[i], "-exit-on-error") == 0)
+			exit_on_error = TRUE;
 		else
 			goto options;
 	}
@@ -2690,6 +2694,7 @@
 				"regular expressions\n");
 			ERROR("\t\t\t\trather than use the default shell "
 				"wildcard\n\t\t\t\texpansion (globbing)\n");
+			ERROR("\t-exit-on-error\t\ttreat normally ignored errors as fatal\n");
 			ERROR("\nDecompressors available:\n");
 			display_compressors("", "");
 		}
