--- linux-2.6.28/net/avm_pa/avm_pa.c
+++ linux-2.6.28/net/avm_pa/avm_pa.c
@@ -6939,11 +6939,13 @@
 
    if (load_reduce == 0) {
       if (ctx->load_reduce) {
+/*
          printk(KERN_INFO "avm_pa: load reduce %d => %d, rate %u (pps %lu ov_pps %lu)\n",
                        ctx->load_reduce, load_reduce,
                        ctx->rate,
                        (unsigned long)ctx->stats.fw_pps,
                        (unsigned long)ctx->stats.overlimit_pps);
+*/
       }
    } else if (ctx->stats.fw_pps > AVM_PA_MINRATE) {
       int change = ctx->load_reduce - load_reduce;
@@ -6975,9 +6977,11 @@
       avm_pa_tbf_update(ctx->rate, ctx->pktbuffer, ctx->pktpeak);
       ctx->tbf_enabled = 1;
    } else {
+/*
       printk(KERN_INFO "avm_pa: load reduce %d => %d, rate %u (pps %lu)\n",
                     ctx->load_reduce, load_reduce,
                     ctx->rate, (unsigned long)ctx->stats.fw_pps);
+*/
    }
    ctx->load_reduce = load_reduce;
 }
