--- inadyn/makefile	2005-09-09 23:16:50.000000000 +0200
+++ inadyn/makefile	2006-11-19 13:24:46.000000000 +0100
@@ -33,8 +33,8 @@
 	$(OUTDIR)/os_unix.o $(OUTDIR)/os_windows.o $(OUTDIR)/tcp.o $(OUTDIR)/inadyn_cmd.o
 OBJ=$(COMMON_OBJ) $(CFG_OBJ)
 
-COMPILE=gcc  -Wall  -pedantic -c  $(ARCH_SPECIFIC_CFLAGS) $(CFLAGS) -o "$(OUTDIR)/$(*F).o" $(CFG_INC) "$<"
-LINK=gcc $(CFLAGS) -o "$(OUTFILE)" $(OBJ) $(CFG_LIB) $(ARCH_SPECIFIC_LIBS)
+COMPILE=$(CC)  -Wall  -pedantic -c  $(ARCH_SPECIFIC_CFLAGS) $(CFLAGS) -o "$(OUTDIR)/$(*F).o" $(CFG_INC) "$<"
+LINK=$(CC) $(CFLAGS) $(LDFLAGS) -o "$(OUTFILE)" $(OBJ) $(CFG_LIB) $(ARCH_SPECIFIC_LIBS)
 
 # Pattern rules
 $(OUTDIR)/%.o : $(SRCDIR)/%.c
