--- inadyn/src/dyndns.c	2007-12-24 01:29:00.000000000 +0100
+++ inadyn/src/dyndns.c	2007-12-24 01:29:48.000000000 +0100
@@ -179,7 +179,8 @@
 	(void)p_sys_info;
 	return sprintf(p_self->p_req_buffer, FREEDNS_UPDATE_MY_IP_REQUEST_FORMAT,
 		p_self->info[infcnt].dyndns_server_name.name,
-		p_self->info[infcnt].dyndns_server_name.port,
+		/*p_self->info[infcnt].dyndns_server_name.port,*/
+		HTTP_DEFAULT_PORT,
 		p_self->info[infcnt].dyndns_server_url,
 		p_self->info[infcnt].alias_info[alcnt].hashes.str,
 		p_self->info[infcnt].dyndns_server_name.name);
