--- Makefile.in
+++ Makefile.in
@@ -373,7 +373,7 @@
     n|no|NO) false;; \
     *) (install-info --version) >/dev/null 2>&1;; \
   esac
-DATA = $(cmake_DATA) $(examples_DATA) $(m4data_DATA) \
+DATA =                                               \
 	$(nodist_pkgconfig_DATA)
 RECURSIVE_CLEAN_TARGETS = mostlyclean-recursive clean-recursive	\
   distclean-recursive maintainer-clean-recursive
@@ -677,7 +677,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = include . doc example fuzz xstc $(am__append_1)
+SUBDIRS = include .                  xstc $(am__append_1)
 DIST_SUBDIRS = include . doc example fuzz python xstc
 AM_CPPFLAGS = -I$(top_builddir)/include -I$(srcdir)/include -DSYSCONFDIR='"$(sysconfdir)"'
 AM_CFLAGS = $(EXTRA_CFLAGS)
@@ -1906,7 +1906,7 @@
 	$(MAKE) $(AM_MAKEFLAGS) $(check_PROGRAMS)
 	$(MAKE) $(AM_MAKEFLAGS) check-local
 check: check-recursive
-all-am: Makefile $(PROGRAMS) $(LTLIBRARIES) $(SCRIPTS) $(DATA) \
+all-am: Makefile             $(LTLIBRARIES)            $(DATA) \
 		config.h
 install-binPROGRAMS: install-libLTLIBRARIES
 
@@ -2033,14 +2033,14 @@
 
 info-am:
 
-install-data-am: install-cmakeDATA install-examplesDATA \
-	install-m4dataDATA install-nodist_pkgconfigDATA
+install-data-am: \
+	install-nodist_pkgconfigDATA
 
 install-dvi: install-dvi-recursive
 
 install-dvi-am:
 
-install-exec-am: install-binPROGRAMS install-libLTLIBRARIES \
+install-exec-am: install-libLTLIBRARIES \
 	install-nodist_binSCRIPTS
 
 install-html: install-html-recursive
