Ticket #2991: fix redraw of WLabel after text change.
Signed-off-by: Andrew Borodin <aborodin@…>

Index: lib/widget/label.c
===================================================================
--- lib/widget/label.c	(revision 30959e0b7d136073dc9bc4905a7b4c826251aec7)
+++ lib/widget/label.c	(revision f23f0613ec4a31fc5c04bb86affcdadd74a21757)
@@ -183,8 +183,8 @@
     }
 
+    widget_redraw (w);
+
     if (newcols < w->cols)
         w->cols = newcols;
-
-    widget_redraw (w);
 }
