Index: src/editor/edit.c
===================================================================
--- src/editor/edit.c	(revision c5924af3c454585a2307a473e8c8e2dcc3cbe4a6)
+++ src/editor/edit.c	(revision 4f1801ca11a50187b51916813a3a331e4e3190ba)
@@ -326,4 +326,6 @@
 
   cleanup:
+    (void) mc_close (file);
+
     if (errmsg != NULL)
     {
@@ -332,7 +334,4 @@
         ret = FALSE;
     }
-
-    if (!ret)
-        (void) mc_close (file);
 
     return ret;
