--- src/filemanager/filegui.c
+++ src/filemanager/filegui.c
@@ -108,7 +108,7 @@
     return 1;
 }
 #else
-#include <string.h>             /* for strverscmp */
+#include "lib/strutil.h"        /* for str_verscmp */
 #include <sys/utsname.h>
 #include <sys/statfs.h>
 #define STAT_STATFS2_BSIZE 1
@@ -120,7 +120,7 @@
     struct utsname name;
 
     if (statvfs_works_cache < 0)
-        statvfs_works_cache = (uname (&name) == 0 && 0 <= strverscmp (name.release, "2.6.36"));
+        statvfs_works_cache = (uname (&name) == 0 && 0 <= str_verscmp (name.release, "2.6.36"));
     return statvfs_works_cache;
 }
 #endif
--- src/filemanager/mountlist.c
+++ src/filemanager/mountlist.c
@@ -305,7 +305,7 @@
     return 1;
 }
 #else
-#include <string.h>             /* for strverscmp */
+#include "lib/strutil.h"        /* for str_verscmp */
 #include <sys/utsname.h>
 #include <sys/statfs.h>
 #define STAT_STATFS2_BSIZE 1
@@ -317,7 +317,7 @@
     struct utsname name;
 
     if (statvfs_works_cache < 0)
-        statvfs_works_cache = (uname (&name) == 0 && 0 <= strverscmp (name.release, "2.6.36"));
+        statvfs_works_cache = (uname (&name) == 0 && 0 <= str_verscmp (name.release, "2.6.36"));
     return statvfs_works_cache;
 }
 #endif
