--- misc/Makefile.am
+++ misc/Makefile.am
@@ -1,6 +1,6 @@
 ## Process this file with automake to create Makefile.in.
 
-SUBDIRS = ext.d macros.d skins syntax
+SUBDIRS = ext.d          skins syntax
 
 LIBFILES_OUT = mc.ext
 
--- misc/Makefile.in
+++ misc/Makefile.in
@@ -471,7 +471,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = ext.d macros.d skins syntax
+SUBDIRS = ext.d          skins syntax
 LIBFILES_OUT = mc.ext
 noinst_DATA = xterm.ad
 PKGDATA_IN = mc.charsets.in
