--- misc/Makefile.am
+++ misc/Makefile.am
@@ -1,6 +1,6 @@
 ## Process this file with automake to create Makefile.in.
 
-SUBDIRS = macros.d skins syntax
+SUBDIRS = skins syntax
 
 LIBFILES_OUT = mc.ext
 
--- misc/Makefile.in
+++ misc/Makefile.in
@@ -371,7 +371,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = macros.d skins syntax
+SUBDIRS = skins syntax
 LIBFILES_OUT = mc.ext
 noinst_DATA = xterm.ad
 pkgdata_DATA = mc.lib $(am__append_1)
