--- doc/hlp/Makefile.am
+++ doc/hlp/Makefile.am
@@ -1,4 +1,7 @@
-DIST_SUBDIRS = es hu it pl ru sr
+DIST_SUBDIRS =
+if USE_NLS
+DIST_SUBDIRS += es hu it pl ru sr
+endif
 if USE_NLS
 SUBDIRS = $(DOC_LINGUAS)
 endif
--- doc/hlp/Makefile.in
+++ doc/hlp/Makefile.in
@@ -365,7 +365,8 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-DIST_SUBDIRS = es hu it pl ru sr
+DIST_SUBDIRS =
+@USE_NLS_TRUE@DIST_SUBDIRS += es hu it pl ru sr
 @USE_NLS_TRUE@SUBDIRS = $(DOC_LINGUAS)
 hlpdir = $(pkgdatadir)/help
 hlp_DATA = mc.hlp
--- doc/Makefile.am
+++ doc/Makefile.am
@@ -1,5 +1,5 @@
-DIST_SUBDIRS = hints man hlp
-SUBDIRS = hints man hlp
+DIST_SUBDIRS = hints hlp
+SUBDIRS = hints hlp
 
 DOCS = FAQ HACKING INSTALL INSTALL.FAST MAINTAINERS README.QNX keybind-migration.txt
 EXTRA_DIST = $(DX_CONFIG) doxygen-include.am $(DOCS)
--- doc/Makefile.in
+++ doc/Makefile.in
@@ -389,8 +389,8 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-DIST_SUBDIRS = hints man hlp
-SUBDIRS = hints man hlp
+DIST_SUBDIRS = hints hlp
+SUBDIRS = hints hlp
 DOCS = FAQ HACKING INSTALL INSTALL.FAST MAINTAINERS README.QNX keybind-migration.txt
 EXTRA_DIST = $(DX_CONFIG) doxygen-include.am $(DOCS)
 @DX_COND_doc_TRUE@@DX_COND_html_TRUE@DX_CLEAN_HTML = @DX_DOCDIR@/html
--- doc/man/Makefile.am
+++ doc/man/Makefile.am
@@ -1,4 +1,7 @@
-DIST_SUBDIRS = es hu it pl ru sr
+DIST_SUBDIRS =
+if USE_NLS
+DIST_SUBDIRS += es hu it pl ru sr
+endif
 if USE_NLS
 SUBDIRS = $(DOC_LINGUAS)
 endif
--- doc/man/Makefile.in
+++ doc/man/Makefile.in
@@ -368,7 +368,8 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-DIST_SUBDIRS = es hu it pl ru sr
+DIST_SUBDIRS =
+@USE_NLS_TRUE@DIST_SUBDIRS += es hu it pl ru sr
 @USE_NLS_TRUE@SUBDIRS = $(DOC_LINGUAS)
 man_MANS = mc.1 mcedit.1 mcview.1
 CLEANFILES = $(man_MANS)
--- misc/Makefile.am
+++ misc/Makefile.am
@@ -20,8 +20,7 @@
 	mc.keymap \
 	mc.keymap.default \
 	mc.keymap.emacs \
-	mc.menu \
-	mc.menu.sr
+	mc.menu
 
 CLEANFILES = $(SCRIPTS_OUT)
 
--- misc/Makefile.in
+++ misc/Makefile.in
@@ -382,8 +382,7 @@
 	mc.keymap \
 	mc.keymap.default \
 	mc.keymap.emacs \
-	mc.menu \
-	mc.menu.sr
+	mc.menu
 
 CLEANFILES = $(SCRIPTS_OUT)
 cfgdir = $(sysconfdir)/@PACKAGE@
--- src/vfs/extfs/helpers/Makefile.am
+++ src/vfs/extfs/helpers/Makefile.am
@@ -3,7 +3,7 @@
 EXTFSCONFFILES = sfs.ini
 
 # Files to install and distribute other than extfs scripts
-EXTFS_MISC  = README README.extfs
+EXTFS_MISC  =
 
 # Scripts hat don't need adaptation to the local system
 EXTFS_CONST = bpp changesetfs gitfs+ patchsetfs rpm trpm u7z
--- src/vfs/extfs/helpers/Makefile.in
+++ src/vfs/extfs/helpers/Makefile.in
@@ -344,7 +344,7 @@
 EXTFSCONFFILES = sfs.ini
 
 # Files to install and distribute other than extfs scripts
-EXTFS_MISC = README README.extfs
+EXTFS_MISC =
 
 # Scripts hat don't need adaptation to the local system
 EXTFS_CONST = bpp changesetfs gitfs+ patchsetfs rpm trpm u7z
--- src/vfs/fish/helpers/Makefile.am
+++ src/vfs/fish/helpers/Makefile.am
@@ -1,7 +1,7 @@
 fishdir = $(libexecdir)/@PACKAGE@/fish
 
 # Files to install and distribute other than fish scripts
-FISH_MISC  = README.fish
+FISH_MISC  =
 
 fish_DATA = $(FISH_MISC)
 fish_SCRIPTS = ls mkdir fexists unlink chown chmod rmdir ln mv hardlink get send append info utime
--- src/vfs/fish/helpers/Makefile.in
+++ src/vfs/fish/helpers/Makefile.in
@@ -331,7 +331,7 @@
 fishdir = $(libexecdir)/@PACKAGE@/fish
 
 # Files to install and distribute other than fish scripts
-FISH_MISC = README.fish
+FISH_MISC =
 fish_DATA = $(FISH_MISC)
 fish_SCRIPTS = ls mkdir fexists unlink chown chmod rmdir ln mv hardlink get send append info utime
 fishconfdir = $(sysconfdir)/@PACKAGE@
