--- src/subshell.c	2012-03-02 13:55:52.018954847 +0100
+++ src/subshell.c	2012-03-05 13:59:36.697459979 +0100
@@ -266,11 +266,14 @@
         putenv (g_strdup (sid_str));
     }
 
+    char *putenv_str = NULL;
     switch (subshell_type)
     {
     case BASH:
+        /* Do we have a custom init file ~/.local/share/mc/bashrc? */
         init_file = mc_config_get_full_path ("bashrc");
 
+        /* Otherwise use ~/.bashrc */
         if (access (init_file, R_OK) == -1)
         {
             g_free (init_file);
@@ -285,9 +288,9 @@
             char *input_file = mc_config_get_full_path ("inputrc");
             if (access (input_file, R_OK) == 0)
             {
-                char *putenv_str = g_strconcat ("INPUTRC=", input_file, NULL);
+                putenv_str = g_strconcat ("INPUTRC=", input_file, NULL);
                 putenv (putenv_str);
-                g_free (putenv_str);
+                /* Do not use "g_free (putenv_str)" here, otherwise INPUTRC will be undefined! */
             }
             g_free (input_file);
         }
@@ -350,6 +353,7 @@
 
     /* If we get this far, everything failed miserably */
     g_free (init_file);
+    g_free (putenv_str);
     _exit (FORK_FAILURE);
 }
 
