--- src/args.c
+++ src/args.c
@@ -95,7 +95,9 @@
 #endif /* ENABLE_SUBSHELL */
 static gboolean mc_args__show_datadirs = FALSE;
 static gboolean mc_args__show_datadirs_extended = FALSE;
+#ifdef ENABLE_SHOW_CONFIGURE_OPTS
 static gboolean mc_args__show_configure_opts = FALSE;
+#endif
 
 static GOptionGroup *main_group;
 
@@ -125,6 +127,7 @@
      NULL
     },
 
+#ifdef ENABLE_SHOW_CONFIGURE_OPTS
     /* show configure options */
     {
      "configure-options", '\0', G_OPTION_FLAG_IN_MAIN, G_OPTION_ARG_NONE,
@@ -132,6 +135,7 @@
      N_("Print configure options"),
      NULL
     },
+#endif
 
     {
      "printwd", 'P', G_OPTION_FLAG_IN_MAIN, G_OPTION_ARG_STRING,
@@ -716,11 +720,13 @@
         return FALSE;
     }
 
+#ifdef ENABLE_SHOW_CONFIGURE_OPTS
     if (mc_args__show_configure_opts)
     {
         show_configure_options ();
         return FALSE;
     }
+#endif
 
     return TRUE;
 }
--- src/textconf.h
+++ src/textconf.h
@@ -17,7 +17,9 @@
 
 extern void show_version (void);
 extern void show_datadirs_extended (void);
+#ifdef ENABLE_SHOW_CONFIGURE_OPTS
 extern void show_configure_options (void);
+#endif
 
 /*** inline functions ****************************************************************************/
 #endif /* MC__TEXTCONF_H */
--- src/textconf.c
+++ src/textconf.c
@@ -228,10 +228,12 @@
 
 /* --------------------------------------------------------------------------------------------- */
 
+#ifdef ENABLE_SHOW_CONFIGURE_OPTS
 void
 show_configure_options (void)
 {
     (void) printf ("%s\n", MC_CONFIGURE_ARGS);
 }
+#endif
 
 /* --------------------------------------------------------------------------------------------- */
