--- edit/syntax.c	2007-07-26 17:35:14.000000000 +0200
+++ edit/syntax.c	2007-07-26 17:35:14.000000000 +0200
@@ -1192,12 +1192,10 @@
     f = mhl_str_dir_plus_file (home_dir, SYNTAX_FILE);
     r = edit_read_syntax_file (edit, pnames, f, edit ? edit->filename : 0,
 			       get_first_editor_line (edit), type);
-    if (r == -1) {
-	edit_free_syntax_rules (edit);
-	message (D_ERROR, _(" Load syntax file "),
-		 _(" Cannot open file %s \n %s "), f,
-		 unix_error_string (errno));
-    } else if (r) {
+    if (r == -1)
+	return;
+
+    if (r) {
 	edit_free_syntax_rules (edit);
 	message (D_ERROR, _(" Load syntax file "),
 		 _(" Error in file %s on line %d "),
 
