--- configure.orig	2009-02-01 20:46:23.000000000 +0100
+++ configure	2011-01-12 15:34:39.909614528 +0100
@@ -32487,7 +32487,7 @@
 
 
 # Check whether --with-termcap was given.
-if test "${with_termcap+set}" = set; then
+if test x"$with_termcap" = xyes; then
   withval=$with_termcap; if test x$with_screen != xmcslang; then
 		{ { echo "$as_me:$LINENO: error: Option \`--with-termcap' only works with \`--with-screen=mcslang'" >&5
 echo "$as_me: error: Option \`--with-termcap' only works with \`--with-screen=mcslang'" >&2;}
