diff -ruN po/POTFILES.in.ori po/POTFILES.in
--- po/POTFILES.in.ori	2007-03-16 10:37:50.000000000 +0100
+++ po/POTFILES.in	2007-03-16 10:38:16.000000000 +0100
@@ -6,19 +6,6 @@
 edit/editoptions.c
 edit/editwidget.c
 edit/syntax.c
-gnome/achown.c
-gnome/background.c
-gnome/boxes.c
-gnome/chmod.c
-gnome/chown.c
-gnome/cmd.c
-gnome/command.c
-gnome/dialog.c
-gnome/dir.c
-gnome/ext.c
-gnome/file.c
-gnome/find.c
-gnome/fixhlp.c
 gnome/gcmd.c
 gnome/gdesktop.c
 gnome/glayout.c
@@ -27,25 +14,6 @@
 gnome/gtools.c
 gnome/gview.c
 gnome/gwidget.c
-gnome/help.c
-gnome/hotlist.c
-gnome/i18n.h
-gnome/info.c
-gnome/layout.c
-gnome/learn.c
-gnome/main.c
-gnome/menu.c
-gnome/option.c
-gnome/panelize.c
-gnome/screen.c
-gnome/subshell.c
-gnome/tree.c
-gnome/user.c
-gnome/util.c
-gnome/utilunix.c
-gnome/view.c
-gnome/win.c
-gnome/wtools.c
 src/achown.c
 src/background.c
 src/boxes.c
