--- vfs/extfs/iso9660.in	2009-02-01 20:30:21.000000000 +0100
+++ vfs/extfs/iso9660.in	2010-06-15 00:46:03.916831191 +0200
@@ -9,19 +9,7 @@
 # tested to comply with isoinfo 2.0's output
 
 test_iso () {
-    CHARSET=`locale charmap 2>/dev/null`
-    if test -z "$CHARSET"; then
-	CHARSET=`locale 2>/dev/null | grep LC_CTYPE | sed -n -e 's/.*\.\(.*\)"$/\1/p'`
-    fi
-    if test -n "$CHARSET"; then
-	CHARSET=`echo "$CHARSET" | tr '[A-Z]' '[a-z]' | sed -e 's/^iso-/iso/'`
-	isoinfo -j $CHARSET -i /dev/null 2>&1 | grep "Unknown charset" >/dev/null && CHARSET=
-    fi
-    if test -n "$CHARSET"; then
-	JOLIET_OPT="-j $CHARSET -J"
-    else
 	JOLIET_OPT="-J"
-    fi
     ISOINFO="isoinfo -R"
     isoinfo -d -i "$1" | grep "NO Joliet" > /dev/null || ISOINFO="$ISOINFO $JOLIET_OPT"
 }
