--- libstdc++-v3/crossconfig.m4.orig	2007-07-29 05:31:45.000000000 +0200
+++ libstdc++-v3/crossconfig.m4	2007-07-29 05:31:47.000000000 +0200
@@ -156,13 +156,6 @@
 
     # For LFS.
     AC_DEFINE(HAVE_INT64_T)
-    case "$target" in
-      *-uclinux*)
-        # Don't enable LFS with uClinux
-        ;;
-      *)
-        AC_DEFINE(_GLIBCXX_USE_LFS)
-    esac
 
     # For showmanyc_helper().
     AC_CHECK_HEADERS(sys/ioctl.h sys/filio.h)
--- libstdc++-v3/configure.orig	2007-07-29 05:31:45.000000000 +0200
+++ libstdc++-v3/configure	2007-07-29 05:31:47.000000000 +0200
@@ -78540,16 +78540,6 @@
 #define HAVE_INT64_T 1
 _ACEOF
 
-    case "$target" in
-      *-uclinux*)
-        # Don't enable LFS with uClibc
-        ;;
-      *)
-        cat >>confdefs.h <<\_ACEOF
-#define _GLIBCXX_USE_LFS 1
-_ACEOF
-
-    esac
 
     # For showmanyc_helper().
 
