From 00c1c0ac1fb9258747c56ff77a4a35beb44bd994 Mon Sep 17 00:00:00 2001
From: Simon McVittie <simon.mcvittie@collabora.co.uk>
Date: Mon, 01 Jul 2013 11:14:02 +0000
Subject: Make the test for #53499 more obviously correct

---
diff --git dbus/dbus-address.c dbus/dbus-address.c
index 6506d21..18f0094 100644
--- dbus/dbus-address.c
+++ dbus/dbus-address.c
@@ -680,7 +680,7 @@ static const EscapeTest escape_tests[] = {
   { "i", "i" },
   { "z", "z" },
   /* Bug: https://bugs.freedesktop.org/show_bug.cgi?id=53499 */
-  { "%c3%b6", "\303\266" }
+  { "%c3%b6", "\xc3\xb6" }
 };
 
 static const char* invalid_escaped_values[] = {
