--- configure
+++ configure
@@ -18684,7 +18684,7 @@
 fi
 
 
-for ac_func in vsnprintf vasprintf nanosleep usleep setenv clearenv unsetenv socketpair getgrouplist fpathconf setrlimit poll setlocale localeconv strtoll strtoull issetugid getresuid
+for ac_func in vsnprintf vasprintf nanosleep usleep setenv clearenv unsetenv socketpair getgrouplist fpathconf setrlimit      setlocale localeconv strtoll strtoull issetugid getresuid
 do :
   as_ac_var=`$as_echo "ac_cv_func_$ac_func" | $as_tr_sh`
 ac_fn_c_check_func "$LINENO" "$ac_func" "$as_ac_var"
@@ -18766,6 +18766,7 @@
 
 
 
+if false; then
 #### Check for broken poll; taken from Glib's configure
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for broken poll" >&5
@@ -18809,6 +18810,7 @@
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $broken_poll" >&5
 $as_echo "$broken_poll" >&6; }
+fi
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for dirfd" >&5
 $as_echo_n "checking for dirfd... " >&6; }
