--- configure
+++ configure
@@ -6639,7 +6639,7 @@
 
 
 
-for ac_header in getopt.h libintl.h limits.h regex.h sys/param.h wchar.h wctype.h stdarg.h magic.h
+for ac_header in getopt.h limits.h regex.h sys/param.h wchar.h wctype.h stdarg.h magic.h
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
 ac_fn_c_check_header_mongrel "$LINENO" "$ac_header" "$as_ac_Header" "$ac_includes_default"
