--- configure.orig	2010-08-07 16:28:49.966221396 +0200
+++ configure	2010-08-07 16:28:50.016248844 +0200
@@ -6758,7 +6758,7 @@
 
 fi
 
-for ac_header in getopt.h libintl.h limits.h regex.h sys/param.h wchar.h wctype.h stdarg.h
+for ac_header in getopt.h limits.h regex.h sys/param.h wchar.h wctype.h stdarg.h
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
 ac_fn_c_check_header_mongrel "$LINENO" "$ac_header" "$as_ac_Header" "$ac_includes_default"
