--- src/files.c.orig	2010-06-24 16:45:54.000000000 +0200
+++ src/files.c	2010-08-06 17:27:51.000000000 +0200
@@ -1505,8 +1505,10 @@
      * specified it interactively), stat and save the value
      * or else we will chase null pointers when we do
      * modtime checks, preserve file times, etc. during backup */
-    if (openfile->current_stat == NULL && !tmp && realexists)
-	stat(realname, openfile->current_stat);
+    if (openfile->current_stat == NULL && !tmp && realexists) {
+        openfile->current_stat = (struct stat *)nmalloc(sizeof(struct stat));
+        stat(realname, openfile->current_stat);
+    }
 
     /* We backup only if the backup toggle is set, the file isn't
      * temporary, and the file already exists.  Furthermore, if we
