--- Rules.mak
+++ Rules.mak
@@ -270,7 +270,7 @@
 # If -msoft-float isn't supported, we want an error anyway.
 # Hmm... might need to revisit this for arm since it has 2 different
 # soft float encodings.
-    CPU_CFLAGS += -msoft-float
+    CPU_CFLAGS += $(if $(strip $(CUSTOM_FLOAT_ABI_CFLAGS)),$(strip $(CUSTOM_FLOAT_ABI_CFLAGS)),-msoft-float)
 ifeq ($(strip $(TARGET_ARCH)),arm)
 # No longer needed with current toolchains, but leave it here for now.
 # If anyone is actually still using gcc 2.95 (say), they can uncomment it.
