--- /dev/null	2009-12-14 23:28:50.862941706 +0100
+++ SysMachine.h	2009-12-15 09:06:33.000000000 +0100
@@ -0,0 +1,24 @@
+#ifndef _MACHDEFS_H
+#define _MACHDEFS_H
+
+
+#undef MACH_BIG_ENDIAN_WORDS
+
+#undef MACH_BIG_ENDIAN_BITFIELD
+
+typedef signed char MachInt8;
+typedef unsigned char MachUInt8;
+#define MACH_TYPE_8BIT char
+
+typedef signed short MachInt16;
+typedef unsigned short MachUInt16;
+#define MACH_TYPE_16BIT short
+
+typedef signed int MachInt32;
+typedef unsigned int MachUInt32;
+#define MACH_TYPE_32BIT int
+
+
+
+#endif
+
