--- Makefile.lnx.orig	2010-02-26 04:33:44.000000000 +0100
+++ Makefile.lnx	2010-05-01 09:33:52.000000000 +0200
@@ -44,12 +44,12 @@
 MAINSRC = MainLinux.cpp
 SYSSRCS = SysDepLinux.cpp SysDepUnix.cpp
 
-ifeq ($(wildcard /usr/include/sys/eventfd.h), )
+#ifeq ($(wildcard /usr/include/sys/eventfd.h), )
     SYSSRCS := $(SYSSRCS) SysOsEventfd_pipe.cpp
-else
-    SYSSRCS := $(SYSSRCS) SysOsEventfd_eventfd.cpp
-    CFLAGS := $(CFLAGS) -DHAS_EVENTFD
-endif
+#else
+#    SYSSRCS := $(SYSSRCS) SysOsEventfd_eventfd.cpp
+#    CFLAGS := $(CFLAGS) -DHAS_EVENTFD
+#endif
 
 
 CFLAGS := $(CFLAGS) -I. -D__UNIX__ -D__LINUX__ -D_REENTRANT=1 -D_THREAD_SAFE=1 -DHAS_SYSMACHINE \
