--- MailRoot/aliases.tab	2009-06-27 21:05:46.000000000 +0200
+++ MailRoot/aliases.tab	2009-12-29 12:01:58.000000000 +0100
@@ -1,2 +0,0 @@
-"xmailserver.test"	"root"	"xmailuser"
-"xmailserver.test"	"postmaster"	"xmailuser"
--- MailRoot/domains.tab	2009-06-27 21:05:46.000000000 +0200
+++ MailRoot/domains.tab	2009-12-29 12:02:22.000000000 +0100
@@ -1 +0,0 @@
-"xmailserver.test"
--- MailRoot/mailusers.tab	2009-06-27 21:05:46.000000000 +0200
+++ MailRoot/mailusers.tab	2009-12-29 12:02:34.000000000 +0100
@@ -1 +0,0 @@
-"xmailserver.test"	"xmailuser"	"1d08040c09"	1	"xmailuser"	"U"
