--- etc/default.Fritz_Box_SpeedportW501V/avm/ar7.cfg	2007-06-04 16:20:05.000000000 +0200
+++ etc/default.Fritz_Box_SpeedportW501V/avm/ar7.cfg	2007-07-11 18:57:28.000000000 +0200
@@ -199,13 +199,13 @@
                 name = "internet";
                 only_crypt_auth = no;
                 local {
-                        username = "";
-                        passwd = "";
+                        username = "5200550445890001@setup.t-online.de";
+                        passwd = "setup";
                 }
                 remoteauth = pppcfg_authtype_chap;
                 remote {
                 }
-                inactivity_timeout = 5m;
+                inactivity_timeout = 3m;
                 bProviderDisconnectPrevention = yes;
                 ProviderDisconnectPreventionInterval = 1d;
                 ProviderDisconnectPreventionHour = 3;
@@ -391,7 +391,7 @@
                 shaperconnection {
                         enabled = yes;
                         targets = "";
-                        name = "Fritz!Box";
+                        name = "Speedport";
                         limiters {
                                 name = "default-out";
                                 bps_limit {
@@ -511,8 +511,42 @@
         }
 }
 
+webui {
+    password = "0000";
+}
+
+ntpclient {
+    server_list = "ntp1.t-online.de", "ptbtime1.ptb.de";
+}
+
+mrouter {
+    igmp_prio = 32;
+}
+
+TR_064 {
+    enabled = yes;
+    username = "dslf-config";
+    password = "0000";
+}
+
 qos {
-        always_set_tos = no;
+        always_set_tos = yes;
+        tcclasstosmap {
+        	tcclass = "sipdns";
+                tos = 192;
+        } {
+        	tcclass = "sip";
+                tos = 192;
+        } {
+        	tcclass = "rtp";
+                tos = 160;
+        } {
+        	tcclass = "igmp";
+                tos = 128;
+        } {
+        	tcclass = "mstvunicast";
+                tos = 96;
+        }
         rules {
            tcclass = "sip";
            rules = "udp.dport 5060";
@@ -530,6 +564,9 @@
         } {
            tcclass = "icmp";
            rules = "ip.proto == icmp";
+        } {
+           tcclass = "mstvunicast";
+           rules = "udp.dport 43962", "udp.dport 47806";
         }
 }
 
