--- linux-2.6.19.2/drivers/net/avm_cpmac/cpmac_if.c.orig	2007-12-04 17:52:56.000000000 +0100
+++ linux-2.6.19.2/drivers/net/avm_cpmac/cpmac_if.c	2008-01-27 00:36:54.000000000 +0100
@@ -20,7 +20,7 @@
 #include <linux/skbuff.h>
 #include <linux/netdevice.h>
 #include <linux/etherdevice.h>
-/*--- #include <linux/if_vlan.h> ---*/ /* TODO Can this line be removed? */
+#include <linux/if_vlan.h>
 #include <linux/proc_fs.h>
 #include <asm/io.h>
 #if !defined(CONFIG_NETCHIP_ADM69961)
