--- Configure.orig	2009-11-24 14:43:38.000000000 +0100
+++ Configure	2009-11-24 14:45:37.000000000 +0100
@@ -2542,7 +2542,6 @@
 # Configure for Linux.
 
   linux)
-    LSOF_TSTBIGF="-D_FILE_OFFSET_BITS=64"
     LSOF_TSTKMEM=0
     if test "X$LSOF_CC" = "X"	# {
     then
@@ -2626,9 +2625,12 @@
 
     # Test for IPv6 support.
 
-    if test -r ${LSOF_INCLUDE}/netinet/ip6.h	# {
+    if test "X$LINUX_HASIPV6" = "XY" -o "X$LINUX_HASIPV6" = "xY" # {
     then
-      LSOF_CFGF="$LSOF_CFGF -DHASIPv6"
+      if test -r ${LSOF_INCLUDE}/netinet/ip6.h	# {
+      then
+        LSOF_CFGF="$LSOF_CFGF -DHASIPv6"
+      fi	# }
     fi	# }
 
     # Test for SELinux support.
