--- Configure
+++ Configure
@@ -189,7 +189,7 @@
 LSOF_TMP2=""
 LSOF_TMPC_BASE=./lsof_Configure_tmp_
 LSOF_TMPC=${LSOF_TMPC_BASE}$$
-LSOF_TSTBIGF=""
+#LSOF_TSTBIGF=""
 LSOF_TSTSUBD="./tests"
 LSOF_TSTCC="${LSOF_TSTSUBD}/config.cc"
 LSOF_TSTCFLG="${LSOF_TSTSUBD}/config.cflags"
@@ -2786,7 +2786,7 @@
 # Configure for Linux.
 
   linux)
-    LSOF_TSTBIGF="-D_FILE_OFFSET_BITS=64"
+    #LSOF_TSTBIGF="-D_FILE_OFFSET_BITS=64"
     LSOF_TSTKMEM=0
     if test "X$LSOF_CC" = "X"	# {
     then
@@ -2942,7 +2942,7 @@
       LSOF_CFGF="$LSOF_CFGF -DHASUXSOCKEPT"
     fi	# }
     LSOF_DIALECT_DIR="linux"
-    LSOF_CFGF="$LSOF_CFGF -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE"
+    #LSOF_CFGF="$LSOF_CFGF -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE"
     ;;
 
 # Configure for NetBSD.
