--- configure	2007-02-12 02:10:24.000000000 +0100
+++ configure	2007-02-12 02:06:57.000000000 +0100
@@ -78866,7 +78866,11 @@
   echo $ac_n "checking for working ttyname_r() implementation""... $ac_c" 1>&6
 echo "configure:78868: checking for working ttyname_r() implementation" >&5
   if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
+    #evaluated by running test program on target platform
+    echo "$ac_t""yes" 1>&6
+    cat >> confdefs.h <<\EOF
+#define HAVE_TTYNAME_R 1
+EOF
 else
   cat > conftest.$ac_ext <<EOF
 #line 78873 "configure"
