--- configure
+++ configure
@@ -9267,7 +9267,7 @@
 
 { echo "$as_me:$LINENO: checking for /proc filesystem" >&5
 echo $ECHO_N "checking for /proc filesystem... $ECHO_C" >&6; }
-if test -d /proc; then
+if true || test -d /proc; then
 	{ echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
 
@@ -9282,7 +9282,7 @@
 
 { echo "$as_me:$LINENO: checking for file /proc/stat" >&5
 echo $ECHO_N "checking for file /proc/stat... $ECHO_C" >&6; }
-if test -r /proc/stat; then
+if true || test -r /proc/stat; then
 	{ echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
 
@@ -9297,7 +9297,7 @@
 
 { echo "$as_me:$LINENO: checking for file /proc/cpuinfo" >&5
 echo $ECHO_N "checking for file /proc/cpuinfo... $ECHO_C" >&6; }
-if test -r /proc/cpuinfo; then
+if true || test -r /proc/cpuinfo; then
 	{ echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
 
@@ -9312,7 +9312,7 @@
 
 { echo "$as_me:$LINENO: checking for file /proc/0/psinfo" >&5
 echo $ECHO_N "checking for file /proc/0/psinfo... $ECHO_C" >&6; }
-if test -r /proc/0/psinfo; then
+if false && test -r /proc/0/psinfo; then
 	{ echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
 
@@ -9327,7 +9327,7 @@
 
 { echo "$as_me:$LINENO: checking for file /proc/loadavg" >&5
 echo $ECHO_N "checking for file /proc/loadavg... $ECHO_C" >&6; }
-if test -r /proc/loadavg; then
+if true || test -r /proc/loadavg; then
 	{ echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
 
@@ -9342,7 +9342,7 @@
 
 { echo "$as_me:$LINENO: checking for file /proc/net/dev" >&5
 echo $ECHO_N "checking for file /proc/net/dev... $ECHO_C" >&6; }
-if test -r /proc/net/dev; then
+if true || test -r /proc/net/dev; then
 	{ echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
 
