--- configure
+++ configure
@@ -7250,7 +7250,7 @@
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for /proc filesystem" >&5
 $as_echo_n "checking for /proc filesystem... " >&6; }
-if test -d /proc; then
+if true || test -d /proc; then
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
 
@@ -7263,7 +7263,7 @@
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for file /proc/stat" >&5
 $as_echo_n "checking for file /proc/stat... " >&6; }
-if test -r /proc/stat; then
+if true || test -r /proc/stat; then
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
 
@@ -7276,7 +7276,7 @@
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for file /proc/cpuinfo" >&5
 $as_echo_n "checking for file /proc/cpuinfo... " >&6; }
-if test -r /proc/cpuinfo; then
+if true || test -r /proc/cpuinfo; then
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
 
@@ -7289,7 +7289,7 @@
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for file /proc/0/psinfo" >&5
 $as_echo_n "checking for file /proc/0/psinfo... " >&6; }
-if test -r /proc/0/psinfo; then
+if false && test -r /proc/0/psinfo; then
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
 
@@ -7302,7 +7302,7 @@
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for file /proc/loadavg" >&5
 $as_echo_n "checking for file /proc/loadavg... " >&6; }
-if test -r /proc/loadavg; then
+if true || test -r /proc/loadavg; then
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
 
@@ -7315,7 +7315,7 @@
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for file /proc/net/dev" >&5
 $as_echo_n "checking for file /proc/net/dev... " >&6; }
-if test -r /proc/net/dev; then
+if true || test -r /proc/net/dev; then
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
 
