--- configure
+++ configure
@@ -7327,6 +7327,8 @@
 fi
 
 
+# HAVE_LONG_LONG_QU remains unset
+if false; then
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for long long format" >&5
 $as_echo_n "checking for long long format... " >&6; }
 if test "$cross_compiling" = yes; then :
@@ -7361,6 +7363,7 @@
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
 fi
+fi
 
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for -rdynamic linking option" >&5
