--- configure
+++ configure
@@ -4453,7 +4453,7 @@
   sys/resource.h pthread.h windows.h process.h conio.h sys/wait.h regex.h \
   stdarg.h winsock2.h pdh.h psapi.h sys/sem.h sys/ipc.h sys/shm.h Winldap.h \
   sys/timeb.h Winber.h lber.h ws2tcpip.h inttypes.h sys/file.h grp.h \
-  execinfo.h libperfstat.h sys/systemcfg.h sys/mnttab.h mntent.h sys/times.h \
+             libperfstat.h sys/systemcfg.h sys/mnttab.h mntent.h sys/times.h \
   dlfcn.h sys/utsname.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
@@ -5677,6 +5677,7 @@
 fi
 
 
+if false; then
 { echo "$as_me:$LINENO: checking for library containing backtrace_symbols" >&5
 echo $ECHO_N "checking for library containing backtrace_symbols... $ECHO_C" >&6; }
 if test "${ac_cv_search_backtrace_symbols+set}" = set; then
@@ -5763,6 +5764,7 @@
 _ACEOF
 
 fi
+fi
 
 
 
