--- configure
+++ configure
@@ -4932,7 +4932,7 @@
 echo "$as_me:$LINENO: result: $ac_cv_lib_ncurses_use_default_colors" >&5
 echo "${ECHO_T}$ac_cv_lib_ncurses_use_default_colors" >&6
 if test $ac_cv_lib_ncurses_use_default_colors = yes; then
-  LDFLAGS="$LDFLAGS -lncurses"; CFLAGS="$CFLAGS -D_HAVE_NCURSES -D_HAVE_NCURSES_COLOR"; color_support="yes"; ncurses_support="yes"
+     LIBS="$LIBS    -lncurses"; CFLAGS="$CFLAGS -D_HAVE_NCURSES -D_HAVE_NCURSES_COLOR"; color_support="yes"; ncurses_support="yes"
 else
   echo "$as_me:$LINENO: checking for use_default_colors in -lncurses" >&5
 echo $ECHO_N "checking for use_default_colors in -lncurses... $ECHO_C" >&6
@@ -4999,7 +4999,7 @@
 echo "$as_me:$LINENO: result: $ac_cv_lib_ncurses_use_default_colors" >&5
 echo "${ECHO_T}$ac_cv_lib_ncurses_use_default_colors" >&6
 if test $ac_cv_lib_ncurses_use_default_colors = yes; then
-  LDFLAGS="$LDFLAGS -lncurses"; CFLAGS="$CFLAGS -D_HAVE_NCURSES"; ncurses_support="yes"; echo "NO TRANSPARENCY SUPPORT in this ncurses lib"
+     LIBS="$LIBS    -lncurses"; CFLAGS="$CFLAGS -D_HAVE_NCURSES"; ncurses_support="yes"; echo "NO TRANSPARENCY SUPPORT in this ncurses lib"
 else
   echo "$as_me:$LINENO: checking for initscr in -lcurses" >&5
 echo $ECHO_N "checking for initscr in -lcurses... $ECHO_C" >&6
