--- usr/www/avm/html/de/fon/sperre1.js.orig	2009-03-17 18:29:26.000000000 +0100
+++ usr/www/avm/html/de/fon/sperre1.js	2009-03-22 01:11:06.000000000 +0100
@@ -68,7 +68,7 @@
 var nummer="";
 if (jslGetValue("uiViewBereichIncoming") == "")
 nummer = jslGetValue("uiViewNummer");
-if (!isZahlRaute(nummer)) {alert(g_mldNonDigits); return;}
+//if (!isZahlRaute(nummer)) {alert(g_mldNonDigits); return;}
 var x = IstActionVorhanden(nummer);
 if (x != "") {
 if (jslGetValue("uiPostInAction"+x) == "1") {
@@ -161,7 +161,7 @@
 }
 } else {
 var nummer = jslGetValue("uiViewNummer");
-if (!isZahlRaute(nummer)) {alert(g_mldNonDigits); return;}
+//if (!isZahlRaute(nummer)) {alert(g_mldNonDigits); return;}
 var x = IstActionVorhanden(nummer);
 if (x != "") {
 if (jslGetValue("uiPostInAction"+x) == "1") {alert(g_txtAlert_1); return;}
