--- configure
+++ configure
@@ -18994,6 +18994,7 @@
 fi
 
 
+if false; then
 echo "$as_me:$LINENO: checking for gethostbyname in -lnsl" >&5
 echo $ECHO_N "checking for gethostbyname in -lnsl... $ECHO_C" >&6
 if test "${ac_cv_lib_nsl_gethostbyname+set}" = set; then
@@ -19066,6 +19067,7 @@
   LIBS="-lnsl $LIBS"
 
 fi
+fi
 
 echo "$as_me:$LINENO: checking for dlopen in -ldl" >&5
 echo $ECHO_N "checking for dlopen in -ldl... $ECHO_C" >&6
