--- signimage/avm_pubkey_to_pkcs8
+++ signimage/avm_pubkey_to_pkcs8
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 ################################################################################
 #                                                                              #
 # convert AVM's format of a public key file to a PEM file usable for "openssl" #
--- signimage/check_signed_image
+++ signimage/check_signed_image
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 ####################################################################################
 #                                                                                  #
 # check, if the specified image contains an AVM-like signature and if it is signed #
--- signimage/generate_signing_key
+++ signimage/generate_signing_key
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 #######################################################################################
 #                                                                                     #
 # generate a RSA key usable to sign a firmware image, which may pass the signature    #
--- signimage/sign_image
+++ signimage/sign_image
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 ####################################################################################
 #                                                                                  #
 # sign a tar archive with our private RSA key                                      #
