--- signimage/image_signing_files.inc
+++ signimage/image_signing_files.inc
@@ -25,7 +25,7 @@
 #                                                        #
 ##########################################################
 [ -z "$name_prefix" ] && name_prefix="${FREETZ_IMAGE_SIGNING_PREFIX}"
-name_prefix="${name_prefix:-$HOME/image_signing}"
+name_prefix="${name_prefix:-$HOME/.freetz.image_signing}"
 ##########################################################
 #                                                        #
 # the extension of the private key file - PEM format,    #
