--- signimage/sign_image
+++ signimage/sign_image
@@ -139,7 +139,6 @@
 		printf "%d" $sum
 	}
 	sum=0
-	cp "$1" /var/tmp/savedheader
 	cmp -l -- "$1" /dev/zero 2>/dev/null | read_values
 }
 ####################################################################################
@@ -371,7 +370,6 @@
 # needed                                                                           #
 #                                                                                  #
 ####################################################################################
-fill_blocks=0
 if [ $(( ( copy_blocks + 2 ) % 20 )) -eq 0 ]; then
 	echo -ne "Repeating first entry as filler ... " 1>&2
 	"$YF_SIGNIMAGE_DD" if="$image_file" of="$tmp/filler.bin" bs=512 count=1 status=none 2>/dev/null
