--- signimage/sign_image
+++ signimage/sign_image
@@ -421,8 +421,11 @@
 # and append another TAR archive containing only our newly created signature file  #
 #                                                                                  #
 ####################################################################################
-"$YF_SIGNIMAGE_TAR" --version 2>/dev/null | grep -q GNU 2>/dev/null && format=" --format=gnu" || format=""
-"$YF_SIGNIMAGE_TAR" $format -C "$tmp" -O -c ./var/signature | "$YF_SIGNIMAGE_DD" bs=512 count=4 status=none 2>/dev/null
+"$YF_SIGNIMAGE_TAR" --version 2>/dev/null | grep -q GNU 2>/dev/null && {
+	format="--format=gnu"
+	owner="--owner=0 --group=0"
+}
+"$YF_SIGNIMAGE_TAR" $format $owner -C "$tmp" -O -c ./var/signature | "$YF_SIGNIMAGE_DD" bs=512 count=4 status=none 2>/dev/null
 show_ok
 ####################################################################################
 #                                                                                  #
