--- eva_tools/eva_discover
+++ eva_tools/eva_discover
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 ###########################################################################
 # USAGE-BEGIN
 #
--- eva_tools/eva_get_environment
+++ eva_tools/eva_get_environment
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 #
 # read environment from FRITZ!Box 
 #
--- eva_tools/eva_store_tffs
+++ eva_tools/eva_store_tffs
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 #
 # file to transfer to the router
 #
--- eva_tools/eva_switch_system
+++ eva_tools/eva_switch_system
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 [ ${#2} -eq 0 -o ${#3} -ne 0 ] && echo "Calling: $0 <interface> <ip address to use>" 2>&1 && exit 1
 #
 # some constants to be changed, if needed
--- eva_tools/eva_to_memory
+++ eva_tools/eva_to_memory
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 #
 # file to transfer to the router
 #
--- eva_tools/image2ram
+++ eva_tools/image2ram
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 if [ -t 0 -o -t 1 ]; then
 	echo "I'm a filter, give me an image file on STDIN and I'll put out an in-memory image on STDOUT." 1>&2
 	exit 2
