From 24eb1cd22451cd5048f686388041996673ce7e44 Mon Sep 17 00:00:00 2001
From: Eugene Rudoy <gene.devel@gmail.com>
Date: Mon, 15 Oct 2018 21:53:56 +0200
Subject: [PATCH] gui_bootmanager: add support for "fixed branding"-approach used in Freetz

Signed-off-by: Eugene Rudoy <gene.devel@gmail.com>
---
 bootmanager/gui_bootmanager | 1 +
 1 file changed, 1 insertion(+)

--- bootmanager/bootmanager
+++ bootmanager/bootmanager
@@ -639,6 +639,7 @@
 (
 	path="${1:-/}${1:+/}$avm_rcconf"
 	b="$(sed -n -e "s|^export OEM\(=\(['\"]\?\)\(.*\)\2\)\?\$|\3|p" "$path")"
+	[ -z "$b" ] && b="$(sed -n -e 's|^OEM_tmp=\("\?\)\([A-Za-z0-9_-]\+\)\1$|\2|p' "$path")" # Freetz approach
 	printf -- '%s' "$b"
 	[ -z "$b" ] && exit 1 || exit 0
 )
