From 24eb1cd22451cd5048f686388041996673ce7e44 Mon Sep 17 00:00:00 2001
From: Eugene Rudoy <gene.devel@gmail.com>
Date: Mon, 15 Oct 2018 21:53:56 +0200
Subject: [PATCH] gui_bootmanager: add support for "fixed branding"-approach used in Freetz

Signed-off-by: Eugene Rudoy <gene.devel@gmail.com>
---
 bootmanager/gui_bootmanager | 1 +
 1 file changed, 1 insertion(+)

diff --git bootmanager/gui_bootmanager bootmanager/gui_bootmanager
index 2a5f6e3..7a9a618 100755
--- bootmanager/gui_bootmanager
+++ bootmanager/gui_bootmanager
@@ -708,6 +708,7 @@ get_fixed_branding()
 (
 	path="${1:-/}${1:+/}$avm_rcconf"
 	b="$(sed -n -e 's|^export OEM\(=\"\?\(.*\)\"\?\)\?$|\2|p' "$path")"
+	[ -z "$b" ] && b="$(sed -n -e 's|^OEM_tmp=\("\?\)\([A-Za-z0-9_-]\+\)\1$|\2|p' "$path")" # Freetz approach
 	printf "%s" "$b"
 	[ -z "$b" ] && exit 1 || exit 0
 )
