--- avm_kernel_config/unpack_kernel.sh
+++ avm_kernel_config/unpack_kernel.sh
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 #######################################################################################################
 #                                                                                                     #
 # unpack LZMA compressed FRITZ!OS kernel                                                              #
