--- signimage/sign_image
+++ signimage/sign_image
@@ -343,14 +343,6 @@
 fi
 ####################################################################################
 #                                                                                  #
-# prepare a temporary directory and cleanup on exit                                #
-#                                                                                  #
-####################################################################################
-tmp=$(mktemp -d)
-[ $? -eq 127 ] && tmp="/tmp/tmp.$(date +%s).$$" && mkdir -p "$tmp"
-trap "rm -r \"$tmp\"" EXIT HUP
-####################################################################################
-#                                                                                  #
 # create a subdirectory 'var' to prepare a TAR file of our signature later         #
 #                                                                                  #
 ####################################################################################
