--- bootmanager/add_to_system_reboot.sh
+++ bootmanager/add_to_system_reboot.sh
@@ -149,14 +149,14 @@
 if check_version $major $minor 7 8; then
 	printf "      Patching file '%s' ...\n" "$LuaFile" 1>&2
 	getLuaPatchText_pre0708 > "$TMP/gui_bootmanager_0_6_tmp"
-	sed -f "$TMP/gui_bootmanager_0_6_tmp" -i "$TargetDir$LuaFile"
+	tools/sed -f "$TMP/gui_bootmanager_0_6_tmp" -i "$TargetDir$LuaFile"
 	rm "$TMP/gui_bootmanager_0_6_tmp"
 else
 	printf "      Patching file '%s' ...\n" "$JsFile" 1>&2
 	getJsPatchText_0708 > $TMP/gui_bootmanager_0_6_tmp
-	sed -f "$TMP/gui_bootmanager_0_6_tmp" -i "$TargetDir$JsFile"
+	tools/sed -f "$TMP/gui_bootmanager_0_6_tmp" -i "$TargetDir$JsFile"
 	printf "      Patching file '%s' ...\n" "$LuaFile" 1>&2
 	getLuaPatchText_0708 > $TMP/gui_bootmanager_0_6_tmp
-	sed -f "$TMP/gui_bootmanager_0_6_tmp" -i "$TargetDir$LuaFile"
+	tools/sed -f "$TMP/gui_bootmanager_0_6_tmp" -i "$TargetDir$LuaFile"
 	rm "$TMP/gui_bootmanager_0_6_tmp"
 fi
