--- squashfs-tools/Makefile
+++ squashfs-tools/Makefile
@@ -24,14 +24,14 @@
 LINKER = $(if $(findstring ++,$(LZMA_LIBNAME)),$(CXX),$(CC))
 
 mksquashfs-lzma: mksquashfs-lzma.o read_fs-lzma.o sort-lzma.o
-	$(LINKER) $^ $(if $(strip $(LZMA_DIR)),-L$(strip $(LZMA_DIR))) -l$(strip $(LZMA_LIBNAME)) -o $@
+	$(LINKER) $(LDFLAGS) $^ $(if $(strip $(LZMA_DIR)),-L$(strip $(LZMA_DIR))) -l$(strip $(LZMA_LIBNAME)) -o $@
 
 mksquashfs-lzma.o: mksquashfs.c mksquashfs.h global.h
 read_fs-lzma.o: read_fs.c read_fs.h global.h
 sort-lzma.o: sort.c global.h
 
 unsquashfs-lzma: unsquashfs-lzma.o
-	$(LINKER) $^ $(if $(strip $(LZMA_DIR)),-L$(strip $(LZMA_DIR))) -l$(strip $(LZMA_LIBNAME)) -o $@
+	$(LINKER) $(LDFLAGS) $^ $(if $(strip $(LZMA_DIR)),-L$(strip $(LZMA_DIR))) -l$(strip $(LZMA_LIBNAME)) -o $@
 
 unsquashfs-lzma.o: unsquashfs.c squashfs_fs.h read_fs.h global.h
 
