--- etc/init.d/rc.S.orig	2007-09-07 18:09:12.000000000 +0200
+++ etc/init.d/rc.S	2007-09-28 16:20:05.000000000 +0200
@@ -239,6 +239,8 @@
  mknod /var/flash/calllog c $tffs_major $((0x8D))
  mknod /var/flash/aura-usb c $tffs_major $((0xA0))
  mknod /var/flash/browser-data c $tffs_major $((0xA8))
+	mknod /var/flash/freetz c $tffs_major $((0x3c))
+	/usr/bin/modload 2>&1 | tee /var/log/mod_load.log
 #file "stdin", 9
 #file "./arch/etc/init.d/rc.default.sh", 0
 echo 1 > /proc/sys/net/ipv4/ip_forward
