--- etc/default.Fritz_Box_SpeedportW701V/avm/ar7.cfg	2007-07-10 13:48:27.000000000 +0200
+++ etc/default.Fritz_Box_SpeedportW701V/avm/ar7.cfg	2007-07-11 18:18:39.000000000 +0200
@@ -199,13 +199,13 @@
                 name = "internet";
                 only_crypt_auth = no;
                 local {
-                        username = "";
-                        passwd = "";
+                        username = "5200550445890001@setup.t-online.de";
+                        passwd = "setup";
                 }
                 remoteauth = pppcfg_authtype_chap;
                 remote {
                 }
-                inactivity_timeout = 5m;
+                inactivity_timeout = 3m;
                 bProviderDisconnectPrevention = yes;
                 ProviderDisconnectPreventionInterval = 1d;
                 ProviderDisconnectPreventionHour = 3;
@@ -391,7 +391,7 @@
                 shaperconnection {
                         enabled = yes;
                         targets = "";
-                        name = "Fritz!Box";
+                        name = "Speedport";
                         limiters {
                                 name = "default-out";
                                 bps_limit {
@@ -470,28 +470,28 @@
                                 shapeOutgoingOnIncomingCounters = no;
                         }
                         out_rules {
-                                name = "fon-rtp";
-                                filter = "tc:rtp,sip";
-                                priority = 3;
-                                limiters = "default-out";
-                        } {
                                 name = "download-tcp-ack";
-                                filter = "tc:tcp_ack";
+                                filter = "tcp and len <= 64";
                                 priority = 2;
                                 limiters = "default-out";
                         } {
                                 name = "dns";
-                                filter = "tc:dns,sipdns,ntpdns,tr069dns";
+                                filter = "udp port 53";
                                 priority = 1;
                                 limiters = "default-out";
                         } {
+                                name = "fon-rtp";
+                                filter = "udp[8] = 0x80 or udp port 5060";
+                                priority = 3;
+                                limiters = "default-out";
+                        } {
                                 name = "http-requests";
-                                filter = "tc:http_req,tr069";
+                                filter = "(tcp dst port 80 or dst port 8080 or dst port 3128) and (len <= 800)";
                                 priority = 1;
                                 limiters = "default-out";
                         } {
                                 name = "pri-out";
-                                filter = "tc:icmp";
+                                filter = "icmp";
                                 priority = 1;
                                 limiters = "default-out";
                         } {
@@ -511,26 +511,16 @@
         }
 }
 
-qos {
-        always_set_tos = no;
-        rules {
-           tcclass = "sip";
-           rules = "udp.dport 5060";
-        } {
-           tcclass = "dns";
-           rules = "udp.dport 53";
-        } {
-           tcclass = "http_req";
-           rules = "tcp.dport 80 ip.len <= 800", 
-                   "tcp.dport 8080 ip.len <= 800", 
-                   "tcp.dport 3128 ip.len <= 800";
-        } {
-           tcclass = "tcp_ack";
-           rules = "ip.proto == tcp ip.len <= 64";
-        } {
-           tcclass = "icmp";
-           rules = "ip.proto == icmp";
-        }
+webui {
+    password = "0000";
+}
+
+ntpclient {
+    server_list = "ntp1.t-online.de", "ptbtime1.ptb.de";
+}
+
+mrouter {
+    igmp_prio = 32;
 }
 
 // EOF
