--- configure
+++ configure
@@ -101035,8 +101035,12 @@
 echo $ECHO_N "checking for working ttyname_r() implementation... $ECHO_C" >&6; }
   if test "$cross_compiling" = yes; then
 
-    { echo "$as_me:$LINENO: result: no, cannot detect working ttyname_r() when cross compiling. posix_ttyname() will be thread-unsafe" >&5
-echo "${ECHO_T}no, cannot detect working ttyname_r() when cross compiling. posix_ttyname() will be thread-unsafe" >&6; }
+#evaluated by running test program on target platform
+{ echo "$as_me:$LINENO: result: yes" >&5
+echo "${ECHO_T}yes" >&6; }
+cat >>confdefs.h <<\_ACEOF
+#define HAVE_TTYNAME_R 1
+_ACEOF
 
 else
   cat >conftest.$ac_ext <<_ACEOF
