--- extra/Configs/Config.mips.orig	2010-04-02 17:34:27.000000000 +0200
+++ extra/Configs/Config.mips	2010-04-11 16:34:43.000000000 +0200
@@ -64,6 +64,12 @@
 config CONFIG_MIPS_ISA_MIPS32
 	bool "MIPS32"
 
+config CONFIG_MIPS_ISA_MIPS32_4KC
+	bool "MIPS32 4KC (FRITZ!Box based on SoCs like AR7 or UR8)"
+
+config CONFIG_MIPS_ISA_MIPS32_24KC
+	bool "MIPS32 24KC (FRITZ!Box based on SoCs like AR9, IKS or VR9)"
+
 config CONFIG_MIPS_ISA_MIPS32R2
 	bool "MIPS32r2"
 
