diff -Naur libpthread/linuxthreads/sysdeps/pthread/pthread.h.orig libpthread/linuxthreads/sysdeps/pthread/pthread.h
--- libpthread/linuxthreads/sysdeps/pthread/pthread.h.orig	2006-12-08 04:19:36.000000000 +0100
+++ libpthread/linuxthreads/sysdeps/pthread/pthread.h	2008-11-26 23:59:29.000000000 +0100
@@ -17,8 +17,8 @@
 
 #include <features.h>
 
-#include <sched.h>
-#include <time.h>
+#include <linux/sched.h>
+#include <linux/time.h>
 
 #define __need_sigset_t
 #include <signal.h>
