--- squashfs-tools/Makefile
+++ squashfs-tools/Makefile
@@ -5,13 +5,18 @@
 CFLAGS := -I$(INCLUDEDIR) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE -O2
 LINKER = $(CC)
 
-LIBS := -lz
+ifeq ($(LINK_COMPRESSION_LIBS_STATICALLY),1)
+LINK_STATICALLY_PRE:=-Wl,-Bstatic
+LINK_STATICALLY_POST:=-Wl,-Bdynamic
+endif
+
+LIBS := $(LINK_STATICALLY_PRE) -lz $(LINK_STATICALLY_POST)
 ifeq ($(LZMA1_SUPPORT),1)
 LZMA_LIBNAME := lzma
 SUFFIX := -multi
 CPPFLAGS += -DLZMA1_SUPPORT $(if $(strip $(LZMA_DIR)),-I$(strip $(LZMA_DIR))/C)
 LINKER = $(if $(findstring ++,$(LZMA_LIBNAME)),$(CXX),$(CC))
-LIBS += $(if $(strip $(LZMA_DIR)),-L$(strip $(LZMA_DIR))) -l$(strip $(LZMA_LIBNAME))
+LIBS += $(if $(strip $(LZMA_DIR)),-L$(strip $(LZMA_DIR))) $(LINK_STATICALLY_PRE) -l$(strip $(LZMA_LIBNAME)) $(LINK_STATICALLY_POST)
 endif
 LIBS += -lpthread -lm
 
