--- configure
+++ configure
@@ -14921,7 +14921,7 @@
 		;;
 	*)
 		OSLIBS=""
-		EXTRALIBS="-Wl,--rpath -Wl,$libdir"
+		EXTRALIBS=""
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: ${EXTRALIBS}" >&5
 $as_echo "${EXTRALIBS}" >&6; }
 		;;
@@ -15046,7 +15046,7 @@
 	FUSE_FLAGS="-DFUSE_USE_VERSION=26"
 	FUSE_INCLUDES="-I${fuse_include_path}"
 	FUSE_LIBS="-L${fuse_lib_path}"
-	LD_EXTRALIBS="$save_LD_EXTRALIBS -Wl,--rpath -Wl,${fuse_lib_path}"
+	LD_EXTRALIBS="$save_LD_EXTRALIBS"
 	CPPFLAGS="$save_CPPFLAGS -D_FILE_OFFSET_BITS=64 $FUSE_FLAGS $FUSE_INCLUDES"
 	LDFLAGS="$save_LDFLAGS $FUSE_LIBS"
 
