diff -Naur smartmontools-7.2/configure smartmontools-7.2.fix/configure
--- smartmontools-7.2/configure	2020-12-30 17:49:29.000000000 +0100
+++ smartmontools-7.2.fix/configure	2021-02-10 19:57:31.467928361 +0100
@@ -7230,11 +7230,11 @@
     CXXFLAGS="$CXXFLAGS $option"
     cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-
+int i;
 int
 main ()
 {
-
+volatile char buf[10]; buf[i] = 0;
   ;
   return 0;
 }
diff -Naur smartmontools-7.2/configure.ac smartmontools-7.2.fix/configure.ac
--- smartmontools-7.2/configure.ac	2020-12-30 17:49:18.000000000 +0100
+++ smartmontools-7.2.fix/configure.ac	2021-02-10 19:57:21.147836006 +0100
@@ -766,7 +766,9 @@
     AC_MSG_CHECKING([whether $CXX supports $option])
     save_CXXFLAGS=$CXXFLAGS
     CXXFLAGS="$CXXFLAGS $option"
-    AC_LINK_IFELSE([AC_LANG_PROGRAM()], [res=yes], [res=no; CXXFLAGS=$save_CXXFLAGS])
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([[int i;]],
+      [[volatile char buf[10]; buf[i] = 0;]])],
+      [res=yes], [res=no; CXXFLAGS=$save_CXXFLAGS])
     AC_MSG_RESULT([$res])
   done
 else
