From 46cbe5357cc65e748171ea476b79c70d73976324 Mon Sep 17 00:00:00 2001
From: Michael Adam <obnox@samba.org>
Date: Wed, 21 Dec 2016 02:18:30 +0100
Subject: [PATCH] network: honour result of inet_ntop in get_ip_string()

Signed-off-by: Michael Adam <obnox@samba.org>
---
 src/network.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

--- src/network.c
+++ src/network.c
@@ -251,6 +251,8 @@ ssize_t readline (int fd, char **whole_buffer)
  */
 char *get_ip_string (struct sockaddr *sa, char *buf, size_t buflen)
 {
+        const char *result;
+
         assert (sa != NULL);
         assert (buf != NULL);
         assert (buflen != 0);
@@ -261,7 +263,8 @@ const char *get_ip_string (struct sockaddr *sa, char *buf, size_t buflen)
                 {
                         struct sockaddr_in *sa_in = (struct sockaddr_in *) sa;
 
-                        inet_ntop (AF_INET, &sa_in->sin_addr, buf, buflen);
+                        result = inet_ntop (AF_INET, &sa_in->sin_addr, buf,
+                                            buflen);
                         break;
                 }
         case AF_INET6:
@@ -269,7 +272,8 @@ const char *get_ip_string (struct sockaddr *sa, char *buf, size_t buflen)
                         struct sockaddr_in6 *sa_in6 =
                             (struct sockaddr_in6 *) sa;
 
-                        inet_ntop (AF_INET6, &sa_in6->sin6_addr, buf, buflen);
+                        result = inet_ntop (AF_INET6, &sa_in6->sin6_addr, buf,
+                                            buflen);
                         break;
                 }
         default:
@@ -277,7 +281,7 @@ const char *get_ip_string (struct sockaddr *sa, char *buf, size_t buflen)
                 return NULL;
         }
 
-        return buf;
+        return result;
 }
 
 /*
