--- Makefile.in
+++ Makefile.in
@@ -317,9 +317,7 @@
 	src \
 	data \
 	etc \
-	docs \
 	m4macros \
-	tests
 
 
 # tools want this on a single line
--- configure
+++ configure
@@ -7107,6 +7107,7 @@
 
 
 
+ac_cv_path_A2X=no
 # Check for asciidoc
 # Extract the first word of "a2x", so it can be a program name with args.
 set dummy a2x; ac_word=$2
@@ -7157,9 +7158,6 @@
   HAVE_A2X_FALSE=
 fi
 
-if test x"$A2X" = x"no"; then
-  as_fn_error $? "Test for asciidoc failed. See the file 'INSTALL' for help." "$LINENO" 5
-fi
 
 ac_config_files="$ac_config_files Makefile src/Makefile data/Makefile data/templates/Makefile etc/Makefile docs/Makefile docs/man5/Makefile docs/man5/tinyproxy.conf.txt docs/man8/Makefile docs/man8/tinyproxy.txt m4macros/Makefile tests/Makefile tests/scripts/Makefile"
 
