--- configure
+++ configure
@@ -5497,6 +5497,7 @@
 
 
 
+if false; then
 unset ac_cv_func_yp_get_default_domain
 ac_fn_c_check_func "$LINENO" "yp_get_default_domain" "ac_cv_func_yp_get_default_domain"
 if test "x$ac_cv_func_yp_get_default_domain" = xyes; then :
@@ -5645,6 +5646,7 @@
   LIBS="-lresolv $LIBS"
 
 fi
+fi
 
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for ANSI C header files" >&5
