--- src/main.c
+++ src/main.c
@@ -336,8 +336,8 @@
         conf->errorpages = NULL;
         conf->stathost = safestrdup (TINYPROXY_STATHOST);
         conf->idletimeout = MAX_IDLE_TIME;
-        conf->logf_name = safestrdup (LOCALSTATEDIR "/log/tinyproxy/tinyproxy.log");
-        conf->pidpath = safestrdup (LOCALSTATEDIR "/run/tinyproxy/tinyproxy.pid");
+        conf->logf_name = safestrdup ("/var/log/tinyproxy.log");
+        conf->pidpath = safestrdup ("/var/run/tinyproxy//tinyproxy.pid");
 }
 
 /**
--- src/child.c
+++ src/child.c
@@ -75,7 +75,7 @@
 
 static void _child_lock_init (void)
 {
-        char lock_file[] = "/tmp/tinyproxy.servers.lock.XXXXXX";
+        char lock_file[] = "/var/tmp/tinyproxy.servers.lock.XXXXXX";
 
         /* Only allow u+rw bits. This may be required for some versions
          * of glibc so that mkstemp() doesn't make us vulnerable.
--- src/heap.c
+++ src/heap.c
@@ -109,9 +109,9 @@
 {
         int fd;
         void *ptr;
-        char buffer[32];
+        char buffer[36];
 
-        static const char *shared_file = "/tmp/tinyproxy.shared.XXXXXX";
+        static const char *shared_file = "/var/tmp/tinyproxy.shared.XXXXXX";
 
         assert (size > 0);
 
