--- configure
+++ configure
@@ -5215,6 +5215,7 @@
 
 
 
+if false; then
 unset ac_cv_func_yp_get_default_domain
 ac_fn_c_check_func "$LINENO" "yp_get_default_domain" "ac_cv_func_yp_get_default_domain"
 if test "x$ac_cv_func_yp_get_default_domain" = x""yes; then :
@@ -5363,6 +5364,7 @@
   LIBS="-lresolv $LIBS"
 
 fi
+fi
 
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for ANSI C header files" >&5
