--- src/child.c	2010-01-10 23:52:04.000000000 +0100
+++ src/child.c	2010-11-30 15:15:20.000000000 +0100
@@ -76,7 +76,7 @@
 
 static void _child_lock_init (void)
 {
-        char lock_file[] = "/tmp/tinyproxy.servers.lock.XXXXXX";
+        char lock_file[] = "/var/tmp/tinyproxy.servers.lock.XXXXXX";
 
         /* Only allow u+rw bits. This may be required for some versions
          * of glibc so that mkstemp() doesn't make us vulnerable.
--- src/heap.c	2010-01-10 23:52:04.000000000 +0100
+++ src/heap.c	2010-11-30 15:15:20.000000000 +0100
@@ -109,9 +109,9 @@
 {
         int fd;
         void *ptr;
-        char buffer[32];
+        char buffer[36];
 
-        static const char *shared_file = "/tmp/tinyproxy.shared.XXXXXX";
+        static const char *shared_file = "/var/tmp/tinyproxy.shared.XXXXXX";
 
         assert (size > 0);
 
