--- linux-2.6.28/drivers/char/ifxmips_mei.c.orig	2011-09-06 08:35:56.080733428 +0200
+++ linux-2.6.28/drivers/char/ifxmips_mei.c	2011-09-06 08:35:44.950873348 +0200
@@ -63,7 +63,7 @@
 #define IFXMIPS_ICU_IM2_IER               IFX_ICU_IM2_IER
 #define IFXMIPS_MEI_INT                   IFX_MEI_INT
 #define IFXMIPS_MEI_DYING_GASP_INT        IFX_MEI_DYING_GASP_INT
-#define IFXMIPS_MEI_BASE_ADDR             IFX_MEI_SPACE_ACCESS
+//#define IFXMIPS_MEI_BASE_ADDR             IFX_MEI_SPACE_ACCESS
 #define IFXMIPS_PMU_PWDCR                 IFX_PMU_PWDCR
 #define IFXMIPS_MPS_CHIPID                IFX_MPS_CHIPID
 
