--- usr/www/all/html/de/home/home.js.orig	Sat Feb 28 11:42:05 2009
+++ usr/www/all/html/de/home/home.js	Sat Feb 28 11:40:36 2009
@@ -696,15 +696,13 @@
 var nPartTotal = Number("<? query ctlusb:settings/storage-part-total ?>");
 if (isNaN(nPartTotal)) nPartTotal = 0;
 var unplugOk = parseInt("<? query ctlusb:settings/unplug-ok ?>", 10) == 1;
-if (nPartTotal > nStor) {
+if (nPartTotal > 0) {
 var strStor = "";
 if (unplugOk) {
 strStor = "Alle USB-Speicher wurden \"sicher entfernt\" und können jetzt abgezogen werden.";
 }
 else {
-strStor = "Mindestens ein USB-Speicher oder Teile davon sind nicht eingebunden. ";
-strStor += "<a href=\"javascript:jslGoTo('home', 'syslog');\">Weitere Informationen.</a>";
-strStor = strStor.fontcolor("#cc0000");
+strStor = "Es wurden " + nStor + " von " + nPartTotal + " Partitionen eingebunden.";
 }
 strArr.push(strStor);
 }
