--- pppd/main.c.orig	2009-11-06 15:01:47.000000000 +0100
+++ pppd/main.c	2009-11-06 15:01:48.000000000 +0100
@@ -1959,9 +1959,11 @@
 		free(p-1);
 		script_env[i] = newstring;
 #ifdef USE_TDB
-		if (iskey && pppdb != NULL)
-		    add_db_key(newstring);
-		update_db_entry();
+		if (pppdb != NULL) {
+		    if (iskey)
+			add_db_key(newstring);
+		    update_db_entry();
+		}
 #endif
 		return;
 	    }
