--- pppd/main.c.orig	2009-07-31 00:40:27.000000000 +0200
+++ pppd/main.c	2009-07-31 00:40:28.000000000 +0200
@@ -1941,9 +1941,11 @@
 		free(p-1);
 		script_env[i] = newstring;
 #ifdef USE_TDB
-		if (iskey && pppdb != NULL)
-		    add_db_key(newstring);
-		update_db_entry();
+		if (pppdb != NULL) {
+		    if (iskey)
+			add_db_key(newstring);
+		    update_db_entry();
+		}
 #endif
 		return;
 	    }
