--- pppd/main.c.orig	2009-07-31 00:40:28.000000000 +0200
+++ pppd/main.c	2009-07-31 00:40:28.000000000 +0200
@@ -1569,6 +1569,8 @@
 	if (errfd == 0 || errfd == 1)
 		errfd = dup(errfd);
 
+	closelog();
+
 	/* dup the in, out, err fds to 0, 1, 2 */
 	if (infd != 0)
 		dup2(infd, 0);
@@ -1577,7 +1579,6 @@
 	if (errfd != 2)
 		dup2(errfd, 2);
 
-	closelog();
 	if (log_to_fd > 2)
 		close(log_to_fd);
 	if (the_channel->close)
