--- pppd/plugins/Makefile.linux.orig	2008-06-22 15:18:31.000000000 +0200
+++ pppd/plugins/Makefile.linux	2008-06-22 15:18:57.000000000 +0200
@@ -9,7 +9,7 @@
 MANDIR = $(DESTDIR)/share/man/man8
 LIBDIR = $(DESTDIR)/lib/pppd/$(VERSION)
 
-SUBDIRS := rp-pppoe pppoatm pppol2tp
+SUBDIRS := rp-pppoe pppol2tp
 # Uncomment the next line to include the radius authentication plugin
 SUBDIRS += radius
 PLUGINS := minconn.so passprompt.so passwordfd.so winbind.so
