--- pppd/plugins/rp-pppoe/plugin.c.orig	2007-07-28 16:58:09.000000000 +0200
+++ pppd/plugins/rp-pppoe/plugin.c	2007-07-28 16:58:12.000000000 +0200
@@ -283,7 +283,7 @@
 	/* Strip off "nic-" */
 	cmd += 4;
     } else if (strlen(cmd) < 4
-	       || (strncmp(cmd, "eth", 3) && strncmp(cmd, "nas", 3)
+	       || (strncmp(cmd, "eth", 3) && strncmp(cmd, "nas", 3) && strncmp(cmd, "vlan", 4) && strncmp(cmd, "ath", 3)
 		   && strncmp(cmd, "tap", 3) && strncmp(cmd, "br", 2))) {
 	return 0;
     }
