--- usr/www/avm/html/de/fon/sperre1.js.orig	2008-09-12 10:16:17.000000000 +0200
+++ usr/www/avm/html/de/fon/sperre1.js	2008-09-12 10:21:05.000000000 +0200
@@ -69,7 +69,7 @@
 var nummer="";
 if (jslGetValue("uiViewBereichIncoming") == "")
 nummer = jslGetValue("uiViewNummer");
-if (!isZahlRaute(nummer)) {alert(g_mldNonDigits); return;}
+//if (!isZahlRaute(nummer)) {alert(g_mldNonDigits); return;}
 var x = IstActionVorhanden(nummer);
 if (x != "") {
 if (jslGetValue("uiPostInAction"+x) == "1") {
@@ -162,7 +162,7 @@
 }
 } else {
 var nummer = jslGetValue("uiViewNummer");
-if (!isZahlRaute(nummer)) {alert(g_mldNonDigits); return;}
+//if (!isZahlRaute(nummer)) {alert(g_mldNonDigits); return;}
 var x = IstActionVorhanden(nummer);
 if (x != "") {
 if (jslGetValue("uiPostInAction"+x) == "1") {alert(g_txtAlert_1); return;}
