--- Makefile.am.orig	2009-12-01 17:53:43.000000000 +0100
+++ Makefile.am	2009-12-01 18:00:48.000000000 +0100
@@ -56,7 +56,7 @@
 	libltdl/config-h.in
 
 libtool: $(srcdir)/ltmain.sh $(top_builddir)/configure.ac
-	$(SHELL) $(top_builddir)/config.status --recheck
+	#$(SHELL) $(top_builddir)/config.status --recheck
 	chmod +x $@
 
 libtoolize: $(srcdir)/libtoolize.in $(top_builddir)/config.status
--- Makefile.in.orig	2009-12-01 17:54:11.000000000 +0100
+++ Makefile.in	2009-12-01 18:01:03.000000000 +0100
@@ -303,7 +303,7 @@
 	esac;
 
 $(top_builddir)/config.status: $(top_srcdir)/configure $(CONFIG_STATUS_DEPENDENCIES)
-	$(SHELL) ./config.status --recheck
+	#$(SHELL) ./config.status --recheck
 
 $(top_srcdir)/configure:  $(am__configure_deps)
 	cd $(srcdir) && $(AUTOCONF)
@@ -786,7 +786,7 @@
 
 
 libtool: $(srcdir)/ltmain.sh $(top_builddir)/configure.ac
-	$(SHELL) $(top_builddir)/config.status --recheck
+	#$(SHELL) $(top_builddir)/config.status --recheck
 	chmod +x $@
 
 libtoolize: $(srcdir)/libtoolize.in $(top_builddir)/config.status
