--- login-utils/checktty.c
+++ login-utils/checktty.c
@@ -311,7 +311,10 @@
 
 	memset(&hints, 0, sizeof(hints));
 	hints.ai_family = AF_UNSPEC;
-	hints.ai_flags = AI_NUMERICHOST |  AI_PASSIVE | AI_ADDRCONFIG;
+	hints.ai_flags = AI_NUMERICHOST |  AI_PASSIVE;
+#ifdef AI_ADDRCONFIG
+	hints.ai_flags |= AI_ADDRCONFIG;
+#endif
 	hints.ai_socktype = SOCK_STREAM;
 
 	for (item = alist; item->range; item++) {
--- login-utils/login.c
+++ login-utils/login.c
@@ -455,7 +455,10 @@
 		struct addrinfo hints, *info = NULL;
 
 		memset(&hints, 0, sizeof(hints));
-		hints.ai_flags = AI_ADDRCONFIG;
+		hints.ai_flags = 0;
+#ifdef AI_ADDRCONFIG
+		hints.ai_flags |= AI_ADDRCONFIG;
+#endif
 
 		hostaddress[0] = 0;
 
