--- scripts/kconfig/symbol.c
+++ scripts/kconfig/symbol.c
@@ -305,7 +305,7 @@
 	struct gstr gs = str_new();
 
 	str_printf(&gs,
-		   "\nWARNING: unmet direct dependencies detected for %s\n",
+		   "\nERROR: unmet direct dependencies detected for %s\n",
 		   sym->name);
 	str_printf(&gs,
 		   "  Depends on [%c]: ",
@@ -319,6 +319,7 @@
 			       "  Selected by [m]:\n");
 
 	fputs(str_get(&gs), stderr);
+	exit(1);
 }
 
 void sym_calc_value(struct symbol *sym)
