--- scripts/kconfig/mconf.c
+++ scripts/kconfig/mconf.c
@@ -504,7 +504,7 @@
 			case P_COMMENT:
 				if (prompt) {
 					child_count++;
-					item_make("---%*c%s", indent + 1, ' ', prompt);
+					item_make("%s%*c%s", (prompt[0] != ' ' ? "---" : "  "), indent + 1, ' ', prompt);
 					item_set_tag(':');
 					item_set_data(menu);
 				}
--- scripts/kconfig/menu.c
+++ scripts/kconfig/menu.c
@@ -159,11 +159,13 @@
 {
 	struct property *prop = menu_add_prop(type, NULL, dep);
 
+	    if (type != P_COMMENT) { // allow leading whitespaces for comments, makes multiline comments possible
 	if (isspace(*prompt)) {
 		prop_warn(prop, "leading whitespace ignored");
 		while (isspace(*prompt))
 			prompt++;
 	}
+	    }
 	if (current_entry->prompt)
 		prop_warn(prop, "prompt redefined");
 
