--- ltdl.m4	2008-02-01 17:33:31.000000000 +0100
+++ ltdl.m4	2010-03-20 17:05:00.000000000 +0100
@@ -186,10 +186,8 @@
     ;;
   esac
   ])
-if test "$libltdl_cv_sys_dlopen_deplibs" != yes; then
  AC_DEFINE([LTDL_DLOPEN_DEPLIBS], [1],
     [Define if the OS needs help to load dependent libraries for dlopen().])
-fi
 ])# AC_LTDL_SYS_DLOPEN_DEPLIBS
 
 
--- libltdl/acinclude.m4	2008-02-01 17:38:42.000000000 +0100
+++ libltdl/acinclude.m4	2010-03-20 17:05:00.000000000 +0100
@@ -6842,10 +6842,8 @@
     ;;
   esac
   ])
-if test "$libltdl_cv_sys_dlopen_deplibs" != yes; then
  AC_DEFINE([LTDL_DLOPEN_DEPLIBS], [1],
     [Define if the OS needs help to load dependent libraries for dlopen().])
-fi
 ])# AC_LTDL_SYS_DLOPEN_DEPLIBS
 
 
--- libltdl/configure	2008-02-01 17:38:49.000000000 +0100
+++ libltdl/configure	2010-03-20 17:05:05.000000000 +0100
@@ -21484,13 +21484,11 @@
 fi
 { echo "$as_me:$LINENO: result: $libltdl_cv_sys_dlopen_deplibs" >&5
 echo "${ECHO_T}$libltdl_cv_sys_dlopen_deplibs" >&6; }
-if test "$libltdl_cv_sys_dlopen_deplibs" != yes; then
 
 cat >>confdefs.h <<\_ACEOF
 #define LTDL_DLOPEN_DEPLIBS 1
 _ACEOF
 
-fi
 
 
 for ac_header in argz.h
