--- configure
+++ configure
@@ -2114,6 +2114,7 @@
 fi
 
 echo "$ac_t""$wi_cv_shared_libgcc" 1>&6
+if false; then
 	if test "$wi_cv_shared_libgcc" != no ; then
 		
 	if test "x$ac_cv_prog_gcc" = "xyes" ; then
@@ -2194,6 +2195,7 @@
 	fi
 
 	fi
+fi
 
 	add_O0="no"
 	if [ "$NOOPTCFLAGS" = "" ] ; then
@@ -3278,6 +3280,7 @@
 fi
 
 echo "$ac_t""$wi_cv_shared_libgcc" 1>&6
+if false; then
 	if test "$wi_cv_shared_libgcc" != no ; then
 		
 	if test "x$ac_cv_prog_gcc" = "xyes" ; then
@@ -3358,6 +3361,7 @@
 	fi
 
 	fi
+fi
 
 	add_O0="no"
 	if [ "$NOOPTCFLAGS" = "" ] ; then
