--- usr/www/avme/html/de/home/home.js.orig	2010-03-26 12:46:42.000000000 +0100
+++ usr/www/avme/html/de/home/home.js	2010-04-08 22:35:44.000000000 +0200
@@ -806,7 +806,7 @@
 var nPartTotal = Number(g_usbhost_data.part_count_total);
 if (isNaN(nPartTotal)) nPartTotal = 0;
 var unplugOk = parseInt(g_usbhost_data.unplug_ok, 10) == 1;
-if (nPartTotal > nStor)
+if (nPartTotal > 0)
 {
 if (unplugOk)
 {
@@ -814,9 +814,7 @@
 }
 else
 {
-strStor = "{?de.home.home.js:5308?} ";
-strStor += "<a href=\"javascript:jslGoTo('home', 'syslog');\">{?de.home.home.js:3744?}</a>";
-strStor = strStor.fontcolor("#cc0000");
+strStor = "Mounted " + nStor + " of " + nPartTotal + " partitions.";
 }
 strArr.push(strStor);
 }
