--- usr/www/avme/html/de/home/home.js.orig	2009-09-15 15:26:15.000000000 +0200
+++ usr/www/avme/html/de/home/home.js	2009-09-16 19:26:35.000000000 +0200
@@ -802,15 +802,13 @@
 var nPartTotal = Number("<? query ctlusb:settings/storage-part-total ?>");
 if (isNaN(nPartTotal)) nPartTotal = 0;
 var unplugOk = parseInt("<? query ctlusb:settings/unplug-ok ?>", 10) == 1;
-if (nPartTotal > nStor) {
+if (nPartTotal > 0) {
 var strStor = "";
 if (unplugOk) {
 strStor = "{?de.home.home.js:912?}";
 }
 else {
-strStor = "{?de.home.home.js:5308?} ";
-strStor += "<a href=\"javascript:jslGoTo('home', 'syslog');\">{?de.home.home.js:3744?}</a>";
-strStor = strStor.fontcolor("#cc0000");
+strStor = nStor + " of " + nPartTotal + " partitions mounted.";
 }
 strArr.push(strStor);
 }
