--- libtransmission/jsonsl.c
+++ libtransmission/jsonsl.c
@@ -870,7 +870,7 @@
     }
     jsn->jprs = malloc(sizeof(jsonsl_jpr_t) * njprs);
     jsn->jpr_count = njprs;
-    jsn->jpr_root = calloc(1, sizeof(int) * njprs * jsn->levels_max);
+    jsn->jpr_root = calloc(1, sizeof(size_t) * njprs * jsn->levels_max);
     memcpy(jsn->jprs, jprs, sizeof(jsonsl_jpr_t) * njprs);
     /* Set the initial jump table values */
 
