--- libtransmission/tr-udp.c
+++ libtransmission/tr-udp.c
@@ -46,8 +46,13 @@
 /* Since we use a single UDP socket in order to implement multiple
    uTP sockets, try to set up huge buffers. */
 
+#ifdef TR_LIGHTWEIGHT
+#define RECV_BUFFER_SIZE (2 * 1024 * 1024)
+#define SEND_BUFFER_SIZE (     512 * 1024)
+#else
 #define RECV_BUFFER_SIZE (4 * 1024 * 1024)
 #define SEND_BUFFER_SIZE (1 * 1024 * 1024)
+#endif
 #define SMALL_BUFFER_SIZE (32 * 1024)
 
 static void set_socket_buffers(tr_socket_t fd, bool large)
