--- third-party/miniupnpc/minissdpc.c
+++ third-party/miniupnpc/minissdpc.c
@@ -637,6 +637,7 @@
 	{
 		if(ipv6) {
 #if !defined(_WIN32)
+#if defined(IPV6_MULTICAST_IF)
 			/* according to MSDN, if_nametoindex() is supported since
 			 * MS Windows Vista and MS Windows Server 2008.
 			 * http://msdn.microsoft.com/en-us/library/bb408409%28v=vs.85%29.aspx */
@@ -645,6 +646,7 @@
 			{
 				PRINT_SOCKET_ERROR("setsockopt IPV6_MULTICAST_IF");
 			}
+#endif
 #else
 #ifdef DEBUG
 			printf("Setting of multicast interface not supported in IPv6 under Windows.\n");
