--- cli/transmissioncli.c.orig	2008-06-15 05:33:39.000000000 +0200
+++ cli/transmissioncli.c	2008-06-15 05:28:38.000000000 +0200
@@ -480,7 +480,7 @@
             case 'i': showInfo = 1; break;
             case 'm': comment = optarg; break;
             case 'n': natTraversal = 1; break;
-            case 'o': savePath = optarg;
+            case 'o': savePath = optarg; break;
             case 'p': bindPort = atoi( optarg ); break;
             case 'r': isPrivate = 1; break;
             case 's': showScrape = 1; break;
