Index: web/javascript/transmission.js
===================================================================
--- web/javascript/transmission.js	(revision 10536)
+++ web/javascript/transmission.js	(revision 10773)
@@ -1567,4 +1567,7 @@
 				if( e ) {
 					var row_index = tr.getTorrentIndex(tr._rows, torrent);
+					//if torrent is not currently visible then delete a hidden row
+					//otherwise the row model gets corrupted because no row gets deleted
+					if( row_index == null ) row_index = tr._rows.length - 1;
 					delete e._torrent; //remove circular refernce to help IE garbage collect
 					tr._rows.splice(row_index, 1)
