diff -burN mk.orig/common.mk mk/common.mk
--- mk.orig/common.mk	2007-05-02 21:15:51.000000000 +0200
+++ mk/common.mk	2007-05-23 15:20:18.000000000 +0200
@@ -1,7 +1,6 @@
 # $Id: common.mk 1739 2007-04-17 22:38:42Z joshe $
 
-TMPCFLAGS   = -g -Wall -W -O3 -funroll-loops -D_FILE_OFFSET_BITS=64 \
-              -D_LARGEFILE_SOURCE -D_GNU_SOURCE \
+TMPCFLAGS   = -g -Wall -W -O3 -funroll-loops -D_GNU_SOURCE \
               -DSYS_$(shell echo $(SYSTEM) | tr a-z A-Z)
 TMPCXXFLAGS = $(TMPCFLAGS)
 TMPLDFLAGS  =
