--- libtransmission/peer-mgr.h	(revision 11264)
+++ libtransmission/peer-mgr.h	(revision 11295)
@@ -111,10 +111,4 @@
     time_t                   chokeChangedAt;
 
-    time_t                   lastBlocksAtTime;
-    int                      blocksAt[60];
-
-    time_t                   lastCancelTime;
-    int                      cancelAt[60];
-
     tr_recentHistory       * blocksSentToClient;
     tr_recentHistory       * blocksSentToPeer;
