--- /dev/null
+++ Net-SSLeay_Makefile.PL
@@ -0,0 +1,13 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use lib 'ext';
+use ExtUtils::MakeMaker;
+
+WriteMakefile(
+    'NAME'             => 'Net::SSLeay',
+    'VERSION_FROM'     => 'lib/Net/SSLeay.pm',
+    'INC'              => '-I$(TARGET_TOOLCHAIN_STAGING_DIR)/include',
+    'LIBS'             => ['-L$(TARGET_TOOLCHAIN_STAGING_DIR)/lib -lssl -lcrypto -lcrypt32 -lz'],
+);
