--- pjlib/src/pj/file_access_unistd.c.orig	2008-12-23 18:27:53.000000000 +0100
+++ pjlib/src/pj/file_access_unistd.c	2009-07-30 23:54:05.000000000 +0200
@@ -49,11 +49,17 @@
 PJ_DEF(pj_off_t) pj_file_size(const char *filename)
 {
     struct stat buf;
+    FILE *fp;
 
     PJ_ASSERT_RETURN(filename, -1);
 
-    if (stat(filename, &buf) != 0)
-	return -1;
+    buf.st_size = -1;
+    fp = fopen(filename, "r");
+    if(fp) {
+	fseek(fp, 0L, SEEK_END);
+	buf.st_size = ftell(fp);
+	fclose(fp);
+    }
 
     return buf.st_size;
 }
