--- CRT.h
+++ CRT.h
@@ -14,7 +14,7 @@
 #include <signal.h>
 #include <stdlib.h>
 #include <stdbool.h>
-#include <execinfo.h>
+/*#include <execinfo.h>*/
 
 #include "String.h"
 
@@ -119,7 +119,7 @@
 
 char* CRT_termType;
 
-void *backtraceArray[128];
+/*void *backtraceArray[128];*/
 
 // TODO: pass an instance of Settings instead.
 
--- CRT.c
+++ CRT.c
@@ -11,7 +11,7 @@
 #include <signal.h>
 #include <stdlib.h>
 #include <stdbool.h>
-#include <execinfo.h>
+/*#include <execinfo.h>*/
 
 #include "String.h"
 
@@ -118,7 +118,7 @@
 
 char* CRT_termType;
 
-void *backtraceArray[128];
+/*void *backtraceArray[128];*/
 
 static void CRT_handleSIGSEGV(int sgn) {
    (void) sgn;
@@ -128,9 +128,9 @@
    #else
    fprintf(stderr, "\n\nhtop " VERSION " aborting. Unsupported platform.\n");
    #endif
-   size_t size = backtrace(backtraceArray, sizeof(backtraceArray));
-   fprintf(stderr, "Backtrace: \n");
-   backtrace_symbols_fd(backtraceArray, size, 2);
+   /*size_t size = backtrace(backtraceArray, sizeof(backtraceArray));*/
+   /*fprintf(stderr, "Backtrace: \n");*/
+   /*backtrace_symbols_fd(backtraceArray, size, 2);*/
    abort();
 }
 
