--- usr/www/all/menus/menu_data.lua
+++ usr/www/all/menus/menu_data.lua
@@ -1189,6 +1189,10 @@
 pageData, data = getPageData()
 local expertMode = data.expertMode
 local checkedIds = {}
+
+-- TODO: find out how to check if Freetz WOL files are avalaible
+local freetzWolInstalled = false
+
 local menuData = {
 ["overview"] = pageData["overview"] and {
 ["txt"] = TXT([[{?menuOverview?}]]),
@@ -2155,6 +2159,16 @@
 ["lnk"] = pageData["liveTv"].lnk,
 ["pos"] = -65
 } or nil,
+["freetz-wol"] = (freetzWolInstalled) and {
+["txt"] = "Freetz WOL",
+["lnk"] = "/cgi-bin/freetz_wol",
+["pos"] = -55
+} or nil,
+["freetz"] = {
+["txt"] = "Freetz",
+["lnk"] = "/cgi-bin/freetz_status",
+["pos"] = -60
+},
 ["endTag"] = nil
 }
 function checkMenu(pid)
