--- frontend/Makefile.in.orig	2010-04-14 02:52:59.000000000 +0200
+++ frontend/Makefile.in	2010-04-30 11:59:52.000000000 +0200
@@ -146,7 +146,7 @@
 LATEX = @LATEX@
 LDFLAGS = @LDFLAGS@
 LIBOBJS = @LIBOBJS@
-LIBS = @LIBS@
+LIBS = 
 LIBTOOL = @LIBTOOL@
 LIBUSB_1_0_CFLAGS = @LIBUSB_1_0_CFLAGS@
 LIBUSB_1_0_LIBS = @LIBUSB_1_0_LIBS@
@@ -257,7 +257,7 @@
 scanimage_LDADD = ../lib/liblib.la ../lib/libfelib.la ../backend/libsane.la
 saned_SOURCES = saned.c
 saned_LDADD = ../sanei/libsanei.la ../lib/liblib.la ../lib/libfelib.la \
-  ../backend/libsane.la @SYSLOG_LIBS@
+  ../backend/libsane.la @SYSLOG_LIBS@ @AVAHI_LIBS@
 
 test_SOURCES = test.c
 test_LDADD = ../lib/liblib.la ../lib/libfelib.la ../backend/libsane.la
