--- linux-2.6.13/arch/mips/kernel/cpu-probe.c
+++ linux-2.6.13/arch/mips/kernel/cpu-probe.c
@@ -87,7 +87,7 @@
         case 0: /*--- 125 MHz ---*/
             if(avm_prev_idle_mode_speedup != avm_idle_mode_speedup) {
                 avm_prev_idle_mode_speedup = avm_idle_mode_speedup;
-                printk("[speedup] -> 125 MHz\n");
+//              printk("[speedup] -> 125 MHz\n");
             }
             if(p_idle < HIGH_SPEED_IDLE) {
                 if(avm_idle_lock_speedup2) {
@@ -121,7 +121,7 @@
         case 1: /*--- 150 MHz ---*/
             if(avm_prev_idle_mode_speedup != avm_idle_mode_speedup) {
                 avm_prev_idle_mode_speedup = avm_idle_mode_speedup;
-                printk("[speedup] -> 150 MHz\n");
+//              printk("[speedup] -> 150 MHz\n");
             }
             if(p_idle < HIGH_SPEED_IDLE) {
                 /*--- Timer aufsetzen fuer normalspeed ---*/
@@ -143,7 +143,7 @@
         case 2: /*--- 62.5 MHz ---*/
             if(avm_prev_idle_mode_speedup != avm_idle_mode_speedup) {
                 avm_prev_idle_mode_speedup = avm_idle_mode_speedup;
-                printk("[speedup] -> 62.5 MHz\n");
+//              printk("[speedup] -> 62.5 MHz\n");
             }
             if(p_idle < LOW_SPEED_IDLE) {
                 /*--- Hochschalten auf 125 MHz ---*/

