--- linux-2.6.13/net/ipv4/mcfastforward.c	2009-07-07 12:45:50.000000000 +0200
+++ linux-2.6.13/net/ipv4/mcfastforward.c	2014-03-24 00:26:41.487576525 +0100
@@ -1067,11 +1067,13 @@
 
 #if 1
    {
+/*
 	  char buf[128];
 	  printk(KERN_DEBUG "mcfw: group %u.%u.%u.%u: query %s:%s %lusec\n",
 			   NIPQUAD(group),
 			   drv->name, portset2str(&portset, buf, sizeof(buf)), 
 			   maxdelay/HZ);
+*/
    }
 #endif
 
