--- etc/init.d/rc.S
+++ etc/init.d/rc.S
@@ -44,21 +44,21 @@
 echo "source files in group ${gruppe} ...";
 if ls /etc/init.d/S${gruppe}[0-9]-* 2>/dev/null ; then
 for skript in `ls /etc/init.d/S${gruppe}[0-9]-* | sort` ; do
-echo "source ${skript}"
 if [ ! -f /var/skip_init ] ; then
+echo "processing ${skript}"
 . ${skript}
 else
 echo "skip_init: ${skript} "
 fi
 done
 fi
-echo "execute files in group ${gruppe} ...";
+echo "executing files in group ${gruppe} ...";
 if ls /etc/init.d/E${gruppe}[0-9]-* 2>/dev/null ; then
 for skript in `ls /etc/init.d/E${gruppe}[0-9]-* | sort` ; do
-echo "execute ${skript}"
 if [ ! -f /var/skip_init ] ; then
+echo "executing ${skript}"
 if ! sh ${skript} ; then
-echo "execute ${skript} failed."
+echo "executing ${skript} failed."
 fi
 else
 echo "skip_init: ${skript} "
