--- configure
+++ configure
@@ -4067,6 +4067,7 @@
 
 
 
+if false; then
 { echo "$as_me:$LINENO: checking for gethostname in -lnsl" >&5
 echo $ECHO_N "checking for gethostname in -lnsl... $ECHO_C" >&6; }
 if test "${ac_cv_lib_nsl_gethostname+set}" = set; then
@@ -4210,6 +4211,7 @@
 fi
 
 fi
+fi
 
 { echo "$as_me:$LINENO: checking for accept in -lsocket" >&5
 echo $ECHO_N "checking for accept in -lsocket... $ECHO_C" >&6; }
