--- Makefile.am	2012-06-07 20:08:50+02:00
+++ Makefile.am	2012-06-07 20:08:50+02:00
@@ -2,7 +2,7 @@
 # have all needed files, that a GNU package needs
 AUTOMAKE_OPTIONS = foreign 1.4
 
-SUBDIRS = src bindings $(LIBFTDI_MODULES) examples doc
+SUBDIRS = src bindings $(LIBFTDI_MODULES)
 
 EXTRA_DIST = libftdi.spec libftdi.spec.in COPYING.LIB COPYING.GPL LICENSE \
              README AUTHORS ChangeLog libftdi-config.in CMakeLists.txt \
--- Makefile.in	2012-06-07 20:08:50+02:00
+++ Makefile.in	2012-06-07 20:08:50+02:00
@@ -270,7 +270,7 @@
 # not a GNU package. You can remove this line, if
 # have all needed files, that a GNU package needs
 AUTOMAKE_OPTIONS = foreign 1.4
-SUBDIRS = src bindings $(LIBFTDI_MODULES) examples doc
+SUBDIRS = src bindings $(LIBFTDI_MODULES)
 EXTRA_DIST = libftdi.spec libftdi.spec.in COPYING.LIB COPYING.GPL LICENSE \
              README AUTHORS ChangeLog libftdi-config.in CMakeLists.txt \
              packages COPYING-CMAKE-SCRIPTS FindUSB.cmake README.mingw \
