--- configure	2011-03-14 21:48:53.000000000 +0100
+++ configure	2012-02-07 19:56:44.513627947 +0100
@@ -5044,6 +5044,9 @@
 	x86_64-*) LINUX_SRCARCH=x86;;
 	powerpc*-*) LINUX_SRCARCH=powerpc;;
 	s390*-*) LINUX_SRCARCH=s390;;
+	mips-*) LINUX_SRCARCH=mips;;
+	mipsel-*) LINUX_SRCARCH=mips;;
+	mipsbe-*) LINUX_SRCARCH=mips;;
 	*) as_fn_error $? "architecture ${host} not supported" "$LINENO" 5 ;;
 	esac
 
