--- configure.in	2008-10-30 22:01:14.000000000 +0100
+++ configure.in	2008-10-30 22:01:00.000000000 +0100
@@ -357,7 +357,7 @@
 if eval "test x$enable_local_backends = xno"; then 
    BACKENDS="net"
 else
-  if test "${BACKENDS}" != "" ; then
+  if true ; then
     AC_MSG_NOTICE([Manually selected backends: ${BACKENDS}])
   else
     BACKENDS="abaton agfafocus apple artec artec_eplus48u as6e avision bh \
