--- Makefile.in.orig	2009-04-30 15:41:20.000000000 +0200
+++ Makefile.in	2009-08-31 00:13:48.000000000 +0200
@@ -244,7 +244,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = include lib sanei backend frontend tools doc po
+SUBDIRS = include lib sanei backend frontend tools
 DIST_SUBDIRS = include lib sanei backend frontend tools doc po japi testsuite
 dist_doc_DATA = AUTHORS ChangeLog COPYING LICENSE NEWS PROBLEMS PROJECTS \
  README README.aix README.beos README.darwin README.djpeg README.freebsd \
