--- Makefile.in.orig	2010-04-14 02:52:59.000000000 +0200
+++ Makefile.in	2010-04-30 11:59:52.000000000 +0200
@@ -290,7 +290,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = include lib sanei backend frontend tools doc po
+SUBDIRS = include lib sanei backend frontend tools
 DIST_SUBDIRS = include lib sanei backend frontend tools doc po japi testsuite
 dist_doc_DATA = AUTHORS ChangeLog COPYING LICENSE NEWS PROBLEMS PROJECTS \
  README README.aix README.beos README.darwin README.djpeg README.freebsd \
