--- Makefile.in.orig	2008-05-19 11:43:49.000000000 +0200
+++ Makefile.in	2008-05-19 11:45:55.000000000 +0200
@@ -55,7 +55,8 @@
 
 .PHONY: all all-recursive clean clean-recursive depend \
   depend-recursive dist install install-recursive libcheck lsm \
-  sane-backends sort-cvsignore uninstall uninstall-recursive
+  sane-backends sort-cvsignore uninstall uninstall-recursive \
+  freetz-install-devel
 
 all: $(DEPEND_RECURSIVE) all-recursive
 
@@ -72,6 +73,14 @@
 
 install: install-recursive
 
+freetz-install-devel:
+	@for subdir in include backend; do		\
+	  target=`echo $@ | sed s/-recursive//`; \
+	  echo making $$target in $$subdir;	\
+	  (cd $$subdir && $(MAKE) $$target)	\
+	   || case "$(MFLAGS)" in *k*) fail=yes;; *) exit 1;; esac; \
+	done && test -z "$$fail"
+
 uninstall: uninstall-recursive
 
 #
--- include/Makefile.in.orig	2008-05-19 11:46:26.000000000 +0200
+++ include/Makefile.in	2008-05-19 11:47:06.000000000 +0200
@@ -51,7 +51,8 @@
 
 DISTFILES = $(SANE_INCLUDES) $(OTHER_INCLUDES)
 
-.PHONY: all clean depend dist distclean install uninstall 
+.PHONY: all clean depend dist distclean install uninstall \
+	freetz-install-devel
 
 all:
 
@@ -63,6 +64,8 @@
 	rm -f Makefile
 	rm -f sane/config.h
 
+freetz-install-devel: install
+
 install:
 	$(MKDIR) $(DESTDIR)$(includedir)/sane
 	@list="$(INSTALLED_INCLUDES)"; for file in $$list; do \
--- backend/Makefile.in.orig	2008-05-19 11:47:26.000000000 +0200
+++ backend/Makefile.in	2008-05-19 11:52:25.000000000 +0200
@@ -208,7 +208,8 @@
   umax1220u.c umax1220u-common.c umax1220u.conf.in \
   v4l.c v4l.conf.in v4l-frequencies.h v4l.h
 
-.PHONY: all clean depend dist distclean install uninstall
+.PHONY: all clean depend dist distclean install uninstall \
+	freetz-install-devel
 
 libsane-%.la: %.lo %-s.lo $(EXTRA) $(LIBOBJS)
 	@$(LIBTOOL) $(MLINK) $(CC) -export-dynamic -o $@ $($*_LIBS) \
@@ -233,6 +234,8 @@
 
 install: $(INSTALL_LOCKPATH) install-be@BE_INSTALL_FLAVOR@ install-libsane install-becfg
 
+freetz-install-devel: install-libsane
+
 install-be:
 	$(MKDIR) $(DESTDIR)$(libdir) $(DESTDIR)$(libsanedir) $(DESTDIR)$(configdir)
 	@# Install all backends in $(prefix)/lib/sane/
