--- configure.orig	2010-04-30 11:59:52.000000000 +0200
+++ configure	2010-04-30 11:59:52.000000000 +0200
@@ -23769,7 +23769,6 @@
   # Some rework will be needed to allow for fast_install
   # before this can be enabled.
   hardcode_into_libs=yes
-  sys_lib_search_path_spec="/usr/lib${libsuff} /lib${libsuff} /usr/local/lib${libsuff}"
   sys_lib_dlsearch_path_spec="/usr/lib${libsuff} /lib${libsuff}"
 
   # Append ld.so.conf contents to the search path
