--- ../sane-backends-1.0.20-orig/configure	2009-08-28 17:13:17.000000000 +0200
+++ configure	2009-09-04 13:21:40.000000000 +0200
@@ -23567,7 +23567,6 @@
   # Some rework will be needed to allow for fast_install
   # before this can be enabled.
   hardcode_into_libs=yes
-  sys_lib_search_path_spec="/usr/lib${libsuff} /lib${libsuff} /usr/local/lib${libsuff}"
   sys_lib_dlsearch_path_spec="/usr/lib${libsuff} /lib${libsuff}"
 
   # Append ld.so.conf contents to the search path
