--- configure.orig	2010-02-28 19:38:37.000000000 +0000
+++ configure	2010-02-28 19:38:37.000000000 +0000
@@ -23564,7 +23564,6 @@
   # Some rework will be needed to allow for fast_install
   # before this can be enabled.
   hardcode_into_libs=yes
-  sys_lib_search_path_spec="/usr/lib${libsuff} /lib${libsuff} /usr/local/lib${libsuff}"
   sys_lib_dlsearch_path_spec="/usr/lib${libsuff} /lib${libsuff}"
 
   # Append ld.so.conf contents to the search path
