--- backend/Makefile.in.orig	2010-04-30 11:59:52.000000000 +0200
+++ backend/Makefile.in	2010-04-30 11:59:52.000000000 +0200
@@ -1805,7 +1805,7 @@
 lib_LTLIBRARIES = libsane.la
 sanelibdir = $(libdir)/sane
 sanelib_LTLIBRARIES = $(BACKEND_LIBS_ENABLED) libsane-dll.la
-COMMON_LIBS = ../lib/liblib.la
+COMMON_LIBS = ../lib/liblib.la $(PTHREAD_LIBS)
 
 # Each backend should define a convenience library that compiles
 # all related files within backend directory.  General guideline
