--- backend/Makefile.in
+++ backend/Makefile.in
@@ -2158,7 +2158,7 @@
 lib_LTLIBRARIES = libsane.la
 sanelibdir = $(libdir)/sane
 sanelib_LTLIBRARIES = $(BACKEND_LIBS_ENABLED) libsane-dll.la
-COMMON_LIBS = ../lib/liblib.la
+COMMON_LIBS = ../lib/liblib.la $(PTHREAD_LIBS)
 
 # Each backend should define a convenience library that compiles
 # all related files within backend directory.  General guideline
