--- frontend/Makefile.in.orig	2011-01-30 19:50:55.000000000 +0100
+++ frontend/Makefile.in	2011-02-14 18:06:57.706188053 +0100
@@ -142,7 +142,7 @@
 LD = @LD@
 LDFLAGS = @LDFLAGS@
 LIBOBJS = @LIBOBJS@
-LIBS = @LIBS@
+LIBS = 
 LIBTOOL = @LIBTOOL@
 LIBUSB_1_0_CFLAGS = @LIBUSB_1_0_CFLAGS@
 LIBUSB_1_0_LIBS = @LIBUSB_1_0_LIBS@
@@ -260,7 +260,7 @@
 scanimage_LDADD = ../lib/liblib.la ../lib/libfelib.la ../backend/libsane.la
 saned_SOURCES = saned.c
 saned_LDADD = ../sanei/libsanei.la ../lib/liblib.la ../lib/libfelib.la \
-  ../backend/libsane.la @SYSLOG_LIBS@
+  ../backend/libsane.la @SYSLOG_LIBS@ @AVAHI_LIBS@
 
 test_SOURCES = test.c
 test_LDADD = ../lib/liblib.la ../lib/libfelib.la ../backend/libsane.la
