--- frontend/Makefile.in	2012-08-28 09:30:11+02:00
+++ frontend/Makefile.in	2012-08-28 09:30:11+02:00
@@ -143,7 +143,7 @@
 LD = @LD@
 LDFLAGS = @LDFLAGS@
 LIBOBJS = @LIBOBJS@
-LIBS = @LIBS@
+LIBS = 
 LIBTOOL = @LIBTOOL@
 LIBUSB_1_0_CFLAGS = @LIBUSB_1_0_CFLAGS@
 LIBUSB_1_0_LIBS = @LIBUSB_1_0_LIBS@
@@ -265,7 +265,7 @@
 
 saned_SOURCES = saned.c
 saned_LDADD = ../backend/libsane.la ../sanei/libsanei.la ../lib/liblib.la \
-             ../lib/libfelib.la @SYSLOG_LIBS@
+             ../lib/libfelib.la @SYSLOG_LIBS@ @AVAHI_LIBS@
 
 test_SOURCES = test.c
 test_LDADD = ../lib/liblib.la ../lib/libfelib.la ../backend/libsane.la
