--- frontend/Makefile.in	2009-04-30 15:41:19.000000000 +0200
+++ frontend/Makefile.in	2009-05-04 11:53:49.669902772 +0200
@@ -144,7 +144,7 @@
 LATEX = @LATEX@
 LDFLAGS = @LDFLAGS@
 LIBOBJS = @LIBOBJS@
-LIBS = @LIBS@
+LIBS = 
 LIBTOOL = @LIBTOOL@
 LIBUSB_1_0_CFLAGS = @LIBUSB_1_0_CFLAGS@
 LIBUSB_1_0_LIBS = @LIBUSB_1_0_LIBS@
@@ -254,7 +254,7 @@
 scanimage_LDADD = ../lib/liblib.la ../lib/libfelib.la ../backend/libsane.la
 saned_SOURCES = saned.c
 saned_LDADD = ../sanei/libsanei.la ../lib/liblib.la ../lib/libfelib.la \
-  ../backend/libsane.la @SYSLOG_LIBS@
+  ../backend/libsane.la @SYSLOG_LIBS@ @AVAHI_LIBS@
 
 test_SOURCES = test.c
 test_LDADD = ../lib/liblib.la ../lib/libfelib.la ../backend/libsane.la
