--- tools/sane-backends.pc.in
+++ tools/sane-backends.pc.in
@@ -2,8 +2,8 @@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
-ldflags=@LDFLAGS@ @GPHOTO2_LDFLAGS@
-libs=@LIBS@ @DL_LIBS@ @LIBV4L_LIBS@ @MATH_LIB@ @TIFF_LIBS@ @JPEG_LIBS@ @GPHOTO2_LIBS@ @SOCKET_LIBS@ @AVAHI_LIBS@ @USB_LIBS@ @SCSI_LIBS@ @RESMGR_LIBS@
+ldflags=@LDFLAGS@
+libs=@LIBS@
 
 Name: SANE Backends
 Description: Backends for SANE, the universal scanner interface
