--- sanei/sanei_usb.c	2009-02-20 19:17:01.000000000 +0100
+++ sanei/sanei_usb.c	2009-08-29 16:27:52.764209580 +0200
@@ -1236,11 +1236,18 @@
 	      /* Loop through all of the alternate settings */
 	      for (a = 0; a < dev->config[c].interface[i].num_altsetting; a++)
 		{
-
 		  DBG (5, "sanei_usb_open: configuration nr: %d\n", c);
 		  DBG (5, "sanei_usb_open:     interface nr: %d\n", i);
 		  DBG (5, "sanei_usb_open:   alt_setting nr: %d\n", a);
 
+                  /* Start by interfaces found in sanei_usb_init */
+                  if (c == 0 && i != devices[devcount].interface_nr)
+                    {
+                      DBG (5, "sanei_usb_open: interface %d not detected as "
+                        "a scanner by sanei_usb_init, ignoring.\n", i);
+                      continue;
+                     }
+
 		  interface = &dev->config[c].interface[i].altsetting[a];
 
 		  /* Now we look for usable endpoints */
@@ -1546,6 +1553,14 @@
 		  DBG (5, "sanei_usb_open:     interface nr: %d\n", i);
 		  DBG (5, "sanei_usb_open:   alt_setting nr: %d\n", a);
 
+                  /* Start by interfaces found in sanei_usb_init */
+                  if (c == 0 && i != devices[devcount].interface_nr)
+                    {
+                      DBG (5, "sanei_usb_open: interface %d not detected as "
+                        "a scanner by sanei_usb_init, ignoring.\n", i);
+                      continue;
+                     }
+
 		  interface = &config->interface[i].altsetting[a];
 
 		  /* Now we look for usable endpoints */
