--- configure	2020-07-12 21:46:14.502264360 +0200
+++ configure	2020-07-12 21:47:37.990006452 +0200
@@ -16409,12 +16409,12 @@
        for ac_func in dlopen
 do :
   ac_fn_c_check_func "$LINENO" "dlopen" "ac_cv_func_dlopen"
-if test "x$ac_cv_func_dlopen" = xyes; then :
+#if test "x$ac_cv_func_dlopen" = xyes; then :
   cat >>confdefs.h <<_ACEOF
 #define HAVE_DLOPEN 1
 _ACEOF
  enable_dynamic=yes
-fi
+#fi
 done
 
        LIBS="${saved_LIBS}"
