Description: Fix some cppcheck warnings.
Forwarded: no
Author: Peter Pentchev <roam@ringlet.net>
Last-Update: 2010-12-24

--- lib/address.c
+++ lib/address.c
@@ -146,6 +146,7 @@
          /* other members have ok default value. */
       }
 
+      // cppcheck-suppress memleakOnRealloc
       if ((socksfdv = realloc(socksfdv, sizeof(*socksfdv) * dc)) == NULL)
          serrx(EXIT_FAILURE, "%s: %s", function, NOMEM);
 
@@ -348,8 +349,9 @@
       /* only network-sockets can be proxied. */
       if (local.sa_family != AF_INET
 #ifdef AF_INET6
-      &&  local.sa_family != AF_INET6)
+      &&  local.sa_family != AF_INET6
 #endif /* AF_INET6 */
+      )
          break;
 
       if ((socksfd = socks_getaddr(s, 0)) != NULL) {
--- lib/Rgethostbyname.c
+++ lib/Rgethostbyname.c
@@ -121,7 +121,7 @@
 
 #if HAVE_IPV6_SUPPORT
       case AF_INET6: {
-         static char ipv6[struct in6_addr)];
+         static char ipv6[struct in6_addr];
 
          hostent->h_length     = sizeof(ipv6);
          *hostent->h_addr_list = ipv6;
--- lib/iobuf.c
+++ lib/iobuf.c
@@ -327,6 +327,7 @@
 
 #if SOCKS_CLIENT
    if (freebuffer == NULL) {
+      // cppcheck-suppress memleakOnRealloc
       if ((iobufferv = realloc(iobufferv, sizeof(*iobufferv) * ++iobufferc))
       == NULL)
          serrx(EXIT_FAILURE, "%s: %s", function, NOMEM);
