--- dlib/interposition.c
+++ dlib/interposition.c
@@ -66,9 +66,9 @@
     HAVE_PROT_SENDTO_4, HAVE_PROT_SENDTO_5, HAVE_PROT_SENDTO_6);
 #endif
 
-#ifndef __USE_GNU
-#define __USE_GNU /* XXX for RTLD_NEXT on Linux */
-#endif /* !__USE_GNU */
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE /* XXX for RTLD_NEXT on Linux */
+#endif /* !_GNU_SOURCE */
 #include <dlfcn.h>
 
 #undef accept
--- include/osdep.h
+++ include/osdep.h
@@ -51,10 +51,11 @@
  *  interposition.c (also for CMSG_)
  */
 
-#ifdef __GNUC__
-#undef __GNUC__
-#define __GNUC__ 0
-#endif /* __GNUC__ */
+#ifdef __GNUC__
+#undef __GNUC__
+#define __GNUC__ 0
+#endif /* __GNUC__ */
+#define _GNU_SOURCE
 
 #endif /* HAVE_LINUX_ECCENTRICITIES */
 
@@ -151,11 +152,11 @@
 #endif /* TIME_WITH_SYS_TIME || (!TIME_WITH_SYS_TIME && !HAVE_SYS_TIME_H) */
 #if HAVE_UNISTD_H
 #ifdef HAVE_DEC_PROTO
-#undef _XOPEN_SOURCE_EXTENDED
+/*#undef _XOPEN_SOURCE_EXTENDED*/
 #endif /* HAVE_DEC_PROTO */
 #include <unistd.h>
 #ifdef HAVE_DEC_PROTO
-#define _XOPEN_SOURCE_EXTENDED 1
+/*#define _XOPEN_SOURCE_EXTENDED 1*/
 #endif /* HAVE_DEC_PROTO */
 #endif /* HAVE_UNISTD_H */
 #if HAVE_RESOLV_H
@@ -191,11 +192,11 @@
  * XXX _XOPEN_SOURCE needed for PTHREAD_MUTEX_ERRORCHECK, define after
  *     all other include files to avoid interference
  */
-#ifdef _XOPEN_SOURCE
-#undef _XOPEN_SOURCE
-#endif /* _XOPEN_SOURCE */
-#define _XOPEN_SOURCE 500
-#undef _FEATURES_H
+/*#ifdef _XOPEN_SOURCE*/
+/*#undef _XOPEN_SOURCE*/
+/*#endif*/ /* _XOPEN_SOURCE */
+/*#define _XOPEN_SOURCE 500*/
+/*#undef _FEATURES_H*/
 #endif /* HAVE_LINUX_ECCENTRICITIES */
 
 #include <pthread.h>
--- lib/address.c
+++ lib/address.c
@@ -45,9 +45,9 @@
 
 #include "interposition.h"
 
-#ifndef __USE_GNU
-#define __USE_GNU /* XXX for RTLD_NEXT on Linux */
-#endif /* !__USE_GNU */
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE /* XXX for RTLD_NEXT on Linux */
+#endif /* !_GNU_SOURCE */
 #include <dlfcn.h>
 
 static const char rcsid[] =
