--- libscompat/getifaddrs.c
+++ libscompat/getifaddrs.c
@@ -83,7 +83,7 @@
 #define lifr_index iflr_index
 #endif
 
-#ifdef AF_NETLINK
+#if 0 /*def AF_NETLINK*/
 
 /*
  * The linux - AF_NETLINK version of getifaddrs - from Usagi.
@@ -1240,7 +1240,7 @@
 void ROKEN_LIB_FUNCTION
 freeifaddrs(struct ifaddrs *ifp)
 {
-#ifdef AF_NETLINK
+#if 0 /*def AF_NETLINK*/
     /* AF_NETLINK method uses a single allocation for all interfaces */
     free(ifp);
 #else
