--- libtransmission/Makefile.in
+++ libtransmission/Makefile.in
@@ -121,7 +121,7 @@
 	list.$(OBJEXT) log.$(OBJEXT) magnet.$(OBJEXT) \
 	makemeta.$(OBJEXT) metainfo.$(OBJEXT) natpmp.$(OBJEXT) \
 	net.$(OBJEXT) peer-io.$(OBJEXT) peer-mgr.$(OBJEXT) \
-	peer-msgs.$(OBJEXT) platform.$(OBJEXT) \
+	peer-msgs.$(OBJEXT) platform.$(OBJEXT) platform-quota.$(OBJEXT) \
 	port-forwarding.$(OBJEXT) ptrarray.$(OBJEXT) quark.$(OBJEXT) \
 	resume.$(OBJEXT) rpcimpl.$(OBJEXT) rpc-server.$(OBJEXT) \
 	session.$(OBJEXT) stats.$(OBJEXT) torrent.$(OBJEXT) \
@@ -761,6 +761,7 @@
   peer-mgr.c \
   peer-msgs.c \
   platform.c \
+  platform-quota.c \
   port-forwarding.c \
   ptrarray.c \
   quark.c \
@@ -818,6 +819,7 @@
   peer-mgr.h \
   peer-msgs.h \
   platform.h \
+  platform-quota.h \
   port-forwarding.h \
   ptrarray.h \
   quark.h \
