--- configure.ac
+++ configure.ac
@@ -155,16 +155,6 @@
 AC_CHECK_FUNC([gethostbyname], [],
   [AC_CHECK_LIB([nsl], [gethostbyname], [], [AC_MSG_ERROR([No nameservice library found])])])
 
-# Find resolver library.
-AC_CHECK_FUNC([res_query], [RESOLV_LIBS=""], [
-  AC_CHECK_LIB([resolv], [__res_query], [RESOLV_LIBS="-lresolv"], [
-    AC_CHECK_LIB([resolv], [res_query], [RESOLV_LIBS="-lresolv"], [
-      AC_CHECK_LIB([bind], [res_query], [RESOLV_LIBS="-lbind"], [
-        AC_MSG_ERROR([No resolver library found])
-      ])
-    ])
-  ])
-])
 dnl MacOS has res_query in libc, but needs libresolv for dn_expand().
 AS_IF([test "x" = "x$RESOLV_LIBS"], [
   AC_CHECK_LIB([resolv], [dn_expand], [RESOLV_LIBS="-lresolv"])
