From 7876bddff32ea262e028701a7029a9a3fb3f000d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Kuzn=C3=ADk?= <ondra@mistotebe.net>
Date: Wed, 9 Apr 2014 19:01:18 +0100
Subject: [PATCH] Fix regex probes always matching (#19)

--- probe.c
+++ probe.c
@@ -229,7 +229,7 @@ static int regex_probe(const char *p, int len, struct proto *proto)
     for (; *probe && regexec(*probe, p, 0, &pos, REG_STARTEND); probe++)
         /* try them all */;
 
-    return (probe != NULL);
+    return (*probe != NULL);
 }
 
 /* 
