--- sslh-main.c
+++ sslh-main.c
@@ -522,13 +522,10 @@
    num_addr_listen = start_listen_sockets(&listen_sockets, addr_listen);
 
    if (!foreground) {
-       if (fork() > 0) exit(0); /* Detach */
-
-       /* New session -- become group leader */
-       if (getuid() == 0) {
-           res = setsid();
-           CHECK_RES_DIE(res, "setsid: already process leader");
-       }
+      if (daemon(0,0) < 0) {
+         fprintf(stderr, "Failed to daemonize: %s\n", strerror(errno));
+         exit(1);
+      }
    }
 
    setup_signals();
