--- etc/fstab.orig	2007-05-29 16:46:27.000000000 +0200
+++ etc/fstab	2007-07-09 17:45:18.000000000 +0200
@@ -1,6 +1,6 @@
 # /etc/fstab: static file system information.
 #
 # <file system> <mount point>   <type>  <options>               <dump>  <pass>
-/dev/root        /              auto    defaults,errors=continue  0       0
-proc             /proc          proc    defaults                  0       0
+proc             /proc          proc    nosuid,nodev,noexec	          0       0
 ramfs		     /var		    ramfs	defaults			      0       0
+sysfs		 /sys		sysfs	nosuid,nodev,noexec	  0	  0
