--- Makefile.am.orig	2009-04-27 14:26:41.000000000 +0200
+++ Makefile.am	2009-05-22 12:38:18.000000000 +0200
@@ -38,7 +38,7 @@
 bin_SCRIPTS = curl-config
 
 SUBDIRS = lib src
-DIST_SUBDIRS = $(SUBDIRS) tests include packages docs
+DIST_SUBDIRS = $(SUBDIRS) tests include packages
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
@@ -53,10 +53,10 @@
 	done)
 
 html:
-	cd docs; make html
+	#cd docs; make html
 
 pdf:
-	cd docs; make pdf
+	#cd docs; make pdf
 
 check: test examples
 
@@ -81,7 +81,7 @@
 endif
 
 examples:
-	@(cd docs/examples; $(MAKE) check)
+	#@(cd docs/examples; $(MAKE) check)
 
 clean-local:
 	@(cd tests; $(MAKE) clean)
@@ -138,12 +138,12 @@
 # We extend the standard install with a custom hook:
 install-data-hook:
 	cd include && $(MAKE) install
-	cd docs && $(MAKE) install
+	#cd docs && $(MAKE) install
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
 	cd include && $(MAKE) uninstall
-	cd docs && $(MAKE) uninstall
+	#cd docs && $(MAKE) uninstall
 
 ca-bundle: lib/mk-ca-bundle.pl
 	@echo "generate a fresh ca-bundle.crt"
--- Makefile.in.orig	2009-05-18 09:07:46.000000000 +0200
+++ Makefile.in	2009-05-22 12:38:18.000000000 +0200
@@ -298,7 +298,7 @@
 
 bin_SCRIPTS = curl-config
 SUBDIRS = lib src
-DIST_SUBDIRS = $(SUBDIRS) tests include packages docs
+DIST_SUBDIRS = $(SUBDIRS) tests include packages
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
 all: all-recursive
@@ -779,10 +779,10 @@
 	done)
 
 html:
-	cd docs; make html
+	#cd docs; make html
 
 pdf:
-	cd docs; make pdf
+	#cd docs; make pdf
 
 check: test examples
 
@@ -802,7 +802,7 @@
 @CROSSCOMPILING_FALSE@	@(cd tests; $(MAKE) all torture-test)
 
 examples:
-	@(cd docs/examples; $(MAKE) check)
+	#@(cd docs/examples; $(MAKE) check)
 
 clean-local:
 	@(cd tests; $(MAKE) clean)
@@ -859,12 +859,12 @@
 # We extend the standard install with a custom hook:
 install-data-hook:
 	cd include && $(MAKE) install
-	cd docs && $(MAKE) install
+	#cd docs && $(MAKE) install
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
 	cd include && $(MAKE) uninstall
-	cd docs && $(MAKE) uninstall
+	#cd docs && $(MAKE) uninstall
 
 ca-bundle: lib/mk-ca-bundle.pl
 	@echo "generate a fresh ca-bundle.crt"
