--- Makefile.am.orig	2008-09-01 09:46:51.000000000 +0200
+++ Makefile.am	2008-09-24 23:30:00.000000000 +0200
@@ -32,7 +32,7 @@
 bin_SCRIPTS = curl-config
 
 SUBDIRS = lib src
-DIST_SUBDIRS = $(SUBDIRS) tests include packages docs
+DIST_SUBDIRS = $(SUBDIRS) tests include packages
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
@@ -47,10 +47,10 @@
 	done)
 
 html:
-	cd docs; make html
+	#cd docs; make html
 
 pdf:
-	cd docs; make pdf
+	#cd docs; make pdf
 
 check: test examples
 
@@ -75,7 +75,7 @@
 endif
 
 examples:
-	@(cd docs/examples; $(MAKE) check)
+	#@(cd docs/examples; $(MAKE) check)
 
 #
 # Build source and binary rpms. For rpm-3.0 and above, the ~/.rpmmacros
@@ -128,12 +128,12 @@
 # We extend the standard install with a custom hook:
 install-data-hook:
 	cd include && $(MAKE) install
-	cd docs && $(MAKE) install
+	#cd docs && $(MAKE) install
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
 	cd include && $(MAKE) uninstall
-	cd docs && $(MAKE) uninstall
+	#cd docs && $(MAKE) uninstall
 
 ca-bundle: lib/mk-ca-bundle.pl
 	@echo "generate a fresh ca-bundle.crt"
--- Makefile.in.orig	2008-09-01 16:39:16.000000000 +0200
+++ Makefile.in	2008-09-24 23:30:00.000000000 +0200
@@ -268,7 +268,7 @@
 
 bin_SCRIPTS = curl-config
 SUBDIRS = lib src
-DIST_SUBDIRS = $(SUBDIRS) tests include packages docs
+DIST_SUBDIRS = $(SUBDIRS) tests include packages
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
 all: all-recursive
@@ -749,10 +749,10 @@
 	done)
 
 html:
-	cd docs; make html
+	#cd docs; make html
 
 pdf:
-	cd docs; make pdf
+	#cd docs; make pdf
 
 check: test examples
 
@@ -772,7 +772,7 @@
 @CROSSCOMPILING_FALSE@	@(cd tests; $(MAKE) all torture-test)
 
 examples:
-	@(cd docs/examples; $(MAKE) check)
+	#@(cd docs/examples; $(MAKE) check)
 
 #
 # Build source and binary rpms. For rpm-3.0 and above, the ~/.rpmmacros
@@ -825,12 +825,12 @@
 # We extend the standard install with a custom hook:
 install-data-hook:
 	cd include && $(MAKE) install
-	cd docs && $(MAKE) install
+	#cd docs && $(MAKE) install
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
 	cd include && $(MAKE) uninstall
-	cd docs && $(MAKE) uninstall
+	#cd docs && $(MAKE) uninstall
 
 ca-bundle: lib/mk-ca-bundle.pl
 	@echo "generate a fresh ca-bundle.crt"
