--- Makefile.am
+++ Makefile.am
@@ -158,7 +158,7 @@
 bin_SCRIPTS = curl-config
 
 SUBDIRS = lib src
-DIST_SUBDIRS = $(SUBDIRS) tests packages scripts include docs
+DIST_SUBDIRS = $(SUBDIRS)       packages scripts include
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
@@ -272,14 +272,14 @@
 # We extend the standard install with a custom hook:
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
-	(cd docs/libcurl && $(MAKE) install)
+#	(cd docs && $(MAKE) install)
+#	(cd docs/libcurl && $(MAKE) install)
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
 	(cd include && $(MAKE) uninstall)
-	(cd docs && $(MAKE) uninstall)
-	(cd docs/libcurl && $(MAKE) uninstall)
+#	(cd docs && $(MAKE) uninstall)
+#	(cd docs/libcurl && $(MAKE) uninstall)
 
 ca-bundle: $(srcdir)/scripts/mk-ca-bundle.pl
 	@echo "generating a fresh ca-bundle.crt"
--- Makefile.in
+++ Makefile.in
@@ -678,7 +678,7 @@
 
 bin_SCRIPTS = curl-config
 SUBDIRS = lib src
-DIST_SUBDIRS = $(SUBDIRS) tests packages scripts include docs
+DIST_SUBDIRS = $(SUBDIRS)       packages scripts include
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
 LIB_VAUTH_CFILES = \
@@ -1769,14 +1769,14 @@
 # We extend the standard install with a custom hook:
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
-	(cd docs/libcurl && $(MAKE) install)
+#	(cd docs && $(MAKE) install)
+#	(cd docs/libcurl && $(MAKE) install)
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
 	(cd include && $(MAKE) uninstall)
-	(cd docs && $(MAKE) uninstall)
-	(cd docs/libcurl && $(MAKE) uninstall)
+#	(cd docs && $(MAKE) uninstall)
+#	(cd docs/libcurl && $(MAKE) uninstall)
 
 ca-bundle: $(srcdir)/scripts/mk-ca-bundle.pl
 	@echo "generating a fresh ca-bundle.crt"
