--- configure.ac
+++ configure.ac
@@ -1542,7 +1542,7 @@
   fi
 
   if test "$OPENSSL_ENABLED" = "1"; then
-    if test -n "$LIB_OPENSSL"; then
+    if false && test -n "$LIB_OPENSSL"; then
        dnl when the ssl shared libs were found in a path that the run-time
        dnl linker doesn't search through, we need to add it to LD_LIBRARY_PATH
        dnl to prevent further configure tests to fail due to this
@@ -1780,7 +1780,7 @@
       if test "x$USE_GNUTLS" = "xyes"; then
         AC_MSG_NOTICE([detected GnuTLS version $version])
 
-        if test -n "$gtlslib"; then
+        if false && test -n "$gtlslib"; then
           dnl when shared libs were found in a path that the run-time
           dnl linker doesn't search through, we need to add it to
           dnl LD_LIBRARY_PATH to prevent further configure tests to fail
@@ -1890,7 +1890,7 @@
       CURL_LIBS="$CURL_LIBS -lpolarssl"
       LIBS="$LIBS -lpolarssl"
 
-      if test -n "$polarssllib"; then
+      if false && test -n "$polarssllib"; then
         dnl when shared libs were found in a path that the run-time
         dnl linker doesn't search through, we need to add it to
         dnl LD_LIBRARY_PATH to prevent further configure tests to fail
--- configure
+++ configure
@@ -20695,7 +20695,7 @@
   fi
 
   if test "$OPENSSL_ENABLED" = "1"; then
-    if test -n "$LIB_OPENSSL"; then
+    if false && test -n "$LIB_OPENSSL"; then
 
        LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$LIB_OPENSSL"
        export LD_LIBRARY_PATH
@@ -21793,7 +21793,7 @@
         { $as_echo "$as_me:${as_lineno-$LINENO}: detected GnuTLS version $version" >&5
 $as_echo "$as_me: detected GnuTLS version $version" >&6;}
 
-        if test -n "$gtlslib"; then
+        if false && test -n "$gtlslib"; then
 
           LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$gtlslib"
           export LD_LIBRARY_PATH
@@ -22042,7 +22042,7 @@
       CURL_LIBS="$CURL_LIBS -lpolarssl"
       LIBS="$LIBS -lpolarssl"
 
-      if test -n "$polarssllib"; then
+      if false && test -n "$polarssllib"; then
 
         LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$polarssllib"
         export LD_LIBRARY_PATH
