--- Makefile.am
+++ Makefile.am
@@ -141,7 +141,7 @@
 bin_SCRIPTS = curl-config
 
 SUBDIRS = lib src include
-DIST_SUBDIRS = $(SUBDIRS) tests packages docs
+DIST_SUBDIRS = $(SUBDIRS)       packages
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
@@ -160,10 +160,10 @@
 	done)
 
 html:
-	cd docs; make html
+	#cd docs; make html
 
 pdf:
-	cd docs; make pdf
+	#cd docs; make pdf
 
 check: test examples check-docs
 
@@ -191,7 +191,7 @@
 endif
 
 examples:
-	@(cd docs/examples; $(MAKE) check)
+	#@(cd docs/examples; $(MAKE) check)
 
 check-docs:
 	@(cd docs/libcurl; $(MAKE) check)
--- Makefile.in
+++ Makefile.in
@@ -598,7 +598,7 @@
 
 bin_SCRIPTS = curl-config
 SUBDIRS = lib src include
-DIST_SUBDIRS = $(SUBDIRS) tests packages docs
+DIST_SUBDIRS = $(SUBDIRS)       packages
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
 LIB_VTLS_CFILES = vtls/openssl.c vtls/gtls.c vtls/vtls.c vtls/nss.c     \
@@ -1302,10 +1302,10 @@
 	done)
 
 html:
-	cd docs; make html
+	#cd docs; make html
 
 pdf:
-	cd docs; make pdf
+	#cd docs; make pdf
 
 check: test examples check-docs
 
@@ -1328,7 +1328,7 @@
 @CROSSCOMPILING_FALSE@	@(cd tests; $(MAKE) all am-test)
 
 examples:
-	@(cd docs/examples; $(MAKE) check)
+	#@(cd docs/examples; $(MAKE) check)
 
 check-docs:
 	@(cd docs/libcurl; $(MAKE) check)
@@ -1392,12 +1392,12 @@
 # We extend the standard install with a custom hook:
 install-data-hook:
 	cd include && $(MAKE) install
-	cd docs && $(MAKE) install
+	#cd docs && $(MAKE) install
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
 	cd include && $(MAKE) uninstall
-	cd docs && $(MAKE) uninstall
+	#cd docs && $(MAKE) uninstall
 
 ca-bundle: lib/mk-ca-bundle.pl
 	@echo "generating a fresh ca-bundle.crt"
