--- Makefile.am.orig	2008-03-21 16:09:08.000000000 +0100
+++ Makefile.am	2008-03-21 16:10:57.000000000 +0100
@@ -29,7 +29,7 @@
 bin_SCRIPTS = curl-config
 
 SUBDIRS = lib src
-DIST_SUBDIRS = $(SUBDIRS) tests include packages docs
+DIST_SUBDIRS = $(SUBDIRS) tests include packages
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
@@ -44,10 +44,10 @@
 	done)
 
 html:
-	cd docs; make html
+	#cd docs; make html
 
 pdf:
-	cd docs; make pdf
+	#cd docs; make pdf
 
 check: test examples
 
@@ -72,7 +72,7 @@
 endif
 
 examples:
-	@(cd docs/examples; $(MAKE) all)
+	#@(cd docs/examples; $(MAKE) all)
 
 #
 # Build source and binary rpms. For rpm-3.0 and above, the ~/.rpmmacros
@@ -125,9 +125,9 @@
 # We extend the standard install with a custom hook:
 install-data-hook:
 	cd include && $(MAKE) install
-	cd docs && $(MAKE) install
+	#cd docs && $(MAKE) install
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
 	cd include && $(MAKE) uninstall
-	cd docs && $(MAKE) uninstall
+	#cd docs && $(MAKE) uninstall
--- Makefile.in.orig	2008-03-21 16:09:58.000000000 +0100
+++ Makefile.in	2008-03-21 16:10:44.000000000 +0100
@@ -260,7 +260,7 @@
 
 bin_SCRIPTS = curl-config
 SUBDIRS = lib src
-DIST_SUBDIRS = $(SUBDIRS) tests include packages docs
+DIST_SUBDIRS = $(SUBDIRS) tests include packages
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
 all: all-recursive
@@ -741,10 +741,10 @@
 	done)
 
 html:
-	cd docs; make html
+	#cd docs; make html
 
 pdf:
-	cd docs; make pdf
+	#cd docs; make pdf
 
 check: test examples
 
@@ -764,7 +764,7 @@
 @CROSSCOMPILING_FALSE@	@(cd tests; $(MAKE) all torture-test)
 
 examples:
-	@(cd docs/examples; $(MAKE) all)
+	#@(cd docs/examples; $(MAKE) all)
 
 #
 # Build source and binary rpms. For rpm-3.0 and above, the ~/.rpmmacros
@@ -817,12 +817,12 @@
 # We extend the standard install with a custom hook:
 install-data-hook:
 	cd include && $(MAKE) install
-	cd docs && $(MAKE) install
+	#cd docs && $(MAKE) install
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
 	cd include && $(MAKE) uninstall
-	cd docs && $(MAKE) uninstall
+	#cd docs && $(MAKE) uninstall
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
