--- lib/Makefile.in	2011-02-02 13:22:52.000000000 +0100
+++ lib/Makefile.in	2011-02-17 20:09:56.000000000 +0100
@@ -73,7 +73,7 @@
 am__installdirs = "$(DESTDIR)$(libdir)"
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES) $(noinst_LTLIBRARIES)
-libcurl_la_LIBADD =
+libcurl_la_LIBADD = -ldl
 am__objects_1 = libcurl_la-file.lo libcurl_la-timeval.lo \
 	libcurl_la-base64.lo libcurl_la-hostip.lo \
 	libcurl_la-progress.lo libcurl_la-formdata.lo \
