--- configure
+++ configure
@@ -20533,8 +20533,8 @@
     PKGCONFIG="no"
 
     if test x$cross_compiling = xyes; then
-            # Extract the first word of "${host}-pkg-config", so it can be a program name with args.
-set dummy ${host}-pkg-config; ac_word=$2
+            # Extract the first word of "pkg-config", so it can be a program name with args.
+set dummy pkg-config; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
 $as_echo_n "checking for $ac_word... " >&6; }
 if test "${ac_cv_path_PKGCONFIG+set}" = set; then :
@@ -21964,8 +21964,8 @@
     PKGCONFIG="no"
 
     if test x$cross_compiling = xyes; then
-            # Extract the first word of "${host}-pkg-config", so it can be a program name with args.
-set dummy ${host}-pkg-config; ac_word=$2
+            # Extract the first word of "pkg-config", so it can be a program name with args.
+set dummy pkg-config; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
 $as_echo_n "checking for $ac_word... " >&6; }
 if test "${ac_cv_path_PKGCONFIG+set}" = set; then :
@@ -22740,8 +22740,8 @@
     PKGCONFIG="no"
 
     if test x$cross_compiling = xyes; then
-            # Extract the first word of "${host}-pkg-config", so it can be a program name with args.
-set dummy ${host}-pkg-config; ac_word=$2
+            # Extract the first word of "pkg-config", so it can be a program name with args.
+set dummy pkg-config; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
 $as_echo_n "checking for $ac_word... " >&6; }
 if test "${ac_cv_path_PKGCONFIG+set}" = set; then :
@@ -23203,8 +23203,8 @@
     PKGCONFIG="no"
 
     if test x$cross_compiling = xyes; then
-            # Extract the first word of "${host}-pkg-config", so it can be a program name with args.
-set dummy ${host}-pkg-config; ac_word=$2
+            # Extract the first word of "pkg-config", so it can be a program name with args.
+set dummy pkg-config; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
 $as_echo_n "checking for $ac_word... " >&6; }
 if test "${ac_cv_path_PKGCONFIG+set}" = set; then :
@@ -23547,8 +23547,8 @@
     PKGCONFIG="no"
 
     if test x$cross_compiling = xyes; then
-            # Extract the first word of "${host}-pkg-config", so it can be a program name with args.
-set dummy ${host}-pkg-config; ac_word=$2
+            # Extract the first word of "pkg-config", so it can be a program name with args.
+set dummy pkg-config; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
 $as_echo_n "checking for $ac_word... " >&6; }
 if test "${ac_cv_path_PKGCONFIG+set}" = set; then :
