--- lib/Makefile.in.orig	2008-10-24 23:27:36.000000000 +0200
+++ lib/Makefile.in	2008-10-24 23:26:13.000000000 +0200
@@ -62,7 +62,7 @@
 am__installdirs = "$(DESTDIR)$(libdir)"
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES)
-libcurl_la_LIBADD =
+libcurl_la_LIBADD = -ldl
 am__objects_1 = file.lo timeval.lo base64.lo hostip.lo progress.lo \
 	formdata.lo cookie.lo http.lo sendf.lo ftp.lo url.lo dict.lo \
 	if2ip.lo speedcheck.lo ldap.lo ssluse.lo version.lo getenv.lo \
