--- src/Makefile.in
+++ src/Makefile.in
@@ -231,7 +231,7 @@
 libcurltool_la_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC \
 	$(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=link $(CCLD) \
 	$(libcurltool_la_CFLAGS) $(CFLAGS) $(libcurltool_la_LDFLAGS) \
-	$(LDFLAGS) -o $@
+	$(LDFLAGS) $(STATIC_LDFLAGS) -o $@
 @BUILD_UNITTESTS_TRUE@am_libcurltool_la_rpath =
 am__objects_6 = slist_wc.$(OBJEXT) tool_binmode.$(OBJEXT) \
 	tool_bname.$(OBJEXT) tool_cb_dbg.$(OBJEXT) \
