--- lib/Makefile.in.orig	2011-06-21 08:46:10.000000000 +0200
+++ lib/Makefile.in	2011-06-26 16:13:06.907417002 +0200
@@ -75,7 +75,7 @@
 am__installdirs = "$(DESTDIR)$(libdir)"
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES) $(noinst_LTLIBRARIES)
-libcurl_la_LIBADD =
+libcurl_la_LIBADD = -ldl
 am__objects_1 = file.lo timeval.lo base64.lo hostip.lo progress.lo \
 	formdata.lo cookie.lo http.lo sendf.lo ftp.lo url.lo dict.lo \
 	if2ip.lo speedcheck.lo ldap.lo ssluse.lo version.lo getenv.lo \
