--- configure
+++ configure
@@ -2750,6 +2750,24 @@
 ## Main body of script. ##
 ## -------------------- ##
 
+if test "x$CPPFLAGS" = "x"; then
+    CPPFLAGS="-D_REENTRANT -D_GNU_SOURCE"
+else
+    curl_addto_bugger="-D_REENTRANT -D_GNU_SOURCE"
+    for i in $curl_addto_bugger; do
+	apr_addto_duplicate="0"
+	for j in $CPPFLAGS; do
+	    if test "x$i" = "x$j"; then
+		apr_addto_duplicate="1"
+		break
+	    fi
+	done
+	if test $apr_addto_duplicate = "0"; then
+	    CPPFLAGS="$CPPFLAGS $i"
+	fi
+    done
+fi
+
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
