--- curl-config.in
+++ curl-config.in
@@ -25,6 +25,7 @@
 
 prefix=@prefix@
 exec_prefix=@exec_prefix@
+libdir=@libdir@
 includedir=@includedir@
 cppflag_curl_staticlib=@CPPFLAG_CURL_STATICLIB@
 
@@ -157,8 +158,8 @@
         ;;
 
     --libs)
-        if test "X@libdir@" != "X/usr/lib" -a "X@libdir@" != "X/usr/lib64"; then
-           CURLLIBDIR="-L@libdir@ "
+        if test "X${libdir}" != "X/usr/lib" -a "X${libdir}" != "X/usr/lib64"; then
+           CURLLIBDIR="-L${libdir} "
         else
            CURLLIBDIR=""
         fi
@@ -174,7 +175,7 @@
 
     --static-libs)
         if test "X@ENABLE_STATIC@" != "Xno" ; then
-          echo @libdir@/libcurl.@libext@ @LDFLAGS@ @LIBCURL_LIBS@
+          echo ${libdir}/libcurl.@libext@ @LDFLAGS@ @LIBCURL_LIBS@
         else
           echo "curl was built with static libraries disabled" >&2
           exit 1
