--- configure
+++ configure
@@ -12505,6 +12505,8 @@
     ;;
   esac
 
+  link_all_deplibs=yes # necessary because of uClibc-pthread bug
+
   ld_shlibs=yes
 
   # On some targets, GNU ld is compatible enough with the native linker
@@ -23544,7 +23546,7 @@
 done
 
 
-      LIBS="-lcyassl -lm $LIBS"
+      LIBS="-lcyassl -lpthread -lm $LIBS"
 
       if test -n "$cyassllib"; then
                                         if test "x$cross_compiling" != "xyes"; then
