--- configure
+++ configure
@@ -12546,6 +12546,8 @@
     ;;
   esac
 
+  link_all_deplibs=yes # necessary because of uClibc-pthread bug
+
   ld_shlibs=yes
 
   # On some targets, GNU ld is compatible enough with the native linker
@@ -23788,7 +23790,7 @@
 done
 
 
-      LIBS="-l$cyassllibname -lm $LIBS"
+      LIBS="-l$cyassllibname -lpthread -lm $LIBS"
 
       if test -n "$cyassllib"; then
                                         if test "x$cross_compiling" != "xyes"; then
