--- configure
+++ configure
@@ -12443,6 +12443,8 @@
     ;;
   esac
 
+  link_all_deplibs=yes # necessary because of uClibc-pthread bug
+
   ld_shlibs=yes
 
   # On some targets, GNU ld is compatible enough with the native linker
@@ -23477,7 +23479,7 @@
 
 
 
-      LIBS="-lcyassl -lm $LIBS"
+      LIBS="-lcyassl -lpthread -lm $LIBS"
 
       if test -n "$cyassllib"; then
                                         if test "x$cross_compiling" != "xyes"; then
