--- configure
+++ configure
@@ -13284,7 +13284,6 @@
 	fi
 	if test "$enableval" = "yes" ; then
 		ENABLE_OWPHP="true"
-	fi
 	if test -z "$SWIG"  ; then
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Cannot find swig program. Look for it at http://www.swig.org or use CPAN" >&5
 $as_echo "$as_me: WARNING: Cannot find swig program. Look for it at http://www.swig.org or use CPAN" >&2;}
@@ -13292,6 +13291,7 @@
 $as_echo "$as_me: WARNING: OWPHP is disabled because swig is not found" >&2;}
 		ENABLE_OWPHP="false"
 	fi
+	fi
 
 else
 
