--- etc/init.d/rc.S
+++ etc/init.d/rc.S
@@ -41,14 +41,16 @@
 echo "source files in group ${gruppe} ...";
 if ls /etc/init.d/S${gruppe}[0-9]-* 2>/dev/null ; then
 for skript in /etc/init.d/S${gruppe}[0-9]-* ; do
+echo "processing ${skript}"
 . ${skript}
 done
 fi
-echo "exceute files in group ${gruppe} ...";
+echo "executing files in group ${gruppe} ...";
 if ls /etc/init.d/E${gruppe}[0-9]-* 2>/dev/null ; then
 for skript in /etc/init.d/E${gruppe}[0-9]-* ; do
+echo "executing ${skript}"
 if ! sh ${skript} ; then
-echo "exceute ${skript} failed.";
+echo "executing ${skript} failed.";
 fi
 done
 fi
