--- include/polarssl/config.h
+++ include/polarssl/config.h
@@ -136,7 +136,7 @@
  *
  * Enable all SSL/TLS debugging messages.
  */
-#define POLARSSL_DEBUG_MSG
+/* #define POLARSSL_DEBUG_MSG */
 
 /**
  * \def POLARSSL_GENPRIME
@@ -203,7 +203,7 @@
  *
  * Enable the checkup functions (*_self_test).
  */
-#define POLARSSL_SELF_TEST
+/* #define POLARSSL_SELF_TEST */
 
 /**
  * \def POLARSSL_X509_ALLOW_UNSUPPORTED_CRITICAL_EXTENSION
@@ -357,7 +357,7 @@
  *
  * This module provides debugging functions.
  */
-#define POLARSSL_DEBUG_C
+/* #define POLARSSL_DEBUG_C */
 
 /**
  * \def POLARSSL_DES_C
@@ -502,7 +502,7 @@
  *
  * This modules adds support for the VIA PadLock on x86.
  */
-#define POLARSSL_PADLOCK_C
+/* #define POLARSSL_PADLOCK_C */
 
 /**
  * \def POLARSSL_PEM_C
