--- etc/init.d/S01-head
+++ etc/init.d/S01-head
@@ -306,6 +306,10 @@
 tffs_nodes_list="$tffs_nodes_list $((0xC9)),websrv_ssl_key.pem"
 tffs_nodes_list="$tffs_nodes_list $((0xCA)),websrv_ssl_cert.pem"
 #########################################################################
+## Freetz
+#########################################################################
+tffs_nodes_list="$tffs_nodes_list $((0x3C)),freetz"
+#########################################################################
 ## NODES anlegen, wenn nicht als _file_ existent
 #########################################################################
 for node in ${tffs_nodes_list} ; do
