--- zebra/kernel_null.c
+++ zebra/kernel_null.c
@@ -3,6 +3,8 @@
 #include <zebra.h>
 #include <log.h>
 
+#include "prefix.h"
+
 #include "zebra/zserv.h"
 #include "zebra/rt.h"
 #include "zebra/redistribute.h"
@@ -10,11 +12,13 @@
 
 int kernel_add_ipv4 (struct prefix *a, struct rib *b) { return 0; }
 #pragma weak kernel_delete_ipv4 = kernel_add_ipv4
+#ifdef HAVE_IPV6
 int kernel_add_ipv6 (struct prefix *a, struct rib *b) { return 0; }
 #pragma weak kernel_delete_ipv6 = kernel_add_ipv6
 int kernel_delete_ipv6_old (struct prefix_ipv6 *dest, struct in6_addr *gate,
                             unsigned int index, int flags, int table)
 { return 0; }
+#endif
 
 int kernel_add_route (struct prefix_ipv4 *a, struct in_addr *b, int c, int d)
 { return 0; }
