--- configure
+++ configure
@@ -19350,6 +19350,7 @@
 fi
 
 
+if false; then
 if test "${glibc}" = "yes"; then
    ac_fn_c_check_header_mongrel "$LINENO" "execinfo.h" "ac_cv_header_execinfo_h" "$ac_includes_default"
 if test "x$ac_cv_header_execinfo_h" = x""yes; then :
@@ -19369,6 +19370,7 @@
 
 
 fi
+fi
 
 ac_fn_c_check_header_compile "$LINENO" "malloc.h" "ac_cv_header_malloc_h" "#ifdef SUNOS_5
 #define _XPG4_2
