--- configure.ac.orig	2008-09-06 06:03:07.000000000 +0000
+++ configure.ac	2009-06-23 09:31:39.000000000 +0000
@@ -50,7 +50,7 @@
 dnl Checks for header files.
 
 AC_HEADER_STDC
-AC_CHECK_HEADERS(getopt.h libintl.h limits.h regex.h sys/param.h wchar.h wctype.h)
+AC_CHECK_HEADERS(getopt.h limits.h regex.h sys/param.h wchar.h wctype.h)
 
 dnl Checks for options.
 
--- configure.orig	2008-09-06 06:03:13.000000000 +0000
+++ configure	2009-06-23 09:31:31.000000000 +0000
@@ -6182,7 +6182,7 @@
 
 
 
-for ac_header in getopt.h libintl.h limits.h regex.h sys/param.h wchar.h wctype.h
+for ac_header in getopt.h limits.h regex.h sys/param.h wchar.h wctype.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
