--- etc/init.d/rc.net
+++ etc/init.d/rc.net
@@ -49,9 +49,15 @@
    WLAN_TEST=0
 fi
 
+# Do we have a websrv or was ist stripped from the firmware?
+_websrv=$(basename $(which /sbin/websrv) 2> /dev/null)
+# Do we have a UPnP server (igdd) or was ist stripped from the firmware?
+_igdd=$(basename $(which igdd) 2> /dev/null)
+
 
 igddstart()
 {
+	 [ "$_igdd" ] || return;
    if [ "`pidof igdd`" = "" ] ; then
       igdd $VERBOSEPARAM
       sleep 1
@@ -208,15 +226,16 @@
 stopall()
 {
    if [ "$CONFIG_DSL" ] ; then
-      AVMDAEMONS="avmike dsld igdd websrv multid"
+      AVMDAEMONS="avmike dsld $_igdd $_websrv multid"
    else
-      AVMDAEMONS="igdd websrv multid"
+      AVMDAEMONS="$_igdd $_websrv multid"
    fi
    termavmwait 999 $AVMDAEMONS
 
    if [ "$CONFIG_WLAN" = y ] ; then
     wlanstop
    fi
+   [ "$_websrv" ] || /etc/init.d/rc.websrv stop
 }
 
 startall()
@@ -254,9 +273,15 @@
      ctlmgr $NICEPARAM $VERBOSEPARAM
    fi
 
+   if [ "$_websrv" ]; then
    if [ "`pidof websrv`" = "" ] ; then
      websrv $NICEPARAM
    fi
+   else
+     if [ "$(/etc/init.d/rc.websrv status)" == "stopped" ]; then
+       /etc/init.d/rc.websrv start
+     fi
+   fi
    fi 
 
    igddstart
@@ -313,9 +338,9 @@
       if [ "$1" ] ; then
          AVMDAEMONS="$*"
       elif [ "$CONFIG_DSL" ] ; then
-         AVMDAEMONS="avmike dsld igdd multid"
+         AVMDAEMONS="avmike dsld $_igdd multid"
       else
-         AVMDAEMONS="igdd multid"
+         AVMDAEMONS="$_igdd multid"
       fi
 
       # stop the daemon with -s
