--- setup.py
+++ setup.py
@@ -788,6 +788,7 @@
             else:
                 readline_extra_link_args = ()
 
+            readline_extra_library_dirs = []
             readline_libs = ['readline']
             if readline_termcap_library:
                 pass # Issue 7384: Already linked against curses or tinfo.
@@ -797,8 +798,9 @@
                                                      ['/usr/lib/termcap'],
                                                      'termcap'):
                 readline_libs.append('termcap')
+                readline_extra_library_dirs.append('/usr/lib/termcap')
             exts.append( Extension('readline', ['readline.c'],
-                                   library_dirs=['/usr/lib/termcap'],
+                                   library_dirs=readline_extra_library_dirs,
                                    extra_link_args=readline_extra_link_args,
                                    libraries=readline_libs) )
         else:
