--- configure_	2016-10-09 03:16:06.297793808 +0200
+++ configure	2016-10-09 03:17:25.354084682 +0200
@@ -18759,6 +18759,7 @@
 if test "x$ac_cv_func_gethostbyname" = xyes; then :
 
 else
+if false; then
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for gethostbyname in -lnsl" >&5
 $as_echo_n "checking for gethostbyname in -lnsl... " >&6; }
 if ${ac_cv_lib_nsl_gethostbyname+:} false; then :
@@ -18798,6 +18799,7 @@
 if test "x$ac_cv_lib_nsl_gethostbyname" = xyes; then :
   LIBNSL="-lnsl"
 fi
+fi
 
 fi
 
