--- ./tools/Makefile.in_	2016-10-09 03:35:35.129761214 +0200
+++ ./tools/Makefile.in	2016-10-09 03:35:47.337322908 +0200
@@ -166,7 +166,7 @@
   done | $(am__uniquify_input)`
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = locktest rpcdebug nlmtest mountstats nfs-iostat rpcgen
+DIST_SUBDIRS = rpcdebug nlmtest mountstats nfs-iostat rpcgen
 am__DIST_COMMON = $(srcdir)/Makefile.in
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 am__relativize = \
@@ -378,7 +378,7 @@
 top_srcdir = @top_srcdir@
 unitdir = @unitdir@
 OPTDIRS = $(am__append_1)
-SUBDIRS = locktest rpcdebug nlmtest mountstats nfs-iostat $(OPTDIRS)
+SUBDIRS = rpcdebug nlmtest mountstats nfs-iostat $(OPTDIRS)
 MAINTAINERCLEANFILES = Makefile.in
 all: all-recursive
 
