--- loginutils/adduser.c	2007-09-03 13:48:47.000000000 +0200
+++ loginutils/adduser.c	2007-10-22 22:43:49.000000000 +0200
@@ -11,6 +11,7 @@
 #include "libbb.h"
 
 #define OPT_DONT_SET_PASS  (1 << 4)
+#define OPT_SYSTEM_ACCOUNT (1 << 5)
 #define OPT_DONT_MAKE_HOME (1 << 6)
 
 
@@ -18,13 +19,24 @@
 /* EDR recoded such that the uid may be passed in *p */
 static int passwd_study(const char *filename, struct passwd *p)
 {
-	enum { min = 500, max = 65000 };
+	//enum { min = 500, max = 65000 };
 	FILE *passwd;
 	/* We are using reentrant fgetpwent_r() in order to avoid
 	 * pulling in static buffers from libc (think static build here) */
 	char buffer[256];
 	struct passwd pw;
 	struct passwd *result;
+	int min, max;
+
+        if (option_mask32 & OPT_SYSTEM_ACCOUNT) {
+		min = 1;
+		max = 999;
+	}
+	else {
+		min = 1000;
+		max = 65000;
+	}
+
 
 	passwd = xfopen(filename, "r");
 
diff
--- include/usage.h	2007-09-03 13:48:46.000000000 +0200
+++ include/usage.h	2007-10-22 23:08:04.000000000 +0200
@@ -27,7 +27,7 @@
        "	-g GECOS	GECOS field\n" \
        "	-s SHELL	Login shell\n" \
        "	-G GROUP	Add user to existing group\n" \
-       "	-S		Create a system user (ignored)\n" \
+       "	-S		Create a system user\n" \
        "	-D		Do not assign a password (logins still possible via ssh)\n" \
        "	-H		Do not create home directory"
 
diff
