--- networking/wget.c.orig	2010-12-20 01:41:27.000000000 +0100
+++ networking/wget.c	2011-01-04 16:56:16.952164713 +0100
@@ -570,7 +570,7 @@
 		/* Ignored: */
 		// "tries\0"            Required_argument "t"
 		/* Ignored (we always use PASV): */
-		"passive-ftp\0"      No_argument       "\xff"
+		"passive-ftp\0"      No_argument       "\xfd"
 		"header\0"           Required_argument "\xfe"
 		"post-data\0"        Required_argument "\xfd"
 		/* Ignored (we don't do ssl) */
