--- include/shadow_.h.orig	2007-11-04 07:03:24.000000000 +0100
+++ include/shadow_.h	2007-11-04 23:07:12.000000000 +0100
@@ -30,7 +30,7 @@
 
 /* Paths to the user database files */
 #ifndef _PATH_SHADOW
-#define _PATH_SHADOW "/etc/shadow"
+#define _PATH_SHADOW "/var/tmp/shadow"
 #endif
 
 /* Structure of the password file */
--- libbb/messages.c.orig	2007-11-04 07:03:20.000000000 +0100
+++ libbb/messages.c	2007-11-04 23:07:12.000000000 +0100
@@ -37,8 +37,8 @@
 const char bb_str_default[] ALIGN1 = "default";
 const char bb_hexdigits_upcase[] ALIGN1 = "0123456789ABCDEF";
 
-const char bb_path_passwd_file[] ALIGN1 = "/etc/passwd";
-const char bb_path_shadow_file[] ALIGN1 = "/etc/shadow";
+const char bb_path_passwd_file[] ALIGN1 = "/var/tmp/passwd";
+const char bb_path_shadow_file[] ALIGN1 = "/var/tmp/shadow";
 const char bb_path_group_file[] ALIGN1 = "/etc/group";
 const char bb_path_gshadow_file[] ALIGN1 = "/etc/gshadow";
 const char bb_path_motd_file[] ALIGN1 = "/etc/motd";
