--- include/shadow_.h.orig	2009-12-12 01:45:08.000000000 +0000
+++ include/shadow_.h	2009-12-13 21:57:26.000000000 +0000
@@ -39,7 +39,7 @@
 
 /* Paths to the user database files */
 #ifndef _PATH_SHADOW
-#define _PATH_SHADOW "/etc/shadow"
+#define _PATH_SHADOW "/var/tmp/shadow"
 #endif
 
 #define setspent    bb_internal_setspent
--- libbb/messages.c.orig	2009-12-12 21:13:28.000000000 +0000
+++ libbb/messages.c	2009-12-13 21:57:26.000000000 +0000
@@ -37,8 +37,8 @@
 const char bb_str_default[] ALIGN1 = "default";
 const char bb_hexdigits_upcase[] ALIGN1 = "0123456789ABCDEF";
 
-const char bb_path_passwd_file[] ALIGN1 = "/etc/passwd";
-const char bb_path_shadow_file[] ALIGN1 = "/etc/shadow";
+const char bb_path_passwd_file[] ALIGN1 = "/var/tmp/passwd";
+const char bb_path_shadow_file[] ALIGN1 = "/var/tmp/shadow";
 const char bb_path_group_file[] ALIGN1 = "/etc/group";
 const char bb_path_gshadow_file[] ALIGN1 = "/etc/gshadow";
 const char bb_path_motd_file[] ALIGN1 = "/etc/motd";
