--- include/shadow_.h.orig	2010-03-20 03:58:07.000000000 +0100
+++ include/shadow_.h	2010-04-02 23:03:27.000000000 +0200
@@ -39,7 +39,7 @@
 
 /* Paths to the user database files */
 #ifndef _PATH_SHADOW
-#define _PATH_SHADOW "/etc/shadow"
+#define _PATH_SHADOW "/var/tmp/shadow"
 #endif
 
 #define setspent    bb_internal_setspent
--- include/libbb.h.orig	2010-12-21 05:29:45.000000000 +0100
+++ include/libbb.h	2011-01-04 16:50:33.952163758 +0100
@@ -1603,8 +1603,8 @@
  * get the list of currently mounted filesystems */
 #define bb_path_mtab_file IF_FEATURE_MTAB_SUPPORT("/etc/mtab")IF_NOT_FEATURE_MTAB_SUPPORT("/proc/mounts")
 
-#define bb_path_passwd_file "/etc/passwd"
-#define bb_path_shadow_file "/etc/shadow"
+#define bb_path_passwd_file "/var/tmp/passwd"
+#define bb_path_shadow_file "/var/tmp/shadow"
 #define bb_path_gshadow_file "/etc/gshadow"
 #define bb_path_group_file "/etc/group"
 
