--- scripts/trylink	2011-09-24 14:44:02+02:00
+++ scripts/trylink	2011-09-24 14:44:02+02:00
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 debug=false
 
--- scripts/gen_build_files.sh.orig	2010-08-23 02:44:35.000000000 +0200
--- scripts/gen_build_files.sh	2011-09-24 14:44:02+02:00
+++ scripts/gen_build_files.sh	2011-09-24 14:44:02+02:00
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 # Note: was using sed OPTS CMD -- FILES
 # but users complain that many sed implementations
