--- scripts/trylink.orig	2010-08-22 10:21:38.000000000 +0200
+++ scripts/trylink	2010-10-08 21:11:24.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 debug=false
 
--- scripts/gen_build_files.sh.orig	2010-08-23 02:44:35.000000000 +0200
+++ scripts/gen_build_files.sh	2010-10-08 21:11:24.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 test $# -ge 2 || { echo "Syntax: $0 SRCTREE OBJTREE"; exit 1; }
 
