--- arch/mipsel/Makefile	2007-01-02 15:20:54.000000000 +0100
+++ arch/mipsel/Makefile	2007-01-02 15:20:54.000000000 +0100
@@ -0,0 +1,8 @@
+# ==========================================================================
+# Build system
+# ==========================================================================
+
+#Override rule from Makefile to link libgcc static
+cmd_busybox__ ?= $(srctree)/scripts/trylink $(CC) -o $@ $(LDFLAGS) \
+-Wl,--warn-common -Wl,--sort-common -Wl,--gc-sections \
+-Wl,--start-group $(busybox-all) -Wl,--end-group -static-libgcc
