--- util-linux/mount.c.orig	2007-09-13 00:18:54.000000000 +0200
+++ util-linux/mount.c	2007-09-13 00:18:54.000000000 +0200
@@ -1561,7 +1561,7 @@
 
 	// Parse remaining options
 
-	opt = getopt32(argc, argv, "o:t:rwanfvs", &opt_o, &fstype);
+	opt = getopt32(argc, argv, "o:t:rwanfvsi", &opt_o, &fstype);
 	if (opt & 0x1) append_mount_options(&cmdopts, opt_o); // -o
 	//if (opt & 0x2) // -t
 	if (opt & 0x4) append_mount_options(&cmdopts, "ro"); // -r
@@ -1571,6 +1571,7 @@
 	if (opt & 0x40) USE_FEATURE_MTAB_SUPPORT(fakeIt = 1); // -f
 	//if (opt & 0x80) // -v: verbose (ignore)
 	//if (opt & 0x100) // -s: sloppy (ignore)
+	//if (opt & 0x200) // -i: don't call mount.<fstype> (ignore)
 	argv += optind;
 	argc -= optind;
 
