diff -burN console-tools.orig/setconsole.c console-tools/setconsole.c
--- console-tools.orig/setconsole.c	2007-03-18 17:59:28.000000000 +0100
+++ console-tools/setconsole.c	2007-05-10 17:28:52.000000000 +0200
@@ -40,7 +40,7 @@
 			device = CONSOLE_DEV;
 	}
 
-	if (-1 == ioctl(xopen(device, O_RDONLY), TIOCCONS)) {
+	if (-1 == ioctl(xopen(device, O_RDWR), TIOCCONS)) {
 		bb_perror_msg_and_die("TIOCCONS");
 	}
 	return EXIT_SUCCESS;
