Fix for telnetd, may be integrated into busybox
--- networking/telnetd.c~	2007-11-04 07:03:15.000000000 +0100
+++ networking/telnetd.c	2007-11-05 12:51:48.000000000 +0100
@@ -384,13 +384,17 @@
 	pid_t pid;
 	struct tsession *ts;
 
-	pid = waitpid(-1, &sig, WNOHANG);
-	if (pid > 0) {
+	for (;;) {
+		pid = waitpid(-1, NULL, WNOHANG);
+		if (pid <= 0)
+			break;
+		if (!(option_mask32 & OPT_WATCHCHILD))
+			continue;
 		ts = sessions;
 		while (ts) {
 			if (ts->shell_pid == pid) {
 				ts->shell_pid = -1;
-				return;
+				break;
 			}
 			ts = ts->next;
 		}
@@ -464,8 +468,7 @@
 	/* We don't want to die if just one session is broken */
 	signal(SIGPIPE, SIG_IGN);
 
-	if (opt & OPT_WATCHCHILD)
-		signal(SIGCHLD, handle_sigchld);
+	signal(SIGCHLD, handle_sigchld);
 
 /*
    This is how the buffers are used. The arrows indicate the movement
