--- shell/ash.c.orig	2009-12-12 21:16:38.000000000 +0000
+++ shell/ash.c	2009-12-13 21:57:27.000000000 +0000
@@ -12351,8 +12351,17 @@
 	const char *p;
 	char **aptr;
 	int flag = argv[0][0] == 'r' ? VREADONLY : VEXPORT;
+	int mask = ~0;
+	int nopt;
+	while ((nopt = nextopt("np"))) {
+		if (nopt == 'n') {
+				mask = ~flag;
+		} else { /* p */
+			break;
+		}
+	}
 
-	if (nextopt("p") != 'p') {
+	if (nopt != 'p') {
 		aptr = argptr;
 		name = *aptr;
 		if (name) {
@@ -12364,10 +12373,12 @@
 					vp = *findvar(hashvar(name), name);
 					if (vp) {
 						vp->flags |= flag;
+						vp->flags &= mask;
 						continue;
 					}
 				}
 				setvar(name, p, flag);
+				setvar(name, p, flag & mask);
 			} while ((name = *++aptr) != NULL);
 			return 0;
 		}
