--- util-linux/mdev.c.orig	2008-09-28 20:04:30.000000000 +0200
+++ util-linux/mdev.c	2008-11-23 14:59:15.000000000 +0100
@@ -19,9 +19,6 @@
 #define root_major (G.root_major)
 #define root_minor (G.root_minor)
 
-/* Prevent infinite loops in /sys symlinks */
-#define MAX_SYSFS_DEPTH 3
-
 /* We use additional 64+ bytes in make_device() */
 #define SCRATCH_SIZE 80
 
@@ -320,15 +317,6 @@
 	return TRUE;
 }
 
-/* Directory callback for /sys/ traversal */
-static int FAST_FUNC dirAction(const char *fileName UNUSED_PARAM,
-		struct stat *statbuf UNUSED_PARAM,
-		void *userData UNUSED_PARAM,
-		int depth)
-{
-	return (depth >= MAX_SYSFS_DEPTH ? SKIP : TRUE);
-}
-
 /* For the full gory details, see linux/Documentation/firmware_class/README
  *
  * Firmware loading works like this:
@@ -424,12 +412,8 @@
 		 * not real directories.
 		 * (kernel's CONFIG_SYSFS_DEPRECATED makes them real dirs,
 		 * but we can't enforce that on users) */
-		recursive_action("/sys/block",
-			ACTION_RECURSE | ACTION_FOLLOWLINKS,
-			fileAction, dirAction, temp, 0);
-		recursive_action("/sys/class",
-			ACTION_RECURSE | ACTION_FOLLOWLINKS,
-			fileAction, dirAction, temp, 0);
+		recursive_action("/sys", ACTION_RECURSE,
+			fileAction, NULL, temp, 0);
 	} else {
 		char *seq;
 		char *action;
