--- editors/awk.c	2007/04/12 00:32:05	18409
+++ editors/awk.c	2007/05/03 22:57:56	18552
@@ -2696,7 +2696,8 @@
 	newfile("/dev/stdout")->F = stdout;
 	newfile("/dev/stderr")->F = stderr;
 
-	for (envp = environ; *envp; envp++) {
+	/* Huh, people report that sometimes environ is NULL. Oh well. */
+	if (environ) for (envp = environ; *envp; envp++) {
 		char *s = xstrdup(*envp);
 		char *s1 = strchr(s, '=');
 		if (s1) {
