--- editors/awk.c.orig	2007-05-20 18:57:56.000000000 +0200
+++ editors/awk.c	2007-07-28 12:42:26.000000000 +0200
@@ -2693,7 +2693,8 @@
 	newfile("/dev/stdout")->F = stdout;
 	newfile("/dev/stderr")->F = stderr;
 
-	for (envp = environ; *envp; envp++) {
+	/* Huh, people report that sometimes environ is NULL. Oh well. */
+	if (environ) for (envp = environ; *envp; envp++) {
 		char *s = xstrdup(*envp);
 		char *s1 = strchr(s, '=');
 		if (s1) {
