--- util-linux/mount.c
+++ util-linux/mount.c
@@ -145,6 +145,7 @@
 //usage:     "\n-o OPT:"
 //usage:	IF_FEATURE_MOUNT_LOOP(
 //usage:     "\n	loop		Ignored (loop devices are autodetected)"
+//usage:     "\n	offset=OFS	The data start is moved OFS bytes into the specified file"
 //usage:	)
 //usage:	IF_FEATURE_MOUNT_FLAGS(
 //usage:     "\n	[a]sync		Writes are [a]synchronous"
@@ -1875,6 +1876,7 @@
 	int rc = -1;
 	unsigned long vfsflags;
 	char *loopFile = NULL, *filteropts = NULL;
+	const char *offsetString = "offset=\0";
 	llist_t *fl = NULL;
 	struct stat st;
 
@@ -1997,9 +1999,27 @@
 	) {
 		// Do we need to allocate a loopback device for it?
 		if (ENABLE_FEATURE_MOUNT_LOOP && S_ISREG(st.st_mode)) {
+			long long loopOffset = 0; // no offset specified
+			char *offsetOption;
 			loopFile = bb_simplify_path(mp->mnt_fsname);
 			mp->mnt_fsname = NULL; // will receive malloced loop dev name
-			if (set_loop(&mp->mnt_fsname, loopFile, 0, /*ro:*/ (vfsflags & MS_RDONLY)) < 0) {
+			if (filteropts && (offsetOption = strstr(filteropts, offsetString))) {
+				if (offsetOption == filteropts || *(offsetOption - 1) == ',') {
+					char *nextOption;
+					if ((nextOption = strchr(offsetOption, ','))) {
+						*(nextOption++) = '\0';
+						loopOffset = xatoull(offsetOption + strlen(offsetString));
+						strcpy(offsetOption, nextOption);
+					}
+					else {
+						loopOffset = xatoull(offsetOption + strlen(offsetString));
+						nextOption = offsetOption - 1;
+						if (offsetOption == filteropts) nextOption = filteropts;
+						*nextOption  = '\0';
+					}
+				}
+			}
+			if (set_loop(&mp->mnt_fsname, loopFile, loopOffset, /*ro:*/ (vfsflags & MS_RDONLY)) < 0) {
 				if (errno == EPERM || errno == EACCES)
 					bb_error_msg(bb_msg_perm_denied_are_you_root);
 				else
