--- networking/httpd.c.orig	2011-01-04 16:56:18.782164713 +0100
+++ networking/httpd.c	2011-01-04 16:56:19.032164713 +0100
@@ -1358,6 +1358,9 @@
 	if (script != NULL)
 		*script = '\0';         /* cut off /PATH_INFO */
 
+	/* REDIRECT_STATUS is required by PHP compiled with --enable-force-cgi-redirect */
+	putenv((char*)"REDIRECT_STATUS=200");
+
 	/* SCRIPT_FILENAME is required by PHP in CGI mode */
 	if (home_httpd[0] == '/') {
 		char *fullpath = concat_path_file(home_httpd, url);
