--- libbb/appletlib.c	2011-10-05 21:12:07+02:00
+++ libbb/appletlib.c	2011-10-05 21:12:07+02:00
@@ -147,12 +147,14 @@
 	return strcmp(name, APPLET_NAME(i));
 }
 #endif
+static const char httpd_applet_name[] ALIGN1 = "httpd";
 int FAST_FUNC find_applet_by_name(const char *name)
 {
+	const char *name_to_search_for = (strncmp(name, httpd_applet_name, 5) != 0) ? name : httpd_applet_name;
 #if NUM_APPLETS > 8
 	/* Do a binary search to find the applet entry given the name. */
 	const char *p;
-	p = bsearch(name, applet_names, ARRAY_SIZE(applet_main), 1, applet_name_compare);
+	p = bsearch(name_to_search_for, applet_names, ARRAY_SIZE(applet_main), 1, applet_name_compare);
 	if (!p)
 		return -1;
 	return p - applet_names;
@@ -161,7 +163,7 @@
 	int i = 0;
 	const char *p = applet_names;
 	while (i < NUM_APPLETS) {
-		if (strcmp(name, p) == 0)
+		if (strcmp(name_to_search_for, p) == 0)
 			return i;
 		p += strlen(p) + 1;
 		i++;
