--- shell/ash.c.ori	2007-01-24 22:34:40.000000000 +0100
+++ shell/ash.c	2007-04-29 15:40:44.000000000 +0200
@@ -6560,7 +6560,9 @@
 		/* turning job control off */
 		fd = ttyfd;
 		pgrp = initialpgrp;
-		xtcsetpgrp(fd, pgrp);
+		/* was xtcsetpgrp, but this can make exiting ash
+		 * with pty already deleted loop forever */
+		tcsetpgrp(fd, pgrp);
 		setpgid(0, pgrp);
 		setsignal(SIGTSTP);
 		setsignal(SIGTTOU);
