Index: util-linux/getopt.c
===================================================================
--- util-linux/getopt.c	(revision 23733)
+++ util-linux/getopt.c	(working copy)
@@ -157,6 +157,14 @@
 	if (quiet_errors) /* No error reporting from getopt(3) */
 		opterr = 0;
 
+	/* We used it already in getopt32(), we *must* reset getopt(3) */
+#ifdef __GLIBC__
+	optind = 0;
+#else /* BSD style */
+	optind = 1;
+	/* optreset = 1; */
+#endif
+
 	while (1) {
 		opt =
 #if ENABLE_GETOPT_LONG
