--- util-linux/getopt.c.orig	2009-12-12 01:45:09.000000000 +0000
+++ util-linux/getopt.c	2009-12-13 21:57:28.000000000 +0000
@@ -166,6 +166,14 @@
 	/* optreset = 1; */
 #endif
 
+	/* We used it already in getopt32(), we *must* reset getopt(3) */
+#ifdef __GLIBC__
+	optind = 0;
+#else /* BSD style */
+	optind = 1;
+	/* optreset = 1; */
+#endif
+
 	while (1) {
 		opt =
 #if ENABLE_FEATURE_GETOPT_LONG
