--- lib/gettext.h      2008-05-16 18:28:43.000000000 +0200
+++ lib/gettext.h      2009-01-03 18:59:05.374184151 +0100
@@ -22,7 +22,6 @@
 #if ENABLE_NLS

 /* Get declarations of GNU message catalog functions.  */
-# include <libintl.h>

 /* You can set the DEFAULT_TEXT_DOMAIN macro to specify the domain used by
    the gettext() and ngettext() macros.  This is an alternative to calling
@@ -54,7 +53,6 @@
 #if defined(__cplusplus) && defined(__GNUG__) && (__GNUC__ >= 3)
 # include <cstdlib>
 # if (__GLIBC__ >= 2) || _GLIBCXX_HAVE_LIBINTL_H
-#  include <libintl.h>
 # endif
 #endif

