--- perl.c	2009-04-22 20:07:41.000000000 +0200
+++ perl.c	2009-09-26 16:18:08.000000000 +0200
@@ -396,11 +396,11 @@
 	("__environ", (unsigned long *) &environ_pointer, NULL);
 #endif /* environ */
 
-#ifndef PERL_MICRO
+//#ifndef PERL_MICRO
 #   ifdef  USE_ENVIRON_ARRAY
     PL_origenviron = environ;
 #   endif
-#endif
+//#endif
 
     /* Use sysconf(_SC_CLK_TCK) if available, if not
      * available or if the sysconf() fails, use the HZ.
@@ -857,7 +857,7 @@
     /* if PERL_USE_SAFE_PUTENV is defined environ will not have been copied
      * so we certainly shouldn't free it here
      */
-#ifndef PERL_MICRO
+//#ifndef PERL_MICRO
 #if defined(USE_ENVIRON_ARRAY) && !defined(PERL_USE_SAFE_PUTENV)
     if (environ != PL_origenviron && !PL_use_safe_putenv
 #ifdef USE_ITHREADS
@@ -877,7 +877,7 @@
 	environ = PL_origenviron;
     }
 #endif
-#endif /* !PERL_MICRO */
+//#endif /* !PERL_MICRO */
 
     if (destruct_level == 0) {
 
@@ -4780,7 +4780,7 @@
 	GvMULTI_on(PL_envgv);
 	hv = GvHVn(PL_envgv);
 	hv_magic(hv, NULL, PERL_MAGIC_env);
-#ifndef PERL_MICRO
+//#ifndef PERL_MICRO
 #ifdef USE_ENVIRON_ARRAY
 	/* Note that if the supplied env parameter is actually a copy
 	   of the global environ then it may now point to free'd memory
@@ -4819,7 +4819,7 @@
 	  }
       }
 #endif /* USE_ENVIRON_ARRAY */
-#endif /* !PERL_MICRO */
+//#endif /* !PERL_MICRO */
     }
     TAINT_NOT;
     if ((tmpgv = gv_fetchpvs("$", GV_ADD|GV_NOTQUAL, SVt_PV))) {
--- perl.h	2009-07-06 13:18:58.000000000 +0200
+++ perl.h	2009-09-26 16:18:08.000000000 +0200
@@ -4190,11 +4190,12 @@
 #else
    /* VMS and some other platforms don't use the environ array */
 #  ifdef USE_ENVIRON_ARRAY
+extern char **  environ;        /* environment variables supplied via exec */
 #    if !defined(DONT_DECLARE_STD) || \
         (defined(__svr4__) && defined(__GNUC__) && defined(sun)) || \
         defined(__sgi) || \
         defined(__DGUX)
-extern char **	environ;	/* environment variables supplied via exec */
+//extern char **	environ;	/* environment variables supplied via exec */
 #    endif
 #  endif
 #endif
--- util.c	2009-08-15 18:36:34.000000000 +0200
+++ util.c	2009-09-26 16:18:28.000000000 +0200
@@ -5801,7 +5801,7 @@
 Perl_my_clearenv(pTHX)
 {
     dVAR;
-#if ! defined(PERL_MICRO)
+//#if ! defined(PERL_MICRO)
 #  if defined(PERL_IMPLICIT_SYS) || defined(WIN32)
     PerlEnv_clearenv();
 #  else /* ! (PERL_IMPLICIT_SYS || WIN32) */
@@ -5849,7 +5849,7 @@
     }
 #    endif /* USE_ENVIRON_ARRAY */
 #  endif /* PERL_IMPLICIT_SYS || WIN32 */
-#endif /* PERL_MICRO */
+//#endif /* PERL_MICRO */
 }
 
 #ifdef PERL_IMPLICIT_CONTEXT
