diff -burN binutils-2.17.50.0.16.orig/libiberty/Makefile.in binutils-2.17.50.0.16/libiberty/Makefile.in
--- libiberty/Makefile.in	2007-05-11 17:23:21.000000000 +0200
+++ libiberty/Makefile.in	2007-07-16 23:02:57.000000000 +0200
@@ -651,7 +651,7 @@
 	else true; fi
 	$(COMPILE.c) $(srcdir)/fibheap.c $(OUTPUT_OPTION)
 
-./filename_cmp.o: $(srcdir)/filename_cmp.c $(INCDIR)/filenames.h \
+./filename_cmp.o: $(srcdir)/filename_cmp.c config.h $(INCDIR)/filenames.h \
 	$(INCDIR)/safe-ctype.h
 	if [ x"$(PICFLAG)" != x ]; then \
 	  $(COMPILE.c) $(PICFLAG) $(srcdir)/filename_cmp.c -o pic/$@; \
