Revert upstream commit because of #1163

From f2657a9968d2f5282f9a7f2e8b93a8d5a57e04fd Mon Sep 17 00:00:00 2001
From: Denys Vlasenko <vda.linux@googlemail.com>
Date: Sun, 24 Oct 2010 00:04:10 +0000
Subject: init: do not clear CRTSCTS (fix from Debian bug 528560)

Signed-off-by: Denys Vlasenko <vda.linux@googlemail.com>
---
--- init/init.c
+++ init/init.c
@@ -361,11 +361,7 @@
 #endif
 
 	/* Make it be sane */
-#ifndef CRTSCTS
-# define CRTSCTS 0
-#endif
-	/* added CRTSCTS to fix Debian bug 528560 */
-	tty.c_cflag &= CBAUD | CBAUDEX | CSIZE | CSTOPB | PARENB | PARODD | CRTSCTS;
+	tty.c_cflag &= CBAUD | CBAUDEX | CSIZE | CSTOPB | PARENB | PARODD;
 	tty.c_cflag |= CREAD | HUPCL | CLOCAL;
 
 	/* input modes */
