--- a/include/time.h.orig	2019-10-15 13:03:34.824587529 +0200
+++ b/include/time.h	2019-10-15 13:04:05.204707796 +0200
@@ -108,7 +108,7 @@
 
 #if !defined __timespec_defined &&				\
     ((defined _TIME_H &&					\
-      (defined __USE_POSIX199309 || defined __USE_MISC)) ||	\
+      (defined __USE_POSIX199309 || defined __USE_MISC || defined __USE_ISOC11)) ||	\
       defined __need_timespec)
 # define __timespec_defined	1
 
