From 6de81f1250fd323c9155de065d5a9dc200a6f20b Mon Sep 17 00:00:00 2001
From: Simon Kelley <simon@thekelleys.org.uk>
Date: Wed, 9 Sep 2015 22:51:13 +0100
Subject: [PATCH] Handle signed dangling CNAME replies to DS queries.

---
 src/dnssec.c |    7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git src/dnssec.c src/dnssec.c
index 4deda24..67ce486 100644
--- src/dnssec.c
+++ src/dnssec.c
@@ -1232,11 +1232,8 @@ int dnssec_validate_ds(time_t now, struct dns_header *header, size_t plen, char
    
   /* If we return STAT_NO_SIG, name contains the name of the DS query */
   if (val == STAT_NO_SIG)
-    {
-      *keyname = 0;
-      return val;
-    }  
-
+    return val;
+  
   /* If the key needed to validate the DS is on the same domain as the DS, we'll
      loop getting nowhere. Stop that now. This can happen of the DS answer comes
      from the DS's zone, and not the parent zone. */
