--- src/config.h	2012-04-29 18:56:52+02:00
+++ src/config.h	2012-04-29 18:56:52+02:00
@@ -33,7 +33,9 @@
 #define SMALLDNAME 40 /* most domain names are smaller than this */
 #define HOSTSFILE "/etc/hosts"
 #define ETHERSFILE "/etc/ethers"
+#ifndef RUNFILE
 #define RUNFILE "/var/run/dnsmasq.pid"
+#endif
 #define DEFLEASE 3600 /* default lease time, 1 hour */
 #define CHUSER "nobody"
 #define CHGRP "dip"
