nfnetlink/nlattr required for ipset feature are available since 2.6.15

--- src/config.h
+++ src/config.h
@@ -135,7 +135,12 @@
 #define HAVE_TFTP
 #define HAVE_SCRIPT
 #define HAVE_AUTH
+#if defined(__linux__)
+#include <linux/version.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,24)
 #define HAVE_IPSET 
+#endif
+#endif
 /* #define HAVE_LUASCRIPT */
 /* #define HAVE_BROKEN_RTC */
 /* #define HAVE_DBUS */
