--- src/config.h.orig	2010-01-21 15:22:06.000000000 +0100
+++ src/config.h	2010-04-02 17:46:53.000000000 +0200
@@ -39,7 +39,7 @@
 #else
 #  define RESOLVFILE "/etc/resolv.conf"
 #endif
-#define RUNFILE "/var/run/dnsmasq.pid"
+#define RUNFILE "/var/run/dnsmasq/dnsmasq.pid"
 
 #ifndef LEASEFILE
 #   if defined(__FreeBSD__) || defined (__OpenBSD__) || defined(__DragonFly__) || defined(__NetBSD__)
@@ -47,7 +47,7 @@
 #   elif defined(__sun__) || defined (__sun)
 #      define LEASEFILE "/var/cache/dnsmasq.leases"
 #   else
-#      define LEASEFILE "/var/lib/misc/dnsmasq.leases"
+#      define LEASEFILE "/var/tmp/dnsmasq.leases"
 #   endif
 #endif
 
@@ -55,7 +55,7 @@
 #   if defined(__FreeBSD__)
 #      define CONFFILE "/usr/local/etc/dnsmasq.conf"
 #   else
-#      define CONFFILE "/etc/dnsmasq.conf"
+#      define CONFFILE "/mod/etc/dnsmasq.conf"
 #   endif
 #endif
 
@@ -211,9 +211,9 @@
 #endif
 #undef HAVE_ARC4RANDOM
 #undef HAVE_SOCKADDR_SA_LEN
-#if !defined(__ARCH_HAS_MMU__) && !defined(__UCLIBC_HAS_MMU__)
-#  define NO_FORK
-#endif
+//#if !defined(__ARCH_HAS_MMU__) && !defined(__UCLIBC_HAS_MMU__)
+//#  define NO_FORK
+//#endif
 #if defined(__UCLIBC_HAS_IPV6__)
 #  ifndef IPV6_V6ONLY
 #    define IPV6_V6ONLY 26
@@ -277,6 +277,17 @@
 #  else
 #    define IPV6_LEVEL IPPROTO_IPV6
 #  endif
+#  if defined(IPV6_RECVPKTINFO)
+#    include <linux/version.h>
+#    if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,14)
+#      if defined IPV6_RECVPKTINFO && defined IPV6_2292PKTINFO
+#        undef IPV6_RECVPKTINFO
+#        undef IPV6_PKTINFO
+#        define IPV6_RECVPKTINFO IPV6_2292PKTINFO
+#        define IPV6_PKTINFO IPV6_2292PKTINFO
+#      endif
+#    endif
+#  endif
 #elif defined(INET_ADDRSTRLEN)
 #  undef HAVE_IPV6
 #  define ADDRSTRLEN INET_ADDRSTRLEN
