--- src/config.h.orig	2006-03-14 15:12:50.000000000 +0100
+++ src/config.h	2006-03-14 15:17:58.000000000 +0100
@@ -35,12 +35,12 @@
 #if defined(__FreeBSD__) || defined (__OpenBSD__)
 #   define LEASEFILE "/var/db/dnsmasq.leases"
 #else
-#   define LEASEFILE "/var/lib/misc/dnsmasq.leases"
+#   define LEASEFILE "/var/tmp/dnsmasq.leases"
 #endif
 #if defined(__FreeBSD__)
 #   define CONFFILE "/usr/local/etc/dnsmasq.conf"
 #else
-#   define CONFFILE "/etc/dnsmasq.conf"
+#   define CONFFILE "/mod/etc/dnsmasq.conf"
 #endif
 #define DEFLEASE 3600 /* default lease time, 1 hour */
 #define CHUSER "nobody"
@@ -224,9 +224,9 @@
 #define HAVE_DEV_RANDOM
 #undef HAVE_SOCKADDR_SA_LEN
 #undef HAVE_PSELECT
-#if !defined(__ARCH_HAS_MMU__)
-#  define NO_FORK
-#endif
+//#if !defined(__ARCH_HAS_MMU__)
+//#  define NO_FORK
+//#endif
 #if !defined(__UCLIBC_HAS_IPV6__)
 #  define NO_IPV6
 #endif
