--- letsencrypt.sh
+++ letsencrypt.sh
@@ -50,7 +50,7 @@
 load_config() {
   # Check for config in various locations
   if [[ -z "${CONFIG:-}" ]]; then
-    for check_config in "/etc/letsencrypt.sh" "/usr/local/etc/letsencrypt.sh" "${PWD}" "${SCRIPTDIR}"; do
+    for check_config in "/mod/etc/letsencrypt.sh" "/etc/letsencrypt.sh" "/usr/local/etc/letsencrypt.sh" "${PWD}" "${SCRIPTDIR}"; do
       if [[ -e "${check_config}/config.sh" ]]; then
         BASEDIR="${check_config}"
         CONFIG="${check_config}/config.sh"
