--- linux-2.6.19/drivers/net/ppp_generic.c
+++ linux-2.6.19/drivers/net/ppp_generic.c
@@ -47,11 +47,6 @@
 #include <linux/mutex.h>
 #include <net/slhc_vj.h>
 #include <asm/atomic.h>
-#if defined(CONFIG_MACH_ADI_FUSIV) || defined(CONFIG_MACH_IKAN_MIPS)
-#ifdef CONFIG_ATM
-#include <atm/atmencap.h>
-#endif
-#endif
 
 #define PPP_VERSION	"2.4.2"
 
@@ -1573,11 +1568,6 @@
 {
 	struct sk_buff *ns;
 	int proto, len, npi;
-#if defined(CONFIG_MACH_ADI_FUSIV) || defined(CONFIG_MACH_IKAN_MIPS)
-#ifdef CONFIG_ATM
-        unsigned char encap;
-#endif
-#endif
 
 	/*
 	 * Decompress the frame, if compressed.
@@ -1650,14 +1640,6 @@
 
 	++ppp->stats.rx_packets;
 	ppp->stats.rx_bytes += skb->len - 2;
-#if defined(CONFIG_MACH_ADI_FUSIV) || defined(CONFIG_MACH_IKAN_MIPS)
-#ifdef CONFIG_ATM
-        encap = skb->encap;
-
-        if ((encap == ENCAPIDROUTEDLLCPPPOA) || (encap == ENCAPIDROUTEDVCPPPOA))
-            proto = PPP_IP;
-#endif
-#endif
 
 	npi = proto_to_npindex(proto);
 	if (npi < 0) {
@@ -1699,15 +1681,8 @@
 		    || ppp->npmode[npi] != NPMODE_PASS) {
 			kfree_skb(skb);
 		} else {
-#if (defined(CONFIG_MACH_ADI_FUSIV) || defined(CONFIG_MACH_IKAN_MIPS)) && defined(CONFIG_ATM)
-             if( (encap != ENCAPIDROUTEDLLCPPPOA) &&
-                 (encap != ENCAPIDROUTEDVCPPPOA) )
-                        skb_pull_rcsum(skb, 2);       /* chop off protocol */
-#else
-                        skb_pull_rcsum(skb, 2);       /* chop off protocol */
-#endif
 			/* chop off protocol */
-			// Fusiv skb_pull_rcsum(skb, 2);
+			skb_pull_rcsum(skb, 2);
 			skb->dev = ppp->dev;
 			skb->protocol = htons(npindex_to_ethertype[npi]);
 			skb->mac.raw = skb->data;
@@ -2817,140 +2792,11 @@
 	*pmap = NULL;
 }
 
-#if CONFIG_ATM
-#if defined(CONFIG_PPPOATM) || defined(CONFIG_PPPOATM_MODULE)
-extern int isPPPModTypePPPoA(struct ppp_channel *);
-#endif
-#endif
-
-#if defined(CONFIG_PPPOE) || defined(CONFIG_PPPOE_MODULE)
-extern int isPPPModTypePPPoE(struct ppp_channel *);
-extern char *getPPPoEIfName(struct ppp_channel *);
-extern int getPPPoESessIdAndAddr(struct ppp_channel *, unsigned char *);
-#endif
-
-char *getPhysicalIfName(char *pIfName)
-{
-    struct net_device *pDev;
-    struct list_head  *list;
-    struct ppp        *ppp;
-
-    if (!pIfName)
-        return NULL;
-
-    pDev = dev_get_by_name(pIfName);
-
-    if (!pDev)
-        return NULL;
-
-    ppp = (struct ppp *) pDev->priv;
-
-    if (!ppp)
-        return NULL;
-
-    list = &ppp->channels;
-
-    if (!list_empty(list))
-    {
-        struct ppp_channel *pppchan;
-        struct channel     *pch;
-
-        list = list->next;
-
-        pch = list_entry(list, struct channel, clist);
-
-        if (!pch)
-            return NULL;
-
-        pppchan = pch->chan;
-
-        if (!pppchan)
-            return NULL;
-
-        if ((pppchan->ops) && (pppchan->ops->start_xmit))
-        {
-#if CONFIG_ATM
-#if defined(CONFIG_PPPOATM) || defined(CONFIG_PPPOATM_MODULE)
-            if (isPPPModTypePPPoA(pppchan))
-                return "atm";
-#endif
-#endif
-
-#if defined(CONFIG_PPPOE) || defined(CONFIG_PPPOE_MODULE)
-            if (isPPPModTypePPPoE(pppchan))
-                return (getPPPoEIfName(pppchan));
-#endif
-        }
-    }
-
-    printk("getPhysicalIfName: FAILED..\r\n");
-
-    dev_put(pDev);
-
-    return NULL;
-}
-
-int getSessIdAndAddr(char *pIfName, unsigned char *addr)
-{
-    struct net_device *pDev;
-    struct list_head  *list;
-    struct ppp        *ppp;
-
-    if (!pIfName)
-        return 0;
-
-    pDev = dev_get_by_name(pIfName);
-
-    if (!pDev)
-        return 0;
-
-    ppp = (struct ppp *) pDev->priv;
-
-    if (!ppp)
-        return 0;
-
-    list = &ppp->channels;
-
-    if (!list_empty(list))
-    {
-        struct ppp_channel *pppchan;
-        struct channel     *pch;
-
-        list = list->next;
-
-        pch = list_entry(list, struct channel, clist);
-
-        if (!pch)
-            return 0;
-
-        pppchan = pch->chan;
-
-        if (!pppchan)
-            return 0;
-
-        if ((pppchan->ops) && (pppchan->ops->start_xmit))
-        {
-#if defined(CONFIG_PPPOE) || defined(CONFIG_PPPOE_MODULE)
-            if (isPPPModTypePPPoE(pppchan))
-                return (getPPPoESessIdAndAddr(pppchan, addr));
-#endif
-        }
-    }
-
-    printk("getSessIdAndAddr: FAILED..\r\n");
-
-    dev_put(pDev);
-
-    return 0;
-}
-
 /* Module/initialization stuff */
 
 module_init(ppp_init);
 module_exit(ppp_cleanup);
 
-EXPORT_SYMBOL(getPhysicalIfName);
-EXPORT_SYMBOL(getSessIdAndAddr);
 EXPORT_SYMBOL(ppp_register_channel);
 EXPORT_SYMBOL(ppp_unregister_channel);
 EXPORT_SYMBOL(ppp_channel_index);
