--- configure
+++ configure
@@ -1993,14 +1993,9 @@
 if enabled pthreads; then
     if check_func pthread_create; then
         :
-    elif check_func pthread_create -pthread; then
-        add_cflags -pthread
-        add_extralibs -pthread
-    elif check_func pthread_create -pthreads; then
-        add_cflags -pthreads
-        add_extralibs -pthreads
-    elif check_func pthread_create -lpthreadGC2; then
-        add_extralibs -lpthreadGC2
+    elif check_func pthread_create -D_REENTRANT -lpthread; then
+        add_cflags -D_REENTRANT
+        add_extralibs -lpthread
     elif ! check_lib pthread.h pthread_create -lpthread; then
         die "ERROR: can't find pthreads library"
     fi
