--- configure
+++ configure
@@ -2012,12 +2012,13 @@
 check_lib math.h sin -lm
 
 # test for C99 functions in math.h
-for func in llrint lrint lrintf round roundf truncf; do
+for func in llrint lrint lrintf round roundf; do
     check_exec <<EOF && enable $func || disable $func
 #include <math.h>
 int main(void) { return ($func(3.999f) > 0)?0:1; }
 EOF
 done
+disable truncf
 
 # these are off by default, so fail if requested and not available
 enabled avisynth   && require2 vfw32 "windows.h vfw.h" AVIFileInit -lvfw32
--- libavcodec/aac.c
+++ libavcodec/aac.c
@@ -946,7 +946,7 @@
 static av_always_inline float flt16_trunc(float pf) {
     int exp;
     pf = frexpf(pf, &exp);
-    pf = ldexpf(truncf(ldexpf(pf, 8)), exp-8);
+    pf = ldexpf(ffmpeg_truncf(ldexpf(pf, 8)), exp-8);
     return pf;
 }
 
--- libavutil/internal.h
+++ libavutil/internal.h
@@ -308,7 +308,7 @@
 #endif /* HAVE_ROUNDF */
 
 #if !HAVE_TRUNCF
-static av_always_inline av_const float truncf(float x)
+static av_always_inline av_const float ffmpeg_truncf(float x)
 {
     return (x > 0) ? floor(x) : ceil(x);
 }
