--- usr/www/avme/en/html/en/fon/routing.js.orig	2008-04-17 17:48:43.000000000 +0200
+++ usr/www/avme/en/html/en/fon/routing.js	2009-09-30 11:53:53.000000000 +0200
@@ -14,6 +14,7 @@
 <script type="text/javascript">
 <? setvariable var:showLCRdata 0 ?>
 <? if eq $var:OEM avm `<? setvariable var:showLCRdata 1 ?>` ?>
+<? if eq $var:OEM tcom `<? setvariable var:showLCRdata 1 ?>` ?>
 <? if eq $var:OEM avme `<? setvariable var:showLCRdata 1 ?>` ?>
 <? if eq $var:OEM freenet `<? setvariable var:showLCRdata 1 ?>` ?>
 <? include ../html/$var:lang/js/val.js ?>
--- usr/www/avme/en/html/en/fon/sip1.html.orig	2008-04-17 17:48:40.000000000 +0200
+++ usr/www/avme/en/html/en/fon/sip1.html	2009-09-30 11:53:53.000000000 +0200
@@ -62,6 +62,20 @@
 <option value="TOnline">$var:Text12</option>
 <option value="Webde">$var:Text11</option>
 `?>
+<? if eq $var:OEM tcom `
+<option value="1u1">$var:Text4</option>
+<option value="AOL">$var:Text9</option>
+<option value="Arcor">$var:Text7</option>
+<option value="Dus">$var:Text13</option>
+<option value="Freenet">$var:Text8</option>
+<option value="GMX">$var:Text5</option>
+<option value="Inode">$var:Text55</option>
+<option value="SuP">$var:Text6</option>
+<option value="Sipgate">$var:Text10</option>
+<option value="Strato">$var:Text44</option>
+<option value="TOnline">$var:Text12</option>
+<option value="Webde">$var:Text11</option>
+`?>
 <option value="Sonst">$var:Text14</option>
 </select></td>
 </tr>
--- usr/www/avme/en/html/en/fon/sip1.js.orig	2009-09-30 11:53:53.000000000 +0200
+++ usr/www/avme/en/html/en/fon/sip1.js	2009-09-30 11:53:53.000000000 +0200
@@ -1,5 +1,6 @@
 <? setvariable var:showtcom 0 ?>
 <? if eq $var:OEM avm '<? setvariable var:showtcom 1 ?>' ?>
+<? if eq $var:OEM tcom '<? setvariable var:showtcom 1 ?>' ?>
 <? if eq $var:OEM aol '<? setvariable var:showtcom 1 ?>' ?>
 <!-- ../fon/sip.js -->
 <style type="text/css">
@@ -553,6 +554,7 @@
 switch (t) {
 case "aol":
 case "avm":
+case "tcom":
 case "other":
 <? if eq $var:OEM avme `
 jslSetValue("uiPostAKN", "0"); break;
--- usr/www/avme/en/html/en/fon/sipoptionen.js.orig	2008-04-17 17:48:38.000000000 +0200
+++ usr/www/avme/en/html/en/fon/sipoptionen.js	2009-09-30 11:53:53.000000000 +0200
@@ -121,11 +121,9 @@
 kompress = jslGetValue("uiPostLowCodecs")=="1" ? 2 : 1;
 }
 OnKompress(kompress);
-if (g_Oem == "avm" || g_Oem == "avme") {
-if (g_Oem == "avme") {
+if (g_Oem == "avm" || g_Oem == "tcom" ||  g_Oem == "avme") {
 jslDisplay("uiEnum", true);
 jslSetChecked("uiViewEnum", jslGetValue("uiPostEnum") == "1");
-}
 jslDisplay("uiVcc", true);
 jslSetChecked("uiViewUseVcc",jslGetValue("uiPostUseSepVcc")=="1");
 var encaps = jslGetValue("uiPostEncaps");
@@ -208,11 +206,9 @@
 jslSetValue("uiPostLowCodecs", jslGetChecked("uiViewDynLowCodec") ? "1" : "0");
 jslEnable("uiPostDynCodec");
 jslEnable("uiPostLowCodecs");
-if (g_Oem == "avm" || g_Oem == "avme") {
-if (g_Oem == "avme") {
+if (g_Oem == "avm" || g_Oem == "tcom" || g_Oem == "avme") {
 jslSetValue("uiPostEnum", jslGetChecked("uiViewEnum") ? "1":"0");
 jslEnable("uiPostEnum");
-}
 jslSetValue("uiPostUseSepVcc", jslGetChecked("uiViewUseVcc") ? "1" : "0");
 jslEnable("uiPostUseSepVcc");
 if (jslGetChecked("uiViewUseVcc")) {
--- usr/www/avme/en/html/en/internet/authform.html.orig	2008-04-17 17:48:43.000000000 +0200
+++ usr/www/avme/en/html/en/internet/authform.html	2009-09-30 11:53:53.000000000 +0200
@@ -13,6 +13,7 @@
 <? if eq $var:OEM freenet `<? setvariable var:allprovider 1 ?>` ?>
 <? if eq $var:OEM aol `<? setvariable var:allprovider 1 ?>` ?>
 <? if eq $var:OEM avm `<? setvariable var:allprovider 1 ?>` ?>
+<? if eq $var:OEM tcom `<? setvariable var:allprovider 1 ?>` ?>
 <? if eq $var:allprovider 1 `
 <option value="1u1">$var:afTxt02</option>
 <option value="AOL">$var:afTxt06</option>
--- usr/www/avme/en/html/en/internet/pp_user.html.orig	2008-04-17 17:48:41.000000000 +0200
+++ usr/www/avme/en/html/en/internet/pp_user.html	2009-09-30 11:53:53.000000000 +0200
@@ -3,7 +3,7 @@
 <? include ../html/config.def ?>
 <? setvariable var:OEM <? query env:status/OEM ?> ?>
 <? setvariable var:BoxName 'FRITZ!Box' ?>
-<? if eq $var:OEM tcom '<? setvariable var:BoxName "Eumex" ?>' ?>
+<? if eq $var:OEM tcom '<? setvariable var:BoxName "FRITZ!Box_Eumex" ?>' ?>
 <? include ../html/$var:lang/internet/pp_user.inc ?>
 <head>
 <title><? echo $var:txt00 ?></title>
--- usr/www/avme/en/html/en/internet/pp_wanbridge.html.orig	2008-04-17 17:48:45.000000000 +0200
+++ usr/www/avme/en/html/en/internet/pp_wanbridge.html	2009-09-30 11:53:53.000000000 +0200
@@ -3,7 +3,7 @@
 <? include ../html/config.def ?>
 <? setvariable var:OEM <? query env:status/OEM ?> ?>
 <? setvariable var:BoxName 'FRITZ!Box' ?>
-<? if eq $var:OEM tcom '<? setvariable var:BoxName "Eumex" ?>' ?>
+<? if eq $var:OEM tcom '<? setvariable var:BoxName "FRITZ!Box_Eumex" ?>' ?>
 <? include ../html/$var:lang/internet/pp_wanbridge.inc ?>
 <head>
 <title><? echo $var:txt00 ?></title>
--- usr/www/avme/en/html/en/menus/menuwatch.html.orig	2008-04-17 17:48:40.000000000 +0200
+++ usr/www/avme/en/html/en/menus/menuwatch.html	2009-09-30 11:53:53.000000000 +0200
@@ -5,7 +5,7 @@
 <? include ../html/$var:lang/menus/menu2.def ?>
 <? setvariable var:OEM <? query env:status/OEM ?> ?>
 <? setvariable var:BoxName 'FRITZ!Box' ?>
-<? if eq $var:OEM tcom '<? setvariable var:BoxName "Eumex" ?>' ?>
+<? if eq $var:OEM tcom '<? setvariable var:BoxName "FRIZT!Box_Eumex" ?>' ?>
 <head>
 <title><? echo $var:BoxName ?></title>
 <meta http-equiv=content-type content="text/html; charset=iso-8859-1">
--- usr/www/avme/en/html/en/system/extended.js.orig	2008-04-17 17:48:38.000000000 +0200
+++ usr/www/avme/en/html/en/system/extended.js	2009-09-30 11:53:53.000000000 +0200
@@ -13,6 +13,7 @@
 jslSetChecked("uiViewExpertMode", jslGetValue("uiPostExpertMode") == "1");
 <? setvariable var:ShowUsePstn 0 ?>
 <? if eq $var:OEM avm `<? setvariable var:ShowUsePstn 1 ?>` ?>
+<? if eq $var:OEM tcom `<? setvariable var:ShowUsePstn 1 ?>` ?>
 <? if eq $var:OEM freenet `<? setvariable var:ShowUsePstn 1 ?>` ?>
 <? if eq $var:ShowUsePstn 1 `
 jslSetChecked("uiViewPstn", jslGetValue("uiPostPstn") == "1");
--- usr/www/avme/en/html/en/first/first_ISP_3.js.orig	2008-04-17 17:48:42.000000000 +0200
+++ usr/www/avme/en/html/en/first/first_ISP_3.js	2009-09-30 11:53:53.000000000 +0200
@@ -1,5 +1,6 @@
 <? setvariable var:showpustefix 0 ?>
 <? if eq $var:OEM avm '<? setvariable var:showpustefix 1 ?>' ?>
+<? if eq $var:OEM tcom '<? setvariable var:showpustefix 1 ?>' ?>
 <? if eq $var:OEM 1und1 '<? setvariable var:showpustefix 1 ?>' ?>
 <!-- ../first/first_ISP_3.js -->
 <style type="text/css">
--- usr/www/avme/en/html/en/first/first_Sip_1.html.orig	2008-04-17 17:48:46.000000000 +0200
+++ usr/www/avme/en/html/en/first/first_Sip_1.html	2009-09-30 11:53:53.000000000 +0200
@@ -58,6 +58,20 @@
 <option value="TOnline">$var:Text12</option>
 <option value="Webde">$var:Text11</option>
 ` ?>
+<? if eq $var:OEM tcom `
+<option value="1u1">$var:Text4</option>
+<option value="AOL">$var:Text9</option>
+<option value="Arcor">$var:Text7</option>
+<option value="Dus">$var:Text13</option>
+<option value="Freenet">$var:Text8</option>
+<option value="GMX">$var:Text5</option>
+<option value="Inode">$var:Text30</option>
+<option value="SuP">$var:Text6</option>
+<option value="Sipgate">$var:Text10</option>
+<option value="Strato">$var:Text28</option>
+<option value="TOnline">$var:Text12</option>
+<option value="Webde">$var:Text11</option>
+` ?>
 <option value="Sonst"><? echo $var:Text14 ?></option>
 </select>
 </p>
--- usr/www/avme/en/html/en/first/first_Sip_1.js.orig	2008-04-17 17:48:46.000000000 +0200
+++ usr/www/avme/en/html/en/first/first_Sip_1.js	2009-09-30 11:53:53.000000000 +0200
@@ -1,5 +1,6 @@
 <? setvariable var:showtcom 0 ?>
 <? if eq $var:OEM avm '<? setvariable var:showtcom 1 ?>' ?>
+<? if eq $var:OEM tcom '<? setvariable var:showtcom 1 ?>' ?>
 <? if eq $var:OEM aol '<? setvariable var:showtcom 1 ?>' ?>
 <!-- ../first/first_Sip_1.js -->
 <style type="text/css">
@@ -129,6 +130,10 @@
 provider = jslGetValue("uiPppProvider");
 jslDisplay( "uiShowProvider", true);
 ` ?>
+<? if eq $var:OEM tcom `
+provider = jslGetValue("uiPppProvider");
+jslDisplay( "uiShowProvider", true);
+` ?>
 } else {
 jslCopyValue("uiView1u1Username", "uiSipRufnr");
 jslSetValue("uiViewArcorUsername", ArcorUsername());
--- usr/www/avme/en/html/en/first/first_Sip_2.js.orig	2008-04-17 17:48:40.000000000 +0200
+++ usr/www/avme/en/html/en/first/first_Sip_2.js	2009-09-30 11:53:53.000000000 +0200
@@ -25,6 +25,10 @@
 var g_txtRufnr_TOnline = "Internet-Telefonnummer:";
 var g_txtUsername_TOnline = "eMail-Adresse:";
 ' ?>
+<? if eq $var:OEM tcom '
+var g_txtRufnr_TOnline = "Internet-Telefonnummer:";
+var g_txtUsername_TOnline = "eMail-Adresse:";
+' ?>
 var g_txtRufnr_Webde = "Internetrufnummer:";
 var g_txtUsername_Webde = "Benutzername:";
 var g_txtRufnr_Dus = "DUS.net-Rufnummer:";
@@ -74,6 +78,12 @@
 SetSpanText("LabelRufnr", g_txtRufnr_TOnline);
 SetSpanText("LabelUserName", g_txtUsername_TOnline);
 ' ?>
+<? if eq $var:OEM tcom '
+} else if (provider == "TOnline") {
+username = jslGetValue("uiSipUsername");
+SetSpanText("LabelRufnr", g_txtRufnr_TOnline);
+SetSpanText("LabelUserName", g_txtUsername_TOnline);
+' ?>
 } else if (provider == "Webde") {
 username = jslGetValue("uiSipUsername");
 SetSpanText("LabelRufnr", g_txtRufnr_Webde);
@@ -152,6 +162,7 @@
 switch (jslGetValue("uiSipRegType")) {
 case "aol":
 case "avm":
+case "tcom":
 case "other":
 <? if eq $var:OEM avme `
 jslSetValue("uiPostAKN", "0"); break;
--- usr/www/avme/en/html/en/first/first_Sip_3.js.orig	2009-09-30 11:53:53.000000000 +0200
+++ usr/www/avme/en/html/en/first/first_Sip_3.js	2009-09-30 11:53:53.000000000 +0200
@@ -30,6 +30,10 @@
 var g_txtRufnr_TOnline = "Internet-Telefonnummer:";
 var g_txtUsername_TOnline = "eMail-Adresse:";
 ' ?>
+<? if eq $var:OEM tcom '
+var g_txtRufnr_TOnline = "Internet-Telefonnummer:";
+var g_txtUsername_TOnline = "eMail-Adresse:";
+' ?>
 var g_txtRufnr_Webde = "Internetrufnummer:";
 var g_txtUsername_Webde = "Benutzername:";
 var g_txtRufnr_Dus = "DUS.net-Rufnummer:";
@@ -89,6 +93,13 @@
 SetSpanText("LabelRufnr", g_txtRufnr_TOnline);
 SetSpanText("LabelUserName", g_txtUsername_TOnline);
 ' ?>
+<? if eq $var:OEM tcom '
+} else if (provider == "TOnline") {
+username = jslGetValue("uiSipUsername");
+registrar = "tel.t-online.de";
+SetSpanText("LabelRufnr", g_txtRufnr_TOnline);
+SetSpanText("LabelUserName", g_txtUsername_TOnline);
+' ?>
 } else if (provider == "Webde") {
 username = jslGetValue("uiSipUsername");
 registrar = "sip.web.de";
--- usr/www/avme/en/html/en/first/first_Start_Sip.js.orig	2008-04-17 17:48:44.000000000 +0200
+++ usr/www/avme/en/html/en/first/first_Start_Sip.js	2009-09-30 11:53:53.000000000 +0200
@@ -1,8 +1,10 @@
 <? setvariable var:showtcom 0 ?>
 <? if eq $var:OEM avm '<? setvariable var:showtcom 1 ?>' ?>
+<? if eq $var:OEM tcom '<? setvariable var:showtcom 1 ?>' ?>
 <? if eq $var:OEM aol '<? setvariable var:showtcom 1 ?>' ?>
 <? setvariable var:showpustefix 0 ?>
 <? if eq $var:OEM avm '<? setvariable var:showpustefix 1 ?>' ?>
+<? if eq $var:OEM tcom '<? setvariable var:showpustefix 1 ?>' ?>
 <? if eq $var:OEM 1und1 '<? setvariable var:showpustefix 1 ?>' ?>
 <!--- ../first/first_Start_Sip.js -->
 <style type="text/css">
