--- usr/www/avme/html/de/home/home.js.orig	2009-01-28 15:56:15.000000000 +0100
+++ usr/www/avme/html/de/home/home.js	2009-02-10 00:05:57.000000000 +0100
@@ -645,12 +645,6 @@
 if (isNaN(nStor)) nStor = 0;
 var nPartTotal = Number("<? query ctlusb:settings/storage-part-total ?>");
 if (isNaN(nStor)) nPartTotal = 0;
-if (nPartTotal > nStor) {
-var str="";
-str += "At least one USB drive or parts of it are not integrated. ";
-str += "<a href=\"javascript:jslGoTo('home', 'syslog');\">more information</a>";
-return str.fontcolor("#cc0000");
-}
 var strArr = new Array();
 if (nUsb == 0)
 {
@@ -659,6 +653,7 @@
 strArr.push(g_usbConnected);
 return strArr.join(" ");
 }
+strArr.push(nStor + " of " + nPartTotal + " partitions mounted.");
 <? if eq '<? query aura:settings/enabled ?>' '1' `
 <? multiquery aura:settings/device/list(class,manufacturer,comment,client) `
 var strAura = UsbDeviceName("<? query aura:settings/$11 ?>", "$13","$14","$12");
