--- Makefile.in
+++ Makefile.in
@@ -377,7 +377,7 @@
 # libtoolize and in case some developer needs to switch back to an
 # old Automake version.
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = doc src tests tune
+SUBDIRS =     src       tune
 nobase_dist_doc_DATA = AUTHORS BUGS COPYING COPYING.LESSER NEWS TODO \
   examples/ReadMe examples/divworst.c examples/rndo-add.c examples/sample.c \
   examples/version.c
