--- linux-2.6.32/drivers/net/avm_cpmac/common/Makefile	2012-07-13 12:12:20+02:00
+++ linux-2.6.32/drivers/net/avm_cpmac/common/Makefile	2012-07-13 12:12:20+02:00
@@ -3,7 +3,7 @@
 # Copyright (C) 2011 AVM GmbH <fritzbox_info@avm.de>
 #
 
-EXTRA_CFLAGS += -DAVMNET_VERSION="\"`svn info $(src)/.. | grep URL | sed -e "s/^.*\/\([0-9.]\+\)-[^\/]*/\1/g"`  -  Revision `svnversion -cn $(src)`  -  `date +\"%a  %e.%m.%Y  %H:%M:%S\"` \""
+EXTRA_CFLAGS += -DAVMNET_VERSION="\"6.54.1.11 - Revision 1939:2217 - Fr 15.02.2013 15:56:01\""
 
 obj-y		+= avmnet_config.o avmnet_common.o
 
