--- etc/init.d/rc.S.orig	2007-09-29 14:26:05.000000000 +0200
+++ etc/init.d/rc.S	2007-09-29 14:26:06.000000000 +0200
@@ -69,8 +69,9 @@
  fi
  echo "SET error,0 = $((0x40 + (4 * ($1 - 1)) + 1))" >/dev/new_led
 }
-mount -n /proc
-mount /var
+mount -o nosuid -t tmpfs dev /dev
+/sbin/makedevs -d /etc/device.table /
+mount -a
 #file "./arch/etc/init.d/rc.watchdog.sh", 0
  echo init-start 120 >/dev/watchdog
 #file "stdin", 117
