--- etc/init.d/rc.S.orig	2008-01-20 19:23:41.000000000 +0100
+++ etc/init.d/rc.S	2008-01-20 19:23:41.000000000 +0100
@@ -86,6 +86,20 @@
 ##########################################################################################
 ##########################################################################################
 echo init-start 120 >/dev/watchdog
+
+# Load boot loader kernel_args API
+. /usr/bin/kernel_args
+# Backup kernel log ring buffer ASAP, so as not to lose its earliest entries
+if ka_isActiveVariable SaveKernelRingBuffer; then
+	ka_decreaseValue SaveKernelRingBuffer
+	dmesg > /var/dmesg-rc.S.log
+fi
+# Initiate inotify-tools file access logging
+if ka_isActiveVariable InotifyBootAnalysis; then
+	ka_decreaseValue InotifyBootAnalysis
+	/etc/init.d/rc.inotify_tools start 
+fi
+
 ##########################################################################################
 ## aus den kernel parameters die für die module ermitteln
 ##########################################################################################
