--- /dev/null
+++ libfdt-freetz/Makefile.freetz
@@ -0,0 +1,32 @@
+CC = gcc
+RM = rm -f
+AR = ar
+RANLIB = ranlib
+
+OPT = -O2
+BITNESS = -m32
+override CFLAGS   += $(OPT) $(BITNESS) -std=c99
+override CPPFLAGS += -I. -D_GNU_SOURCE
+
+include Makefile.libfdt
+
+LIBFDT_a = libfdt.a
+
+all: $(LIBFDT_a)
+
+$(LIBFDT_a): $(LIBFDT_OBJS)
+	-$(RM) $@
+	$(AR) rcu $@ $?
+	$(RANLIB) $@
+
+install: $(LIBFDT_a)
+	@mkdir -p $(PREFIX)/include/ $(PREFIX)/lib/
+	cp *fdt*.h $(PREFIX)/include/
+	cp $(LIBFDT_a) $(PREFIX)/lib/
+
+$(LIBFDT_SRCS): $(LIBFDT_INCLUDES)
+
+clean:
+	-$(RM) *.o *.a
+
+.PHONY: all clean
