--- configure.orig	2008-05-31 17:18:42.000000000 +0200
+++ configure	2008-10-04 11:36:11.000000000 +0200
@@ -15580,7 +15580,7 @@
 if test "${rb_cv_frptr+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  for frptr in	     _IO_read_ptr      _ptr	     __ptr	     bufpos	     _p	; do
+  for frptr in	     _IO_read_ptr      _ptr	     __ptr	     bufpos	     _p        __bufpos	; do
     cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -15643,7 +15643,7 @@
     if test "${rb_cv_frend+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  for frend in	         _IO_read_end	         bufread 	    ; do
+  for frend in	         _IO_read_end	         bufread         __bufread 	    ; do
         cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
--- configure.in.orig	2008-05-31 16:57:11.000000000 +0200
+++ configure.in	2008-10-04 11:36:11.000000000 +0200
@@ -834,6 +834,7 @@
      __ptr	dnl
      bufpos	dnl
      _p	dnl
+     __bufpos	dnl
 ; do
     AC_TRY_COMPILE([#include <stdio.h>
 ],
@@ -853,6 +854,7 @@
     [for frend in	dnl
          _IO_read_end	dnl
          bufread 	dnl
+         __bufread 	dnl
     ; do
         AC_TRY_COMPILE([#include <stdio.h>
     ],
