--- configure	2007-06-08 09:51:42.000000000 +0100
+++ configure	2007-08-29 09:44:29.000000000 +0100
@@ -14957,7 +14957,7 @@
 if test "${rb_cv_frptr+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  for frptr in	     _IO_read_ptr      _ptr	     __ptr	     bufpos	     _p	; do
+  for frptr in	     _IO_read_ptr      _ptr	     __ptr	     bufpos	     _p        __bufpos	; do
     cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -15020,7 +15020,7 @@
     if test "${rb_cv_frend+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  for frend in	         _IO_read_end	         bufread 	    ; do
+  for frend in	         _IO_read_end	         bufread         __bufread 	    ; do
         cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
--- configure.in	2007-05-22 19:19:07.000000000 +0100
+++ configure.in	2007-08-29 09:42:47.000000000 +0100
@@ -707,6 +707,7 @@
      __ptr	dnl
      bufpos	dnl
      _p	dnl
+     __bufpos	dnl
 ; do
     AC_TRY_COMPILE([#include <stdio.h>
 ],
@@ -726,6 +727,7 @@
     [for frend in	dnl
          _IO_read_end	dnl
          bufread 	dnl
+         __bufread 	dnl
     ; do
         AC_TRY_COMPILE([#include <stdio.h>
     ],
