--- configure.orig	2007-03-12 20:19:03.000000000 +0100
+++ configure	2009-02-13 15:44:07.000000000 +0100
@@ -14952,7 +14952,7 @@
 if test "${rb_cv_frptr+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  for frptr in	     _IO_read_ptr      _ptr	     __ptr	     bufpos	     _p	; do
+  for frptr in	     _IO_read_ptr      _ptr	     __ptr	     bufpos	     _p        __bufpos	; do
     cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -15015,7 +15015,7 @@
     if test "${rb_cv_frend+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  for frend in	         _IO_read_end	         bufread 	    ; do
+  for frend in	         _IO_read_end	         bufread         __bufread 	    ; do
         cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
--- configure.in.orig	2007-02-28 14:23:42.000000000 +0100
+++ configure.in	2009-02-13 15:44:07.000000000 +0100
@@ -702,6 +702,7 @@
      __ptr	dnl
      bufpos	dnl
      _p	dnl
+     __bufpos	dnl
 ; do
     AC_TRY_COMPILE([#include <stdio.h>
 ],
@@ -721,6 +722,7 @@
     [for frend in	dnl
          _IO_read_end	dnl
          bufread 	dnl
+         __bufread 	dnl
     ; do
         AC_TRY_COMPILE([#include <stdio.h>
     ],
