--- linux-2.6.13.1/arch/mips/kernel/head.S.orig	2005-09-14 12:22:35.000000000 +0200
+++ linux-2.6.13.1/arch/mips/kernel/head.S	2007-07-28 16:57:36.000000000 +0200
@@ -122,11 +122,15 @@
 #endif
 	.endm
 
+
+	j kernel_entry
+	nop
+
 	/*
 	 * Reserved space for exception handlers.
 	 * Necessary for machines which link their kernels at KSEG0.
 	 */
-	.fill	0x400
+	.align 10
 
 EXPORT(stext)					# used for profiling
 EXPORT(_stext)
