--- linux-2.6.13.1/net/ipv4/mcfastforward.c.orig	2007-04-18 17:35:17.000000000 +0200
+++ linux-2.6.13.1/net/ipv4/mcfastforward.c	2007-06-06 00:47:22.000000000 +0200
@@ -1424,6 +1424,16 @@
    return mcfw_multicast_forward(sourceid, skb, iph);
 }
 
+
+/* dirty hack */
+void mcfw_multicast_forward_ethernet_enable(void) {
+};
+void mcfw_multicast_forward_disable(void) {
+};
+
+EXPORT_SYMBOL(mcfw_multicast_forward_ethernet_enable);
+EXPORT_SYMBOL(mcfw_multicast_forward_disable);
+
 EXPORT_SYMBOL(mcfw_netdriver_register);
 EXPORT_SYMBOL(mcfw_netdriver_unregister);
 EXPORT_SYMBOL(_mcfw_snoop_recv);
