diff -burN linux-2.6.13.1.orig/Makefile linux-2.6.13.1/Makefile
--- linux-2.6.13.1.orig/Makefile	2007-04-22 14:19:55.000000000 +0200
+++ linux-2.6.13.1/Makefile	2007-04-22 15:07:29.000000000 +0200
@@ -932,7 +932,10 @@
 endif
 .PHONY: _modinst_post
 _modinst_post: _modinst_
+# Do not run depmod while cross-compiling
+ifeq ([ uname -m ],$(ARCH))
 	if [ -r System.map -a -x $(DEPMOD) ]; then $(DEPMOD) -ae -F System.map $(depmod_opts) $(KERNELRELEASE); fi
+endif
 
 else # CONFIG_MODULES
 
