--- linux-2.6.13.1/fs/Kconfig.orig	2005-10-19 09:40:38.000000000 +0200
+++ linux-2.6.13.1/fs/Kconfig	2007-07-28 12:44:45.000000000 +0200
@@ -328,6 +328,9 @@
 
 source "fs/xfs/Kconfig"
 
+config MINI_FO
+	tristate "Mini fanout overlay filesystem"
+
 config MINIX_FS
 	tristate "Minix fs support"
 	help
--- linux-2.6.13.1/fs/Makefile.orig	2005-10-19 09:40:38.000000000 +0200
+++ linux-2.6.13.1/fs/Makefile	2007-07-28 12:44:45.000000000 +0200
@@ -59,6 +59,7 @@
 obj-$(CONFIG_RAMFS)		+= ramfs/
 obj-$(CONFIG_HUGETLBFS)		+= hugetlbfs/
 obj-$(CONFIG_CODA_FS)		+= coda/
+obj-$(CONFIG_MINI_FO)		+= mini_fo/
 obj-$(CONFIG_MINIX_FS)		+= minix/
 obj-$(CONFIG_FAT_FS)		+= fat/
 obj-$(CONFIG_MSDOS_FS)		+= msdos/
