--- linux-2.6.13.1/fs/Kconfig.orig	2007-07-28 12:44:45.000000000 +0200
+++ linux-2.6.13.1/fs/Kconfig	2007-07-28 12:44:46.000000000 +0200
@@ -1362,6 +1362,16 @@
 	  Say Y here if you want to try writing to UFS partitions. This is
 	  experimental, so you should back up your UFS partitions beforehand.
 
+config UNION_FS
+	tristate "Union fs support"
+	depends on EXPERIMENTAL
+	help
+	  Unionfs is a stackable unification file system, which can
+	  appear to merge the contents of several directories (branches),
+	  while keeping their physical content separate.
+
+	  see <http://www.fsl.cs.sunysb.edu/project-unionfs.html> for details
+
 endmenu
 
 menu "Network File Systems"
--- linux-2.6.13.1/fs/Makefile.orig	2007-07-28 12:44:45.000000000 +0200
+++ linux-2.6.13.1/fs/Makefile	2007-07-28 12:44:46.000000000 +0200
@@ -101,3 +101,4 @@
 obj-$(CONFIG_HPPFS)		+= hppfs/
 obj-$(CONFIG_DEBUG_FS)		+= debugfs/
 obj-$(CONFIG_YFFS_FS)		+= yaffs2/
+obj-$(CONFIG_UNION_FS)          += unionfs/
