--- source3/configure
+++ source3/configure
@@ -17599,6 +17599,7 @@
 
 #####################################
 # needed for SRV lookups
+if false; then
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for dn_expand in -lresolv" >&5
 $as_echo_n "checking for dn_expand in -lresolv... " >&6; }
 if ${ac_cv_lib_resolv_dn_expand+:} false; then :
@@ -17733,6 +17734,7 @@
   LIBS="-lresolv $LIBS"
 
 fi
+fi
 
 
 #
