--- linux-2.6.39/GNUmakefile
+++ linux-2.6.39/GNUmakefile
@@ -105,13 +105,6 @@
 		avm-subprojects='$(avm-subprojects)' \
 		USERINCLUDE_AVM_SUBPROJECTS='$(USERINCLUDE_AVM_SUBPROJECTS)' \
 		LINUXINCLUDE_AVM_SUBPROJECTS='$(LINUXINCLUDE_AVM_SUBPROJECTS)'
-ifeq ($(KBUILD_EXTMOD)$(M),)
-    ifneq ($(AVM_KCONFIG_CONFIG),$(KCONFIG_CONFIG),)
-	$(if $(filter-out $(no-config-diff-targets),$@),\
-		@scripts/diffconfig $(AVM_KCONFIG_CONFIG) $(KCONFIG_CONFIG) | \
-		 sed -re '1{s|^|Linux config has changed compared to $(AVM_KCONFIG_CONFIG):\n|}') >&2
-    endif
-endif
 
 $(KCONFIG_CONFIG): $(wildcard $(AVM_KCONFIG_CONFIG))
 	$(if $<,@echo "UPD $< -> $@")
