--- usr/www/avme/en/html/en/fon/sperre1.js.orig	2008-01-07 13:34:52.000000000 +0100
+++ usr/www/avme/en/html/en/fon/sperre1.js	2008-01-16 01:00:50.000000000 +0100
@@ -48,7 +48,7 @@
 switch (mode) {
 case "In": {
 var nummer = jslGetValue("uiViewNummer");
-if (!isZahlRaute(nummer)) {alert(g_mldNonDigits); return;}
+//if (!isZahlRaute(nummer)) {alert(g_mldNonDigits); return;}
 var x = IstActionVorhanden(nummer);
 if (x != "") {
 if (jslGetValue("uiPostInAction"+x) == "1") {
@@ -139,7 +139,7 @@
 }
 } else {
 var nummer = jslGetValue("uiViewNummer");
-if (!isZahlRaute(nummer)) {alert(g_mldNonDigits); return;}
+//if (!isZahlRaute(nummer)) {alert(g_mldNonDigits); return;}
 var x = IstActionVorhanden(nummer);
 if (x != "") {
 if (jslGetValue("uiPostInAction"+x) == "1") {alert(g_txtAlert_1); return;}
