--- src/ip.h
+++ src/ip.h
@@ -92,14 +92,12 @@
 /*Wrap getifaddrs for Windows, earlier Solaris.
 */
 
-#if !(defined(__sun) && defined(__SVR4)) && defined(HAVE_IFADDRS_H)
+#if !defined(_WIN32) && (defined(HAVE_IFADDRS_H) && !(defined(__sun) && defined(__SVR4)))
 
 #define ifaddrs_w ifaddrs
 
 #else
 
-#ifdef _WIN32
-
 /*Wrap ifaddrs, getifaddrs for Windows, earlier Solaris.
 */
 struct ifaddrs_w
@@ -122,7 +120,6 @@
 }ifaddrs_w;
 
 #endif
-#endif
 
 typedef struct IPV6_PRIV_EXT
 {
