diff -Naur a/man/inadyn.8 b/man/inadyn.8
--- man/inadyn.8	2013-01-19 08:39:22.000000000 +0100
+++ man/inadyn.8	2013-06-09 15:44:48.104802122 +0200
@@ -79,6 +79,8 @@
 .TP
 http://www.tunnelbroker.net,
 .TP
+http://dns.he.net,
+.TP
 http://www.dynsip.org
 .TP
 http://www.dhis.org
@@ -159,7 +161,7 @@
 
 There is only one acceptable service for http://freedns.afraid.org, which is default@freedns.afraid.org. 
 
-Other services are: default@zoneedit.com, default@no-ip.com, default@easydns.com, dyndns@3322.org, default@sitelutions.com, default@dnsomatic.com, ipv6tb@he.net, default@tzo.com, default@dynsip.org, default@dhis.org, default@majimoto.net, default@zerigo.com, custom@http_svr_basic_auth. The default service is dyndns@dyndns.org, which is believed to be used by most users, at least initially.
+Other services are: default@zoneedit.com, default@no-ip.com, default@easydns.com, dyndns@3322.org, default@sitelutions.com, default@dnsomatic.com, ipv6tb@he.net, default@he.net, default@tzo.com, default@dynsip.org, default@dhis.org, default@majimoto.net, default@zerigo.com, custom@http_svr_basic_auth. The default service is dyndns@dyndns.org, which is believed to be used by most users, at least initially.
 .TP
 .I \-\-proxy_server[:port]
 An http proxy server name and port. The default is none.
diff -Naur a/man/inadyn-mt.8 b/man/inadyn-mt.8
--- man/inadyn-mt.8	2013-01-19 08:39:22.000000000 +0100
+++ man/inadyn-mt.8	2013-06-09 15:44:48.104802122 +0200
@@ -79,6 +79,8 @@
 .TP
 http://www.tunnelbroker.net,
 .TP
+http://dns.he.net,
+.TP
 http://www.dynsip.org
 .TP
 http://www.dhis.org
@@ -159,7 +161,7 @@
 
 There is only one acceptable service for http://freedns.afraid.org, which is default@freedns.afraid.org. 
 
-Other services are: default@zoneedit.com, default@no-ip.com, default@easydns.com, dyndns@3322.org, default@sitelutions.com, default@dnsomatic.com, ipv6tb@he.net, default@tzo.com, default@dynsip.org, default@dhis.org, default@majimoto.net, default@zerigo.com, custom@http_svr_basic_auth. The default service is dyndns@dyndns.org, which is believed to be used by most users, at least initially.
+Other services are: default@zoneedit.com, default@no-ip.com, default@easydns.com, dyndns@3322.org, default@sitelutions.com, default@dnsomatic.com, ipv6tb@he.net, default@he.net, default@tzo.com, default@dynsip.org, default@dhis.org, default@majimoto.net, default@zerigo.com, custom@http_svr_basic_auth. The default service is dyndns@dyndns.org, which is believed to be used by most users, at least initially.
 .TP
 .I \-\-proxy_server[:port]
 An http proxy server name and port. The default is none.
diff -Naur a/readme.html b/readme.html
--- readme.html	2013-06-09 15:48:51.988791866 +0200
+++ readme.html	2013-06-09 15:44:48.108802122 +0200
@@ -57,6 +57,7 @@
   <li><a href="http://www.sitelutions.com">sitelutions.com</a></li>
   <li><a href="http://www.dnsomatic.com">dnsomatic.com</a></li>
   <li><a href="http://ipv6tb.he.net">ipv6tb.he.net</a></li>
+  <li><a href="http://dns.he.net">dns.he.net</a></li>
   <li><a href="http://www.tzo.com">tzo.com</a></li>
   <li><a href="http://www.dynsip.org">dynsip.org</a></li>
   <li><a href="http://www.dhis.org">dhis.org</a></li>
@@ -321,6 +322,7 @@
 -For sitelutions.com:  default@sitelutions.com<br>
 -For dnsomatic.com:  default@dnsomatic.com<br>
 -For tunnelbroker.net:  ipv6tb@he.net<br>
+-For dns.he.net:  default@he.net<br>
 -For tzo.com:  default@tzo.com<br>
 -For dynsip.org:  default@dynsip.org<br>
 -For dhis.org:  default@dhis.org<br>
diff -Naur a/src/dyndns.c b/src/dyndns.c
--- src/dyndns.c	2013-06-09 15:48:51.992791866 +0200
+++ src/dyndns.c	2013-06-09 15:47:47.384794583 +0200
@@ -225,6 +225,7 @@
 static int get_req_for_sitelutions_http_dns_server(DYN_DNS_CLIENT *p_self,char srv_cnt, int cnt,  DYNDNS_SYSTEM *p_sys_info);
 static int get_req_for_tzo_http_dns_server(DYN_DNS_CLIENT *p_self,char srv_cnt, int cnt, DYNDNS_SYSTEM *p_sys_info);
 static int get_req_for_he_ipv6_server(DYN_DNS_CLIENT *p_self,char srv_cnt, int cnt, DYNDNS_SYSTEM *p_sys_info);
+static int get_req_for_he_dns_server(DYN_DNS_CLIENT *p_self,char srv_cnt, int cnt, DYNDNS_SYSTEM *p_sys_info);
 static int get_req_for_dhis_http_dns_server(DYN_DNS_CLIENT *p_self,char srv_cnt, int cnt, DYNDNS_SYSTEM *p_sys_info);
 static int get_req_for_majimoto_http_dns_server(DYN_DNS_CLIENT *p_self,char srv_cnt, int cnt, DYNDNS_SYSTEM *p_sys_info);
 static int get_req_for_zerigo_http_dns_server(DYN_DNS_CLIENT *p_self,char srv_cnt, int cnt, DYNDNS_SYSTEM *p_sys_info);
@@ -359,6 +360,14 @@
 			DYNDNS_MY_IP_SERVER, DYNDNS_MY_IP_SERVER_URL,
 			"ipv4.tunnelbroker.net", "/ipv4_end.php?", NULL}},
 
+		{HE_DEFAULT,
+			{"default@he.net", NULL,
+			(DNS_SYSTEM_SRV_RESPONSE_OK_FUNC)is_dyndns_server_rsp_ok,
+			(DNS_SYSTEM_SRV_RESPONSE_CONFIG_FUNC)is_dyndns_server_rsp_config,
+			(DNS_SYSTEM_REQUEST_FUNC) get_req_for_he_dns_server,
+			"checkip.dns.he.net", "/",
+			"dyn.dns.he.net", "/nic/update?", NULL}},
+
 		/* Support for dynsip.org by milkfish, from DD-WRT */
 		{DYNSIP_DEFAULT,
 			{"default@dynsip.org", NULL,
@@ -706,6 +715,19 @@
 				p_self->info[srv_cnt].dyndns_server_name.name);
 }
 
+static int get_req_for_he_dns_server(DYN_DNS_CLIENT *p_self,char srv_cnt, int cnt,  DYNDNS_SYSTEM *p_sys_info)
+{
+
+	(void)p_sys_info;
+
+	return sprintf(p_self->p_req_buffer, HE_DEFAULT_UPDATE_MY_IP_REQUEST_FORMAT,
+	               p_self->info[srv_cnt].dyndns_server_name.url,
+				   p_self->info[srv_cnt].alias_info.names[cnt].name,
+				   p_self->info[srv_cnt].i_face.my_ip_address.name[ip_store],
+				   p_self->info[srv_cnt].credentials.p_enc_usr_passwd_buffer,				   
+				   p_self->info[srv_cnt].dyndns_server_name.name);
+}
+
 static int get_req_for_dhis_http_dns_server(DYN_DNS_CLIENT *p_self,char srv_cnt, int cnt,  DYNDNS_SYSTEM *p_sys_info)
 {
 
diff -Naur a/src/dyndns.h b/src/dyndns.h
--- src/dyndns.h	2013-06-09 15:48:51.992791866 +0200
+++ src/dyndns.h	2013-06-09 15:44:48.108802122 +0200
@@ -97,6 +97,7 @@
 	TZO_DEFAULT,
 	DNSOMATIC_DEFAULT,
 	HE_IPV6TB,
+	HE_DEFAULT,
 	DYNSIP_DEFAULT,
 	DHIS_DEFAULT,
 	MAJIMOTO_DEFAULT,
@@ -241,6 +242,16 @@
 	"Host: %s\r\n"							\
 	"User-Agent: "DYNDNS_AGENT_NAME " " DYNDNS_EMAIL_ADDR"\r\n\r\n"
 
+/* HE DynDNS specific update request format */
+#define HE_DEFAULT_UPDATE_MY_IP_REQUEST_FORMAT				\
+	"GET %s"							\
+	"hostname=%s&"							\
+	"myip=%s "							\
+	"HTTP/1.0\r\n"							\
+	"Authorization: Basic %s\r\n"	\
+	"Host: %s\r\n"							\
+	"User-Agent: "DYNDNS_AGENT_NAME " " DYNDNS_EMAIL_ADDR"\r\n\r\n"
+
 /* is.dhis.org specific update request format */
 #define DHIS_MY_IP_UPDATE_REQUEST_FORMAT				\
 	"GET %s" \
diff -Naur a/src/inadyn_cmd.c b/src/inadyn_cmd.c
--- src/inadyn_cmd.c	2013-06-09 15:48:51.996791866 +0200
+++ src/inadyn_cmd.c	2013-06-09 15:44:48.108802122 +0200
@@ -196,6 +196,7 @@
 		 "\t\t-For sitelutions.com:  default@sitelutions.com\n"
 		 "\t\t-For dnsomatic.com:  default@dnsomatic.com\n"
 		 "\t\t-For tunnelbroker.net:  ipv6tb@he.net\n"
+		 "\t\t-For dns.he.net:  default@he.net\n"
 		 "\t\t-For tzo.com:  default@tzo.com\n"
 		 "\t\t-For dynsip.org:  default@dynsip.org\n"
 		 "\t\t-For dhis.org:  default@dhis.org\n"
