--- man/inadyn.8
+++ man/inadyn.8
@@ -81,6 +81,8 @@
 .TP
 http://dns.he.net,
 .TP
+http://www.changeip.com,
+.TP
 http://www.dynsip.org
 .TP
 http://www.dhis.org
@@ -161,7 +163,7 @@
 
 There is only one acceptable service for http://freedns.afraid.org, which is default@freedns.afraid.org. 
 
-Other services are: default@zoneedit.com, default@no-ip.com, default@easydns.com, dyndns@3322.org, default@sitelutions.com, default@dnsomatic.com, ipv6tb@he.net, default@he.net, default@tzo.com, default@dynsip.org, default@dhis.org, default@majimoto.net, default@zerigo.com, custom@http_svr_basic_auth. The default service is dyndns@dyndns.org, which is believed to be used by most users, at least initially.
+Other services are: default@zoneedit.com, default@no-ip.com, default@easydns.com, dyndns@3322.org, default@sitelutions.com, default@dnsomatic.com, ipv6tb@he.net, default@he.net, default@changeip.com, default@tzo.com, default@dynsip.org, default@dhis.org, default@majimoto.net, default@zerigo.com, custom@http_svr_basic_auth. The default service is dyndns@dyndns.org, which is believed to be used by most users, at least initially.
 .TP
 .I \-\-proxy_server[:port]
 An http proxy server name and port. The default is none.
--- man/inadyn-mt.8
+++ man/inadyn-mt.8
@@ -81,6 +81,8 @@
 .TP
 http://dns.he.net,
 .TP
+http://www.changeip.com,
+.TP
 http://www.dynsip.org
 .TP
 http://www.dhis.org
@@ -161,7 +163,7 @@
 
 There is only one acceptable service for http://freedns.afraid.org, which is default@freedns.afraid.org. 
 
-Other services are: default@zoneedit.com, default@no-ip.com, default@easydns.com, dyndns@3322.org, default@sitelutions.com, default@dnsomatic.com, ipv6tb@he.net, default@he.net, default@tzo.com, default@dynsip.org, default@dhis.org, default@majimoto.net, default@zerigo.com, custom@http_svr_basic_auth. The default service is dyndns@dyndns.org, which is believed to be used by most users, at least initially.
+Other services are: default@zoneedit.com, default@no-ip.com, default@easydns.com, dyndns@3322.org, default@sitelutions.com, default@dnsomatic.com, ipv6tb@he.net, default@he.net, default@changeip.com, default@tzo.com, default@dynsip.org, default@dhis.org, default@majimoto.net, default@zerigo.com, custom@http_svr_basic_auth. The default service is dyndns@dyndns.org, which is believed to be used by most users, at least initially.
 .TP
 .I \-\-proxy_server[:port]
 An http proxy server name and port. The default is none.
--- readme.html
+++ readme.html
@@ -58,6 +58,7 @@
   <li><a href="http://www.dnsomatic.com">dnsomatic.com</a></li>
   <li><a href="http://ipv6tb.he.net">ipv6tb.he.net</a></li>
   <li><a href="http://dns.he.net">dns.he.net</a></li>
+  <li><a href="http://www.changeip.com">www.changeip.com</a></li>
   <li><a href="http://www.tzo.com">tzo.com</a></li>
   <li><a href="http://www.dynsip.org">dynsip.org</a></li>
   <li><a href="http://www.dhis.org">dhis.org</a></li>
@@ -323,6 +324,7 @@
 -For dnsomatic.com:  default@dnsomatic.com<br>
 -For tunnelbroker.net:  ipv6tb@he.net<br>
 -For dns.he.net:  default@he.net<br>
+-For changeip.com:  default@changeip.com<br>
 -For tzo.com:  default@tzo.com<br>
 -For dynsip.org:  default@dynsip.org<br>
 -For dhis.org:  default@dhis.org<br>
--- src/dyndns.c
+++ src/dyndns.c
@@ -226,6 +226,7 @@
 static int get_req_for_tzo_http_dns_server(DYN_DNS_CLIENT *p_self,char srv_cnt, int cnt, DYNDNS_SYSTEM *p_sys_info);
 static int get_req_for_he_ipv6_server(DYN_DNS_CLIENT *p_self,char srv_cnt, int cnt, DYNDNS_SYSTEM *p_sys_info);
 static int get_req_for_he_dns_server(DYN_DNS_CLIENT *p_self,char srv_cnt, int cnt, DYNDNS_SYSTEM *p_sys_info);
+static int get_req_for_changeip_http_dns_server(DYN_DNS_CLIENT *p_self,char srv_cnt, int cnt,  DYNDNS_SYSTEM *p_sys_info);
 static int get_req_for_dhis_http_dns_server(DYN_DNS_CLIENT *p_self,char srv_cnt, int cnt, DYNDNS_SYSTEM *p_sys_info);
 static int get_req_for_majimoto_http_dns_server(DYN_DNS_CLIENT *p_self,char srv_cnt, int cnt, DYNDNS_SYSTEM *p_sys_info);
 static int get_req_for_zerigo_http_dns_server(DYN_DNS_CLIENT *p_self,char srv_cnt, int cnt, DYNDNS_SYSTEM *p_sys_info);
@@ -368,6 +369,14 @@
 			"checkip.dns.he.net", "/",
 			"dyn.dns.he.net", "/nic/update?", NULL}},
 
+		{CHANGEIP_DEFAULT,
+			{"default@changeip.com", NULL,
+			(DNS_SYSTEM_SRV_RESPONSE_OK_FUNC)is_dyndns_server_rsp_ok,
+			(DNS_SYSTEM_SRV_RESPONSE_CONFIG_FUNC)is_dyndns_server_rsp_config,
+			(DNS_SYSTEM_REQUEST_FUNC) get_req_for_changeip_http_dns_server,
+			"ip.changeip.com", "/",
+			"nic.changeip.com", "/nic/update?system=dyndns&hostname=", ""}},
+
 		/* Support for dynsip.org by milkfish, from DD-WRT */
 		{DYNSIP_DEFAULT,
 			{"default@dynsip.org", NULL,
@@ -728,6 +737,17 @@
 				   p_self->info[srv_cnt].dyndns_server_name.name);
 }
 
+static int get_req_for_changeip_http_dns_server(DYN_DNS_CLIENT *p_self,char srv_cnt, int cnt,  DYNDNS_SYSTEM *p_sys_info)
+{
+	(void)p_sys_info;
+	return sprintf(p_self->p_req_buffer, GENERIC_CHANGEIP_AUTH_MY_IP_REQUEST_FORMAT,
+	               p_self->info[srv_cnt].dyndns_server_name.url,
+	               p_self->info[srv_cnt].alias_info.names[cnt].name,
+	               p_self->info[srv_cnt].i_face.my_ip_address.name[ip_store],
+	               p_self->info[srv_cnt].credentials.p_enc_usr_passwd_buffer,
+	               p_self->info[srv_cnt].dyndns_server_name.name);
+}
+
 static int get_req_for_dhis_http_dns_server(DYN_DNS_CLIENT *p_self,char srv_cnt, int cnt,  DYNDNS_SYSTEM *p_sys_info)
 {
 
--- src/dyndns.h
+++ src/dyndns.h
@@ -98,6 +98,7 @@
 	DNSOMATIC_DEFAULT,
 	HE_IPV6TB,
 	HE_DEFAULT,
+	CHANGEIP_DEFAULT,
 	DYNSIP_DEFAULT,
 	DHIS_DEFAULT,
 	MAJIMOTO_DEFAULT,
@@ -252,6 +253,13 @@
 	"Host: %s\r\n"							\
 	"User-Agent: "DYNDNS_AGENT_NAME " " DYNDNS_EMAIL_ADDR"\r\n\r\n"
 
+#define GENERIC_CHANGEIP_AUTH_MY_IP_REQUEST_FORMAT \
+	"GET %s%s&ip=%s " \
+	"HTTP/1.0\r\n" \
+	"Authorization: Basic %s\r\n" \
+	"Host: %s\r\n" \
+	"User-Agent: "DYNDNS_AGENT_NAME " " DYNDNS_EMAIL_ADDR"\r\n\r\n"
+
 /* is.dhis.org specific update request format */
 #define DHIS_MY_IP_UPDATE_REQUEST_FORMAT				\
 	"GET %s" \
--- src/inadyn_cmd.c
+++ src/inadyn_cmd.c
@@ -197,6 +197,7 @@
 		 "\t\t-For dnsomatic.com:  default@dnsomatic.com\n"
 		 "\t\t-For tunnelbroker.net:  ipv6tb@he.net\n"
 		 "\t\t-For dns.he.net:  default@he.net\n"
+		 "\t\t-For changeip.com:  default@changeip.com\n"
 		 "\t\t-For tzo.com:  default@tzo.com\n"
 		 "\t\t-For dynsip.org:  default@dynsip.org\n"
 		 "\t\t-For dhis.org:  default@dhis.org\n"
