--- extra/servers_additional.cfg
+++ extra/servers_additional.cfg
@@ -15,3 +15,4 @@
 default@strato.com --dyndns_server_name dyndns.strato.com --dyndns_server_url /nic/update?system=dyndns&hostname= --svr_rsp_success "good,nochg" --svr_ip_param myip
 default@system-ns.com --dyndns_server_name system-ns.com --dyndns_server_url /api?type=dynamic&command=set&domain= --svr_rsp_success "0" --svr_rsp_other "99" --svr_ip_param ip blahblah token
 mondo@freedns.afraid.org --dyndns_server_name freedns.afraid.org --dyndns_server_url /dynamic/update.php? --svr_rsp_success "has not changed" --svr_ip_in_rsp --svr_ip_param address blah blah
+default@tunnelbroker.net --dyndns_server_name ipv4.tunnelbroker.net --dyndns_server_url /nic/update?hostname= --ip_server_name checkip.dns.he.net / --svr_rsp_success "good,nochg" --svr_rsp_other "dnserr,911" --svr_ip_param myip
