--- extra/servers_additional.cfg
+++ extra/servers_additional.cfg
@@ -1,5 +1,5 @@
 default@dtdns.com --dyndns_server_name www.dtdns.com --dyndns_server_url /api/autodns.cfm?id= --ip_server_name myip.dtdns.com / --svr_ip_param ip usr pw --svr_rsp_success "Host ($ALIAS) now points to ($IP_ADDR)."
-default@changeip.com --dyndns_server_name ip.changeip.com --dyndns_server_url /nic/update?system=dyndns&hostname= --ip_server_name ip.changeip.com / --svr_rsp_success "good,nochg" --svr_ip_param myip
+default@changeip.com --dyndns_server_name nic.changeip.com --dyndns_server_url /nic/update?hostname= --ip_server_name ip.changeip.com / --svr_rsp_success "good,nochg" --svr_ip_param myip
 default@dnsexit.com --dyndns_server_name update.dnsexit.com --dyndns_server_url /RemoteUpdate.sv?host= --ip_server_name ip.dnsexit.com / --svr_ip_param myip login password --svr_rsp_success "0=,1=" --svr_rsp_other "11="
 ipv4@nsupdate.info --dyndns_server_name ipv4.nsupdate.info --dyndns_server_url /nic/update?system=dyndns&hostname= --ip_server_name ipv4.nsupdate.info /myip --svr_rsp_success "good,nochg" --svr_ip_param myip
 ipv6@nsupdate.info --dyndns_server_name ipv6.nsupdate.info --dyndns_server_url /nic/update?system=dyndns&hostname= --ip_server_name ipv6.nsupdate.info /myip --svr_rsp_success "good,nochg" --svr_ip_param myip
