--- linux-2.6.28/drivers/net/avm_cpmac/Makefile
+++ linux-2.6.28/drivers/net/avm_cpmac/Makefile
@@ -6,7 +6,7 @@
 
 
 EXTRA_CFLAGS := -DCPMAC_SIGNAL_CONGESTION -DCPPHY_USE_SWITCH -Wall -Wextra -Werror
-EXTRA_CFLAGS += -DAVM_CPMAC_VERSION="\"`svn info $(src) | grep URL | sed -e "s/^.*\/\([0-9.]\+\)-[^\/]*/\1/g"`  -  Revision `svnversion -cn $(src)`  -  `date +\"%a  %e.%m.%Y  %H:%M:%S\"` \""
+EXTRA_CFLAGS += -DAVM_CPMAC_VERSION="\"1.108.1.32.1.15 - Revision 811:1568 - Mi  19.10.2011 11:06:17 \""
 
 NOTANGLE=notangle
 NOWEAVE=noweave
@@ -23,7 +23,7 @@
 			   cpphy_entry.o cpphy_ar8216_ar.o cpphy_mgmt.o \
 			   cpmac_fusiv_if.o cpmac_puma_if.o cpphy_switch.o cpmac_product_conf.o
 
-$(obj)/cpphy_entry.o: $(src)/cpphy_entry.c DO_COPY FORCE
+$(obj)/cpphy_entry.o: $(src)/cpphy_entry.c FORCE
 
 DO_COPY:
 	mkdir -p $(FRITZ_BOX_BUILD_DIR)/$(KERNEL_BUILD)/filesystem/include/linux 
