--- linux-2.6.28/arch/mips/include/asm/mach-infineon/ifxmips_mei_interface.h.orig	2011-02-24 16:53:13.000000000 +0100
+++ linux-2.6.28/arch/mips/include/asm/mach-infineon/ifxmips_mei_interface.h	2011-12-19 20:31:22.998408129 +0100
@@ -427,6 +427,8 @@
 #define MEI_CRLF        "\n"
 #define MEI_DBG_PREFIX  "MEI_DRV: "
 
+#define MEI_DEBUG_DISABLE
+
 #ifdef MEI_DEBUG_DISABLE
 #define MEI_DEBUG(module, level, body)   ((void)0)
 #else
--- linux-2.6.28/drivers/char/ifxmips_mei.c.orig	2011-12-19 21:54:03.000000000 +0100
+++ linux-2.6.28/drivers/char/ifxmips_mei.c	2011-12-19 20:39:36.000000000 +0100
@@ -3766,9 +3766,10 @@
 
       *ppos += (p - buf);
    }
+/*
    else if (strcmp(entry->name, "DBG_ENABLE") == 0)
    {
-      if (*ppos > 0) /* Assume reading completed in previous read */
+      if (*ppos > 0)
          return 0;
 
       for (i = 0; i < MEI_DBG_MOD_LAST; i++)
@@ -3779,6 +3780,7 @@
 
       *ppos += (p - buf);
    }
+*/
    else if (strcmp(entry->name, "fw_version") == 0)
    {
       if (*ppos > 0) /* Assume reading completed in previous read */
