--- fortune.c.old	2008-02-15 13:20:30.000000000 +0000
+++ fortune.c	2010-01-28 15:44:02.000000000 +0000
@@ -50,8 +50,7 @@
 #warning Your system headers say that mmap is not supported!
 #endif
 
-#define VERSION "1.0"
-#define FORTUNEDIR "/usr/share/games/fortunes"
+#define VERSION "1.2"
 
 struct option const long_options[] =
 {
@@ -175,6 +174,12 @@
   int c;
   char *re;
 
+  char *FORTUNEDIR;
+
+  FORTUNEDIR = getenv("FORTUNE_DIR");
+  if (FORTUNEDIR == NULL)
+    FORTUNEDIR = "/var/media/fortunes";
+
   progname=argv[0]; re=NULL;
   while ((c = getopt_long(argc, argv, "h?m:V",
 			  long_options, (int *) 0)) != -1) {
