--- src/Makefile
+++ src/Makefile
@@ -544,7 +544,7 @@
 # additional libraries for linker calls                                                               #
 #                                                                                                     #
 #######################################################################################################
-LIBS = -lcrypto
+LDLIBS = -lcrypto
 #######################################################################################################
 #                                                                                                     #
 # target binary name                                                                                  #
@@ -613,9 +613,6 @@
 # build targets and rules                                                                             #
 #                                                                                                     #
 #######################################################################################################
-%.o: %.c
-	$(CC) $(CFLAGS) -c $< -o $@
-
 all:	$(binfile)
 
 .PHONY:	all install_$(binfile) links uninstall clean install $(CMDS)
@@ -627,7 +624,6 @@
 $(OBJS_MAIN):	$(APPLETS) $(CONFIG) $(HDRS_MAIN) $(SRCS_MAIN) $(USE_MAIN)
 
 $(binfile):	$(OBJS_COMMON) $(OBJS_APP) $(OBJS_MAIN)
-	$(CC) $(LDFLAGS) -L. $(LIBS) -o $@ $(filter %.o,$^)
 
 $(APPLETS): | $(CMDS)
 	@$(RM) $@ 2>/dev/null || true
