--- src/decoder.c
+++ src/decoder.c
@@ -99,6 +99,7 @@
 	{ .name = "decode_secret", .ep = &decode_secret_main },
 	{ .name = "decode_secrets", .ep = &decode_secrets_main },
 	{ .name = "decode_passwords", .ep = &decode_secrets_main },
+	{ .name = "decrypt-fritzos-cfg", .ep = &decode_secrets_main }, // Freetz only alias
 	{ .name = "decode_export", .ep = &decode_export_main },
 	{ .name = NULL, .ep = NULL }
 };
