--- usr/www/all/html/de/home/foncalls.js.orig	2010-02-23 09:48:52.000000000 +0100
+++ usr/www/all/html/de/home/foncalls.js	2010-02-23 18:40:18.000000000 +0100
@@ -355,17 +355,17 @@
 var buchname = GetBuchName(name);
 var buchnrtype = GetBuchNumberType(telbooknumbertyp);
 var buchentryinfo = GetBuchEntryInfo(buchname,buchnrtype);
-if (nr=="" && buchname=="") return g_txtUnbekannt;
+if (nr=="" && buchname=="") return g_sym0 + g_txtUnbekannt;
 if ("<? query telcfg:settings/UseClickToDial ?>" == "1") {
-if (nr == "") return span(buchentryinfo);
-if (buchname == "") {
-return "<nobr>" +
-"<a href=\"javascript:Dial('"+nr+"')\" title=\""+nr+"\">"+nr+"</a></nobr>";
-}
-return "<nobr><a href=\"javascript:Dial('"+nr+"')\" title=\""+buchentryinfo+"\">"+buchentryinfo+"</a></nobr>";
+if (nr == "") return g_sym0 + span(buchentryinfo);
+if (buchname == "") return "<nobr>" + uiRufnummerInfo (nr) +"<a href=\"javascript:Dial('"+nr+"')\" title=\""+nr+"\">"+nr+"</a></nobr>";
+return "<nobr>" +  g_sym0 + "<a href=\"javascript:Dial('"+nr+"')\" title=\""+buchentryinfo+"\">"+buchentryinfo+"</a></nobr>";
 } else {
-return span(buchname=="" ? nr:buchentryinfo);
+return (buchname=="" ? uiRufnummerInfo(nr):g_sym0) + span(buchname=="" ? nr:buchentryinfo);
+}
 }
+function uiRufnummerInfo (nr) {
+return "<a href=\"http://www.dasoertliche.de/Controller?form_name=search_inv&ph=" + encodeURI (nr) + "\" target=\"_blank\" title=\"R&uuml;ckw&auml;rtssuche bei dasoertliche.de nach "+nr+"\"><img src=\"../html/<? echo $var:lang ?>/images/bearbeiten.gif\"></a>";
 }
 function uiRouteDisplay (n, t) {
 var name;
