--- Scripts/pilconvert.py
+++ Scripts/pilconvert.py
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#! /usr/bin/env python
 #
 # The Python Imaging Library.
 # $Id$
--- Scripts/pilfile.py
+++ Scripts/pilfile.py
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#! /usr/bin/env python
 #
 # The Python Imaging Library.
 # $Id$
--- Scripts/pilfont.py
+++ Scripts/pilfont.py
@@ -1,3 +1,5 @@
+#! /usr/bin/env python
+
 #
 # The Python Imaging Library
 # $Id$
--- Scripts/pilprint.py
+++ Scripts/pilprint.py
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#! /usr/bin/env python
 #
 # The Python Imaging Library.
 # $Id$
