--- bfd/elf32-mips.c.orig	2005-12-20 23:43:55.000000000 +0100
+++ bfd/elf32-mips.c	2007-07-29 05:45:00.000000000 +0200
@@ -1614,7 +1614,7 @@
 
 /* The SVR4 MIPS ABI says that this should be 0x10000, and Linux uses
    page sizes of up to that limit, so we need to respect it.  */
-#define ELF_MAXPAGESIZE			0x10000
+#define ELF_MAXPAGESIZE			0x1000
 #define elf32_bed			elf32_tradbed
 
 /* Include the target file again for this target.  */
