--- configure
+++ configure
@@ -14967,8 +14967,8 @@
   shlibext="$acl_cv_shlibext"
   hardcode_libdir_flag_spec="$acl_cv_hardcode_libdir_flag_spec"
   hardcode_libdir_separator="$acl_cv_hardcode_libdir_separator"
-  hardcode_direct="$acl_cv_hardcode_direct"
-  hardcode_minus_L="$acl_cv_hardcode_minus_L"
+  hardcode_direct=no
+  hardcode_minus_L=no
     # Check whether --enable-rpath was given.
 if test "${enable_rpath+set}" = set; then :
   enableval=$enable_rpath; :
