
  https://github.com/shellinabox/shellinabox/pull/489/commits/15f9e6b54aa16f718e6e2f40026a1085d47ce33a

diff --git a/configure.ac b/configure.ac
index 05ab1bb..f7134e5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2,7 +2,7 @@ AC_PREREQ(2.57)
 
 dnl This is one of the locations where the authoritative version
 dnl  number is stored.  The other is in the debian/changelog.
-AC_INIT(shellinabox, 2.20, markus@shellinabox.com)
+AC_INIT(shellinabox, 2.21, markus@shellinabox.com)
 if test -e .git; then
 VCS_REVISION=" (revision `cd $srcdir && git log -1 --format=format:%h`)"
 else
