--- Makefile.am
+++ Makefile.am
@@ -126,7 +126,7 @@
 
 shellinaboxd_LDADD   = liblogging.la                                          \
                        libhttp.la
-shellinaboxd_LDFLAGS = -static
+shellinaboxd_LDFLAGS =
 ## Added this for compatibility with older versions of autoconf/automake
 docdir               = ${datadir}/doc/${PACKAGE}
 
--- Makefile.in
+++ Makefile.in
@@ -523,7 +523,7 @@
 shellinaboxd_LDADD = liblogging.la                                          \
                        libhttp.la
 
-shellinaboxd_LDFLAGS = -static
+shellinaboxd_LDFLAGS =
 symbolname = \
   sed -e 's/.*\/\([^.]*\)[.].*/\1/'                                           \
       -e 't0'                                                                 \
