--- linux-2.6.28/drivers/usb/serial/ftdi_sio.h
+++ linux-2.6.28/drivers/usb/serial/ftdi_sio.h
@@ -91,6 +91,15 @@
 #define FTDI_XF_635_PID 0xFC0D	/* 635: 20x4 Character Display */
 #define FTDI_XF_640_PID 0xFC0E	/* 640: Two line Display */
 #define FTDI_XF_642_PID 0xFC0F	/* 642: Two line Display */
+#define WBE_VID                      0x104F
+#define WBE_SMARTMOUSE_PID1   0x0001  /* Smartmouse USB */
+#define WBE_SMARTMOUSE_PID2   0x0002  /* Smartmouse USB */
+#define WBE_SMARTMOUSE_PID3   0x0003  /* Smartmouse USB */
+#define WBE_SMARTMOUSE_PID4   0x0004  /* Smartmouse USB */
+#define FTDI_8U232AM_PID1 0x6001 /* FTDI USB2Serial */
+#define FTDI_8U232AM_PID2 0x6002 /* FTDI USB2Serial */
+#define FTDI_8U232AM_PID3 0x6003 /* FTDI USB2Serial */
+#define FTDI_8U232AM_PID4 0x6004 /* FTDI USB2Serial */
 
 /* Video Networks Limited / Homechoice in the UK use an ftdi-based device for their 1Mb */
 /* broadband internet service.  The following PID is exhibited by the usb device supplied */
--- linux-2.6.28/drivers/usb/serial/ftdi_sio.c
+++ linux-2.6.28/drivers/usb/serial/ftdi_sio.c
@@ -666,6 +664,14 @@
 	{ USB_DEVICE(ADI_VID, ADI_GNICE_PID),
 		.driver_info = (kernel_ulong_t)&ftdi_jtag_quirk },
 	{ USB_DEVICE(JETI_VID, JETI_SPC1201_PID) },
+	{ USB_DEVICE(WBE_VID, WBE_SMARTMOUSE_PID1) },
+	{ USB_DEVICE(WBE_VID, WBE_SMARTMOUSE_PID2) },
+	{ USB_DEVICE(WBE_VID, WBE_SMARTMOUSE_PID3) },
+	{ USB_DEVICE(WBE_VID, WBE_SMARTMOUSE_PID4) },
+	{ USB_DEVICE(FTDI_VID, FTDI_8U232AM_PID1) },
+	{ USB_DEVICE(FTDI_VID, FTDI_8U232AM_PID2) },
+	{ USB_DEVICE(FTDI_VID, FTDI_8U232AM_PID3) },
+	{ USB_DEVICE(FTDI_VID, FTDI_8U232AM_PID4) },
 	{ },					/* Optional parameter entry */
 	{ }					/* Terminating entry */
 };
