--- linux-2.6.32/drivers/net/ifxmips_switch_api/GPHY/ifxmips_vr9_gphy.c
+++ linux-2.6.32/drivers/net/ifxmips_switch_api/GPHY/ifxmips_vr9_gphy.c
@@ -27,25 +27,25 @@
 #include <asm/io.h>
 #include <asm/irq.h>
 
-#include <asm/ifx/ifx_types.h>
-#include <asm/ifx/ifx_regs.h>
-#include <asm/ifx/common_routines.h>
-#include <asm/ifx/ifx_pmu.h>
-#include <asm/ifx/ifx_rcu.h>
-#include <asm/ifx/ifx_pmu.h>
-#include <asm/ifx/ifx_gpio.h>
-#include <ifx_ethsw_core_platform.h>
+#include <ifx_types.h>
+#include <ifx_regs.h>
+#include <common_routines.h>
+#include <ifx_pmu.h>
+#include <ifx_rcu.h>
+#include <ifx_pmu.h>
+#include <ifx_gpio.h>
+#include <switch_api/ifx_ethsw_core_platform.h>
 //#if ( defined(CONFIG_VR9) || defined(CONFIG_AR10) || defined(CONFIG_HN1) )
 #if ( defined(CONFIG_AR10) || defined(CONFIG_HN1) )
 #if defined(CONFIG_GE_MODE)
-	#include "gphy_fw_ge.h"
+	#include <switch_api/gphy_fw_ge.h>
 #endif 
 #if defined(CONFIG_FE_MODE)
-	#include "gphy_fw_fe.h"
+	#include <switch_api/gphy_fw_fe.h>
 #endif
 #endif /* defined(CONFIG_VR9) || defined(CONFIG_AR10) || defined(CONFIG_HN1) */
 #if  defined(CONFIG_AR10) 
-#include <asm/ifx/ar10/ar10.h>
+#include <ar10.h>
 #endif
 
 #if defined(CONFIG_GE_MODE)
