--- Makefile.am
+++ Makefile.am
@@ -31,7 +31,7 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = doc include m4 man src examples test build objs microbench
+SUBDIRS =     include        src               build objs
 
 EXTRA_DIST = \
 	COPYING.FDL \
--- Makefile.in
+++ Makefile.in
@@ -397,7 +397,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = doc include m4 man src examples test build objs microbench
+SUBDIRS =     include        src               build objs
 EXTRA_DIST = \
 	COPYING.FDL \
 	COPYING.GPL \
--- src/Makefile.am
+++ src/Makefile.am
@@ -21,7 +21,7 @@
 endif
 
 if FLaC__WITH_CPPLIBS
-CPPLIBS_DIRS = libFLAC++ test_libFLAC++
+CPPLIBS_DIRS = libFLAC++
 endif
 
 SUBDIRS = \
@@ -30,11 +30,6 @@
 	flac \
 	metaflac \
 	$(XMMS_DIRS) \
-	test_grabbag \
-	test_libs_common \
-	test_libFLAC \
-	test_seeking \
-	test_streams \
 	utils \
 	$(CPPLIBS_DIRS)
 
--- src/Makefile.in
+++ src/Makefile.in
@@ -363,18 +363,13 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 @FLaC__HAS_XMMS_TRUE@XMMS_DIRS = plugin_common plugin_xmms
-@FLaC__WITH_CPPLIBS_TRUE@CPPLIBS_DIRS = libFLAC++ test_libFLAC++
+@FLaC__WITH_CPPLIBS_TRUE@CPPLIBS_DIRS = libFLAC++
 SUBDIRS = \
 	libFLAC \
 	share \
 	flac \
 	metaflac \
 	$(XMMS_DIRS) \
-	test_grabbag \
-	test_libs_common \
-	test_libFLAC \
-	test_seeking \
-	test_streams \
 	utils \
 	$(CPPLIBS_DIRS)
 
