--- configure
+++ configure
@@ -20573,6 +20573,9 @@
 	CFLAGS="$CFLAGS -Wall -Wextra -Wstrict-prototypes -Wmissing-prototypes -Waggregate-return -Wcast-align -Wnested-externs -Wshadow -Wundef -Wmissing-declarations -Winline " # -Wcast-qual -Wbad-function-cast -Wwrite-strings -Wconversion
 	CXXFLAGS="$CXXFLAGS -Wall -Wextra -Wcast-align -Wshadow -Wwrite-strings -Wctor-dtor-privacy -Wnon-virtual-dtor -Wreorder -Wsign-promo -Wundef " # -Wcast-qual -Wbad-function-cast -Wwrite-strings -Woverloaded-virtual -Wmissing-declarations
 
+	CFLAGS="$CFLAGS -Wno-unused-function "
+	CXXFLAGS="$CXXFLAGS -Wno-unused-function "
+
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking if $CC accepts -Wdeclaration-after-statement" >&5
 $as_echo_n "checking if $CC accepts -Wdeclaration-after-statement... " >&6; }
 	ac_add_cflags__old_cflags="$CFLAGS"
