--- Makefile
+++ Makefile
@@ -190,9 +190,9 @@
 	@if [ "$(V)" = "0" ]; then \
 		echo " [LD] $@ ($^)";	\
 	else	\
-		echo "$(CC) -shared -Xlinker -x -o $@ $^ $(LIBLINUX)";	\
+		echo "$(CC) -shared -Xlinker -x -o $@ $^ $(LIBS) $(LIBLINUX)";	\
 	fi
-	@$(CC) -shared -Xlinker -x -o $@ $^ $(LIBLINUX)
+	@$(CC) -shared -Xlinker -x -o $@ $^ $(LIBS) $(LIBLINUX)
 
 install: all
 	$(INSTALL) -d -m 755 $(MODULES_DIR)
