--- configure
+++ configure
@@ -6665,7 +6665,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $found_b64_ntop" >&5
 $as_echo "$found_b64_ntop" >&6; }
 OLD_LIBS="$LIBS"
-if test "x$found_b64_ntop" = xno; then
+if test "x$found_b64_ntop" = xno_DISABLED; then
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for b64_ntop with -lresolv" >&5
 $as_echo_n "checking for b64_ntop with -lresolv... " >&6; }
 	LIBS="$OLD_LIBS -lresolv"
--- configure.ac
+++ configure.ac
@@ -431,7 +431,7 @@
 )
 AC_MSG_RESULT($found_b64_ntop)
 OLD_LIBS="$LIBS"
-if test "x$found_b64_ntop" = xno; then
+if test "x$found_b64_ntop" = xno_DISABLED; then
 	AC_MSG_CHECKING(for b64_ntop with -lresolv)
 	LIBS="$OLD_LIBS -lresolv"
 	AC_LINK_IFELSE([AC_LANG_PROGRAM(
