--- configure
+++ configure
@@ -5086,6 +5086,7 @@
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
 
+if false; then
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for b64_ntop with -lresolv" >&5
 $as_echo_n "checking for b64_ntop with -lresolv... " >&6; }
 	LIBS="$LIBS -lresolv"
@@ -5117,6 +5118,7 @@
 $as_echo "no" >&6; }
 	fi
 fi
+fi
 if test "x$found_b64_ntop" = xyes; then
 	$as_echo "#define HAVE_B64_NTOP 1" >>confdefs.h
 
