--- Makefile.in
+++ Makefile.in
@@ -88,7 +88,7 @@
 
 # Set flags for gcc. gcc4 whines abouts silly stuff so it needs slightly
 # different flags.
-@IS_GCC_TRUE@am__append_2 = -std=gnu99 -O2
+@IS_GCC_TRUE@am__append_2 = -std=gnu99
 @IS_DEBUG_TRUE@@IS_GCC_TRUE@am__append_3 = -g -Wno-long-long -Wall -W \
 @IS_DEBUG_TRUE@@IS_GCC_TRUE@	-Wnested-externs -Wformat=2 \
 @IS_DEBUG_TRUE@@IS_GCC_TRUE@	-Wmissing-prototypes \
