--- Makefile.in
+++ Makefile.in
@@ -72,7 +72,7 @@
 @IS_DEBUG_TRUE@@IS_GCC_TRUE@	-Winline -Wcast-align \
 @IS_DEBUG_TRUE@@IS_GCC_TRUE@	-Wdeclaration-after-statement
 @IS_DEBUG_TRUE@@IS_GCC_TRUE@am__append_4 = -DDEBUG
-@IS_DEBUG_FALSE@@IS_GCC_TRUE@am__append_5 = -O2
+@IS_DEBUG_FALSE@@IS_GCC_TRUE@am__append_5 =
 @IS_GCC4_TRUE@@IS_GCC_TRUE@am__append_6 = -iquote. -I/usr/local/include
 @IS_DEBUG_TRUE@@IS_GCC4_TRUE@@IS_GCC_TRUE@am__append_7 = -Wno-pointer-sign
 @IS_GCC4_FALSE@@IS_GCC_TRUE@am__append_8 = -I. -I- -I/usr/local/include
