--- linux-2.6.32/drivers/net/ifxmips_switch_api/ifx_ethsw_flow_api.c.orig	2012-10-15 11:54:01.000000000 +0200
+++ linux-2.6.32/drivers/net/ifxmips_switch_api/ifx_ethsw_flow_api.c	2013-04-06 17:28:28.685332427 +0200
@@ -17,6 +17,7 @@
 #ifdef IFX_ETHSW_API_COC
 	#include <switch_api/ifx_ethsw_pm.h>
 #endif
+#include <switch_api/ifx_ethsw_api.h>
 #define MULTI_PCE_RULES_INDEX	40
 #define FLOW_TABLE_DEBUG	0
 /********************/
