--- linux-2.6.32/drivers/net/avm_cpmac/common/avmnet_config.c.orig	2013-04-06 17:03:54.789398139 +0200
+++ linux-2.6.32/drivers/net/avm_cpmac/common/avmnet_config.c	2013-04-06 17:03:31.321399186 +0200
@@ -759,6 +759,7 @@
     }
 
     AVMNET_INFO("[%s] done\n", __func__ );
+    AVMNET_INFO("AVMNET: compiled with sizeof(struct sk_buff) = %d\n", sizeof(struct sk_buff));
     return 0;
 }
 arch_initcall(avmnet_cfg_init);
