--- linux-2.6.32/net/atm/pppoatm.c	2012-05-08 11:04:56.364466566 +0200
+++ linux-2.6.32/net/atm/pppoatm.c	2012-05-08 11:05:47.593087265 +0200
@@ -319,8 +319,6 @@ static int pppoatm_assign_vcc(struct atm
 	atmvcc->user_back = pvcc;
 	atmvcc->push = pppoatm_push;
 	atmvcc->pop = pppoatm_pop;
-    if ( ppa_hook_mpoa_setup )
-        ppa_hook_mpoa_setup(atmvcc, 2, pvcc->encaps == e_llc ? 1 : 0);  //  PPPoA
 	__module_get(THIS_MODULE);
 #if defined(CONFIG_IFX_PPA_A6) || defined(CONFIG_IFX_PPA_A5) || defined(CONFIG_IFX_PPA_A4)
 	if ( ppa_hook_mpoa_setup )
