--- linux-2.6.32/drivers/net/avm_cpmac/Makefile
+++ linux-2.6.32/drivers/net/avm_cpmac/Makefile
@@ -12,7 +12,7 @@
 #EXTRA_CFLAGS += -Wpadded
 #EXTRA_CFLAGS += -Wunreachable-code
 EXTRA_CFLAGS += -Wall -Wextra -Werror
-EXTRA_CFLAGS += -DAVM_CPMAC_VERSION="\"`svn info $(src) | grep URL | sed -e "s/^.*\/\([0-9.]\+\)-[^\/]*/\1/g"`  -  Revision `svnversion -cn $(src)`  -  `date +\"%a  %e.%m.%Y  %H:%M:%S\"` \""
+EXTRA_CFLAGS += -DAVM_CPMAC_VERSION="\"1.108.1.101 - Revision 811:1778 - Fr 13.04.2012 13:09:32\""
 
 # declare multi-module main driver.
 obj-$(CONFIG_AVM_CPMAC)	+= avm_cpmac.o
