--- configure.in
+++ configure.in
@@ -903,6 +903,7 @@
      __ptr	dnl
      bufpos	dnl
      _p	dnl
+     __bufpos	dnl
 ; do
     AC_TRY_COMPILE([#include <stdio.h>
 ],
@@ -922,6 +923,7 @@
     [for frend in	dnl
          _IO_read_end	dnl
          bufread 	dnl
+         __bufread 	dnl
     ; do
         AC_TRY_COMPILE([#include <stdio.h>
     ],
--- configure
+++ configure
@@ -9822,7 +9822,7 @@
 if ${rb_cv_frptr+:} false; then :
   $as_echo_n "(cached) " >&6
 else
-  for frptr in	     _IO_read_ptr      _ptr	     __ptr	     bufpos	     _p	; do
+  for frptr in	     _IO_read_ptr      _ptr	     __ptr	     bufpos	     _p	     __bufpos	; do
     cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <stdio.h>
@@ -9861,7 +9861,7 @@
     if ${rb_cv_frend+:} false; then :
   $as_echo_n "(cached) " >&6
 else
-  for frend in	         _IO_read_end	         bufread 	    ; do
+  for frend in	         _IO_read_end	         bufread 	         __bufread	    ; do
         cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <stdio.h>
