--- configure.orig	2012-01-24 21:13:55.000000000 +0100
+++ configure	2012-01-24 21:14:06.609734925 +0100
@@ -31823,8 +31823,6 @@
                         CPPFLAGS="-I${bdbdir}/include${subdir} $CPPFLAGS"
                         LDFLAGS="-L$bdblibdir $LDFLAGS"
 
-                                                                                                                                                                        eval export $shlibpath_var=$bdblibdir
-
     atalk_cv_bdb_version=no
     maj=$atalk_cv_bdb_MAJOR
     min=$atalk_cv_bdb_MINOR
