--- etc/init.d/S08-tffs.orig	2010-09-15 13:39:31.000000000 +0200
+++ etc/init.d/S08-tffs	2010-09-28 01:56:31.089189998 +0200
@@ -109,6 +109,10 @@
 tffs_nodes_list="$tffs_nodes_list $((0xC9)),websrv_ssl_key.pem"
 tffs_nodes_list="$tffs_nodes_list $((0xCA)),websrv_ssl_cert.pem"
 #########################################################################
+## Freetz
+#########################################################################
+tffs_nodes_list="$tffs_nodes_list $((0x3C)),freetz"
+#########################################################################
 ## NODES anlegen, wenn nicht als _file_ existent
 #########################################################################
 for node in ${tffs_nodes_list} ; do
