--- usr/www/all/html/de/home/home.js.orig	2009-11-12 13:13:06.000000000 +0100
+++ usr/www/all/html/de/home/home.js	2009-11-12 18:59:47.675697638 +0100
@@ -799,7 +799,7 @@
 var nPartTotal = Number(g_usbhost_data.part_count_total);
 if (isNaN(nPartTotal)) nPartTotal = 0;
 var unplugOk = parseInt(g_usbhost_data.unplug_ok, 10) == 1;
-if (nPartTotal > nStor)
+if (nPartTotal > 0) 
 {
 if (unplugOk)
 {
@@ -807,9 +807,7 @@
 }
 else
 {
-strStor = "{?de.home.home.js:5308?} ";
-strStor += "<a href=\"javascript:jslGoTo('home', 'syslog');\">{?de.home.home.js:3744?}</a>";
-strStor = strStor.fontcolor("#cc0000");
+strStor = "Es wurden " + nStor + " von " + nPartTotal + " Partitionen eingebunden.";
 }
 strArr.push(strStor);
 }
