--- extensions/Makefile.orig	2006-12-04 12:15:19.000000000 +0100
+++ extensions/Makefile	2007-07-28 12:45:28.000000000 +0200
@@ -13,6 +13,10 @@
 PF6_EXT_SE_SLIB:=SECMARK CONNSECMARK
 endif
 
+
+# ipp2p
+PF_EXT_SLIB += ipp2p
+
 # Optionals
 PF_EXT_SLIB_OPTS:=$(foreach T,$(wildcard extensions/.*-test),$(shell KERNEL_DIR=$(KERNEL_DIR) $(T)))
 PF6_EXT_SLIB_OPTS:=$(foreach T,$(wildcard extensions/.*-test6),$(shell KERNEL_DIR=$(KERNEL_DIR) $(T)))
