--- usr/www/all/html/de/fon/sperre1.js.orig	2008-05-07 13:17:20.000000000 +0200
+++ usr/www/all/html/de/fon/sperre1.js	2008-05-08 22:01:59.000000000 +0200
@@ -58,7 +58,7 @@
 switch (mode) {
 case "In": {
 var nummer = jslGetValue("uiViewNummer");
-if (!isZahlRaute(nummer)) {alert(g_mldNonDigits); return;}
+//if (!isZahlRaute(nummer)) {alert(g_mldNonDigits); return;}
 var x = IstActionVorhanden(nummer);
 if (x != "") {
 if (jslGetValue("uiPostInAction"+x) == "1") {
@@ -151,7 +151,7 @@
 }
 } else {
 var nummer = jslGetValue("uiViewNummer");
-if (!isZahlRaute(nummer)) {alert(g_mldNonDigits); return;}
+//if (!isZahlRaute(nummer)) {alert(g_mldNonDigits); return;}
 var x = IstActionVorhanden(nummer);
 if (x != "") {
 if (jslGetValue("uiPostInAction"+x) == "1") {alert(g_txtAlert_1); return;}
