--- usr/www/all/html/de/home/home.js	Fri Feb 27 16:14:38 2009
+++ usr/www/all/html/de/home/home.js	Fri Feb 27 17:59:49 2009
@@ -774,15 +774,13 @@
 var nPartTotal = Number("<? query ctlusb:settings/storage-part-total ?>");
 if (isNaN(nPartTotal)) nPartTotal = 0;
 var unplugOk = parseInt("<? query ctlusb:settings/unplug-ok ?>", 10) == 1;
-if (nPartTotal > nStor) {
+if (nPartTotal > 0) {
 var strStor = "";
 if (unplugOk) {
 strStor = "{?de.home.home.js:1071?}";
 }
 else {
-strStor = "{?de.home.home.js:5308?} ";
-strStor += "<a href=\"javascript:jslGoTo('home', 'syslog');\">{?de.home.home.js:3744?}</a>";
-strStor = strStor.fontcolor("#cc0000");
+strStor = "Es wurden " + nStor + " von " + nPartTotal + " Partitionen eingebunden.";
 }
 strArr.push(strStor);
 }
