--- usr/www/all/html/de/home/home.html
+++ usr/www/all/html/de/home/home.html
@@ -166,6 +166,11 @@
 <td class="tdLed">&nbsp;</td>
 <td><script type="text/javascript">document.write(WeckerDisplay(2));</script></td>
 </tr>
+<tr id="trWecker3" style="display: none">
+<td class="tdName"><a href="javascript:jslGoToWecker(2)">Weckruf 3 </a></td>
+<td class="tdLed">&nbsp;</td>
+<td><script type="text/javascript">document.write(WeckerDisplay(3));</script></td>
+</tr>
 </table>
 </div></div></div></div></div></div>
 <? include ../html/de/home/rbb_refresh.html ?>
--- usr/www/all/html/de/home/home.js
+++ usr/www/all/html/de/home/home.js
@@ -224,6 +224,7 @@
 var NachtEnabled = "<? query box:settings/night_time_control_enabled ?>";
 var Wecker1 = "<? query telcfg:settings/AlarmClock0/Active ?>";
 var Wecker2 = "<? query telcfg:settings/AlarmClock1/Active ?>";
+var Wecker3 = "<? query telcfg:settings/AlarmClock2/Active ?>";
 <? if eq $var:showPortfreigabe 1 `
 var showPortfreigabeInfo = "0";
 <? multiquery forwardrules:settings/rule/list(activated) `
@@ -299,7 +300,7 @@
 <? if eq $var:showPortfreigabe 1 `
 || showPortfreigabeInfo == "1"
 ` ?>
-|| g_callthroughActive == true || g_tamActive > 0 || g_tamDisplay > 0 || g_intFaxActive == "1" || Wecker1 == "1" || Wecker2 == "1" || RufsperreAktiv == true || g_RufumleitungAktiv == true
+|| g_callthroughActive == true || g_tamActive > 0 || g_tamDisplay > 0 || g_intFaxActive == "1" || Wecker1 == "1" || Wecker2 == "1" || Wecker3 == "1" || RufsperreAktiv == true || g_RufumleitungAktiv == true
 || showRemoteHttps == true
 || showDynDns == true
 || showOnlineCnt == "1"
@@ -309,6 +310,7 @@
 }
 jslDisplay("trWecker1",Wecker1 == "1");
 jslDisplay("trWecker2",Wecker2 == "1");
+jslDisplay("trWecker3",Wecker3 == "1");
 jslDisplay("trTam",g_tamActive > 0 || g_tamDisplay > 0);
 jslDisplay("trIntFax", g_intFaxActive == "1");
 jslDisplay("trSperre",RufsperreAktiv == true);
@@ -1357,8 +1359,10 @@
 }
 return DisplayString;
 }
+
 function WeckerDisplay(Wecker)
 {
+
 <? setvariable var:wnr '0' ?>
 var Active = "<? query telcfg:settings/AlarmClock<? echo $var:wnr ?>/Active ?>";
 var Time = "<? query telcfg:settings/AlarmClock<? echo $var:wnr ?>/Time ?>";
@@ -1368,13 +1372,29 @@
 {
 return WeckerString(Active,Time,Num,Weekdays);
 }
+
 <? setvariable var:wnr '1' ?>
 Active = "<? query telcfg:settings/AlarmClock<? echo $var:wnr ?>/Active ?>";
 Time = "<? query telcfg:settings/AlarmClock<? echo $var:wnr ?>/Time ?>";
 Num = "<? query telcfg:settings/AlarmClock<? echo $var:wnr ?>/Number ?>";
 Weekdays = "<? query telcfg:settings/AlarmClock<? echo $var:wnr ?>/Weekdays ?>";
+if (Wecker == 2)
+{
 return WeckerString(Active,Time,Num,Weekdays);
 }
+
+<? setvariable var:wnr '2' ?>
+Active = "<? query telcfg:settings/AlarmClock<? echo $var:wnr ?>/Active ?>";
+Time = "<? query telcfg:settings/AlarmClock<? echo $var:wnr ?>/Time ?>";
+Num = "<? query telcfg:settings/AlarmClock<? echo $var:wnr ?>/Number ?>";
+Weekdays = "<? query telcfg:settings/AlarmClock<? echo $var:wnr ?>/Weekdays ?>";
+if (Wecker == 3)
+{
+return WeckerString(Active,Time,Num,Weekdays);
+}
+
+}
+
 <? if eq $var:showPortfreigabe `1` `
 function PortfwDisplay(){
 var infotext="";
@@ -1593,6 +1613,8 @@
 jslSetValue("uiPostWeckernr",0);break;
 case 1:
 jslSetValue("uiPostWeckernr",1);break;
+case 2:
+jslSetValue("uiPostWeckernr",2);break;
 }
 jslGoTo('fon','wecker');
 }
--- usr/www/all/html/de/fon/wecker.html
+++ usr/www/all/html/de/fon/wecker.html
@@ -3,14 +3,25 @@
 {?txtWakeupCall?}
 </div></div></div></div></div></div>
 <div class="backframe"><div class="ecklm"><div class="eckrm"><div class="rundrb"><div class="rundlb"><div class="foredialog"><div class="ie_b10">
+
 <ul class="tabs" id="uiView0">
 <li class="tabs_on"><div class="activtab">{?388:2379?}</div></li>
 <li><a href="javascript:DoTabs(1)">{?388:1017?}</a></li>
+<li><a href="javascript:DoTabs(2)">Weckruf 3</a></li>
 </ul>
+
 <ul class="tabs" id="uiView1" style="display:none">
 <li><a href="javascript:DoTabs(0)">{?388:6066?}</a></li>
 <li class="tabs_on"><div class="activtab">{?388:6416?}</div></li>
+<li><a href="javascript:DoTabs(2)">Weckruf 3</a></li>
 </ul>
+
+<ul class="tabs" id="uiView2" style="display:none">
+<li><a href="javascript:DoTabs(0)">{?388:6066?}</a></li>
+<li><a href="javascript:DoTabs(1)">{?388:1017?}</a></li>
+<li class="tabs_on"><div class="activtab">Weckruf 3</div></li>
+</ul>
+
 <div class="backdialog"><div class="ecklm"><div class="eckrm"><div class="ecklb"><div class="eckrb"><div class="foredialog">
 <p class="mb10">{?388:3268?}</p>
 <p class="mb10">{?388:2396?}</p>
--- usr/www/all/html/de/fon/wecker.js
+++ usr/www/all/html/de/fon/wecker.js
@@ -24,8 +24,9 @@
 var mCount = "<? query telcfg:settings/Foncontrol ?>";
 ` ?>
 var nr = jslGetValue("uiPostWeckernr");
-jslDisplay("uiView0", nr != "1");
+jslDisplay("uiView0", nr == "0");
 jslDisplay("uiView1", nr == "1");
+jslDisplay("uiView2", nr == "2");
 jslSetChecked("uiViewActive", jslGetValue("uiPostActive") == "1");
 var zeit = jslGetValue("uiPostZeit");
 if (zeit.length == 4) {
