--- linux-2.6.13.1/drivers/mtd/maps/ohio-flash.c	2007-07-09 01:22:13.000000000 +0200
+++ linux-2.6.13.1/drivers/mtd/maps/ohio-flash.c	2007-07-09 01:22:42.000000000 +0200
@@ -425,16 +425,20 @@
                     partition_info[i].size   = hidden_size;
                     partition_info[i].offset = hidden_start;
                 }
+/*
                 p = prom_getenv("nfsroot");
                 if(p) {
                     root_dev_setup(p);
                 } else {
+*/
                     int mtd_index = i;
                     if (mymtd->type != MTD_SPIFLASH)
                         mtd_index += 1;         /*--- beim SPI-Flash gibt es eine Partition weniger (phys_mapped flash) ---*/
                     sprintf(buff, "/dev/mtdblock%u", mtd_index);
                     root_dev_setup(buff);
+/*
                 }
+*/
             }
         }
         if(jffs2_size == 0)
