--- src/cjson.c
+++ src/cjson.c
@@ -350,9 +350,9 @@
         return NULL;
     }
 
-    if (needed > LLONG_MAX)
+    if (needed > SIZE_MAX)
     {
-        /* sizes bigger than LLONG_MAX are currently not supported */
+        /* sizes bigger than SIZE_MAX are currently not supported */
         return NULL;
     }
 
@@ -367,12 +367,12 @@
     }
 
     /* calculate new buffer size */
-    if (needed > (LLONG_MAX / 2))
+    if (needed > (SIZE_MAX / 2))
     {
-        /* overflow of int, use LLONG_MAX if possible */
-        if (needed <= LLONG_MAX)
+        /* overflow of size_t, use SIZE_MAX if possible */
+        if (needed <= SIZE_MAX)
         {
-            newsize = LLONG_MAX;
+            newsize = SIZE_MAX;
         }
         else
         {
