--- configure
+++ configure
@@ -27606,6 +27606,8 @@
     as_fn_error $? "Could not find a glib-genmarshal in your PATH" "$LINENO" 5
   fi
 
+GLIB_COMPILE_SCHEMAS=false
+if false; then
   # Extract the first word of "glib-compile-schemas", so it can be a program name with args.
 set dummy glib-compile-schemas; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
@@ -27650,7 +27652,10 @@
   if test x$GLIB_COMPILE_SCHEMAS = xno; then
     as_fn_error $? "Could not find a glib-compile-schemas in your PATH" "$LINENO" 5
   fi
+fi
 
+GLIB_COMPILE_RESOURCES=false
+if false; then
   # Extract the first word of "glib-compile-resources", so it can be a program name with args.
 set dummy glib-compile-resources; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
@@ -27696,6 +27701,7 @@
     as_fn_error $? "Could not find a glib-compile-resources in your PATH" "$LINENO" 5
   fi
 fi
+fi
 
 
 
