--- configure
+++ configure
@@ -600,19 +600,13 @@
 # Factoring default headers for most tests.
 ac_includes_default="\
 #include <stdio.h>
-#ifdef HAVE_SYS_TYPES_H
 # include <sys/types.h>
-#endif
-#ifdef HAVE_SYS_STAT_H
 # include <sys/stat.h>
-#endif
 #ifdef STDC_HEADERS
 # include <stdlib.h>
 # include <stddef.h>
 #else
-# ifdef HAVE_STDLIB_H
 #  include <stdlib.h>
-# endif
 #endif
 #ifdef HAVE_STRING_H
 # if !defined STDC_HEADERS && defined HAVE_MEMORY_H
@@ -623,15 +617,10 @@
 #ifdef HAVE_STRINGS_H
 # include <strings.h>
 #endif
-#ifdef HAVE_INTTYPES_H
 # include <inttypes.h>
-#endif
-#ifdef HAVE_STDINT_H
 # include <stdint.h>
-#endif
-#ifdef HAVE_UNISTD_H
 # include <unistd.h>
-#endif"
+"
 
 ac_subst_vars='am__EXEEXT_FALSE
 am__EXEEXT_TRUE
