--- configure
+++ configure
@@ -28580,6 +28580,7 @@
   $as_echo_n "(cached) " >&6
 else
   if ac_fn_c_compute_int "$LINENO" "(long int) offsetof (ac__type_alignof_, y)" "ac_cv_alignof_guint32"        "$ac_includes_default
+#include <stddef.h>
 typedef unsigned $gint32 guint32;
 
 #ifndef offsetof
@@ -28623,6 +28624,7 @@
   $as_echo_n "(cached) " >&6
 else
   if ac_fn_c_compute_int "$LINENO" "(long int) offsetof (ac__type_alignof_, y)" "ac_cv_alignof_guint64"        "$ac_includes_default
+#include <stddef.h>
 typedef unsigned $gint64 guint64;
 
 #ifndef offsetof
@@ -28665,6 +28667,7 @@
   $as_echo_n "(cached) " >&6
 else
   if ac_fn_c_compute_int "$LINENO" "(long int) offsetof (ac__type_alignof_, y)" "ac_cv_alignof_unsigned_long"        "$ac_includes_default
+#include <stddef.h>
 #ifndef offsetof
 # define offsetof(type, member) ((char *) &((type *) 0)->member - (char *) 0)
 #endif
