--- modcgi.c.orig	2010-08-27 15:21:46.000000000 +0200
+++ modcgi.c	2010-08-27 15:23:52.000000000 +0200
@@ -84,7 +84,7 @@
     }
 
     if ((ts = fread (result, sizeof (char), length, stdin)) < length) {
-      fprintf (stderr, "error reading post data, %d bytes read, %d expedted\n",
+      fprintf (stderr, "error reading post data, %d bytes read, %d expected\n",
 	       ts, length);
     }
     result[length] = '\0';
