--- configure
+++ configure
@@ -6108,6 +6108,7 @@
 
 fi
 
+if false; then
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for long long off_t" >&5
 $as_echo_n "checking for long long off_t... " >&6; }
 if test "${ac_cv_have_long_long_off_t+set}" = set; then :
@@ -6167,6 +6168,7 @@
 $as_echo "#define HAVE_LONG_LONG_RLIM_T 1" >>confdefs.h
 
 fi
+fi
 
 ac_fn_c_check_type "$LINENO" "struct opthdr" "ac_cv_type_struct_opthdr" "#include <sys/socket.h>
 "
