diff --git a/acinclude.m4 b/acinclude.m4
index 1741ab9..708f8cc 100644
--- acinclude.m4
+++ acinclude.m4
@@ -287,8 +287,8 @@
 		return 0;
 	return 1;
 }
-]])],[ac_cv_have_little_endian_long_long=yes],[ac_cv_have_little_endian_long_long=no],[# Should try to guess here
-ac_cv_have_little_endian_long_long=no
+]])],[ac_cv_have_little_endian_long_long=yes],[ac_cv_have_little_endian_long_long=no],[
+ac_cv_have_little_endian_long_long=$ac_cv_c_littleendian
 ])])
 AC_MSG_RESULT($ac_cv_have_little_endian_long_long)
 if test "$ac_cv_have_little_endian_long_long" = yes
diff --git a/configure.ac b/configure.ac
index 96a9a66..bc4dbb1 100644
--- configure.ac
+++ configure.ac
@@ -185,6 +185,7 @@ AC_PROG_CPP
 AC_PROG_GCC_TRADITIONAL
 AC_PROG_INSTALL
 AC_C_CONST
+AC_C_BIGENDIAN
 AC_HEADER_STDC
 AC_HEADER_STDBOOL
 AC_HEADER_DIRENT
