--- configure
+++ configure
@@ -30717,24 +30717,6 @@
   $ac_cs_success || { (exit 1); exit 1; }
 fi
 
-
-{ echo "$as_me:$LINENO: checking in" >&5
-echo $ECHO_N "checking in... $ECHO_C" >&6; }
-{ echo "$as_me:$LINENO: result: and out again" >&5
-echo "${ECHO_T}and out again" >&6; }
-
-echo $ECHO_N "ordering CD from http://tobi.oetiker.ch/wish $ECHO_C" 1>&6
-sleep 1
-echo $ECHO_N ".$ECHO_C" 1>&6
-sleep 1
-echo $ECHO_N ".$ECHO_C" 1>&6
-sleep 1
-echo $ECHO_N ".$ECHO_C" 1>&6
-sleep 1
-echo $ECHO_N ".$ECHO_C" 1>&6
-sleep 1
-{ echo "$as_me:$LINENO: result:  just kidding ;-)" >&5
-echo "${ECHO_T} just kidding ;-)" >&6; }
 echo
 echo "----------------------------------------------------------------"
 echo "Config is DONE!"
