--- src/ngircd/ngircd.c.orig	2010-11-14 09:50:47.000000000 +0100
+++ src/ngircd/ngircd.c		2010-11-14 09:51:05.000000000 +0100
@@ -102,7 +102,7 @@
 #ifdef SNIFFER
 	NGIRCd_Sniffer = false;
 #endif
-	strlcpy( NGIRCd_ConfFile, SYSCONFDIR, sizeof( NGIRCd_ConfFile ));
+	/* strlcpy( NGIRCd_ConfFile, SYSCONFDIR, sizeof( NGIRCd_ConfFile )); */
 	strlcat( NGIRCd_ConfFile, CONFIG_FILE, sizeof( NGIRCd_ConfFile ));
 
 	Fill_Version( );
