--- src/ngircd/conf.c.orig	2010-11-14 09:53:58.000000000 +0100
+++ src/ngircd/conf.c		2010-11-14 09:54:16.000000000 +0100
@@ -572,7 +572,7 @@
 	strcpy(Conf_ServerAdmin2, "");
 	strcpy(Conf_ServerAdminMail, "");
 
-	strlcpy(Conf_MotdFile, SYSCONFDIR, sizeof(Conf_MotdFile));
+	/* strlcpy(Conf_MotdFile, SYSCONFDIR, sizeof(Conf_MotdFile)); */
 	strlcat(Conf_MotdFile, MOTD_FILE, sizeof(Conf_MotdFile));
 
 	Conf_UID = Conf_GID = 0;
