diff -burN rcapid.orig/configure rcapid/configure
--- configure	2000-02-04 16:21:46.000000000 +0100
+++ configure	2007-07-01 01:51:54.000000000 +0200
@@ -1402,7 +1402,7 @@
 	if test "$DOTEST" = "y" || test "$CONFIG_KERNELDIR" != "" ; then
 		echo $ac_n "checking for linux kernel source in ${tst_kerneldir}""... $ac_c" 1>&6
 echo "configure:1405: checking for linux kernel source in ${tst_kerneldir}" >&5
-		CPPFLAGS="-nostdinc -I${tst_kerneldir}/include -I${tst_kerneldir}/drivers/isdn"
+		CPPFLAGS="-nostdinc -I${tst_kerneldir}/include -I${tst_kerneldir}/drivers/isdn/i4l"
 		cat > conftest.$ac_ext <<EOF
 #line 1408 "configure"
 #include "confdefs.h"
