diff -burN rcapid.orig/Makefile.in rcapid/Makefile.in
--- rcapid.orig/Makefile.in	2000-02-11 11:50:11.000000000 +0100
+++ rcapid/Makefile.in	2007-07-01 02:11:14.000000000 +0200
@@ -72,9 +72,9 @@
 MAINTAINERCLEANFILES    = configure aclocal.m4 Makefile.in config.h.in \
 			  stamp-h.in comperr
 
-INCLUDES = -I../capi20 -I$(CONFIG_KERNELDIR)/drivers/isdn/avmb1 $(all_includes)
-CFLAGS = -Wall -O3
-LDFLAGS = -L../capi20/.libs -L../capi20 $(all_libraries)
+INCLUDES = -I$(CONFIG_KERNELDIR)/drivers/isdn/avmb1 $(all_includes)
+CFLAGS = @CFLAGS@
+LDFLAGS = $(all_libraries)
 LDADD = -lcapi20
 
 sbin_PROGRAMS = rcapid
