--- usr/www/all/html/de/home/home.js.orig	2009-10-16 14:23:49.000000000 +0200
+++ usr/www/all/html/de/home/home.js	2009-10-17 09:55:05.883422392 +0200
@@ -806,7 +806,7 @@
 var nPartTotal = Number(g_usbhost_data.part_count_total);
 if (isNaN(nPartTotal)) nPartTotal = 0;
 var unplugOk = parseInt(g_usbhost_data.unplug_ok, 10) == 1;
-if (nPartTotal > nStor)
+if (nPartTotal > 0) 
 {
 if (unplugOk)
 {
@@ -814,9 +814,7 @@
 }
 else
 {
-strStor = "{?de.home.home.js:5308?} ";
-strStor += "<a href=\"javascript:jslGoTo('home', 'syslog');\">{?de.home.home.js:3744?}</a>";
-strStor = strStor.fontcolor("#cc0000");
+strStor = "Es wurden " + nStor + " von " + nPartTotal + " Partitionen eingebunden.";
 }
 strArr.push(strStor);
 }
