--- linux-2.6.32/drivers/block/compcache/Makefile	2012-05-07 23:30:53+02:00
+++ linux-2.6.32/drivers/block/compcache/Makefile	2012-05-07 23:30:53+02:00
@@ -6,7 +6,7 @@
 ramzswap-objs	:=	ramzswap_drv.o xvmalloc.o
 
 ifneq ($(CONFIG_COMPCACHE),n)
-extra-y			+= rzscontrol ramzswap.sh
+#extra-y			+= rzscontrol ramzswap.sh
 
 rzscontrol-objs	:= rzscontrol.o
 
--- linux-2.6.32/drivers/rtc/Makefile	2012-05-07 23:30:53+02:00
+++ linux-2.6.32/drivers/rtc/Makefile	2012-05-07 23:30:53+02:00
@@ -91,7 +91,7 @@
 ifeq ($(CONFIG_RTC_DRV_AVM_REF_CLOCK),m)
 CFLAGS_rtc-avm.o += -DRTC_AVM_DEVICE_COUNT=1 # -DRTC_AVM_DEBUG=1
 
-extra-y			 += S09-rtc
+#extra-y			 += S09-rtc
 
 $(src)/S09-rtc: FORCE
 	rm -rf $(src)/S09-rtc

