--- configure
+++ configure
@@ -16806,6 +16806,7 @@
 if test "x$ac_cv_func_gethostbyname" = x""yes; then :
 
 else
+if false; then
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for gethostbyname in -lnsl" >&5
 $as_echo_n "checking for gethostbyname in -lnsl... " >&6; }
 if test "${ac_cv_lib_nsl_gethostbyname+set}" = set; then :
@@ -16845,6 +16846,7 @@
 if test "x$ac_cv_lib_nsl_gethostbyname" = x""yes; then :
   LIBNSL="-lnsl"
 fi
+fi
 
 fi
 
