--- configure.ac.orig	2010-09-28 14:24:16.000000000 +0200
+++ configure.ac	2011-04-26 17:53:26.431300280 +0200
@@ -303,7 +303,7 @@
 dnl *************************************************************
 dnl Check for headers
 dnl *************************************************************
-AC_CHECK_HEADERS([arpa/inet.h fcntl.h libintl.h limits.h \
+AC_CHECK_HEADERS([arpa/inet.h fcntl.h limits.h \
                  malloc.h memory.h netdb.h netinet/in.h paths.h \
                  stdlib.h string.h sys/file.h sys/ioctl.h sys/mount.h \
                  sys/param.h sys/socket.h sys/time.h sys/vfs.h \
--- configure.orig	2010-09-28 14:32:34.000000000 +0200
+++ configure	2011-04-26 17:53:26.441300281 +0200
@@ -18184,7 +18184,7 @@
 
 
 
-for ac_header in arpa/inet.h fcntl.h libintl.h limits.h \
+for ac_header in arpa/inet.h fcntl.h limits.h \
                  malloc.h memory.h netdb.h netinet/in.h paths.h \
                  stdlib.h string.h sys/file.h sys/ioctl.h sys/mount.h \
                  sys/param.h sys/socket.h sys/time.h sys/vfs.h \
