--- configure
+++ configure
@@ -12038,6 +12038,7 @@
 
 
 
+if false; then
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking dynamic linker characteristics" >&5
 $as_echo_n "checking dynamic linker characteristics... " >&6; }
 
@@ -12700,6 +12701,7 @@
 if test "$GCC" = yes; then
   variables_saved_for_relink="$variables_saved_for_relink GCC_EXEC_PREFIX COMPILER_PATH LIBRARY_PATH"
 fi
+fi
 
 if test "${lt_cv_sys_lib_search_path_spec+set}" = set; then
   sys_lib_search_path_spec="$lt_cv_sys_lib_search_path_spec"
@@ -15546,6 +15548,7 @@
 
 
 
+if false; then
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking dynamic linker characteristics" >&5
 $as_echo_n "checking dynamic linker characteristics... " >&6; }
 
@@ -16236,6 +16239,7 @@
   # Fast installation is not necessary
   enable_fast_install=needless
 fi
+fi
 
 
 
