--- Makefile.in
+++ Makefile.in
@@ -418,7 +418,7 @@
 @BUILD_DOC_FALSE@doc = 
 @BUILD_DOC_TRUE@doc = doc
 DIST_SUBDIRS = m4 compat mpi cipher random src doc tests
-SUBDIRS = compat mpi cipher random src $(doc) tests
+SUBDIRS = compat mpi cipher random src
 EXTRA_DIST = autogen.sh autogen.rc README.GIT LICENSES      	           \
              ChangeLog-2011 build-aux/ChangeLog-2011 doc/ChangeLog-2011    \
              m4/ChangeLog-2011 cipher/ChangeLog-2011 src/ChangeLog-2011    \
