--- Makefile.in
+++ Makefile.in
@@ -313,7 +313,7 @@
 # (A suitable gitlog-to-changelog script can be found in GnuPG master.)
 GITLOG_TO_CHANGELOG = gitlog-to-changelog
 DIST_SUBDIRS = m4 compat mpi cipher random src doc tests
-SUBDIRS = compat mpi cipher random src doc tests
+SUBDIRS = compat mpi cipher random src
 EXTRA_DIST = autogen.sh README.GIT \
 	     scripts/ChangeLog-2011 doc/ChangeLog-2011 ChangeLog-2011 \
              m4/ChangeLog-2011 cipher/ChangeLog-2011 src/ChangeLog-2011 \
