--- openbsd-compat/bsd-adjfreq.c
+++ openbsd-compat/bsd-adjfreq.c
@@ -21,7 +21,7 @@
 
 #if (defined(__linux__) || defined(__FreeBSD__) || defined(__DragonFly__) \
 	|| defined(__sun__)) \
-	&& !defined(HAVE_ADJFREQ) && defined(HAVE_NTP_ADJTIME)
+	&& !defined(HAVE_ADJFREQ) && defined(HAVE_ADJTIMEX)
 
 #include <sys/types.h>
 #include <unistd.h>
@@ -53,7 +53,7 @@
 #endif
 		txc.freq = *freq / 1e3 / (1LL << 16);
 
-		if ((ntp_adjtime(&txc)) == -1)
+		if ((adjtimex(&txc)) == -1)
 			log_warn("ntp_adjtime (2) failed");
 
 		log_debug("ntp_adjtime adjusted frequency by %fppm",
@@ -61,7 +61,7 @@
 	}
 	if (oldfreq != NULL) {
 		txc.modes = 0;
-		if ((ntp_adjtime(&txc)) == -1) {
+		if ((adjtimex(&txc)) == -1) {
 			log_warn("ntp_adjtime (1) failed");
 			return -1;
 		}
