--- ntp.c.orig	2011-07-03 23:51:07.000000000 +0200
+++ ntp.c	2011-07-31 21:02:39.000000000 +0200
@@ -116,6 +116,13 @@
 
 	if (socketpair(AF_UNIX, SOCK_STREAM, PF_UNSPEC, pipe_dns) == -1)
 		fatal("socketpair");
+
+	if (!nconf->debug) {
+		dup2(nullfd, STDIN_FILENO);
+		dup2(nullfd, STDOUT_FILENO);
+		dup2(nullfd, STDERR_FILENO);
+	}
+	close(nullfd);
 	close(pipe_prnt[0]);  /* close prior creating child process! */
 	dns_pid = ntp_dns(pipe_dns, nconf, pw);
 	close(pipe_dns[1]);
@@ -135,12 +142,7 @@
 	if (chdir("/") == -1)
 		fatal("chdir(\"/\")");
 
-	if (!nconf->debug) {
-		dup2(nullfd, STDIN_FILENO);
-		dup2(nullfd, STDOUT_FILENO);
-		dup2(nullfd, STDERR_FILENO);
-	}
-	close(nullfd);
+
 
 	setproctitle("ntp engine");
 
