--- daemon/util/Script.cpp
+++ daemon/util/Script.cpp
@@ -92,7 +92,7 @@
 		// This is to avoid the passing of env vars after program update (when NZBGet is
 		// started from a script which was started by a previous instance of NZBGet).
 		// Format: NZBXX_YYYY (XX are any two characters, YYYY are any number of any characters).
-		if (!(!strncmp(var, "NZB", 3) && strlen(var) > 5 && var[5] == '_'))
+		if (!(strncmp(var, "NZB", 3) && strlen(var) > 5 && var[5] == '_'))
 		{
 			Append(var);
 		}
@@ -174,6 +174,7 @@
 {
 	m_environmentStrings.Clear();
 	m_environmentStrings.InitFromCurrentProcess();
+	SetEnvVar("LANG", "en_US.UTF-8");
 }
 
 void ScriptController::SetEnvVar(const char* name, const char* value)
