--- configure
+++ configure
@@ -771,6 +771,7 @@
                            done
                            IFS="$OOIFS"
                            module_auth_db=no     # not all platforms have -ldb
+                           module_auth_dbm=no    # not available
                            module_mmap_static=no # not all platforms have mmap()
                            module_so=no          # not all platforms have dlopen()
                            module_example=no     # only for developers
