--- configure.orig	2008-12-05 14:27:37.000000000 +0100
+++ configure	2008-12-05 18:12:42.000000000 +0100
@@ -767,6 +767,7 @@
                            done
                            IFS="$OOIFS"
                            module_auth_db=no     # not all platforms have -ldb
+                           module_auth_dbm=no    # not available
                            module_mmap_static=no # not all platforms have mmap()
                            module_so=no          # not all platforms have dlopen()
                            module_example=no     # only for developers
