--- src/ap/Makefile.tmpl
+++ src/ap/Makefile.tmpl
@@ -24,7 +24,7 @@
 
 $(LIB): $(OBJS)
 	rm -f $@
-	ar cr $@ $(OBJS)
+	$(AR) cr $@ $(OBJS)
 	$(RANLIB) $@
 
 # We really don't expect end users to use this rule.  It works only with
--- src/Configure
+++ src/Configure
@@ -2654,7 +2654,7 @@
 
 dummy $(LIB): $(OBJS)
 	rm -f $@
-	ar cr $@ $(OBJS)
+	$(AR) cr $@ $(OBJS)
 	$(RANLIB) $@
 
 .SUFFIXES: .o .so .dll
--- src/lib/expat-lite/Makefile.tmpl
+++ src/lib/expat-lite/Makefile.tmpl
@@ -11,7 +11,7 @@
 
 libexpat.a: $(OBJS)
 	rm -f libexpat.a
-	ar cr libexpat.a $(OBJS)
+	$(AR) cr libexpat.a $(OBJS)
 	$(RANLIB) libexpat.a
 
 clean:
--- src/main/Makefile.tmpl
+++ src/main/Makefile.tmpl
@@ -20,7 +20,7 @@
 
 $(LIB): $(OBJS)
 	rm -f $@
-	ar cr $@ $(OBJS)
+	$(AR) cr $@ $(OBJS)
 	$(RANLIB) $@
 
 clean:
--- src/modules/proxy/Makefile.tmpl
+++ src/modules/proxy/Makefile.tmpl
@@ -14,7 +14,7 @@
 
 libproxy.a: $(OBJS)
 	rm -f $@
-	ar cr $@ $(OBJS)
+	$(AR) cr $@ $(OBJS)
 	$(RANLIB) $@
 
 libproxy.so: $(OBJS_PIC)
--- src/os/bs2000/Makefile.tmpl
+++ src/os/bs2000/Makefile.tmpl
@@ -11,7 +11,7 @@
 
 $(LIB): $(OBJS)
 	rm -f $@
-	ar cr $@ $(OBJS)
+	$(AR) cr $@ $(OBJS)
 	$(RANLIB) $@
 
 .c.o:
--- src/os/cygwin/Makefile.tmpl
+++ src/os/cygwin/Makefile.tmpl
@@ -11,7 +11,7 @@
 
 $(LIB): $(OBJS)
 	rm -f $@
-	ar cr $@ $(OBJS)
+	$(AR) cr $@ $(OBJS)
 	$(RANLIB) $@
 
 .c.o:
--- src/os/mpeix/Makefile.tmpl
+++ src/os/mpeix/Makefile.tmpl
@@ -12,7 +12,7 @@
 
 $(LIB): $(LIBOBJS)
 	rm -f $@
-	ar cr $@ $(LIBOBJS)
+	$(AR) cr $@ $(LIBOBJS)
 	$(RANLIB) $@
 
 .c.o:
--- src/os/os2/Makefile.tmpl
+++ src/os/os2/Makefile.tmpl
@@ -12,7 +12,7 @@
 
 $(LIB): $(OBJS)
 	rm -f $@
-	ar cr $@ $(OBJS)
+	$(AR) cr $@ $(OBJS)
 	$(RANLIB) $@
 
 .c.o:
--- src/os/os390/Makefile.tmpl
+++ src/os/os390/Makefile.tmpl
@@ -11,7 +11,7 @@
 
 $(LIB): $(OBJS)
 	rm -f $@
-	ar cr $@ $(OBJS)
+	$(AR) cr $@ $(OBJS)
 	$(RANLIB) $@
 
 .c.o:
--- src/os/tpf/Makefile.tmpl
+++ src/os/tpf/Makefile.tmpl
@@ -11,7 +11,7 @@
 
 $(LIB): $(OBJS)
 	rm -f $@
-	ar cr $@ $(OBJS)
+	$(AR) cr $@ $(OBJS)
 	$(RANLIB) $@
 
 .c.o:
--- src/os/unix/Makefile.tmpl
+++ src/os/unix/Makefile.tmpl
@@ -11,7 +11,7 @@
 
 $(LIB): $(OBJS)
 	rm -f $@
-	ar cr $@ $(OBJS)
+	$(AR) cr $@ $(OBJS)
 	$(RANLIB) $@
 
 .c.o:
--- src/regex/Makefile.tmpl
+++ src/regex/Makefile.tmpl
@@ -39,7 +39,7 @@
 
 libregex.a: $(OBJPRODN)
 	rm -f libregex.a
-	ar cr libregex.a $(OBJPRODN)
+	$(AR) cr libregex.a $(OBJPRODN)
 	$(RANLIB) libregex.a
 
 default:	r
