--- src/server.c
+++ src/server.c
@@ -273,10 +273,14 @@
 		if (bind(sockets[1], (struct sockaddr *) addresses[1], sizeof (struct sockaddr_in6)) < 0)
 			Log_fatal("bind %s %d: %s", Util_addressToString(addresses[1]), Util_addressToPort(addresses[1]), strerror(errno));
 		val = 0xe0;
+#if defined(IPV6_TCLASS)
 		if (setsockopt(sockets[1], IPPROTO_IPV6, IPV6_TCLASS, &val, sizeof(val)) < 0)
+#endif
 			Log_warn("Server: Failed to set TOS for UDP Socket");
 		val = 0x80;
+#if defined(IPV6_TCLASS)
 		if (setsockopt(sockets[1], IPPROTO_IPV6, IPV6_TCLASS, &val, sizeof(val)) < 0)
+#endif
 			Log_warn("Server: Failed to set TOS for UDP Socket");
 
 		fcntl(sockets[1], F_SETFL, O_NONBLOCK);
