--- src/ssl.c
+++ src/ssl.c
@@ -144,9 +144,11 @@
 	 * Read the certificate again so that it gets last in chain. Later releases like 0.14.0 works
 	 * fine with the extra certificate, so I don't see any harm in doing so.
 	 */
+#if 0
 	rc = x509parse_crtfile(&certificate, crtfile);
 	if (rc != 0)
 		Log_fatal("Could not read certificate file %s", crtfile);
+#endif
 	
 	free(ca_file);
 }
@@ -217,7 +219,7 @@
     ssl_set_ciphers(ssl, ciphers);
     ssl_set_session(ssl, 0, 0, ssn);
 
-    ssl_set_ca_chain(ssl, certificate.next, NULL, NULL);
+    ssl_set_ca_chain(ssl, (certificate.next ? certificate.next : &certificate), NULL, NULL);
     ssl_set_own_cert(ssl, &certificate, &key);
     ssl_set_dh_param(ssl, my_dhm_P, my_dhm_G);
 
