--- linux-2.6.28/net/ipv4/ip_output.c
+++ linux-2.6.28/net/ipv4/ip_output.c
@@ -94,9 +94,9 @@
 #include <linux/if_pppox.h>
 #include <linux/if_ether.h>
 #include <linux/if_vlan.h>
-#include <../../../fusiv_src/kernel/ap_code/inc/netpro/vx185/commonconfig.h>
-#include <../../../fusiv_src/kernel/ap_code/inc/netpro/apprehdr.h>
-#include <../../../fusiv_src/kernel/ap_code/inc/netpro/buftrans.h>
+#include <../include/netpro/vx185/commonconfig.h>
+#include <../include/netpro/apprehdr.h>
+#include <../include/netpro/buftrans.h>
 #if CONFIG_IPSEC_AP_SUPPORT
 #include <netpro/cen.h>
 #include <netpro/secap_cen.h>
