--- linux-2.6.28/drivers/net/avm_cpmac/Makefile
+++ linux-2.6.28/drivers/net/avm_cpmac/Makefile
@@ -23,13 +23,13 @@
 			   cpphy_entry.o cpphy_ar8216_ar.o cpphy_mgmt.o \
 			   cpmac_fusiv_if.o cpmac_puma_if.o cpphy_switch.o cpmac_product_conf.o
 			
-$(obj)/cpphy_entry.o: $(src)/cpphy_entry.c DO_COPY FORCE
+$(obj)/cpphy_entry.o: $(src)/cpphy_entry.c FORCE
 
 DO_COPY:
 	mkdir -p $(FRITZ_BOX_BUILD_DIR)/$(KERNEL_BUILD)/filesystem/include/linux 
 	cp -f $(src)/linux_avm_cpmac.h $(FRITZ_BOX_BUILD_DIR)/$(KERNEL_BUILD)/filesystem/include/linux/avm_cpmac.h
 
-extra-y	+= cpmacconfig
+#extra-y	+= cpmacconfig
 
 cpmacconfig-objs	:= cpmacconfig.o
 
