--- src/Makefile.in
+++ src/Makefile.in
@@ -1276,6 +1276,7 @@
 CFLAGS_FOR_BUILD = @CFLAGS_FOR_BUILD@
 CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
+CPPFLAGS += $(if $(strip $(filter -lpthread,@FAM_LIBS@ @LIBEV_LIBS@ @LIBUNWIND_LIBS@ @LUA_LIBS@ @MYSQL_LIBS@ @SQLITE_LIBS@ @UUID_LIBS@ @XML_LIBS@)),-D_REENTRANT)
 CPPFLAGS_FOR_BUILD = @CPPFLAGS_FOR_BUILD@
 CPP_FOR_BUILD = @CPP_FOR_BUILD@
 CRYPTO_LIB = @CRYPTO_LIB@
@@ -1732,6 +1733,7 @@
 @LIGHTTPD_STATIC_TRUE@	$(am__append_41) $(am__append_44) \
 @LIGHTTPD_STATIC_TRUE@	$(am__append_46) $(am__append_49) \
 @LIGHTTPD_STATIC_TRUE@	$(am__append_52) $(am__append_55)
+lighttpd_LDADD += $(strip $(filter -lpthread,$(FAM_LIBS) $(LIBEV_LIBS) $(LIBUNWIND_LIBS) $(LUA_LIBS) $(MYSQL_LIBS) $(SQLITE_LIBS) $(UUID_LIBS) $(XML_LIBS)))
 @LIGHTTPD_STATIC_FALSE@lighttpd_LDFLAGS = -export-dynamic
 @LIGHTTPD_STATIC_TRUE@lighttpd_LDFLAGS = -export-dynamic
 t_test_common_SOURCES = t/test_common.c \
