--- src/Makefile.in
+++ src/Makefile.in
@@ -1247,6 +1247,7 @@
 CFLAGS_FOR_BUILD = @CFLAGS_FOR_BUILD@
 CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
+CPPFLAGS += $(if $(strip $(filter -lpthread,@FAM_LIBS@ @LIBEV_LIBS@ @LIBUNWIND_LIBS@ @LUA_LIBS@ @MYSQL_LIBS@ @SQLITE_LIBS@ @UUID_LIBS@ @XML_LIBS@)),-D_REENTRANT)
 CPPFLAGS_FOR_BUILD = @CPPFLAGS_FOR_BUILD@
 CPP_FOR_BUILD = @CPP_FOR_BUILD@
 CRYPTO_LIB = @CRYPTO_LIB@
@@ -1684,6 +1685,7 @@
 @LIGHTTPD_STATIC_TRUE@	$(am__append_29) $(am__append_32) \
 @LIGHTTPD_STATIC_TRUE@	$(am__append_35) $(am__append_37) \
 @LIGHTTPD_STATIC_TRUE@	$(am__append_39) $(am__append_40)
+lighttpd_LDADD += $(strip $(filter -lpthread,$(FAM_LIBS) $(LIBEV_LIBS) $(LIBUNWIND_LIBS) $(LUA_LIBS) $(MYSQL_LIBS) $(SQLITE_LIBS) $(UUID_LIBS) $(XML_LIBS)))
 @LIGHTTPD_STATIC_FALSE@lighttpd_LDFLAGS = -export-dynamic
 @LIGHTTPD_STATIC_TRUE@lighttpd_LDFLAGS = -export-dynamic
 t_test_array_SOURCES = t/test_array.c array.c data_array.c data_integer.c data_string.c buffer.c
