--- C/7zip/Compress/LZMA/LZMAEncoder.cpp
+++ C/7zip/Compress/LZMA/LZMAEncoder.cpp
@@ -1203,9 +1203,9 @@
 
   if (repLens[repMaxIndex] >= 2)
   {
-    if (repLens[repMaxIndex] + 1 >= lenMain || 
-        repLens[repMaxIndex] + 2 >= lenMain && (backMain > (1 << 9)) ||
-        repLens[repMaxIndex] + 3 >= lenMain && (backMain > (1 << 15)))
+    if ((repLens[repMaxIndex] + 1 >= lenMain) || 
+        (repLens[repMaxIndex] + 2 >= lenMain && (backMain > (1 << 9))) ||
+        (repLens[repMaxIndex] + 3 >= lenMain && (backMain > (1 << 15))))
     {
       backRes = repMaxIndex;
       lenRes = repLens[repMaxIndex];
@@ -1219,10 +1219,10 @@
     if (_longestMatchLength >= 2)
     {
       UInt32 newDistance = matchDistances[_numDistancePairs - 1];
-      if (_longestMatchLength >= lenMain && newDistance < backMain || 
-          _longestMatchLength == lenMain + 1 && !ChangePair(backMain, newDistance) ||
-          _longestMatchLength > lenMain + 1 ||
-          _longestMatchLength + 1 >= lenMain && lenMain >= 3 && ChangePair(newDistance, backMain))
+      if ((_longestMatchLength >= lenMain && newDistance < backMain) || 
+          (_longestMatchLength == lenMain + 1 && !ChangePair(backMain, newDistance)) ||
+          (_longestMatchLength > lenMain + 1) ||
+          (_longestMatchLength + 1 >= lenMain && lenMain >= 3 && ChangePair(newDistance, backMain)))
       {
         _longestMatchWasFound = true;
         backRes = UInt32(-1);
--- C/7zip/Compress/LZMA/LZMADecoder.cpp
+++ C/7zip/Compress/LZMA/LZMADecoder.cpp
@@ -309,7 +309,7 @@
 
 STDMETHODIMP CDecoder::SetOutStreamSize(const UInt64 *outSize)
 {
-  if (_outSizeDefined = (outSize != NULL))
+  if ( (_outSizeDefined = (outSize != NULL)) )
     _outSize = *outSize;
   _remainLen = kLenIdNeedInit;
   _outWindowStream.Init();
--- C/7zip/Compress/LZMA_Alone/LzmaAlone.cpp
+++ C/7zip/Compress/LZMA_Alone/LzmaAlone.cpp
@@ -368,7 +368,7 @@
     if(parser[NKey::kFastBytes].ThereIs)
       if (!GetNumber(parser[NKey::kFastBytes].PostStrings[0], numFastBytes))
         IncorrectCommand();
-    if (matchFinderCyclesDefined = parser[NKey::kMatchFinderCycles].ThereIs)
+    if ( (matchFinderCyclesDefined = parser[NKey::kMatchFinderCycles].ThereIs) )
       if (!GetNumber(parser[NKey::kMatchFinderCycles].PostStrings[0], matchFinderCycles))
         IncorrectCommand();
     if(parser[NKey::kLitContext].ThereIs)
