--- Makefile.org	2010-05-12 18:31:33.000000000 +0200
+++ Makefile	2010-05-12 18:31:57.000000000 +0200
@@ -13,16 +13,6 @@ endif
 # Allow people to override OpenSSL and build it statically, if they need
 # a special build for the DTLS support. $(OPENSSL) points to the build 
 # dir; there's no need to install it anywhere (we link it statically).
-ifdef OPENSSL
-SSL_CFLAGS += -I$(OPENSSL)/include
-SSL_LDFLAGS += $(OPENSSL)/libssl.a $(OPENSSL)/libcrypto.a
-else
-ifeq ($(wildcard /usr/include/openssl),)
-$(error "No OpenSSL in /usr/include/openssl. Cannot continue");
-endif
-SSL_CFLAGS += -I/usr/include/openssl
-SSL_LDFLAGS += -lcrypto -lssl
-endif
 
 XML2_CFLAGS += $(shell xml2-config --cflags) 
 XML2_LDFLAGS += $(shell xml2-config --libs)
