--- bin/prepare_fwupgrade.orig	2006-09-28 10:54:05.000000000 +0200
+++ bin/prepare_fwupgrade	2007-06-25 13:48:11.000000000 +0200
@@ -37,7 +37,7 @@
 voipd -s
 avmike -s
 killall printserv
-/sbin/igdd -s
+[ -x /sbin/igdd ] && /sbin/igdd -s
 if [ "$1" = "start_from_internet" ] ; then
 # we will download the update image from the internet so
 # keep websrv and dsld running
@@ -93,6 +93,7 @@
 killall avmike
 killall dsld
 killall websrv
+[ -x /etc/init.d/rc.websrv ] && /etc/init.d/rc.websrv stop 2> /dev/null
 killall user_drv
 killall wpa_authenticator
 killall wstart
@@ -106,6 +107,7 @@
 killall -9 dsld
 killall -9 thttpd
 killall -9 websrv
+[ -x /etc/init.d/rc.websrv ] && /etc/init.d/rc.websrv kill 2> /dev/null
 killall -9 user_drv
 killall -9 wpa_authenticator
 killall -9 wstart
--- etc/init.d/rc.net.orig	2007-06-25 13:48:10.000000000 +0200
+++ etc/init.d/rc.net	2007-06-25 13:48:11.000000000 +0200
@@ -43,8 +43,20 @@
    WLAN_TEST=0
 fi
 
+# Do we have a websrv or was ist stripped from the firmware?
+_websrv=$(basename $(which websrv) 2> /dev/null)
+# Do we have a UPnP server (igdd) or was ist stripped from the firmware?
+_igdd=$(basename $(which igdd) 2> /dev/null)
+# Does a multid option to start without UPnP device (-u) exist?
+_multid_upnp=$(multid -? 2>&1 | grep upnp)
+# Set multid "no UPnP" option, if
+#   a) it has the parameter at all AND
+#   b) igdd binary does *not* exist
+[ "$_multid_upnp" ] && [ ! "$_igdd" ] && MULTIDPARAM="-u"
+
 igddstart()
 {
+   [ "$_igdd" ] || return;
    if [ "`pidof igdd`" = "" ] ; then
       igdd
       sleep 1
@@ -208,14 +213,15 @@
 stopall()
 {
    if [ "$CONFIG_DSL" ] ; then
-      AVMDAEMONS="avmike dsld igdd websrv multid"
+      AVMDAEMONS="avmike dsld $_igdd $_websrv multid"
    else
-      AVMDAEMONS="igdd websrv multid"
+      AVMDAEMONS="$_igdd $_websrv multid"
    fi
    DAEMONS=
    for i in $AVMDAEMONS ; do
     eval $i -s
    done
+   [ "$_websrv" ] || /etc/init.d/rc.websrv stop
    sleep 3
    for i in $DAEMONS $AVMDAEMONS ; do
      killall -15 $i
@@ -249,9 +255,15 @@
      wlanstart
    fi
 
+   if [ "$_websrv" ]; then
    if [ "`pidof websrv`" = "" ] ; then
      websrv $NICEPARAM
    fi
+   else
+     if [ "$(/etc/init.d/rc.websrv status)" == "stopped" ]; then
+       /etc/init.d/rc.websrv start
+     fi
+   fi
 
    igddstart
    multidstart
@@ -303,9 +315,9 @@
       if [ "$1" ] ; then
          AVMDAEMONS="$*"
       elif [ "$CONFIG_DSL" ] ; then
-         AVMDAEMONS="avmike dsld igdd multid"
+         AVMDAEMONS="avmike dsld $_igdd multid"
       else
-         AVMDAEMONS="igdd multid"
+         AVMDAEMONS="$_igdd multid"
       fi
 
       # stop the daemon with -s
