--- linux-2.6.32.21/Makefile.orig	2010-08-27 01:42:20.000000000 +0200
+++ linux-2.6.32.21/Makefile	2011-03-20 17:08:27.921534751 +0100
@@ -839,6 +839,9 @@
 	$(Q)echo 'cmd_$@ := $(cmd_vmlinux-modpost)' > $(dot-target).cmd
 endef
 
+vmlinux.eva_pad: vmlinux
+	echo dummy > $@
+
 # vmlinux image - including updated kernel symbols
 vmlinux: $(vmlinux-lds) $(vmlinux-init) $(vmlinux-main) vmlinux.o $(kallsyms.o) FORCE
 ifdef CONFIG_HEADERS_CHECK
