--- src/openvpn/mtcp.c
+++ src/openvpn/mtcp.c
@@ -542,9 +542,23 @@
        */
       if (touched && IS_SIG (&touched->context))
 	{
+	  int exit_after_close = 0;
+
+	  if (touched->context.c2.link_socket->inetd == INETD_NOWAIT)
+	    {
+	      exit_after_close = 1;
+	    }
+
 	  if (mi == touched)
 	    mi = NULL;
 	  multi_close_instance_on_signal (m, touched);
+
+	  /*
+	   * If this socket results from an inetd-nowait - connection
+	   * it has to be the only one and we have to exit.
+	   */
+	  if (exit_after_close)
+	    m->top.sig->signal_received = SIGTERM;
 	}
     }
 
--- src/openvpn/options.c
+++ src/openvpn/options.c
@@ -1856,8 +1856,10 @@
       )
     msg (M_USAGE, "--inetd nowait can only be used in TLS mode");
 
+/*
   if (options->inetd == INETD_NOWAIT && dev != DEV_TYPE_TAP)
     msg (M_USAGE, "--inetd nowait only makes sense in --dev tap mode");
+*/
 
 
   if (options->lladdr && dev != DEV_TYPE_TAP)
@@ -2040,8 +2042,10 @@
 #endif
       if (options->shaper)
 	msg (M_USAGE, "--shaper cannot be used with --mode server");
+#if 0
       if (options->inetd)
 	msg (M_USAGE, "--inetd cannot be used with --mode server");
+#endif
       if (options->ipchange)
 	msg (M_USAGE, "--ipchange cannot be used with --mode server (use --client-connect instead)");
       if (!(proto_is_dgram(ce->proto) || ce->proto == PROTO_TCPv4_SERVER
@@ -2425,8 +2429,10 @@
    * In forking TCP server mode, you don't need to ifconfig
    * the tap device (the assumption is that it will be bridged).
    */
+/*
   if (options->inetd == INETD_NOWAIT)
     options->ifconfig_noexec = true;
+*/
 
 #ifdef WIN32
   if ((dev == DEV_TYPE_TUN || dev == DEV_TYPE_TAP) && !options->route_delay_defined)
--- src/openvpn/socket.c
+++ src/openvpn/socket.c
@@ -1418,7 +1418,9 @@
       ASSERT (sock->info.proto == PROTO_TCPv4_SERVER
 	      || sock->info.proto == PROTO_TCPv6_SERVER
 	     );
+/*
       ASSERT (!sock->inetd);
+*/
       sock->sd = accept_from->sd;
     }
 
