--- ./linux-3.10/net/avm_pa/avm_pa.c
+++ ./linux-3.10/net/avm_pa/avm_pa.c
@@ -424,7 +424,7 @@
 #define AVM_PA_MINRATE                  1000
 #define AVM_PA_DEFAULT_PKTBUFFER        1024
 #define AVM_PA_DEFAULT_PKTPEAK           256
-#define AVM_PA_DEFAULT_TELEPHONY_REDUCE   65
+#define AVM_PA_DEFAULT_TELEPHONY_REDUCE    1
 
 #define AVM_PA_EST_DEFAULT_IDX                0 /* 0 - 5 => 0.25sec - 8sec */
 #define AVM_PA_EST_DEFAULT_EWMA_LOG           3 /* 1 - 31 */
