--- configure
+++ configure
@@ -5187,7 +5187,7 @@
   # Note, version OpenSSL 0.9.7j and higher requires -lcrypto even on Linux.
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libssl" >&5
 $as_echo_n "checking for libssl... " >&6; }
-  if ${sc_cv_have_libssl+:} false; then :
+  if false; then # ignore cached value otherwise SSL libs are not added to LIBS
   $as_echo_n "(cached) " >&6
 else
    LIBS0="$LIBS"
