--- GNUmakefile.in	2006-09-29 22:10:01.000000000 +0200
+++ GNUmakefile.in	2006-09-29 22:11:34.000000000 +0200
@@ -246,7 +246,7 @@
 CFLAGS = @CFLAGS@ @CPPFLAGS@ $(OTHER_CFLAGS) $(SPECIAL_CFLAGS) -Wall \
          @STATIC_PCRE_ONLY@ -Ipcre 
 
-LDFLAGS = $(DEBUG_CFLAGS) $(SPECIAL_CFLAGS)
+LDFLAGS = @LDFLAGS@ $(DEBUG_CFLAGS) $(SPECIAL_CFLAGS)
 
 
 #############################################################################
