--- GNUmakefile.in.orig	2007-07-28 12:42:17.000000000 +0200
+++ GNUmakefile.in	2007-07-28 12:42:17.000000000 +0200
@@ -984,8 +984,9 @@
 
 # An auxiliary program makes the PCRE default character table source
 
-pcre/chartables.c:   pcre/dftables@EXEEXT@
-		pcre/dftables@EXEEXT@ >pcre/chartables.c
+#pcre/chartables.c:   pcre/dftables@EXEEXT@
+#		pcre/dftables@EXEEXT@ >pcre/chartables.c
+pcre/chartables.c:
 
 pcre/dftables@EXEEXT@:       pcre/dftables.c pcre/maketables.c pcre/pcre.h pcre/internal.h pcre/config.h
 		$(CC) -o pcre/dftables@EXEEXT@ $(CFLAGS) pcre/dftables.c
