--- etc/hotplug/storage.orig	2008-08-01 11:41:17.000000000 +0200
+++ etc/hotplug/storage	2008-08-13 16:23:32.000000000 +0200
@@ -11,6 +11,10 @@
 FIRST=$6
 MSTART=/sbin/start_mediasrv
 MSTOP=/sbin/stop_mediasrv
+
+# Do we have a Ftp server (ftpd) or was ist stripped from the firmware?
+_ftpd=$(basename $(which ftpd) 2> /dev/null)
+
 ## Wait for removal of filesystem semaphores (mask: $1, maximal wait period: $2)
 wait4remove () {
 local COUNT=0
@@ -138,6 +142,7 @@
 ## ftp server control
 ENABLED=`echo usbhost.ftp_server_enabled|ar7cfgctl -s`
 if [ "$ENABLED" != "no" ] ; then
+if [ "$_ftpd" ] ; then
 if test -x /bin/inetdctl ; then
 inetdctl enable ftpd
 else
@@ -154,6 +159,7 @@
 ftpd -D -q $READONLYOPTION -t 120 -m 15 $USERSOPTION -h "$CONFIG_PRODUKT_NAME" &
 fi
 fi
+fi
 else
 if test -x /bin/inetdctl ; then
 inetdctl disable ftpd
--- etc/hotplug/run_mount.orig	2008-08-01 11:41:05.000000000 +0200
+++ etc/hotplug/run_mount	2008-08-13 16:23:32.000000000 +0200
@@ -14,6 +14,10 @@
 MSTART=/sbin/start_mediasrv
 TR069START=/bin/tr069starter
 SYSFS=/var/sysfs
+
+# Do we have a Ftp server (ftpd) or was ist stripped from the firmware?
+_ftpd=$(basename $(which ftpd) 2> /dev/null)
+
 ## Generate a frontend'able 'nice name' to be used as mount point:
 nicename () {
 ## Find device's strings, remove/translate evil chars...
@@ -79,6 +83,7 @@
 eventadd 140 $MNTNAME
 ENABLED=`echo usbhost.ftp_server_enabled|ar7cfgctl -s`
 if [ "$ENABLED" != "no" ] ; then
+if [ "$_ftpd" ] ; then
 if test -x /bin/inetdctl; then
 inetdctl enable ftpd
 else
@@ -98,6 +103,7 @@
 fi
 fi
 fi
+fi
 test -x $TR069START && $TR069START $MNTNAME
 if test -x /etc/samba_control ; then
 /etc/samba_control reconfig
