--- configure
+++ configure
@@ -7166,6 +7166,7 @@
 fi
 rm -f core conftest.err conftest.$ac_objext conftest.beam conftest.$ac_ext
 } # clang
+if false; then
 	{
 	{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if $LD supports link flag -Wl,-z,retpolineplt" >&5
 printf %s "checking if $LD supports link flag -Wl,-z,retpolineplt... " >&6; }
@@ -7213,6 +7214,7 @@
 rm -f core conftest.err conftest.$ac_objext conftest.beam \
     conftest$ac_exeext conftest.$ac_ext
 }
+fi
 	{
 	{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if $CC supports compile flag -D_FORTIFY_SOURCE=2" >&5
 printf %s "checking if $CC supports compile flag -D_FORTIFY_SOURCE=2... " >&6; }
