--- usr/www/all/menus/menu_data.lua
+++ usr/www/all/menus/menu_data.lua
@@ -1955,6 +1955,20 @@
 ["lnk"] = pageData["liveTv"].lnk,
 ["pos"] = -65
 } or nil
+-- TODO: find out how to check if Freetz WOL files are avalaible
+local freetzWolInstalled = false
+if freetzWolInstalled then
+menuData["freetz-wol"] = {
+["txt"] = "Freetz WOL",
+["lnk"] = "/cgi-bin/freetz_wol",
+["pos"] = -55
+}
+end
+menuData["freetz"] = {
+["txt"] = "Freetz",
+["lnk"] = "/cgi-bin/freetz_status",
+["pos"] = -60
+}
 function checkMenu(pid)
 local exists = false
 checkedIds[pid] = true
