--- Makefile.in
+++ Makefile.in
@@ -513,7 +513,7 @@
 # series won't be binary-compatible.
 @INSTALL_LIBEVENT_TRUE@dist_bin_SCRIPTS = event_rpcgen.py
 pkgconfigdir = $(libdir)/pkgconfig
-LIBEVENT_PKGCONFIG = libevent.pc $(am__append_2) $(am__append_4)
+LIBEVENT_PKGCONFIG = libevent.pc
 
 # These sources are conditionally added by configure.in or conditionally
 # included from other files.
@@ -532,12 +532,11 @@
 	Makefile.nmake test/Makefile.nmake \
 	$(PLATFORM_DEPENDENT_SRC)
 
-LIBEVENT_LIBS_LA = libevent.la libevent_core.la libevent_extra.la \
-	$(am__append_1) $(am__append_3)
+LIBEVENT_LIBS_LA = libevent.la
 @INSTALL_LIBEVENT_TRUE@lib_LTLIBRARIES = $(LIBEVENT_LIBS_LA)
 @INSTALL_LIBEVENT_TRUE@pkgconfig_DATA = $(LIBEVENT_PKGCONFIG)
 @INSTALL_LIBEVENT_FALSE@noinst_LTLIBRARIES = $(LIBEVENT_LIBS_LA)
-SUBDIRS = . include sample test
+SUBDIRS = . include
 @BUILD_WIN32_FALSE@SYS_LIBS = 
 @BUILD_WIN32_TRUE@SYS_LIBS = -lws2_32 -lshell32 -ladvapi32
 @BUILD_WIN32_FALSE@SYS_SRC = $(am__append_5) $(am__append_6) \
@@ -1243,7 +1242,7 @@
 
 install-dvi-am:
 
-install-exec-am: install-dist_binSCRIPTS install-libLTLIBRARIES
+install-exec-am:                         install-libLTLIBRARIES
 
 install-html: install-html-recursive
 
