From: Joachim Bauch <mail@joachim-bauch.de>
Date: Mon, 3 Sep 2012 17:49:15 +0200
Subject: [PATCH] Prevent reference leak of bufferevent if getaddrinfo fails.

---
 bufferevent_sock.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git bufferevent_sock.c bufferevent_sock.c
index 50270be..e1fcadc 100644
--- bufferevent_sock.c
+++ bufferevent_sock.c
@@ -502,6 +502,7 @@ bufferevent_socket_connect_hostname(struct bufferevent *bev,
 	} else {
 		bufferevent_unsuspend_write(bev, BEV_SUSPEND_LOOKUP);
 		bufferevent_unsuspend_read(bev, BEV_SUSPEND_LOOKUP);
+		bufferevent_decref(bev);
 		return -1;
 	}
 }
--
