From: George Danchev <danchev@spnet.net>
Date: Wed, 12 Sep 2012 20:12:03 -0400
Subject: [PATCH] FIx another memleak on OOM in evdns

---
 evdns.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git evdns.c evdns.c
index 7b0fc2b..9e5a0f2 100644
--- evdns.c
+++ evdns.c
@@ -3159,6 +3159,8 @@ search_request_new(struct evdns_base *base, struct evdns_request *handle,
 		handle->search_origname = mm_strdup(name);
 		if (handle->search_origname == NULL) {
 			/* XXX Should we dealloc req? If yes, how? */
+			if (req)
+				mm_free(req);
 			return NULL;
 		}
 		handle->search_state = base->global_search_state;
--
