--- etc/init.d/S08-tffs.orig	2011-05-04 15:24:23.000000000 +0200
+++ etc/init.d/S08-tffs	2011-05-06 18:49:40.534483954 +0200
@@ -110,6 +110,10 @@
 tffs_nodes_list="$tffs_nodes_list $((0xCA)),websrv_ssl_cert.pem"
 tffs_nodes_list="$tffs_nodes_list $((0xE0)),smartmeter.cfg"
 #########################################################################
+## Freetz
+#########################################################################
+tffs_nodes_list="$tffs_nodes_list $((0x3C)),freetz"
+#########################################################################
 ## NODES anlegen, wenn nicht als _file_ existent
 #########################################################################
 for node in ${tffs_nodes_list} ; do
