--- extra/servers_additional.cfg
+++ extra/servers_additional.cfg
@@ -25,3 +25,4 @@
 mondo@freedns.afraid.org --dyndns_server_name freedns.afraid.org --dyndns_server_url /dynamic/update.php? --svr_rsp_success "has not changed" --svr_ip_in_rsp --svr_ip_param address blah blah
 default@he.net --dyndns_server_name dyn.dns.he.net --dyndns_server_url /nic/update? --ip_server_name checkip.dns.he.net / --svr_rsp_success "good,nochg" --svr_rsp_other "dnserr,911" --svr_ip_param myip
 default@dy.fi --dyndns_server_name www.dy.fi --dyndns_server_url /nic/update?hostname= --ip_server_name checkip.dy.fi / --svr_rsp_success "good,nochg" --svr_ip_param myip
+default@tunnelbroker.net --dyndns_server_name ipv4.tunnelbroker.net --dyndns_server_url /nic/update?hostname= --ip_server_name checkip.dns.he.net / --svr_rsp_success "good,nochg" --svr_rsp_other "dnserr,911" --svr_ip_param myip
