--- linux-2.6.28/drivers/net/avm_cpmac/Makefile.orig	2010-09-15 14:30:13.000000000 +0200
+++ linux-2.6.28/drivers/net/avm_cpmac/Makefile	2010-10-14 15:38:44.100544900 +0200
@@ -6,7 +6,7 @@
 
 EXTRA_CFLAGS := -DCPMAC_SIGNAL_CONGESTION -DCPPHY_USE_SWITCH
 EXTRA_CFLAGS += -Wall -Wextra -Werror
-EXTRA_CFLAGS += -DAVM_CPMAC_VERSION="\"`svn info $(src) | grep URL | sed -e "s/^.*\/\([0-9.]\+\)-[^\/]*/\1/g"`  -  Revision `svnversion -cn $(src)`  -  `date +\"%a  %e.%m.%Y  %H:%M:%S\"` \""
+EXTRA_CFLAGS += -DAVM_CPMAC_VERSION="\"1.108.1.32.1.3 - Revision 811:1122 - Fr 10.09.2010 16:11:49\""
 
 # declare multi-module main driver.
 obj-$(CONFIG_AVM_CPMAC)	+= avm_cpmac.o
@@ -17,13 +17,13 @@
 			   cpphy_entry.o cpphy_ar8216_ar.o cpphy_mgmt.o \
 			   cpmac_fusiv_if.o cpmac_puma_if.o cpphy_switch.o cpmac_product_conf.o
 			
-$(obj)/cpphy_entry.o: $(src)/cpphy_entry.c DO_COPY FORCE
+$(obj)/cpphy_entry.o: $(src)/cpphy_entry.c FORCE
 
 DO_COPY:
 	mkdir -p $(FRITZ_BOX_BUILD_DIR)/$(KERNEL_BUILD)/filesystem/include/linux 
 	cp -f $(src)/linux_avm_cpmac.h $(FRITZ_BOX_BUILD_DIR)/$(KERNEL_BUILD)/filesystem/include/linux/avm_cpmac.h
 
-extra-y	+= cpmacconfig
+#extra-y	+= cpmacconfig
 
 cpmacconfig-objs	:= cpmacconfig.o
 
