--- linux-2.6.39/drivers/usb/serial/ftdi_sio.c	2012-03-15 11:53:32.000000000 +0100
+++ linux-2.6.39/drivers/usb/serial/ftdi_sio.c	2019-03-10 18:58:43.405590206 +0100
@@ -828,6 +828,19 @@
 		.driver_info = (kernel_ulong_t)&ftdi_jtag_quirk },
 	{ USB_DEVICE(ST_VID, ST_STMCLT1030_PID),
 		.driver_info = (kernel_ulong_t)&ftdi_stmclite_quirk },
+	{ USB_DEVICE(WBE_VID, WBE_SMARTMOUSE_PID1) },
+	{ USB_DEVICE(WBE_VID, WBE_SMARTMOUSE_PID2) },
+	{ USB_DEVICE(WBE_VID, WBE_SMARTMOUSE_PID3) },
+	{ USB_DEVICE(WBE_VID, WBE_SMARTMOUSE_PID4) },
+	{ USB_DEVICE(FTDI_VID, FTDI_8U232AM_PID1) },
+	{ USB_DEVICE(FTDI_VID, FTDI_8U232AM_PID2) },
+	{ USB_DEVICE(FTDI_VID, FTDI_8U232AM_PID3) },
+	{ USB_DEVICE(FTDI_VID, FTDI_8U232AM_PID4) },
+	{ USB_DEVICE(FTDI_VID, FTDI_RF_R106) },
+	{ USB_DEVICE(QUAD_VID, QUADMOUSE_PID1) },
+	{ USB_DEVICE(QUAD_VID, QUADMOUSE_PID2) },
+	{ USB_DEVICE(QUAD_VID, QUADMOUSE_PID3) },
+	{ USB_DEVICE(QUAD_VID, QUADMOUSE_PID4) },
 	{ },					/* Optional parameter entry */
 	{ }					/* Terminating entry */
 };
--- linux-2.6.39/drivers/usb/serial/ftdi_sio_ids.h	2012-03-15 11:53:32.000000000 +0100
+++ linux-2.6.39/drivers/usb/serial/ftdi_sio_ids.h	2019-03-10 19:05:26.951858652 +0100
@@ -468,6 +468,30 @@
 #define FTDI_XF_640_PID 0xFC0E	/* 640: Two line Display */
 #define FTDI_XF_642_PID 0xFC0F	/* 642: Two line Display */
 
+#define FTDI_8U232AM_PID1	0x6001	/* FTDI USB2Serial */
+#define FTDI_8U232AM_PID2	0x6002	/* FTDI USB2Serial */
+#define FTDI_8U232AM_PID3	0x6003	/* FTDI USB2Serial */
+#define FTDI_8U232AM_PID4	0x6004	/* FTDI USB2Serial */
+#define FTDI_RF_R106		0x8A28	/* ZigBee controller */
+
+/**/
+#define WBE_VID 0x104F
+/*
+ */
+#define WBE_SMARTMOUSE_PID1	0x0001	/* Smartmouse USB */
+#define WBE_SMARTMOUSE_PID2	0x0002	/* Smartmouse USB */
+#define WBE_SMARTMOUSE_PID3	0x0003	/* Smartmouse USB */
+#define WBE_SMARTMOUSE_PID4	0x0004	/* Smartmouse USB */
+
+/**/
+#define QUAD_VID 0x0403
+/*
+ */
+#define QUADMOUSE_PID1		0x0001	/* Quadmouse USB */
+#define QUADMOUSE_PID2		0x0002	/* Quadmouse USB */
+#define QUADMOUSE_PID3		0x0003	/* Quadmouse USB */
+#define QUADMOUSE_PID4		0x0004	/* Quadmouse USB */
+
 /*
  * Video Networks Limited / Homechoice in the UK use an ftdi-based device
  * for their 1Mb broadband internet service.  The following PID is exhibited
