--- coreutils/df.c
+++ coreutils/df.c
@@ -249,11 +249,14 @@
 			bb_simple_perror_msg(mount_point);
 			goto set_error;
 		}
+#if 0
+		/* not necessary after switching to using statvfs */
 		/* Some uclibc versions were seen to lose f_frsize
 		 * (kernel does return it, but then uclibc does not copy it)
 		 */
 		if (s.f_frsize == 0)
 			s.f_frsize = s.f_bsize;
+#endif
 
 		if ((s.f_blocks > 0) || !mount_table || (opt & OPT_ALL)) {
 			unsigned long long blocks_used;
