--- libbb/die_if_bad_username.c
+++ libbb/die_if_bad_username.c
@@ -12,6 +12,15 @@
 #define LOGIN_NAME_MAX 256
 #endif
 
+#if defined(__UCLIBC__) && UCLIBC_VERSION < KERNEL_VERSION(0, 9, 29)
+/* workaround the build problem resulting from the fact that uClibc versions prior to 0.9.29 don't provide LOGIN_NAME_MAX symbol
+ * (available since the following commit http://git.uclibc.org/uClibc/commit?id=eb1c3012e66fc6a966d5d90e14752a13d3a2f407)
+ */
+#ifndef LOGIN_NAME_MAX
+#define LOGIN_NAME_MAX 256
+#endif
+#endif
+
 /* To avoid problems, the username should consist only of
  * letters, digits, underscores, periods, at signs and dashes,
  * and not start with a dash (as defined by IEEE Std 1003.1-2001).
