--- applets/applet_tables.c
+++ applets/applet_tables.c
@@ -67,6 +67,7 @@
 {
 	int i, j;
 	char tmp1[PATH_MAX], tmp2[PATH_MAX];
+	unsigned MAX_APPLET_NAME_LEN = 1;
 
 	// In find_applet_by_name(), before linear search, narrow it down
 	// by looking at N "equidistant" names. With ~350 applets:
@@ -139,8 +140,8 @@
 	printf("const char applet_names[] ALIGN1 = \"\"\n");
 	for (i = 0; i < NUM_APPLETS; i++) {
 		printf("\"%s\" \"\\0\"\n", applets[i].name);
-//		if (MAX_APPLET_NAME_LEN < strlen(applets[i].name))
-//			MAX_APPLET_NAME_LEN = strlen(applets[i].name);
+		if (MAX_APPLET_NAME_LEN < strlen(applets[i].name))
+			MAX_APPLET_NAME_LEN = strlen(applets[i].name);
 	}
 	printf(";\n\n");
 
@@ -208,8 +209,8 @@
 #endif
 	//printf("#endif /* SKIP_definitions */\n");
 
-//	printf("\n");
-//	printf("#define MAX_APPLET_NAME_LEN %u\n", MAX_APPLET_NAME_LEN);
+	printf("\n");
+	printf("#define MAX_APPLET_NAME_LEN %u\n", MAX_APPLET_NAME_LEN);
 
 	if (argv[2]) {
 		FILE *fp;
