--- networking/httpd.c
+++ networking/httpd.c
@@ -1592,6 +1592,9 @@
 	if (script != NULL)
 		*script = '\0';         /* cut off /PATH_INFO */
 
+	/* REDIRECT_STATUS is required by PHP compiled with --enable-force-cgi-redirect */
+	putenv((char*)"REDIRECT_STATUS=200");
+
 	/* SCRIPT_FILENAME is required by PHP in CGI mode */
 	if (home_httpd[0] == '/') {
 		char *fullpath = concat_path_file(home_httpd, url);
