--- etc/hotplug/storage.orig	2008-02-21 14:02:12.000000000 +0100
+++ etc/hotplug/storage	2008-02-22 20:05:11.000000000 +0100
@@ -15,6 +15,9 @@
 MSTART=/sbin/start_mediasrv
 MSTOP=/sbin/stop_mediasrv
 
+# Do we have a Ftp server (ftpd) or was ist stripped from the firmware?
+_ftpd=$(basename $(which ftpd) 2> /dev/null)
+
 # Ask the system's configuration for the status of the readonly flag:
 # (Translates flag's yes/no to ro/rw)
 #
@@ -205,11 +208,13 @@
 	if [ "$ENABLED" != "no" ] ; then
 		if [ "`pidof ftpd`" = "" ] ; then
 			USERS_ENABLED=`echo usbhost.users_enabled|ar7cfgctl -s`
-			if [ "$USERS_ENABLED" = "yes" ] ; then
-				ftpd -D -q -t 120 -m 15 -U -h "$CONFIG_PRODUKT_NAME" &
-			else
-				ftpd -D -q -t 120 -m 15 -h "$CONFIG_PRODUKT_NAME" &
-			fi
+      if [ "$_ftpd" ] ; then
+        if [ "$USERS_ENABLED" = "yes" ] ; then
+          ftpd -D -q -t 120 -m 15 -U -h "$CONFIG_PRODUKT_NAME" &
+        else
+          ftpd -D -q -t 120 -m 15 -h "$CONFIG_PRODUKT_NAME" &
+        fi
+      fi
 		fi
 	else
 		killall ftpd
