--- etc/hotplug/storage.orig	2008-02-17 13:36:00.000000000 +0100
+++ etc/hotplug/storage	2008-02-17 13:34:20.000000000 +0100
@@ -14,6 +14,9 @@
 MSTART=/sbin/start_mediasrv
 MSTOP=/sbin/stop_mediasrv
 
+# Do we have a Ftp server (ftpd) or was ist stripped from the firmware?
+_ftpd=$(basename $(which ftpd) 2> /dev/null)
+
 # Ask the system's configuration for the status of the readonly flag:
 # (Translates flag's yes/no to ro/rw)
 #
@@ -215,8 +218,10 @@
 	# ftp server control
 	ENABLED=`echo usbhost.ftp_server_enabled|ar7cfgctl -s`
 	if [ "$ENABLED" != "no" ] ; then
-		if [ "`pidof ftpd`" = "" ] ; then
-			ftpd -D -q -t 120 -m 15 -h "$CONFIG_PRODUKT_NAME" &
+		if [ "$_ftpd" ] ; then
+			if [ "`pidof ftpd`" = "" ] ; then
+				ftpd -D -q -t 120 -m 15 -h "$CONFIG_PRODUKT_NAME" &
+			fi
 		fi
 	else
 		killall ftpd
