--- linux-2.6.32/drivers/block/compcache/Makefile
+++ linux-2.6.32/drivers/block/compcache/Makefile
@@ -6,7 +6,7 @@
 ramzswap-objs	:=	ramzswap_drv.o xvmalloc.o
 
 ifneq ($(CONFIG_COMPCACHE),n)
-extra-y			+= rzscontrol ramzswap.sh
+#extra-y			+= rzscontrol ramzswap.sh
 
 rzscontrol-objs	:= rzscontrol.o
 
--- linux-2.6.32/drivers/rtc/Makefile
+++ linux-2.6.32/drivers/rtc/Makefile
@@ -91,7 +91,7 @@
 ifeq ($(CONFIG_RTC_DRV_AVM_REF_CLOCK),m)
 CFLAGS_rtc-avm.o += -DRTC_AVM_DEVICE_COUNT=1 # -DRTC_AVM_DEBUG=1
 
-extra-y			 += S09-rtc
+#extra-y			 += S09-rtc
 
 $(src)/S09-rtc: FORCE
 	rm -rf $(src)/S09-rtc
--- linux-2.6.32/drivers/net/avm_cpmac/lib/notify_avmnet/Makefile
+++ linux-2.6.32/drivers/net/avm_cpmac/lib/notify_avmnet/Makefile
@@ -3,7 +3,7 @@
 $(info avmnet/lib/notify: Guessing INSTALL_MOD_PATH: $(INSTALL_MOD_PATH))
 endif
 
-extra-y			+= notify_avmnet INSTALL_EXTRA_TARGETS
+#extra-y			+= notify_avmnet INSTALL_EXTRA_TARGETS
 clean-files		+= libtest libnotify_avmnet.so.0.1 libnotify_avmnet.so.0 libnotify_avmnet.so notify_avmnet.o
 
 quiet_cmd_target_cc_o_c = EXTRACC $@
