--- ext/gd/libgd/gd_compat.c
+++ ext/gd/libgd/gd_compat.c
@@ -33,3 +33,6 @@
 }
 #endif
 
+#if !HAVE_GD_BUNDLED
+#include "gd_security.c" // overflow2
+#endif
--- ext/gd/libgd/gd_security.c
+++ ext/gd/libgd/gd_security.c
@@ -15,7 +15,24 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <limits.h>
-#include "gd.h"
+#if HAVE_GD_BUNDLED
+# include "gd.h"
+#else
+# include <gd.h>
+#endif
+
+#if !HAVE_GD_BUNDLED
+#include "php.h"
+static void php_gd_error(const char *format, ...) {
+	va_list args;
+
+	TSRMLS_FETCH();
+
+	va_start(args, format);
+	php_verror(NULL, "", E_WARNING, format, args TSRMLS_CC);
+	va_end(args);
+}
+#endif
 
 int overflow2(int a, int b)
 {
