--- lib/gnutls.pc.in
+++ lib/gnutls.pc.in
@@ -18,7 +18,5 @@
 Description: Transport Security Layer implementation for the GNU system
 URL: http://www.gnutls.org/
 Version: @VERSION@
-Libs: -L${libdir} -lgnutls
-Libs.private: @LIBZ@ @LIBINTL@ @LIBSOCKET@ @LIBPTHREAD@ @LIBICONV@ @P11_KIT_LIBS@ @LIB_SELECT@ @TSS_LIBS@ @GMP_LIBS@
-@GNUTLS_REQUIRES_PRIVATE@
+Libs: -L${libdir} -lgnutls @LIBTASN1_LIBS@ @LIBZ@ @LIBINTL@ @LIBSOCKET@ @LIBPTHREAD@ @LIBICONV@ @P11_KIT_LIBS@ @LIB_SELECT@ @TSS_LIBS@ @HOGWEED_LIBS@ @NETTLE_LIBS@ @GMP_LIBS@
 Cflags: -I${includedir}
