--- lib/gnutls.pc.in
+++ lib/gnutls.pc.in
@@ -19,7 +19,5 @@
 Description: Transport Security Layer implementation for the GNU system
 URL: http://www.gnu.org/software/gnutls/
 Version: @VERSION@
-Libs: -L${libdir} -lgnutls
-Libs.private: @LTLIBGCRYPT@ @LTLIBNETTLE@ @NETTLE_LIBS@ @GNUTLS_ZLIB_LIBS_PRIVATE@
-@GNUTLS_REQUIRES_PRIVATE@
+Libs: -L${libdir} -lgnutls @LTLIBGCRYPT@ @LIBTASN1_LIBS@
 Cflags: -I${includedir}
--- libextra/gnutls-extra.pc.in
+++ libextra/gnutls-extra.pc.in
@@ -21,6 +21,5 @@
 URL: http://www.gnu.org/software/gnutls/
 Requires: gnutls
 Version: @VERSION@
-Libs: -L${libdir} -lgnutls-extra
-Libs.private: @LIBGNUTLS_EXTRA_LIBS@
+Libs: -L${libdir} -lgnutls-extra -lgnutls @LTLIBGCRYPT@ @LIBTASN1_LIBS@
 Cflags: -I${includedir}
