--- scripts/kconfig/mconf.c
+++ scripts/kconfig/mconf.c
@@ -588,7 +588,7 @@
 				if (sym_is_changeable(sym))
 					item_make("[%c]", val == no ? ' ' : '*');
 				else
-					item_make("-%c-", val == no ? ' ' : '*');
+					item_make("[%c]", val == no ? '-' : 'X');
 				item_set_tag('t');
 				item_set_data(menu);
 				break;
@@ -604,7 +604,7 @@
 					else
 						item_make("<%c>", ch);
 				} else
-					item_make("-%c-", ch);
+					item_make("[%c]", 'X');
 				item_set_tag('t');
 				item_set_data(menu);
 				break;
