--- scripts/kconfig/mconf.c
+++ scripts/kconfig/mconf.c
@@ -821,9 +821,9 @@
 		dialog_clear();
 		res = dialog_checklist(prompt ? prompt : "Main Menu",
 					radiolist_instructions,
-					MENUBOX_HEIGTH_MIN,
+					getmaxy(stdscr) - CHECKLIST_HEIGTH_MIN,
 					MENUBOX_WIDTH_MIN,
-					CHECKLIST_HEIGTH_MIN);
+					getmaxy(stdscr) - MENUBOX_HEIGTH_MIN);
 		selected = item_activate_selected();
 		switch (res) {
 		case 0:
