--- linux-2.6.32/include/linux/capi.h.orig	2010-04-08 13:00:21.000000000 +0200
+++ linux-2.6.32/include/linux/capi.h	2011-05-09 11:39:54.916470788 +0200
@@ -14,6 +14,7 @@
 
 #include <linux/types.h>
 #include <linux/ioctl.h>
+#include <linux/compiler.h>
 #ifndef __KERNEL__
 #include <linux/kernelcapi.h>
 #endif
--- linux-2.6.32/include/linux/kernelcapi.h.orig	2010-04-08 13:00:21.000000000 +0200
+++ linux-2.6.32/include/linux/kernelcapi.h	2011-05-09 11:39:54.916470788 +0200
@@ -45,6 +45,7 @@
 
 #ifdef __KERNEL__
 
+#include <linux/compiler.h>
 #include <linux/list.h>
 #include <linux/skbuff.h>
 #include <linux/workqueue.h>
