--- linux-2.6.32/net/atm/pppoatm.c
+++ linux-2.6.32/net/atm/pppoatm.c
@@ -319,8 +319,6 @@
 	atmvcc->user_back = pvcc;
 	atmvcc->push = pppoatm_push;
 	atmvcc->pop = pppoatm_pop;
-    if ( ppa_hook_mpoa_setup )
-        ppa_hook_mpoa_setup(atmvcc, 2, pvcc->encaps == e_llc ? 1 : 0);  //  PPPoA
 	__module_get(THIS_MODULE);
 #if defined(CONFIG_IFX_PPA_A6) || defined(CONFIG_IFX_PPA_A5) || defined(CONFIG_IFX_PPA_A4)
 	if ( ppa_hook_mpoa_setup )
