--- configure.orig	2008-03-12 19:36:07.000000000 +0100
+++ configure	2008-03-12 19:34:27.000000000 +0100
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #
 # Licensed to the Apache Software Foundation (ASF) under one or more
 # contributor license agreements.  See the NOTICE file distributed with
@@ -177,7 +177,7 @@
 ##
 ##  determine path to sh, it's not /bin/sh on ALL systems
 ##
-SHELL=/bin/sh
+SHELL=/bin/bash
 if [ ! -f "$SHELL" ]; then
     SHELL="`$aux/PrintPath sh`"
     if [ "x$SHELL" = "x" ]; then
