--- m4/package_utilies.m4
+++ m4/package_utilies.m4
@@ -27,7 +27,7 @@
 #
 # Generic macro, sets up all of the global build variables.
 # The following environment variables may be set to override defaults:
-#  CC MAKE LIBTOOL TAR ZIP MAKEDEPEND AWK SED ECHO SORT
+#  CC MAKE LIBTOOL TAR GZIP MAKEDEPEND AWK SED ECHO SORT
 #  MSGFMT MSGMERGE XGETTEXT RPM
 #
 AC_DEFUN([AC_PACKAGE_UTILITIES],
@@ -51,12 +51,12 @@
     fi
     tar=$TAR
     AC_SUBST(tar)
-    if test -z "$ZIP"; then
-        AC_PATH_PROG(ZIP, gzip,, /bin:/usr/bin:/usr/local/bin:/usr/freeware/bin)
+    if test -z "$GZIP"; then
+        AC_PATH_PROG(GZIP, gzip,, /bin:/usr/bin:/usr/local/bin:/usr/freeware/bin)
     fi
 
-    zip=$ZIP
-    AC_SUBST(zip)
+    gzip=$GZIP
+    AC_SUBST(gzip)
 
     if test -z "$MAKEDEPEND"; then
         AC_PATH_PROG(MAKEDEPEND, makedepend, /bin/true)
--- build/Makefile
+++ build/Makefile
@@ -45,7 +45,7 @@
 	    exit 1; \
 	else \
 	    unset TAPE; \
-	    $(TAR) -T $$CDIR/$@ -cf - | $(ZIP) --best > $$CDIR/$(SRCTAR); \
+	    $(TAR) -T $$CDIR/$@ -cf - | $(GZIP) --best > $$CDIR/$(SRCTAR); \
 	    echo Wrote: $$CDIR/$(SRCTAR); \
 	fi
 
--- build/tar/Makefile
+++ build/tar/Makefile
@@ -32,5 +32,5 @@
 		$$1 == "f" { printf (".%s\n", $$6); } \
 		$$1 == "d" { next; } \
 		$$1 == "l" { printf (".%s\n", $$3); }' \
-	| $(TAR) -T - -cf - | $(ZIP) --best > $$HERE/$(BINTAR)
+	| $(TAR) -T - -cf - | $(GZIP) --best > $$HERE/$(BINTAR)
 	@echo Wrote: `pwd`/$(BINTAR)
--- doc/Makefile
+++ doc/Makefile
@@ -28,7 +28,7 @@
 include $(BUILDRULES)
 
 CHANGES.gz:
-	$(ZIP) --best -c < CHANGES > $@
+	$(GZIP) --best -c < CHANGES > $@
 
 install: default
 	$(SUBDIRS_MAKERULE)
--- include/builddefs.in
+++ include/builddefs.in
@@ -53,7 +53,7 @@
 AWK		= @awk@
 SED		= @sed@
 TAR		= @tar@
-ZIP		= @zip@
+GZIP		= @gzip@
 MAKE		= @make@
 ECHO		= @echo@
 SORT		= @sort@
@@ -72,7 +72,7 @@
 ENABLE_SHARED	= @enable_shared@
 ENABLE_GETTEXT	= @enable_gettext@
 
-HAVE_ZIPPED_MANPAGES = @have_zipped_manpages@
+HAVE_GZIPPED_MANPAGES = @have_gzipped_manpages@
 
 ifneq "$(findstring $(PKG_PLATFORM), linux gnu gnu/kfreebsd gnu/knetbsd)" ""
 PCFLAGS = -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64
--- include/buildmacros
+++ include/buildmacros
@@ -113,8 +113,8 @@
 			[ -z "$$m" -o "$$m" = "\\" ] && continue; \
 			t=$(MAN_DEST)/$$m.$(MAN_SECTION); \
 			if $$first; then \
-				if $(HAVE_ZIPPED_MANPAGES); then \
-					$(ZIP) -9 -c $$d > $$d.gz; _sfx=.gz; \
+				if $(HAVE_GZIPPED_MANPAGES); then \
+					$(GZIP) -9 -c $$d > $$d.gz; _sfx=.gz; \
 				fi; \
 				u=$$m.$(MAN_SECTION)$$_sfx; \
 				echo $(INSTALL) -m 644 $${d}$$_sfx $${t}$$_sfx;\
@@ -141,7 +141,7 @@
 MAN_MAKERULE = \
 	@for f in *.[12345678] ""; do \
 		if test ! -z "$$f"; then \
-			$(ZIP) --best -c < $$f > $$f.gz; \
+			$(GZIP) --best -c < $$f > $$f.gz; \
 		fi; \
 	done
 
--- configure
+++ configure
@@ -789,7 +789,7 @@
 LIBOBJS
 ALLOCA
 libdirsuffix
-have_zipped_manpages
+have_gzipped_manpages
 rpmbuild
 RPMBUILD
 rpm_version
@@ -810,8 +810,8 @@
 AWK
 makedepend
 MAKEDEPEND
-zip
-ZIP
+gzip
+GZIP
 tar
 TAR
 make
@@ -11955,17 +11955,17 @@
     fi
     tar=$TAR
 
-    if test -z "$ZIP"; then
+    if test -z "$GZIP"; then
         # Extract the first word of "gzip", so it can be a program name with args.
 set dummy gzip; ac_word=$2
 { $as_echo "$as_me:$LINENO: checking for $ac_word" >&5
 $as_echo_n "checking for $ac_word... " >&6; }
-if test "${ac_cv_path_ZIP+set}" = set; then
+if test "${ac_cv_path_GZIP+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  case $ZIP in
+  case $GZIP in
   [\\/]* | ?:[\\/]*)
-  ac_cv_path_ZIP="$ZIP" # Let the user override the test with a path.
+  ac_cv_path_GZIP="$GZIP" # Let the user override the test with a path.
   ;;
   *)
   as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
@@ -11976,7 +11976,7 @@
   test -z "$as_dir" && as_dir=.
   for ac_exec_ext in '' $ac_executable_extensions; do
   if { test -f "$as_dir/$ac_word$ac_exec_ext" && $as_test_x "$as_dir/$ac_word$ac_exec_ext"; }; then
-    ac_cv_path_ZIP="$as_dir/$ac_word$ac_exec_ext"
+    ac_cv_path_GZIP="$as_dir/$ac_word$ac_exec_ext"
     $as_echo "$as_me:$LINENO: found $as_dir/$ac_word$ac_exec_ext" >&5
     break 2
   fi
@@ -11987,10 +11987,10 @@
   ;;
 esac
 fi
-ZIP=$ac_cv_path_ZIP
-if test -n "$ZIP"; then
-  { $as_echo "$as_me:$LINENO: result: $ZIP" >&5
-$as_echo "$ZIP" >&6; }
+GZIP=$ac_cv_path_GZIP
+if test -n "$GZIP"; then
+  { $as_echo "$as_me:$LINENO: result: $GZIP" >&5
+$as_echo "$GZIP" >&6; }
 else
   { $as_echo "$as_me:$LINENO: result: no" >&5
 $as_echo "no" >&6; }
@@ -11999,7 +11999,7 @@
 
     fi
 
-    zip=$ZIP
+    gzip=$GZIP
 
 
     if test -z "$MAKEDEPEND"; then
@@ -12508,11 +12508,11 @@
     fi
 
 
- have_zipped_manpages=false
+ have_gzipped_manpages=false
     for d in ${prefix}/share/man ${prefix}/man ; do
         if test -f $d/man1/man.1.gz
         then
-            have_zipped_manpages=true
+            have_gzipped_manpages=true
             break
         fi
     done
