--- Makefile.in
+++ Makefile.in
@@ -725,7 +725,7 @@
 @ENABLE_ERROR_INJECTION_TRUE@libinject_errors_la_LDFLAGS = -module
 generate_ver = ver="`{ $(srcdir)/scripts/version || echo '$(VERSION)' ; } | sed 's/.*/\#define JQ_VERSION \"&\"/'`"
 jq_SOURCES = src/main.c src/version.h
-jq_LDFLAGS = -static-libtool-libs $(am__append_7)
+jq_LDFLAGS =                      $(am__append_7)
 jq_LDADD = libjq.la -lm $(am__append_6)
 
 ### Tests (make check)
