--- linux-2.6.39/drivers/net/avm_cpmac/common/Makefile
+++ linux-2.6.39/drivers/net/avm_cpmac/common/Makefile
@@ -3,7 +3,7 @@
 # Copyright (C) 2011 AVM GmbH <fritzbox_info@avm.de>
 #
 
-AVMNET_VERSION = $(shell echo "`cat $(obj)/.avmnet-version` - `date -r $(obj)/.avmnet-version`")
+AVMNET_VERSION = "6.315-vr9_compatibility - Tue Mar 31 21:14:55 CEST 2020"

 obj-y	+= avmnet_config.o avmnet_common.o .avmnet-version.o
 
@@ -17,6 +17,7 @@
 quiet_cmd_set_executable = CHMOD+x $@
       cmd_set_executable = chmod +x $@
 
+ifdef DETERMINE_AVMNET_VERSION_USING_VCS
 .INTERMEDIATE: $(obj)/.avmnet-version.new
 
 $(obj)/.avmnet-version.new: FORCE
@@ -36,6 +37,12 @@
 	@echo "  GEN     $@"
 	@echo '#include <avmnet_common.h>' > $@
 	@echo 'const char *avmnet_version = "$(shell echo "`cat $<``date -r $<`")";' >> $@
+else
+$(obj)/.avmnet-version.c:
+	@echo "  GEN     $@"
+	@echo '#include <avmnet_common.h>' > $@
+	@echo 'const char *avmnet_version = "$(subst ",,$(AVMNET_VERSION))";' >> $@
+endif
 
 $(obj)/.avmnet-version.o: $(obj)/.avmnet-version.c
 
