--- configure.in
+++ configure.in
@@ -4,13 +4,11 @@
 AC_INIT([chan_dongle],[1.1],[http://code.google.com/p/asterisk-chan-dongle/issues/list],[chan_dongle],[http://code.google.com/p/asterisk-chan-dongle])
 PACKAGE_REVISION="34"
 AC_CANONICAL_TARGET
-AM_INIT_AUTOMAKE
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_SRCDIR([chan_dongle.c])
 AC_CANONICAL_SYSTEM
 AC_CANONICAL_HOST
 
-dnl AM_INIT_AUTOMAKE
 
 CPPFLAGS="$CPPFLAGS -D_GNU_SOURCE"
 
