--- Makefile.am
+++ Makefile.am
@@ -15,7 +15,7 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 ACLOCAL_AMFLAGS = -I m4 
-SUBDIRS=gnu lib src etc doc po
+SUBDIRS=gnu lib src etc
 
 # We never remove files from the configuration directory
 distuninstallcheck_listfiles = \
--- configure.ac
+++ configure.ac
@@ -104,8 +104,5 @@
 AC_CONFIG_FILES([Makefile
                  gnu/Makefile
 		 lib/Makefile
-		 src/Makefile
-		 etc/Makefile
-		 doc/Makefile
-		 po/Makefile.in])
+		 src/Makefile])
 AC_OUTPUT
