--- linux-2.6.32/arch/mips/kernel/process.c
+++ linux-2.6.32/arch/mips/kernel/process.c
@@ -267,6 +267,7 @@
             } else {
                 pg_fault_avg = 0;
             }
+            if(pcpuidle->fullrun > 9)
             printk(KERN_WARNING"[%u]system-load %d %s loadavg %lu.%lu %lu.%lu %lu.%lu - %d tasks:%lu %% curr:%s(%lu %%) max:%s(%lu %%, pid:%d), readytorun: %ld"
                     ", pgfault %lu/s (max %lu avg %lu.%lu)"
                     "\n", 
