--- configure
+++ configure
@@ -16641,6 +16641,7 @@
 $as_echo_n "checking if have htonll defined... " >&6; }
 
     have_htonll="no"
+if false; then
     if test "$cross_compiling" = yes; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
@@ -16679,6 +16680,7 @@
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
 fi
+fi
 
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $have_htonll" >&5
