--- src/plugin.c	2010-11-27 11:10:38.000000000 +0100
+++ src/plugin.c	2011-02-12 12:05:06.000000000 +0100
@@ -631,12 +631,6 @@
 					sstrerror (errno, errbuf, sizeof (errbuf)));
 			continue;
 		}
-		else if (!S_ISREG (statbuf.st_mode))
-		{
-			/* don't follow symlinks */
-			WARNING ("stat %s: not a regular file", filename);
-			continue;
-		}

 		if (plugin_load_file (filename, flags) == 0)
 		{
