--- comgt.c.orig	2006-10-20 16:49:46.000000000 +0200
+++ comgt.c	2010-05-23 11:00:06.545502659 +0200
@@ -1642,7 +1642,7 @@
       //scriptfile[0] = '\0';
     }
     if((fp=fopen(scriptfile,"r"))==NULL) {
-		strcpy(scriptfilepath,"/etc/comgt/");
+		strcpy(scriptfilepath,"/mod/etc/comgt/");
 		strcat(scriptfilepath,scriptfile);
 		if((fp=fopen(scriptfilepath,"r"))==NULL) {
       		sprintf(msg,"Could not open scriptfile \"%s\".\n",scriptfile);
