--- configure
+++ configure
@@ -716,14 +716,15 @@
     return 1;
 }
 EOF
-Z_U4=""
-if try $CC $CFLAGS $test.c -o $test && Z_U4=`./$test` && test -n "$Z_U4"; then
+# evaluated by running the test-program above on target
+Z_U4="unsigned"
+#if try $CC $CFLAGS $test.c -o $test && Z_U4=`./$test` && test -n "$Z_U4"; then
   sed < zconf.h "/#define Z_U4/s/\/\* \.\/configure may/#define Z_U4 $Z_U4   \/* .\/configure put the/" > zconf.temp.h
   mv zconf.temp.h zconf.h
   echo "Looking for a four-byte integer type... Found." | tee -a configure.log
-else
-  echo "Looking for a four-byte integer type... Not found." | tee -a configure.log
-fi
+#else
+#  echo "Looking for a four-byte integer type... Not found." | tee -a configure.log
+#fi
 
 # clean up files produced by running the compiler and linker
 rm -f $test.[co] $test $test$shared_ext $test.gcno
