--- linux-2.6.28/include/netpro/hostconfig.h
+++ linux-2.6.28/include/netpro/hostconfig.h
@@ -18,8 +18,8 @@
 //AVMbk
 //#include <defines.h>
 //#include <sysconf.h>
-#include "../../../drivers/inc/defines.h"
-#include "../../../ap_code/inc/sysconf.h"
+#include "../../fusiv_src/kernel/drivers/inc/defines.h"
+#include "../../fusiv_src/kernel/ap_code/inc/sysconf.h"
 
 		/* APU identifiers */
 #define MAC1_ID                         1
