--- Makefile.pre.in
+++ Makefile.pre.in
@@ -241,6 +241,8 @@
 ##########################################################################
 # Parser
 PGEN=		Parser/pgen$(EXE)
+# override for cross-compile builds
+PGEN_FOR_BUILD=$(PGEN)
 
 PSRCS=		\
 		Parser/acceler.c \
@@ -639,7 +641,7 @@
 $(GRAMMAR_H): $(GRAMMAR_INPUT) $(PGENSRCS)
 		@$(MKDIR_P) Include
 		$(MAKE) $(PGEN)
-		$(PGEN) $(GRAMMAR_INPUT) $(GRAMMAR_H) $(GRAMMAR_C)
+		$(PGEN_FOR_BUILD) $(GRAMMAR_INPUT) $(GRAMMAR_H) $(GRAMMAR_C)
 $(GRAMMAR_C): $(GRAMMAR_H) $(GRAMMAR_INPUT) $(PGENSRCS)
 		$(MAKE) $(GRAMMAR_H)
 		touch $(GRAMMAR_C)
