--- setup.py
+++ setup.py
@@ -618,6 +618,16 @@
         if host_platform in ['darwin', 'beos']:
             math_libs = []
 
+        # search for the 3rd-party libraries required to build modules in a directory
+        # specified by CROSS_TOOLCHAIN_SYSROOT environment variable
+        cross_toolchain_sysroot = os.getenv('CROSS_TOOLCHAIN_SYSROOT', '')
+        if cross_toolchain_sysroot:
+            if not cross_toolchain_sysroot.endswith(os.sep): cross_toolchain_sysroot += os.sep
+            print ('CROSS_TOOLCHAIN_SYSROOT: adding ' + cross_toolchain_sysroot + 'lib' + ' to lib_dirs')
+            lib_dirs += [ cross_toolchain_sysroot + 'lib' ]
+            print ('CROSS_TOOLCHAIN_SYSROOT: adding ' + cross_toolchain_sysroot + 'include' + ' to inc_dirs')
+            inc_dirs += [ cross_toolchain_sysroot + 'include' ]
+
         # XXX Omitted modules: gl, pure, dl, SGI-specific modules
 
         #
