From ac9be9b75658c9350ce058e6268a91db8ba6ecce Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Sun, 12 Feb 2017 21:17:24 +0100
Subject: * src/main.c: Remove double 'verbose' option

Fixes: #50290
---
 src/main.c | 1 -
 1 file changed, 1 deletion(-)

index 581a33d..311a91f 100644
--- src/main.c
+++ src/main.c
@@ -424,7 +424,6 @@ static struct cmdline_option option_data[] =
     { "user", 0, OPT_VALUE, "user", -1 },
     { "user-agent", 'U', OPT_VALUE, "useragent", -1 },
     { "verbose", 'v', OPT_BOOLEAN, "verbose", -1 },
-    { "verbose", 0, OPT_BOOLEAN, "verbose", -1 },
     { "version", 'V', OPT_FUNCALL, (void *) print_version, no_argument },
     { "wait", 'w', OPT_VALUE, "wait", -1 },
     { "waitretry", 0, OPT_VALUE, "waitretry", -1 },
