--- configure
+++ configure
@@ -33730,6 +33730,7 @@
 
 if test x"$with_ssl" = xopenssl; then :
 
+if test x"$with_zlib" != xno; then :
         { $as_echo "$as_me:${as_lineno-$LINENO}: checking for compress in -lz" >&5
 $as_echo_n "checking for compress in -lz... " >&6; }
 if ${ac_cv_lib_z_compress+:} false; then :
@@ -33774,6 +33775,7 @@
   LIBS="-lz $LIBS"
 
 fi
+fi
 
 
                     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for dlopen in -ldl" >&5
@@ -34570,6 +34572,7 @@
   if test x"$with_ssl" != xno; then :
 
 
+if test x"$with_zlib" != xno; then :
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for compress in -lz" >&5
 $as_echo_n "checking for compress in -lz... " >&6; }
 if ${ac_cv_lib_z_compress+:} false; then :
@@ -34614,6 +34617,7 @@
   LIBS="-lz $LIBS"
 
 fi
+fi
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for gpg_err_init in -lgpg-error" >&5
 $as_echo_n "checking for gpg_err_init in -lgpg-error... " >&6; }
