--- setup.py
+++ setup.py
@@ -9,7 +9,7 @@
 PY_PACKAGE = "curl"
 VERSION = "7.19.0.2"
 
-import glob, os, re, sys, string, subprocess
+import glob, os, re, sys, string
 import distutils
 from distutils.core import setup
 from distutils.extension import Extension
@@ -100,6 +100,7 @@
     # Run curl-config --libs and --static-libs.  Some platforms may not
     # support one or the other of these curl-config options, so gracefully
     # tolerate failure of either, but not both.
+    '''
     optbuf = ""
     for option in ["--libs", "--static-libs"]:
         p = subprocess.Popen("'%s' %s" % (CURL_CONFIG, option), shell=True,
@@ -111,6 +112,8 @@
         raise Exception("Neither curl-config --libs nor curl-config --static-libs" +
             " produced output")
     libs = split_quoted(optbuf)
+    '''
+    libs = split_quoted(os.popen("'%s' --libs" % CURL_CONFIG).read())
 
     for e in libs:
         if e[:2] == "-l":
