--- noip2.c.orig	2008-11-21 23:19:54.000000000 +0100
+++ noip2.c	2009-09-06 00:53:51.000000000 +0200
@@ -233,10 +233,10 @@
 #define GROUP			2
 #define DOMAIN			3
 #ifndef PREFIX
-  #define PREFIX		"/usr/local"
+  #define PREFIX		"/var/tmp/flash/mod"
 #endif
-#define CONFIG_FILEPATH		PREFIX"/etc"
-#define CONFIG_FILENAME		PREFIX"/etc/no-ip2.conf"
+#define CONFIG_FILEPATH		PREFIX""
+#define CONFIG_FILENAME		PREFIX"/no-ip2.conf"
 #define CONFSTRLEN		1024
 #define MAX_DEVLEN		16
 #define MAX_INSTANCE		4
