--- linux-2.6.32/net/netfilter/nf_conntrack_core.c
+++ linux-2.6.32/net/netfilter/nf_conntrack_core.c
@@ -541,7 +541,6 @@
 	struct net *net = nf_ct_net(ignored_conntrack);
 	struct nf_conntrack_tuple_hash *h;
 	struct hlist_nulls_node *n;
-	unsigned int hash = hash_conntrack(tuple);
 #ifdef CONFIG_ATHRS_HW_NAT
         int (*athr_tuple_taken)(const struct nf_conntrack_tuple *,
                                 const struct nf_conn *);
