--- Makefile.in
+++ Makefile.in
@@ -213,7 +213,7 @@
 use_convenience_ltdl_FALSE = @use_convenience_ltdl_FALSE@
 use_convenience_ltdl_TRUE = @use_convenience_ltdl_TRUE@
 @use_convenience_ltdl_TRUE@OPT_LTDL_DIR = libltdl
-SUBDIRS = $(OPT_LTDL_DIR) src doc scripts contrib
+SUBDIRS = $(OPT_LTDL_DIR) src     scripts contrib
 INCLUDES = $(LTDLINCL)
 EXTRA_DIST = TODO RELNOTES siproxd.spec autogen.sh
 all: config.h
