--- /dev/null
+++ usr/www/all/html/de/internet/labor_dsl.frm
@@ -0,0 +1,9 @@
+<input type="hidden" name="sar:settings/MaxDownstreamRate" value="<? query sar:settings/MaxDownstreamRate ?>" id="uiPostMaxDownstreamRate">
+<input type="hidden" name="sar:settings/MaxUpstreamRate" value="<? query sar:settings/MaxUpstreamRate ?>" id="uiPostMaxUpstreamRate">
+<input type="hidden" name="sar:settings/ControlBitfield" value="<? query sar:settings/ControlBitfield ?>" id="uiPostControlBitfield">
+<input type="hidden" name="sar:settings/RFI" value="<? query sar:settings/RFI ?>" id="uiPostRFI">
+<input type="hidden" name="sar:settings/DownstreamBlackoutBandStart" value="<? query sar:settings/DownstreamBlackoutBandStart ?>" id="uiPostDownstreamBlackoutBandStart">
+<input type="hidden" name="sar:settings/DownstreamBlackoutBandEnd" value="<? query sar:settings/DownstreamBlackoutBandEnd ?>" id="uiPostDownstreamBlackoutBandEnd">
+<input type="hidden" name="sar:settings/DownstreamPcbOffset" value="<? query sar:settings/DownstreamPcbOffset ?>" id="iPostDownstreamPcbOffset">
+<input type="hidden" name="sar:settings/UpstreamPcbOffset" value="<? query sar:settings/UpstreamPcbOffset ?>" id="iPostUpstreamPcbOffset">
+<input type="hidden" name="sar:settings/DownstreamMarginOffset" value="<? query sar:settings/DownstreamMarginOffset ?>" id="iPostDownstreamMarginOffset">
--- /dev/null
+++ usr/www/all/html/de/internet/labor_dsl.html
@@ -0,0 +1,103 @@
+<div id="content">
+<div class="backtitel"><div class="rundrt"><div class="rundlt"><div class="ecklb"><div class="eckrb"><div class="foretitel">
+DSL-Informationen
+</div></div></div></div></div></div>
+<div class="backframe"><div class="ecklm"><div class="eckrm"><div class="rundrb"><div class="rundlb"><div class="foredialog"><div class="ie_b10">
+<ul class="tabs">
+<li><a href="javascript:uiDoOverviewPage()">&Uuml;bersicht</a></li>
+<li><a href="javascript:uiDoADSLPage()">DSL</a></li>
+<li><a href="javascript:uiDoBitsPage()">Spektrum</a></li>
+<li><a href="javascript:uiDoATMPage()">Statistik</a></li>
+<li class="tabs_on"><div class="activtab">Einstellungen</div></li>
+<li><a href="javascript:uiDoFeedbackPage()">Feedback</a></li>
+</ul>
+<div class="backdialog"><div class="ecklm"><div class="eckrm"><div class="ecklb"><div class="eckrb"><div class="foredialog">
+<fieldset class="groups">
+<legend class="mb10">Bevorzugter Latenzpfad</legend>
+<div>
+<input type="radio" id="LP_Auto" name="bl" value="1">Automatisch</input>
+</div>
+<div>
+<input type="radio" id="LP_Fast" name="bl">Fast</input>
+</div>
+<div>
+<input type="radio" id="LP_Interleaved" name="bl">Interleaved</input>
+</div>
+</fieldset>
+<fieldset class="groups">
+<legend class="mb10">St&ouml;rungen durch Rundfunksender (RFI)</legend>
+<div class="paramline">
+<p><input type="checkbox" id="uiViewAutoRFI""/>Automatisch ermitteln</p>
+</div>
+<br>
+<div class="paramline">
+<p class="paramname"><label for="uiViewRFI">Zus&auml;tzlicher RFI Tr&auml;ger</label></p>
+<p><input type="text" size="5" maxlength="3" id="uiViewRFI" class="Eingabefeld"/></p>
+</div>
+</fieldset>
+<fieldset class="groups">
+<legend class="mb10">&Uuml;bertragungsraten</legend>
+<div class="paramline">
+<p class="paramname"><label for="uiViewMaxDownstreamRate">Maximale Downstream Rate (kbit/s)</label></p>
+<p><input type="text" size="5" maxlength="5" id="uiViewMaxDownstreamRate" class="Eingabefeld"/></p>
+</div>
+<div class="paramline">
+<p class="paramname"><label for="uiViewMaxUpstreamRate">Maximale Upstream Rate (kbit/s)</label></p>
+<p><input type="text" size="5" maxlength="5" id="uiViewMaxUpstreamRate" class="Eingabefeld"/></p>
+</div>
+</fieldset>
+<fieldset class="groups">
+<legend class="mb10">&Uuml;bertragungsbereich im Downstream ausblenden</legend>
+<div class="paramline">
+<p class="paramname"><label for="uiViewDownstreamBlackoutBandStart">Tr&auml;gerband ausblenden Start (Tr&auml;gernummer)</label></p>
+<p><input type="text" size="5" maxlength="5" id="uiViewDownstreamBlackoutBandStart" class="Eingabefeld"/></p>
+</div>
+<div class="paramline">
+<p class="paramname"><label for="uiViewDownstreamBlackoutBandEnd">Tr&auml;gerband ausblenden Ende (Tr&auml;gernummer)</label></p>
+<p><input type="text" size="5" maxlength="3" id="uiViewDownstreamBlackoutBandEnd" class="Eingabefeld"/></p>
+</div>
+</fieldset>
+<fieldset class="groups">
+<legend class="mb10">St&ouml;rabstandsmarge (SNR-Margin)</legend>
+<div class="paramline">
+<p>
+negative ganze Zahlen: kleinere St&ouml;rabstandsmarge (geringere Stabilit&auml;t) und  h&ouml;here Datenrate<br>
+positive ganze Zahlen: gr&ouml;&szlig;ere St&ouml;rabstandsmarge (h&ouml;here Stabilit&auml;t) und geringere Datenrate
+</p><br>
+<p class="paramname"><label for="iViewDownstreamMarginOffset">Ver&auml;ndere Downstream SNR-Margin um (ganze dB)</label></p>
+<p><input type="text" size="5" maxlength="3" id="iViewDownstreamMarginOffset" class="Eingabefeld"/></p>
+</div>
+</fieldset>
+<fieldset class="groups">
+<legend class="mb10">Anpassung der Sendeleistung/Powercutback (PCB)</legend>
+<div class="paramline">
+<p>
+negative ganze Zahlen: Verst&auml;rkung der Sendeleistung (wird nicht immer unterst&uuml;zt)<br>
+positive ganze Zahlen: Abschw&auml;chung der Sendeleistung
+</p>
+<br>
+<p class="paramname"><label for="iViewDownstreamPcbOffset">Ver&auml;ndere Downstream PCB um (ganze dB)</label></p>
+<p><input type="text" size="5" maxlength="3" id="iViewDownstreamPcbOffset" class="Eingabefeld"/></p>
+</div>
+<div class="paramline">
+<p class="paramname"><label for="iViewUpstreamPcbOffset">Ver&auml;ndere Upstream PCB um (ganze dB)</label></p>
+<p><input type="text" size="5" maxlength="3" id="iViewUpstreamPcbOffset" class="Eingabefeld"/></p>
+</div>
+</fieldset>
+<fieldset class="groups">
+<legend class="mb10">Diverses</legend>
+<div class="paramline">
+<p><input type="checkbox" id="uiViewOptLongLine"/>Optimierung f&uuml;r lange Leitungen aktivieren</p>
+</div>
+<div class="paramline">
+<p><input type="checkbox" id="uiViewOptShortLine"/>Optimierung f&uuml;r kurze Leitungen aktivieren</p>
+</div>
+<div class="paramline">
+<p><input type="checkbox" id="uiViewL2Off"/>Energiesparmodus f&uuml;r DSL aktivieren</p>
+</div>
+</fieldset>
+<!-- Trenner einbauen -->
+</div></div></div></div></div></div>
+<? include ../html/$var:lang/home/rbb_applycancelreset.html ?>
+</div></div></div></div></div></div></div>
+</div>
--- /dev/null
+++ usr/www/all/html/de/internet/labor_dsl.js
@@ -0,0 +1,159 @@
+<!-- ../internet/labor_dsl.js -->
+<? include ../html/$var:lang/internet/labor_dsl.inc ?>
+<? setvariable var:dslguiversion "<? query sar:status/gui_version_number ?>" ?>
+<STYLE type = "text/css">
+<!--
+<? if gt $var:dslguiversion 1 `
+.SlideDiv{margin-bottom:10px;}
+.SNR_td_tadio{
+width:40px;
+}
+` `
+.SlideDiv{margin-bottom:10px;}
+.SNR_td_tadio{
+width:20px;
+}
+` ?>
+.paramname { float: left; clear: both; width: 325px; text-align:left;margin-left: 2px;}
+.groups { margin-top: 5px;margin-bottom: 10px;}
+-->
+</STYLE>
+<script type="text/javascript">
+<? include ../html/$var:lang/js/jsl.js ?>
+<? include ../html/$var:lang/internet/awatch.js ?>
+function Select(id, index) {
+var obj = document.getElementById(id);
+if (obj) obj.selectedIndex = index;
+}
+function OnClickedOptDefault() {
+}
+function OnClickedOptPerformance() {
+}
+function OnClickedOptStability() {
+}
+function uiDoRefresh() {
+}
+var g_mldEingabeZahl = "Fehlerhafte Eingabe: Die Eingaben duerfen nur ganze Zahlen enthalten!";
+function valIsZahl(nummer) {
+if (nummer.match("[^0-9]") != null) return false;
+return true;
+}
+function valIsGanzeZahl(nummer) {
+if (nummer.match("[^\-0-9]") != null) return false;
+if (nummer.match("[0-9]-") != null) return false;
+if (nummer == "-" ) return false;
+return true;
+}
+function uiDoOnLoad() {
+var rfi = document.getElementById("uiViewAutoRFI");
+var optlong = document.getElementById("uiViewOptLongLine");
+var optshort = document.getElementById("uiViewOptShortLine");
+var l2off = document.getElementById("uiViewL2Off");
+var ctlbits = jslGetValue("uiPostControlBitfield");
+if (ctlbits & 1)
+rfi.checked=false;
+else
+rfi.checked=true;
+if (ctlbits & 8)
+optshort.checked=false;
+else
+optshort.checked=true;
+if (ctlbits & 16)
+optlong.checked=false;
+else
+optlong.checked=true;
+if (ctlbits & 32)
+l2off.checked=false;
+else
+l2off.checked=true;
+var auto = document.getElementById("LP_Auto");
+var fast = document.getElementById("LP_Fast");
+var interleaved = document.getElementById("LP_Interleaved");
+if (ctlbits & 2)
+fast.checked=true;
+else if (ctlbits & 4)
+interleaved.checked=true;
+else
+auto.checked=true;
+jslCopyValue("uiViewMaxDownstreamRate","uiPostMaxDownstreamRate");
+jslCopyValue("uiViewMaxUpstreamRate","uiPostMaxUpstreamRate");
+jslCopyValue("uiViewRFI","uiPostRFI");
+jslCopyValue("uiViewDownstreamBlackoutBandStart","uiPostDownstreamBlackoutBandStart");
+jslCopyValue("uiViewDownstreamBlackoutBandEnd","uiPostDownstreamBlackoutBandEnd");
+jslCopyValue("iViewDownstreamMarginOffset","iPostDownstreamMarginOffset");
+document.getElementById("iViewDownstreamMarginOffset").value /= 10;
+jslCopyValue("iViewDownstreamPcbOffset","iPostDownstreamPcbOffset");
+jslCopyValue("iViewUpstreamPcbOffset","iPostUpstreamPcbOffset");
+}
+function uiDoHelp() {
+jslPopHelp("hilfe_dslinfo_einstellungen");
+}
+function uiDoSave() {
+alert("Denken Sie daran, die Einstellungen zurueckzusetzen, bevor Sie eine Firmware installieren, die die DSL Experteneinstellungen nicht enthaelt. Beachten Sie bitte auch die Hinweise in der Hilfe. Die Internetverbindung wird nun kurz getrennt.");
+var z = 0;
+var rfi = document.getElementById("uiViewAutoRFI");
+var optlong = document.getElementById("uiViewOptLongLine");
+var optshort = document.getElementById("uiViewOptShortLine");
+var l2off = document.getElementById("uiViewL2Off");
+var ctlbits = jslGetValue("uiPostControlBitfield");
+if (rfi.checked)
+ctlbits &= ~1;
+else
+ctlbits |= 1;
+if (optlong.checked)
+ctlbits &= ~16;
+else
+ctlbits |= 16;
+if (optshort.checked)
+ctlbits &= ~8;
+else
+ctlbits |= 8;
+if (l2off.checked)
+ctlbits &= ~(32);
+else
+ctlbits |= 32;
+var fast = document.getElementById("LP_Fast");
+var interleaved = document.getElementById("LP_Interleaved");
+ctlbits &= ~6;
+if (fast.checked)
+ctlbits |= 2;
+else if (interleaved.checked)
+ctlbits |= 4;
+jslSetValue("uiPostControlBitfield", ctlbits);
+( valIsZahl(jslGetValue("uiViewMaxDownstreamRate")) ? jslCopyValue("uiPostMaxDownstreamRate", "uiViewMaxDownstreamRate" ) : z++);
+( valIsZahl(jslGetValue("uiViewMaxUpstreamRate")) ? jslCopyValue("uiPostMaxUpstreamRate", "uiViewMaxUpstreamRate" ) : z++);
+( valIsZahl(jslGetValue("uiViewRFI")) ? jslCopyValue("uiPostRFI","uiViewRFI") : z++);
+( valIsZahl(jslGetValue("uiViewDownstreamBlackoutBandStart")) ? jslCopyValue("uiPostDownstreamBlackoutBandStart","uiViewDownstreamBlackoutBandStart") : z++);
+( valIsZahl(jslGetValue("uiViewDownstreamBlackoutBandEnd")) ? jslCopyValue("uiPostDownstreamBlackoutBandEnd","uiViewDownstreamBlackoutBandEnd") : z++);
+( valIsGanzeZahl(jslGetValue("iViewDownstreamMarginOffset")) ? jslCopyValue("iPostDownstreamMarginOffset","iViewDownstreamMarginOffset") : z++);
+document.getElementById("iPostDownstreamMarginOffset").value *= 10;
+( valIsGanzeZahl(jslGetValue("iViewDownstreamPcbOffset")) ? jslCopyValue("iPostDownstreamPcbOffset","iViewDownstreamPcbOffset") : z++);
+( valIsGanzeZahl(jslGetValue("iViewUpstreamPcbOffset")) ? jslCopyValue("iPostUpstreamPcbOffset","iViewUpstreamPcbOffset") : z++);
+if (z>0) alert(g_mldEingabeZahl);
+jslFormSubmitEx("internet", "labor_dsl");
+}
+function uiDoCancel() {
+jslGoTo("internet", "overview");
+}
+function uiDoReset() {
+var res = confirm("Wollen Sie alle Einstellungen zurücksetzen?");
+if (res == false) {
+return;
+}
+jslSetValue("uiPostMaxDownstreamRate", "0" );
+jslSetValue("uiPostMaxUpstreamRate", "0" );
+jslSetValue("uiPostRFI","0");
+jslSetValue("uiPostDownstreamBlackoutBandStart","0");
+jslSetValue("uiPostDownstreamBlackoutBandEnd","0");
+jslSetValue("iPostDownstreamMarginOffset", 0);
+jslSetValue("iPostDownstreamPcbOffset", 0);
+jslSetValue("iPostUpstreamPcbOffset", 0);
+jslSetValue("uiPostControlBitfield",0);
+alert("Die Standardeinstellungen werden nun wiederhergestellt und die Internetverbindung wird dazu nun kurz getrennt!");
+jslFormSubmitEx("internet", "labor_dsl");
+}
+function uiDoFeedback()
+{
+jslGoTo('home','feedback');
+}
+</script>
