--- lib/zebra.h.orig	2009-11-30 12:50:11.000000000 +0100
+++ lib/zebra.h	2009-11-30 12:50:16.000000000 +0100
@@ -25,6 +25,26 @@
 #include "config.h"
 #endif /* HAVE_CONFIG_H */
 
+#ifdef HAVE_IPV6
+#if defined IPV6_RECVHOPLIMIT || defined IPV6_RECVPKTINFO
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,14)
+#if defined IPV6_RECVHOPLIMIT && defined IPV6_2292HOPLIMIT
+#undef IPV6_RECVHOPLIMIT
+#undef IPV6_HOPLIMIT
+#define IPV6_RECVHOPLIMIT IPV6_2292HOPLIMIT
+#define IPV6_HOPLIMIT IPV6_2292HOPLIMIT
+#endif
+#if defined IPV6_RECVPKTINFO && defined IPV6_2292PKTINFO
+#undef IPV6_RECVPKTINFO
+#undef IPV6_PKTINFO
+#define IPV6_RECVPKTINFO IPV6_2292PKTINFO
+#define IPV6_PKTINFO IPV6_2292PKTINFO
+#endif
+#endif
+#endif
+#endif
+
 #ifdef SUNOS_5
 #define _XPG4_2
 #define __EXTENSIONS__
--- zebra/kernel_null.c
+++ zebra/kernel_null.c
@@ -3,6 +3,8 @@
 #include <zebra.h>
 #include <log.h>
 
+#include "prefix.h"
+
 #include "zebra/zserv.h"
 #include "zebra/rt.h"
 #include "zebra/redistribute.h"
@@ -10,11 +12,13 @@
 
 int kernel_add_ipv4 (struct prefix *a, struct rib *b) { return 0; }
 #pragma weak kernel_delete_ipv4 = kernel_add_ipv4
+#ifdef HAVE_IPV6
 int kernel_add_ipv6 (struct prefix *a, struct rib *b) { return 0; }
 #pragma weak kernel_delete_ipv6 = kernel_add_ipv6
 int kernel_delete_ipv6_old (struct prefix_ipv6 *dest, struct in6_addr *gate,
                             unsigned int index, int flags, int table)
 { return 0; }
+#endif
 
 int kernel_add_route (struct prefix_ipv4 *a, struct in_addr *b, int c, int d)
 { return 0; }
