--- linux-2.6.32/arch/mips/kernel/process.c	2011-12-05 12:48:04.000000000 +0100
+++ linux-2.6.32/arch/mips/kernel/process.c	2012-05-10 00:53:59.138054377 +0200
@@ -230,6 +230,7 @@
             } else {
                 pg_fault_avg = 0;
             }
+            if(pcpuidle->fullrun > 1)
             printk("[%lu][%u]system-load %d %s %d tasks:%lu %% curr:%s(%lu %%) max:%s(%lu %%, pid:%d), readytorun: %ld"
                     ", pgfault %lu/s (max %lu avg %lu.%lu)"
                     "\n", 
