--- source/configure.orig	2009-09-30 15:08:58.000000000 +0200
+++ source/configure	2010-04-02 18:47:28.000000000 +0200
@@ -37218,7 +37218,7 @@
 See \`config.log' for more details." >&5
 $as_echo "$as_me: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }; }
+   }; }
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
@@ -49549,7 +49549,7 @@
 See \`config.log' for more details." >&5
 $as_echo "$as_me: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }; }
+   }; }
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
