--- etc/init.d/rc.S.orig	2007-06-08 13:17:02.000000000 +0200
+++ etc/init.d/rc.S	2007-06-25 13:28:31.000000000 +0200
@@ -374,6 +374,8 @@
  mknod /var/flash/calllog c $tffs_major $((0x8D))
  mknod /var/flash/aura-usb c $tffs_major $((0xA0))
  mknod /var/flash/browser-data c $tffs_major $((0xA8))
+ mknod /var/flash/ds_mod c $tffs_major $((0x3c))
+ /usr/bin/modload 2>&1 | tee /var/log/mod_load.log
 #file "stdin", 9
 #file "./arch/etc/init.d/rc.cpmac.sh", 0
 ##########################################################################################
