--- configure
+++ configure
@@ -11727,8 +11727,8 @@
         case $host in
         *-*-linux*)
                 DEBUG="-g -Wall -fsigned-char"
-                CFLAGS="-O20 -Wall -ffast-math -fsigned-char"
-                PROFILE="-Wall -W -pg -g -O20 -ffast-math -fsigned-char"
+                CFLAGS="-Wall -ffast-math -fsigned-char"
+                PROFILE="-Wall -W -pg -g -ffast-math -fsigned-char"
 		;;
         sparc-sun-*)
                 DEBUG="-g -Wall -fsigned-char"
@@ -11742,8 +11742,8 @@
 		;;
         *)
                 DEBUG="-g -Wall -fsigned-char"
-                CFLAGS="-O20 -fsigned-char"
-                PROFILE="-O20 -g -pg -fsigned-char"
+                CFLAGS="-fsigned-char"
+                PROFILE="-g -pg -fsigned-char"
 		;;
         esac
 fi
