From: Brent Bottles
Subject: Fix leak in gdImageCopyResized
Origin: upstream, http://cvs.php.net/viewvc.cgi/gd/libgd/src/gd.c?r1=1.49.2.24&r2=1.49.2.25
Bug: http://bugs.libgd.org/145
Last-Update: 20091105
--- NEWS
+++ NEWS
@@ -1,6 +1,7 @@
 'GD                                                                         NEWS
 |||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
 GD 2.0.36 (2007-11-xx)
+145, Fixed keak in gdImageCopyResized (Patch by Brent Bottles)
  81, Fixed gdImageCopy with true color image, the transparent color was ignored
      (Pierre, Mark Fisher)
  88, Fixed support of PNG grayscale image with alpha channel (Pierre)
--- gd.c
+++ gd.c
@@ -1,4 +1,4 @@
-/* $Id: gd.c,v 1.49.2.24 2007/11/20 13:42:11 pajoye Exp $ */
+/* $Id: gd.c,v 1.49.2.25 2008/02/01 01:52:25 pajoye Exp $ */
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
@@ -2486,6 +2486,7 @@
 
   sty = (int *) gdMalloc (sizeof (int) * srcH);
 	if (!sty) {
+		gdFree(stx);
 		return;
 	}
 
