From 63aae8a69bd2052e8601a957f31c3d5873829b89 Mon Sep 17 00:00:00 2001
From: Pierre Joye <pajoye@php.net>
Date: Mon, 3 Jan 2011 09:22:58 +0000
Subject: [PATCH] - stream output support for image outputs functions

diff --git ext/gd/libgd/gd_io.h ext/gd/libgd/gd_io.h
index b893751..a4d66bb 100644
--- gd_io.h
+++ gd_io.h
@@ -26,6 +26,7 @@ typedef struct gdIOCtx {
 
   void (*gd_free) (struct gdIOCtx *);
 
+  void *data;
 }
 gdIOCtx;
 
