--- libavutil/libm.h
+++ libavutil/libm.h
@@ -454,7 +454,15 @@
 }
 #endif /* HAVE_ROUNDF */
 
+#if defined(__UCLIBC__) && __UCLIBC_MAJOR__ == 0 && __UCLIBC_MINOR__ == 9 && __UCLIBC_SUBLEVEL__ <= 29
+#define NO_TRUNC_IN_UCLIBC
+#endif
+
 #if !HAVE_TRUNC
+#ifdef NO_TRUNC_IN_UCLIBC
+#undef truncf
+#define trunc ffmpeg_trunc
+#endif
 static av_always_inline av_const double trunc(double x)
 {
     return (x > 0) ? floor(x) : ceil(x);
@@ -462,6 +470,10 @@
 #endif /* HAVE_TRUNC */
 
 #if !HAVE_TRUNCF
+#ifdef NO_TRUNC_IN_UCLIBC
+#undef truncf
+#define truncf ffmpeg_truncf
+#endif
 static av_always_inline av_const float truncf(float x)
 {
     return (x > 0) ? floor(x) : ceil(x);
