--- linux-2.6.19.2/drivers/char/avm_new/Makefile.26.orig	2007-06-05 15:23:33.000000000 +0200
+++ linux-2.6.19.2/drivers/char/avm_new/Makefile.26	2007-12-21 22:18:54.000000000 +0100
@@ -97,3 +97,5 @@
 obj-$(CONFIG_AVM_SAMMEL) += avm.o 
 
 # EXTRA_CFLAGS	+= -g -ggdb -save-temps
+EXTRA_CFLAGS    += -DAVM_NEW_HWREV_LIST="\" 118 117 110 114 129 115 108 94 127 122 101 120 102 121 104 103 105 95 107 106 128 79 112 111 \""
+
