--- linux-2.6.19.2/drivers/isdn/capi_oslib/ca_sched.c.orig	2007-05-08 18:29:22.000000000 +0200
+++ linux-2.6.19.2/drivers/isdn/capi_oslib/ca_sched.c	2007-12-21 23:21:14.000000000 +0100
@@ -5,14 +5,8 @@
 #include <linux/string.h>
 #include <linux/interrupt.h>
 #include <linux/timer.h>
-#if defined(CONFIG_MIPS_UR8)
-#include <linux/pcmlink_ul.h>
-#define ubik2_tasklet_control_enter_critical pcmlink_tasklet_control_enter_critical
-#define ubik2_tasklet_control_leave_critical pcmlink_tasklet_control_leave_critical
-#else/*--- #if defined(CONFIG_MIPS_UR8) ---*/
 #include <linux/ubik2_interface.h>
 
-#endif/*--- #else ---*//*--- #if defined(CONFIG_MIPS_UR8) ---*/
 #include <asm/atomic.h>
 #include <asm/semaphore.h>
 #include <linux/new_capi.h>
