--- linux-2.6.19/drivers/net/avm_cpmac/cpmac_debug.h
+++ linux-2.6.19/drivers/net/avm_cpmac/cpmac_debug.h
@@ -39,14 +39,11 @@
 /*--- #define DEB_SUPPORT(a...) ---*/
 #define DEB_ERR(a...) printk(KERN_ERR "[cpmac] " a);
 #define DEB_WARN(a...) printk(KERN_WARNING "[cpmac] " a);
-/*--- #define DEB_INFO(a...) printk (KERN_ERR "[cpmac] " a); ---*/
-#define DEB_INFO(a...)
-/*--- #define DEB_INFOTRC(a...) printk(KERN_ERR "[cpmac] " a); ---*/
-#define DEB_INFOTRC(a...) 
+#define DEB_INFO(a...) printk (KERN_ERR "[cpmac] " a);
+#define DEB_INFOTRC(a...) printk(KERN_ERR "[cpmac] " a);
 /*--- #define DEB_TRC(a...) printk(KERN_ERR "[cpmac] " a); ---*/
 #define DEB_TRC(a...)
-/*--- #define DEB_TEST(a...) DEB_ERR("[test] " a); ---*/
-#define DEB_TEST(a...)
+#define DEB_TEST(a...) DEB_ERR("[test] " a);
 
 #define assert(i)  BUG_ON(!(i))
 /*--- #define assert(exp) ---*/
