--- linux-2.6.19.2/drivers/usb/core/sysfs.c	2007-01-11 08:38:19.000000000 +0100
+++ linux-2.6.19.2/drivers/usb/core/sysfs.c	2013-02-03 01:05:35.570144915 +0100
@@ -343,11 +343,14 @@
 {
 	struct usb_host_interface *iface_desc;
 	int i;
+/*	printk("REJECTED usb_create_intf_ep_files()\n");	*/
 
 	iface_desc = intf->cur_altsetting;
+/*
 	for (i = 0; i < iface_desc->desc.bNumEndpoints; ++i)
 		usb_create_ep_files(&intf->dev, &iface_desc->endpoint[i],
 				udev);
+*/
 }
 
 static inline void usb_remove_intf_ep_files(struct usb_interface *intf)
