--- linux-2.6.19.2/init/main.c.orig	2007-08-16 13:23:39.000000000 +0200
+++ linux-2.6.19.2/init/main.c	2007-11-28 23:55:10.000000000 +0100
@@ -106,7 +106,7 @@
 enum system_states system_state;
 EXPORT_SYMBOL(system_state);
 
-extern int avm_led_load_config(void);
+/* extern int avm_led_load_config(void); */
 
 /*
  * Boot command-line arguments
@@ -783,7 +783,7 @@
 		printk(KERN_WARNING "Warning: unable to open an initial console.\n");
 
 #if defined(CONFIG_MIPS_OHIO) || defined(CONFIG_MIPS_AR7) || defined(CONFIG_MIPS_UR8)
-    avm_led_load_config(); /* HACK! Find a better point to start it! */
+/*    avm_led_load_config();*/ /* HACK! Find a better point to start it! */
 #endif /*--- #if defined(CONFIG_MIPS_OHIO) || defined(CONFIG_MIPS_AR7) || defined(CONFIG_MIPS_UR8) ---*/
 
 	(void) sys_dup(0);
