--- libpcre/chartables.c.orig	2005-01-08 13:00:26.000000000 +0100
+++ libpcre/chartables.c	2005-01-08 12:59:49.000000000 +0100
@@ -0,0 +1,13 @@
+/*************************************************
+*      Perl-Compatible Regular Expressions       *
+*************************************************/
+
+/* This file is automatically written by the dftables auxiliary 
+program. If you edit it by hand, you might like to edit the Makefile to 
+prevent its ever being regenerated.
+
+This file is #included in the compilation of pcre.c to build the default
+character tables which are used when no tables are passed to the compile
+function. */
+
+/* End of chartables.c */
