--- ev.c
+++ ev.c
@@ -3078,27 +3078,35 @@
 /*****************************************************************************/
 
 #if EV_USE_IOCP
+# warning using "ev_iocp.c"
 # include "ev_iocp.c"
 #endif
 #if EV_USE_PORT
+# warning using "ev_port.c"
 # include "ev_port.c"
 #endif
 #if EV_USE_KQUEUE
+# warning using "ev_kqueue.c"
 # include "ev_kqueue.c"
 #endif
 #if EV_USE_EPOLL
+# warning using "ev_epoll.c"
 # include "ev_epoll.c"
 #endif
 #if EV_USE_LINUXAIO
+# warning using "ev_linuxaio.c"
 # include "ev_linuxaio.c"
 #endif
 #if EV_USE_IOURING
+# warning using "ev_iouring.c"
 # include "ev_iouring.c"
 #endif
 #if EV_USE_POLL
+# warning using "ev_poll.c"
 # include "ev_poll.c"
 #endif
 #if EV_USE_SELECT
+# warning using "ev_select.c"
 # include "ev_select.c"
 #endif
 
