--- src/Makefile.orig	2009-02-07 15:51:29.000000000 +0100
+++ src/Makefile	2009-02-07 15:51:33.000000000 +0100
@@ -1,5 +1,5 @@
 # remove -fno-exceptions  if it causes problems with stdio library on some platforms (ARM)
-
+CXX=mipsel-linux-g++
 PREFIX=/usr
 BINDIR=$(PREFIX)/bin
 INCDIR=$(PREFIX)/include/espeak
@@ -24,7 +24,7 @@
 LN_SF = /bin/ln -sf
 MKDIR = mkdir -p
 
-AUDIO = portaudio
+#AUDIO = portaudio
 #AUDIO = pulseaudio
 #AUDIO = sada
 
@@ -60,17 +60,17 @@
 
 SRCS1=$(speak_SOURCES)
 OBJS1=$(patsubst %.cpp,%.o,$(SRCS1))
-LIBS1=-lstdc++ $(LIB_AUDIO) -lpthread $(EXTRA_LIBS)
+LIBS1=-static -lstdc++ $(EXTRA_LIBS)
 
 SRCS2=$(libespeak_SOURCES)
 OBJS2=$(patsubst %.cpp,x_%.o,$(SRCS2))
-LIBS2=-lstdc++ $(LIB_AUDIO) -lpthread
+LIBS2=-static -lstdc++ -Wl,-Bdynamic -lpthread
 
 SRCS3 = espeak.cpp
 OBJS3=$(patsubst %.cpp,%.o,$(SRCS3))
-LIBS3=-lstdc++ -L . -lespeak
+LIBS3=-lstdc++ -L . -lespeak
 
-CXXFLAGS=-O2
+CXXFLAGS=-Os -march=mips32 -mips32 -pipe
 
 
 all: $(BIN_NAME) $(LIB_NAME) $(STATIC_LIB_NAME) $(BIN2_NAME)
