--- etc/init.d/rc.S	2006-03-10 13:47:39.000000000 +0100
+++ etc/init.d/rc.S	2006-03-10 13:47:59.000000000 +0100
@@ -431,8 +431,7 @@
 fi
  echo "usb storage: nothing to do ...."
  tmp_iso_path=/lib/modules/cdrom.iso
- modprobe avalanche_usb use_cdrom=1 iso_path=${tmp_iso_path} cdrom_fallback=1
- echo "usb client: CDROM Fallback=$CONFIG_CDROM_FALLBACK ISO=${tmp_iso_path}"
+ modprobe avalanche_usb use_cdrom=0
 #file "stdin", 35
 #file "./arch/etc/init.d/rc.hostname.sh", 0
 HOSTNAME=`echo servercfg.hostname | ar7cfgctl -s`
