--- etc/init.d/rc.S.orig	2006-03-10 13:47:39.000000000 +0100
+++ etc/init.d/rc.S	2006-03-10 13:47:59.000000000 +0100
@@ -438,7 +438,7 @@
 #file "stdin", 30
 #file "./arch/etc/init.d/rc.usb.sh", 0
  tmp_iso_path=/lib/modules/cdrom.iso
- modprobe avalanche_usb use_cdrom=1 iso_path=${tmp_iso_path} cdrom_fallback=0
+ modprobe avalanche_usb use_cdrom=0
  echo "usb client: CDROM Fallback=$CONFIG_CDROM_FALLBACK ISO=${tmp_iso_path}"
 #file "stdin", 34
 #file "./arch/etc/init.d/rc.hostname.sh", 0
