--- tunables.c.orig	2008-07-30 03:52:23.000000000 +0200
+++ tunables.c	2008-11-18 02:49:39.000000000 +0100
@@ -104,19 +104,19 @@
 /* -rw------- */
 unsigned int tunable_chown_upload_mode = 0600;
 
-const char* tunable_secure_chroot_dir = "/usr/share/empty";
+const char* tunable_secure_chroot_dir = "/var/run/vsftpd";
 const char* tunable_ftp_username = "ftp";
 const char* tunable_chown_username = "root";
 const char* tunable_xferlog_file = "/var/log/xferlog";
 const char* tunable_vsftpd_log_file = "/var/log/vsftpd.log";
 const char* tunable_message_file = ".message";
-const char* tunable_nopriv_user = "nobody";
+const char* tunable_nopriv_user = "root";
 const char* tunable_ftpd_banner = 0;
-const char* tunable_banned_email_file = "/etc/vsftpd.banned_emails";
-const char* tunable_chroot_list_file = "/etc/vsftpd.chroot_list";
+const char* tunable_banned_email_file = "/mod/etc/vsftpd.banned_emails";
+const char* tunable_chroot_list_file = "/mod/etc/vsftpd.chroot_list";
 const char* tunable_pam_service_name = "ftp";
 const char* tunable_guest_username = "ftp";
-const char* tunable_userlist_file = "/etc/vsftpd.user_list";
+const char* tunable_userlist_file = "/mod/etc/vsftpd.user_list";
 const char* tunable_anon_root = 0;
 const char* tunable_local_root = 0;
 const char* tunable_banner_file = 0;
@@ -129,7 +129,7 @@
 const char* tunable_deny_file = 0;
 const char* tunable_user_sub_token = 0;
 const char* tunable_email_password_file = "/etc/vsftpd.email_passwords";
-const char* tunable_rsa_cert_file = "/usr/share/ssl/certs/vsftpd.pem";
+const char* tunable_rsa_cert_file = "/var/tmp/flash/vsftpd.pem";
 const char* tunable_dsa_cert_file = 0;
 const char* tunable_ssl_ciphers = "DES-CBC3-SHA";
 const char* tunable_rsa_private_key_file = 0;
