--- sysdeputil.c	2005-05-24 00:34:23.000000000 +0200
+++ sysdeputil.c	2007-03-02 17:50:30.969786936 +0100
@@ -159,7 +159,6 @@
 #include <linux/capability.h>
 #include <errno.h>
 #include <syscall.h>
-_syscall2(int, capset, cap_user_header_t, header, const cap_user_data_t, data)
 /* Gross HACK to avoid warnings - linux headers overlap glibc headers */
 #undef __NFDBITS
 #undef __FDMASK
Only in vsftpd.old: vsftpd-2.0.4
