--- twoprocess.c.orig	2009-01-15 02:03:04.000000000 +0100
+++ twoprocess.c	2009-05-06 13:30:33.457287613 +0200
@@ -372,13 +372,6 @@
     str_free(&chroot_str);
     str_free(&chdir_str);
     str_free(&userdir_str);
-    /* Guard against the config error of having the anonymous ftp tree owned
-     * by the user we are running as
-     */
-    if (was_anon && vsf_sysutil_write_access("/"))
-    {
-      die("vsftpd: refusing to run with writable anonymous root");
-    }
     p_sess->is_anonymous = anon;
     process_post_login(p_sess);
     bug("should not get here: common_do_login");
