From 4c67d036565cd93388389b1693f0d72551dde6f7 Mon Sep 17 00:00:00 2001
From: Michele Giacomoli <michele.giacomoli@mynet.it>
Date: Mon, 19 Sep 2022 16:50:43 +0200
Subject: [PATCH] Fix utx var typo

---
 loginrec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/loginrec.c b/loginrec.c
index af10d959c..f93e12e9f 100644
--- loginrec.c
+++ loginrec.c
@@ -829,7 +829,7 @@
 		return 0;
 	}
 # else
-	if (!utmpx_write_direct(li, &ut)) {
+	if (!utmpx_write_direct(li, &utx)) {
 		dropbear_log(LOG_WARNING, "utmpx_perform_login: utmp_write_direct() failed");
 		return 0;
 	}
