--- extra/Configs/Config.mips
+++ extra/Configs/Config.mips
@@ -64,6 +64,12 @@
 config CONFIG_MIPS_ISA_MIPS32
 	bool "MIPS32"
 
+config CONFIG_MIPS_ISA_MIPS32_4KC
+	bool "MIPS32 4KC (FRITZ!Box based on SoCs like AR7 or UR8)"
+
+config CONFIG_MIPS_ISA_MIPS32_24KC
+	bool "MIPS32 24KC (FRITZ!Box based on SoCs like AR9, IKS or VR9)"
+
 config CONFIG_MIPS_ISA_MIPS32R2
 	bool "MIPS32r2"
 
--- Rules.mak
+++ Rules.mak
@@ -363,6 +363,8 @@
 	CPU_CFLAGS-$(CONFIG_MIPS_ISA_3)+=-mips3 -mtune=mips3
 	CPU_CFLAGS-$(CONFIG_MIPS_ISA_4)+=-mips4 -mtune=mips4
 	CPU_CFLAGS-$(CONFIG_MIPS_ISA_MIPS32)+=-mips32 -mtune=mips32
+	CPU_CFLAGS-$(CONFIG_MIPS_ISA_MIPS32_4KC)+=-march=4kc -mtune=4kc
+	CPU_CFLAGS-$(CONFIG_MIPS_ISA_MIPS32_24KC)+=-march=24kc -mtune=24kc
 	CPU_CFLAGS-$(CONFIG_MIPS_ISA_MIPS32R2)+=-march=mips32r2 -mtune=mips32r2
 	CPU_CFLAGS-$(CONFIG_MIPS_ISA_MIPS64)+=-mips64 -mtune=mips32
 	ifeq ($(strip $(ARCH_BIG_ENDIAN)),y)
