--- configure.ac.orig	2009-06-02 14:43:05.000000000 +0000
+++ configure.ac	2009-06-23 09:03:49.000000000 +0000
@@ -259,7 +259,7 @@
 dnl *************************************************************
 dnl Check for headers
 dnl *************************************************************
-AC_CHECK_HEADERS([arpa/inet.h fcntl.h libintl.h limits.h \
+AC_CHECK_HEADERS([arpa/inet.h fcntl.h limits.h \
                  malloc.h memory.h netdb.h netinet/in.h paths.h \
                  stdlib.h string.h sys/file.h sys/ioctl.h sys/mount.h \
                  sys/param.h sys/socket.h sys/time.h sys/vfs.h \
--- configure.orig	2009-06-02 15:23:26.000000000 +0000
+++ configure	2009-06-23 09:04:02.000000000 +0000
@@ -26211,7 +26211,7 @@
 
 
 
-for ac_header in arpa/inet.h fcntl.h libintl.h limits.h \
+for ac_header in arpa/inet.h fcntl.h limits.h \
                  malloc.h memory.h netdb.h netinet/in.h paths.h \
                  stdlib.h string.h sys/file.h sys/ioctl.h sys/mount.h \
                  sys/param.h sys/socket.h sys/time.h sys/vfs.h \
