--- source/configure
+++ source/configure
@@ -37813,6 +37813,7 @@
 #####################################
 # needed for SRV lookups
 
+if false; then
 { $as_echo "$as_me:$LINENO: checking for dn_expand in -lresolv" >&5
 $as_echo_n "checking for dn_expand in -lresolv... " >&6; }
 if test "${ac_cv_lib_resolv_dn_expand+set}" = set; then
@@ -38036,6 +38037,7 @@
   LIBS="-lresolv $LIBS"
 
 fi
+fi
 
 
 #
