--- lib/et/error_message.c	2009-08-13 03:39:57.000000000 +0200
+++ lib/et/error_message.c	2009-11-22 00:12:50.000000000 +0100
@@ -42,6 +42,7 @@
 #include "error_table.h"
 #include "internal.h"
 
+#undef TLS
 #ifdef TLS
 #define THREAD_LOCAL static TLS
 #else
--- lib/uuid/gen_uuid.c	2009-08-13 03:39:57.000000000 +0200
+++ lib/uuid/gen_uuid.c	2009-11-22 00:13:00.000000000 +0100
@@ -98,6 +98,7 @@
 #define rand() 		random()
 #endif
 
+#undef TLS
 #ifdef TLS
 #define THREAD_LOCAL static TLS
 #else
