--- e2fsck/super.c.orig	2010-02-23 05:40:50.000000000 +0100
+++ e2fsck/super.c	2010-04-27 20:20:12.000000000 +0200
@@ -821,6 +821,7 @@
 			    0, accept_time_fudge, &accept_time_fudge);
 	ctx->time_fudge = accept_time_fudge ? 86400 : 0;
 
+#if 0 /* at boot time we have no valid date/time yet so disable this check */
 	/*
 	 * Check to see if the superblock last mount time or last
 	 * write time is in the future.
@@ -845,6 +846,7 @@
 			ext2fs_mark_super_dirty(fs);
 		}
 	}
+#endif	
 
 	/*
 	 * Move the ext3 journal file, if necessary.
