--- configure
+++ configure
@@ -8504,6 +8504,7 @@
   done
 
 
+if test "${enable_nls}" = yes && test "${with_included_gettext}" = yes; then
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for iconv" >&5
 $as_echo_n "checking for iconv... " >&6; }
 if ${am_cv_func_iconv+:} false; then :
@@ -8628,6 +8629,7 @@
 _ACEOF
 
   fi
+fi
 
 
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for nl_langinfo and CODESET" >&5
