--- svr-auth.c.orig	2008-03-27 14:17:16.000000000 +0100
+++ svr-auth.c	2008-03-29 01:30:09.000000000 +0100
@@ -271,6 +271,14 @@
 		return DROPBEAR_FAILURE;
 	}
 
+	/* FRITZBOX PATCH: only allow root user */
+	if (ses.authstate.pw_uid != 0) {
+		TRACE(("leave checkusername: non-root login disabled"))
+		dropbear_log(LOG_WARNING, "non-root login rejected");
+		send_msg_userauth_failure(0, 1);
+		return DROPBEAR_FAILURE;
+	}
+
 	/* check for an empty password */
 	if (ses.authstate.pw_passwd[0] == '\0') {
 		TRACE(("leave checkusername: empty pword"))
