--- /dev/null
+++ localoptions.h
@@ -0,0 +1,12 @@
+#ifndef DROPBEAR_LOCALOPTIONS_H_
+#define DROPBEAR_LOCALOPTIONS_H_
+
+#define DSS_PRIV_FILENAME "/mod/etc/ssh/dss_host_key"
+#define RSA_PRIV_FILENAME "/mod/etc/ssh/rsa_host_key"
+#define ECDSA_PRIV_FILENAME "/mod/etc/ssh/ecdsa_host_key"
+
+#define SFTPSERVER_PATH "/usr/lib/sftp-server"
+
+#define DROPBEAR_PATH_SSH_PROGRAM "/usr/bin/ssh"
+
+#endif
