--- svr-chansession.c
+++ svr-chansession.c
@@ -630,10 +630,10 @@
 			}
 		}
 		if (issubsys) {
-#ifdef SFTPSERVER_PATH
+#ifdef DROPBEAR_SFTPSERVER
 			if ((cmdlen == 4) && strncmp(chansess->cmd, "sftp", 4) == 0) {
 				m_free(chansess->cmd);
-				chansess->cmd = m_strdup(SFTPSERVER_PATH);
+				chansess->cmd = m_strdup(svr_opts.sftppath);
 			} else 
 #endif
 			{
--- svr-runopts.c
+++ svr-runopts.c
@@ -49,6 +49,10 @@
 					"-r rsakeyfile	Use rsakeyfile for the RSA host key\n"
 					"		(default: %s)\n"
 #endif
+#ifdef DROPBEAR_SFTPSERVER
+					"-S sftpserver	Set path to sftp server\n"
+					"		(default: %s)\n"
+#endif
 					"-F		Don't fork into background\n"
 #ifdef DISABLE_SYSLOG
 					"(Syslog support not compiled in, using stderr)\n"
@@ -94,6 +98,9 @@
 #ifdef DROPBEAR_RSA
 					RSA_PRIV_FILENAME,
 #endif
+#ifdef DROPBEAR_SFTPSERVER
+					SFTPSERVER_PATH,
+#endif
 					DROPBEAR_MAX_PORTS, DROPBEAR_DEFPORT, DROPBEAR_PIDFILE,
 					DEFAULT_RECV_WINDOW, DEFAULT_KEEPALIVE, DEFAULT_IDLE_TIMEOUT);
 }
@@ -110,6 +117,7 @@
 	/* see printhelp() for options */
 	svr_opts.rsakeyfile = NULL;
 	svr_opts.dsskeyfile = NULL;
+	svr_opts.sftppath = NULL;
 	svr_opts.bannerfile = NULL;
 	svr_opts.banner = NULL;
 	svr_opts.forkbg = 1;
@@ -180,6 +188,11 @@
 					next = &svr_opts.rsakeyfile;
 					break;
 #endif
+#ifdef DROPBEAR_SFTPSERVER
+				case 'S':
+					next = &svr_opts.sftppath;
+					break;
+#endif
 				case 'F':
 					svr_opts.forkbg = 0;
 					break;
@@ -278,6 +291,9 @@
 	if (svr_opts.rsakeyfile == NULL) {
 		svr_opts.rsakeyfile = RSA_PRIV_FILENAME;
 	}
+	if (svr_opts.sftppath == NULL) {
+		svr_opts.sftppath = SFTPSERVER_PATH;
+	}
 
 	if (svr_opts.bannerfile) {
 		struct stat buf;
--- runopts.h
+++ runopts.h
@@ -62,6 +62,7 @@
 
 	char * rsakeyfile;
 	char * dsskeyfile;
+	char * sftppath;
 	char * bannerfile;
 
 	int forkbg;
