--- include/libssh2.h
+++ include/libssh2.h
@@ -46,12 +46,12 @@
    to make the BANNER define (used by src/session.c) be a valid SSH
    banner. Release versions have no appended strings and may of course not
    have dashes either. */
-#define LIBSSH2_VERSION                             "1.7.0_DEV"
+#define LIBSSH2_VERSION                             "1.8.0"
 
 /* The numeric version number is also available "in parts" by using these
    defines: */
 #define LIBSSH2_VERSION_MAJOR                       1
-#define LIBSSH2_VERSION_MINOR                       7
+#define LIBSSH2_VERSION_MINOR                       8
 #define LIBSSH2_VERSION_PATCH                       0
 
 /* This is the numeric version of the libssh2 version number, meant for easier
@@ -69,7 +69,7 @@
    and it is always a greater number in a more recent release. It makes
    comparisons with greater than and less than work.
 */
-#define LIBSSH2_VERSION_NUM                         0x010700
+#define LIBSSH2_VERSION_NUM                         0x010800
 
 /*
  * This is the date and time when the full source package was created. The
