--- linux-3.10/include/uapi/linux/Kbuild
+++ linux-3.10/include/uapi/linux/Kbuild
@@ -422,6 +422,7 @@
 header-y += xfrm.h
 header-y += hw_breakpoint.h
 
+# Freetz: everything below was added by AVM (i.e. it's the delta between AVM and vanilla version of this file)
 header-y += ar7wdt.h
 header-y += autoconf.h
 header-y += avm_debug.h
@@ -438,17 +439,17 @@
 header-y += avm_proc.h
 header-y += avm_profile.h
 header-y += avmnet.h
-header-y += capi_codec.h
+#header-y += capi_codec.h
 header-y += capi_oslib.h
-header-y += capidef.h
-header-y += codec_dsp.h
-header-y += import_rtp.h
+#header-y += capidef.h
+#header-y += codec_dsp.h
+#header-y += import_rtp.h
 header-y += kconfig.h
 header-y += new_capi.h
-header-y += pcmlink_audio.h
+#header-y += pcmlink_audio.h
 header-y += pcmlink_ul.h
-header-y += pcmroute.h
-header-y += rtp_fastpath.h
+#header-y += pcmroute.h
+#header-y += rtp_fastpath.h
 header-y += simple_proc.h
 header-y += tffs.h
 header-y += zugriff.h
