--- Makefile.in
+++ Makefile.in
@@ -43464,6 +43464,7 @@
 all-stage4-target-libstdc++-v3: maybe-configure-stage4-target-libgomp
 all-stageprofile-target-libstdc++-v3: maybe-configure-stageprofile-target-libgomp
 all-stagefeedback-target-libstdc++-v3: maybe-configure-stagefeedback-target-libgomp
+install-target-libiberty: maybe-install-libiberty
 all-target-libgloss: maybe-all-target-newlib
 all-target-winsup: maybe-all-target-libtermcap
 configure-target-newlib: maybe-all-binutils
--- boehm-gc/Makefile.in
+++ boehm-gc/Makefile.in
@@ -626,6 +626,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -674,7 +675,7 @@
 
 install-dvi-am:
 
-install-exec-am: install-multi
+install-exec-am:
 
 install-html: install-html-recursive
 
--- libffi/Makefile.in
+++ libffi/Makefile.in
@@ -1162,6 +1162,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -1244,7 +1245,7 @@
 
 install-dvi-am:
 
-install-exec-am: install-multi install-toolexeclibLTLIBRARIES
+install-exec-am: install-toolexeclibLTLIBRARIES
 
 install-html: install-html-recursive
 
--- libgfortran/Makefile.in
+++ libgfortran/Makefile.in
@@ -5756,6 +5756,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-am
 install-strip:
@@ -5806,7 +5807,7 @@
 
 install-dvi-am:
 
-install-exec-am: install-cafexeclibLTLIBRARIES install-multi \
+install-exec-am: install-cafexeclibLTLIBRARIES \
 	install-myexeclibLTLIBRARIES install-toolexeclibDATA \
 	install-toolexeclibLTLIBRARIES
 
--- libgo/Makefile.in
+++ libgo/Makefile.in
@@ -3849,6 +3849,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -3899,7 +3900,7 @@
 
 install-dvi-am:
 
-install-exec-am: install-multi install-toolexeclibLIBRARIES \
+install-exec-am: install-toolexeclibLIBRARIES \
 	install-toolexeclibLTLIBRARIES install-toolexeclibgoDATA \
 	install-toolexeclibgoarchiveDATA \
 	install-toolexeclibgocompressDATA \
--- libgomp/Makefile.in
+++ libgomp/Makefile.in
@@ -858,6 +858,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -919,7 +920,7 @@
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(dvidir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(dvidir)" || exit $$?; \
 	done
-install-exec-am: install-multi install-nodist_toolexeclibHEADERS \
+install-exec-am: install-nodist_toolexeclibHEADERS \
 	install-toolexeclibLTLIBRARIES
 
 install-html: install-html-recursive
--- libjava/classpath/Makefile.in
+++ libjava/classpath/Makefile.in
@@ -546,6 +546,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -592,7 +593,7 @@
 
 install-dvi-am:
 
-install-exec-am: install-multi
+install-exec-am:
 
 install-html: install-html-recursive
 
--- libjava/libltdl/Makefile.in
+++ libjava/libltdl/Makefile.in
@@ -467,6 +467,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-am
 install-strip:
@@ -516,7 +517,7 @@
 
 install-dvi-am:
 
-install-exec-am: install-libLTLIBRARIES install-multi
+install-exec-am: install-libLTLIBRARIES
 
 install-html: install-html-am
 
--- libjava/Makefile.in
+++ libjava/Makefile.in
@@ -10344,6 +10344,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -10461,7 +10462,7 @@
 
 install-exec-am: install-binPROGRAMS install-binSCRIPTS \
 	install-dbexecDATA install-dbexecLTLIBRARIES \
-	install-libexecsubPROGRAMS install-multi \
+	install-libexecsubPROGRAMS \
 	install-toolexeclibLTLIBRARIES install-toolexecmainlibDATA
 	@$(NORMAL_INSTALL)
 	$(MAKE) $(AM_MAKEFLAGS) install-exec-hook
--- libmudflap/Makefile.in
+++ libmudflap/Makefile.in
@@ -698,6 +698,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -746,7 +747,7 @@
 
 install-dvi-am:
 
-install-exec-am: install-multi install-toolexeclibLTLIBRARIES
+install-exec-am: install-toolexeclibLTLIBRARIES
 
 install-html: install-html-recursive
 
--- libquadmath/Makefile.in
+++ libquadmath/Makefile.in
@@ -1197,6 +1197,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-am
 install-strip:
@@ -1263,7 +1264,7 @@
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(dvidir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(dvidir)" || exit $$?; \
 	done
-install-exec-am: install-multi install-toolexeclibLTLIBRARIES
+install-exec-am: install-toolexeclibLTLIBRARIES
 
 install-html: install-html-am
 
--- libssp/Makefile.in
+++ libssp/Makefile.in
@@ -583,6 +583,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-am
 install-strip:
@@ -631,7 +632,7 @@
 
 install-dvi-am:
 
-install-exec-am: install-multi install-toolexeclibLTLIBRARIES
+install-exec-am: install-toolexeclibLTLIBRARIES
 
 install-html: install-html-am
 
--- libstdc++-v3/Makefile.in
+++ libstdc++-v3/Makefile.in
@@ -576,6 +576,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -614,7 +615,7 @@
 
 install-dvi-am:
 
-install-exec-am: install-multi
+install-exec-am:
 
 install-html-am:
 
--- zlib/Makefile.in
+++ zlib/Makefile.in
@@ -559,6 +559,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-am
 install-strip:
@@ -606,7 +607,7 @@
 
 install-dvi-am:
 
-install-exec-am: install-multi install-toolexeclibLIBRARIES
+install-exec-am: install-toolexeclibLIBRARIES
 
 install-html: install-html-am
 
