From: Nick Clifton <nickc@redhat.com>
Date: Fri, 25 Mar 2011 17:14:02 +0000
Subject: [PATCH] 	* ar.c (move_members): Plug memory leak.

diff --git binutils/ar.c binutils/ar.c
index 0866a54..6320a28 100644
--- binutils/ar.c
+++ binutils/ar.c
@@ -1185,9 +1185,8 @@ delete_members (bfd *arch, char **files_to_delete)
 static void
 move_members (bfd *arch, char **files_to_move)
 {
-  bfd **after_bfd;		/* New entries go after this one.  */
-  bfd **current_ptr_ptr;	/* cdr pointer into contents.  */
-  const char *tmp = NULL;
+  bfd **after_bfd;		/* New entries go after this one */
+  bfd **current_ptr_ptr;	/* cdr pointer into contents */
 
   for (; *files_to_move; ++files_to_move)
     {
@@ -1195,11 +1194,8 @@ move_members (bfd *arch, char **files_to_move)
       while (*current_ptr_ptr)
 	{
 	  bfd *current_ptr = *current_ptr_ptr;
-
-	  if (tmp != NULL)
-	    free ((char *) tmp);
-	  tmp = normalize (*files_to_move, arch);
-	  if (FILENAME_CMP (tmp, current_ptr->filename) == 0)
+	  if (FILENAME_CMP (normalize (*files_to_move, arch),
+			    current_ptr->filename) == 0)
 	    {
 	      /* Move this file to the end of the list - first cut from
 		 where it is.  */
@@ -1223,13 +1219,10 @@ move_members (bfd *arch, char **files_to_move)
       /* xgettext:c-format */
       fatal (_("no entry %s in archive %s!"), *files_to_move, arch->filename);
 
+    next_file:;
-    next_file:
-      ;
     }
 
   write_archive (arch);
-  if (tmp != NULL)
-    free ((char *) tmp);
 }
 
 /* Ought to default to replacing in place, but this is existing practice!  */
--
1.7.3.4
