--- setup.py
+++ setup.py
@@ -198,11 +198,13 @@
       ext_modules = [mkExtension('crypto'), mkExtension('rand'),
                      mkExtension('SSL')],
       py_modules  = ['OpenSSL.__init__', 'OpenSSL.tsafe',
-                     'OpenSSL.version', 'OpenSSL.test.__init__',
-                     'OpenSSL.test.util',
-                     'OpenSSL.test.test_crypto',
-                     'OpenSSL.test.test_rand',
-                     'OpenSSL.test.test_ssl'],
+                     'OpenSSL.version' #,
+                     #'OpenSSL.test.__init__',
+                     #'OpenSSL.test.util',
+                     #'OpenSSL.test.test_crypto',
+                     #'OpenSSL.test.test_rand',
+                     #'OpenSSL.test.test_ssl'
+                     ],
       zip_safe = False,
       cmdclass = {"build_ext": BuildExtension},
       description = 'Python wrapper module around the OpenSSL library',
