--- lib/Makefile
+++ lib/Makefile
@@ -56,7 +56,7 @@ endif
 
 all : ${ALL_DEPS}
 ifeq ($(WITH_SHARED_LIBRARIES),yes)
-	$(MAKE) -C cpp
+#	$(MAKE) -C cpp
 endif
 
 install : all
@@ -75,7 +75,7 @@ endif
 	$(INSTALL) -m644 ../libmosquitto.pc.in "${DESTDIR}${libdir}/pkgconfig/libmosquitto.pc"
 	sed -i -e "s#@CMAKE_INSTALL_PREFIX@#${prefix}#" -e "s#@VERSION@#${VERSION}#" "${DESTDIR}${libdir}/pkgconfig/libmosquitto.pc"
 ifeq ($(WITH_SHARED_LIBRARIES),yes)
-	$(MAKE) -C cpp install
+#	$(MAKE) -C cpp install
 endif
 
 uninstall :
@@ -88,7 +88,7 @@ reallyclean : clean
 
 clean :
 	-rm -f *.o libmosquitto.so.${SOVERSION} libmosquitto.so libmosquitto.a *.gcno *.gcda
-	$(MAKE) -C cpp clean
+#	$(MAKE) -C cpp clean
 
 libmosquitto.so.${SOVERSION} : ${MOSQ_OBJS}
 	${CROSS_COMPILE}$(CC) -shared $(LIB_LDFLAGS) $^ -o $@ ${LIB_LIBADD}
