--- linux-2.6.13.1/drivers/Makefile.orig	2007-03-19 16:41:47.000000000 +0100
+++ linux-2.6.13.1/drivers/Makefile	2009-04-03 01:32:14.000000000 +0200
@@ -30,7 +30,6 @@
 obj-$(CONFIG_NUBUS)		+= nubus/
 obj-$(CONFIG_ATM)		+= atm/
 obj-$(CONFIG_UR8_DSL)	+= dsl/ur8/
-obj-$(CONFIG_VINAX)		+= vinax/
 obj-$(CONFIG_PPC_PMAC)		+= macintosh/
 obj-$(CONFIG_IDE)		+= ide/
 obj-$(CONFIG_FC4)		+= fc4/
--- linux-2.6.13.1/drivers/net/Kconfig.orig	2009-04-03 01:42:01.000000000 +0200
+++ linux-2.6.13.1/drivers/net/Kconfig	2009-04-03 01:43:47.000000000 +0200
@@ -2226,8 +2226,6 @@
 
 source "drivers/atm/Kconfig"
 
-source "drivers/vinax/Kconfig"
-
 source "drivers/s390/net/Kconfig"
 
 config ISERIES_VETH
