--- gcc/config/mips/linux64.h
+++ gcc/config/mips/linux64.h
@@ -31,8 +31,9 @@
 
 #undef LIB_SPEC
 #define LIB_SPEC "\
+%{pthread:-lpthread} \
 %{shared: -lc} \
-%{!shared: %{pthread:-lpthread} \
+%{!shared: \
   %{profile:-lc_p} %{!profile: -lc}}"
 
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
--- gcc/config/mips/linux.h
+++ gcc/config/mips/linux.h
@@ -173,8 +173,9 @@
 
 #undef LIB_SPEC
 #define LIB_SPEC "\
+%{pthread:-lpthread} \
 %{shared: -lc} \
-%{!shared: %{pthread:-lpthread} \
+%{!shared: \
   %{profile:-lc_p} %{!profile: -lc}}"
 
 #define MD_UNWIND_SUPPORT "config/mips/linux-unwind.h"
