diff --strip-trailing-cr -x CVS -x .svn -urN boehm-gc/Makefile.in boehm-gc/Makefile.in
--- boehm-gc/Makefile.in	2008-05-19 16:28:07.000000000 +0200
+++ boehm-gc/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -668,6 +668,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -709,7 +710,7 @@
 
 install-data-am:
 
-install-exec-am: install-multi
+install-exec-am:
 
 install-info: install-info-recursive
 
diff --strip-trailing-cr -x CVS -x .svn -urN libffi/Makefile.in libffi/Makefile.in
--- libffi/Makefile.in	2008-05-19 16:28:07.000000000 +0200
+++ libffi/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -1210,6 +1210,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -1283,7 +1284,7 @@
 
 install-data-am:
 
-install-exec-am: install-multi install-toolexeclibLTLIBRARIES
+install-exec-am: install-toolexeclibLTLIBRARIES
 
 install-info: install-info-recursive
 
diff --strip-trailing-cr -x CVS -x .svn -urN libgfortran/Makefile.in libgfortran/Makefile.in
--- libgfortran/Makefile.in	2008-05-19 16:28:07.000000000 +0200
+++ libgfortran/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -2907,6 +2907,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-am
 install-strip:
@@ -2948,7 +2949,7 @@
 
 install-data-am:
 
-install-exec-am: install-multi install-myexeclibLTLIBRARIES \
+install-exec-am: install-myexeclibLTLIBRARIES \
 	install-toolexeclibLTLIBRARIES
 
 install-info: install-info-am
diff --strip-trailing-cr -x CVS -x .svn -urN libgomp/Makefile.in libgomp/Makefile.in
--- libgomp/Makefile.in	2008-05-19 16:28:07.000000000 +0200
+++ libgomp/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -895,6 +895,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -941,7 +942,7 @@
 install-data-am: install-info-am install-nodist_fincludeHEADERS \
 	install-nodist_libsubincludeHEADERS
 
-install-exec-am: install-multi install-nodist_toolexeclibHEADERS \
+install-exec-am: install-nodist_toolexeclibHEADERS \
 	install-toolexeclibLTLIBRARIES
 
 install-info: install-info-recursive
diff --strip-trailing-cr -x CVS -x .svn -urN libiberty/Makefile.in libiberty/Makefile.in
--- libiberty/Makefile.in	2007-03-03 20:29:51.000000000 +0100
+++ libiberty/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -106,7 +106,7 @@
 
 # FIXME: add @BUILD_INFO@ once we're sure it works for everyone.
 all: stamp-picdir $(TARGETLIB) needed-list required-list all-subdir
-	@: $(MAKE) ; exec $(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=all
+	@: $(MAKE) ; $(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=all
 
 .PHONY: check installcheck
 check: check-subdir
@@ -356,14 +356,14 @@
 	    ${INSTALL_DATA} $$h $(DESTDIR)$${thd}; \
 	  done; \
 	fi
-	@$(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=install
+	@: $(MAKE) ; $(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=install
 
 install_to_tooldir: all
 	${mkinstalldirs} $(DESTDIR)$(tooldir)/lib/$(MULTIOSDIR)
 	$(INSTALL_DATA) $(TARGETLIB) $(DESTDIR)$(tooldir)/lib/$(MULTIOSDIR)/$(TARGETLIB)n
 	( cd $(DESTDIR)$(tooldir)/lib/$(MULTIOSDIR) ; chmod 644 $(TARGETLIB)n; $(RANLIB) $(TARGETLIB)n )
 	mv -f $(DESTDIR)$(tooldir)/lib/$(MULTIOSDIR)/$(TARGETLIB)n $(DESTDIR)$(tooldir)/lib/$(MULTIOSDIR)/$(TARGETLIB)
-	@$(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=install
+	@: $(MAKE) ; $(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=install
 
 # needed-list is used by libstdc++.  NEEDED is the list of functions
 # to include there.  Do not add anything LGPL to this list; libstdc++
diff --strip-trailing-cr -x CVS -x .svn -urN libjava/classpath/Makefile.in libjava/classpath/Makefile.in
--- libjava/classpath/Makefile.in	2007-01-17 19:10:26.000000000 +0100
+++ libjava/classpath/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -686,6 +686,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -725,7 +726,7 @@
 
 install-data-am:
 
-install-exec-am: install-multi
+install-exec-am:
 
 install-info: install-info-recursive
 
diff --strip-trailing-cr -x CVS -x .svn -urN libjava/libltdl/Makefile.in libjava/libltdl/Makefile.in
--- libjava/libltdl/Makefile.in	2007-01-17 19:10:26.000000000 +0100
+++ libjava/libltdl/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -585,6 +585,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-am
 install-strip:
@@ -627,7 +628,7 @@
 
 install-data-am: install-data-local install-includeHEADERS
 
-install-exec-am: install-libLTLIBRARIES install-multi
+install-exec-am: install-libLTLIBRARIES
 
 install-info: install-info-am
 
diff --strip-trailing-cr -x CVS -x .svn -urN libjava/Makefile.in libjava/Makefile.in
--- libjava/Makefile.in	2008-05-19 16:28:07.000000000 +0200
+++ libjava/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -9005,6 +9005,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -9105,7 +9106,7 @@
 install-data-am: install-data-local install-jarDATA
 
 install-exec-am: install-binPROGRAMS install-binSCRIPTS \
-	install-dbexecDATA install-dbexecLTLIBRARIES install-multi \
+	install-dbexecDATA install-dbexecLTLIBRARIES \
 	install-toolexeclibLTLIBRARIES install-toolexecmainlibDATA
 	@$(NORMAL_INSTALL)
 	$(MAKE) $(AM_MAKEFLAGS) install-exec-hook
@@ -11940,9 +11941,9 @@
 maintainer-clean-recursive: maintainer-clean-multi
 
 all-multi:
-	: $(MAKE) ; exec $(MULTIDO) $(AM_MAKEFLAGS) DO=all multi-do
+	: $(MAKE) ; $(MULTIDO) $(AM_MAKEFLAGS) DO=all multi-do
 install-multi:
-	$(MULTIDO) $(AM_MAKEFLAGS) DO=install multi-do
+	: $(MAKE) ; $(MULTIDO) $(AM_MAKEFLAGS) DO=install multi-do
 mostlyclean-multi:
 	$(MULTICLEAN) $(AM_MAKEFLAGS) DO=mostlyclean multi-clean
 clean-multi:
diff --strip-trailing-cr -x CVS -x .svn -urN libmudflap/Makefile.in libmudflap/Makefile.in
--- libmudflap/Makefile.in	2006-06-21 18:35:27.000000000 +0200
+++ libmudflap/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -751,6 +751,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -792,7 +793,7 @@
 
 install-data-am: install-nobase_libsubincludeHEADERS
 
-install-exec-am: install-multi install-toolexeclibLTLIBRARIES
+install-exec-am: install-toolexeclibLTLIBRARIES
 
 install-info: install-info-recursive
 
diff --strip-trailing-cr -x CVS -x .svn -urN libobjc/Makefile.in libobjc/Makefile.in
--- libobjc/Makefile.in	2006-10-15 09:42:57.000000000 +0200
+++ libobjc/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -138,7 +138,7 @@
 	"tooldir=$(tooldir)"
 
 all: libobjc$(libext).la $(OBJC_BOEHM_GC)
-	: $(MAKE) ; exec $(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=all
+	: $(MAKE) ; $(MULTIDO) $(FLAGS_TO_PASS) multi-do DO=all
 
 # User-visible header files.
 
@@ -338,7 +338,7 @@
 	  $(LIBTOOL_INSTALL) $(INSTALL) libobjc_gc$(libext).la \
 				$(DESTDIR)$(toolexeclibdir);\
 	fi
-	$(MULTIDO) $(FLAGS_TO_PASS) multi-do DO="$@"
+	: $(MAKE) ; $(MULTIDO) $(FLAGS_TO_PASS) multi-do DO="$@"
 	@-$(LIBTOOL) --mode=finish $(DESTDIR)$(toolexeclibdir)
 
 # Copy Objective C headers to installation include directory.
diff --strip-trailing-cr -x CVS -x .svn -urN libssp/Makefile.in libssp/Makefile.in
--- libssp/Makefile.in	2006-09-29 23:18:51.000000000 +0200
+++ libssp/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -663,6 +663,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-am
 install-strip:
@@ -704,7 +705,7 @@
 
 install-data-am: install-nobase_libsubincludeHEADERS
 
-install-exec-am: install-multi install-toolexeclibLTLIBRARIES
+install-exec-am: install-toolexeclibLTLIBRARIES
 
 install-info: install-info-am
 
diff --strip-trailing-cr -x CVS -x .svn -urN libstdc++-v3/Makefile.in libstdc++-v3/Makefile.in
--- libstdc++-v3/Makefile.in	2006-10-16 21:08:22.000000000 +0200
+++ libstdc++-v3/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -672,6 +672,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-recursive
 install-strip:
@@ -711,7 +712,7 @@
 
 install-data-am:
 
-install-exec-am: install-multi
+install-exec-am:
 
 install-info: install-info-recursive
 
diff --strip-trailing-cr -x CVS -x .svn -urN Makefile.in Makefile.in
--- Makefile.in	2006-12-29 18:47:06.000000000 +0100
+++ Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -43791,6 +43791,7 @@
 all-target-libstdc++-v3: maybe-all-target-libiberty
 install-target-libssp: maybe-install-gcc
 install-target-libgomp: maybe-install-gcc
+install-target-libiberty: maybe-install-libiberty
 all-target-libgloss: maybe-configure-target-newlib
 all-target-winsup: maybe-all-target-libiberty
 all-target-winsup: maybe-all-target-libtermcap
diff --strip-trailing-cr -x CVS -x .svn -urN zlib/Makefile.in zlib/Makefile.in
--- zlib/Makefile.in	2007-01-17 19:10:26.000000000 +0100
+++ zlib/Makefile.in	2010-06-02 00:00:00.000000000 +0200
@@ -518,6 +518,7 @@
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-multi
 
 installcheck: installcheck-am
 install-strip:
@@ -556,7 +557,7 @@
 
 install-data-am:
 
-install-exec-am: install-multi install-toolexeclibLIBRARIES
+install-exec-am: install-toolexeclibLIBRARIES
 
 install-info: install-info-am
 
