--- usr/www/all/menus/menu_data.lua	2015-03-26 11:54:28.000000000 +0100
+++ usr/www/all/menus/menu_data.lua	2015-08-02 12:59:44.814095528 +0200
@@ -19,7 +19,7 @@
 menu = "main"
 }
 menu.add_item{
-submenu = "net",
+submenu = "fet",
 text = [[{?menuHomeNetwork?}]],
 short = [[{?menuHomeNetworkShort?}]],
 menu = "main"
@@ -279,7 +279,7 @@
 menu.add_item{
 text = [[{?menuNetwork?}]],
 short = [[{?menuNetworkShort?}]],
-menu = "net",
+menu = "fet",
 tabs = {
 { page = "/net/network_user_devices.lua", text = [[{?3505:134?}]]},
 { page = "/net/network_settings.lua", text = [[{?3505:557?}]] }
