--- src/ccache.cpp
+++ src/ccache.cpp
@@ -1965,7 +1965,7 @@
 set_up_config(Config& config)
 {
   const std::string home_dir = Util::get_home_directory();
-  const std::string legacy_ccache_dir = home_dir + "/.ccache";
+  const std::string legacy_ccache_dir = home_dir + "/.freetz-ccache";
   const bool legacy_ccache_dir_exists =
     Stat::stat(legacy_ccache_dir).is_directory();
   const char* const env_xdg_cache_home = getenv("XDG_CACHE_HOME");
