--- linux-2.6.28.10/arch/mips/Makefile
+++ linux-2.6.28.10/arch/mips/Makefile
@@ -645,6 +645,9 @@
 cflags-$(CONFIG_AMAZON_S) += -I$(srctree)/arch/mips/include/asm/mach-amazone_s -save-temps
 load-$(CONFIG_AMAZON_S) += 0xFFFFFFFF80002000
 
+# temporary until string.h is fixed
+cflags-y += -ffreestanding
+
 cflags-y			+= -I$(srctree)/arch/mips/include/asm/mach-generic
 drivers-$(CONFIG_PCI)		+= arch/mips/pci/
 
