--- configure
+++ configure
@@ -19354,6 +19354,7 @@
 
 
 
+if false; then
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for Mach-O dynamic module iteration functions" >&5
 $as_echo_n "checking for Mach-O dynamic module iteration functions... " >&6; }
 if ${ac_cv_mach_o_dynamic_module_iteration_works+:} false; then :
@@ -19440,6 +19441,7 @@
 
 fi
 rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
+fi
 
 
   # Check whether --enable-keychain was given.
