--- configure.in.orig	2007-12-29 20:30:55.000000000 +0000
+++ configure.in	2009-05-09 12:23:38.000000000 +0000
@@ -211,25 +211,4 @@
 		src/Makefile \
 		src/include/Makefile \
 		src/lib/Makefile \
-		src/test/Makefile \
-		src/debug/Makefile \
-		src/qgen/Makefile \
-		src/saal/Makefile \
-		src/sigd/Makefile \
-		src/maint/Makefile \
-		src/arpd/Makefile \
-		src/ilmid/Makefile \
-		src/ilmid/asn1/Makefile \
-		src/man/Makefile \
-		src/led/Makefile \
-		src/lane/Makefile \
-		src/mpoad/Makefile \
-		src/oamd/Makefile \
-		src/switch/Makefile \
-		src/switch/debug/Makefile \
-		src/switch/tcp/Makefile \
-		src/br2684/Makefile \
-		src/config/Makefile \
-		src/extra/Makefile \
-		src/extra/linux-atm.spec
 		)
--- src/Makefile.am.orig	2005-04-21 14:29:28.000000000 +0000
+++ src/Makefile.am	2009-05-09 12:23:38.000000000 +0000
@@ -1,3 +1,2 @@
-SUBDIRS = include lib test debug qgen saal sigd maint arpd ilmid man led lane \
-		mpoad oamd switch config extra br2684
+SUBDIRS = include lib
 
