--- configure
+++ configure
@@ -4257,13 +4257,16 @@
   for xc_word in $CFLAGS; do
     case "$xc_word" in
       -D*)
-        xc_bad_var_cflags=yes
+        CPPFLAGS="$CPPFLAGS $xc_word"
+        #xc_bad_var_cflags=yes
         ;;
       -U*)
-        xc_bad_var_cflags=yes
+        CPPFLAGS="$CPPFLAGS $xc_word"
+        #xc_bad_var_cflags=yes
         ;;
       -I*)
-        xc_bad_var_cflags=yes
+        CPPFLAGS="$CPPFLAGS $xc_word"
+        #xc_bad_var_cflags=yes
         ;;
       -rpath*)
         xc_bad_var_cflags=yes
@@ -45321,13 +45324,16 @@
   for xc_word in $CFLAGS; do
     case "$xc_word" in
       -D*)
-        xc_bad_var_cflags=yes
+        #xc_bad_var_cflags=yes
+        CPPFLAGS="$CPPFLAGS $xc_word"
         ;;
       -U*)
-        xc_bad_var_cflags=yes
+        #xc_bad_var_cflags=yes
+        CPPFLAGS="$CPPFLAGS $xc_word"
         ;;
       -I*)
-        xc_bad_var_cflags=yes
+        #xc_bad_var_cflags=yes
+        CPPFLAGS="$CPPFLAGS $xc_word"
         ;;
       -rpath*)
         xc_bad_var_cflags=yes
