--- linux-2.6.32/drivers/block/compcache/Makefile	2012-07-13 12:12:20+02:00
+++ linux-2.6.32/drivers/block/compcache/Makefile	2012-07-13 12:12:20+02:00
@@ -6,7 +6,7 @@
 ramzswap-objs	:=	ramzswap_drv.o xvmalloc.o
 
 ifneq ($(CONFIG_COMPCACHE),n)
-extra-y			+= rzscontrol ramzswap.sh
+#extra-y			+= rzscontrol ramzswap.sh
 
 rzscontrol-objs	:= rzscontrol.o
 
--- linux-2.6.32/drivers/rtc/Makefile	2012-07-13 12:12:20+02:00
+++ linux-2.6.32/drivers/rtc/Makefile	2012-07-13 12:12:20+02:00
@@ -91,7 +91,7 @@
 ifeq ($(CONFIG_RTC_DRV_AVM_REF_CLOCK),m)
 CFLAGS_rtc-avm.o += -DRTC_AVM_DEVICE_COUNT=1 # -DRTC_AVM_DEBUG=1
 
-extra-y			 += S09-rtc
+#extra-y			 += S09-rtc
 
 $(src)/S09-rtc: FORCE
 	rm -rf $(src)/S09-rtc
--- linux-2.6.32/drivers/net/ifxmips_switch_api/Makefile	2012-07-13 12:12:20+02:00
+++ linux-2.6.32/drivers/net/ifxmips_switch_api/Makefile	2012-07-13 12:12:20+02:00
@@ -9,7 +9,7 @@
 drv_switch_api-objs	:= ifx_ethsw_linux.o
 drv_switch_api-objs	+= ifx_ethsw_init.o
 
-extra-y			+= switch_utility
+#extra-y			+= switch_utility
 
 ##################### AR9 ######################
 ifeq ($(CONFIG_AR9),y)
--- linux-2.6.32/drivers/net/avm_cpmac/Makefile	2012-07-13 12:12:20+02:00
+++ linux-2.6.32/drivers/net/avm_cpmac/Makefile	2012-07-13 12:12:20+02:00
@@ -99,7 +99,7 @@
 # Create source files on demand
 EXTRACT_SOURCE: $(SOURCE_FILES:%=$(src)/%) $(src)/cpmac_product_conf.c
 
-extra-y	+= cpmacconfig
+#extra-y	+= cpmacconfig
 
 cpmacconfig-objs	:= cpmacconfig.o
 
