diff -Naur extra/Configs/Config.mips.orig extra/Configs/Config.mips
--- extra/Configs/Config.mips.orig	2008-09-25 17:07:27.000000000 +0200
+++ extra/Configs/Config.mips	2008-11-27 11:13:09.000000000 +0100
@@ -68,6 +68,9 @@
 config CONFIG_MIPS_ISA_MIPS32
 	bool "MIPS32"
 
+config CONFIG_MIPS_ISA_MIPS32_4KC
+	bool "MIPS32 4KC (texas ar7, fritz!box)"
+
 config CONFIG_MIPS_ISA_MIPS32R2
 	bool "MIPS32r2"
 
