--- etc/init.d/S08-tffs.orig	2010-09-15 13:39:31.000000000 +0200
--- etc/init.d/S08-tffs	2011-11-18 18:35:55+01:00
+++ etc/init.d/S08-tffs	2011-11-18 18:35:55+01:00
@@ -111,6 +111,10 @@
 tffs_nodes_list="$tffs_nodes_list $((0xCA)),websrv_ssl_cert.pem"
 tffs_nodes_list="$tffs_nodes_list $((0xE0)),smartmeter.cfg"
 #########################################################################
+## Freetz
+#########################################################################
+tffs_nodes_list="$tffs_nodes_list $((0x3C)),freetz"
+#########################################################################
 ## NODES anlegen, wenn nicht als _file_ existent
 #########################################################################
 for node in ${tffs_nodes_list} ; do
