--- Makefile.in
+++ Makefile.in
@@ -366,10 +366,6 @@
 revision-install:
 	test -d $(DESTDIR)$(includedir)/subversion-1 || \
 	  $(MKDIR) $(DESTDIR)$(includedir)/subversion-1
-	(subversion/svnversion/svnversion $(top_srcdir) ||  \
-	 svnversion $(top_srcdir) ||                        \
-	 echo "unknown";                                    \
-	) > $(DESTDIR)$(includedir)/subversion-1/svn-revision.txt
 
 install-static: @INSTALL_STATIC_RULES@
 
