--- configure
+++ configure
@@ -23314,11 +23314,9 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for inet_ntop with IPv6 support" >&5
 $as_echo_n "checking for inet_ntop with IPv6 support... " >&6; }
 if test "$cross_compiling" = yes; then :
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: assuming inet_ntop needed" >&5
-$as_echo "assuming inet_ntop needed" >&6; }
-	ISC_EXTRA_OBJS="$ISC_EXTRA_OBJS inet_ntop.$O"
-	ISC_EXTRA_SRCS="$ISC_EXTRA_SRCS inet_ntop.c"
-	ISC_PLATFORM_NEEDNTOP="#define ISC_PLATFORM_NEEDNTOP 1"
+  { $as_echo "$as_me:${as_lineno-$LINENO}: result: assuming inet_ntop is not needed" >&5
+$as_echo "assuming inet_ntop is not needed" >&6; }
+	ISC_PLATFORM_NEEDNTOP="#undef ISC_PLATFORM_NEEDNTOP"
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
