--- lib/export/isc/Makefile.in
+++ lib/export/isc/Makefile.in
@@ -47,6 +47,7 @@
 		unix/errno2result.@O@ \
 		unix/file.@O@ \
 		unix/fsaccess.@O@ \
+		unix/net.@O@ \
 		unix/stdio.@O@ \
 		unix/stdtime.@O@ unix/strerror.@O@ unix/time.@O@
 
--- lib/export/isc/unix/Makefile.in
+++ lib/export/isc/unix/Makefile.in
@@ -36,6 +36,7 @@
 		dir.@O@ \
 		errno2result.@O@ \
 		file.@O@ fsaccess.@O@ \
+		net.@O@ \
 		stdio.@O@ stdtime.@O@ strerror.@O@ \
 		time.@O@ \
 		${ISCDRIVEROBJS}
@@ -47,6 +48,7 @@
 		dir.c \
 		errno2result.c \
 		file.c fsaccess.c \
+		net.c \
 		stdio.c stdtime.c strerror.c \
 		time.c \
 		${ISCDRIVERSRCS}
