--- configure
+++ configure
@@ -9705,6 +9705,7 @@
 LDFLAGS="$LDFLAGS_SAVE"
 
 
+if false; then
 for ac_header in execinfo.h
 do
 as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
@@ -9854,6 +9855,7 @@
 fi
 
 done
+fi
 
 
 { $as_echo "$as_me:$LINENO: checking for time_t" >&5
@@ -12653,6 +12655,7 @@
 
 
 
+if false; then
 for ac_func in backtrace backtrace_symbols
 do
 as_ac_var=`$as_echo "ac_cv_func_$ac_func" | $as_tr_sh`
@@ -12752,6 +12755,7 @@
 
 fi
 done
+fi
 
 # Checks for typedefs, structures, and compiler characteristics.
 { $as_echo "$as_me:$LINENO: checking for an ANSI C-conforming const" >&5
