--- linux-2.6.19.2/arch/mips/mm/cache.c.orig	2008-09-02 11:49:50.000000000 +0200
+++ linux-2.6.19.2/arch/mips/mm/cache.c	2010-10-28 17:19:13.664312274 +0200
@@ -32,6 +32,8 @@
 void (*flush_data_cache_page)(unsigned long addr);
 void (*flush_icache_all)(void);
 
+/* ### Hack to make fuse work ### */
+EXPORT_SYMBOL(flush_cache_page);
 EXPORT_SYMBOL(flush_data_cache_page);
 
 #ifdef CONFIG_DMA_NONCOHERENT
