--- configure
+++ configure
@@ -5737,6 +5737,7 @@
 
 
 
+if false; then
 for flag in -fPIE; do
   as_CACHEVAR=`$as_echo "ax_cv_check_cflags__$flag" | $as_tr_sh`
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether C compiler accepts $flag" >&5
@@ -5797,11 +5798,13 @@
 fi
 
 done
+fi
 
 
 
 
 
+if false; then
 for flag in -fPIE -pie; do
   as_CACHEVAR=`$as_echo "ax_cv_check_ldflags__$flag" | $as_tr_sh`
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the linker accepts $flag" >&5
@@ -5863,6 +5866,7 @@
 fi
 
 done
+fi
 
 
 
