--- configure
+++ configure
@@ -5711,6 +5711,7 @@
 
 
 
+if false; then
 for flag in -fPIE; do
   as_CACHEVAR=`$as_echo "ax_cv_check_cflags__$flag" | $as_tr_sh`
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether C compiler accepts $flag" >&5
@@ -5771,6 +5772,7 @@
 fi
 
 done
+fi
 
 
 
@@ -5841,6 +5843,7 @@
 
 
 
+if false; then
 for flag in -fPIE -pie; do
   as_CACHEVAR=`$as_echo "ax_cv_check_ldflags__$flag" | $as_tr_sh`
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the linker accepts $flag" >&5
@@ -5902,6 +5905,7 @@
 fi
 
 done
+fi
 
 
 
