--- configure
+++ configure
@@ -6510,7 +6510,7 @@
 
 
 
-for flag in -fPIE; do
+false && for flag in -fPIE; do
   as_CACHEVAR=`$as_echo "ax_cv_check_cflags__$flag" | $as_tr_sh`
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether C compiler accepts $flag" >&5
 $as_echo_n "checking whether C compiler accepts $flag... " >&6; }
@@ -6666,7 +6666,7 @@
 
 
 
-for flag in -fPIE -pie; do
+false && for flag in -fPIE -pie; do
   as_CACHEVAR=`$as_echo "ax_cv_check_ldflags__$flag" | $as_tr_sh`
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the linker accepts $flag" >&5
 $as_echo_n "checking whether the linker accepts $flag... " >&6; }
