--- configure
+++ configure
@@ -5722,6 +5722,7 @@
 
 
 
+if false; then
 for flag in -fPIE; do
   as_CACHEVAR=`$as_echo "ax_cv_check_cflags__$flag" | $as_tr_sh`
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether C compiler accepts $flag" >&5
@@ -5782,11 +5783,13 @@
 fi
 
 done
+fi
 
 
 
 
 
+if false; then
 for flag in -fPIE -pie; do
   as_CACHEVAR=`$as_echo "ax_cv_check_ldflags__$flag" | $as_tr_sh`
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the linker accepts $flag" >&5
@@ -5848,6 +5851,7 @@
 fi
 
 done
+fi
 
 
 
