--- configure
+++ configure
@@ -4315,6 +4315,7 @@
     conftest$ac_exeext conftest.$ac_ext
 CFLAGS="$valid_CFLAGS"; LDFLAGS="$valid_LDFLAGS"
 
+if false; then
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether $CC accepts -pie" >&5
 $as_echo_n "checking whether $CC accepts -pie... " >&6; }
 valid_CFLAGS="$CFLAGS"; CFLAGS="$CFLAGS -fPIE"
@@ -4341,6 +4342,7 @@
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 CFLAGS="$valid_CFLAGS"; LDFLAGS="$valid_LDFLAGS"
+fi
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether $CC accepts -Wall" >&5
 $as_echo_n "checking whether $CC accepts -Wall... " >&6; }
