--- configure
+++ configure
@@ -15002,7 +15002,7 @@
 
 
 valid_CPPFLAGS="$CPPFLAGS"; CPPFLAGS="$CPPFLAGS -I$SSLDIR/include"
-valid_LIBS="$LIBS"; LIBS="$LIBS -L$SSLDIR/lib64 -L$SSLDIR/lib -lssl -lcrypto"
+valid_LIBS="$LIBS"; LIBS="$LIBS                 -L$SSLDIR/lib -lssl -lcrypto -ldl"
 
 if test "x$use_fips" = "xauto"; then
     for ac_func in FIPS_mode_set
--- src/Makefile.in
+++ src/Makefile.in
@@ -378,7 +378,7 @@
 	-DLIBDIR='"$(pkglibdir)"' -DCONFDIR='"$(sysconfdir)/stunnel"'
 
 # SSL library
-stunnel_LDFLAGS = -L$(SSLDIR)/lib64 -L$(SSLDIR)/lib -lssl -lcrypto
+stunnel_LDFLAGS =                   -L$(SSLDIR)/lib -lssl -lcrypto -ldl
 
 # stunnel3 script
 edit = sed \
