--- configure
+++ configure
@@ -13250,7 +13250,7 @@
 
 
 valid_CPPFLAGS="$CPPFLAGS"; CPPFLAGS="$CPPFLAGS -I$SSLDIR/include"
-valid_LIBS="$LIBS"; LIBS="$LIBS -L$SSLDIR/lib64 -L$SSLDIR/lib -lssl -lcrypto"
+valid_LIBS="$LIBS"; LIBS="$LIBS                 -L$SSLDIR/lib -lssl -lcrypto -ldl"
 
 as_ac_Header=`$as_echo "ac_cv_header_$SSLDIR/include/openssl/engine.h" | $as_tr_sh`
 ac_fn_c_check_header_mongrel "$LINENO" "$SSLDIR/include/openssl/engine.h" "$as_ac_Header" "$ac_includes_default"
--- src/Makefile.in
+++ src/Makefile.in
@@ -281,7 +281,7 @@
 stunnel_LDFLAGF = @stunnel_LDFLAGF@
 
 # SSL library
-stunnel_LDFLAGS = -L$(SSLDIR)/lib64 -L$(SSLDIR)/lib -lssl -lcrypto
+stunnel_LDFLAGS =                   -L$(SSLDIR)/lib -lssl -lcrypto -ldl
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
