--- configure
+++ configure
@@ -14970,7 +14970,7 @@
 
 
 valid_CPPFLAGS="$CPPFLAGS"; CPPFLAGS="$CPPFLAGS -I$SSLDIR/include"
-valid_LIBS="$LIBS"; LIBS="$LIBS -L$SSLDIR/lib64 -L$SSLDIR/lib -lssl -lcrypto"
+valid_LIBS="$LIBS"; LIBS="$LIBS                 -L$SSLDIR/lib -lssl -lcrypto -ldl"
 
 as_ac_Header=`$as_echo "ac_cv_header_$SSLDIR/include/openssl/engine.h" | $as_tr_sh`
 ac_fn_c_check_header_mongrel "$LINENO" "$SSLDIR/include/openssl/engine.h" "$as_ac_Header" "$ac_includes_default"
--- src/Makefile.in
+++ src/Makefile.in
@@ -390,7 +390,7 @@
 	-DLIBDIR='"$(pkglibdir)"' -DCONFDIR='"$(sysconfdir)/stunnel"'
 
 # SSL library
-stunnel_LDFLAGS = -L$(SSLDIR)/lib64 -L$(SSLDIR)/lib -lssl -lcrypto
+stunnel_LDFLAGS =                   -L$(SSLDIR)/lib -lssl -lcrypto -ldl
 
 # Win32 executable
 EXTRA_DIST = make.bat makece.bat makew32.bat mingw.mak evc.mak vc.mak \
