--- src/Makefile.in
+++ src/Makefile.in
@@ -272,8 +272,10 @@
 
 # Additional preprocesor definitions
 stunnel_CPPFLAGS = -I$(SSLDIR)/include \
-	-DLIBDIR='"$(pkglibdir)"' -DCONFDIR='"$(sysconfdir)/stunnel"' \
-	-DPIDFILE='"$(localstatedir)/run/stunnel/stunnel.pid"'
+	-DLIBDIR='"$(pkglibdir)"' -DCONFDIR='"/mod/etc"' \
+	-DPIDFILE='"/var/run/stunnel.pid"' \
+	-DFREETZ_DEFAULT_CERT='"/tmp/flash/stunnel/certs.pem"' \
+	-DFREETZ_DEFAULT_KEY='"/tmp/flash/stunnel/key.pem"'
 
 # Win32 executable
 EXTRA_DIST = nogui.c make.bat makece.bat makew32.bat mingw.mak evc.mak \
--- src/options.c
+++ src/options.c
@@ -653,7 +653,7 @@
     /* cert */
     switch(cmd) {
     case CMD_INIT:
-        section->cert=NULL;
+        section->cert=FREETZ_DEFAULT_CERT;
         break;
     case CMD_EXEC:
         if(strcasecmp(opt, "cert"))
@@ -661,7 +661,8 @@
         section->cert=str_dup(arg);
         return NULL; /* OK */
     case CMD_DEFAULT:
-        break; /* no default certificate */
+        s_log(LOG_NOTICE, "%-15s = %s", "cert", FREETZ_DEFAULT_CERT);
+        break;
     case CMD_HELP:
         s_log(LOG_NOTICE, "%-15s = certificate chain", "cert");
         break;
@@ -947,7 +948,7 @@
     /* key */
     switch(cmd) {
     case CMD_INIT:
-        section->key=NULL;
+        section->key=FREETZ_DEFAULT_KEY;
         break;
     case CMD_EXEC:
         if(strcasecmp(opt, "key"))
@@ -955,6 +956,7 @@
         section->key=str_dup(arg);
         return NULL; /* OK */
     case CMD_DEFAULT:
+        s_log(LOG_NOTICE, "%-15s = %s", "key", FREETZ_DEFAULT_KEY);
         break;
     case CMD_HELP:
         s_log(LOG_NOTICE, "%-15s = certificate private key", "key");
