--- src/Makefile.in	2010-03-01 17:55:59.000000000 +0100
+++ src/Makefile.in	2010-03-27 23:17:03.000000000 +0100
@@ -227,7 +227,7 @@
 INCLUDES = -I/usr/kerberos/include
 
 # Additional compiler flags
-AM_CPPFLAGS = -DLIBDIR='"$(pkglibdir)"' -DCONFDIR='"$(sysconfdir)/stunnel"' -DPIDFILE='"$(localstatedir)/run/stunnel/stunnel.pid"'
+AM_CPPFLAGS = -DLIBDIR='"$(pkglibdir)"' -DCONFDIR='"/mod/etc"' -DPIDFILE='"/var/run/stunnel.pid"'
 
 # Win32 executable
 EXTRA_DIST = stunnel.exe make.bat mingw.mak makece.bat evc.mak vc.mak nogui.c os2.mak
--- src/options.c	2010-03-23 15:58:34.000000000 +0100
+++ src/options.c	2010-03-27 23:14:44.000000000 +0100
@@ -613,7 +613,7 @@
     switch(cmd) {
     case CMD_INIT:
 #ifdef CONFDIR
-        section->cert=CONFDIR CONFSEPARATOR "stunnel.pem";
+        section->cert="/tmp/flash/.stunnel/certs.pem";
 #else
         section->cert="stunnel.pem";
 #endif
@@ -626,7 +626,7 @@
         return NULL; /* OK */
     case CMD_DEFAULT:
 #ifdef CONFDIR
-        s_log(LOG_NOTICE, "%-15s = %s", "cert", CONFDIR CONFSEPARATOR "stunnel.pem");
+        s_log(LOG_NOTICE, "%-15s = %s", "cert", "/tmp/flash/.stunnel/certs.pem");
 #else
         s_log(LOG_NOTICE, "%-15s = %s", "cert", "stunnel.pem");
 #endif
@@ -882,7 +882,7 @@
     /* key */
     switch(cmd) {
     case CMD_INIT:
-        section->key=NULL;
+        section->key="/tmp/flash/.stunnel/key.pem";
         break;
     case CMD_EXEC:
         if(strcasecmp(opt, "key"))
