--- src/Makefile.in
+++ src/Makefile.in
@@ -310,8 +310,10 @@
 
 # Additional preprocesor definitions
 stunnel_CPPFLAGS =                         -I$(SSLDIR)/include \
-	-DLIBDIR='"$(pkglibdir)"' -DCONFDIR='"$(sysconfdir)/stunnel"' \
-	-DPIDFILE='"$(localstatedir)/run/stunnel/stunnel.pid"'
+	-DLIBDIR='"$(pkglibdir)"' -DCONFDIR='"/mod/etc"' \
+	-DPIDFILE='"/var/run/stunnel.pid"' \
+	-DFREETZ_DEFAULT_CERT='"/tmp/flash/stunnel/certs.pem"' \
+	-DFREETZ_DEFAULT_KEY='"/tmp/flash/stunnel/key.pem"'
 
 # Win32 executable
 EXTRA_DIST = make.bat makece.bat makew32.bat mingw.mak evc.mak vc.mak \
--- src/options.c
+++ src/options.c
@@ -916,7 +916,7 @@
     /* cert */
     switch(cmd) {
     case CMD_BEGIN:
-        section->cert=NULL;
+        section->cert=FREETZ_DEFAULT_CERT;
         break;
     case CMD_EXEC:
         if(strcasecmp(opt, "cert"))
@@ -933,7 +933,8 @@
     case CMD_FREE:
         break;
     case CMD_DEFAULT:
-        break; /* no default certificate */
+        s_log(LOG_NOTICE, "%-22s = %s", "cert", FREETZ_DEFAULT_CERT);
+        break;
     case CMD_HELP:
         s_log(LOG_NOTICE, "%-22s = certificate chain", "cert");
         break;
@@ -1302,7 +1303,7 @@
     /* key */
     switch(cmd) {
     case CMD_BEGIN:
-        section->key=NULL;
+        section->key=FREETZ_DEFAULT_KEY;
         break;
     case CMD_EXEC:
         if(strcasecmp(opt, "key"))
@@ -1314,6 +1315,7 @@
     case CMD_FREE:
         break;
     case CMD_DEFAULT:
+        s_log(LOG_NOTICE, "%-22s = %s", "key", FREETZ_DEFAULT_KEY);
         break;
     case CMD_HELP:
         s_log(LOG_NOTICE, "%-22s = certificate private key", "key");
