--- src/options.c.orig	2007-12-29 02:14:10.000000000 +0100
+++ src/options.c	2007-12-29 02:16:19.000000000 +0100
@@ -250,7 +250,7 @@
 #ifndef USE_WIN32
     switch(cmd) {
     case CMD_INIT:
-        options.pidfile=PIDFILE;
+        options.pidfile="/var/run/stunnel.pid";
         break;
     case CMD_EXEC:
         if(strcasecmp(opt, "pid"))
@@ -261,7 +261,7 @@
             options.pidfile=NULL; /* empty -> do not create a pid file */
         return NULL; /* OK */
     case CMD_DEFAULT:
-        log_raw("%-15s = %s", "pid", PIDFILE);
+        log_raw("%-15s = %s", "pid", "/var/run/stunnel.pid");
         break;
     case CMD_HELP:
         log_raw("%-15s = pid file (empty to disable creating)", "pid");
@@ -534,7 +534,7 @@
     switch(cmd) {
     case CMD_INIT:
 #ifdef CONFDIR
-        section->cert=CONFDIR CONFSEPARATOR "stunnel.pem";
+        section->cert="/tmp/flash/.stunnel/cert.pem";
 #else
         section->cert="stunnel.pem";
 #endif
@@ -1241,7 +1241,7 @@
 
 void parse_config(char *name, char *parameter) {
 #ifdef CONFDIR
-    char *default_config_file=CONFDIR CONFSEPARATOR "stunnel.conf";
+    char *default_config_file="/mod/etc/stunnel.conf";
 #else
     char *default_config_file="stunnel.conf";
 #endif
