--- src/options.c.orig	2008-06-21 23:18:23.000000000 +0200
+++ src/options.c	2008-12-25 01:12:28.000000000 +0100
@@ -285,7 +285,7 @@
 #ifndef USE_WIN32
     switch(cmd) {
     case CMD_INIT:
-        options.pidfile=PIDFILE;
+        options.pidfile="/var/run/stunnel.pid";
         break;
     case CMD_EXEC:
         if(strcasecmp(opt, "pid"))
@@ -296,7 +296,7 @@
             options.pidfile=NULL; /* empty -> do not create a pid file */
         return NULL; /* OK */
     case CMD_DEFAULT:
-        s_log(LOG_RAW, "%-15s = %s", "pid", PIDFILE);
+        s_log(LOG_RAW, "%-15s = %s", "pid", "/var/run/stunnel.pid");
         break;
     case CMD_HELP:
         s_log(LOG_RAW, "%-15s = pid file (empty to disable creating)", "pid");
@@ -606,7 +606,7 @@
     switch(cmd) {
     case CMD_INIT:
 #ifdef CONFDIR
-        section->cert=CONFDIR CONFSEPARATOR "stunnel.pem";
+        section->cert="/tmp/flash/.stunnel/certs.pem";
 #else
         section->cert="stunnel.pem";
 #endif
@@ -843,7 +843,7 @@
     /* key */
     switch(cmd) {
     case CMD_INIT:
-        section->key=NULL;
+        section->key="/tmp/flash/.stunnel/key.pem";
         break;
     case CMD_EXEC:
         if(strcasecmp(opt, "key"))
@@ -1350,7 +1350,7 @@
 
 void parse_config(char *name, char *parameter) {
 #ifdef CONFDIR
-    char *default_config_file=CONFDIR CONFSEPARATOR "stunnel.conf";
+    char *default_config_file="/mod/etc/stunnel.conf";
 #else
     char *default_config_file="stunnel.conf";
 #endif
