--- src/Makefile.in
+++ src/Makefile.in
@@ -253,7 +253,7 @@
 INCLUDES = -I/usr/kerberos/include
 
 # Additional compiler flags
-AM_CPPFLAGS = -DLIBDIR='"$(pkglibdir)"' -DCONFDIR='"$(sysconfdir)/stunnel"' -DPIDFILE='"$(localstatedir)/run/stunnel/stunnel.pid"'
+AM_CPPFLAGS = -DLIBDIR='"$(pkglibdir)"' -DCONFDIR='"/mod/etc"' -DPIDFILE='"/var/run/stunnel.pid"'
 
 # Win32 executable
 EXTRA_DIST = stunnel.exe make.bat mingw.mak makece.bat evc.mak vc.mak nogui.c os2.mak
--- src/options.c
+++ src/options.c
@@ -620,7 +620,7 @@
     /* cert */
     switch(cmd) {
     case CMD_INIT:
-        section->cert=NULL;
+        section->cert="/tmp/flash/.stunnel/certs.pem";
         break;
     case CMD_EXEC:
         if(strcasecmp(opt, "cert"))
@@ -629,7 +629,7 @@
         return NULL; /* OK */
     case CMD_DEFAULT:
 #ifdef CONFDIR
-        s_log(LOG_NOTICE, "%-15s = %s", "cert", CONFDIR CONFSEPARATOR "stunnel.pem");
+        s_log(LOG_NOTICE, "%-15s = %s", "cert", "/tmp/flash/.stunnel/certs.pem");
 #else
         s_log(LOG_NOTICE, "%-15s = %s", "cert", "stunnel.pem");
 #endif
@@ -912,7 +912,7 @@
     /* key */
     switch(cmd) {
     case CMD_INIT:
-        section->key=NULL;
+        section->key="/tmp/flash/.stunnel/key.pem";
         break;
     case CMD_EXEC:
         if(strcasecmp(opt, "key"))
@@ -920,6 +920,7 @@
         section->key=stralloc(arg);
         return NULL; /* OK */
     case CMD_DEFAULT:
+        s_log(LOG_NOTICE, "%-15s = %s", "key", "/tmp/flash/.stunnel/key.pem");
         break;
     case CMD_HELP:
         s_log(LOG_NOTICE, "%-15s = certificate private key", "key");
