--- agent/mibgroup/ucd-snmp/pass.c.ori	2012-02-28 17:19:31.000000000 +0100
+++ agent/mibgroup/ucd-snmp/pass.c	2012-02-28 17:21:26.000000000 +0100
@@ -27,6 +27,9 @@
 #ifdef WIN32
 #include <limits.h>
 #endif
+#ifndef ASN_INTEGER64
+#define ASN_INTEGER64        (ASN_APPLICATION | 10)
+#endif
 
 #include <net-snmp/net-snmp-includes.h>
 #include <net-snmp/agent/net-snmp-agent-includes.h>
--- agent/mibgroup/ucd-snmp/pass_persist.c.ori	2012-02-28 17:21:55.000000000 +0100
+++ agent/mibgroup/ucd-snmp/pass_persist.c	2012-02-28 17:22:36.000000000 +0100
@@ -29,6 +29,9 @@
 #ifdef WIN32
 #include <limits.h>
 #endif
+#ifndef ASN_INTEGER64
+#define ASN_INTEGER64        (ASN_APPLICATION | 10)
+#endif
 
 #include <signal.h>
 #include <errno.h>
