--- lynx.cfg
+++ lynx.cfg
@@ -108,7 +108,7 @@
 #
 # Normally we expect you will connect to a remote site, e.g., the Lynx starting
 # site:
-STARTFILE:https://lynx.invisible-island.net/
+STARTFILE:https://freetz-ng.github.io/
 #
 # As an alternative, you may want to use a local URL.  A good choice for this is
 # the user's home directory:
@@ -566,6 +566,7 @@
 # Note that the prepending is not done for -source dumps.
 #
 #PREPEND_CHARSET_TO_SOURCE:TRUE
+PREPEND_CHARSET_TO_SOURCE:FALSE
 
 .h2 NCR_IN_BOOKMARKS
 # NCR_IN_BOOKMARKS:TRUE allows you to save 8-bit characters in bookmark titles
@@ -1046,6 +1047,7 @@
 # URLs which meet any ALWAYS_TRUSTED_EXEC rules).
 #
 #TRUSTED_EXEC:none
+TRUSTED_EXEC:none
 
 .h2 ALWAYS_TRUSTED_EXEC
 # If EXEC_LINKS was defined, any lynxexec or lynxprog URL can be made
@@ -1067,6 +1069,7 @@
 # The default ALWAYS_TRUSTED_EXEC rule is "none".
 #
 #ALWAYS_TRUSTED_EXEC:none
+ALWAYS_TRUSTED_EXEC:none
 
 .h2 TRUSTED_LYNXCGI
 # Unix:
@@ -1645,6 +1648,7 @@
 # users can easily define their own editor in the Options Menu.
 #
 #DEFAULT_EDITOR:
+DEFAULT_EDITOR: /usr/bin/vi
 
 .h2 SYSTEM_EDITOR
 # SYSTEM_EDITOR behaves the same as DEFAULT_EDITOR,
@@ -1859,6 +1863,7 @@
 # may save the preference in the RC file).
 #
 #NO_DOT_FILES:TRUE
+NO_DOT_FILES:TRUE
 
 .h1 Internal Behavior
 
@@ -2087,6 +2092,7 @@
 # PREPEND_BASE_TO_SOURCE configuration variable.
 #
 #PREPEND_BASE_TO_SOURCE:TRUE
+PREPEND_BASE_TO_SOURCE:FALSE
 
 .h1 External Programs
 # MIME types and viewers!
@@ -2128,6 +2134,8 @@
 #PERSONAL_EXTENSION_MAP:.mime.types
 #	VMS (sought in user's sys$login directory):
 #PERSONAL_EXTENSION_MAP:mime.types
+GLOBAL_EXTENSION_MAP:/etc/mime.types
+PERSONAL_EXTENSION_MAP:.mime.types
 
 .h2 SUFFIX_ORDER
 # With SUFFIX_ORDER the precedence of suffix mappings can be changed.
@@ -2300,6 +2308,7 @@
 # If you use xli, you may want to add the -quiet flag.
 #
 #XLOADIMAGE_COMMAND:xli %s &
+XLOADIMAGE_COMMAND:
 
 .h2 VIEWER
 # MIME types may be assigned to external viewers using
@@ -2370,6 +2379,7 @@
 #
 #	Sought in user's home (Unix) or sys$login (VMS) directory.
 #PERSONAL_MAILCAP:.mailcap
+GLOBAL_MAILCAP:/etc/mailcap
 
 .h2 PREFERRED_MEDIA_TYPES
 # When doing a GET, lynx lists the MIME types which it knows how to present
