--- userdefs.h
+++ userdefs.h
@@ -105,7 +105,8 @@
  * mailcap files (see the examples in the samples directory).
  */
 #ifndef LYNX_CFG_FILE
-#define LYNX_CFG_FILE "Lynx_Dir:lynx.cfg"
+/* #define LYNX_CFG_FILE "Lynx_Dir:lynx.cfg" */
+#define LYNX_CFG_FILE "/etc/lynx.cfg"
 #endif /* LYNX_CFG_FILE */
 
 #ifndef LYNX_CFG_PATH
@@ -119,8 +120,10 @@
  * Mappings in these global and personal files override any SUFFIX
  * definitions in lynx.cfg and built-in defaults from src/HTInit.c.
  */
-#define GLOBAL_EXTENSION_MAP "Lynx_Dir:mime.types"
-#define PERSONAL_EXTENSION_MAP "mime.types"
+/* #define GLOBAL_EXTENSION_MAP "Lynx_Dir:mime.types" */
+#define GLOBAL_EXTENSION_MAP "/etc/mime.types"
+/* #define PERSONAL_EXTENSION_MAP "mime.types" */
+#define PERSONAL_EXTENSION_MAP ".mime.types"
 
 /**************************
  * The MAILCAP file allows you to map file MIME types to
@@ -129,7 +132,8 @@
  * Mappings in these global and personal files override any VIEWER
  * definitions in lynx.cfg and built-in defaults from src/HTInit.c.
  */
-#define GLOBAL_MAILCAP "Lynx_Dir:mailcap"
+/* #define GLOBAL_MAILCAP "Lynx_Dir:mailcap" */
+#define GLOBAL_MAILCAP "/etc/mailcap"
 #define PERSONAL_MAILCAP ".mailcap"
 
 /**************************
@@ -146,7 +150,8 @@
  * Make this NULL if you don't have such a viewer or don't want to
  * use any default viewers for image types.
  */
-#define XLOADIMAGE_COMMAND "xv %s"
+/* #define XLOADIMAGE_COMMAND "xv %s" */
+#define XLOADIMAGE_COMMAND NULL
 
 /**************************
  * SYSTEM_MAIL must be defined here to your mail sending command,
@@ -300,8 +305,10 @@
 #define LYNX_CFG_PATH "."
 #define LYNX_CFG_FILE "./lynx.cfg"
 #else
-#define LYNX_CFG_PATH "/usr/local/lib"
-#define LYNX_CFG_FILE "/usr/local/lib/lynx.cfg"
+/* #define LYNX_CFG_PATH "/usr/local/lib" */
+/* #define LYNX_CFG_FILE "/usr/local/lib/lynx.cfg" */
+#define LYNX_CFG_PATH "/etc"
+#define LYNX_CFG_FILE "/etc/lynx.cfg"
 #endif /* DOSPATH */
 #endif /* LYNX_CFG_FILE */
 #endif /* HAVE_CONFIG_H */
@@ -317,7 +324,8 @@
  * Mappings in these global and personal files override any SUFFIX
  * definitions in lynx.cfg and built-in defaults from src/HTInit.c.
  */
-#define GLOBAL_EXTENSION_MAP MIME_LIBDIR "mime.types"
+/* #define GLOBAL_EXTENSION_MAP MIME_LIBDIR "mime.types" */
+#define GLOBAL_EXTENSION_MAP "/etc/mime.types"
 #define PERSONAL_EXTENSION_MAP "~/.mime.types"
 
 /**************************
@@ -327,7 +335,8 @@
  * Mappings in these global and personal files override any VIEWER
  * definitions in lynx.cfg and built-in defaults from src/HTInit.c.
  */
-#define GLOBAL_MAILCAP MIME_LIBDIR "mailcap"
+/* #define GLOBAL_MAILCAP MIME_LIBDIR "mailcap" */
+#define GLOBAL_MAILCAP "/etc/mailcap"
 #define PERSONAL_MAILCAP "~/.mailcap"
 
 /**************************
