--- lynx2-8-5.orig/userdefs.h	2004-02-02 21:02:28.000000000 +0100
+++ lynx2-8-5/userdefs.h	2007-03-02 13:15:19.054353410 +0100
@@ -100,7 +100,8 @@
  * mailcap files (see the examples in the samples directory).
  */
 #ifndef LYNX_CFG_FILE
-#define LYNX_CFG_FILE "Lynx_Dir:lynx.cfg"
+/* #define LYNX_CFG_FILE "Lynx_Dir:lynx.cfg" */
+#define LYNX_CFG_FILE "/etc/lynx.cfg"
 #endif /* LYNX_CFG_FILE */
 
 /**************************
@@ -110,8 +111,10 @@
  * Mappings in these global and personal files override any SUFFIX
  * definitions in lynx.cfg and built-in defaults from src/HTInit.c.
  */
-#define GLOBAL_EXTENSION_MAP "Lynx_Dir:mime.types"
-#define PERSONAL_EXTENSION_MAP "mime.types"
+/* #define GLOBAL_EXTENSION_MAP "Lynx_Dir:mime.types" */
+#define GLOBAL_EXTENSION_MAP "/etc/mime.types"
+/* #define PERSONAL_EXTENSION_MAP "mime.types" */
+#define PERSONAL_EXTENSION_MAP ".mime.types"
 
 /**************************
  * The MAILCAP file allows you to map file MIME types to
@@ -120,7 +123,8 @@
  * Mappings in these global and personal files override any VIEWER
  * definitions in lynx.cfg and built-in defaults from src/HTInit.c.
  */
-#define GLOBAL_MAILCAP "Lynx_Dir:mailcap"
+/* #define GLOBAL_MAILCAP "Lynx_Dir:mailcap" */
+#define GLOBAL_MAILCAP "/etc/mailcap"
 #define PERSONAL_MAILCAP ".mailcap"
 
 /**************************
@@ -137,7 +141,8 @@
  * Make this NULL if you don't have such a viewer or don't want to
  * use any default viewers for image types.
  */
-#define XLOADIMAGE_COMMAND "xv %s"
+/* #define XLOADIMAGE_COMMAND "xv %s" */
+#define XLOADIMAGE_COMMAND NULL
 
 /**************************
  * SYSTEM_MAIL must be defined here to your mail sending command,
@@ -284,7 +289,8 @@
 #ifdef DOSPATH
 #define LYNX_CFG_FILE "./lynx.cfg"
 #else
-#define LYNX_CFG_FILE "/usr/local/lib/lynx.cfg"
+/* #define LYNX_CFG_FILE "/usr/local/lib/lynx.cfg" */
+#define LYNX_CFG_FILE "/etc/lynx.cfg"
 #endif /* DOSPATH */
 #endif /* LYNX_CFG_FILE */
 #endif /* HAVE_CONFIG_H */
@@ -296,7 +302,8 @@
  * Mappings in these global and personal files override any SUFFIX
  * definitions in lynx.cfg and built-in defaults from src/HTInit.c.
  */
-#define GLOBAL_EXTENSION_MAP "/usr/local/lib/mosaic/mime.types"
+/* #define GLOBAL_EXTENSION_MAP "/usr/local/lib/mosaic/mime.types" */
+#define GLOBAL_EXTENSION_MAP "/etc/mime.types"
 #define PERSONAL_EXTENSION_MAP ".mime.types"
 
 /**************************
@@ -306,7 +313,8 @@
  * Mappings in these global and personal files override any VIEWER
  * definitions in lynx.cfg and built-in defaults from src/HTInit.c.
  */
-#define GLOBAL_MAILCAP "/usr/local/lib/mosaic/mailcap"
+/* #define GLOBAL_MAILCAP "/usr/local/lib/mosaic/mailcap" */
+#define GLOBAL_MAILCAP "/etc/mailcap"
 #define PERSONAL_MAILCAP ".mailcap"
 
 /**************************
