--- Makefile	2017-08-13 21:37:50.000000000 +0200
+++ Makefile	2017-08-13 21:48:24.768087383 +0200
@@ -23,19 +23,18 @@
 # tools
 #
 CC = $(CROSS_COMPILE)gcc
-RM = rm
 AR = $(CROSS_COMPILE)ar
 RANLIB = $(CROSS_COMPILE)ranlib
 #
 # flags for calling the tools
 #
-override CFLAGS += -W -Wall -std=c99 -O2 -fvisibility=hidden 
-#override CFLAGS += -W -Wall -std=c99 -O0 -ggdb -fvisibility=hidden
-#
-# how to build objects from sources
-#
-%.o: %.c
-	$(CC) $(CFLAGS) -c $< -o $@
+ifdef DO_DEBUG
+OPT = -O0 -ggdb
+else
+OPT = -O2
+override CPPFLAGS += -DNDEBUG
+endif
+override CFLAGS += -W -Wall -std=c99 $(OPT) -fvisibility=hidden
 #
 # generate position independent code for the library
 #
@@ -62,7 +61,7 @@
 # shared library
 #
 $(LIB_SO): $(LIB_OBJS) $(LIBHDR) libcrypt.h
-	$(CC) -shared -o $@ $(filter %.o,$^)
+	$(CC) -shared $(filter %.o,$^) -o $@
 #
 # static library
 #
@@ -74,7 +73,7 @@
 # the CLI binary
 #
 $(BINARY): $(BIN_OBJS) $(LIB_SO)
-	$(CC) $(LDFLAGS) $(filter %.o,$^) -L. -l$(BASENAME) -o $@ $(LIBS)
+	$(CC) $(LDFLAGS) $(filter %.o,$^) -L. -l$(BASENAME) -o $@
 #
 # everything to make, if header file changes
 #
