--- Makefile
+++ Makefile
@@ -31,19 +31,12 @@
 #
 override CFLAGS += -W -Wall -std=c99 -O2 -fvisibility=hidden 
 #override CFLAGS += -W -Wall -std=c99 -O0 -ggdb -fvisibility=hidden
-#
-# how to build objects from sources
-#
-%.o: %.c
-	$(CC) $(CFLAGS) -c $< -o $@
+
 #
 # generate position independent code for the library
 #
 $(LIB_OBJS): CFLAGS += -fPIC
-#
-# link binaries with this libraries too
-#
-LIBS = 
+
 #
 # targets to make
 #
@@ -73,8 +66,9 @@
 #
 # the CLI binary
 #
+$(BINARY): LDFLAGS+=-L.
+$(BINARY): LDLIBS+=-l$(BASENAME)
 $(BINARY): $(BIN_OBJS) | $(LIB_SO)
-	$(CC) $(LDFLAGS) $^ -L. -l$(BASENAME) -o $@ $(LIBS)
 #
 # everything to make, if header file changes
 #
