--- Makefile
+++ Makefile
@@ -29,8 +29,13 @@
 #
 # flags for calling the tools
 #
-override CFLAGS += -W -Wall -std=c99 -O2 -fvisibility=hidden 
-#override CFLAGS += -W -Wall -std=c99 -O0 -ggdb -fvisibility=hidden
+ifdef DO_DEBUG
+OPT = -O0 -ggdb
+else
+OPT = -O2
+override CPPFLAGS += -DNDEBUG
+endif
+override CFLAGS += -W -Wall -std=c99 $(OPT) -fvisibility=hidden
 
 #
 # generate position independent code for the library
