--- Makefile.in.orig	2009-04-23 22:35:03.000000000 +0200
+++ Makefile.in	2009-07-28 11:40:53.000000000 +0200
@@ -190,7 +190,7 @@
 target_vendor = @target_vendor@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = src ctaocrypt examples testsuite
+SUBDIRS = src ctaocrypt
 EXTRA_DIST = cyassl.dsp cyassl.dsw certs/*.pem certs/*.der certs/*.txt \
     lib/dummy cyassl.sln cyassl.vcproj cyassl-iphone.xcodeproj/project.pbxproj
 
