--- builds/unix/configure	2007-05-15 18:51:19.000000000 +0200
+++ builds/unix/configure	2007-05-15 18:53:02.000000000 +0200
@@ -3270,7 +3270,7 @@
   test -z "$as_dir" && as_dir=.
   for ac_exec_ext in '' $ac_executable_extensions; do
   if { test -f "$as_dir/$ac_word$ac_exec_ext" && $as_test_x "$as_dir/$ac_word$ac_exec_ext"; }; then
-    ac_cv_prog_CC_BUILD="${build-gcc}"
+    ac_cv_prog_CC_BUILD="${build}-gcc"
     echo "$as_me:$LINENO: found $as_dir/$ac_word$ac_exec_ext" >&5
     break 2
   fi
--- builds/unix/configure.ac	2007-05-15 18:51:19.000000000 +0200
+++ builds/unix/configure.ac	2007-05-15 18:53:25.000000000 +0200
@@ -40,7 +40,7 @@
 # checks for native programs to generate building tool
 
 if test ${cross_compiling} = yes; then
-  AC_CHECK_PROG(CC_BUILD, ${build}-gcc, ${build-gcc})
+  AC_CHECK_PROG(CC_BUILD, ${build}-gcc, ${build}-gcc)
   test -z "${CC_BUILD}" && AC_CHECK_PROG(CC_BUILD, gcc, gcc)
   test -z "${CC_BUILD}" && AC_CHECK_PROG(CC_BUILD, cc, cc, , , /usr/ucb/cc)
   test -z "${CC_BUILD}" && AC_MSG_ERROR([cannot find native C compiler])
--- builds/unix/configure.raw	2007-05-15 18:51:19.000000000 +0200
+++ builds/unix/configure.raw	2007-05-15 18:54:07.000000000 +0200
@@ -40,7 +40,7 @@
 # checks for native programs to generate building tool
 
 if test ${cross_compiling} = yes; then
-  AC_CHECK_PROG(CC_BUILD, ${build}-gcc, ${build-gcc})
+  AC_CHECK_PROG(CC_BUILD, ${build}-gcc, ${build}-gcc)
   test -z "${CC_BUILD}" && AC_CHECK_PROG(CC_BUILD, gcc, gcc)
   test -z "${CC_BUILD}" && AC_CHECK_PROG(CC_BUILD, cc, cc, , , /usr/ucb/cc)
   test -z "${CC_BUILD}" && AC_MSG_ERROR([cannot find native C compiler])
