--- Makefile.in.orig	2008-07-25 19:28:11.000000000 +0200
+++ Makefile.in	2008-11-22 15:46:46.000000000 +0100
@@ -190,7 +190,7 @@
 target_vendor = @target_vendor@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = src ctaocrypt examples testsuite
+SUBDIRS = src ctaocrypt
 EXTRA_DIST = cyassl.dsp cyassl.dsw certs/*.pem certs/*.der certs/*.txt \
     lib/dummy cyassl.sln cyassl.vcproj
 
