--- include/polarssl/config.h
+++ include/polarssl/config.h
@@ -71,12 +71,12 @@
 /*
  * Enable all SSL/TLS debugging messages.
  */
-#define POLARSSL_DEBUG_MSG
+/* #define POLARSSL_DEBUG_MSG */
 
 /*
  * Enable the checkup functions (*_self_test).
  */
-#define POLARSSL_SELF_TEST
+/* #define POLARSSL_SELF_TEST */
 
 /*
  * Enable the prime-number generation code.
@@ -156,7 +156,7 @@
  *
  * This module provides debugging functions.
  */
-#define POLARSSL_DEBUG_C
+/* #define POLARSSL_DEBUG_C */
 
 /*
  * Module:  library/des.c
@@ -229,7 +229,7 @@
  *
  * This modules adds support for the VIA PadLock on x86.
  */
-#define POLARSSL_PADLOCK_C
+/* #define POLARSSL_PADLOCK_C */
 
 /*
  * Module:  library/rsa.c
