--- library/debug.c
+++ library/debug.c
@@ -182,7 +182,7 @@
     prefix[maxlen] = '\0';
     maxlen = sizeof( str ) - 1;
 
-    while( crt != NULL && crt->next != NULL )
+    while( crt != NULL )
     {
         char buf[1024];
         x509parse_cert_info( buf, sizeof( buf ) - 1, prefix, crt );
--- library/ssl_srv.c
+++ library/ssl_srv.c
@@ -485,7 +485,7 @@
     p += 2;
     crt = ssl->ca_chain;
 
-    while( crt != NULL && crt->next != NULL )
+    while( crt != NULL )
     {
         if( p - buf > 4096 )
             break;
--- library/ssl_tls.c
+++ library/ssl_tls.c
@@ -1160,7 +1160,7 @@
     i = 7;
     crt = ssl->own_cert;
 
-    while( crt != NULL && crt->next != NULL )
+    while( crt != NULL )
     {
         n = crt->raw.len;
         if( i + 3 + n > SSL_MAX_CONTENT_LEN )
