--- Makefile.am	2010-01-18 08:52:26.000000000 +0100
+++ Makefile.am	2010-03-30 16:30:07.000000000 +0200
@@ -27,7 +27,7 @@
 	windows/test-parser/test-parser.vcproj			\
 	windows/test-tree/test-tree.vcproj
 
-SUBDIRS = gl lib src examples tests doc
+SUBDIRS = gl lib
 
 ACLOCAL_AMFLAGS = -I m4 -I gl/m4 -I lib/glm4
 
--- Makefile.in	2010-03-15 14:32:18.000000000 +0100
+++ Makefile.in	2010-03-30 16:30:21.000000000 +0200
@@ -524,7 +524,7 @@
 	windows/libtasn1/libtasn1.vcproj \
 	windows/test-parser/test-parser.vcproj \
 	windows/test-tree/test-tree.vcproj cfg.mk maint.mk .clcopying
-SUBDIRS = gl lib src examples tests doc
+SUBDIRS = gl lib
 ACLOCAL_AMFLAGS = -I m4 -I gl/m4 -I lib/glm4
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
