--- configure.ac.orig	2008-04-02 19:17:39.000000000 +0200
+++ configure.ac	2008-12-23 17:41:04.000000000 +0100
@@ -45,7 +45,7 @@
 AM_C_PROTOTYPES
 AC_CHECK_VA_COPY
 
-AC_CHECK_HEADERS(float.h glob.h langinfo.h libintl.h mcheck.h unistd.h)
+AC_CHECK_HEADERS(float.h glob.h langinfo.h mcheck.h unistd.h)
 
 # For some systems we know that we have ld_version scripts.
 # Use it then as default.
@@ -69,7 +69,7 @@
 AC_CHECK_FUNC(setreuid, [], [
     AC_CHECK_LIB(ucb, setreuid, [if echo $LIBS | grep -- -lucb >/dev/null ;then :; else LIBS="$LIBS -lc -lucb" USEUCB=y;fi])
 ])
-AC_CHECK_FUNCS(getuid geteuid iconv mtrace __secure_getenv setregid stpcpy strerror vasprintf)
+AC_CHECK_FUNCS(getuid geteuid mtrace __secure_getenv setregid stpcpy strerror vasprintf)
 
 AM_GNU_GETTEXT([external])
 AM_ICONV_LINK
