--- crypto/hmac/hmac.c.orig	2008-06-16 18:56:40.000000000 +0200
+++ crypto/hmac/hmac.c	2008-10-04 22:55:02.000000000 +0200
@@ -178,3 +178,11 @@
 	EVP_MD_CTX_set_flags(&ctx->md_ctx, flags);
 	}
 
+// Define alias names for compatibility with AVMs libcrypto
+#define	DEFINE_ALIAS(n)	extern __typeof (n) ossl_##n __attribute__ ((alias (#n)));
+
+DEFINE_ALIAS (HMAC);
+DEFINE_ALIAS (HMAC_Init);
+DEFINE_ALIAS (HMAC_Init_ex);
+DEFINE_ALIAS (HMAC_Update);
+DEFINE_ALIAS (HMAC_Final);
