--- ltmain.sh.orig	2005-07-29 15:45:48.000000000 +0200
+++ ltmain.sh	2006-02-25 13:30:52.861058049 +0100
@@ -1886,7 +1886,9 @@
 	      fi
 	    done
 	  done
-	  if test "$found" != yes; then
+	  # cross compile hack
+	  if true; then
+	  #if test "$found" != yes; then
 	    # deplib doesn't seem to be a libtool library
 	    if test "$linkmode,$pass" = "prog,link"; then
 	      compile_deplibs="$deplib $compile_deplibs"
