--- Makefile.am.orig	2008-06-16 14:36:35.000000000 +0200
+++ Makefile.am	2008-11-09 01:31:07.000000000 +0100
@@ -2,7 +2,7 @@
 # have all needed files, that a GNU package needs
 AUTOMAKE_OPTIONS = foreign 1.4
 
-SUBDIRS = src examples doc
+SUBDIRS = src
 
 EXTRA_DIST = libftdi.spec libftdi.spec.in COPYING.LIB README AUTHORS ChangeLog libftdi-config.in
 
--- Makefile.in.orig	2008-09-09 13:38:34.000000000 +0200
+++ Makefile.in	2008-11-09 01:31:12.000000000 +0100
@@ -198,7 +198,7 @@
 # not a GNU package. You can remove this line, if
 # have all needed files, that a GNU package needs
 AUTOMAKE_OPTIONS = foreign 1.4
-SUBDIRS = src examples doc
+SUBDIRS = src
 EXTRA_DIST = libftdi.spec libftdi.spec.in COPYING.LIB README AUTHORS ChangeLog libftdi-config.in
 bin_SCRIPTS = libftdi-config
 
