--- configure.orig	2004-02-05 10:34:07.000000000 +0100
+++ configure	2008-05-17 01:18:45.000000000 +0200
@@ -19059,10 +19059,6 @@
 	-mno-cygwin)
 	    shift
 	    ;;
-	-m*)
-	    arch="$arch $1"
-	    shift
-	    ;;
 	-O2)
 	    optimize="-O"
 	    shift
@@ -19070,10 +19066,6 @@
 	-fomit-frame-pointer)
 	    shift
 	    ;;
-	-O*|-f*)
-	    optimize="$optimize $1"
-	    shift
-	    ;;
 	*)
 	    CFLAGS="$CFLAGS $1"
 	    shift
