--- dist/Makefile.in	2009-12-18 23:26:01.000000000 +0100
+++ dist/Makefile.in	2010-01-16 22:46:48.000000000 +0100
@@ -925,9 +925,9 @@
 # Library and standard utilities install.
 ##################################################
 library_install: install_setup
-library_install: install_include install_lib install_utilities install_docs
+library_install: install_include install_lib
 
-uninstall: uninstall_include uninstall_lib uninstall_utilities uninstall_docs
+uninstall: uninstall_include uninstall_lib
 
 install_setup:
 	@test -d $(DESTDIR)$(prefix) || \
