--- crypto/hmac/hmac.c.orig	2010-06-15 19:25:09.000000000 +0200
+++ crypto/hmac/hmac.c	2011-05-10 10:22:20.003532100 +0200
@@ -212,3 +212,12 @@
 	EVP_MD_CTX_set_flags(&ctx->o_ctx, flags);
 	EVP_MD_CTX_set_flags(&ctx->md_ctx, flags);
 	}
+
+// Define alias names for compatibility with AVMs libcrypto
+#define	DEFINE_ALIAS(n)	extern __typeof (n) ossl_##n __attribute__ ((alias (#n)));
+
+DEFINE_ALIAS (HMAC);
+DEFINE_ALIAS (HMAC_Init);
+DEFINE_ALIAS (HMAC_Init_ex);
+DEFINE_ALIAS (HMAC_Update);
+DEFINE_ALIAS (HMAC_Final);
