diff -burN pjproject-0.7.0.orig/Makefile pjproject-0.7.0/Makefile
--- pjproject-0.7.0.orig/Makefile	2007-04-22 14:48:30.000000000 +0200
+++ pjproject-0.7.0/Makefile	2007-07-01 02:35:16.000000000 +0200
@@ -16,16 +16,16 @@
 		fi; \
 	done
 
-LIBS = 	pjlib/lib/libpj-$(TARGET_NAME).a \
-	pjlib-util/lib/libpjlib-util-$(TARGET_NAME).a \
-	pjnath/lib/libpjnath-$(TARGET_NAME).a \
-	pjmedia/lib/libpjmedia-$(TARGET_NAME).a \
-	pjmedia/lib/libpjmedia-codec-$(TARGET_NAME).a \
-       	pjsip/lib/libpjsip-$(TARGET_NAME).a \
-	pjsip/lib/libpjsip-ua-$(TARGET_NAME).a \
-	pjsip/lib/libpjsip-simple-$(TARGET_NAME).a \
-	pjsip/lib/libpjsua-$(TARGET_NAME).a
-BINS = 	pjsip-apps/bin/pjsua-$(TARGET_NAME)$(HOST_EXE) 
+LIBS = 	pjlib/lib/libpj.a \
+	pjlib-util/lib/libpjlib-util.a \
+	pjnath/lib/libpjnath.a \
+	pjmedia/lib/libpjmedia.a \
+	pjmedia/lib/libpjmedia-codec.a \
+       	pjsip/lib/libpjsip.a \
+	pjsip/lib/libpjsip-ua.a \
+	pjsip/lib/libpjsip-simple.a \
+	pjsip/lib/libpjsua.a
+BINS = 	pjsip-apps/bin/pjsua$(HOST_EXE) 
 
 size:
 	@echo -n 'Date: '
@@ -56,14 +56,11 @@
 		cp /tmp/id $$f; \
 	done
 
-prefix = /usr/local
+prefix = /usr
 install:
 	mkdir -p $(DESTDIR)$(prefix)/lib
 	cp -L $$(find . -name '*.a') $(DESTDIR)$(prefix)/lib
 	mkdir -p $(DESTDIR)$(prefix)/include
 	cp -RL $$(find  . -name include) $(DESTDIR)$(prefix)
-	cd $(DESTDIR)$(prefix)/lib && for i in $$(find . -name 'libpj*a'); do\
-		ln -s $$i $$(echo $$i | sed -e "s/-$(TARGET_NAME)//");\
-	done
 	mkdir -p $(DESTDIR)$(prefix)/lib/pkgconfig
 	sed -e "s!@PREFIX@!$(DESTDIR)$(prefix)!" libpj.pc.in > $(DESTDIR)/$(prefix)/lib/pkgconfig/libpj.pc
diff -burN pjproject-0.7.0.orig/build.mak.in pjproject-0.7.0/build.mak.in
--- pjproject-0.7.0.orig/build.mak.in	2007-04-17 18:16:01.000000000 +0200
+++ pjproject-0.7.0/build.mak.in	2007-07-01 02:49:28.000000000 +0200
@@ -8,22 +8,22 @@
 export LINUX_POLL := @ac_linux_poll@ 
 
 # Determine which party libraries to use
-export APP_THIRD_PARTY_LIBS := -lresample-$(TARGET_NAME)
+export APP_THIRD_PARTY_LIBS := -lresample
 
 ifneq (@ac_no_gsm_codec@,1)
-APP_THIRD_PARTY_LIBS += -lgsmcodec-$(TARGET_NAME)
+APP_THIRD_PARTY_LIBS += -lgsmcodec
 endif
 
 ifneq (@ac_no_speex_codec@,1)
-APP_THIRD_PARTY_LIBS += -lspeex-$(TARGET_NAME)
+APP_THIRD_PARTY_LIBS += -lspeex
 endif
 
 ifneq (@ac_no_ilbc_codec@,1)
-APP_THIRD_PARTY_LIBS += -lilbccodec-$(TARGET_NAME)
+APP_THIRD_PARTY_LIBS += -lilbccodec
 endif
 
 ifneq ($(findstring pa,@ac_pjmedia_snd@),)
-APP_THIRD_PARTY_LIBS += -lportaudio-$(TARGET_NAME)
+APP_THIRD_PARTY_LIBS += -lportaudio
 endif
 
 
@@ -45,16 +45,16 @@
 	-L$(PJDIR)/pjsip/lib\
 	-L$(PJDIR)/third_party/lib\
 	@LDFLAGS@
-export APP_LDLIBS := -lpjsua-$(TARGET_NAME)\
-	-lpjsip-ua-$(TARGET_NAME)\
-	-lpjsip-simple-$(TARGET_NAME)\
-	-lpjsip-$(TARGET_NAME)\
-	-lpjmedia-codec-$(TARGET_NAME)\
-	-lpjmedia-$(TARGET_NAME)\
-	-lpjnath-$(TARGET_NAME)\
-	-lpjlib-util-$(TARGET_NAME)\
+export APP_LDLIBS := -lpjsua \
+	-lpjsip-ua \
+	-lpjsip-simple \
+	-lpjsip \
+	-lpjmedia-codec \
+	-lpjmedia \
+	-lpjnath \
+	-lpjlib-util \
 	$(APP_THIRD_PARTY_LIBS)\
-	-lpj-$(TARGET_NAME)\
+	-lpj \
 	@LIBS@
 
 export PJ_DIR := $(PJDIR)
diff -burN pjproject-0.7.0.orig/pjlib/build/Makefile pjproject-0.7.0/pjlib/build/Makefile
--- pjproject-0.7.0.orig/pjlib/build/Makefile	2007-04-09 23:28:15.000000000 +0200
+++ pjproject-0.7.0/pjlib/build/Makefile	2007-07-01 02:53:45.000000000 +0200
@@ -4,7 +4,7 @@
 RULES_MAK := $(PJDIR)/build/rules.mak
 
 
-export PJLIB_LIB := ../lib/libpj-$(TARGET_NAME)$(LIBEXT)
+export PJLIB_LIB := ../lib/libpj$(LIBEXT)
 
 ###############################################################################
 # Gather all flags.
@@ -43,7 +43,7 @@
 		    util.o
 export TEST_CFLAGS += $(_CFLAGS)
 export TEST_LDFLAGS += $(_LDFLAGS)
-export TEST_EXE := ../bin/pjlib-test-$(TARGET_NAME)$(HOST_EXE)
+export TEST_EXE := ../bin/pjlib-test$(HOST_EXE)
 
 	
 export CC_OUT CC AR RANLIB HOST_MV HOST_RM HOST_RMDIR HOST_MKDIR OBJEXT LD LDOUT 
@@ -65,7 +65,7 @@
 depend: ../include/pj/config_site.h
 	$(MAKE) -f $(RULES_MAK) APP=PJLIB app=pjlib depend
 	$(MAKE) -f $(RULES_MAK) APP=TEST app=pjlib-test depend
-	echo '$(TEST_EXE): $(PJLIB_LIB)' >> .pjlib-test-$(TARGET_NAME).depend
+	echo '$(TEST_EXE): $(PJLIB_LIB)' >> .pjlib-test.depend
 
 
 .PHONY: dep depend pjlib pjlib-test clean realclean distclean
@@ -95,8 +95,8 @@
 	$(MAKE) -f $(RULES_MAK) APP=TEST app=pjlib-test clean
 
 realclean:
-	$(subst @@,$(subst /,$(HOST_PSEP),.pjlib-$(TARGET_NAME).depend),$(HOST_RMR))
-	$(subst @@,$(subst /,$(HOST_PSEP),.pjlib-test-$(TARGET_NAME).depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.pjlib.depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.pjlib-test.depend),$(HOST_RMR))
 	$(MAKE) -f $(RULES_MAK) APP=PJLIB app=pjlib realclean
 	$(MAKE) -f $(RULES_MAK) APP=TEST app=pjlib-test realclean
 
diff -burN pjproject-0.7.0.orig/pjlib-util/build/Makefile pjproject-0.7.0/pjlib-util/build/Makefile
--- pjproject-0.7.0.orig/pjlib-util/build/Makefile	2007-04-09 23:28:15.000000000 +0200
+++ pjproject-0.7.0/pjlib-util/build/Makefile	2007-07-01 02:53:32.000000000 +0200
@@ -7,8 +7,8 @@
 
 RULES_MAK := $(PJDIR)/build/rules.mak
 
-PJLIB_LIB:=$(PJDIR)/pjlib/lib/libpj-$(TARGET_NAME)$(LIBEXT)
-export PJLIB_UTIL_LIB:=../lib/libpjlib-util-$(TARGET_NAME)$(LIBEXT)
+PJLIB_LIB:=$(PJDIR)/pjlib/lib/libpj$(LIBEXT)
+export PJLIB_UTIL_LIB:=../lib/libpjlib-util$(LIBEXT)
 
 ###############################################################################
 # Gather all flags.
@@ -40,7 +40,7 @@
 export UTIL_TEST_OBJS += xml.o encryption.o stun.o test.o 
 export UTIL_TEST_CFLAGS += $(_CFLAGS)
 export UTIL_TEST_LDFLAGS += $(_LDFLAGS)
-export UTIL_TEST_EXE:=../bin/pjlib-util-test-$(TARGET_NAME)$(HOST_EXE)
+export UTIL_TEST_EXE:=../bin/pjlib-util-test$(HOST_EXE)
 
 	
 export CC_OUT CC AR RANLIB HOST_MV HOST_RM HOST_RMDIR HOST_MKDIR OBJEXT LD LDOUT 
@@ -64,7 +64,7 @@
 pjlib-util:
 	$(MAKE) -f $(RULES_MAK) APP=PJLIB_UTIL app=pjlib-util $(PJLIB_UTIL_LIB)
 
-pjlib-util-test: $(PJLIB_LIB)
+pjlib-util-test: $(PJLIB_LIB) pjlib-util
 	$(MAKE) -f $(RULES_MAK) APP=UTIL_TEST app=pjlib-util-test $(UTIL_TEST_EXE)
 
 .PHONY: ../lib/pjlib-util.ko
@@ -81,14 +81,14 @@
 	$(MAKE) -f $(RULES_MAK) APP=UTIL_TEST app=pjlib-util-test $@
 
 realclean:
-	$(subst @@,$(subst /,$(HOST_PSEP),.pjlib-util-$(TARGET_NAME).depend),$(HOST_RMR))
-	$(subst @@,$(subst /,$(HOST_PSEP),.pjlib-util-test-$(TARGET_NAME).depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.pjlib-util.depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.pjlib-util-test.depend),$(HOST_RMR))
 	$(MAKE) -f $(RULES_MAK) APP=PJLIB_UTIL app=pjlib-util $@
 	$(MAKE) -f $(RULES_MAK) APP=UTIL_TEST app=pjlib-util-test $@
 
 depend:
 	$(MAKE) -f $(RULES_MAK) APP=PJLIB_UTIL app=pjlib-util $@
 	$(MAKE) -f $(RULES_MAK) APP=UTIL_TEST app=pjlib-util-test $@
-	echo '$(TEST_EXE): $(PJLIB_UTIL_LIB)' >> .pjlib-util-test-$(TARGET_NAME).depend; \
+	echo '$(TEST_EXE): $(PJLIB_UTIL_LIB)' >> .pjlib-util-test.depend; \
 
 
diff -burN pjproject-0.7.0.orig/pjmedia/build/Makefile pjproject-0.7.0/pjmedia/build/Makefile
--- pjproject-0.7.0.orig/pjmedia/build/Makefile	2007-04-18 11:24:31.000000000 +0200
+++ pjproject-0.7.0/pjmedia/build/Makefile	2007-07-01 02:53:23.000000000 +0200
@@ -7,13 +7,13 @@
 
 RULES_MAK := $(PJDIR)/build/rules.mak
 
-PJLIB_LIB:=$(PJDIR)/pjlib/lib/libpj-$(TARGET_NAME)$(LIBEXT)
-PJLIB_UTIL_LIB:=$(PJDIR)/pjlib-util/lib/libpjlib-util-$(TARGET_NAME)$(LIBEXT)
-PJNATH_LIB:=$(PJDIR)/pjnath/lib/libpjnath-$(TARGET_NAME)$(LIBEXT)
-
-export PJMEDIA_LIB:=../lib/libpjmedia-$(TARGET_NAME)$(LIBEXT)
-export PJMEDIA_CODEC_LIB:=../lib/libpjmedia-codec-$(TARGET_NAME)$(LIBEXT)
-export PJSDP_LIB:=../lib/libpjsdp-$(TARGET_NAME)$(LIBEXT)
+PJLIB_LIB:=$(PJDIR)/pjlib/lib/libpj$(LIBEXT)
+PJLIB_UTIL_LIB:=$(PJDIR)/pjlib-util/lib/libpjlib-util$(LIBEXT)
+PJNATH_LIB:=$(PJDIR)/pjnath/lib/libpjnath$(LIBEXT)
+
+export PJMEDIA_LIB:=../lib/libpjmedia$(LIBEXT)
+export PJMEDIA_CODEC_LIB:=../lib/libpjmedia-codec$(LIBEXT)
+export PJSDP_LIB:=../lib/libpjsdp$(LIBEXT)
 
 NULLSOUND_OBJS := nullsound.o
 
@@ -85,7 +85,7 @@
 export PJMEDIA_TEST_OBJS += jbuf_test.o main.o rtp_test.o sdp_neg_test.o test.o
 export PJMEDIA_TEST_CFLAGS += $(_CFLAGS)
 export PJMEDIA_TEST_LDFLAGS += $(_LDFLAGS)
-export PJMEDIA_TEST_EXE:=../bin/pjmedia-test-$(TARGET_NAME)$(HOST_EXE)
+export PJMEDIA_TEST_EXE:=../bin/pjmedia-test$(HOST_EXE)
 
 	
 export CC_OUT CC AR RANLIB HOST_MV HOST_RM HOST_RMDIR HOST_MKDIR OBJEXT LD LDOUT 
@@ -140,10 +140,10 @@
 	$(MAKE) -f $(RULES_MAK) APP=PJMEDIA_TEST app=pjmedia-test $@
 
 realclean:
-	$(subst @@,$(subst /,$(HOST_PSEP),.pjmedia-$(TARGET_NAME).depend),$(HOST_RMR))
-	$(subst @@,$(subst /,$(HOST_PSEP),.pjmedia-codec-$(TARGET_NAME).depend),$(HOST_RMR))
-	$(subst @@,$(subst /,$(HOST_PSEP),.pjmedia-test-$(TARGET_NAME).depend),$(HOST_RMR))
-	$(subst @@,$(subst /,$(HOST_PSEP),.pjsdp-$(TARGET_NAME).depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.pjmedia.depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.pjmedia-codec.depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.pjmedia-test.depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.pjsdp.depend),$(HOST_RMR))
 	
 	$(MAKE) -f $(RULES_MAK) APP=PJMEDIA app=pjmedia $@
 	$(MAKE) -f $(RULES_MAK) APP=PJMEDIA_CODEC app=pjmedia-codec $@
@@ -155,8 +155,8 @@
 	$(MAKE) -f $(RULES_MAK) APP=PJMEDIA_CODEC app=pjmedia-codec $@
 	$(MAKE) -f $(RULES_MAK) APP=PJMEDIA_TEST app=pjmedia-test $@
 	$(MAKE) -f $(RULES_MAK) APP=PJSDP app=pjsdp $@
-	echo '$(PJMEDIA_LIB): .pjmedia-$(TARGET_NAME).depend' >> .pjmedia-$(TARGET_NAME).depend; \
-	echo '$(PJMEDIA_CODEC_LIB): .pjmedia-codec-$(TARGET_NAME).depend' >> .pjmedia-codec-$(TARGET_NAME).depend; \
-	echo '$(PJMEDIA_TEST_EXE): $(PJMEDIA_LIB) $(PJMEDIA_CODEC_LIB)' >> .pjmedia-test-$(TARGET_NAME).depend; \
+	echo '$(PJMEDIA_LIB): .pjmedia.depend' >> .pjmedia.depend; \
+	echo '$(PJMEDIA_CODEC_LIB): .pjmedia-codec.depend' >> .pjmedia-codec.depend; \
+	echo '$(PJMEDIA_TEST_EXE): $(PJMEDIA_LIB) $(PJMEDIA_CODEC_LIB)' >> .pjmedia-test.depend; \
 
 
diff -burN pjproject-0.7.0.orig/pjnath/build/Makefile pjproject-0.7.0/pjnath/build/Makefile
--- pjproject-0.7.0.orig/pjnath/build/Makefile	2007-05-11 21:02:46.000000000 +0200
+++ pjproject-0.7.0/pjnath/build/Makefile	2007-07-01 02:53:11.000000000 +0200
@@ -7,9 +7,9 @@
 
 RULES_MAK := $(PJDIR)/build/rules.mak
 
-PJLIB_LIB:=../../pjlib/lib/libpj-$(TARGET_NAME)$(LIBEXT)
-PJLIB_UTIL_LIB:=../../pjlib-util/lib/libpjlib-util-$(TARGET_NAME)$(LIBEXT)
-export PJNATH_LIB:=../lib/libpjnath-$(TARGET_NAME)$(LIBEXT)
+PJLIB_LIB:=../../pjlib/lib/libpj$(LIBEXT)
+PJLIB_UTIL_LIB:=../../pjlib-util/lib/libpjlib-util$(LIBEXT)
+export PJNATH_LIB:=../lib/libpjnath$(LIBEXT)
 
 ###############################################################################
 # Gather all flags.
@@ -41,7 +41,7 @@
 export PJNATH_TEST_OBJS += ice_test.o stun.o test.o
 export PJNATH_TEST_CFLAGS += $(_CFLAGS)
 export PJNATH_TEST_LDFLAGS += $(_LDFLAGS)
-export PJNATH_TEST_EXE:=../bin/pjnath-test-$(TARGET_NAME)$(HOST_EXE)
+export PJNATH_TEST_EXE:=../bin/pjnath-test$(HOST_EXE)
 
 	
 ###############################################################################
@@ -51,7 +51,7 @@
 export PJSTUN_CLIENT_OBJS += client_main.o
 export PJSTUN_CLIENT_CFLAGS += $(_CFLAGS)
 export PJSTUN_CLIENT_LDFLAGS += $(_LDFLAGS)
-export PJSTUN_CLIENT_EXE:=../bin/pjstun-client-$(TARGET_NAME)$(HOST_EXE)
+export PJSTUN_CLIENT_EXE:=../bin/pjstun-client$(HOST_EXE)
 
 ###############################################################################
 # Defines for building test server application
@@ -61,7 +61,7 @@
 				main.o
 export PJSTUN_SRV_TEST_CFLAGS += $(_CFLAGS)
 export PJSTUN_SRV_TEST_LDFLAGS += $(_LDFLAGS)
-export PJSTUN_SRV_TEST_EXE:=../bin/pjstun-srv-test-$(TARGET_NAME)$(HOST_EXE)
+export PJSTUN_SRV_TEST_EXE:=../bin/pjstun-srv-test$(HOST_EXE)
 
 	
 	
@@ -86,13 +86,13 @@
 pjnath:
 	$(MAKE) -f $(RULES_MAK) APP=PJNATH app=pjnath $(PJNATH_LIB)
 
-pjnath-test: $(PJLIB_LIB) $(PJLIB_UTIL_LIB) $(PJNATH_LIB)
+pjnath-test: $(PJLIB_LIB) $(PJLIB_UTIL_LIB) pjnath
 	$(MAKE) -f $(RULES_MAK) APP=PJNATH_TEST app=pjnath-test $(PJNATH_TEST_EXE)
 
-pjstun-client: $(PJLIB_LIB) $(PJLIB_UTIL_LIB) $(PJNATH_LIB)
+pjstun-client: $(PJLIB_LIB) $(PJLIB_UTIL_LIB) pjnath
 	$(MAKE) -f $(RULES_MAK) APP=PJSTUN_CLIENT app=pjstun-client $(PJSTUN_CLIENT_EXE)
 
-pjstun-srv-test: $(PJLIB_LIB) $(PJLIB_UTIL_LIB) $(PJNATH_LIB)
+pjstun-srv-test: $(PJLIB_LIB) $(PJLIB_UTIL_LIB) pjnath
 	$(MAKE) -f $(RULES_MAK) APP=PJSTUN_SRV_TEST app=pjstun-srv-test $(PJSTUN_SRV_TEST_EXE)
 
 .PHONY: ../lib/pjnath.ko
@@ -111,10 +111,10 @@
 	$(MAKE) -f $(RULES_MAK) APP=PJSTUN_SRV_TEST app=pjstun-srv-test $@
 
 realclean:
-	$(subst @@,$(subst /,$(HOST_PSEP),.pjnath-$(TARGET_NAME).depend),$(HOST_RMR))
-	$(subst @@,$(subst /,$(HOST_PSEP),.pjnath-test-$(TARGET_NAME).depend),$(HOST_RMR))
-	$(subst @@,$(subst /,$(HOST_PSEP),.pjstun-client-$(TARGET_NAME).depend),$(HOST_RMR))
-	$(subst @@,$(subst /,$(HOST_PSEP),.pjstun-srv-test-$(TARGET_NAME).depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.pjnath.depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.pjnath-test.depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.pjstun-client.depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.pjstun-srv-test.depend),$(HOST_RMR))
 	$(MAKE) -f $(RULES_MAK) APP=PJNATH app=pjnath $@
 	$(MAKE) -f $(RULES_MAK) APP=PJNATH_TEST app=pjnath-test $@
 	$(MAKE) -f $(RULES_MAK) APP=PJSTUN_CLIENT app=pjstun-client $@
@@ -125,6 +125,6 @@
 	$(MAKE) -f $(RULES_MAK) APP=PJNATH_TEST app=pjnath-test $@
 	$(MAKE) -f $(RULES_MAK) APP=PJSTUN_CLIENT app=pjstun-client $@
 	$(MAKE) -f $(RULES_MAK) APP=PJSTUN_SRV_TEST app=pjstun-srv-test $@
-	echo '$(TEST_EXE): $(PJNATH_LIB)' >> .pjnath-test-$(TARGET_NAME).depend; \
+	echo '$(TEST_EXE): $(PJNATH_LIB)' >> .pjnath-test.depend; \
 
 
diff -burN pjproject-0.7.0.orig/pjsip/build/Makefile pjproject-0.7.0/pjsip/build/Makefile
--- pjproject-0.7.0.orig/pjsip/build/Makefile	2007-05-11 17:14:34.000000000 +0200
+++ pjproject-0.7.0/pjsip/build/Makefile	2007-07-01 02:52:57.000000000 +0200
@@ -8,15 +8,15 @@
 
 RULES_MAK := $(PJDIR)/build/rules.mak
 
-PJLIB_LIB:=../../pjlib/lib/libpj-$(TARGET_NAME)$(LIBEXT)
-PJLIB_UTIL_LIB:=../../pjlib-util/lib/libpjlib-util-$(TARGET_NAME)$(LIBEXT)
-PJMEDIA_LIB:=../../pjmedia/lib/libpjmedia-$(TARGET_NAME)$(LIBEXT)
-PJMEDIA_CODEC_LIB:=../../pjmedia/lib/libpjmedia-codec-$(TARGET_NAME)$(LIBEXT)
-
-export PJSIP_LIB:=../lib/libpjsip-$(TARGET_NAME)$(LIBEXT)
-export PJSIP_UA_LIB:=../lib/libpjsip-ua-$(TARGET_NAME)$(LIBEXT)
-export PJSIP_SIMPLE_LIB:=../lib/libpjsip-simple-$(TARGET_NAME)$(LIBEXT)
-export PJSUA_LIB_LIB=../lib/libpjsua-$(TARGET_NAME)$(LIBEXT)
+PJLIB_LIB:=../../pjlib/lib/libpj$(LIBEXT)
+PJLIB_UTIL_LIB:=../../pjlib-util/lib/libpjlib-util$(LIBEXT)
+PJMEDIA_LIB:=../../pjmedia/lib/libpjmedia$(LIBEXT)
+PJMEDIA_CODEC_LIB:=../../pjmedia/lib/libpjmedia-codec$(LIBEXT)
+
+export PJSIP_LIB:=../lib/libpjsip$(LIBEXT)
+export PJSIP_UA_LIB:=../lib/libpjsip-ua$(LIBEXT)
+export PJSIP_SIMPLE_LIB:=../lib/libpjsip-simple$(LIBEXT)
+export PJSUA_LIB_LIB=../lib/libpjsua$(LIBEXT)
 
 
 ###############################################################################
@@ -90,7 +90,7 @@
 		    tsx_uas_test.o txdata_test.o uri_test.o
 export TEST_CFLAGS += $(_CFLAGS)
 export TEST_LDFLAGS += $(PJ_LDFLAGS) $(PJ_LDLIBS) $(LDFLAGS)
-export TEST_EXE := ../bin/pjsip-test-$(TARGET_NAME)$(HOST_EXE)
+export TEST_EXE := ../bin/pjsip-test$(HOST_EXE)
 
 	
 export CC_OUT CC AR RANLIB HOST_MV HOST_RM HOST_RMDIR HOST_MKDIR OBJEXT LD LDOUT 
@@ -155,10 +155,10 @@
 	$(MAKE) -f $(RULES_MAK) APP=TEST app=pjsip-test $@
 
 realclean:
-	$(subst @@,$(subst /,$(HOST_PSEP),.pjsip-$(TARGET_NAME).depend),$(HOST_RMR))
-	$(subst @@,$(subst /,$(HOST_PSEP),.pjsip-ua-$(TARGET_NAME).depend),$(HOST_RMR))
-	$(subst @@,$(subst /,$(HOST_PSEP),.pjsip-simple-$(TARGET_NAME).depend),$(HOST_RMR))
-	$(subst @@,$(subst /,$(HOST_PSEP),.pjsua-lib-$(TARGET_NAME).depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.pjsip.depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.pjsip-ua.depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.pjsip-simple.depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.pjsua-lib.depend),$(HOST_RMR))
 	$(MAKE) -f $(RULES_MAK) APP=PJSIP app=pjsip $@
 	$(MAKE) -f $(RULES_MAK) APP=PJSIP_UA app=pjsip-ua $@
 	$(MAKE) -f $(RULES_MAK) APP=PJSIP_SIMPLE app=pjsip-simple $@
diff -burN pjproject-0.7.0.orig/pjsip-apps/build/Makefile pjproject-0.7.0/pjsip-apps/build/Makefile
--- pjproject-0.7.0.orig/pjsip-apps/build/Makefile	2007-04-17 18:16:01.000000000 +0200
+++ pjproject-0.7.0/pjsip-apps/build/Makefile	2007-07-01 02:54:03.000000000 +0200
@@ -4,15 +4,15 @@
 
 RULES_MAK := $(PJDIR)/build/rules.mak
 
-PJLIB_LIB:=../../pjlib/lib/libpj-$(TARGET_NAME)$(LIBEXT)
-PJLIB_UTIL_LIB:=../../pjlib-util/lib/libpjlib-util-$(TARGET_NAME)$(LIBEXT)
-PJNATH_LIB:=../../pjnath/lib/libpjnath-$(TARGET_NAME)$(LIBEXT)
-PJMEDIA_LIB:=../../pjmedia/lib/libpjmedia-$(TARGET_NAME)$(LIBEXT)
-PJMEDIA_CODEC_LIB:=../../pjmedia/lib/libpjmedia-codec-$(TARGET_NAME)$(LIBEXT)
-PJSIP_LIB:=../../pjsip/lib/libpjsip-$(TARGET_NAME)$(LIBEXT)
-PJSIP_UA_LIB:=../../pjsip/lib/libpjsip-ua-$(TARGET_NAME)$(LIBEXT)
-PJSIP_SIMPLE_LIB:=../../pjsip/lib/libpjsip-simple-$(TARGET_NAME)$(LIBEXT)
-PJSUA_LIB_LIB=../../pjsip/lib/libpjsua-$(TARGET_NAME)$(LIBEXT)
+PJLIB_LIB:=../../pjlib/lib/libpj$(LIBEXT)
+PJLIB_UTIL_LIB:=../../pjlib-util/lib/libpjlib-util$(LIBEXT)
+PJNATH_LIB:=../../pjnath/lib/libpjnath$(LIBEXT)
+PJMEDIA_LIB:=../../pjmedia/lib/libpjmedia$(LIBEXT)
+PJMEDIA_CODEC_LIB:=../../pjmedia/lib/libpjmedia-codec$(LIBEXT)
+PJSIP_LIB:=../../pjsip/lib/libpjsip$(LIBEXT)
+PJSIP_UA_LIB:=../../pjsip/lib/libpjsip-ua$(LIBEXT)
+PJSIP_SIMPLE_LIB:=../../pjsip/lib/libpjsip-simple$(LIBEXT)
+PJSUA_LIB_LIB=../../pjsip/lib/libpjsua$(LIBEXT)
 
 
 ###############################################################################
@@ -35,7 +35,7 @@
 			main.o pjsua_app.o
 export PJSUA_CFLAGS += $(_CFLAGS)
 export PJSUA_LDFLAGS += $(APP_LDFLAGS) $(APP_LDLIBS)
-export PJSUA_EXE:=../bin/pjsua-$(TARGET_NAME)$(HOST_EXE)
+export PJSUA_EXE:=../bin/pjsua$(HOST_EXE)
 
 
 export CC_OUT CC AR RANLIB HOST_MV HOST_RM HOST_RMDIR HOST_MKDIR OBJEXT LD LDOUT 
@@ -71,7 +71,7 @@
 	$(MAKE) -f $(RULES_MAK) APP=PJSUA app=pjsua $@
 	$(MAKE) -f Samples.mak $@
 	@if test "$@" = "depend"; then \
-	  echo '$(PJSUA_EXE): $(PJSIP_LIB) $(PJSIP_UA_LIB) $(PJSIP_SIMPLE) $(PJSUA_LIB_LIB) $(PJLIB_LIB) $(PJLIB_UTIL_LIB) $(PJNATH_LIB) $(PJMEDIA_LIB) $(PJMEDIA_CODEC_LIB) Makefile' >> .pjsua-$(TARGET_NAME).depend; \
+	  echo '$(PJSUA_EXE): $(PJSIP_LIB) $(PJSIP_UA_LIB) $(PJSIP_SIMPLE) $(PJSUA_LIB_LIB) $(PJLIB_LIB) $(PJLIB_UTIL_LIB) $(PJNATH_LIB) $(PJMEDIA_LIB) $(PJMEDIA_CODEC_LIB) Makefile' >> .pjsua.depend; \
 	fi
 
 
diff -burN pjproject-0.7.0.orig/third_party/build/gsm/Makefile pjproject-0.7.0/third_party/build/gsm/Makefile
--- pjproject-0.7.0.orig/third_party/build/gsm/Makefile	2007-04-09 19:13:10.000000000 +0200
+++ pjproject-0.7.0/third_party/build/gsm/Makefile	2007-07-01 03:00:02.000000000 +0200
@@ -5,7 +5,7 @@
 
 RULES_MAK := $(PJDIR)/build/rules.mak
 
-export GSM_CODEC_LIB := ../../lib/libgsmcodec-$(TARGET_NAME)$(LIBEXT)
+export GSM_CODEC_LIB := ../../lib/libgsmcodec$(LIBEXT)
 
 ###############################################################################
 # Gather all flags.
@@ -54,12 +54,12 @@
 	$(MAKE) -f $(RULES_MAK) APP=GSM_CODEC app=libgsmcodec $@
 
 realclean:
-	$(subst @@,$(subst /,$(HOST_PSEP),.ilbc-$(TARGET_NAME).depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.ilbc.depend),$(HOST_RMR))
 	
 	$(MAKE) -f $(RULES_MAK) APP=GSM_CODEC app=libgsmcodec $@
 
 depend:
 	$(MAKE) -f $(RULES_MAK) APP=GSM_CODEC app=libgsmcodec $@
-	echo '$(GSM_CODEC_LIB): .libgsmcodec-$(TARGET_NAME).depend' >> .libgsmcodec-$(TARGET_NAME).depend; \
+	echo '$(GSM_CODEC_LIB): .libgsmcodec.depend' >> .libgsmcodec.depend; \
 
 
diff -burN pjproject-0.7.0.orig/third_party/build/ilbc/Makefile pjproject-0.7.0/third_party/build/ilbc/Makefile
--- pjproject-0.7.0.orig/third_party/build/ilbc/Makefile	2007-04-09 19:13:10.000000000 +0200
+++ pjproject-0.7.0/third_party/build/ilbc/Makefile	2007-07-01 02:59:52.000000000 +0200
@@ -5,7 +5,7 @@
 
 RULES_MAK := $(PJDIR)/build/rules.mak
 
-export ILBC_LIB := ../../lib/libilbccodec-$(TARGET_NAME)$(LIBEXT)
+export ILBC_LIB := ../../lib/libilbccodec$(LIBEXT)
 
 ###############################################################################
 # Gather all flags.
diff -burN pjproject-0.7.0.orig/third_party/build/portaudio/Makefile pjproject-0.7.0/third_party/build/portaudio/Makefile
--- pjproject-0.7.0.orig/third_party/build/portaudio/Makefile	2007-04-18 11:24:31.000000000 +0200
+++ pjproject-0.7.0/third_party/build/portaudio/Makefile	2007-07-01 02:59:47.000000000 +0200
@@ -6,7 +6,7 @@
 
 RULES_MAK := $(PJDIR)/build/rules.mak
 
-export PORTAUDIO_LIB:=../../lib/libportaudio-$(TARGET_NAME)$(LIBEXT)
+export PORTAUDIO_LIB:=../../lib/libportaudio$(LIBEXT)
 
 PORTAUDIO_OBJS += pa_allocation.o \
 		  pa_converters.o \
@@ -59,11 +59,11 @@
 	$(MAKE) -f $(RULES_MAK) APP=PORTAUDIO app=libportaudio $@
 
 realclean:
-	$(subst @@,$(subst /,$(HOST_PSEP),.libportaudio-$(TARGET_NAME).depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.libportaudio.depend),$(HOST_RMR))
 	
 	$(MAKE) -f $(RULES_MAK) APP=PORTAUDIO app=libportaudio $@
 
 depend:
 	$(MAKE) -f $(RULES_MAK) APP=PORTAUDIO app=libportaudio $@
-	echo '$(PORTAUDIO_LIB): .libportaudio-$(TARGET_NAME).depend' >> .libportaudio-$(TARGET_NAME).depend; \
+	echo '$(PORTAUDIO_LIB): .libportaudio.depend' >> .libportaudio.depend; \
 
diff -burN pjproject-0.7.0.orig/third_party/build/resample/Makefile pjproject-0.7.0/third_party/build/resample/Makefile
--- pjproject-0.7.0.orig/third_party/build/resample/Makefile	2007-04-09 19:13:10.000000000 +0200
+++ pjproject-0.7.0/third_party/build/resample/Makefile	2007-07-01 02:59:38.000000000 +0200
@@ -5,7 +5,7 @@
 
 RULES_MAK := $(PJDIR)/build/rules.mak
 
-export RESAMPLE_LIB := ../../lib/libresample-$(TARGET_NAME)$(LIBEXT)
+export RESAMPLE_LIB := ../../lib/libresample$(LIBEXT)
 
 ###############################################################################
 # Gather all flags.
@@ -48,12 +48,12 @@
 	$(MAKE) -f $(RULES_MAK) APP=RESAMPLE app=libresample $@
 
 realclean:
-	$(subst @@,$(subst /,$(HOST_PSEP),.ilbc-$(TARGET_NAME).depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.ilbc.depend),$(HOST_RMR))
 	
 	$(MAKE) -f $(RULES_MAK) APP=RESAMPLE app=libresample $@
 
 depend:
 	$(MAKE) -f $(RULES_MAK) APP=RESAMPLE app=libresample $@
-	echo '$(RESAMPLE_LIB): .libresample-$(TARGET_NAME).depend' >> .libresample-$(TARGET_NAME).depend; \
+	echo '$(RESAMPLE_LIB): .libresample.depend' >> .libresample.depend; \
 
 
diff -burN pjproject-0.7.0.orig/third_party/build/speex/Makefile pjproject-0.7.0/third_party/build/speex/Makefile
--- pjproject-0.7.0.orig/third_party/build/speex/Makefile	2007-04-09 19:13:10.000000000 +0200
+++ pjproject-0.7.0/third_party/build/speex/Makefile	2007-07-01 02:59:31.000000000 +0200
@@ -5,7 +5,7 @@
 
 RULES_MAK := $(PJDIR)/build/rules.mak
 
-export SPEEX_LIB := ../../lib/libspeex-$(TARGET_NAME)$(LIBEXT)
+export SPEEX_LIB := ../../lib/libspeex$(LIBEXT)
 
 ###############################################################################
 # Gather all flags.
@@ -63,12 +63,12 @@
 	$(MAKE) -f $(RULES_MAK) APP=SPEEX app=libspeex $@
 
 realclean:
-	$(subst @@,$(subst /,$(HOST_PSEP),.ilbc-$(TARGET_NAME).depend),$(HOST_RMR))
+	$(subst @@,$(subst /,$(HOST_PSEP),.ilbc.depend),$(HOST_RMR))
 	
 	$(MAKE) -f $(RULES_MAK) APP=SPEEX app=libspeex $@
 
 depend:
 	$(MAKE) -f $(RULES_MAK) APP=SPEEX app=libspeex $@
-	echo '$(SPEEX_LIB): .libspeex-$(TARGET_NAME).depend' >> .libspeex-$(TARGET_NAME).depend; \
+	echo '$(SPEEX_LIB): .libspeex.depend' >> .libspeex.depend; \
 
 
 	