diff -ruN Makefile.in.orig Makefile.in
--- Makefile.in.orig	2003-12-15 02:35:03.000000000 +0100
+++ Makefile.in	2007-02-01 08:01:46.000000000 +0100
@@ -47,6 +47,9 @@
 DEFS = @DEFS@ @V_DEFS@
 LIBS = @V_LIBS@
 DYEXT = @DYEXT@
+LDSHARED=$(CC) -shared -Wl,-soname,$(SHAREDLIBM)
+
+
 
 # Standard CFLAGS
 CFLAGS = $(CCOPT) $(INCLS) $(DEFS)
@@ -56,6 +59,11 @@
 INSTALL_DATA = @INSTALL_DATA@
 RANLIB = @RANLIB@
 
+STATICLIB=libpcap.a
+SHAREDLIB=libpcap.so
+SHAREDLIBV=$(SHAREDLIB).0.9.5
+SHAREDLIBM=$(SHAREDLIB).0.8
+
 #
 # Flex and bison allow you to specify the prefixes of the global symbols
 # used by the generated parser.  This allows programs to use lex/yacc
@@ -95,24 +103,29 @@
 TAGFILES = \
 	$(SRC) $(HDR) $(TAGHDR)
 
-CLEANFILES = $(OBJ) libpcap.a $(GENSRC) $(GENHDR) lex.yy.c
+CLEANFILES = $(OBJ) $(STATICLIB) $(GENSRC) $(GENHDR) lex.yy.c
 
-all: libpcap.a
+all: $(STATICLIB) shared
 
-libpcap.a: $(OBJ)
+$(STATICLIB): $(OBJ)
 	@rm -f $@
 	ar rc $@ $(OBJ) $(LIBS)
 	$(RANLIB) $@
 
 shared: libpcap.$(DYEXT)
 
+$(SHAREDLIB): $(SHAREDLIBV)
+
 #
 # XXX - this works with GNU ld, but won't necessarily work with native
 # ld on, for example, various SVR4-flavored platforms, or Digital UNIX.
 #
-libpcap.so: $(OBJ)
-	@rm -f $@
-	ld -shared -o $@.`cat VERSION` $(OBJ)
+
+$(SHAREDLIBV): $(OBJ)
+	$(LDSHARED) -lc -o $@ $(OBJ)
+	rm -f $(SHAREDLIB) $(SHAREDLIBM)
+	ln -s $@ $(SHAREDLIB)
+	ln -s $@ $(SHAREDLIBM)
 
 # the following rule succeeds, but the result is untested.
 libpcap.dylib: $(OBJ)
@@ -170,11 +183,11 @@
 bpf_filter.o: bpf_filter.c
 	$(CC) $(CFLAGS) -c bpf_filter.c
 
-install: libpcap.a 
+install: $(STATICLIB)
 	[ -d $(DESTDIR)$(libdir) ] || \
 	    (mkdir -p $(DESTDIR)$(libdir); chmod 755 $(DESTDIR)$(libdir))
-	$(INSTALL_DATA) libpcap.a $(DESTDIR)$(libdir)/libpcap.a
-	$(RANLIB) $(DESTDIR)$(libdir)/libpcap.a
+	$(INSTALL_DATA) $(STATICLIB) $(DESTDIR)$(libdir)/$(STATICLIB)
+	$(RANLIB) $(DESTDIR)$(libdir)/$(STATICLIB)
 	[ -d $(DESTDIR)$(includedir) ] || \
 	    (mkdir -p $(DESTDIR)$(includedir); chmod 755 $(DESTDIR)$(includedir))
 	$(INSTALL_DATA) $(srcdir)/pcap.h $(DESTDIR)$(includedir)/pcap.h
@@ -188,21 +201,30 @@
 	    $(DESTDIR)$(mandir)/man3/pcap.3
 
 install-shared: install-shared-$(DYEXT)
-install-shared-so: libpcap.so
-	$(INSTALL_PROGRAM) libpcap.so.`cat VERSION` $(DESTDIR)$(libdir)/libpcap.so.`cat VERSION`
+install-shared-so: $(SHAREDLIB)
+	$(INSTALL_PROGRAM) $(SHAREDLIBV) $(DESTDIR)$(libdir)/$(SHAREDLIBV)
+	cd $(DESTDIR)$(libdir); if test -f $(SHAREDLIBV); then \
+	  rm -f $(SHAREDLIB) $(SHAREDLIBM); \
+	  ln -s $(SHAREDLIBV) $(SHAREDLIB); \
+	  ln -s $(SHAREDLIBV) $(SHAREDLIBM); \
+	  (ldconfig || true)  >/dev/null 2>&1; \
+	fi
+
 install-shared-dylib: libpcap.dylib
 	$(INSTALL_PROGRAM) libpcap.`cat VERSION`.dylib $(DESTDIR)$(libdir)/libpcap.`cat VERSION`.dylib
 	VER=`cat VERSION`; cd $(DESTDIR)$(libdir) && ln -sf libpcap.$$VER.dylib libpcap.0.dylib;  ln -sf libpcap.0.dylib libpcap.dylib
 
 uninstall:
-	rm -f $(DESTDIR)$(libdir)/libpcap.a
+	rm -f $(DESTDIR)$(libdir)/$(STATICLIB)
+	rm -f $(DESTDIR)$(libdir)/$(SHAREDLIBM)
+	rm -f $(DESTDIR)$(libdir)/$(SHAREDLIBV)
 	rm -f $(DESTDIR)$(includedir)/pcap.h
 	rm -f $(DESTDIR)$(includedir)/pcap-bpf.h
 	rm -f $(DESTDIR)$(includedir)/pcap-namedb.h
 	rm -f $(DESTDIR)$(mandir)/man3/pcap.3
 
 clean:
-	rm -f $(CLEANFILES) libpcap*.dylib libpcap.so*
+	rm -f $(CLEANFILES) libpcap*.dylib $(SHAREDLIB)*
 
 distclean: clean
 	rm -f Makefile config.cache config.log config.status \
