--- Makefile.in
+++ Makefile.in
@@ -564,8 +564,8 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = include . doc example xstc @PYTHON_SUBDIR@
-DIST_SUBDIRS = include . doc example python xstc
+SUBDIRS = include . xstc @PYTHON_SUBDIR@
+DIST_SUBDIRS = include . python xstc
 INCLUDES = -I$(top_builddir)/include -I@srcdir@/include @THREAD_CFLAGS@ @Z_CFLAGS@ 
 bin_SCRIPTS = xml2-config
 lib_LTLIBRARIES = libxml2.la
@@ -1608,7 +1608,7 @@
 
 info-am:
 
-install-data-am: install-data-local install-m4dataDATA install-man \
+install-data-am: \
 	install-pkgconfigDATA
 
 install-dvi: install-dvi-recursive
