--- Makefile.am.orig	2010-02-19 15:44:23.000000000 +0100
+++ Makefile.am	2010-02-19 15:45:52.000000000 +0100
@@ -2,7 +2,7 @@
 
 AUTOMAKE_OPTIONS = foreign 1.6 dist-zip
 
-SUBDIRS = src include doc
+SUBDIRS = src include
 
 m4datadir = $(datadir)/aclocal
 m4data_DATA = ogg.m4
--- Makefile.in.orig	2009-06-24 17:27:07.000000000 +0200
+++ Makefile.in	2010-02-19 15:59:34.000000000 +0100
@@ -205,7 +205,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = foreign 1.6 dist-zip
-SUBDIRS = src include doc
+SUBDIRS = src include
 m4datadir = $(datadir)/aclocal
 m4data_DATA = ogg.m4
 pkgconfigdir = $(libdir)/pkgconfig
