--- configure.in.orig	2007-09-07 14:07:59.000000000 +0200
+++ configure.in	2008-06-08 15:04:17.000000000 +0200
@@ -70,7 +70,7 @@
   AC_DEFINE(__LIBELF_HEADER_ELF_H, [<sys/elf.h>])
 fi
 
-AC_CHECK_HEADERS(ar.h libelf.h nlist.h gelf.h)
+AC_CHECK_HEADERS(ar.h libelf.h nlist.h )
 AC_MSG_CHECKING([whether to install <libelf.h>, <nlist.h> and <gelf.h>])
 AC_ARG_ENABLE(compat,
   [  --enable-compat         install <libelf.h>, <nlist.h> and <gelf.h> (default: auto)],
