--- configure.orig	2008-02-26 22:34:02.000000000 +0100
+++ configure	2010-10-10 20:12:27.535482913 +0200
@@ -7606,7 +7606,7 @@
 
 
 
-for ac_header in inttypes.h libintl.h limits.h locale.h malloc.h netinet/in.h stddef.h stdlib.h string.h strings.h unistd.h
+for ac_header in inttypes.h limits.h locale.h malloc.h netinet/in.h stddef.h stdlib.h string.h strings.h unistd.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if eval "test \"\${$as_ac_Header+set}\" = set"; then
