--- pcap-linux.c.orig	2008-10-28 02:52:19.000000000 +0100
+++ pcap-linux.c	2010-02-10 20:30:20.000000000 +0100
@@ -1921,8 +1921,8 @@
 	int ret;
 
 	if (handle->opt.buffer_size == 0) {
-		/* by default request 2M for the ring buffer */
-		handle->opt.buffer_size = 2*1024*1024;
+		/* by default request 1M for the ring buffer */
+		handle->opt.buffer_size = 1*1024*1024;
 	}
 	ret = prepare_tpacket_socket(handle);
 	if (ret == 0)
