--- Makefile.in	2010-03-15 16:18:25.000000000 +0100
+++ Makefile.in	2010-03-21 00:10:00.000000000 +0100
@@ -562,8 +562,8 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = include . doc example xstc @PYTHON_SUBDIR@
-DIST_SUBDIRS = include . doc example python xstc
+SUBDIRS = include . xstc @PYTHON_SUBDIR@
+DIST_SUBDIRS = include . python xstc
 INCLUDES = -I$(top_builddir)/include -I@srcdir@/include @THREAD_CFLAGS@ @Z_CFLAGS@ 
 bin_SCRIPTS = xml2-config
 lib_LTLIBRARIES = libxml2.la
@@ -1606,7 +1606,7 @@
 
 info-am:
 
-install-data-am: install-data-local install-m4dataDATA install-man \
+install-data-am: \
 	install-pkgconfigDATA
 
 install-dvi: install-dvi-recursive
