--- configure
+++ configure
@@ -42144,10 +42144,10 @@
     if test "$GCC" = yes ; then
         case "${host}" in
             *-pc-os2_emx | *-pc-os2-emx )
-                                                OPTIMISE_CFLAGS="-O2"
+                                                OPTIMISE_CFLAGS=""
             ;;
             *)
-                                                                OPTIMISE_CFLAGS="-O2 -fno-strict-aliasing"
+                                                                OPTIMISE_CFLAGS="-fno-strict-aliasing"
             ;;
         esac
     else
