--- include/xyssl/config.h.orig	2008-03-16 15:14:56.000000000 +0100
+++ include/xyssl/config.h	2009-07-29 12:24:48.000000000 +0200
@@ -81,7 +81,7 @@
  *      SSL_RSA_RC4_128_MD5
  *      SSL_RSA_RC4_128_SHA
  */
-#define XYSSL_ARC4_C
+/*#define XYSSL_ARC4_C*/
 
 /*
  * Module:  library/base64.c
@@ -139,7 +139,7 @@
  *      SSL_EDH_RSA_DES_168_SHA
  *      SSL_EDH_RSA_AES_256_SHA
  */
-#define XYSSL_DHM_C
+/*#define XYSSL_DHM_C*/
 
 /*
  * Module:  library/havege.c
--- library/Makefile.orig	2009-07-29 12:24:48.000000000 +0200
+++ library/Makefile	2009-07-29 12:24:48.000000000 +0200
@@ -17,12 +17,12 @@
 # OSX shared library extension:
 # DLEXT=dylib
 
-OBJS=	aes.o		arc4.o		base64.o	\
+OBJS=	aes.o				base64.o	\
 	bignum.o	certs.o		debug.o		\
-	des.o		dhm.o		havege.o	\
-	md2.o		md4.o		md5.o		\
+	des.o				havege.o	\
+					md5.o		\
 	net.o		padlock.o	rsa.o		\
-	sha1.o		sha2.o		sha4.o		\
+	sha1.o				sha4.o		\
 	ssl_cli.o	ssl_srv.o	ssl_tls.o	\
 	timing.o	x509parse.o x509write.o
 
