--- include/xyssl/config.h	2007-07-07 07:07:00.000000000 +0200
+++ include/xyssl/config.h	2007-09-01 02:48:30.000000000 +0200
@@ -75,7 +75,7 @@
  *      SSL_RSA_RC4_128_MD5
  *      SSL_RSA_RC4_128_SHA
  */
-#define XYSSL_ARC4_C
+/*#define XYSSL_ARC4_C*/
 
 /*
  * Module:  library/base64.c
@@ -133,7 +133,7 @@
  *      SSL_EDH_RSA_DES_168_SHA
  *      SSL_EDH_RSA_AES_256_SHA
  */
-#define XYSSL_DHM_C
+/*#define XYSSL_DHM_C*/
 
 /*
  * Module:  library/havege.c
@@ -149,7 +149,7 @@
  *
  * This module enables support for MD2-signed X.509 certificates.
  */
-#define XYSSL_MD2_C
+/*#define XYSSL_MD2_C*/
 
 /*
  * Module:  library/md4.c
@@ -157,7 +157,7 @@
  *
  * This module enables support for MD4-signed X.509 certificates.
  */
-#define XYSSL_MD4_C
+/*#define XYSSL_MD4_C*/
 
 /*
  * Module:  library/md5.c
--- library/Makefile	2007-07-07 07:07:00.000000000 +0200
+++ library/Makefile	2007-09-01 02:49:23.000000000 +0200
@@ -17,12 +17,12 @@
 # OSX shared library extension:
 # DLEXT=dylib
 
-OBJS=	aes.o		arc4.o		base64.o	\
-	bignum.o	certs.o		debug.o		\
-	des.o		dhm.o		havege.o	\
-	md2.o		md4.o		md5.o		\
+OBJS=	aes.o				base64.o	\
+	bignum.o	certs.o				\
+	des.o				havege.o	\
+					md5.o		\
 	net.o		padlock.o	rsa.o		\
-	sha1.o		sha2.o		sha4.o		\
+	sha1.o				sha4.o		\
 	ssl_cli.o	ssl_srv.o	ssl_tls.o	\
 	timing.o	x509parse.o
 
