--- Configure.orig	2011-05-10 10:22:48.079133965 +0200
+++ Configure	2011-05-10 10:22:48.319010949 +0200
@@ -1,4 +1,4 @@
-:
+#!/usr/bin/perl
 eval 'exec perl -S $0 ${1+"$@"}'
     if $running_under_some_shell;
 ##
--- tools/c_rehash.in.orig	2010-04-15 01:07:28.000000000 +0200
+++ tools/c_rehash.in	2011-05-10 10:22:48.319010949 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 
 
 # Perl c_rehash script, scan all files in a directory
--- util/clean-depend.pl.orig	2007-09-19 16:53:18.000000000 +0200
+++ util/clean-depend.pl	2011-05-10 10:22:48.319010949 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl
 # Clean the dependency list in a makefile of standard includes...
 # Written by Ben Laurie <ben@algroup.co.uk> 19 Jan 1999
 
--- util/mkdef.pl.orig	2010-07-25 18:56:06.000000000 +0200
+++ util/mkdef.pl	2011-05-10 10:22:48.319010949 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl
 #
 # generate a .def file
 #
--- util/mkerr.pl.orig	2011-02-08 18:10:53.000000000 +0100
+++ util/mkerr.pl	2011-05-10 10:22:48.319010949 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl
 
 my $config = "crypto/err/openssl.ec";
 my $hprefix = "openssl/";
--- util/mkstack.pl.orig	2011-02-03 13:04:40.000000000 +0100
+++ util/mkstack.pl	2011-05-10 10:22:48.319010949 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl
 
 # This is a utility that searches out "DECLARE_STACK_OF()"
 # declarations in .h and .c files, and updates/creates/replaces
--- util/pod2man.pl.orig	2010-01-05 18:33:09.000000000 +0100
+++ util/pod2man.pl	2011-05-10 10:22:48.319010949 +0200
@@ -1,4 +1,4 @@
-: #!/usr/bin/perl-5.005
+#!/usr/bin/perl
     eval 'exec /usr/bin/perl -S $0 ${1+"$@"}'
 	if $running_under_some_shell;
 
--- util/selftest.pl.orig	2009-04-13 13:22:55.000000000 +0200
+++ util/selftest.pl	2011-05-10 10:22:48.319010949 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl
 #
 # Run the test suite and generate a report
 #
