--- third_party/build/milenage/Makefile.orig	2007-12-26 17:44:00.000000000 +0100
+++ third_party/build/milenage/Makefile	2007-12-26 17:44:11.000000000 +0100
@@ -5,7 +5,7 @@
 
 RULES_MAK := $(PJDIR)/build/rules.mak
 
-export MILENAGE_LIB := ../../lib/libmilenage-$(TARGET_NAME)$(LIBEXT)
+export MILENAGE_LIB := ../../lib/libmilenage$(LIBEXT)
 
 ###############################################################################
 # Gather all flags.
