--- Makefile.org.orig	2011-05-10 10:26:05.567854953 +0200
+++ Makefile.org	2011-05-10 10:26:05.967649926 +0200
@@ -242,17 +242,17 @@
 build_libs: build_crypto build_ssl build_engines

 build_crypto:
-	@dir=crypto; target=all; $(BUILD_ONE_CMD)
-build_ssl:
-	@dir=ssl; target=all; $(BUILD_ONE_CMD)
-build_engines:
-	@dir=engines; target=all; $(BUILD_ONE_CMD)
-build_apps:
-	@dir=apps; target=all; $(BUILD_ONE_CMD)
-build_tests:
-	@dir=test; target=all; $(BUILD_ONE_CMD)
-build_tools:
-	@dir=tools; target=all; $(BUILD_ONE_CMD)
+	@+dir=crypto; target=all; $(BUILD_ONE_CMD)
+build_ssl: build_crypto
+	@+dir=ssl; target=all; $(BUILD_ONE_CMD)
+build_engines: build_crypto
+	@+dir=engines; target=all; $(BUILD_ONE_CMD)
+build_apps: build_libs
+	@+dir=apps; target=all; $(BUILD_ONE_CMD)
+build_tests: build_libs
+	@+dir=test; target=all; $(BUILD_ONE_CMD)
+build_tools: build_libs
+	@+dir=tools; target=all; $(BUILD_ONE_CMD)

 all_testapps: build_libs build_testapps
 build_testapps:
@@ -377,7 +377,7 @@
 
 files:
 	$(PERL) $(TOP)/util/files.pl Makefile > $(TOP)/MINFO
-	@set -e; target=files; $(RECURSIVE_BUILD_CMD)
+	@+set -e; target=files; $(RECURSIVE_BUILD_CMD)
 
 links:
 	@$(PERL) $(TOP)/util/mkdir-p.pl include/openssl
@@ -508,7 +508,7 @@
 	(cp $$i $(INSTALL_PREFIX)$(INSTALLTOP)/include/openssl/$$i; \
 	chmod 644 $(INSTALL_PREFIX)$(INSTALLTOP)/include/openssl/$$i ); \
 	done;
-	@set -e; target=install; $(RECURSIVE_BUILD_CMD)
+	@+set -e; target=install; $(RECURSIVE_BUILD_CMD)
 	@set -e; for i in $(LIBS) ;\
 	do \
 		if [ -f "$$i" ]; then \
--- crypto/Makefile.orig	2010-07-27 00:09:59.000000000 +0200
+++ crypto/Makefile	2011-05-10 10:26:05.967649926 +0200
@@ -85,11 +85,11 @@
 	@if [ -z "$(THIS)" ]; then $(MAKE) -f $(TOP)/Makefile reflect THIS=$@; fi
 
 subdirs:
-	@target=all; $(RECURSIVE_MAKE)
+	@+target=all; $(RECURSIVE_MAKE)
 
 files:
 	$(PERL) $(TOP)/util/files.pl Makefile >> $(TOP)/MINFO
-	@target=files; $(RECURSIVE_MAKE)
+	@+target=files; $(RECURSIVE_MAKE)
 
 links:
 	@$(PERL) $(TOP)/util/mklink.pl ../include/openssl $(EXHEADER)
@@ -110,7 +110,7 @@
 	fi
 
 libs:
-	@target=lib; $(RECURSIVE_MAKE)
+	@+target=lib; $(RECURSIVE_MAKE)
 
 install:
 	@[ -n "$(INSTALLTOP)" ] # should be set by top Makefile...
@@ -119,7 +119,7 @@
 	(cp $$i $(INSTALL_PREFIX)$(INSTALLTOP)/include/openssl/$$i; \
 	chmod 644 $(INSTALL_PREFIX)$(INSTALLTOP)/include/openssl/$$i ); \
 	done;
-	@target=install; $(RECURSIVE_MAKE)
+	@+target=install; $(RECURSIVE_MAKE)
 
 lint:
 	@target=lint; $(RECURSIVE_MAKE)
