--- Configure.orig	2011-05-10 10:26:05.028131742 +0200
+++ Configure	2011-05-10 10:26:06.287485902 +0200
@@ -664,7 +664,7 @@
 my $processor="";
 my $default_ranlib;
 my $perl;
-
+my $quiet;

 # All of the following is disabled by default (RC5 was enabled before 0.9.8):

@@ -858,6 +858,10 @@
 				{
 				$cross_compile_prefix=$1;
 				}
+			elsif (/^--quiet$/)
+				{
+				$quiet=1
+				}
 			else
 				{
 				print STDERR $usage;
