--- src/qgen/Makefile.am.orig	2008-01-01 00:14:51.000000000 +0000
+++ src/qgen/Makefile.am	2009-05-09 12:53:50.000000000 +0000
@@ -3,7 +3,7 @@
 
 qgen_SOURCES = common.c common.h file.c file.h first.c ql_y.y ql_l.l qgen.c \
 		qgen.h second.c third.c
-qgen_LDADD = -lfl
+qgen_LDADD = 
 
 q_dump_SOURCES = common.c
 q_dump_LDADD = qd.dump.standalone.o
--- src/qgen/Makefile.in.orig	2009-05-09 12:46:38.000000000 +0000
+++ src/qgen/Makefile.in	2009-05-09 12:53:50.000000000 +0000
@@ -210,7 +210,7 @@
 qgen_SOURCES = common.c common.h file.c file.h first.c ql_y.y ql_l.l qgen.c \
 		qgen.h second.c third.c
 
-qgen_LDADD = -lfl
+qgen_LDADD =
 q_dump_SOURCES = common.c
 q_dump_LDADD = qd.dump.standalone.o
 # FIXME: paulsch: We don't really depend on qd.dump.o or q.out.o here, but this
--- src/qgen/ql_l.l.orig	2001-10-09 22:33:07.000000000 +0000
+++ src/qgen/ql_l.l	2009-05-09 12:53:50.000000000 +0000
@@ -14,6 +14,11 @@
 #include "qgen.h"
 #include "ql_y.h"
 
+int yywrap(void)
+{
+        return 1;
+}
+
 
 typedef struct _tree {
     struct _tree *left,*right;
--- src/sigd/cfg_l.l.orig	2008-01-01 00:14:52.000000000 +0000
+++ src/sigd/cfg_l.l	2009-05-09 12:53:50.000000000 +0000
@@ -16,6 +16,10 @@
 
 #include "cfg_y.h"
 
+int yywrap(void)
+{
+        return 1;
+}
 
 static int lineno = 1;
 static int token; /* f@#%ing flex doesn't grok return after BEGIN */
--- src/sigd/Makefile.am.orig	2008-01-01 00:14:52.000000000 +0000
+++ src/sigd/Makefile.am	2009-05-09 12:53:50.000000000 +0000
@@ -8,7 +8,7 @@
 			$(top_builddir)/src/qgen/qd.dump.o \
 			$(top_builddir)/src/lib/libatm.la \
 			$(top_builddir)/src/saal/libsaal.a
-atmsigd_LDADD = $(atmsigd_XTRAS) -lfl
+atmsigd_LDADD = $(atmsigd_XTRAS) 
 atmsigd_DEPENDENCIES = mess.c $(atmsigd_XTRAS)
 
 CLEANFILES = mess.c
--- src/sigd/Makefile.in.orig	2009-05-09 12:46:39.000000000 +0000
+++ src/sigd/Makefile.in	2009-05-09 12:53:50.000000000 +0000
@@ -225,7 +225,7 @@
 			$(top_builddir)/src/lib/libatm.la \
 			$(top_builddir)/src/saal/libsaal.a
 
-atmsigd_LDADD = $(atmsigd_XTRAS) -lfl
+atmsigd_LDADD = $(atmsigd_XTRAS) 
 atmsigd_DEPENDENCIES = mess.c $(atmsigd_XTRAS)
 CLEANFILES = mess.c
 sysconf_DATA = atmsigd.conf
--- src/switch/debug/debug.c.orig	2008-01-01 00:14:52.000000000 +0000
+++ src/switch/debug/debug.c	2009-05-09 12:53:50.000000000 +0000
@@ -20,6 +20,11 @@
 
 #define PRV(call) ((FAB *) (call)->fab)
 
+int yywrap(void)
+{
+        return 1;
+}
+
 
 typedef struct _fab {
     CALL *next; /* relay.c may not keep track of calls, but WE are */
--- src/switch/debug/Makefile.am.orig	2008-01-01 00:14:52.000000000 +0000
+++ src/switch/debug/Makefile.am	2009-05-09 12:53:50.000000000 +0000
@@ -5,7 +5,7 @@
 sw_debug_SOURCES = debug.c
 sw_debug_XTRAS = $(top_builddir)/src/switch/libsw.a \
 			$(top_builddir)/src/lib/libatm.la
-sw_debug_LDADD = $(sw_debug_XTRAS) -lfl
+sw_debug_LDADD = $(sw_debug_XTRAS) 
 			
 sw_debug_DEPENDENCIES = $(sw_debug_XTRAS)
 
--- src/switch/debug/Makefile.in.orig	2009-05-09 12:46:39.000000000 +0000
+++ src/switch/debug/Makefile.in	2009-05-09 12:53:50.000000000 +0000
@@ -193,7 +193,7 @@
 sw_debug_XTRAS = $(top_builddir)/src/switch/libsw.a \
 			$(top_builddir)/src/lib/libatm.la
 
-sw_debug_LDADD = $(sw_debug_XTRAS) -lfl
+sw_debug_LDADD = $(sw_debug_XTRAS)
 sw_debug_DEPENDENCIES = $(sw_debug_XTRAS)
 EXTRA_DIST = demo README
 all: all-am
--- src/switch/tcp/Makefile.am.orig	2008-01-01 00:14:52.000000000 +0000
+++ src/switch/tcp/Makefile.am	2009-05-09 12:53:50.000000000 +0000
@@ -5,7 +5,7 @@
 sw_tcp_SOURCES = tcpsw.c
 sw_tcp_XTRAS = $(top_builddir)/src/switch/libsw.a \
 		$(top_builddir)/src/lib/libatm.la
-sw_tcp_LDADD = $(sw_tcp_XTRAS) -lfl
+sw_tcp_LDADD = $(sw_tcp_XTRAS) 
 sw_tcp_DEPENDENCIES = $(sw_tcp_XTRAS)
 
 EXTRA_DIST = mkfiles README
--- src/switch/tcp/Makefile.in.orig	2009-05-09 12:46:39.000000000 +0000
+++ src/switch/tcp/Makefile.in	2009-05-09 12:53:50.000000000 +0000
@@ -193,7 +193,7 @@
 sw_tcp_XTRAS = $(top_builddir)/src/switch/libsw.a \
 		$(top_builddir)/src/lib/libatm.la
 
-sw_tcp_LDADD = $(sw_tcp_XTRAS) -lfl
+sw_tcp_LDADD = $(sw_tcp_XTRAS) 
 sw_tcp_DEPENDENCIES = $(sw_tcp_XTRAS)
 EXTRA_DIST = mkfiles README
 all: all-am
--- src/switch/tcp/tcpsw.c.orig	2008-01-01 00:14:52.000000000 +0000
+++ src/switch/tcp/tcpsw.c	2009-05-09 12:53:50.000000000 +0000
@@ -35,6 +35,10 @@
 #define MAX_PACKET (ATM_MAX_AAL5_PDU+sizeof(struct atmtcp_hdr))
 #define BUFFER_SIZE (MAX_PACKET*2)
 
+int yywrap(void)
+{
+        return 1;
+}
 
 typedef struct _table {
     struct _link *out;	/* output port */
--- src/test/ispl_l.l.orig	2001-10-09 22:33:08.000000000 +0000
+++ src/test/ispl_l.l	2009-05-09 12:53:50.000000000 +0000
@@ -17,6 +17,11 @@
 #include "isp.h"
 #include "ispl_y.h"
 
+int yywrap(void)
+{
+	return 1;
+}
+
 
 static int lineno = 1;
 
--- src/test/Makefile.am.orig	2008-01-01 00:14:53.000000000 +0000
+++ src/test/Makefile.am	2009-05-09 12:53:50.000000000 +0000
@@ -20,7 +20,7 @@
 bw_SOURCES = bw.c
 isp_SOURCES = isp.c isp.h ispl_y.y ispl_l.l
 isp_XTRAS = $(LDADD)
-isp_LDADD = $(isp_XTRAS) -lfl
+isp_LDADD = $(isp_XTRAS) 
 isp_DEPENDENCIES = $(isp_XTRAS) errnos.inc
 window_SOURCES = window.c
 
--- src/test/Makefile.in.orig	2009-05-09 12:46:39.000000000 +0000
+++ src/test/Makefile.in	2009-05-09 12:53:50.000000000 +0000
@@ -277,7 +277,7 @@
 bw_SOURCES = bw.c
 isp_SOURCES = isp.c isp.h ispl_y.y ispl_l.l
 isp_XTRAS = $(LDADD)
-isp_LDADD = $(isp_XTRAS) -lfl
+isp_LDADD = $(isp_XTRAS) 
 isp_DEPENDENCIES = $(isp_XTRAS) errnos.inc
 window_SOURCES = window.c
 CLEANFILES = errnos.inc
