--- pcre-config.in	2009-09-01 18:21:54.000000000 +0200
+++ pcre-config.in	2009-11-17 21:31:08.000000000 +0100
@@ -21,6 +21,10 @@
   libR=" -Wl,-R@libdir@"
   ;;
 esac
+libS=
+if test @libdir@ != /usr/lib ; then
+  libS=-L@libdir@
+fi
 
 while test $# -gt 0; do
   case "$1" in
@@ -55,10 +59,10 @@
       echo $includes @PCRE_STATIC_CFLAG@
       ;;
     --libs-posix)
-      echo -L@libdir@$libR -lpcreposix -lpcre
+      echo $libS$libR -lpcreposix -lpcre
       ;;
     --libs)
-      echo -L@libdir@$libR -lpcre
+      echo $libS$libR -lpcre
       ;;
     *)
       echo "${usage}" 1>&2
