--- src/mount_davfs.c
+++ src/mount_davfs.c
@@ -98,6 +98,7 @@
 
 /* The mointpoint as taken from the command line. */
 static char *mpoint;
+static char mpoint_canonicalized_buf[PATH_MAX+1];
 
 /* The type of the kernel file system used. */
 static char *kernel_fs;
@@ -758,7 +759,7 @@
     struct passwd *pw;
 
     if (*mpoint != '/') {
-        char *mp = canonicalize_file_name(mpoint);
+        char *mp = realpath(mpoint, mpoint_canonicalized_buf);
         if (!mp)
             error(EXIT_FAILURE, 0,
                   _("can't evaluate path of mount point %s"), mpoint);
--- src/umount_davfs.c
+++ src/umount_davfs.c
@@ -26,6 +26,9 @@
 #ifdef HAVE_LIBINTL_H
 #include <libintl.h>
 #endif
+#ifdef HAVE_LIMITS_H
+#include <limits.h>
+#endif
 #ifdef HAVE_LOCALE_H
 #include <locale.h>
 #endif
@@ -120,7 +123,8 @@
     if (optind < (argc - 1))
         error(EXIT_FAILURE, 0, _("too many arguments"));
 
-    char *mpoint = canonicalize_file_name(argv[optind]);
+    char mpoint_canonicalized_buf[PATH_MAX+1];
+    char *mpoint = realpath(argv[optind], mpoint_canonicalized_buf);
     if (!mpoint)
         mpoint = argv[optind];
     if (!mpoint || *mpoint != '/')
