--- configure.orig	2009-09-23 12:42:26.000000000 +0200
+++ configure	2009-09-23 12:42:36.000000000 +0200
@@ -5546,7 +5546,7 @@
 echo "${ECHO_T}$ne_cv_lib_neon" >&6; }
     if test "$ne_cv_lib_neon" = "yes"; then
        ne_cv_lib_neonver=no
-       for v in 25 26 27 28; do
+       for v in 25 26 27 28 29; do
           case $ne_libver in
           0.$v.*) ne_cv_lib_neonver=yes ;;
           esac
