--- configure.orig	2010-04-30 21:35:56.000000000 +0200
+++ configure	2010-09-20 01:45:44.000000000 +0200
@@ -6933,7 +6933,7 @@
 
 
 
-for ac_header in argz.h fcntl.h libintl.h langinfo.h limits.h locale.h mntent.h stddef.h stdint.h stdlib.h string.h sys/mount.h sys/time.h syslog.h termios.h unistd.h utime.h
+for ac_header in argz.h fcntl.h langinfo.h limits.h locale.h mntent.h stddef.h stdint.h stdlib.h string.h sys/mount.h sys/time.h syslog.h termios.h unistd.h utime.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
