--- configure.in
+++ configure.in
@@ -20,7 +20,7 @@
 	[asterisk],
 	AS_HELP_STRING([--with-asterisk=path], [set asterisk headers location]),
 	[ if test "x$with_asterisk" = "xyes" -o "x$with_asterisk" = "xno" ; then  AC_MSG_ERROR([Invalid --with-asterisk=path value]); fi ],
-	[ with_asterisk="../include /usr/include /usr/local/include /opt/local/include" ]
+	[ with_asterisk="../include" ]
 )
 
 AC_ARG_ENABLE(
@@ -88,7 +88,7 @@
 )
 
 AC_HEADER_FIND([asterisk.h], $with_asterisk)
-AC_HEADER_FIND([iconv.h], /usr/include /usr/local/include /opt/local/include)
+AC_CHECK_HEADER([iconv.h])
 
 AC_DEFINE([ICONV_CONST],[], [Define to const if you has iconv() const declaration of input buffer])
 AC_MSG_CHECKING([for iconv use const inbuf])
