--- Rules.mak
+++ Rules.mak
@@ -217,12 +217,16 @@
 ifeq ($(TARGET_ARCH),mips)
 	CPU_LDFLAGS-$(ARCH_LITTLE_ENDIAN)+=-EL
 	CPU_LDFLAGS-$(ARCH_BIG_ENDIAN)+=-EB
+ifneq ($(strip $(MIPS_CUSTOM_ARCH_CPU_CFLAGS)),)
+	CPU_CFLAGS-y+=$(strip $(MIPS_CUSTOM_ARCH_CPU_CFLAGS))
+else
 	CPU_CFLAGS-$(CONFIG_MIPS_ISA_1)+=-mips1
 	CPU_CFLAGS-$(CONFIG_MIPS_ISA_2)+=-mips2 -mtune=mips2
 	CPU_CFLAGS-$(CONFIG_MIPS_ISA_3)+=-mips3 -mtune=mips3
 	CPU_CFLAGS-$(CONFIG_MIPS_ISA_4)+=-mips4 -mtune=mips4
 	CPU_CFLAGS-$(CONFIG_MIPS_ISA_MIPS32)+=-mips32 -mtune=mips32
 	CPU_CFLAGS-$(CONFIG_MIPS_ISA_MIPS64)+=-mips64 -mtune=mips32
+endif
 	ifeq ($(strip $(ARCH_BIG_ENDIAN)),y)
 		CPU_LDFLAGS-$(CONFIG_MIPS_N64_ABI)+=-melf64btsmip
 		CPU_LDFLAGS-$(CONFIG_MIPS_O32_ABI)+=-melf32btsmip
