--- libc.orig/stdlib/Makefile.in	2006-11-29 22:04:35.000000000 +0100
+++ libc/stdlib/Makefile.in	2007-09-29 16:22:06.000000000 +0200
@@ -68,7 +68,7 @@
 # wcstod wcstof wcstold
 
 # multi source _atexit.c
-CSRC += __cxa_atexit.c __cxa_finalize.c __exit_handler.c exit.c on_exit.c
+CSRC += __cxa_atexit.c __cxa_finalize.c __exit_handler.c exit.c on_exit.c atexit.c
 ifeq ($(COMPAT_ATEXIT),y)
 CSRC += old_atexit.c
 endif
