--- libc/misc/wchar/wchar.c.old.ak	2007-07-03 10:48:57.000000000 +0200
+++ libc/misc/wchar/wchar.c	2007-07-03 10:54:29.000000000 +0200
@@ -1240,7 +1240,7 @@
 	"\x08\xec""UCS-4\x00"		/* always BE */
 	"\x0a\xec""UCS-4BE\x00"
 	"\x0a\xed""UCS-4LE\x00"
-	"\x09\fe4""UTF-32\x00"		/* platform endian with BOM */
+	"\x09\xe4""UTF-32\x00"		/* platform endian with BOM */
 	"\x0b\xe4""UTF-32BE\x00"
 	"\x0b\xe5""UTF-32LE\x00"
 	"\x08\xe2""UCS-2\x00"		/* always BE */
