--- Makefile.in
+++ Makefile.in
@@ -26,8 +26,8 @@
 include $(top_srcdir)libcrypt/Makefile.in
 include $(top_srcdir)libintl/Makefile.in
 include $(top_srcdir)libm/Makefile.in
-include $(top_srcdir)libnsl/Makefile.in
-include $(top_srcdir)libresolv/Makefile.in
+#include $(top_srcdir)libnsl/Makefile.in
+#include $(top_srcdir)libresolv/Makefile.in
 include $(top_srcdir)libutil/Makefile.in
 include $(top_srcdir)libpthread/Makefile.in
 include $(top_srcdir)librt/Makefile.in
