--- configure_	2018-03-27 00:13:41.000000000 +0200
+++ configure	2019-03-12 00:12:40.185509863 +0100
@@ -4948,7 +4948,7 @@
 
 fi
 
-for ac_header in arpa/inet.h fcntl.h inttypes.h libintl.h limits.h malloc.h \
+for ac_header in arpa/inet.h fcntl.h inttypes.h           limits.h malloc.h \
                   netdb.h netinet/in.h stddef.h stdint.h stdlib.h string.h \
                   sys/socket.h sys/time.h termios.h unistd.h
 do :
