--- linux-3.10/arch/mips/kernel/Makefile
+++ linux-3.10/arch/mips/kernel/Makefile
@@ -2,7 +2,9 @@
 # Makefile for the Linux/MIPS kernel.
 #
 
-extra-y		:= head.o vmlinux.lds
+extra-y		:= head.o avm_kernel_config_area.o vmlinux.lds
+
+$(obj)/vmlinux.lds: $(obj)/avm_kernel_config_area.o
 
 obj-y		+= cpu-probe.o branch.o entry.o genex.o idle.o irq.o process.o \
 		   prom.o ptrace.o reset.o setup.o signal.o syscall.o \
