diff -urN debootstrap-0.3.3.2.orig/target/usr/lib/debootstrap/functions debootstrap-0.3.3.2/target/usr/lib/debootstrap/functions
--- debootstrap-0.3.3.2.orig/target/usr/lib/debootstrap/functions	2007-02-17 14:34:58.000000000 +0100
+++ debootstrap-0.3.3.2/target/usr/lib/debootstrap/functions	2007-03-02 22:09:50.915296372 +0100
@@ -976,7 +976,7 @@
   N_EXIT_THINGS=0
 }
 
-trap "exit_function" 0
+#trap "exit_function" 0
 trap "exit 129" 1
 trap "error 130 INTERRUPTED \"Interrupt caught ... exiting\"" 2
 trap "exit 131" 3
