--- module/config/default.conf
+++ module/config/default.conf
@@ -25,7 +25,7 @@
 	int log_size : "Size in kb" = 100
 	bool log_rotate : "Log Rotate" = True
 general - "General":
-	en;de;fr;it;es;nl;sv;ru;pl;cs;sr;pt_BR language : "Language" = en
+	en language : "Language" = en
 	folder download_folder : "Download Folder" = Downloads
 	bool debug_mode : "Debug Mode" = False
 	bool checksum : "Use Checksum" = False
