--- module/setup.py
+++ module/setup.py
@@ -128,11 +128,6 @@
                 print _("Only needed for some hosters and as freeuser.")
                 print ""
 
-            if not gui:
-                print _("** Gui not available **")
-                print _("The Graphical User Interface.")
-                print ""
-
             if not js:
                 print _("** no JavaScript engine found **")
                 print _("You will need this for some Click'N'Load links. Install Spidermonkey, ossp-js, pyv8 or rhino")
@@ -216,10 +211,7 @@
 
         captcha = pil and tesser
 
-        print ""
-
-        gui = self.check_module("PyQt4")
-        self.print_dep("PyQt4", gui)
+        gui = False
 
         print ""
         jinja = True
