--- src/Makefile.in.orig	2007-08-28 15:47:01.000000000 +0200
+++ src/Makefile.in	2008-04-27 21:25:52.000000000 +0200
@@ -834,7 +834,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 SUBDIRS = $(am__append_1) $(am__append_2)
-@COMPILER_IS_GCC_TRUE@AM_CFLAGS = -Wall -Werror
+# AM_CFLAGS = -Wall -Werror
 collectd_SOURCES = collectd.c collectd.h \
 		   utils_avltree.c utils_avltree.h \
 		   utils_mount.c utils_mount.h \
