--- linux-2.6.13/drivers/net/avm_cpmac/cpmac_eth.c
+++ linux-2.6.13/drivers/net/avm_cpmac/cpmac_eth.c
@@ -384,8 +384,11 @@
     case SIOCGMIIPHY:
     case SIOCGMIIREG:
     case SIOCSMIIREG:
+    case AVM_CPMAC_IOCTL_INFO:
+    case AVM_CPMAC_IOCTL_GENERIC:
     case AVM_CPMAC_IOCTL_SET_CONFIG_N:
     case AVM_CPMAC_IOCTL_GET_CONFIG_N:
+
         if(real_dev->do_ioctl && netif_device_present(real_dev)) 
             err = real_dev->do_ioctl(real_dev, &ifrr, cmd);
         break;
