--- client/client.c
+++ client/client.c
@@ -28,6 +28,7 @@
 #include <sys/socket.h>
 #include <sys/un.h>
 #include <sys/types.h>
+#include <sys/select.h>
 
 #include "nest/bird.h"
 #include "lib/resource.h"
 #include "lib/resource.h"
--- filter/tree.c
+++ filter/tree.c
@@ -6,8 +6,9 @@
  *	Can be freely distributed and used under the terms of the GNU GPL.
  */
 
-#include "lib/alloca.h"
+#include <stdlib.h> /* malloc, free */
 #include "nest/bird.h"
+#include "lib/alloca.h" /* alloca - requires HAVE_ALLOCA_H defined in one of the headers included from nest/bird.h */
 #include "conf/conf.h"
 #include "filter/filter.h"
 
--- lib/string.h
+++ lib/string.h
@@ -11,6 +11,7 @@
 
 #include <stdarg.h>
 #include <string.h>
+#include <strings.h> /* bzero */
 
 int bsprintf(char *str, const char *fmt, ...);
 int bvsprintf(char *str, const char *fmt, va_list args);
