diff -urN bird-1.0.11~/tools/Makefile.in bird-1.0.11/tools/Makefile.in
--- bird-1.0.11~/tools/Makefile.in	2004-05-31 19:53:02.000000000 +0200
+++ bird-1.0.11/tools/Makefile.in	2006-07-10 02:43:05.000000000 +0200
@@ -43,20 +43,20 @@
 	cd $(srcdir) ; etags -lc `find $(static-dirs) $(addprefix $(objdir)/,$(dynamic-dirs)) $(client-dirs) -name *.[chY]`
 
 install: all
-	$(INSTALL) -d $(sbindir) $(sysconfdir) $(localstatedir)
-	$(INSTALL_PROGRAM) -s $(exedir)/bird $(sbindir)/
+	$(INSTALL) -d $(DESTDIR)/$(sbindir) $(DESTDIR)/$(sysconfdir) $(DESTDIR)/$(localstatedir)
+	$(INSTALL_PROGRAM) -s $(exedir)/bird $(DESTDIR)/$(sbindir)/
 	if test -n "@CLIENT@" ; then								\
-		$(INSTALL_PROGRAM) -s $(exedir)/birdc $(sbindir)/ ;				\
+		$(INSTALL_PROGRAM) -s $(exedir)/birdc $(DESTDIR)/$(sbindir)/ ;				\
 	fi
-	if ! test -f $(sysconfdir)/bird.conf ; then						\
-		$(INSTALL_DATA) $(srcdir)/doc/bird.conf.example $(sysconfdir)/bird.conf ;	\
+	if ! test -f $(DESTDIR)/$(sysconfdir)/bird.conf ; then						\
+		$(INSTALL_DATA) $(srcdir)/doc/bird.conf.example $(DESTDIR)/$(sysconfdir)/bird.conf ;	\
 	else											\
 		echo "Not overwriting old bird.conf" ;						\
 	fi
 
 install-docs:
-	$(INSTALL) -d $(docdir)
-	$(INSTALL_DATA) $(srcdir)/doc/{bird,prog}{,-*}.html $(docdir)/
+	$(INSTALL) -d $(DESTDIR)/$(docdir)
+	$(INSTALL_DATA) $(srcdir)/doc/{bird,prog}{,-*}.html $(DESTDIR)/$(docdir)/
 
 clean:
 	find . -name "*.[oa]" -o -name core -o -name depend -o -name "*.html" | xargs rm -f
