--- client/client.c
+++ client/client.c
@@ -256,7 +256,7 @@
 
   bzero(&sa, sizeof(sa));
   sa.sun_family = AF_UNIX;
-  strcpy(sa.sun_path, server_path);
+  strncpy(sa.sun_path, server_path, sizeof(sa.sun_path));
   if (connect(server_fd, (struct sockaddr *) &sa, SUN_LEN(&sa)) < 0)
     DIE("Unable to connect to server control socket (%s)", server_path);
   if (fcntl(server_fd, F_SETFL, O_NONBLOCK) < 0)
--- sysdep/unix/io.c
+++ sysdep/unix/io.c
@@ -1515,7 +1515,7 @@
 
   /* Path length checked in test_old_bird() */
   sa.sun_family = AF_UNIX;
-  strcpy(sa.sun_path, name);
+  strncpy(sa.sun_path, name, sizeof(sa.sun_path));
 
   if (bind(fd, (struct sockaddr *) &sa, SUN_LEN(&sa)) < 0)
     return -1;
