--- client/client.c.orig	2009-08-16 22:42:37.000000000 +0200
+++ client/client.c	2009-08-24 12:02:29.000000000 +0200
@@ -322,7 +322,7 @@
 
   bzero(&sa, sizeof(sa));
   sa.sun_family = AF_UNIX;
-  strcpy(sa.sun_path, server_path);
+  strncpy(sa.sun_path, server_path, sizeof(sa.sun_path));
   if (connect(server_fd, (struct sockaddr *) &sa, SUN_LEN(&sa)) < 0)
     die("Unable to connect to server control socket (%s): %m", server_path);
   if (fcntl(server_fd, F_SETFL, O_NONBLOCK) < 0)
--- sysdep/unix/io.c.orig	2009-06-28 09:33:51.000000000 +0200
+++ sysdep/unix/io.c	2009-08-24 12:02:29.000000000 +0200
@@ -1016,7 +1016,7 @@
     die("sk_open_unix: path too long");
 
   sa.sun_family = AF_UNIX;
-  strcpy(sa.sun_path, name);
+  strncpy(sa.sun_path, name, sizeof(sa.sun_path));
   if (bind(fd, (struct sockaddr *) &sa, SUN_LEN(&sa)) < 0)
     ERR("bind");
   if (listen(fd, 8))
