diff -urN bird-1.0.11/conf/Makefile bird-1.0.11/conf/Makefile
--- conf/Makefile	2000-03-12 18:04:04.000000000 -0500
+++ conf/Makefile	2006-07-13 10:30:06.000000000 -0400
@@ -24,6 +24,6 @@
 	$(M4) -P $(conf-src)/gen_commands.m4 $(srcdir)/client/cmds.m4 $(conf-fragments) | sort >commands.h
 
 cf-lex.c: cf-lex.l
-	$(FLEX) $(FLEX_DEBUG) -sB8 -ocf-lex.c -Pcf_ cf-lex.l
+	$(FLEX) $(FLEX_DEBUG) -s -B -8 -ocf-lex.c -Pcf_ cf-lex.l
 
 depend: keywords.h commands.h cf-parse.tab.c cf-lex.c
