--- tools/Rules.in
+++ tools/Rules.in
@@ -19,7 +19,7 @@
 all-dirs:=$(static-dirs) $(dynamic-dirs) $(client-dirs) $(doc-dirs)
 clean-dirs:=$(all-dirs) proto sysdep
 
-CPPFLAGS=-I$(root-rel) -I$(srcdir) @CPPFLAGS@
+CPPFLAGS=-I$(root-rel) -I$(srcdir) @CPPFLAGS@ -D_XOPEN_SOURCE=600
 CFLAGS=$(CPPFLAGS) @CFLAGS@
 LDFLAGS=@LDFLAGS@
 LIBS=@LIBS@
@@ -68,7 +68,7 @@
 all.o: $(objs)
 #       $(LD) -r -o $@ $^
 #	Changed to $(CC) because $(LD) has problems with crosscompiling
-	$(CC) -nostdlib -r -o $@ $^
+	$(CC) $(CFLAGS) -nostdlib -r -o $@ $^
 
 endif
 
