--- sysdep/unix/io.c.orig
+++ sysdep/unix/io.c
@@ -679,8 +679,16 @@
  * don't have IPV6_RECVPKTINFO we suppose the OS implements the older
  * RFC and we use IPV6_PKTINFO.
  */
-#ifndef IPV6_RECVPKTINFO
-#define IPV6_RECVPKTINFO IPV6_PKTINFO
+#ifdef IPV6_RECVPKTINFO
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,14)
+#if defined IPV6_RECVPKTINFO && defined IPV6_2292PKTINFO
+#undef IPV6_RECVPKTINFO
+#undef IPV6_PKTINFO
+#define IPV6_RECVPKTINFO IPV6_2292PKTINFO
+#define IPV6_PKTINFO IPV6_2292PKTINFO
+#endif
+#endif
 #endif
 
 static char *
