--- src/openvpn/tun.c
+++ src/openvpn/tun.c
@@ -2074,6 +2074,11 @@
 # define TUNSETGROUP   _IOW('T', 206, int)
 #endif
 
+/* TUNSETGROUP appeared in 2.6.23 */
+#ifndef TUNSETGROUP
+#define TUNSETGROUP _IOW('T', 206, int)
+#endif
+
 void
 tuncfg(const char *dev, const char *dev_type, const char *dev_node, int persist_mode, const char *username, const char *groupname, const struct tuntap_options *options)
 {
