--- src/smsd_cfg.h
+++ src/smsd_cfg.h
@@ -21,7 +21,7 @@
 #include <signal.h>
 
 #ifndef __FreeBSD__
-#define DEFAULT_CONFIGFILE "/etc/smsd.conf"
+#define DEFAULT_CONFIGFILE "/tmp/flash/smstools3/smsd.conf"
 #else
 #define DEFAULT_CONFIGFILE "%%PREFIX%%/etc/smsd.conf"
 #endif
@@ -36,7 +36,7 @@
 
 #define MM_CORE_FNAME "/tmp/mm_smsd_%i" /* %i is PID */
 
-#define NUMS 64
+#define NUMBER_OF_MODEMS 64
 #define SIZE_NUM 16
 
 // 3.1.18: Definitions for process_id's:
@@ -141,7 +141,7 @@
 typedef struct
 {
   char name[32]; 		// Name of the queue
-  char numbers[NUMS][SIZE_NUM];	// Phone numbers assigned to this queue
+  char numbers[NUMBER_OF_MODEMS][SIZE_NUM];	// Phone numbers assigned to this queue
   char directory[PATH_MAX];		// Queue directory
 } _queue;
 
