--- configure
+++ configure
@@ -3650,7 +3650,7 @@
 -L$PJSIP_PATH/pjmedia/lib \
 -L$PJSIP_PATH/pjsip/lib \
 -L$PJSIP_PATH/third_party/lib \
--L${prefix}/usr/local/lib -L${prefix}/usr/lib -L${prefix}/lib"
+"
 
 CPPFLAGS="$CPPFLAGS \
 -I$PJSIP_PATH/pjlib/include \
@@ -3660,7 +3660,7 @@
 -I$PJSIP_PATH/pjsip/include \
 -I$PJSIP_PATH/third_party/speex/include \
 -I$PJSIP_PATH/third_party/build/speex \
--I${prefix}/usr/local/include -I${prefix}/usr/include -I${prefix}/include"
+"
 
 ###########################################################################################
 # == Checks for libcapi20 ==
