diff -burN ../dtmfbox-0.4.1_rc4-src_orig/capi_funcs.c ./capi_funcs.c
--- ../dtmfbox-0.4.1_rc4-src_orig/capi_funcs.c	2008-06-03 18:59:19.000000000 +0200
+++ ./capi_funcs.c	2008-06-03 18:59:42.000000000 +0200
@@ -293,6 +293,7 @@
 
 void capi_send_connect_resp(connection_t* mCON, int reject)
 {
+    #if DTMFBOX_USE_CAPI
     _cmsg CMSG1;
     connection_t *CUR_CON = (connection_t*)mCON;
     
@@ -317,6 +318,7 @@
                  NULL,        // Useruserdata
                  NULL);       // Facilitydataarray
 
+  #endif
 }
 
 #if DTMFBOX_USE_CAPI
