--- Makefile.mipsel.orig	2008-02-09 21:41:46.000000000 +0100
+++ Makefile.mipsel	2008-02-09 21:43:05.000000000 +0100
@@ -1,23 +1,19 @@
 CROSS=mipsel-linux-
-PJPATH=./pjproject
 
-CC=$(CROSS)gcc
-STRIP=$(CROSS)strip
+CC:=$(CROSS)gcc
+STRIP:=$(CROSS)strip
 
 CAPI_LIB=-lcapi20
 CAPI_CFLAG=-DDTMFBOX_USE_CAPI=1
 SPEEX_CFLAG=-I$(PJPATH)/third_party/speex/include \
 	    -I$(PJPATH)/third_party/build/speex
 
-include $(PJPATH)/build.mak
-
-CC = $(APP_CC)
-LDFLAGS = $(APP_LDFLAGS)
-LDLIBS = -Wl,-Bstatic $(APP_LDLIBS) \
+LDFLAGS := $(LDFLAGS)
+LDLIBS := -Wl,-Bstatic $(LDLIBS) \
 	-static-libgcc -Wl,-Bdynamic -lpthread $(CAPI_LIB)
 
-CFLAGS = $(APP_CFLAGS)
-CPPFLAGS = -w $(CAPI_CFLAG) $(SPEEX_CFLAG) ${CFLAGS}
+CFLAGS := ${CFLAGS}
+CPPFLAGS := -w $(CAPI_CFLAG) $(SPEEX_CFLAG) $(CFLAGS)
 DESTDIR=
 
 PCH=global_headers.h
