--- Makefile.mipsel.orig	2007-12-26 15:03:30.000000000 +0100
+++ Makefile.mipsel	2007-12-26 15:03:12.000000000 +0100
@@ -1,21 +1,17 @@
-CROSS=mipsel-linux-
-PJPATH=./pjproject
+CROSS:=mipsel-linux-
 
-CC=$(CROSS)gcc
-STRIP=$(CROSS)strip
+CC:=$(CROSS)gcc
+STRIP:=$(CROSS)strip
 
 CAPI_LIB=-lcapi20
 CAPI_CFLAG=-DDTMFBOX_USE_CAPI=1
 
-include $(PJPATH)/build.mak
+LDFLAGS:=$(LDFLAGS)
+LDLIBS:=-Wl,-Bstatic $(LDLIBS) \
+	-Wl,-Bdynamic -lpthread $(CAPI_LIB)
 
-CC = $(APP_CC)
-LDFLAGS = $(APP_LDFLAGS)
-LDLIBS = -Wl,-Bstatic $(APP_LDLIBS) \
-	-static-libgcc -Wl,-Bdynamic -lpthread $(CAPI_LIB)
-
-CFLAGS = $(APP_CFLAGS)
-CPPFLAGS = -w $(CAPI_CFLAG) ${CFLAGS}
+CFLAGS:=$(CFLAGS)
+CPPFLAGS:=-w $(CAPI_CFLAG) $(CFLAGS)
 DESTDIR=
 
 PCH=global_headers.h
