--- linux-2.6.32/drivers/net/avm_cpmac/common/Makefile
+++ linux-2.6.32/drivers/net/avm_cpmac/common/Makefile
@@ -3,7 +3,7 @@
 # Copyright (C) 2011 AVM GmbH <fritzbox_info@avm.de>
 #
 
-EXTRA_CFLAGS += -DAVMNET_VERSION="\"`svn info $(src)/.. | grep URL | sed -e "s/^.*\/\([0-9.]\+\)-[^\/]*/\1/g"`  -  Revision `svnversion -cn $(src)/..`  -  `date +\"%a  %e.%m.%Y  %H:%M:%S\"` \""
+EXTRA_CFLAGS += -DAVMNET_VERSION="\"6.106 - Revision 1496:2462 - Do 20.02.2014 11:04:09\""
 
 obj-y	+= avmnet_config.o avmnet_common.o
 
