--- linux-2.6.13/drivers/char/avm_new/Makefile.26
+++ linux-2.6.13/drivers/char/avm_new/Makefile.26
@@ -92,3 +92,5 @@
 obj-$(CONFIG_AVM_SAMMEL) += avm.o 
 
 # EXTRA_CFLAGS	+= -g -ggdb -save-temps
+EXTRA_CFLAGS    += -DAVM_NEW_HWREV_LIST="\" 118 117 110 114 129 155 156 115 108 94 127 137 122 139 145 150 152 154 141 135 146 153 143 101 144 136 134 120 102 121 104 103 105 95 107 106 128 79 112 131 111 147 148 149 \""
+
--- linux-2.6.13/drivers/net/avm_cpmac/Makefile.26
+++ linux-2.6.13/drivers/net/avm_cpmac/Makefile.26
@@ -23,6 +23,10 @@
 
 $(obj)/dummy.o: $(src)/linux_avm_cpmac.h
 	touch $(obj)/dummy.o
-	cp -f $(src)/linux_avm_cpmac.h $(FRITZ_BOX_BUILD_DIR)/$(KERNEL_BUILD)/filesystem/include/linux/avm_cpmac.h
 
-EXTRA_CFLAGS += -DAVM_CPMAC_VERSION="\"`svn info $(src) | grep URL`  -  Revision `svnversion -cn $(src)`  -  `date` \""
+EXTRA_CFLAGS += -DAVM_CPMAC_VERSION="\"URL: svn://EmbeddedVM/home/SVN/drivers/cpmac/tags/1.104-r759-trunk-PHY_Zuordnung_7212  -  Revision 538:760  -  Mi 16. Feb 13:01:10 CET 2009\""
+EXTRA_CFLAGS  += -DAVM_CPMAC_SWITCH_HWREV_LIST="\"79 84 94 95 101 102 103 104 105 107 110 111 115 117 119 120 121 122 125 126 127 131 134 135 136 137 139 140 141 144 145 146 147 148 152 153 156\""
+EXTRA_CFLAGS  += -DAVM_CPMAC_VINAX_HWREV_LIST="\"105 134 135 146 153 \""
+EXTRA_CFLAGS  += -DAVM_CPMAC_NO_PHY0_HWREV_LIST="\"79 84 138 154 \""
+EXTRA_CFLAGS  += -DAVM_CPMAC_NO_PHY1_HWREV_LIST="\"K 58 60 61 64 65 66 78 122 135 136 137 139 141 143 144 145 146 147 148 149 153 \""
+EXTRA_CFLAGS  += -DAVM_CPMAC_NQOS_HWREV_LIST="\" \""
