--- src/othptab.c.orig	2010-09-23 16:23:46.000000000 +0200
+++ src/othptab.c	2010-09-23 16:23:46.000000000 +0200
@@ -18,6 +18,7 @@
 
 #include <stdlib.h>
 #include <asm/types.h>
+#include <sys/types.h>
 #include <sys/socket.h>
 #include <linux/if_ether.h>
 #include <linux/if_tr.h>
--- src/tr.c.orig	2005-09-13 08:42:54.000000000 +0200
+++ src/tr.c	2010-09-23 16:23:46.000000000 +0200
@@ -7,6 +7,7 @@
  */
 
 #include <asm/types.h>
+#include <sys/types.h>
 #include <linux/if_tr.h>
 #include <netinet/in.h>
 
