diff -urN src/othptab.c iptraf-3.0.0.new/src/othptab.c
--- src/othptab.c	2005-09-13 08:42:54.000000000 +0200
+++ iptraf-3.0.0.new/src/othptab.c	2009-01-16 15:45:08.000000000 +0100
@@ -17,6 +17,7 @@
 ***/
 
 #include <asm/types.h>
+#include <sys/types.h>
 #include <linux/if_ether.h>
 #include <linux/if_tr.h>
 #include <linux/if_fddi.h>
diff -urN src/tr.c iptraf-3.0.0.new/src/tr.c
--- src/tr.c	2005-09-13 08:42:54.000000000 +0200
+++ iptraf-3.0.0.new/src/tr.c	2009-01-16 15:47:47.000000000 +0100
@@ -7,6 +7,7 @@
  */
 
 #include <asm/types.h>
+#include <sys/types.h>
 #include <linux/if_tr.h>
 #include <netinet/in.h>
 
