--- src/Makefile.orig	2005-09-13 09:11:17.000000000 +0000
+++ src/Makefile	2010-02-06 16:16:20.000000000 +0000
@@ -73,7 +73,7 @@
 # The IPTraf working directory; if you change this.  Starting with this
 # version, this definition overrides dirs.h.
 
-WORKDIR		= /var/local/iptraf
+WORKDIR		= /var/tmp/
 
 # The IPTraf log file directory.  IPTraf log files are placed here.
 # This definition overrides dirs.h
