--- src/Makefile.orig	2009-05-17 17:39:39.000000000 +0200
+++ src/Makefile	2009-05-17 17:42:52.000000000 +0200
@@ -74,7 +74,7 @@
 # The IPTraf working directory; if you change this.  Starting with this
 # version, this definition overrides dirs.h.
 
-WORKDIR		= /var/local/iptraf
+WORKDIR		= /var/tmp/
 
 # The IPTraf log file directory.  IPTraf log files are placed here.
 # This definition overrides dirs.h
