--- src/dirs.h.orig	2009-05-16 00:08:21.000000000 +0200
+++ src/dirs.h	2009-05-16 00:09:08.000000000 +0200
@@ -26,11 +26,11 @@
  */
 
 #ifndef WORKDIR
-#define WORKDIR		"/var/local/iptraf"
+#define WORKDIR		"/var/tmp"
 #endif
 
 #ifndef LOGDIR
-#define LOGDIR		"/var/log/iptraf"
+#define LOGDIR		"/var/log"
 #endif
 
 #ifndef EXECDIR
@@ -46,7 +46,7 @@
  */
 
 #ifndef LOCKDIR
-#define LOCKDIR		"/var/run/iptraf"
+#define LOCKDIR		"/var/run"
 #endif
 
 /***
