--- bird-1.0.11/sysdep/config.h	2005-02-21 11:05:22.000000000 +0100
+++ bird-1.0.11/sysdep/config.h	2007-03-05 14:13:26.238371888 +0100
@@ -38,16 +38,16 @@
 /* Path to configuration file */
 #ifdef IPV6
 #  ifdef DEBUGGING
-#    define PATH_CONFIG "bird-6.conf"
-#    define PATH_CONTROL_SOCKET "bird-6.ctl"
+#    define PATH_CONFIG PATH_CONFIG_DIR "/bird-6.conf"
+#    define PATH_CONTROL_SOCKET PATH_CONTROL_SOCKET_DIR "/bird-6.ctl"
 #  else
 #    define PATH_CONFIG PATH_CONFIG_DIR "/bird-6.conf"
 #    define PATH_CONTROL_SOCKET PATH_CONTROL_SOCKET_DIR "/bird-6.ctl"
 #  endif
 #else
 #  ifdef DEBUGGING
-#    define PATH_CONFIG "bird.conf"
-#    define PATH_CONTROL_SOCKET "bird.ctl"
+#    define PATH_CONFIG PATH_CONFIG_DIR "/bird.conf"
+#    define PATH_CONTROL_SOCKET PATH_CONTROL_SOCKET_DIR "/bird.ctl"
 #  else
 #    define PATH_CONFIG PATH_CONFIG_DIR "/bird.conf"
 #    define PATH_CONTROL_SOCKET PATH_CONTROL_SOCKET_DIR "/bird.ctl"
