--- ./src/igmp.c	2009-05-13 21:43:18.000000000 +0200
+++ ./src/igmp.c	2014-04-23 14:44:06.633905832 +0200
@@ -149,14 +149,6 @@
                         return;
                     }
                     else if(!isAdressValidForIf(checkVIF, src)) {
-                        struct IfDesc *downVIF = getIfByAddress(src);
-                        if (downVIF && downVIF->state & IF_STATE_DOWNSTREAM) {
-                            my_log(LOG_NOTICE, 0, "The source address %s for group %s is from downstream VIF[%d]. Ignoring.",
-                                inetFmt(src, s1), inetFmt(dst, s2), i);
-                        } else {
-                            my_log(LOG_WARNING, 0, "The source address %s for group %s, is not in any valid net for upstream VIF[%d].",
-                                inetFmt(src, s1), inetFmt(dst, s2), i);
-                        }
                     } else {
                         // Activate the route.
                         int vifindex = checkVIF->index;
