--- etc/init.d/rc.S
+++ etc/init.d/rc.S
@@ -266,6 +266,7 @@
 #########################################################################
 ## DECT
 #########################################################################
+mknod /var/flash/dect_misc c $tffs_major $((0xB0))
 mknod /var/flash/aura-usb c $tffs_major $((0xA0))
 mknod /var/flash/configd c $tffs_major $((0xA1))
 #########################################################################
@@ -1809,6 +1810,16 @@
 fi
 ## VAG2 + VAG3
 isdn_params=""
+ isdn_params="${isdn_params} dect_hw=2"
+ if /usr/bin/checkempty /var/flash/dect_misc; then
+ isdn_params="${isdn_params} dect_on=1"
+ else
+ if /bin/testvalue /var/flash/dect_misc 1 0 0; then
+ isdn_params="${isdn_params} dect_on=0"
+ else
+ isdn_params="${isdn_params} dect_on=1"
+ fi
+ fi
 ## ########### Annex A DTMF/EC settings ###########
 ## ########### done ###########
 ## ########### optional pcmlink settings ###########
@@ -1865,6 +1876,15 @@
 ## LED - Deamon
 ##########################################################################################
 ln -s /dev/new_led /var/led
+case $OEM in
+ tcom|avm)
+ echo "STATES isdn,0 = 0 -> 2, 1 -> 4" >/dev/new_led
+ echo "MAP isdn,0 TO ab,1" >/dev/new_led
+ echo "STATES ab,1 = 1 -> 18, 0 -> 2, 4 -> 1" >/dev/new_led
+ ;;
+ avme)
+ ;;
+esac
 ##########################################################################################
 ## Box Spezifische Einstellungen
 ##########################################################################################
