--- linux-2.6.13/drivers/usb/serial/ftdi_sio.c
+++ linux-2.6.13/drivers/usb/serial/ftdi_sio.c
@@ -407,6 +407,8 @@
 	{ USB_DEVICE(FTDI_VID, FTDI_GUDEADS_E88E_PID) },
 	{ USB_DEVICE(FTDI_VID, FTDI_GUDEADS_E88F_PID) },
 	{ USB_DEVICE(FTDI_VID, FTDI_ELV_UO100_PID) },
+	{ USB_DEVICE(FTDI_VID, FTDI_ELV_FHZ1000_PID) },
+	{ USB_DEVICE(FTDI_VID, FTDI_ELV_FHZ1300_PID) },
 	{ USB_DEVICE(FTDI_VID, FTDI_ELV_UM100_PID) },
 	{ USB_DEVICE(FTDI_VID, LINX_SDMUSBQSS_PID) },
 	{ USB_DEVICE(FTDI_VID, LINX_MASTERDEVEL2_PID) },
--- linux-2.6.13/drivers/usb/serial/ftdi_sio.h
+++ linux-2.6.13/drivers/usb/serial/ftdi_sio.h
@@ -156,6 +156,9 @@
 /* ELV USB Module UM100 (PID sent by Arnim Laeuger) */
 #define FTDI_ELV_UM100_PID     0xFB5A  /* Product Id */

+#define FTDI_ELV_FHZ1000_PID   0xF06F  /* Product Id */
+#define FTDI_ELV_FHZ1300_PID   0xE0E8  /* Product Id */
+
 /*
  * Definitions for ID TECH (www.idt-net.com) devices
  */
