--- Makefile.in.orig	2008-01-30 00:55:46.000000000 +0100
+++ Makefile.in	2008-02-06 20:13:50.000000000 +0100
@@ -409,7 +409,7 @@
 	-test -z "$(sbin_PROGRAMS)" || rm -f $(sbin_PROGRAMS)
 openvpn$(EXEEXT): $(openvpn_OBJECTS) $(openvpn_DEPENDENCIES) 
 	@rm -f openvpn$(EXEEXT)
-	$(LINK) $(openvpn_LDFLAGS) $(openvpn_OBJECTS) $(openvpn_LDADD) $(LIBS)
+	$(LINK) $(openvpn_LDFLAGS) $(openvpn_OBJECTS) $(openvpn_LDADD) $(LIBS) $(LDL)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
