--- Makefile.in	2009-06-07 08:47:58.000000000 +0200
+++ Makefile.in	2009-06-20 11:30:54.000000000 +0200
@@ -455,7 +455,7 @@
 	-test -z "$(sbin_PROGRAMS)" || rm -f $(sbin_PROGRAMS)
 openvpn$(EXEEXT): $(openvpn_OBJECTS) $(openvpn_DEPENDENCIES) 
 	@rm -f openvpn$(EXEEXT)
-	$(LINK) $(openvpn_OBJECTS) $(openvpn_LDADD) $(LIBS)
+	$(LINK) $(openvpn_OBJECTS) $(openvpn_LDADD) $(LIBS) $(LDL)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
