--- library/Makefile
+++ library/Makefile
@@ -34,6 +34,13 @@
 LDFLAGS += -lws2_32
 endif
 
+VERSION=0.0.0
+VERSION_MAJOR=$(word 1,$(subst ., ,$(VERSION)))
+VERSION_MINOR=$(word 2,$(subst ., ,$(VERSION)))
+LIB_SO=libpolarssl.$(DLEXT)
+LIB_SO_VERSION=$(LIB_SO).$(VERSION)
+LIB_SO_SONAME=$(LIB_SO).$(VERSION_MAJOR).$(VERSION_MINOR)
+
 OBJS=	aes.o		arc4.o		asn1parse.o		\
 		asn1write.o base64.o	bignum.o		\
 		blowfish.o	camellia.o					\
@@ -61,7 +68,7 @@
 
 static: libpolarssl.a
 
-shared: libpolarssl.$(DLEXT)
+shared: $(LIB_SO_VERSION)
 
 libpolarssl.a: $(OBJS)
 	echo "  AR    $@"
@@ -69,9 +76,11 @@
 	echo "  RL    $@"
 	$(AR) s $@
 
-libpolarssl.so: libpolarssl.a
+$(LIB_SO_VERSION): $(OBJS)
 	echo "  LD    $@"
-	$(CC) ${LDFLAGS} -shared -Wl,-soname,$(SONAME) -o $@ $(OBJS)
+	$(CC) ${LDFLAGS} -shared -Wl,-soname,$(LIB_SO_SONAME) -o $@ $(OBJS) \
+	&& ln -fs $@ $(LIB_SO_SONAME) \
+	&& ln -fs $(LIB_SO_SONAME) $(LIB_SO)
 
 libpolarssl.dylib: libpolarssl.a
 	echo "  LD    $@"
