--- etc/hotplug/storage.orig	2009-02-25 10:42:05.000000000 +0100
+++ etc/hotplug/storage	2009-02-27 09:02:47.000000000 +0100
@@ -13,6 +13,10 @@
 FIRST=$6
 MSTART=/sbin/start_mediasrv
 MSTOP=/sbin/stop_mediasrv
+
+# Do we have a Ftp server (ftpd) or was ist stripped from the firmware?
+_ftpd=$(basename $(which ftpd) 2> /dev/null)
+
 ## Wait for removal of filesystem semaphores (mask: $1, maximal wait period: $2)
 wait4remove () {
 local COUNT=0
@@ -146,6 +150,7 @@
 ## ftp server control
 ENABLED=`echo usbhost.ftp_server_enabled|usbcfgctl -s`
 if [ "$ENABLED" != "no" ] ; then
+if [ "$_ftpd" ] ; then
 if test -x /bin/inetdctl ; then
 inetdctl enable ftpd
 else
@@ -162,6 +167,7 @@
 ftpd -D -q $READONLYOPTION -t 120 -m 15 $USERSOPTION -h "$CONFIG_PRODUKT_NAME" &
 fi
 fi
+fi
 else
 if test -x /bin/inetdctl ; then
 inetdctl disable ftpd
--- etc/hotplug/run_mount.orig	2009-02-25 10:41:52.000000000 +0100
+++ etc/hotplug/run_mount	2009-02-27 09:02:47.000000000 +0100
@@ -15,6 +15,10 @@
 MSTART=/sbin/start_mediasrv
 TR069START=/bin/tr069starter
 SYSFS=/var/sysfs
+
+# Do we have a Ftp server (ftpd) or was ist stripped from the firmware?
+_ftpd=$(basename $(which ftpd) 2> /dev/null)
+
 ## Generate a frontend'able 'nice name' to be used as mount point:
 nicename () {
 ## Find device's strings, remove/translate evil chars...
@@ -115,6 +119,7 @@
 sh /sbin/start_plugin.sh usb_mount $MNTPATH > /dev/ttyS0
 ENABLED=`echo usbhost.ftp_server_enabled|usbcfgctl -s`
 if [ "$ENABLED" != "no" ] ; then
+if [ "$_ftpd" ] ; then
 if test -x /bin/inetdctl; then
 inetdctl enable ftpd
 else
@@ -134,6 +139,7 @@
 fi
 fi
 fi
+fi
 test -x $TR069START && $TR069START $MNTNAME
 if test -x /etc/samba_control ; then
 /etc/samba_control reconfig
