--- Makefile.in
+++ Makefile.in
@@ -259,7 +259,7 @@
 radvd_SOURCES = $(COMMON_SRC) radvd.c timer.c send.c process.c interface.c \
 	device.c device-common.c privsep.c gram.y gram.h scanner.l
 
-radvd_LDADD = -lfl
+radvd_LDADD =
 EXTRA_radvd_SOURCES = device-linux.c device-bsd44.c privsep-linux.c privsep-bsd44.c
 radvdump_SOURCES = $(COMMON_SRC) radvdump.c
 man_MANS = radvd.8 radvdump.8 radvd.conf.5
--- scanner.l
+++ scanner.l
@@ -24,6 +24,8 @@
 int num_lines = 1;
 %}
 
+%option noyywrap
+
 digit		[0-9]
 number		({digit})+
 snum		-?({digit})+
