--- recv.c.orig	2006-10-09 08:21:59.000000000 +0200
+++ recv.c	2009-01-13 19:39:03.000000000 +0100
@@ -18,6 +18,14 @@
 #include <includes.h>
 #include <radvd.h>
 
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,19) && (defined(__UCLIBC__) && __UCLIBC_MAJOR__ == 0 && __UCLIBC_MINOR__ == 9 && __UCLIBC_SUBLEVEL__ >= 29)
+# undef IPV6_HOPLIMIT
+# undef IPV6_PKTINFO
+# define IPV6_HOPLIMIT IPV6_2292HOPLIMIT
+# define IPV6_PKTINFO IPV6_2292PKTINFO
+#endif
+
 int
 recv_rs_ra(int sock, unsigned char *msg, struct sockaddr_in6 *addr,
                  struct in6_pktinfo **pkt_info, int *hoplimit)
--- send.c.orig	2008-10-15 07:34:35.000000000 +0200
+++ send.c	2009-01-13 19:40:05.000000000 +0100
@@ -18,6 +18,12 @@
 #include <includes.h>
 #include <radvd.h>
 
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,19) && (defined(__UCLIBC__) && __UCLIBC_MAJOR__ == 0 && __UCLIBC_MINOR__ == 9 && __UCLIBC_SUBLEVEL__ >= 29)
+# undef IPV6_PKTINFO
+# define IPV6_PKTINFO IPV6_2292PKTINFO
+#endif
+
 void
 send_ra(int sock, struct Interface *iface, struct in6_addr *dest)
 {
--- socket.c.orig	2005-10-18 21:22:00.000000000 +0200
+++ socket.c	2009-01-13 19:37:03.000000000 +0100
@@ -18,6 +18,13 @@
 #include <includes.h>
 #include <radvd.h>
 
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,19) && (defined(__UCLIBC__) && __UCLIBC_MAJOR__ == 0 && __UCLIBC_MINOR__ == 9 && __UCLIBC_SUBLEVEL__ >= 29)
+# undef IPV6_RECVHOPLIMIT
+# undef IPV6_RECVPKTINFO
+# define IPV6_RECVHOPLIMIT IPV6_2292HOPLIMIT
+# define IPV6_RECVPKTINFO IPV6_2292PKTINFO
+#else
 /* Note: these are applicable to receiving sockopts only */
 #if defined IPV6_HOPLIMIT && !defined IPV6_RECVHOPLIMIT
 # define IPV6_RECVHOPLIMIT IPV6_HOPLIMIT
@@ -26,6 +33,7 @@
 #if defined IPV6_PKTINFO && !defined IPV6_RECVPKTINFO
 # define IPV6_RECVPKTINFO IPV6_PKTINFO
 #endif
+#endif
 
 int
 open_icmpv6_socket(void)
