--- config.h.in
+++ config.h.in
@@ -190,6 +190,9 @@
 /* Define to 1 if `st_atimespec' is member of `struct stat'. */
 #undef HAVE_STRUCT_STAT_ST_ATIMESPEC
 
+/* Define to 1 if `st_atimensec' is member of `struct stat'. */
+#undef HAVE_STRUCT_STAT_ST_ATIMENSEC
+
 /* Define to 1 if `st_blocks' is member of `struct stat'. */
 #undef HAVE_STRUCT_STAT_ST_BLOCKS
 
--- configure.ac
+++ configure.ac
@@ -280,6 +280,7 @@
 AC_CHECK_MEMBERS([struct stat.st_rdev])
 AC_CHECK_MEMBERS([struct stat.st_atim])
 AC_CHECK_MEMBERS([struct stat.st_atimespec])
+AC_CHECK_MEMBERS([struct stat.st_atimensec])
 
 # For the 'nfconv' patch (Mac OS X only):
 case "${target_os}" in
--- libfuse-lite/fuse_misc.h
+++ libfuse-lite/fuse_misc.h
@@ -36,6 +36,12 @@
 #define ST_MTIM_NSEC(stbuf) ((stbuf)->st_mtimespec.tv_nsec)
 #define ST_ATIM_NSEC_SET(stbuf, val) (stbuf)->st_atimespec.tv_nsec = (val)
 #define ST_MTIM_NSEC_SET(stbuf, val) (stbuf)->st_mtimespec.tv_nsec = (val)
+#elif defined(HAVE_STRUCT_STAT_ST_ATIMENSEC)
+#define ST_ATIM_NSEC(stbuf) ((stbuf)->st_atimensec)
+#define ST_CTIM_NSEC(stbuf) ((stbuf)->st_ctimensec)
+#define ST_MTIM_NSEC(stbuf) ((stbuf)->st_mtimensec)
+#define ST_ATIM_NSEC_SET(stbuf, val) (stbuf)->st_atimensec = (val)
+#define ST_MTIM_NSEC_SET(stbuf, val) (stbuf)->st_mtimensec = (val)
 #else
 #define ST_ATIM_NSEC(stbuf) 0
 #define ST_CTIM_NSEC(stbuf) 0
--- src/ntfs-3g.c
+++ src/ntfs-3g.c
@@ -870,6 +870,28 @@
  	stbuf->st_atim = ntfs2timespec(ni->last_access_time);
  	stbuf->st_ctim = ntfs2timespec(ni->last_mft_change_time);
  	stbuf->st_mtim = ntfs2timespec(ni->last_data_change_time);
+#elif defined(HAVE_STRUCT_STAT_ST_ATIMENSEC)
+	{
+		struct timespec l_st_atim = ntfs2timespec(ni->last_access_time);
+		struct timespec l_st_ctim = ntfs2timespec(ni->last_mft_change_time);
+		struct timespec l_st_mtim = ntfs2timespec(ni->last_data_change_time);
+		stbuf->st_atime = l_st_atim.tv_sec;
+		stbuf->st_atimensec = l_st_atim.tv_nsec;
+		stbuf->st_ctime = l_st_ctim.tv_sec;
+		stbuf->st_ctimensec = l_st_ctim.tv_nsec;
+		stbuf->st_mtime = l_st_mtim.tv_sec;
+		stbuf->st_mtimensec = l_st_mtim.tv_nsec;
+	}
+#elif defined(__UCLIBC__) && __UCLIBC_MAJOR__==0 && __UCLIBC_MINOR__==9 && __UCLIBC_SUBLEVEL__==28
+	{
+		// uClibc-0.9.28 doesn't support nanoseconds
+		struct timespec l_st_tim = ntfs2timespec(ni->last_access_time);
+		stbuf->st_atime = l_st_tim.tv_sec;
+		l_st_tim = ntfs2timespec(ni->last_mft_change_time);
+		stbuf->st_ctime = l_st_tim.tv_sec;
+		l_st_tim = ntfs2timespec(ni->last_data_change_time);
+		stbuf->st_mtime = l_st_tim.tv_sec;
+	}
 #else
 	#error "No known timespec member in struct stat!"
 #endif
--- configure
+++ configure
@@ -21888,6 +21888,108 @@
 
 fi
 
+{ echo "$as_me:$LINENO: checking for struct stat.st_atimensec" >&5
+echo $ECHO_N "checking for struct stat.st_atimensec... $ECHO_C" >&6; }
+if test "${ac_cv_member_struct_stat_st_atimensec+set}" = set; then
+  echo $ECHO_N "(cached) $ECHO_C" >&6
+else
+  cat >conftest.$ac_ext <<_ACEOF
+/* confdefs.h.  */
+_ACEOF
+cat confdefs.h >>conftest.$ac_ext
+cat >>conftest.$ac_ext <<_ACEOF
+/* end confdefs.h.  */
+$ac_includes_default
+int
+main ()
+{
+static struct stat ac_aggr;
+if (ac_aggr.st_atimensec)
+return 0;
+  ;
+  return 0;
+}
+_ACEOF
+rm -f conftest.$ac_objext
+if { (ac_try="$ac_compile"
+case "(($ac_try" in
+  *\"* | *\`* | *\\*) ac_try_echo=\$ac_try;;
+  *) ac_try_echo=$ac_try;;
+esac
+eval "echo \"\$as_me:$LINENO: $ac_try_echo\"") >&5
+  (eval "$ac_compile") 2>conftest.er1
+  ac_status=$?
+  grep -v '^ *+' conftest.er1 >conftest.err
+  rm -f conftest.er1
+  cat conftest.err >&5
+  echo "$as_me:$LINENO: \$? = $ac_status" >&5
+  (exit $ac_status); } && {
+	 test -z "$ac_c_werror_flag" ||
+	 test ! -s conftest.err
+       } && test -s conftest.$ac_objext; then
+  ac_cv_member_struct_stat_st_atimensec=yes
+else
+  echo "$as_me: failed program was:" >&5
+sed 's/^/| /' conftest.$ac_ext >&5
+
+	cat >conftest.$ac_ext <<_ACEOF
+/* confdefs.h.  */
+_ACEOF
+cat confdefs.h >>conftest.$ac_ext
+cat >>conftest.$ac_ext <<_ACEOF
+/* end confdefs.h.  */
+$ac_includes_default
+int
+main ()
+{
+static struct stat ac_aggr;
+if (sizeof ac_aggr.st_atimensec)
+return 0;
+  ;
+  return 0;
+}
+_ACEOF
+rm -f conftest.$ac_objext
+if { (ac_try="$ac_compile"
+case "(($ac_try" in
+  *\"* | *\`* | *\\*) ac_try_echo=\$ac_try;;
+  *) ac_try_echo=$ac_try;;
+esac
+eval "echo \"\$as_me:$LINENO: $ac_try_echo\"") >&5
+  (eval "$ac_compile") 2>conftest.er1
+  ac_status=$?
+  grep -v '^ *+' conftest.er1 >conftest.err
+  rm -f conftest.er1
+  cat conftest.err >&5
+  echo "$as_me:$LINENO: \$? = $ac_status" >&5
+  (exit $ac_status); } && {
+	 test -z "$ac_c_werror_flag" ||
+	 test ! -s conftest.err
+       } && test -s conftest.$ac_objext; then
+  ac_cv_member_struct_stat_st_atimensec=yes
+else
+  echo "$as_me: failed program was:" >&5
+sed 's/^/| /' conftest.$ac_ext >&5
+
+	ac_cv_member_struct_stat_st_atimensec=no
+fi
+
+rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
+fi
+
+rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
+fi
+{ echo "$as_me:$LINENO: result: $ac_cv_member_struct_stat_st_atimensec" >&5
+echo "${ECHO_T}$ac_cv_member_struct_stat_st_atimensec" >&6; }
+if test $ac_cv_member_struct_stat_st_atimensec = yes; then
+
+cat >>confdefs.h <<_ACEOF
+#define HAVE_STRUCT_STAT_ST_ATIMENSEC 1
+_ACEOF
+
+
+fi
+
 
 # For the 'nfconv' patch (Mac OS X only):
 case "${target_os}" in
