--- configure.ac.orig	2008-08-11 18:10:29.000000000 +0200
+++ configure.ac	2008-09-08 21:21:51.000000000 +0200
@@ -232,7 +232,7 @@
 
 # Checks for header files.
 AC_HEADER_STDC
-AC_CHECK_HEADERS([ctype.h fcntl.h libgen.h libintl.h limits.h locale.h \
+AC_CHECK_HEADERS([ctype.h fcntl.h libgen.h limits.h locale.h \
 	mntent.h stddef.h stdint.h stdlib.h stdio.h stdarg.h string.h \
 	strings.h errno.h time.h unistd.h utime.h wchar.h getopt.h features.h \
 	endian.h byteswap.h sys/byteorder.h sys/endian.h sys/param.h \
