--- configure.ac.orig	2008-01-28 20:06:29.000000000 +0100
+++ configure.ac	2008-02-10 13:27:34.000000000 +0100
@@ -218,7 +218,7 @@
 
 # Checks for header files.
 AC_HEADER_STDC
-AC_CHECK_HEADERS([ctype.h fcntl.h libgen.h libintl.h limits.h locale.h \
+AC_CHECK_HEADERS([ctype.h fcntl.h libgen.h limits.h locale.h \
 	mntent.h stddef.h stdint.h stdlib.h stdio.h stdarg.h string.h \
 	strings.h errno.h time.h unistd.h utime.h wchar.h getopt.h features.h \
 	endian.h byteswap.h sys/byteorder.h sys/endian.h sys/param.h \
