--- configure.ac	2008-11-13 21:49:39.000000000 +0100
+++ configure.ac	2008-12-21 00:00:00.000000000 +0100
@@ -251,7 +251,7 @@
 
 # Checks for header files.
 AC_HEADER_STDC
-AC_CHECK_HEADERS([ctype.h fcntl.h libgen.h libintl.h limits.h locale.h \
+AC_CHECK_HEADERS([ctype.h fcntl.h libgen.h limits.h locale.h \
 	mntent.h stddef.h stdint.h stdlib.h stdio.h stdarg.h string.h \
 	strings.h errno.h time.h unistd.h utime.h wchar.h getopt.h features.h \
 	endian.h byteswap.h sys/byteorder.h sys/endian.h sys/param.h \
--- configure	2009-11-13 21:51:34.000000000 +0100
+++ configure	2009-12-21 00:00:00.000000000 +0100
@@ -20692,7 +20692,7 @@
 
 
 
-for ac_header in ctype.h fcntl.h libgen.h libintl.h limits.h locale.h \
+for ac_header in ctype.h fcntl.h libgen.h limits.h locale.h \
 	mntent.h stddef.h stdint.h stdlib.h stdio.h stdarg.h string.h \
 	strings.h errno.h time.h unistd.h utime.h wchar.h getopt.h features.h \
 	endian.h byteswap.h sys/byteorder.h sys/endian.h sys/param.h \
