--- configure.ac
+++ configure.ac
@@ -474,7 +474,7 @@
 # Checks for header files.
 AC_HEADER_STDC
 AC_HEADER_MAJOR
-AC_CHECK_HEADERS([ctype.h fcntl.h libgen.h libintl.h limits.h locale.h \
+AC_CHECK_HEADERS([ctype.h fcntl.h libgen.h           limits.h locale.h \
 	mntent.h stddef.h stdint.h stdlib.h stdio.h stdarg.h string.h \
 	strings.h errno.h time.h unistd.h utime.h wchar.h getopt.h features.h \
 	regex.h endian.h byteswap.h sys/byteorder.h sys/disk.h sys/endian.h \
--- configure
+++ configure
@@ -13830,7 +13830,7 @@
   fi
 fi
 
-for ac_header in ctype.h fcntl.h libgen.h libintl.h limits.h locale.h \
+for ac_header in ctype.h fcntl.h libgen.h           limits.h locale.h \
 	mntent.h stddef.h stdint.h stdlib.h stdio.h stdarg.h string.h \
 	strings.h errno.h time.h unistd.h utime.h wchar.h getopt.h features.h \
 	regex.h endian.h byteswap.h sys/byteorder.h sys/disk.h sys/endian.h \
