--- configure
+++ configure
@@ -9148,6 +9148,7 @@
   ;;
 esac
 
+if false; then
 { echo "$as_me:$LINENO: checking dynamic linker characteristics" >&5
 echo $ECHO_N "checking dynamic linker characteristics... $ECHO_C" >&6; }
 library_names_spec=
@@ -9810,6 +9811,7 @@
   # Fast installation is not necessary
   enable_fast_install=needless
 fi
+fi
 
 striplib=
 old_striplib=
@@ -13163,6 +13165,7 @@
   ;;
 esac
 
+if false; then
 { echo "$as_me:$LINENO: checking dynamic linker characteristics" >&5
 echo $ECHO_N "checking dynamic linker characteristics... $ECHO_C" >&6; }
 library_names_spec=
@@ -13773,6 +13776,7 @@
   # Fast installation is not necessary
   enable_fast_install=needless
 fi
+fi
 
 
 # The else clause should only fire when bootstrapping the
@@ -15784,6 +15788,7 @@
   ;;
 esac
 
+if false; then
 { echo "$as_me:$LINENO: checking dynamic linker characteristics" >&5
 echo $ECHO_N "checking dynamic linker characteristics... $ECHO_C" >&6; }
 library_names_spec=
@@ -16394,6 +16399,7 @@
   # Fast installation is not necessary
   enable_fast_install=needless
 fi
+fi
 
 
 # The else clause should only fire when bootstrapping the
@@ -18405,6 +18411,7 @@
   ;;
 esac
 
+if false; then
 { echo "$as_me:$LINENO: checking dynamic linker characteristics" >&5
 echo $ECHO_N "checking dynamic linker characteristics... $ECHO_C" >&6; }
 library_names_spec=
@@ -19015,6 +19022,7 @@
   # Fast installation is not necessary
   enable_fast_install=needless
 fi
+fi
 
 
 # The else clause should only fire when bootstrapping the
