--- configure	2009-11-13 21:51:34.000000000 +0100
+++ configure	2009-12-21 00:00:00.000000000 +0100
@@ -9068,6 +9068,7 @@
   ;;
 esac
 
+if false; then
 { echo "$as_me:$LINENO: checking dynamic linker characteristics" >&5
 echo $ECHO_N "checking dynamic linker characteristics... $ECHO_C" >&6; }
 library_names_spec=
@@ -9730,6 +9731,7 @@
   # Fast installation is not necessary
   enable_fast_install=needless
 fi
+fi
 
 striplib=
 old_striplib=
@@ -13083,6 +13085,7 @@
   ;;
 esac
 
+if false; then
 { echo "$as_me:$LINENO: checking dynamic linker characteristics" >&5
 echo $ECHO_N "checking dynamic linker characteristics... $ECHO_C" >&6; }
 library_names_spec=
@@ -13693,6 +13696,7 @@
   # Fast installation is not necessary
   enable_fast_install=needless
 fi
+fi
 
 
 # The else clause should only fire when bootstrapping the
@@ -15704,6 +15708,7 @@
   ;;
 esac
 
+if false; then
 { echo "$as_me:$LINENO: checking dynamic linker characteristics" >&5
 echo $ECHO_N "checking dynamic linker characteristics... $ECHO_C" >&6; }
 library_names_spec=
@@ -16314,6 +16319,7 @@
   # Fast installation is not necessary
   enable_fast_install=needless
 fi
+fi
 
 
 # The else clause should only fire when bootstrapping the
@@ -18325,6 +18331,7 @@
   ;;
 esac
 
+if false; then
 { echo "$as_me:$LINENO: checking dynamic linker characteristics" >&5
 echo $ECHO_N "checking dynamic linker characteristics... $ECHO_C" >&6; }
 library_names_spec=
@@ -18935,6 +18942,7 @@
   # Fast installation is not necessary
   enable_fast_install=needless
 fi
+fi
 
 
 # The else clause should only fire when bootstrapping the
