--- configure
+++ configure
@@ -9123,6 +9123,7 @@
   ;;
 esac
 
+if false; then
 { echo "$as_me:$LINENO: checking dynamic linker characteristics" >&5
 echo $ECHO_N "checking dynamic linker characteristics... $ECHO_C" >&6; }
 library_names_spec=
@@ -9785,6 +9786,7 @@
   # Fast installation is not necessary
   enable_fast_install=needless
 fi
+fi
 
 striplib=
 old_striplib=
@@ -13138,6 +13140,7 @@
   ;;
 esac
 
+if false; then
 { echo "$as_me:$LINENO: checking dynamic linker characteristics" >&5
 echo $ECHO_N "checking dynamic linker characteristics... $ECHO_C" >&6; }
 library_names_spec=
@@ -13748,6 +13751,7 @@
   # Fast installation is not necessary
   enable_fast_install=needless
 fi
+fi
 
 
 # The else clause should only fire when bootstrapping the
@@ -15759,6 +15763,7 @@
   ;;
 esac
 
+if false; then
 { echo "$as_me:$LINENO: checking dynamic linker characteristics" >&5
 echo $ECHO_N "checking dynamic linker characteristics... $ECHO_C" >&6; }
 library_names_spec=
@@ -16369,6 +16374,7 @@
   # Fast installation is not necessary
   enable_fast_install=needless
 fi
+fi
 
 
 # The else clause should only fire when bootstrapping the
@@ -18380,6 +18386,7 @@
   ;;
 esac
 
+if false; then
 { echo "$as_me:$LINENO: checking dynamic linker characteristics" >&5
 echo $ECHO_N "checking dynamic linker characteristics... $ECHO_C" >&6; }
 library_names_spec=
@@ -18990,6 +18997,7 @@
   # Fast installation is not necessary
   enable_fast_install=needless
 fi
+fi
 
 
 # The else clause should only fire when bootstrapping the
