--- usr/www/avme/html/de/home/home.js.orig	2010-10-29 15:42:53.000000000 +0200
+++ usr/www/avme/html/de/home/home.js	2011-01-13 15:40:17.490905570 +0100
@@ -723,7 +723,7 @@
 var nPartTotal = Number(g_usbhost_data.part_count_total);
 if (isNaN(nPartTotal)) nPartTotal = 0;
 var unplugOk = parseInt(g_usbhost_data.unplug_ok, 10) == 1;
-if (nPartTotal > nStor)
+if (nPartTotal > 0)
 {
 if (unplugOk)
 {
@@ -731,9 +731,7 @@
 }
 else
 {
-strStor = "{?895:5106?} ";
-strStor += "<a href=\"javascript:syslog_jslSubmitFormEx(5);\">{?895:8842?}</a>";
-strStor = strStor.fontcolor("#cc0000");
+strStor = "Mounted " + nStor + " of " + nPartTotal + " partitions.";
 }
 strArr.push(strStor);
 }
--- usr/www/avme/html/de/usb/status.js.orig	2010-10-29 15:42:40.000000000 +0200
+++ usr/www/avme/html/de/usb/status.js	2011-01-13 15:40:17.490905570 +0100
@@ -481,8 +481,7 @@
 }
 else
 {
-str += "<font color=\"#cc0000\">";
-str += "{?2798:8143?}";
+strStor = "Mounted " + nStor + " of " + nPartTotal + " partitions.";
 str += "</font>";
 str += "<a href=\"javascript:goto_syslog_usb();\">{?2798:9581?}</a>";
 }
