--- logincontrol.c
+++ logincontrol.c
@@ -18,6 +18,7 @@
 #include <string.h>
 #include <time.h>
 #include <sys/types.h>
+#include <sys/wait.h>
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <errno.h>
@@ -229,12 +230,24 @@
 	return is;
 }
 
-static void Event(unsigned  id, const char *display_name)
+static void Event(unsigned id, const char *display_name)
 {
-	char buf[256];
-	snprintf(buf, sizeof(buf)-1, "/sbin/eventadd %u %s", id, display_name);
-	buf[sizeof(buf)-1] = '\0';
-	system(buf);
+	int status;
+	pid_t pid;
+
+	pid = fork();
+	if (pid < 0) {
+		return;
+	}
+	if (pid == 0) { /* child */
+		char id_buf[32];
+		snprintf(id_buf, sizeof(id_buf), "%u", id);
+		id_buf[sizeof(id_buf)-1] = '\0';
+		(void)execl("/sbin/eventadd", "eventadd", id_buf, display_name, NULL);
+		exit(1);
+	}
+	/* parent */
+	(void)waitpid( pid, &status, 0);
 }
 
 
