--- avm_acl.h
+++ avm_acl.h
@@ -11,6 +11,9 @@
  *
  */
 
+#ifndef AVMACL2_AVM_ACL_H_
+#define AVMACL2_AVM_ACL_H_
+
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -57,6 +60,7 @@
 }
 #endif
 
+#endif /* !AVMACL2_AVM_ACL_H_ */
 
 
 
--- logincontrol.h
+++ logincontrol.h
@@ -11,6 +11,9 @@
  *
  */
 
+#ifndef AVMACL2_LOGINCONTROL_H_
+#define AVMACL2_LOGINCONTROL_H_
+
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -32,3 +35,4 @@
 }
 #endif
 
+#endif /* !AVMACL2_LOGINCONTROL_H_ */
