--- src/chrtrans/makefile.in
+++ src/chrtrans/makefile.in
@@ -21,6 +21,12 @@
 CFLAGS		= @CFLAGS@
 EXTRA_CPPFLAGS	= @EXTRA_CPPFLAGS@
 CPPFLAGS	= @CPPFLAGS@
+HOSTCC		= gcc
+HOSTCPP	= gcc -E
+HOSTLD		= ld
+HOSTCFLAGS	= -Os -W -Wall -pipe -Wbad-function-cast -Wcast-align -Wcast-qual -Winline -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wpointer-arith -Wshadow -Wstrict-prototypes -Wundef
+HOSTDEFS	= -DHAVE_CONFIG_H
+HOSTCPPFLAGS	= -DLINUX -D_GNU_SOURCE
 
 WWWINC		= WWW/Library/Implementation
 
@@ -32,7 +38,9 @@
 BUILD_CPPFLAGS	= @BUILD_CPPFLAGS@ @DEFS@
 
 BUILD_LIBS	= @BUILD_LIBS@
+HOSTLIBS	= #
 BUILD_LDFLAGS	= @BUILD_LDFLAGS@
+HOSTLDFLAGS	= #
 
 x		= @EXEEXT@
 o		= .@BUILD_OBJEXT@
@@ -52,6 +60,15 @@
 
 CTAGS		= @CTAGS@
 
+HOSTCPP_OPTS	= $(HOSTDEFS) $(HOSTCPPFLAGS) \
+		-I$(top_builddir) \
+		-I$(top_srcdir)/src \
+		-I$(top_srcdir)/src/chrtrans \
+		-I$(top_srcdir)/$(WWWINC) \
+		-I$(top_srcdir)/ \
+		$(INTLDIR_CPPFLAGS) $(SITE_DEFS)
+HOSTCC_OPTS	= $(HOSTCPP_OPTS) $(HOSTCFLAGS)
+
 #
 # This file contains the font map for the default (hardware) font
 #
@@ -117,7 +134,7 @@
 C_SRC		= $(OBJS:$o=.c)
 
 $(MAKEUCTB) : $(OBJS)
-	$(BUILD_CC) $(CC_OPTS) $(BUILD_LDFLAGS) -o $@ $(OBJS) $(BUILD_LIBS)
+	$(HOSTCC) $(HOSTCC_OPTS) $(HOSTLDFLAGS) -o $@ $(OBJS) $(HOSTLIBS)
 
 makeuctb$o : $(srcdir)/UCkd.h $(srcdir)/makeuctb.c
 
@@ -125,11 +142,11 @@
 
 .c$o:
 	@RULE_CC@
-	@ECHO_CC@$(BUILD_CC) $(CC_OPTS) -c $(srcdir)/$*.c
+	@ECHO_CC@$(HOSTCC) $(HOSTCC_OPTS) -c $(srcdir)/$*.c
 
 .c.i:
 	@RULE_CC@
-	@ECHO_CC@$(BUILD_CPP) $(CPP_OPTS) $(srcdir)/$*.c >$@
+	@ECHO_CC@$(HOSTCPP) $(HOSTCPP_OPTS) $(srcdir)/$*.c >$@
 
 .tbl.h:
 	./$(MAKEUCTB) $(srcdir)/$*.tbl $*.h
