--- linux-2.6.13/drivers/net/avm_cpmac/cpmac_eth.c
+++ linux-2.6.13/drivers/net/avm_cpmac/cpmac_eth.c
@@ -333,6 +333,8 @@
     case SIOCGMIIPHY:
     case SIOCGMIIREG:
     case SIOCSMIIREG:
+    case AVM_CPMAC_IOCTL_INFO:
+    case AVM_CPMAC_IOCTL_GENERIC:
     case AVM_CPMAC_IOCTL_SET_CONFIG_N:
     case AVM_CPMAC_IOCTL_GET_CONFIG_N:
         if(real_dev->do_ioctl && netif_device_present(real_dev))
