--- source3/configure
+++ source3/configure
@@ -7755,6 +7755,7 @@
 $as_echo "$ac_cv_prog_gnu_ld" >&6; }
 
 
+if false; then
 if test "$ac_cv_prog_gnu_ld" = "yes"; then
 	ac_cv_gnu_ld_version=`$CC -Wl,-v /dev/null 2>&1 < /dev/null | grep "GNU ld"`
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking GNU ld release date" >&5
@@ -7793,6 +7794,9 @@
            fi
         fi
 fi
+fi
+#ac_cv_gnu_ld_no_default_allow_shlib_undefined=no
+ac_cv_gnu_ld_version_script=yes
 
 
 
