--- configure
+++ configure
@@ -21725,6 +21725,7 @@
 fi
 
 
+if false; then
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for main in -lnsl" >&5
 $as_echo_n "checking for main in -lnsl... " >&6; }
 if ${ac_cv_lib_nsl_main+:} false; then :
@@ -21763,6 +21764,7 @@
   LIBS="-lnsl $LIBS"
 
 fi
+fi
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for main in -lsocket" >&5
 $as_echo_n "checking for main in -lsocket... " >&6; }
@@ -22141,6 +22143,7 @@
 		echo "skipping libresolv checks for $host"
 		;;
 	*)
+if false; then
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for inet_aton in -lresolv" >&5
 $as_echo_n "checking for inet_aton in -lresolv... " >&6; }
 if ${ac_cv_lib_resolv_inet_aton+:} false; then :
@@ -22263,6 +22266,7 @@
   LIBS="-lresolv $LIBS"
 
 fi
+fi
 
 		;;
     esac
