--- linux-2.6.28.10/drivers/net/avm_cpmac/Makefile	2012-02-14 14:19:47+01:00
+++ linux-2.6.28.10/drivers/net/avm_cpmac/Makefile	2012-02-14 14:19:47+01:00
@@ -12,7 +12,7 @@
 #EXTRA_CFLAGS += -Wmissing-declarations -Wmissing-field-initializers
 ##EXTRA_CFLAGS += -Wswitch-default -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wcast-align -Wconversion -Wpadded 
 ##EXTRA_CFLAGS += -Wredundant-decls -Wunsafe-loop-optimizations
-EXTRA_CFLAGS += -DAVM_CPMAC_VERSION="\"`svn info $(src) | grep URL | sed -e "s/^.*\/\([0-9.]\+\)-[^\/]*/\1/g"`  -  Revision `svnversion -cn $(src)`  -  `date +\"%a  %e.%m.%Y  %H:%M:%S\"` \""
+EXTRA_CFLAGS += -DAVM_CPMAC_VERSION="\"5.14 - Revision 1228:1440 - Mi  5.10.2011 13:42:56 \""
 
 NOTANGLE=notangle
 NOWEAVE=noweave
