--- linux-2.6.28.10/drivers/char/ifxmips_mei.c.orig	2011-09-06 08:35:56.080733428 +0200
+++ linux-2.6.28.10/drivers/char/ifxmips_mei.c	2011-09-06 08:35:44.950873348 +0200
@@ -63,7 +63,7 @@
 #define IFXMIPS_ICU_IM2_IER               IFX_ICU_IM2_IER
 #define IFXMIPS_MEI_INT                   IFX_MEI_INT
 #define IFXMIPS_MEI_DYING_GASP_INT        IFX_MEI_DYING_GASP_INT
-#define IFXMIPS_MEI_BASE_ADDR             IFX_MEI_SPACE_ACCESS
+//#define IFXMIPS_MEI_BASE_ADDR             IFX_MEI_SPACE_ACCESS
 #define IFXMIPS_PMU_PWDCR                 IFX_PMU_PWDCR
 #define IFXMIPS_MPS_CHIPID                IFX_MPS_CHIPID
 
--- linux-2.6.28.10/drivers/usb/Kconfig.orig	2011-11-09 07:26:36.607408904 +0100
+++ linux-2.6.28.10/drivers/usb/Kconfig	2011-11-08 22:49:12.000000000 +0100
@@ -104,7 +104,7 @@
 
 source "drivers/usb/host/Kconfig"
 
-
+source "drivers/usb/musb/Kconfig"
 
 source "drivers/usb/class/Kconfig"
 
