--- addons/chan_mobile.c
+++ addons/chan_mobile.c
@@ -27,7 +27,6 @@
 
 /*** MODULEINFO
 	<depend>bluetooth</depend>
-	<defaultenabled>no</defaultenabled>
 	<support_level>extended</support_level>
  ***/
 
--- addons/chan_ooh323.c
+++ addons/chan_ooh323.c
@@ -18,7 +18,6 @@
 
 
 /*** MODULEINFO
-	<defaultenabled>no</defaultenabled>
 	<support_level>extended</support_level>
  ***/
 
--- addons/format_mp3.c
+++ addons/format_mp3.c
@@ -28,7 +28,6 @@
  */
 
 /*** MODULEINFO
-	<defaultenabled>no</defaultenabled>
 	<support_level>extended</support_level>
  ***/
 
--- apps/app_meetme.c
+++ apps/app_meetme.c
@@ -31,7 +31,6 @@
 
 /*** MODULEINFO
 	<depend>dahdi</depend>
-	<defaultenabled>no</defaultenabled>
 	<support_level>extended</support_level>
 	<replacement>app_confbridge</replacement>
  ***/
--- res/res_pktccops.c
+++ res/res_pktccops.c
@@ -31,7 +31,6 @@
  */
 
 /*** MODULEINFO
-        <defaultenabled>no</defaultenabled>
 	<support_level>extended</support_level>
  ***/
 
