--- linux-2.6.19.2/net/ipv4/netfilter/ip_conntrack_core.c.orig	2011-02-08 22:01:35.000000000 +0100
+++ linux-2.6.19.2/net/ipv4/netfilter/ip_conntrack_core.c	2011-02-27 14:16:21.456003485 +0100
@@ -55,9 +55,11 @@
 #else
 #define DEBUGP(format, args...)
 #endif
+/*
 #if defined(CONFIG_FUSIV_KERNEL_AP_2_AP) || defined(CONFIG_FUSIV_KERNEL_AP_2_AP_MODULE)
         void (*ap2apRouteFlowDelete_ptr)(void *) = NULL;
 #endif
+*/
 
 DEFINE_RWLOCK(ip_conntrack_lock);
 
@@ -315,6 +317,7 @@
 	DEBUGP("destroy_conntrack(%p)\n", ct);
 	IP_NF_ASSERT(atomic_read(&nfct->use) == 0);
 	IP_NF_ASSERT(!timer_pending(&ct->timeout));
+/*
 #if defined(CONFIG_FUSIV_KERNEL_AP_2_AP) || defined(CONFIG_FUSIV_KERNEL_AP_2_AP_MODULE)
         if(ap2apRouteFlowDelete_ptr != NULL)
         {
@@ -323,6 +326,7 @@
         else
                 printk("\ndestroy_conntrack: ap2ap_lkm not initialized properly. ..\n");
 #endif
+*/
 
 	ip_conntrack_event(IPCT_DESTROY, ct);
 	set_bit(IPS_DYING_BIT, &ct->status);
@@ -1556,6 +1560,8 @@
 
 	return -ENOMEM;
 }
+/*
 #if defined(CONFIG_FUSIV_KERNEL_AP_2_AP) || defined(CONFIG_FUSIV_KERNEL_AP_2_AP_MODULE)
         EXPORT_SYMBOL(ap2apRouteFlowDelete_ptr);
 #endif
+*/
