--- EXAMPLE.conf.def
+++ EXAMPLE.conf.def
@@ -6,7 +6,7 @@
 ###############################################################################
 #
 # This file is intended to only be an example.  If, however, you want
-# to use it, it should be placed in SYSCONFDIR/snmp/snmpd.conf.
+# to use it, it should be placed in /etc/snmp/snmpd.conf.
 # When the snmpd agent starts up, this is where it will look for it.
 #
 # You might be interested in generating your own snmpd.conf file using
@@ -58,31 +58,37 @@
 # from):
 
 #       sec.name  source          community
-com2sec local     localhost       COMMUNITY
-com2sec mynetwork NETWORK/24      COMMUNITY
+com2sec paranoid  default         public
+#com2sec readonly  default         public
+#com2sec readwrite default         private
 
 ####
 # Second, map the security names into group names:
 
 #             	sec.model  sec.name
-group MyRWGroup	v1         local
-group MyRWGroup	v2c        local
-group MyRWGroup	usm        local
-group MyROGroup v1         mynetwork
-group MyROGroup v2c        mynetwork
-group MyROGroup usm        mynetwork
+group MyROSystem v1        paranoid
+group MyROSystem v2c       paranoid
+group MyROSystem usm       paranoid
+group MyROGroup v1         readonly
+group MyROGroup v2c        readonly
+group MyROGroup usm        readonly
+group MyRWGroup v1         readwrite
+group MyRWGroup v2c        readwrite
+group MyRWGroup usm        readwrite
 
 ####
 # Third, create a view for us to let the groups have rights to:
 
 #           incl/excl subtree                          mask
 view all    included  .1                               80
+view system included  .iso.org.dod.internet.mgmt.mib-2.system
 
 ####
 # Finally, grant the 2 groups access to the 1 view with different
 # write permissions:
 
 #                context sec.model sec.level match  read   write  notif
+access MyROSystem ""     any       noauth    exact  system none   none
 access MyROGroup ""      any       noauth    exact  all    none   none
 access MyRWGroup ""      any       noauth    exact  all    all    none
 
@@ -100,8 +106,8 @@ access MyRWGroup ""      any       noaut
 # value of an object whose value is given here will fail with an error
 # status of notWritable.
 
-syslocation Right here, right now.
-syscontact Me <me@somewhere.org>
+syslocation Unknown (configure /etc/snmp/snmpd.local.conf)
+syscontact Root <root@localhost> (configure /etc/snmp/snmpd.local.conf)
 
 # Example output of snmpwalk:
 #   % snmpwalk -v 1 -c public localhost system
@@ -135,13 +141,13 @@ syscontact Me <me@somewhere.org>
 #
 
 #  Make sure mountd is running
-proc mountd
+#proc mountd
 
 #  Make sure there are no more than 4 ntalkds running, but 0 is ok too.
-proc ntalkd 4
+#proc ntalkd 4
 
 #  Make sure at least one sendmail, but less than or equal to 10 are running.
-proc sendmail 10 1
+#proc sendmail 10 1
 
 #  A snmpwalk of the prTable would look something like this:
 # 
@@ -199,7 +205,7 @@ proc sendmail 10 1
 #  ARGS:     optional arguments to be passed to the program
 
 # a simple hello world
-exec echotest /bin/echo hello world
+#exec echotest /bin/echo hello world
 
 # Run a shell script containing:
 #
@@ -250,7 +256,7 @@ exec echotest /bin/echo hello world
 
 # Check the / partition and make sure it contains at least 10 megs.
 
-disk / 10000
+#disk / 10000
 
 # % snmpwalk -v 1 -c public localhost .EXTENSIBLEDOTMIB.DISKMIBNUM
 # enterprises.ucdavis.diskTable.dskEntry.diskIndex.1 = 0
@@ -279,7 +285,7 @@ disk / 10000
 # 15MAX:  Similar, but for 15 min average.
 
 # Check for loads:
-load 12 14 14
+#load 12 14 14
 
 # % snmpwalk -v 1 -c public localhost .EXTENSIBLEDOTMIB.LOADAVEMIBNUM
 # enterprises.ucdavis.loadTable.laEntry.loadaveIndex.1 = 1
@@ -365,7 +371,7 @@ load 12 14 14
 #           (commented out here since it requires that you place the
 #           script in the right location. (its not installed by default))
 
-# pass .EXTENSIBLEDOTMIB.255 /bin/sh PREFIX/local/passtest
+# pass .EXTENSIBLEDOTMIB.255 /bin/sh /usr/local/passtest
 
 # % snmpwalk -v 1 -c public localhost .EXTENSIBLEDOTMIB.255
 # enterprises.ucdavis.255.1 = "life the universe and everything"
--- /dev/null
+++ EXAMPLE-trap.conf
@@ -0,0 +1,18 @@
+###############################################################################
+#
+# EXAMPLE-trap.conf:
+#   An example configuration file for configuring the Net-SNMP snmptrapd agent.
+#
+###############################################################################
+#
+# This file is intended to only be an example.  If, however, you want
+# to use it, it should be placed in /etc/snmp/snmptrapd.conf.
+# When the snmptrapd agent starts up, this is where it will look for it.
+#
+# All lines beginning with a '#' are comments and are intended for you
+# to read.  All other lines are configuration commands for the agent.
+
+#
+# PLEASE: read the snmptrapd.conf(5) manual page as well!
+#
+
