--- configure
+++ configure
@@ -36577,6 +36577,8 @@
     libmagic/is_tar.c libmagic/magic.c libmagic/print.c \
     libmagic/readcdf.c libmagic/softmagic.c"
 
+# strcasestr is avaliable in all versions of uClibc since 0.9.28, no need to check
+if false; then
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for strcasestr" >&5
 $as_echo_n "checking for strcasestr... " >&6; }
   if test "$cross_compiling" = yes; then :
@@ -36631,6 +36633,7 @@
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
 fi
+fi
 
 
 
--- ext/xmlrpc/libxmlrpc/xml_to_soap.c
+++ ext/xmlrpc/libxmlrpc/xml_to_soap.c
@@ -319,7 +319,7 @@
 			   are supposed to know the type of action based on state, which strikes me as a bit lame.
 			   Anyway, we don't have that state info, thus we use Response as a heuristic. */
 			rtype =
-#ifdef strcasestr
+#if 1 /* strcasestr is a function declared in a regular way and not as #define, it's available in all version of uClibc since 0.9.28 */
 			strcasestr(el->name, "response") ? xmlrpc_request_response : xmlrpc_request_call;
 #else
 			strstr(el->name, "esponse") ? xmlrpc_request_response : xmlrpc_request_call;
