--- configure
+++ configure
@@ -66551,6 +66551,11 @@
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: $msg" >&5
 $as_echo "$msg" >&6; }
 
+# evaluated by running test program on the target
+flock_type=linux
+$as_echo "#define HAVE_FLOCK_LINUX /**/" >>confdefs.h
+
+if false; then
 flock_type=unknown
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking \"whether flock struct is linux ordered\"" >&5
 $as_echo_n "checking \"whether flock struct is linux ordered\"... " >&6; }
@@ -66628,6 +66633,7 @@
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
 fi
+fi
 
 
 if test "$flock_type" = "unknown"; then
