commit d9f033aa43299daa610c37243c623eccdfb8db82
Author: Justin Maggard <jmaggard@netgear.com>
Date:   Sat May 20 00:16:33 2017 -0700

    clients: add Movian detection and enable subtitles

index d32e92d..e889e74 100644
--- clients.c
+++ clients.c
@@ -254,6 +254,13 @@ struct client_type_s client_types[] =
 	  EUserAgent
 	},
 
+	{ EMovian,
+	  FLAG_CAPTION_RES,
+	  "Movian",
+	  "Movian",
+	  EUserAgent
+	},
+
 	{ EKodi,
 	  FLAG_DLNA | FLAG_MIME_AVI_AVI | FLAG_CAPTION_RES,
 	  "Kodi",
index a393b72..f982ea5 100644
--- clients.h
+++ clients.h
@@ -80,6 +80,7 @@ enum client_types {
 	EBubbleUPnP,
 	ENetFrontLivingConnect,
 	EKodi,
+	EMovian,
 	EStandardDLNA150,
 	EStandardUPnP
 };
