--- Makefile.in
+++ Makefile.in
@@ -333,7 +333,7 @@
 	@LIBEXIF_LIBS@ \
 	@LIBINTL@ \
 	@LIBICONV@ \
-	-lFLAC  $(flacoggflag) $(vorbisflag)
+	-lFLAC $(vorbisflag) $(flacoggflag)
 
 minidlnad_LDFLAGS = @STATIC_LDFLAGS@
 testupnpdescgen_SOURCES = testupnpdescgen.c upnpdescgen.c
@@ -344,7 +344,7 @@
 	@LIBAVFORMAT_LIBS@ \
 	@LIBAVUTIL_LIBS@ \
 	@LIBEXIF_LIBS@ \
-	-lFLAC  $(flacoggflag) $(vorbisflag)
+	-lFLAC $(vorbisflag) $(flacoggflag)
 
 SUFFIXES = .tmpl .
 GENERATED_FILES = \
