--- Makefile.in
+++ Makefile.in
@@ -410,7 +410,7 @@
 	@LIBEXIF_LIBS@ \
 	@LIBINTL@ \
 	@LIBICONV@ \
-	-lFLAC $(flacogglibs) $(vorbislibs) $(avahilibs)
+	-lFLAC $(vorbislibs) $(flacogglibs) $(avahilibs)
 
 minidlnad_LDFLAGS = @STATIC_LDFLAGS@
 testupnpdescgen_SOURCES = testupnpdescgen.c upnpdescgen.c
@@ -421,7 +421,7 @@
 	@LIBAVFORMAT_LIBS@ \
 	@LIBAVUTIL_LIBS@ \
 	@LIBEXIF_LIBS@ \
-	-lFLAC $(flacogglibs) $(vorbislibs) $(avahilibs)
+	-lFLAC $(vorbislibs) $(flacogglibs) $(avahilibs)
 
 SUFFIXES = .tmpl .
 GENERATED_FILES = \
