--- genconfig.sh
+++ genconfig.sh
@@ -26,12 +26,12 @@
 # Database path
 DB_PATH="/tmp/minidlna"
 # Log path
-LOG_PATH="${DB_PATH}"
+LOG_PATH="/var/log"
 
 # detecting the OS name and version
-OS_NAME=`uname -s`
-OS_VERSION=`uname -r`
-TIVO="/*#define TIVO_SUPPORT*/"
+OS_NAME="FRITZ!Box"
+OS_URL="http://sourceforge.net/projects/minidlna/"
+TIVO="#define TIVO_SUPPORT"
 NETGEAR="/*#define NETGEAR*/"
 READYNAS="/*#define READYNAS*/"
 PNPX="#define PNPX 0"
@@ -40,22 +40,22 @@
 
 # Detect if there are missing headers
 # NOTE: This check only works with a normal distro
-[ ! -e "/usr/include/sqlite3.h" ] && MISSING="libsqlite3 $MISSING"
-[ ! -e "/usr/include/jpeglib.h" ] && MISSING="libjpeg $MISSING"
-[ ! -e "/usr/include/libexif/exif-loader.h" ] && MISSING="libexif $MISSING"
-[ ! -e "/usr/include/id3tag.h" ] && MISSING="libid3tag $MISSING"
-[ ! -e "/usr/include/ogg/ogg.h" ] && MISSING="libogg $MISSING"
-[ ! -e "/usr/include/vorbis/codec.h" ] && MISSING="libvorbis $MISSING"
-[ ! -e "/usr/include/FLAC/metadata.h" ] && MISSING="libflac $MISSING"
-[ ! -e "/usr/include/ffmpeg/avutil.h" -a \
-  ! -e "/usr/include/libavutil/avutil.h" -a \
-  ! -e "/usr/include/ffmpeg/libavutil/avutil.h" ] && MISSING="libavutil $MISSING"
-[ ! -e "/usr/include/ffmpeg/avformat.h" -a \
-  ! -e "/usr/include/libavformat/avformat.h" -a \
-  ! -e "/usr/include/ffmpeg/libavformat/avformat.h" ] && MISSING="libavformat $MISSING"
-[ ! -e "/usr/include/ffmpeg/avcodec.h" -a \
-  ! -e "/usr/include/libavcodec/avcodec.h" -a \
-  ! -e "/usr/include/ffmpeg/libavcodec/avcodec.h" ] && MISSING="libavcodec $MISSING"
+[ ! -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/sqlite3.h" ] && MISSING="libsqlite3 $MISSING"
+[ ! -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/jpeglib.h" ] && MISSING="libjpeg $MISSING"
+[ ! -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/libexif/exif-loader.h" ] && MISSING="libexif $MISSING"
+[ ! -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/id3tag.h" ] && MISSING="libid3tag $MISSING"
+[ ! -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/ogg/ogg.h" ] && MISSING="libogg $MISSING"
+[ ! -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/vorbis/codec.h" ] && MISSING="libvorbis $MISSING"
+[ ! -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/FLAC/metadata.h" ] && MISSING="libflac $MISSING"
+[ ! -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/ffmpeg/avutil.h" -a \
+  ! -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/libavutil/avutil.h" -a \
+  ! -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/ffmpeg/libavutil/avutil.h" ] && MISSING="libavutil $MISSING"
+[ ! -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/ffmpeg/avformat.h" -a \
+  ! -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/libavformat/avformat.h" -a \
+  ! -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/ffmpeg/libavformat/avformat.h" ] && MISSING="libavformat $MISSING"
+[ ! -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/ffmpeg/avcodec.h" -a \
+  ! -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/libavcodec/avcodec.h" -a \
+  ! -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/ffmpeg/libavcodec/avcodec.h" ] && MISSING="libavcodec $MISSING"
 if [ -n "$MISSING" ]; then
 	echo -e "\nERROR!  Cannot continue."
 	echo -e "The following required libraries are either missing, or are missing development headers:\n"
@@ -72,6 +72,7 @@
 echo "" >> ${CONFIGFILE}
 
 # OS Specific stuff
+if false; then
 case $OS_NAME in
 	OpenBSD)
 		MAJORVER=`echo $OS_VERSION | cut -d. -f1`
@@ -155,49 +156,49 @@
 		exit 1
 		;;
 esac
+fi
 
 echo "#define OS_NAME			\"$OS_NAME\"" >> ${CONFIGFILE}
-echo "#define OS_VERSION		\"$OS_NAME/$OS_VERSION\"" >> ${CONFIGFILE}
 echo "#define OS_URL			\"${OS_URL}\"" >> ${CONFIGFILE}
 echo "" >> ${CONFIGFILE}
 
-if [ -e "/usr/include/libavutil/avutil.h" ]; then
+if [ -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/libavutil/avutil.h" ]; then
 	echo "#define HAVE_LIBAVUTIL_AVUTIL_H 1" >> ${CONFIGFILE}
-elif [ -e "/usr/include/ffmpeg/avutil.h" ]; then
+elif [ -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/ffmpeg/avutil.h" ]; then
 	echo "#define HAVE_FFMPEG_AVUTIL_H 1" >> ${CONFIGFILE}
-elif [ -e "/usr/include/ffmpeg/libavutil/avutil.h" ]; then
+elif [ -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/ffmpeg/libavutil/avutil.h" ]; then
 	echo "#define HAVE_FFMPEG_LIBAVUTIL_AVUTIL_H 1" >> ${CONFIGFILE}
-elif [ -e "/usr/include/libav/libavutil/avutil.h" ]; then
+elif [ -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/libav/libavutil/avutil.h" ]; then
 	echo "#define HAVE_LIBAV_LIBAVUTIL_AVUTIL_H 1" >> ${CONFIGFILE}
-elif [ -e "/usr/include/libav/avutil.h" ]; then
+elif [ -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/libav/avutil.h" ]; then
 	echo "#define HAVE_LIBAV_AVUTIL_H 1" >> ${CONFIGFILE}
-elif [ -e "/usr/include/avutil.h" ]; then
+elif [ -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/avutil.h" ]; then
 	echo "#define HAVE_AVUTIL_H 1" >> ${CONFIGFILE}
 fi
-if [ -e "/usr/include/libavformat/avformat.h" ]; then
+if [ -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/libavformat/avformat.h" ]; then
 	echo "#define HAVE_LIBAVFORMAT_AVFORMAT_H 1" >> ${CONFIGFILE}
-elif [ -e "/usr/include/ffmpeg/avformat.h" ]; then
+elif [ -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/ffmpeg/avformat.h" ]; then
 	echo "#define HAVE_FFMPEG_AVFORMAT_H 1" >> ${CONFIGFILE}
-elif [ -e "/usr/include/ffmpeg/libavformat/avformat.h" ]; then
+elif [ -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/ffmpeg/libavformat/avformat.h" ]; then
 	echo "#define HAVE_FFMPEG_LIBAVFORMAT_AVFORMAT_H 1" >> ${CONFIGFILE}
-elif [ -e "/usr/include/libav/libavformat/avformat.h" ]; then
+elif [ -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/libav/libavformat/avformat.h" ]; then
 	echo "#define HAVE_LIBAV_LIBAVFORMAT_AVFORMAT_H 1" >> ${CONFIGFILE}
-elif [ -e "/usr/include/libav/avformat.h" ]; then
+elif [ -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/libav/avformat.h" ]; then
 	echo "#define HAVE_LIBAV_AVFORMAT_H 1" >> ${CONFIGFILE}
-elif [ -e "/usr/include/avformat.h" ]; then
+elif [ -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/avformat.h" ]; then
 	echo "#define HAVE_AVFORMAT_H 1" >> ${CONFIGFILE}
 fi
-if [ -e "/usr/include/libavcodec/avcodec.h" ]; then
+if [ -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/libavcodec/avcodec.h" ]; then
 	echo "#define HAVE_LIBAVCODEC_AVCODEC_H 1" >> ${CONFIGFILE}
-elif [ -e "/usr/include/ffmpeg/avcodec.h" ]; then
+elif [ -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/ffmpeg/avcodec.h" ]; then
 	echo "#define HAVE_FFMPEG_AVCODEC_H 1" >> ${CONFIGFILE}
-elif [ -e "/usr/include/ffmpeg/libavcodec/avcodec.h" ]; then
+elif [ -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/ffmpeg/libavcodec/avcodec.h" ]; then
 	echo "#define HAVE_FFMPEG_LIBAVCODEC_AVCODEC_H 1" >> ${CONFIGFILE}
-elif [ -e "/usr/include/libav/libavcodec/avcodec" ]; then
+elif [ -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/libav/libavcodec/avcodec" ]; then
 	echo "#define HAVE_LIBAV_LIBAVCODEC_AVCODEC_H 1" >> ${CONFIGFILE}
-elif [ -e "/usr/include/libav/avcodec.h" ]; then
+elif [ -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/libav/avcodec.h" ]; then
 	echo "#define HAVE_LIBAV_AVCODEC_H 1" >> ${CONFIGFILE}
-elif [ -e "/usr/include/avcodec.h" ]; then
+elif [ -e "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/avcodec.h" ]; then
 	echo "#define HAVE_AVCODEC_H 1" >> ${CONFIGFILE}
 fi
 echo >> ${CONFIGFILE}
@@ -216,7 +217,7 @@
 echo "" >> ${CONFIGFILE}
 
 echo "/* Enable if the system inotify.h exists.  Otherwise our own inotify.h will be used. */" >> ${CONFIGFILE}
-if [ -f /usr/include/sys/inotify.h ]; then
+if [ -f "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/sys/inotify.h" ]; then
 echo "#define HAVE_INOTIFY_H" >> ${CONFIGFILE}
 else
 echo "/*#define HAVE_INOTIFY_H*/" >> ${CONFIGFILE}
@@ -224,7 +225,7 @@
 echo "" >> ${CONFIGFILE}
 
 echo "/* Enable if the system iconv.h exists.  ID3 tag reading in various character sets will not work properly otherwise. */" >> ${CONFIGFILE}
-if [ -f /usr/include/iconv.h ]; then
+if [ -f "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/iconv.h" ]; then
 echo "#define HAVE_ICONV_H" >> ${CONFIGFILE}
 else
 echo -e "\nWARNING!!  Iconv support not found.  ID3 tag reading may not work."
@@ -233,11 +234,11 @@
 echo "" >> ${CONFIGFILE}
 
 echo "/* Enable if the system libintl.h exists for NLS support. */" >> ${CONFIGFILE}
-if [ -f /usr/include/libintl.h ]; then
-echo "#define ENABLE_NLS" >> ${CONFIGFILE}
-else
+#if [ -f "${CROSS_TOOLCHAIN_SYSROOT}/usr/include/libintl.h" ]; then
+#echo "#define ENABLE_NLS" >> ${CONFIGFILE}
+#else
 echo "/*#define ENABLE_NLS*/" >> ${CONFIGFILE}
-fi
+#fi
 echo "" >> ${CONFIGFILE}
 
 echo "/* Enable NETGEAR-specific tweaks. */" >> ${CONFIGFILE}
--- Makefile
+++ Makefile
@@ -10,11 +10,9 @@
 # or :
 # $ make install
 #
-#CFLAGS = -Wall -O -D_GNU_SOURCE -g -DDEBUG
-#CFLAGS = -Wall -g -Os -D_GNU_SOURCE
-CFLAGS = -Wall -g -O3 -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64
-#STATIC_LINKING: CFLAGS += -DSTATIC
-#STATIC_LINKING: LDFLAGS = -static
+override CFLAGS   += -Wall -pthread -ffunction-sections -fdata-sections
+override CPPFLAGS += -D_GNU_SOURCE
+LDFLAGS  += -Wl,--gc-sections
 CC = gcc
 RM = rm -f
 INSTALL = install
@@ -32,16 +30,19 @@
            tagutils/textutils.o tagutils/misc.o tagutils/tagutils.o \
            playlist.o image_utils.o albumart.o log.o
 
-ALLOBJS = $(BASEOBJS) $(LNXOBJS)
+ALLOBJS = $(BASEOBJS)
 
-LIBS = -lpthread -lexif -ljpeg -lsqlite3 -lavformat -lavutil -lavcodec -lid3tag -lFLAC -logg -lvorbis
-#STATIC_LINKING: LIBS = -lvorbis -logg -lm -lsqlite3 -lpthread -lexif -ljpeg -lFLAC -lm -lid3tag -lz -lavformat -lavutil -lavcodec -lm
+LIBS = -lavformat -lavcodec -lavutil -lexif -lFLAC $(ICONV_LIB) -lid3tag -ljpeg -lsqlite3 -lvorbis -logg -lpthread
+ifeq ($(strip $(LINK_STATICALLY)),yes)
+LDFLAGS += -static
+LIBS += $(FFMPEG_EXTRA_LIBS) -lz -lm -ldl
+endif
 
 TESTUPNPDESCGENOBJS = testupnpdescgen.o upnpdescgen.o
 
 EXECUTABLES = minidlna testupnpdescgen
 
-.PHONY:	all clean distclean install depend
+.PHONY:	all clean distclean install
 
 all:	$(EXECUTABLES)
 
@@ -53,7 +54,7 @@
 distclean: clean
 	$(RM) config.h
 
-install:	minidlna
+install: minidlna
 	$(INSTALL) -d $(SBININSTALLDIR)
 	$(INSTALL) minidlna $(SBININSTALLDIR)
 
@@ -61,21 +62,11 @@
 	$(INSTALL) -d $(ETCINSTALLDIR)
 	$(INSTALL) --mode=0644 minidlna.conf $(ETCINSTALLDIR)
 
-minidlna:	$(BASEOBJS) $(LNXOBJS) $(LIBS)
-	@echo Linking $@
-	@$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(BASEOBJS) $(LNXOBJS) $(LIBS)
+minidlna: $(BASEOBJS)
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) -o $@ $(BASEOBJS) $(LIBS)
 
-
-testupnpdescgen:	$(TESTUPNPDESCGENOBJS)
-	@echo Linking $@
-	@$(CC) $(CFLAGS) -o $@ $(TESTUPNPDESCGENOBJS)
-
-config.h:	genconfig.sh
-	./genconfig.sh
-
-depend:	config.h
-	makedepend -f$(MAKEFILE_LIST) -Y \
-	$(ALLOBJS:.o=.c) $(TESTUPNPDESCGENOBJS:.o=.c) 2>/dev/null
+testupnpdescgen: $(TESTUPNPDESCGENOBJS)
+	$(CC) $(CFLAGS) $(CPPFLAGS) -o $@ $(TESTUPNPDESCGENOBJS)
 
 # DO NOT DELETE
 
@@ -125,6 +116,4 @@
 .SUFFIXES: .c .o
 
 .c.o:
-	@echo Compiling $*.c
-	@$(CC) $(CFLAGS) -o $@ -c $< && exit 0; \
-		echo "The following command failed:  $(CC) $(CFLAGS) -o $@ -c $<" && false
+	$(CC) $(CFLAGS) $(CPPFLAGS) -o $@ -c $<
--- upnphttp.h
+++ upnphttp.h
@@ -56,7 +56,7 @@
 #include "config.h"
 
 /* server: HTTP header returned in all HTTP responses : */
-#define MINIDLNA_SERVER_STRING	OS_VERSION " DLNADOC/1.50 UPnP/1.0 " SERVER_NAME "/" MINIDLNA_VERSION
+#define MINIDLNA_SERVER_STRING	"DLNADOC/1.50 UPnP/1.0 " SERVER_NAME "/" MINIDLNA_VERSION
 
 /*
  states :
