--- 2.6.32/platform/linux/tarball-tag/sal/io.c
+++ 2.6.32/platform/linux/tarball-tag/sal/io.c
@@ -41,7 +41,8 @@
 #include <linux/fs.h>
 #include <asm/uaccess.h>
 #include <linux/ide.h>
-#include <linux/smp_lock.h>
+#include <linux/mutex.h>
+
 #include <linux/time.h>
 
 #include "sal/sync.h"
@@ -54,6 +55,8 @@
 
 #include "linux_ver.h" // SAL_HZ
 
+DEFINE_MUTEX(fs_mutex);
+
 NDAS_SAL_API sal_file sal_file_open(const char * filename, int flags, int mode)
 {
 	int linux_flags = 0;
@@ -95,7 +98,7 @@
 	int size32;
 #endif
 
-	lock_kernel();
+	mutex_lock(&fs_mutex);
 
         if (filp->f_dentry &&
 		filp->f_dentry->d_inode &&
@@ -158,7 +161,7 @@
 	} else {
 		ret = FALSE;
 	}
-	unlock_kernel();
+	mutex_unlock(&fs_mutex);
 	return ret;
 }
 
