--- libexslt/Makefile.in
+++ libexslt/Makefile.in
@@ -763,7 +763,7 @@
 
 info-am:
 
-install-data-am: install-exsltincHEADERS install-man \
+install-data-am: install-exsltincHEADERS             \
 	install-nodist_exsltincHEADERS
 
 install-dvi: install-dvi-am
--- libxslt/Makefile.in
+++ libxslt/Makefile.in
@@ -812,7 +812,7 @@
 
 info-am:
 
-install-data-am: install-man install-nodist_xsltincHEADERS \
+install-data-am:             install-nodist_xsltincHEADERS \
 	install-xsltincHEADERS
 
 install-dvi: install-dvi-am
--- Makefile.in
+++ Makefile.in
@@ -415,12 +415,9 @@
 SUBDIRS = \
 	libxslt \
 	libexslt \
-	xsltproc \
-	doc \
-	$(PYTHON_SUBDIR) \
-	tests
+	xsltproc
 
-DIST_SUBDIRS = libxslt libexslt xsltproc python doc tests
+DIST_SUBDIRS = libxslt libexslt xsltproc
 confexecdir = $(libdir)
 confexec_DATA = xsltConf.sh
 bin_SCRIPTS = xslt-config
@@ -963,13 +960,13 @@
 
 info-am:
 
-install-data-am: install-m4dataDATA install-pkgconfigDATA
+install-data-am:                    install-pkgconfigDATA
 
 install-dvi: install-dvi-recursive
 
 install-dvi-am:
 
-install-exec-am: install-binSCRIPTS install-confexecDATA
+install-exec-am: install-binSCRIPTS
 
 install-html: install-html-recursive
 
