--- src/auto/configure
+++ src/auto/configure
@@ -10823,6 +10823,7 @@
 fi
 rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
 
+if false; then
 ac_fn_c_check_header_mongrel "$LINENO" "elf.h" "ac_cv_header_elf_h" "$ac_includes_default"
 if test "x$ac_cv_header_elf_h" = xyes; then :
   HAS_ELF=1
@@ -10870,6 +10871,7 @@
 fi
 
 fi
+fi
 
 ac_header_dirent=no
 for ac_hdr in dirent.h sys/ndir.h sys/dir.h ndir.h; do
