--- src/auto/configure
+++ src/auto/configure
@@ -13039,6 +13039,7 @@
 fi
 rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
 
+if false; then
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for timer_create" >&5
 $as_echo_n "checking for timer_create... " >&6; }
 save_LIBS="$LIBS"
@@ -13125,6 +13126,7 @@
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
 fi
+fi
 
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether stat() ignores a trailing slash" >&5
