--- Rules.mak
+++ Rules.mak
@@ -341,8 +341,10 @@
 
 # Some nice CPU specific optimizations
 ifeq ($(TARGET_ARCH),i386)
-$(eval $(call check-gcc-var,-fomit-frame-pointer))
-	OPTIMIZATION += $(CFLAG_-fomit-frame-pointer)
+
+# DDO: we need frame-pointers for backtraces
+#$(eval $(call check-gcc-var,-fomit-frame-pointer))
+#	OPTIMIZATION += $(CFLAG_-fomit-frame-pointer)
 
 ifeq ($(CONFIG_386)$(CONFIG_486)$(CONFIG_586),y)
 	# TODO: Change this to a gcc version check.  This bug
