--- usr/www/all/html/de/home/home.js.orig	2009-08-28 11:18:44.000000000 +0000
+++ usr/www/all/html/de/home/home.js	2009-08-31 19:29:11.000000000 +0000
@@ -759,7 +759,7 @@
 var nPartTotal = Number(g_usbhost_data.part_count_total);
 if (isNaN(nPartTotal)) nPartTotal = 0;
 var unplugOk = parseInt(g_usbhost_data.unplug_ok, 10) == 1;
-if (nPartTotal > nStor)
+if (nPartTotal > 0)
 {
 if (unplugOk)
 {
@@ -767,9 +767,7 @@
 }
 else
 {
-strStor = "{?de.home.home.js:5308?} ";
-strStor += "<a href=\"javascript:jslGoTo('home', 'syslog');\">{?de.home.home.js:3744?}</a>";
-strStor = strStor.fontcolor("#cc0000");
+strStor = "Es wurden " + nStor + " von " + nPartTotal + " Partitionen eingebunden.";
 }
 strArr.push(strStor);
 }
