--- Makefile
+++ Makefile
@@ -151,6 +151,8 @@
 	PLATFORM := Linux
 	C_CXX_FLAGS += -DTC_UNIX -DTC_LINUX
 
+	C_CXX_FLAGS += -pthread
+
 	ifeq "$(TC_BUILD_CONFIG)" "Release"
 		C_CXX_FLAGS += -fdata-sections -ffunction-sections
 		LFLAGS += -Wl,--gc-sections
--- Main/Main.make
+++ Main/Main.make
@@ -92,7 +92,7 @@
 
 $(APPNAME): $(LIBS) $(OBJS)
 	@echo Linking $@
-	$(CXX) -o $(APPNAME) $(LFLAGS) $(OBJS) $(LIBS) $(FUSE_LIBS) $(WX_LIBS)
+	$(CXX) -o $(APPNAME) $(LFLAGS) $(OBJS) $(LIBS) $(FUSE_LIBS) $(WX_LIBS) -lpthread
 
 ifeq "$(TC_BUILD_CONFIG)" "Release"
 ifndef NOSTRIP
