--- Makefile
+++ Makefile
@@ -37,7 +37,7 @@
 
 export CFLAGS := -Wall
 export CXXFLAGS := -Wall -Wno-unused-parameter
-C_CXX_FLAGS := -MMD -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGE_FILES -I$(BASE_DIR) -I$(BASE_DIR)/Crypto
+C_CXX_FLAGS := -MMD $(LFS_FLAGS) -I$(BASE_DIR) -I$(BASE_DIR)/Crypto
 export ASFLAGS := -Ox -D __GNUC__
 export LFLAGS :=
 
@@ -87,7 +87,7 @@
 
 ifeq "$(TC_BUILD_CONFIG)" "Release"
 
-	C_CXX_FLAGS += -O2 -fno-strict-aliasing  # Do not enable strict aliasing
+	C_CXX_FLAGS += -fno-strict-aliasing  # Do not enable strict aliasing
 	export WX_BUILD_DIR ?= $(BASE_DIR)/wxrelease
 	WX_CONFIGURE_FLAGS += --disable-debug_flag --disable-debug_gdb --disable-debug_info
 
@@ -155,7 +155,7 @@
 		C_CXX_FLAGS += -fdata-sections -ffunction-sections
 		LFLAGS += -Wl,--gc-sections
 
-		ifneq "$(shell ld --help 2>&1 | grep sysv | wc -l)" "0"
+		ifneq "$(shell $(LD) --help 2>&1 | grep sysv | wc -l)" "0"
 			LFLAGS += -Wl,--hash-style=sysv
 		endif
 
