--- Makefile
+++ Makefile
@@ -152,15 +152,19 @@
 	C_CXX_FLAGS += -DTC_UNIX -DTC_LINUX
 
 	ifeq "$(TC_BUILD_CONFIG)" "Release"
-		C_CXX_FLAGS += -fdata-sections -ffunction-sections
-		LFLAGS += -Wl,--gc-sections
+#		C_CXX_FLAGS += -fdata-sections -ffunction-sections
+#		LFLAGS += -Wl,--gc-sections
 
-		ifneq "$(shell $(LD) --help 2>&1 | grep sysv | wc -l)" "0"
-			LFLAGS += -Wl,--hash-style=sysv
-		endif
-
-		WXCONFIG_CFLAGS += -fdata-sections -ffunction-sections
-		WXCONFIG_CXXFLAGS += -fdata-sections -ffunction-sections
+#		ifneq "$(shell $(LD) --help 2>&1 | grep sysv | wc -l)" "0"
+#			LFLAGS += -Wl,--hash-style=sysv
+#		endif
+
+#		WXCONFIG_CFLAGS += -fdata-sections -ffunction-sections
+#		WXCONFIG_CXXFLAGS += -fdata-sections -ffunction-sections
+
+# truecrypt depends on pthread, don't assume fuse or wxWidgets depends on it too, add required flags explicitly
+		C_CXX_FLAGS += -pthread
+		LFLAGS += -lpthread
 	endif
 
 endif
