--- autoconf/Makefile.in
+++ autoconf/Makefile.in
@@ -51,12 +51,12 @@
 	@exit 1
 static: makefiles slang.pc
 	cd src; $(MAKE) static
-	cd slsh; $(MAKE) static
+#	cd slsh; $(MAKE) static
 modules:
 	cd modules; $(MAKE) all
 elf: makefiles slang.pc
 	cd src; $(MAKE) elf
-	cd slsh; $(MAKE) all
+#	cd slsh; $(MAKE) all
 	cd modules; $(MAKE) all
 	@echo Use make install to install it.
 runtests: all
@@ -77,14 +77,14 @@
 	cd demo; $(MAKE) clean
 install-static: install-pkgconfig
 	cd src; $(MAKE) install-static
-	cd slsh; $(MAKE) install-static
+#	cd slsh; $(MAKE) install-static
 install-pkgconfig: slang.pc
 	$(MKINSDIR) $(DEST_PKGCONFIGDIR)
 	$(INSTALL_DATA) slang.pc $(DEST_PKGCONFIGDIR)/
 install-elf: install-pkgconfig
 	cd src; $(MAKE) install-elf
-	@echo Now installing slsh
-	cd slsh; $(MAKE) install
+#	@echo Now installing slsh
+#	cd slsh; $(MAKE) install
 	@echo Now installing the modules
 	cd modules; $(MAKE) install
 	@echo ""
