--- modules/Makefile.in
+++ modules/Makefile.in
@@ -85,7 +85,7 @@
 DEST_CMAPSDIR	= $(DESTDIR)$(CMAPS_DIR)
 DEST_HELPDIR	= $(DESTDIR)$(HELP_DIR)
 # ---------------------------------------------------------------------------
-RPATH = @RPATH@
+#RPATH = @RPATH@
 
 LIBS = $(LDFLAGS) $(RPATH) $(SLANG_LIB) $(DL_LIB) @M_LIB@
 INCS = $(SLANG_SRCINC)
--- slsh/Makefile.in
+++ slsh/Makefile.in
@@ -23,7 +23,7 @@
 #---------------------------------------------------------------------------
 DYNAMIC_LIBS	= @TERMCAP@ @DYNAMIC_LINK_LIB@ @LIBS@ @M_LIB@
 STATIC_LIBS	= @TERMCAP@ @LIBS@ @M_LIB@
-RPATH		= @RPATH@
+#RPATH		= @RPATH@
 #----------------------------------------------------------------------------
 INSTALL		= @INSTALL@
 INSTALL_DATA	= @INSTALL_DATA@
