--- modules/Makefile.in
+++ modules/Makefile.in
@@ -75,7 +75,7 @@
 DEST_CMAPSDIR	= $(DESTDIR)$(CMAPS_DIR)
 DEST_HELPDIR	= $(DESTDIR)$(HELP_DIR)
 #---------------------------------------------------------------------------
-RPATH = @RPATH@
+#RPATH = @RPATH@
 
 LIBS = $(LDFLAGS) $(RPATH) $(SLANG_LIB) $(DL_LIB) @M_LIB@
 INCS = $(SLANG_SRCINC)
--- slsh/Makefile.in
+++ slsh/Makefile.in
@@ -19,7 +19,7 @@
 SLANG_SRCLIB	= -L@ELFDIR@
 #---------------------------------------------------------------------------
 OTHER_LIBS	= @TERMCAP@ @DYNAMIC_LINK_LIB@ @LIBS@ @M_LIB@
-RPATH		= @RPATH@
+#RPATH		= @RPATH@
 #----------------------------------------------------------------------------
 INSTALL		= @INSTALL@
 INSTALL_DATA	= @INSTALL_DATA@
