--- usr/www/all/html/de/home/home.js.orig	2006-11-12 10:40:37.000000000 +0100
+++ usr/www/all/html/de/home/home.js	2006-11-12 10:53:55.000000000 +0100
@@ -62,7 +62,7 @@
 	SetSpanText("Status$1", uiSipState("$1"));
 ' ?>
 	if (!showFon) jslDisplay("FonAll", true);
-	var signed = "<? query box:status/signed_firmware ?>";
+	var signed = "1";
 	if (g_showEventlink != 0) jslDisplay("uiViewEventlink", true);
 	jslDisplay("uiNotSigned", signed == "0");
 	jslZebrastreifen("tAnsch", "#F8F8F8", "#FFFFFF");
