--- Makefile.in
+++ Makefile.in
@@ -3029,11 +3029,10 @@
 
 info-am:
 
-install-data-am: install-dist_docDATA install-dist_htmlDATA \
-	install-htmlDATA install-includeHEADERS install-man \
+install-data-am: \
+	install-includeHEADERS \
 	install-nodist_includeHEADERS install-pkgconfigDATA
 	@$(NORMAL_INSTALL)
-	$(MAKE) $(AM_MAKEFLAGS) install-data-hook
 install-dvi: install-dvi-am
 
 install-dvi-am:
