--- pcre-config.in
+++ pcre-config.in
@@ -3,6 +3,8 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 exec_prefix_set=no
+includedir=@includedir@
+libdir=@libdir@
 
 if test @enable_cpp@ = yes ; then
   usage="Usage: pcre-config [--prefix] [--exec-prefix] [--version] [--libs] [--libs-posix] [--libs-cpp] [--cflags] [--cflags-posix]"
@@ -18,16 +20,16 @@
 libR=
 case `uname -s` in
   *SunOS*)
-  libR=" -R@libdir@"
+  libR=" -R${libdir}"
   ;;
   *BSD*)
-  libR=" -Wl,-R@libdir@"
+  libR=" -Wl,-R${libdir}"
   ;;
 esac
 
 libS=
-if test @libdir@ != /usr/lib ; then
-  libS=-L@libdir@
+if test ${libdir} != /usr/lib ; then
+  libS=-L${libdir}
 fi
 
 while test $# -gt 0; do
@@ -57,8 +59,8 @@
       echo @PACKAGE_VERSION@
       ;;
     --cflags | --cflags-posix)
-      if test @includedir@ != /usr/include ; then
-        includes=-I@includedir@
+      if test ${includedir} != /usr/include ; then
+        includes=-I${includedir}
       fi
       echo $includes @PCRE_STATIC_CFLAG@
       ;;
