--- configure	2016-12-15 20:22:23.195976902 +0100
+++ configure	2016-12-15 20:28:05.605998610 +0100
@@ -15217,7 +15217,7 @@
 
 done
 
-for ac_header in execinfo.h paths.h pwd.h grp.h uuid/uuid.h
+for ac_header in paths.h pwd.h grp.h uuid/uuid.h
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
 ac_fn_c_check_header_mongrel "$LINENO" "$ac_header" "$as_ac_Header" "$ac_includes_default"
@@ -15971,6 +15971,7 @@
   return 0;
 }
 _ACEOF
+if false; then
 for ac_lib in '' execinfo; do
   if test -z "$ac_lib"; then
     ac_res="none required"
@@ -15987,6 +15988,7 @@
   break
 fi
 done
+fi
 if ${ac_cv_search_backtrace+:} false; then :
 
 else
