--- src/mklibs.orig	2010-09-20 22:33:22.000000000 +0200
+++ src/mklibs	2010-09-21 15:03:28.604621874 +0200
@@ -421,8 +421,8 @@
                     if debuglevel >= DEBUG_VERBOSE:
                         print "Adding rpath " + rpath_elem + " for " + obj
                     lib_rpath.append(rpath_elem)
-        else:
-            print "warning: " + obj + " may need rpath, but --root not specified"
+#        else:
+#            print "warning: " + obj + " may need rpath, but --root not specified"
 
 lib_path.extend(lib_rpath)
 
