--- src/mklibs.orig	2012-05-17 12:34:02.000000000 +0200
+++ src/mklibs	2012-05-21 20:23:40.225139359 +0200
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2
 
 # mklibs.py: An automated way to create a minimal /lib/ directory.
 #
--- src/mklibs-copy.orig	2010-06-02 19:47:32.000000000 +0200
+++ src/mklibs-copy	2012-05-21 20:23:56.661138626 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # mklibs.py: An automated way to create a minimal /lib/ directory.
 #
