--- C/7zip/Compress/LZMA/LZMADecoder.cpp.orig	2005-09-22 10:07:26.000000000 +0200
+++ C/7zip/Compress/LZMA/LZMADecoder.cpp	2007-07-28 12:36:08.000000000 +0200
@@ -274,12 +274,17 @@
   Byte remainder = (Byte)(properties[0] / 9);
   int lp = remainder % 5;
   int pb = remainder / 5;
-  if (pb > NLength::kNumPosStatesBitsMax)
-    return E_INVALIDARG;
-  _posStateMask = (1 << pb) - 1;
   UInt32 dictionarySize = 0;
   for (int i = 0; i < 4; i++)
     dictionarySize += ((UInt32)(properties[1 + i])) << (i * 8);
+  return SetDecoderPropertiesRaw(lc, lp, pb, dictionarySize);
+}
+
+STDMETHODIMP CDecoder::SetDecoderPropertiesRaw(int lc, int lp, int pb, UInt32 dictionarySize)
+{
+  if (pb > NLength::kNumPosStatesBitsMax)
+    return E_INVALIDARG;
+  _posStateMask = (1 << pb) - 1;
   if (!_outWindowStream.Create(dictionarySize))
     return E_OUTOFMEMORY;
   if (!_literalDecoder.Create(lp, lc))
--- C/7zip/Compress/LZMA/LZMADecoder.h.orig	2006-05-08 16:50:54.000000000 +0200
+++ C/7zip/Compress/LZMA/LZMADecoder.h	2007-07-28 12:36:08.000000000 +0200
@@ -231,6 +231,7 @@
       ICompressProgressInfo *progress);
 
   STDMETHOD(SetDecoderProperties2)(const Byte *data, UInt32 size);
+  STDMETHOD(SetDecoderPropertiesRaw)(int lc, int lp, int pb, UInt32 dictionarySize);
 
   STDMETHOD(GetInStreamProcessedSize)(UInt64 *value);
 
