--- Makefile.am.orig	2008-08-04 00:26:49.000000000 +0200
+++ Makefile.am	2008-11-22 13:46:00.000000000 +0100
@@ -1,5 +1,5 @@
 AUTOMAKE_OPTIONS=foreign
-SUBDIRS=scripts doc test
+SUBDIRS=scripts test
 
 noinst_LTLIBRARIES = libcommunicate.la
 libcommunicate_la_SOURCES = communicate.c
--- Makefile.in.orig	2008-10-04 20:49:19.000000000 +0200
+++ Makefile.in	2008-11-22 13:46:00.000000000 +0100
@@ -237,7 +237,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = foreign
-SUBDIRS = scripts doc test
+SUBDIRS = scripts test
 noinst_LTLIBRARIES = libcommunicate.la
 libcommunicate_la_SOURCES = communicate.c
 lib_LTLIBRARIES = libfakeroot.la
