--- faked.c	2007-05-10 16:55:56.000000000 +0200
+++ faked.c	2005-08-22 01:49:52.000000000 +0200
@@ -464,22 +464,6 @@
     roots = "/";
 #endif
 
-  do {
-    int r,fd=0;
-    struct stat s;
-    r=stat(save_file,&s);
-    if (r<0) return 0;
-    if (!(s.st_mode&S_IFIFO)) break;
-    fd=open(save_file,O_WRONLY|O_NONBLOCK);
-    if (fd<0) {
-      sleep(1);
-      continue;
-    }
-    close(fd);
-    break;
-  } while (1);
-
-
   f=fopen(save_file, "w");
   if(!f)
     return 0;
