--- configure	2007-10-17 19:35:11.000000000 +0200
+++ configure	2007-10-17 19:36:13.000000000 +0200
@@ -12550,7 +12550,7 @@
 
 { echo "$as_me:$LINENO: checking for a working shell..." >&5
 echo $ECHO_N "checking for a working shell...... $ECHO_C" >&6; }
-for i in /usr/xpg6/bin/sh /usr/xpg4/bin/sh /usr/ccs/bin/sh /bin/sh /usr/bin/sh /bin/ksh /bin/bash /usr/local/bin/bash ; do
+for i in /bin/bash /usr/local/bin/bash /usr/xpg6/bin/sh /usr/xpg4/bin/sh /usr/ccs/bin/sh /bin/sh /usr/bin/sh /bin/ksh ; do
     $i -c 'if ! false ; then echo $(echo ok) ; fi' >config.$$ 2>/dev/null
     j=`cat config.$$`
     rm -f config.$$
