diff -burN squashfs-tools/mksquashfs.c squashfs-tools/mksquashfs.c
--- squashfs-tools/mksquashfs.c	2005-09-09 00:34:28.000000000 +0200
+++ squashfs-tools/mksquashfs.c	2007-05-13 18:54:46.000000000 +0200
@@ -1780,14 +1780,6 @@
 		}
 	}
 
-
-	for(i = 0; i < source; i++)
-		if(stat(source_path[i], &buf) == -1) {
-			fprintf(stderr, "Cannot stat source directory \"%s\" because %s\n", source_path[i], strerror(errno));
-			EXIT_MKSQUASHFS();
-		}
-
-
 	destination_file = argv[source + 1];
 	if(stat(argv[source + 1], &buf) == -1) {
 		if(errno == ENOENT) { /* Does not exist */
