diff -Nur squashfs2.2-r2/squashfs-tools/mksquashfs.c squashfs-tools/mksquashfs.c
--- squashfs2.2-r2/squashfs-tools/mksquashfs.c	2005-09-09 00:34:28.000000000 +0200
+++ squashfs-tools/mksquashfs.c	2005-09-20 21:29:13.000000000 +0200
@@ -34,7 +34,13 @@
 #include <dirent.h>
 #include <string.h>
 #include <zlib.h>
-#include <endian.h>
+#ifdef __APPLE__
+	#include <machine/endian.h>
+	#define __BYTE_ORDER BYTE_ORDER
+	#define __BIG_ENDIAN BIG_ENDIAN
+#else
+	#include <endian.h>
+#endif
 #include <stdlib.h>
 #include <signal.h>
 #include <setjmp.h>
diff -Nur squashfs2.2-r2/squashfs-tools/read_fs.c squashfs-tools/read_fs.c
--- squashfs2.2-r2/squashfs-tools/read_fs.c	2005-09-08 01:00:39.000000000 +0200
+++ squashfs-tools/read_fs.c	2005-09-20 21:31:19.000000000 +0200
@@ -34,8 +34,13 @@
 #include <string.h>
 #include <zlib.h>
 #include <sys/mman.h>
-#include <endian.h>
-
+#ifdef __APPLE__
+	#include <machine/endian.h>
+	#define __BYTE_ORDER BYTE_ORDER
+	#define __BIG_ENDIAN BIG_ENDIAN
+#else
+	#include <endian.h>
+#endif 
 #include <squashfs_fs.h>
 #include "read_fs.h"
 #include "global.h"
