--- scripts/kconfig/mconf.c	2011-06-06 00:57:46.000000000 +0200
+++ scripts/kconfig/mconf.c	2011-10-02 00:31:28.000000000 +0200
@@ -463,7 +463,7 @@
 				if (sym_is_changable(sym))
 					item_make("[%c]", val == no ? ' ' : '*');
 				else
-					item_make("-%c-", val == no ? ' ' : '*');
+					item_make("[%c]", val == no ? '-' : 'X');
 				item_set_tag('t');
 				item_set_data(menu);
 				break;
@@ -479,7 +479,7 @@
 					else
 						item_make("<%c>", ch);
 				} else
-					item_make("-%c-", ch);
+					item_make("[%c]", 'X');
 				item_set_tag('t');
 				item_set_data(menu);
 				break;

