--- scripts/kconfig/symbol.c
+++ scripts/kconfig/symbol.c
@@ -370,12 +370,14 @@
 				    sym->dir_dep.expr);
 				fprintf(stderr, "warning: (");
 				expr_fprint(e, stderr);
-				fprintf(stderr, ") selects %s which has unmet direct dependencies (",
+				fprintf(stderr, ") can NOT select %s because of unmet direct dependencies (",
 					sym->name);
 				expr_fprint(sym->dir_dep.expr, stderr);
 				fprintf(stderr, ")\n");
 				expr_free(e);
+				newval.tri = no;
 			}
+			else
 			newval.tri = EXPR_OR(newval.tri, sym->rev_dep.tri);
 		}
 		if (newval.tri == mod && sym_get_type(sym) == S_BOOLEAN)
