--- libfakeroot.c
+++ libfakeroot.c
@@ -206,6 +206,8 @@ void load_library_symbols(void){
 
 #endif /* LIBFAKEROOT_DEBUGGING */
   for(i=0; next_wrap[i].doit; i++){
+ /* clear dlerror() just in case dlsym() legitimately returns NULL */
+    msg = dlerror();
     *(next_wrap[i].doit)=dlsym(get_libc(), next_wrap[i].name);
     if ( (msg = dlerror()) != NULL){
       fprintf (stderr, "dlsym(%s): %s\n", next_wrap[i].name, msg);
