--- archival/libarchive/data_extract_all.c	2014-01-01 00:45:23.000000000 +0100
+++ archival/libarchive/data_extract_all.c	2014-01-13 21:31:41.454593800 +0100
@@ -128,6 +128,7 @@
 			free(dst_name);
 		}
 #endif
+		res = 0;
 		break;
 	}
 	case S_IFDIR:
@@ -168,7 +169,7 @@
 		bb_error_msg_and_die("unrecognized file type");
 	}
 
-	if (!S_ISLNK(file_header->mode)) {
+	if (res == 0 && !S_ISLNK(file_header->mode)) {
 		if (!(archive_handle->ah_flags & ARCHIVE_DONT_RESTORE_OWNER)) {
 			uid_t uid = file_header->uid;
 			gid_t gid = file_header->gid;
