--- linux-2.6.28/arch/mips/kernel/process.c
+++ linux-2.6.28/arch/mips/kernel/process.c
@@ -100,6 +100,7 @@
             if(cpu_idlecount.fullrun >= 10) {
                 printk("system-load 100 %% curr: %s runnable: %ld\n", current->comm, nr_running());
             } else {
+              if(cpu_idlecount.fullrun > 9)
                 printk("system-load %d curr: %s runnable: %ld\n", cpu_idlecount.fullrun, current->comm, nr_running());
             }
             cpu_idlecount.fullrun          = 0;
