--- configure
+++ configure
@@ -4851,7 +4851,7 @@
 
 
 
-CPPFLAGS="-D__UNIX__"
+CPPFLAGS+=" -D__UNIX__"
 
 
 
@@ -8764,8 +8764,11 @@
 
   { $as_echo "$as_me:$LINENO: checking for Ogg" >&5
 $as_echo_n "checking for Ogg... " >&6; }
-  no_ogg=""
-
+  if test "x$sr_disable_oggvorbis_support" = "xyes" ; then
+    no_ogg="yes"
+  else
+    no_ogg=""
+  fi
 
   if test "x$enable_oggtest" = "xyes" ; then
     ac_save_CFLAGS="$CFLAGS"
@@ -8895,7 +8898,11 @@
 
   { $as_echo "$as_me:$LINENO: checking for Vorbis" >&5
 $as_echo_n "checking for Vorbis... " >&6; }
-  no_vorbis=""
+  if test "x$sr_disable_oggvorbis_support" = "xyes" ; then
+    no_vorbis="yes"
+  else
+    no_vorbis=""
+  fi
 
 
   if test "x$enable_vorbistest" = "xyes" ; then
--- Makefile.in
+++ Makefile.in
@@ -636,7 +636,7 @@
 	esac;
 
 $(top_builddir)/config.status: $(top_srcdir)/configure $(CONFIG_STATUS_DEPENDENCIES)
-	$(SHELL) ./config.status --recheck
+	#$(SHELL) ./config.status --recheck
 
 $(top_srcdir)/configure:  $(am__configure_deps)
 	cd $(srcdir) && $(AUTOCONF)
