--- scripts/gen_build_files.sh
+++ scripts/gen_build_files.sh
@@ -68,6 +68,7 @@
 	fi
 }
 
+if [ "$FREETZ_GENERATE_CONFIG_IN_ONLY" != y ]; then
 # (Re)generate include/applets.h
 sed -n 's@^//applet:@@p' "$srctree"/*/*.c "$srctree"/*/*/*.c \
 | generate \
@@ -91,6 +92,7 @@
 	"$srctree/include/usage.src.h" \
 	"include/usage.h" \
 	"/* DO NOT EDIT. This file is generated from usage.src.h */"
+fi
 
 # (Re)generate */Kbuild and */Config.in
 # We skip .dotdirs - makes git/svn/etc users happier
@@ -98,6 +100,7 @@
 | while read -r d; do
 	d="${d#./}"
 
+if [ "$FREETZ_GENERATE_CONFIG_IN_ONLY" != y ]; then
 	src="$srctree/$d/Kbuild.src"
 	dst="$d/Kbuild"
 	if test -f "$src"; then
@@ -108,6 +111,7 @@
 			"${src}" "${dst}" \
 			"# DO NOT EDIT. This file is generated from Kbuild.src"
 	fi
+fi
 
 	src="$srctree/$d/Config.src"
 	dst="$d/Config.in"
