--- gcc/config/t-linux
+++ gcc/config/t-linux
@@ -21,6 +21,9 @@
 CRTSTUFF_T_CFLAGS_S = $(CRTSTUFF_T_CFLAGS) -fPIC
 # Compile libgcc2.a with pic.
 TARGET_LIBGCC2_CFLAGS = -fPIC
+LIB2FUNCS_EXCLUDE += _clz _clzsi2 __ctzsi2 _ffssi2
+LIB2FUNCS_EXCLUDE += __mulsc3 _muldc3 _mulxc3 _multc3 _divsc3 _divdc3 _divxc3 _divtc3
+LIB2FUNCS_EXCLUDE += _absvsi2 _absvdi2 _addvsi3 _addvdi3 _subvsi3 _subvdi3 _mulvsi3 _mulvdi3 _negvsi2 _negvdi2
 
 # Override t-slibgcc-elf-ver to export some libgcc symbols with
 # the symbol versions that glibc used.
