--- avahi-client.pc.in.orig	2010-06-25 02:54:22.000000000 +0200
+++ avahi-client.pc.in	2011-09-11 13:52:44.712636845 +0200
@@ -1,7 +1,7 @@
 prefix=@prefix@
-exec_prefix=${prefix}
+exec_prefix=@prefix@
 libdir=@libdir@
-includedir=${prefix}/include
+includedir=@prefix@/include
 
 Name: avahi-client
 Description: Avahi Multicast DNS Responder (Client Support)
--- avahi-core.pc.in.orig	2010-06-25 02:54:22.000000000 +0200
+++ avahi-core.pc.in	2011-09-11 13:52:27.342855212 +0200
@@ -1,7 +1,7 @@
 prefix=@prefix@
-exec_prefix=${prefix}
+exec_prefix=@prefix@
 libdir=@libdir@
-includedir=${prefix}/include
+includedir=@prefix@/include
 
 Name: avahi-core
 Description: Avahi Multicast DNS Responder (Embeddable Stack)
