--- configure
+++ configure
@@ -3057,9 +3057,6 @@
 
 
 
-if test -n "$CFLAGS" || test -n "$CC" ; then
-   user_redefine_cc=yes
-fi
 
 # Check for programs
 ac_ext=c
@@ -4069,7 +4066,7 @@
 fi
 rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
 
-if test -z "$user_redefine_cc" ; then
+if true ; then
    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for grep that handles long lines and -e" >&5
 $as_echo_n "checking for grep that handles long lines and -e... " >&6; }
 if test "${ac_cv_path_GREP+set}" = set; then :
